<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header">
			<h1> </h1>
		</div>
	



<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <p>##RSEM</p>
<h3 id="1-Get-matrix-of-gene-expression-FPKM-or-TPM-or-reads-count"><a href="#1-Get-matrix-of-gene-expression-FPKM-or-TPM-or-reads-count" class="headerlink" title="1. Get matrix of gene expression (FPKM or TPM),or reads count"></a>1. Get matrix of gene expression (FPKM or TPM),or reads count</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"> </span><br><span class="line"><span class="meta"> #</span><span class="bash">  rsem-generate-data-matrix is a perl script, be sure of the prameter inside is <span class="built_in">set</span> as getting counts or FPKM or TPM</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> vim pl file to make target col as 6 <span class="keyword">for</span> FPKM</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> vim pl file to make target col as 5 <span class="keyword">for</span> TPM</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> vim pl file to make target col as 4 <span class="keyword">for</span> counts</span></span><br><span class="line"><span class="meta"> #</span><span class="bash"><span class="comment"># rsem-generate-data-matrix PATH</span></span></span><br><span class="line"> rsem-generate-data-matrix results_file/S*gene* &gt; 30.counts.matrix</span><br><span class="line"> </span><br><span class="line"><span class="meta"> #</span><span class="bash"><span class="comment"># remove reduant information</span></span></span><br><span class="line"> sed -i 's/results_file\///g' 30.counts.matrix </span><br><span class="line"> sed -i 's/.genes.results//g' 30.counts.matrix</span><br><span class="line"> sed -i 's/"//g' 30.counts.matrix </span><br><span class="line"> awk '&#123;if( $0 !~"^L")print $0&#125;' 30.counts.matrix &gt; 30.count.matrix </span><br><span class="line"> </span><br><span class="line"></span><br><span class="line"><span class="meta"> #</span><span class="bash"><span class="comment"># vim rsem-generate-data-matrix to get </span></span></span><br><span class="line"> vim ~/software/RSEM-1.3.2/rsem-generate-data-matrix</span><br><span class="line"> rsem-generate-data-matrix S*gene* &gt; 39.FPKM.matrix</span><br><span class="line"> awk '&#123;if( $0 !~"^L")print $0&#125;' 30.FPKM.matrix &gt; 30.count.matrix</span><br></pre></td></tr></table></figure>

<p>###2. Primary filter</p>
<p>#####filter condition: counts &gt; 10 in 90 percent samples </p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># filter by counts &gt;10 in 90 percent samples</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line">ab = &#123;&#125;</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> count_matrix:</span><br><span class="line">    head_include = count_matrix.readline().split(<span class="string">'\t'</span>)</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> count_matrix:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        m = <span class="number">0</span></span><br><span class="line">        <span class="keyword">for</span> i <span class="keyword">in</span> range(len(lst1)<span class="number">-1</span>):</span><br><span class="line">            <span class="keyword">if</span> float(lst1[i+<span class="number">1</span>].strip()) &gt; <span class="number">10.0</span>:</span><br><span class="line">                m += <span class="number">1</span></span><br><span class="line">        <span class="keyword">if</span> m &gt; int(<span class="number">0.9</span>*len(head_include)):</span><br><span class="line">            ab[lst1[<span class="number">0</span>]] = <span class="string">"yes"</span></span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            ab[lst1[<span class="number">0</span>]] = <span class="string">"bad"</span></span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">2</span>]) <span class="keyword">as</span> fpkm_matrix:</span><br><span class="line">    print(fpkm_matrix.readline().rstrip())</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> fpkm_matrix:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        <span class="keyword">if</span> ab[lst1[<span class="number">0</span>]] == <span class="string">"yes"</span>:</span><br><span class="line">            print(row.strip())</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> use reads counts filter FPKM matrix</span></span><br><span class="line">python f.py  30.counts.matrix 30.FPKM.noLG.matrix  &gt; 30.FPKM.noLG.filtered.matrix</span><br></pre></td></tr></table></figure>

<h3 id="3-trinity"><a href="#3-trinity" class="headerlink" title="3.trinity"></a>3.trinity</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/PtR --matrix 32.filter.noLG.FPKM.matrix --samples 32samples.txt --log2 --min_rowSums 10 --compare_replicates</span><br><span class="line">~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/PtR --matrix  32.filter.noLG.FPKM.matrix --min_rowSums 10 -s 32samples.txt --log2 --CPM --sample_cor_matrix</span><br><span class="line">~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/PtR --matrix  32.filter.noLG.FPKM.matrix -s 32samples.txt --min_rowSums 10 --log2 --CPM --center_rows   --prin_comp 8</span><br></pre></td></tr></table></figure>

<p>##WGCNA</p>
<h3 id="How-many-samples-does-WGCNA-need"><a href="#How-many-samples-does-WGCNA-need" class="headerlink" title="How many samples does WGCNA need ?"></a>How many samples does WGCNA need ?</h3><p>20 samples</p>
<p>If at all possible, one should <strong>have</strong> at least 20 <strong>samples</strong>; as with <strong>any</strong> analysis methods, more <strong>samples</strong> usually lead to more robust and refined results</p>
<h3 id="If-replicated-samples-could-use-as-individuals"><a href="#If-replicated-samples-could-use-as-individuals" class="headerlink" title="If replicated samples could use as individuals?"></a>If replicated samples could use as individuals?</h3><p>There is a biostar question about that</p>
<blockquote>
<p>Q: Hi, in WGCNA manual I learn that the minimum sample size for WGCNA is 15. If I have 10 different treatment, each treatment has three replicates, do I have 30 samples or just 10 samples?</p>
<p>A: The value of 15 seems arbitrary to me - I am not sure that it is a hard threshold. If the developer of WGCNA went by the Aztec system of numbers, maybe he would have said that 20 is the minimum (Aztecs had 20-base system of numbers).</p>
</blockquote>
<p>This answer get 3 stars include me :)</p>
<blockquote>
<p><a href="https://www.biostars.org/p/342696/" target="_blank" rel="noopener">https://www.biostars.org/p/342696/</a></p>
</blockquote>
<p>###Does gene set need normalized?</p>
<p>Dec 24, 2017 - These <em>normalization</em> methods make a big difference if one wants to compare expression of gene A to expression of gene B; b                             ut <em>WGCNA</em> calculates correlations for which gene-wise scaling factors make no difference. (Sample-wise scaling factors of course do, so samples do <em>need</em> to be <em>normalized</em>.)</p>
<h3 id="Whether-should-I-filter-gene-sets"><a href="#Whether-should-I-filter-gene-sets" class="headerlink" title="Whether  should I filter gene sets"></a>Whether  should I filter gene sets</h3><blockquote>
<p>offical:</p>
<p>We <strong>do</strong> not recommend <strong>filtering genes</strong> by differential expression. <strong>WGCNA</strong> is designed to be an unsupervised analysis method that clusters <strong>genes</strong> based on their expression profiles.</p>
</blockquote>
<p>But it really need a lot memory, 16G PC is not available for gene sets large than 50k</p>
<h3 id="Answear-two-forementioned-question"><a href="#Answear-two-forementioned-question" class="headerlink" title="Answear two forementioned question :"></a>Answear two forementioned question :</h3><p>Can WGCNA be used to analyze RNA-Seq data?</p>
<p>Yes. As far as WGCNA is concerned, working with (properly normalized) RNA-seq data isn’t really any different from working with (properly normalized) microarray data.</p>
<p><strong>We suggest removing features whose counts are consistently low</strong> (for example, removing all features that have a count of less than say 10 in more than 90% of the samples) because such low-expressed features tend to reflect noise and correlations based on counts that are mostly zero aren’t really meaningful. The actual thresholds should be based on experimental design, sequencing depth and sample counts.</p>
<p><strong>We then recommend a variance-stabilizing transformation. For example,</strong> <strong>package DESeq2 implements the function</strong> <strong>varianceStabilizingTransformation which we have found useful, but one</strong> <strong>could also start with normalized counts (or RPKM/FPKM data) and</strong> <strong>log-transform them using log2(x+1).</strong> For highly expressed features, the differences between full variance stabilization and a simple log transformation are small.</p>
<p><strong>Whether one uses RPKM, FPKM, or simply normalized counts doesn’t make</strong> <strong>a whole lot of difference for WGCNA analysis as long as all samples</strong> <strong>were processed the same way.</strong> These normalization methods make a big difference if one wants to compare expression of gene A to expression of gene B; but WGCNA calculates correlations for which gene-wise scaling factors make no difference. (Sample-wise scaling factors of course do, so samples do need to be normalized.)</p>
<p>If data come from different batches, we recommend to check for batch effects and, if needed, adjust for them. We use ComBat for batch effect removal but other methods should also work.</p>
<p>Finally, we usually check quantile scatterplots to make sure there are no systematic shifts between samples; if sample quantiles show correlations (which they usually do), quantile normalization can be used to remove this effect.</p>
<blockquote>
<p><a href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html" target="_blank" rel="noopener">OFFICIAL FAQ</a></p>
</blockquote>
<h5 id="Final-I-decided"><a href="#Final-I-decided" class="headerlink" title="Final I decided:"></a>Final I decided:</h5><p>The authors do recommend to remove the “noisier” genes, either by mean expression or variance, but they also recommmend to not filter by differential expression.</p>
<ol>
<li>Use reads count to filter: if reads count of one gene in 90% samples is less than <strong><em>10</em></strong> , remove it (i got 20902 genes,15 SPL genes included finally)    </li>
</ol>
<h2 id="WGCNA-Process"><a href="#WGCNA-Process" class="headerlink" title="WGCNA Process"></a>WGCNA Process</h2><h4 id="Samples-30-samples-during-5-period-of-batatas-devolopment"><a href="#Samples-30-samples-during-5-period-of-batatas-devolopment" class="headerlink" title="Samples: 30 samples during 5 period of batatas devolopment"></a>Samples: 30 samples during 5 period of batatas devolopment</h4><p>10TR 20TR 30FR 30SR 40FR 40SR 50FR 50SR</p>
<p>Only after 30 days, we could recognize SR based on the diameter &gt; 2.5mm (and DAT10TR DAT20TR is FR).</p>
<p>each period has 4 rep</p>
<p>go</p>
<h4 id="Target-Use-WGCNA-distinguishes-which-genes-have-the-same-express-pattern-and-find-the-gene-most-correlated-to-SPL-genes-in-the-same-module"><a href="#Target-Use-WGCNA-distinguishes-which-genes-have-the-same-express-pattern-and-find-the-gene-most-correlated-to-SPL-genes-in-the-same-module" class="headerlink" title="Target: Use WGCNA distinguishes which genes have the same express pattern and find the gene most correlated to SPL genes in the same module"></a>Target: Use WGCNA distinguishes which genes have the same express pattern and find the gene most correlated to <em>SPL</em> genes in the same module</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(WGCNA)</span><br><span class="line">allowWGCNAThreads(<span class="number">12</span>)</span><br><span class="line">options(stringsAsFactors = <span class="literal">FALSE</span>)</span><br><span class="line">setwd(<span class="string">"~/DATA/WGCNA/"</span>)</span><br><span class="line">expro=read.table(<span class="string">'rename.30.FPKM.noLG.filtered.matrix'</span>,row.names = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># filter most 25% variant genes</span></span><br><span class="line"><span class="comment">#m.vars=apply(expro,1,var)</span></span><br><span class="line"><span class="comment">#expro.upper=expro[which(m.vars&gt;quantile(m.vars, probs = seq(0, 1, 0.25))[4]),]</span></span><br><span class="line"><span class="comment">#dim(expro.upper)</span></span><br><span class="line"><span class="comment">#datExpr=as.data.frame(t(expro.upper));</span></span><br><span class="line"><span class="comment">#nGenes = ncol(datExpr)</span></span><br><span class="line"><span class="comment">#nSamples = nrow(datExpr)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># do not perform filter</span></span><br><span class="line">datExpr=as.data.frame(t(expro));</span><br><span class="line">nGenes = ncol(datExpr)</span><br><span class="line">nSamples = nrow(datExpr)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## check outlier</span></span><br><span class="line">gsg = goodSamplesGenes(datExpr, verbose = <span class="number">3</span>);</span><br><span class="line">gsg$allOK</span><br><span class="line">sampleTree = hclust(dist(datExpr), method = <span class="string">"average"</span>)</span><br><span class="line">plot(sampleTree, main = <span class="string">"Sample clustering to detect outliers"</span></span><br><span class="line">     , sub=<span class="string">""</span>, xlab=<span class="string">""</span>)</span><br><span class="line"><span class="comment"># save(datExpr, file = "FPKM-01-dataInput.RData")</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## if outlier exsit, then remove gsg False genes using this function </span></span><br><span class="line"><span class="comment">#if (!gsg$allOK)&#123;</span></span><br><span class="line"><span class="comment">## Optionally, print the gene and sample names that were removed:</span></span><br><span class="line"><span class="comment">#if (sum(!gsg$goodGenes)&gt;0) </span></span><br><span class="line"><span class="comment"># printFlush(paste("Removing genes:", </span></span><br><span class="line"><span class="comment"># paste(names(datExpr)[!gsg$goodGenes], collapse = ",")));</span></span><br><span class="line"><span class="comment">#if (sum(!gsg$goodSamples)&gt;0) </span></span><br><span class="line"><span class="comment"># printFlush(paste("Removing samples:", </span></span><br><span class="line"><span class="comment"># paste(rownames(datExpr)[!gsg$goodSamples], collapse = ",")));</span></span><br><span class="line"><span class="comment"># Remove the offending genes and samples from the data:</span></span><br><span class="line"><span class="comment">#datExpr = datExpr[gsg$goodSamples, gsg$goodGenes]</span></span><br><span class="line"><span class="comment">#&#125;</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## Log2 fold change</span></span><br><span class="line"><span class="comment">#datExpr = log2(datExpr+1)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## try different soft threshold,from 1 to 25</span></span><br><span class="line">powers = c(seq(<span class="number">1</span>,<span class="number">25</span>,by = <span class="number">1</span>))</span><br><span class="line">sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = <span class="number">5</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">## plot the 25 value</span></span><br><span class="line">par(mfrow = c(<span class="number">1</span>,<span class="number">2</span>));</span><br><span class="line">cex1 = <span class="number">0.8</span>;</span><br><span class="line">plot(sft$fitIndices[,<span class="number">1</span>], -sign(sft$fitIndices[,<span class="number">3</span>])*sft$fitIndices[,<span class="number">2</span>],</span><br><span class="line">     xlab=<span class="string">"Soft Threshold (power)"</span>,ylab=<span class="string">"Scale Free Topology Model Fit,signed R^2"</span>,type=<span class="string">"n"</span>,</span><br><span class="line">     main = paste(<span class="string">"Scale independence"</span>));</span><br><span class="line">text(sft$fitIndices[,<span class="number">1</span>], -sign(sft$fitIndices[,<span class="number">3</span>])*sft$fitIndices[,<span class="number">2</span>],</span><br><span class="line">     labels=powers,cex=cex1,col=<span class="string">"red"</span>);</span><br><span class="line">abline(h=<span class="number">0.80</span>,col=<span class="string">"red"</span>)</span><br><span class="line">plot(sft$fitIndices[,<span class="number">1</span>], sft$fitIndices[,<span class="number">5</span>],</span><br><span class="line">     xlab=<span class="string">"Soft Threshold (power)"</span>,ylab=<span class="string">"Mean Connectivity"</span>, type=<span class="string">"n"</span>,</span><br><span class="line">     main = paste(<span class="string">"Mean connectivity"</span>))</span><br><span class="line">text(sft$fitIndices[,<span class="number">1</span>], sft$fitIndices[,<span class="number">5</span>], labels=powers, cex=cex1,col=<span class="string">"red"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">#### get an estimate power value(Closest to 0.9)</span></span><br><span class="line">sft$powerEstimate</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## One-step network construction and module detection##</span></span><br><span class="line">cor = WGCNA::cor</span><br><span class="line">net = blockwiseModules(</span><br><span class="line">  <span class="comment"># input data</span></span><br><span class="line">  datExpr, </span><br><span class="line">  <span class="comment"># calculate correlation coefficient</span></span><br><span class="line">  power = sft$powerEstimate, </span><br><span class="line">  corType = <span class="string">"pearson"</span>,</span><br><span class="line">  networkType = <span class="string">"unsigned"</span>,</span><br><span class="line">  maxBlockSize = <span class="number">6000</span>,</span><br><span class="line">  TOMType = <span class="string">"unsigned"</span>,</span><br><span class="line">  deepSplit = <span class="number">2</span>,</span><br><span class="line">  minModuleSize = <span class="number">30</span>,</span><br><span class="line">  reassignThreshold = <span class="number">0</span>,</span><br><span class="line">  <span class="comment"># distance among modules less than 0.15, otherwise correlation among module more than 0.85 will be merged</span></span><br><span class="line">  mergeCutHeight = <span class="number">0.15</span>,</span><br><span class="line">  numericLabels = <span class="literal">TRUE</span>, </span><br><span class="line">  pamRespectsDendro = <span class="literal">FALSE</span>,)</span><br><span class="line"><span class="comment">## show how many modules identified</span></span><br><span class="line">table(net$colors)</span><br><span class="line"><span class="comment">## plot results</span></span><br><span class="line"><span class="comment">## Convert labels to colors for plotting</span></span><br><span class="line">mergedColors = labels2colors(net$colors)</span><br><span class="line"><span class="comment">## Plot the dendrogram and the module colors underneath</span></span><br><span class="line">plotDendroAndColors(net$dendrograms[[<span class="number">1</span>]], mergedColors[net$blockGenes[[<span class="number">1</span>]]],<span class="string">"Module colors"</span>,</span><br><span class="line">                    dendroLabels = <span class="literal">FALSE</span>, hang = <span class="number">0.03</span>,</span><br><span class="line">                    addGuide = <span class="literal">TRUE</span>, guideHang = <span class="number">0.05</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## get all edge file and node file</span></span><br><span class="line"></span><br><span class="line">moduleLabels = net$colors</span><br><span class="line">moduleColors = labels2colors(net$colors)</span><br><span class="line">modProbes = names(datExpr)</span><br><span class="line">modules = <span class="string">"all_without_normalize"</span></span><br><span class="line">TOM = TOMsimilarityFromExpr(datExpr, power = sft$powerEstimate);</span><br><span class="line">cyt = exportNetworkToCytoscape(TOM,</span><br><span class="line">                               edgeFile = paste(<span class="string">"CytoscapeInput-edges-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               nodeFile = paste(<span class="string">"CytoscapeInput-nodes-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               weighted = <span class="literal">TRUE</span>,</span><br><span class="line">                               threshold = <span class="number">0.02</span>,</span><br><span class="line">                               nodeNames = modProbes,</span><br><span class="line">                               nodeAttr = moduleColors)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Or retrieve specific module edges</span></span><br><span class="line">modules = c(<span class="string">"turquoise"</span>)</span><br><span class="line"><span class="comment"># 要可视化的基因</span></span><br><span class="line">probes = colnames(datExpr)</span><br><span class="line">inModule = is.finite(match(moduleColors, modules));</span><br><span class="line">modProbes = probes[inModule]</span><br><span class="line"><span class="comment"># 候选基因的 TOM</span></span><br><span class="line">modTOM = TOM[inModule, inModule]</span><br><span class="line"></span><br><span class="line">dimnames(modTOM) = list(modProbes, modProbes)</span><br><span class="line"><span class="comment"># Export the network into edge and node list files Cytoscape can read</span></span><br><span class="line">cyt = exportNetworkToCytoscape(modTOM,</span><br><span class="line">                               edgeFile = paste(<span class="string">"CytoscapeInput-edges-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               nodeFile = paste(<span class="string">"CytoscapeInput-nodes-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               weighted = <span class="literal">TRUE</span>,</span><br><span class="line">                               threshold = <span class="number">0.02</span>,</span><br><span class="line">                               nodeNames = modProbes,</span><br><span class="line">                               nodeAttr = moduleColors[inModule])</span><br><span class="line"><span class="comment">########</span></span><br><span class="line"><span class="comment"># If our target genes are not clustered in one module as expected</span></span><br><span class="line"><span class="comment"># may be reconstruct net in more sensitive parameter</span></span><br><span class="line"><span class="comment">#######</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># get module</span></span><br><span class="line">MEs0 = moduleEigengenes(datExpr, moduleColors)$eigengenes</span><br><span class="line">MEs = orderMEs(MEs0)</span><br><span class="line">modNames = substring(names(MEs), <span class="number">3</span>)</span><br><span class="line">geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = <span class="string">"p"</span>))</span><br><span class="line">MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), </span><br><span class="line">                                          nSamples))</span><br><span class="line">names(geneModuleMembership) = paste(<span class="string">"MM"</span>, modNames, sep=<span class="string">""</span>)</span><br><span class="line">names(MMPvalue) = paste(<span class="string">"p.MM"</span>, modNames, sep=<span class="string">""</span>)</span><br><span class="line"></span><br><span class="line">write.csv(MMPvalue,file = <span class="string">"MMPvalue.csv"</span>,quote=<span class="literal">F</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">HubGenes &lt;- chooseTopHubInEachModule(datExpr,modules)</span><br><span class="line">write.table (HubGenes,file = <span class="string">"HubGenes_of_turquoise_module.txt"</span>,quote=<span class="literal">F</span>,sep=<span class="string">'\t'</span>)</span><br></pre></td></tr></table></figure>

<h3 id="SPL-gene-analysis"><a href="#SPL-gene-analysis" class="headerlink" title="SPL gene analysis"></a>SPL gene analysis</h3><h4 id="get-target-gene-color-and-id"><a href="#get-target-gene-color-and-id" class="headerlink" title="get target gene color and id"></a>get target gene color and id</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">get top gene based on weight</span></span><br><span class="line">less  spl.ori.id | while read line; do grep -w $&#123;line&#125;  CytoscapeInput-nodes-all_without_normalize.txt; done &gt; spl.color</span><br><span class="line">mkdir result</span><br><span class="line">awk '&#123;print "grep -w \""$1"\" 0.05.CytoscapeInput-edges-all.txt &gt; result/"$1"."$3".result"&#125;' spl.color &gt; go</span><br><span class="line">sh go</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> direction among genes <span class="keyword">in</span> sample color are lacking of direction, cuz we constrcut net use <span class="string">"unsigned"</span> method</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> therefore we could sort the edge file by 3rd column</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> either we should filter those replicated genes <span class="keyword">in</span> 1st 2nd column </span></span><br><span class="line">ls *result.2 | sed 's/.result.2//g' | while read line;do python sort.py $&#123;line&#125;.result &gt; $&#123;line&#125;.id.2;done</span><br></pre></td></tr></table></figure>

<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## sort.py </span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line">i = <span class="number">0</span></span><br><span class="line">lst3 = []</span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">sort_list</span><span class="params">(one_list)</span>:</span></span><br><span class="line">  <span class="keyword">return</span> list(set(one_list))</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_data:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_data:</span><br><span class="line">        i += <span class="number">1</span></span><br><span class="line">file_len = i</span><br><span class="line">sort_file = <span class="string">''</span>.join(sorted(open(sys.argv[<span class="number">1</span>]), key=<span class="keyword">lambda</span> s: s.split()[<span class="number">2</span>],reverse=<span class="number">1</span>))</span><br><span class="line">lst1 = sort_file.strip().split(<span class="string">'\n'</span>)</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> range(int(file_len*<span class="number">0.1</span>)):</span><br><span class="line">    lst2 = lst1[i].split(<span class="string">'\t'</span>)</span><br><span class="line">    lst3.append(lst2[<span class="number">0</span>])</span><br><span class="line">    lst3.append(lst2[<span class="number">1</span>])</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> range(len(sort_list(lst3))):</span><br><span class="line">    print(sort_list(lst3)[i<span class="number">-1</span>])</span><br></pre></td></tr></table></figure>

<h4 id="get-gene-based-on-FPKM-cor"><a href="#get-gene-based-on-FPKM-cor" class="headerlink" title="get gene based on FPKM cor"></a>get gene based on FPKM cor</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> only analyze gene <span class="keyword">in</span> turquoise</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> get gene based on FPKM cor</span></span><br><span class="line">less  turquoise.gene.id | while read line  </span><br><span class="line">do</span><br><span class="line">grep -w $&#123;line&#125; ../../9_RSEM/88/88.FPKM.noLGfiltered.matrix</span><br><span class="line">done &gt; turquoise.gene.FPKM.matrix</span><br></pre></td></tr></table></figure>

<h4 id="get-correlation-ship-between-spl-gene-and-turquoise-gene"><a href="#get-correlation-ship-between-spl-gene-and-turquoise-gene" class="headerlink" title="get correlation ship between spl gene and turquoise gene"></a>get correlation ship between spl gene and turquoise gene</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(circlize)</span><br><span class="line"><span class="keyword">library</span>(corrplot)</span><br><span class="line"><span class="keyword">library</span>(pheatmap)</span><br><span class="line">setwd(<span class="string">"~/data/gene_family/downstream/10_GO_KEGG"</span>)</span><br><span class="line"></span><br><span class="line">gene &lt;- read.delim(<span class="string">'hubgene/turquoise.gene.FPKM.matrix'</span>, row.names = <span class="number">1</span>,header = <span class="literal">T</span>, sep = <span class="string">'\t'</span>)</span><br><span class="line">gene &lt;- t(gene) </span><br><span class="line">gene &lt;- log2(gene+<span class="number">1</span>) </span><br><span class="line">gene_cor &lt;- cor(gene, method = <span class="string">'pearson'</span>)</span><br><span class="line">df &lt;- as.data.frame(gene_cor)</span><br><span class="line"></span><br><span class="line">spl_turquoise &lt;- df[,which(names(df)%<span class="keyword">in</span>%c(<span class="string">"g34536"</span>,<span class="string">"g46868"</span>,<span class="string">"g54228"</span>,<span class="string">"g31999"</span>,<span class="string">"g44906"</span>))]</span><br><span class="line">gene_cor &lt;- as.matrix(spl_turquoise)</span><br><span class="line"></span><br><span class="line">col &lt;- colorRampPalette(c(<span class="string">"#BB4444"</span>, <span class="string">"#EE9988"</span>, <span class="string">"#FFFFFF"</span>, <span class="string">"#77AADD"</span>, <span class="string">"#4477AA"</span>))</span><br><span class="line">pheatmap(spl_turquoise,</span><br><span class="line">         labels_row = <span class="literal">NULL</span>,</span><br><span class="line">         angle_col = <span class="string">"45"</span>,</span><br><span class="line">         cluster_row = <span class="literal">T</span>,</span><br><span class="line">         )</span><br><span class="line">  </span><br><span class="line">write.csv(spl_turquoise, file = <span class="string">"5spl_turquoise.gene.cor.matrix"</span>,quote = <span class="literal">F</span>)</span><br></pre></td></tr></table></figure>

<h2 id="eggNOG-mapper-and-ClusterProfile"><a href="#eggNOG-mapper-and-ClusterProfile" class="headerlink" title="eggNOG mapper and ClusterProfile"></a>eggNOG mapper and ClusterProfile</h2><p>we first submit our whole protome to <a href="http://eggnog-mapper.embl.de/" target="_blank" rel="noopener">eggNOG website</a>,  against 33090 database representing plants.</p>
<p>Then we use R script to create a R package named </p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> we should modify some cloumn name and necessary content both <span class="keyword">in</span> these two file</span></span><br><span class="line">query_seqs.fa.emapper.annotations</span><br><span class="line"></span><br><span class="line">nohup Rscript  makeOrgPackageFromEmapper.R 1&gt;make.log 2&gt;err.log</span><br></pre></td></tr></table></figure>

<h4 id="makeOrgPackageFromEmapper-R"><a href="#makeOrgPackageFromEmapper-R" class="headerlink" title="makeOrgPackageFromEmapper.R"></a>makeOrgPackageFromEmapper.R</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(tidyverse)</span><br><span class="line"><span class="keyword">library</span>(stringr)</span><br><span class="line"><span class="keyword">library</span>(KEGGREST)</span><br><span class="line"><span class="keyword">library</span>(AnnotationForge)</span><br><span class="line"></span><br><span class="line"><span class="comment">#' Title</span></span><br><span class="line"><span class="comment">#'</span></span><br><span class="line"><span class="comment">#' @param f_emapper_anno eggnog-mapper annotation result</span></span><br><span class="line"><span class="comment">#' @param author Who is the creator of this package? like "xxx &lt;xxx@xxx.xx&gt;"</span></span><br><span class="line"><span class="comment">#' @param tax_id The Taxonomy ID that represents your organism. (NCBI has a nice online browser for finding the one you need)</span></span><br><span class="line"><span class="comment">#' @param genus Single string indicating the genus</span></span><br><span class="line"><span class="comment">#' @param species Single string indicating the species</span></span><br><span class="line"><span class="comment">#'</span></span><br><span class="line"><span class="comment">#' @return OrgDb name</span></span><br><span class="line"><span class="comment">#' @export</span></span><br><span class="line"><span class="comment">#'</span></span><br><span class="line"><span class="comment">#' @examples</span></span><br><span class="line">makeOrgPackageFromEmapper &lt;- <span class="keyword">function</span>(f_emapper_anno, </span><br><span class="line">                                      author, </span><br><span class="line">                                      tax_id = <span class="string">"0"</span>, </span><br><span class="line">                                      genus = <span class="string">"default"</span>, </span><br><span class="line">                                      species = <span class="string">"default"</span>) &#123;</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># f_emapper_anno = "input/sesame.emapper.annotations"</span></span><br><span class="line">  <span class="comment"># author = "zhangxudong &lt;zhangxudong@genek.tv&gt;"</span></span><br><span class="line">  <span class="comment"># tax_id = "4182"</span></span><br><span class="line">  <span class="comment"># genus = "Sesamum"</span></span><br><span class="line">  <span class="comment"># species = "indicum"</span></span><br><span class="line">  </span><br><span class="line">  <span class="comment"># read emapper result</span></span><br><span class="line">  emapper &lt;- read_delim(f_emapper_anno,</span><br><span class="line">                        <span class="string">"\t"</span>, escape_double = <span class="literal">FALSE</span>, trim_ws = <span class="literal">TRUE</span>)</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract gene name from emapper</span></span><br><span class="line">  gene_info &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(GID = query_name, GENENAME = `eggNOG annot`) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract go annotation from emapper</span></span><br><span class="line">  gos &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(query_name, GO_terms) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  gene2go = data.frame(GID = character(),</span><br><span class="line">                       GO = character(),</span><br><span class="line">                       EVIDENCE = character())</span><br><span class="line">  </span><br><span class="line">  <span class="keyword">for</span> (row <span class="keyword">in</span> <span class="number">1</span>:nrow(gos)) &#123;</span><br><span class="line">    the_gid &lt;- gos[row, <span class="string">"query_name"</span>][[<span class="number">1</span>]]</span><br><span class="line">    the_gos &lt;- str_split(gos[row,<span class="string">"GO_terms"</span>], <span class="string">","</span>, simplify = <span class="literal">FALSE</span>)[[<span class="number">1</span>]]</span><br><span class="line">    </span><br><span class="line">    df_temp &lt;- data_frame(GID = rep(the_gid, length(the_gos)),</span><br><span class="line">                          GO = the_gos,</span><br><span class="line">                          EVIDENCE = rep(<span class="string">"IEA"</span>, length(the_gos)))</span><br><span class="line">    gene2go &lt;- rbind(gene2go, df_temp)</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract kegg pathway annotation from emapper ----------------------------</span></span><br><span class="line">  kos &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(query_name, KEGG_KOs) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  gene2ko = data.frame(GID = character(),</span><br><span class="line">                       Ko = character())</span><br><span class="line">  </span><br><span class="line">  df_temp &lt;- list()</span><br><span class="line">  <span class="keyword">for</span> (row <span class="keyword">in</span> <span class="number">1</span>:nrow(kos)) &#123;</span><br><span class="line">    the_gid &lt;- kos[row, <span class="string">"query_name"</span>][[<span class="number">1</span>]]</span><br><span class="line">    the_kos &lt;- str_split(kos[row,<span class="string">"KEGG_KOs"</span>], <span class="string">","</span>, simplify = <span class="literal">FALSE</span>)[[<span class="number">1</span>]]</span><br><span class="line">    </span><br><span class="line">    df_temp[[row]] &lt;- data_frame(GID = rep(the_gid, length(the_kos)),</span><br><span class="line">                                 Ko = the_kos)</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  gene2ko &lt;- bind_rows(df_temp)</span><br><span class="line">  </span><br><span class="line">  load(file = <span class="string">"kegg_info.RData"</span>)</span><br><span class="line">  gene2pathway &lt;- gene2ko %&gt;% left_join(ko2pathway, by = <span class="string">"Ko"</span>) %&gt;%</span><br><span class="line">    left_join(pathway2name, by = <span class="string">"Pathway"</span>) %&gt;%</span><br><span class="line">    dplyr::select(GID, Ko, Pathway, Pathway_Name, Pathway_Class, Pathway_Subclass) %&gt;%</span><br><span class="line">    distinct() %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract COG annotation from emapper -------------------------------------</span></span><br><span class="line">  cog_info &lt;- read_delim(<span class="string">"cog_funclass.tab"</span>, </span><br><span class="line">                         <span class="string">"\t"</span>, escape_double = <span class="literal">FALSE</span>, trim_ws = <span class="literal">TRUE</span>)</span><br><span class="line">  </span><br><span class="line">  cogs &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(query_name, COG = `COG cat`) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  gene2cog = data.frame(GID = character(),</span><br><span class="line">                        COG = character())</span><br><span class="line">  </span><br><span class="line">  df_temp &lt;- list()</span><br><span class="line">  <span class="keyword">for</span> (row <span class="keyword">in</span> <span class="number">1</span>:nrow(cogs)) &#123;</span><br><span class="line">    the_gid &lt;- cogs[row, <span class="string">"query_name"</span>][[<span class="number">1</span>]]</span><br><span class="line">    the_cogs &lt;- str_trim(str_split(cogs[row,<span class="string">"COG"</span>], <span class="string">","</span>, simplify = <span class="literal">FALSE</span>)[[<span class="number">1</span>]])</span><br><span class="line">    </span><br><span class="line">    df_temp[[row]] &lt;- data_frame(GID = rep(the_gid, length(the_cogs)),</span><br><span class="line">                                 COG = the_cogs)</span><br><span class="line">  &#125;</span><br><span class="line">  gene2cog &lt;- bind_rows(df_temp)</span><br><span class="line">  </span><br><span class="line">  gene2cog &lt;- gene2cog %&gt;% left_join(cog_info, by = <span class="string">"COG"</span>)</span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">  <span class="comment"># make OrgDb</span></span><br><span class="line">  makeOrgPackage(gene_info=gene_info,</span><br><span class="line">                 go=gene2go,</span><br><span class="line">                 <span class="comment">#ko=gene2ko,</span></span><br><span class="line">                 pathway=gene2pathway,</span><br><span class="line">                 cog=gene2cog,</span><br><span class="line">                 version=<span class="string">"0.0.2"</span>,</span><br><span class="line">                 maintainer=author,</span><br><span class="line">                 author=author,</span><br><span class="line">                 outputDir = <span class="string">"."</span>,</span><br><span class="line">                 tax_id=tax_id,</span><br><span class="line">                 genus=genus,</span><br><span class="line">                 species=species,</span><br><span class="line">                 goTable=<span class="string">"go"</span>)</span><br><span class="line">  </span><br><span class="line">  my_orgdb &lt;- str_c(<span class="string">"org."</span>, str_to_upper(str_sub(genus, <span class="number">1</span>, <span class="number">1</span>)) , species, <span class="string">".eg.db"</span>, sep = <span class="string">""</span>)</span><br><span class="line">  <span class="keyword">return</span>(my_orgdb)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">my_orgdb &lt;- makeOrgPackageFromEmapper(<span class="string">"query_seqs.fa.emapper.annotations"</span>, </span><br><span class="line">                                      <span class="string">"shy &lt;2020180495@jsnu.edu.cn&gt;"</span>, </span><br><span class="line">                                      tax_id = <span class="string">"4120"</span>, </span><br><span class="line">                                      genus = <span class="string">"Ipomoea"</span>, </span><br><span class="line">                                      species = <span class="string">"batatas"</span>)</span><br></pre></td></tr></table></figure>

<h3 id="And-i-also-use-kobas-enrichment-methods-but-it-still-queued"><a href="#And-i-also-use-kobas-enrichment-methods-but-it-still-queued" class="headerlink" title="And i also use kobas enrichment methods, but it still queued"></a>And i also use kobas enrichment methods, but it still queued</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"># ipomoea nil</span><br><span class="line">http://kobas.cbi.pku.edu.cn/kobas3/retrieve/?taskid=8ee9e38a368d4ce98ea359f629b0ca38</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># sl</span><br><span class="line"></span><br><span class="line">page:http://kobas.cbi.pku.edu.cn/kobas3/retrieve/?taskid=48bed844140b48a7a4e76e8aca4fa16a</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># enrichment nil </span><br><span class="line"></span><br><span class="line">http://kobas.cbi.pku.edu.cn/kobas3/retrieve/?taskid=bd863ad74ea74fa9adc91b4bfe080520</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">#enrichment sl</span><br></pre></td></tr></table></figure>

<h4 id="There-is-a-very-funny-bug"><a href="#There-is-a-very-funny-bug" class="headerlink" title="There is a very funny bug"></a>There is a very funny bug</h4><p>when i first visit kobas webwites, I find they only have two species included in protein blast tags (human and mouse), after I pee back I found many items loaded…. So I could use Ipomoea nil as reference species…</p>
<h4 id="GO-and-KEGG-annotate-by-ClusterProfile"><a href="#GO-and-KEGG-annotate-by-ClusterProfile" class="headerlink" title="GO and KEGG annotate by ClusterProfile"></a>GO and KEGG annotate by ClusterProfile</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(readr)</span><br><span class="line"><span class="keyword">library</span>(clusterProfiler)</span><br><span class="line"><span class="keyword">library</span>(DOSE)</span><br><span class="line"><span class="keyword">library</span>(purrr)</span><br><span class="line"><span class="keyword">library</span>(tidyverse)</span><br><span class="line"><span class="keyword">library</span>(org.Ibatatas.eg.db)</span><br><span class="line">setwd(<span class="string">"~/data/gene_family/downstream/10_GO_KEGG/"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 安装构建的R包</span></span><br><span class="line">my_orgdb &lt;- <span class="string">"org.Ibatatas.eg.db/"</span></span><br><span class="line"><span class="keyword">if</span> (requireNamespace(my_orgdb, quietly = <span class="literal">TRUE</span>))</span><br><span class="line">  remove.packages(my_orgdb)</span><br><span class="line">install.packages(my_orgdb, repos = <span class="literal">NULL</span>, type = <span class="string">"source"</span>)</span><br><span class="line"><span class="keyword">library</span>(org.Ibatatas.eg.db)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">load(<span class="string">"kegg_info.RData"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">gene_list &lt;- read.table(<span class="string">"hubgene/CytoscapeInput-edges-turquoise.top10.id"</span>, header = <span class="literal">FALSE</span>)</span><br><span class="line">gene_list &lt;- as.character(gene_list$V1)</span><br><span class="line"><span class="comment">#gene_list2 &lt;- read.table("../39/30/gene_filter/final/top10_percent.g34536.brown.f", header = FALSE)</span></span><br><span class="line"><span class="comment">#gene_list2 &lt;- as.character(gene_list2$V1)</span></span><br><span class="line"></span><br><span class="line">ego &lt;- enrichGO(gene          = gene_list, <span class="comment">#差异基因 vector</span></span><br><span class="line">                keyType       = <span class="string">"GID"</span>,  <span class="comment">#差异基因的 ID 类型，需要是 OrgDb 支持的</span></span><br><span class="line">                OrgDb         = org.Ibatatas.eg.db, <span class="comment">#对应的OrgDb</span></span><br><span class="line">                ont           = <span class="string">"BP"</span>, <span class="comment">#GO 分类名称，CC BP MF </span></span><br><span class="line">                pvalueCutoff  = <span class="number">0.05</span>, <span class="comment">#Pvalue 阈值</span></span><br><span class="line">                qvalueCutoff  = <span class="number">0.05</span>, <span class="comment">#qvalue 阈值</span></span><br><span class="line">                pAdjustMethod = <span class="string">"BH"</span>, <span class="comment">#Pvalue 矫正方法 </span></span><br><span class="line">                readable      = <span class="literal">FALSE</span>) <span class="comment">#TRUE 则展示SYMBOL，FALSE 则展示原来的ID</span></span><br><span class="line"><span class="comment">#将 ego 对象转换为dataframe，新版本可以用as.data.frame(ego)</span></span><br><span class="line">ego_results&lt;-as.data.frame(ego)</span><br><span class="line">barplot(ego, showCategory=<span class="number">20</span>, x = <span class="string">"GeneRatio"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">pathway2gene &lt;- AnnotationDbi::select(org.Ibatatas.eg.db, </span><br><span class="line">                                      keys = keys(org.Ibatatas.eg.db), </span><br><span class="line">                                      columns = c(<span class="string">"Pathway"</span>)) %&gt;%</span><br><span class="line">  na.omit() %&gt;%</span><br><span class="line">  dplyr::select(Pathway, GID)</span><br><span class="line"></span><br><span class="line">ekp &lt;- enricher(gene_list, </span><br><span class="line">                TERM2GENE = pathway2gene, </span><br><span class="line">                TERM2NAME = pathway2name, </span><br><span class="line">                pvalueCutoff = <span class="number">0.05</span>, </span><br><span class="line">                qvalueCutoff = <span class="number">0.2</span>,</span><br><span class="line">                pAdjustMethod = <span class="string">"BH"</span>,</span><br><span class="line">                minGSSize = <span class="number">1</span>)</span><br><span class="line">ekp_results &lt;- as.data.frame(ekp)</span><br><span class="line"></span><br><span class="line">barplot(ekp, showCategory=<span class="number">20</span>, x = <span class="string">"GeneRatio"</span>)</span><br><span class="line"></span><br><span class="line">dotplot(ekp)</span><br><span class="line"></span><br><span class="line">emapplot(ekp)</span><br></pre></td></tr></table></figure>

<h3 id="use-CytoScape-construct-network-plot"><a href="#use-CytoScape-construct-network-plot" class="headerlink" title="use CytoScape construct network plot"></a>use CytoScape construct network plot</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p><img src="file:///Users/sunhaoyun/Library/Application%20Support/typora-user-images/image-20200620111509583.png?lastModify=1592810480" alt="image-20200620111509583"></p>
<h3 id="Hub-based-analysis"><a href="#Hub-based-analysis" class="headerlink" title="Hub-based analysis"></a>Hub-based analysis</h3><blockquote>
<p><a href="https://www.karger.com/Article/Fulltext/495205" target="_blank" rel="noopener">Weighted Gene Correlation Network Analysis (WGCNA) Detected Loss of MAGI2 Promotes Chronic Kidney Disease (CKD) by Podocyte Damage</a></p>
</blockquote>
<p>Highly connected hub nodes are central to the network’s architecture [<a href="https://www.karger.com/Article/Fulltext/495205#ref6" target="_blank" rel="noopener">6</a>] and studies suggested that genes more centralized in the network are more likely to be key drivers to respective cellular functions compared with peripheral genes [<a href="https://www.karger.com/Article/Fulltext/495205#ref26" target="_blank" rel="noopener">26</a>]. Therefore, more important nodes can be obtained by identifying hub nodes. In this study, the top 5% of nodes with highest connectivity were defined as hub genes in each module [<a href="https://www.karger.com/Article/Fulltext/495205#ref27" target="_blank" rel="noopener">27</a>]. Fig. <a href="https://www.karger.com/Article/Fulltext/495205#f03" target="_blank" rel="noopener">3</a>A-D are the networks of hub genes in the brown, yellow, green and red modules, only displaying connections with weight (w) above threshold values of 0.3, 0.15, 0.05, and 0.05, respectively. Clearly, hub genes (red nodes) exhibited high connectivity with neighboring genes whose functions are consistent with GO and KEGG findings. Taking the brown module as an example, which showed functional enrichment in metabolic and oxidation-reduction processes, some hub genes have been also reported to participate in similar processes, e.g. the SLC13 Family (SLC13A3) [<a href="https://www.karger.com/Article/Fulltext/495205#ref28" target="_blank" rel="noopener">28</a>, <a href="https://www.karger.com/Article/Fulltext/495205#ref29" target="_blank" rel="noopener">29</a>], SLC22 Family (SLC22A6) [<a href="https://www.karger.com/Article/Fulltext/495205#ref30" target="_blank" rel="noopener">30</a>] or MSRA that plays a protective role in the progression of UUO-induced kidney fibrosis via suppression of fibrotic responses caused by oxidative stress [<a href="https://www.karger.com/Article/Fulltext/495205#ref31" target="_blank" rel="noopener">31</a>]. Similarly, some hub genes in the yellow module have been identified as pro-inflammatory factors in a variety of human diseases, including CTSS and ITGB2 [<a href="https://www.karger.com/Article/Fulltext/495205#ref32" target="_blank" rel="noopener">32</a>], consistent with the BP term or KEGG pathway enriched in the yellow module. Meanwhile, some well-known inflammatory factors also had high connectivity to these hub genes, e.g. IL1B, CXCR4 and CASP1. These observations further supported the notion that proposed roles for hub genes of unknown functions may be inferred from clusters of genes similarly expressed across many biological conditions.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">awk &apos;&#123;if ($1 ~ &quot;^g&quot;)print $1&quot;.t1&quot;&#125;&apos; 0.1.CytoscapeInput-nodes-turquoise.txt  &gt; 0.1.gene.id</span><br><span class="line">awk &apos;&#123;if ($1 ~ &quot;^g&quot;)print $1&quot;.t1&quot;&#125;&apos; 0.05.CytoscapeInput-nodes-turquoise.txt  &gt; 0.05.gene.id</span><br><span class="line">awk &apos;&#123;if ($1 ~ &quot;^g&quot;)print $1&quot;.t1&quot;&#125;&apos; 0.02.CytoscapeInput-nodes-turquoise.txt  &gt; 0.02.gene.id</span><br></pre></td></tr></table></figure>

<p>![image-20200701084031314](/Users/sunhaoyun/Library/Application Support/typora-user-images/image-20200701084031314.png)</p>
<p>![image-20200701085601228](/Users/sunhaoyun/Library/Application Support/typora-user-images/image-20200701085601228.png)</p>
<p>To identify the key modules co-expressed with FT, we calculated themodule-trait association and ranked genes by their correlation with the FPKM value of PmFT.Finally,top 50 candidate genes (R2 &gt; 0.6) co-expressed with PmFTand 11 putative FTinteracting factors inArabidopsisflowering pathway </p>
<p>by “WGCNA” R package (version 1.4.9) [<a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-018-1526-8#ref-CR55" target="_blank" rel="noopener">55</a>], the parameters were set as follows: The power was 9; the minModuleSize was 30 and the cutHeight was 0.25. Finally, we visualized the sub-network using Cytoscape (version 3.4.0) program [<a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-018-1526-8#ref-CR56" target="_blank" rel="noopener">56</a>]. An additional investigation of the putative functions of <em>GhSK</em> genes and their co-expression genes was based on Gene Ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) enrichment analyses.</p>
	  
	</div>

	<div>
  	<center>
	<div class="pagination">

    
    
    <a href="/2020/06/22/KaKs_calculator/" type="button" class="btn btn-default"><i
                class="fa fa-arrow-circle-o-left"></i> Prev</a>
    

    <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
    
    <a href="/2020/05/28/wget_ncbi/" type="button" class="btn btn-default ">Next<i
                class="fa fa-arrow-circle-o-right"></i></a>
    

    
</div>

    </center>
	</div>
	
	<!-- comment -->
	
<section id="comment">
    <h2 class="title">Comments</h2>

    
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2020-06-16 
	</div>
	

	<!-- categories -->
    

	<!-- tags -->
		

	<!-- toc -->
	<div class="meta-widget">
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2021 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
