<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>主要记录千人基因组数据分析 | Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="主要记录千人基因组数据分析">
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header">
			<h1> 主要记录千人基因组数据分析</h1>
		</div>
	



<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <h3 id="Part-1"><a href="#Part-1" class="headerlink" title="Part 1."></a>Part 1.</h3><h4 id="数据下载"><a href="#数据下载" class="headerlink" title="数据下载"></a>数据下载</h4><blockquote>
<p>感觉再做一些可以去迅雷入职了</p>
</blockquote>
<p>首先下载alignment的list   </p>
<p>eg. igsr_Northern\ Han\ in\ China\ (HGDP)_undefined.tsv</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> 0.提取第一列保存为1_ftp_sites</span><br><span class="line">awk '&#123;print $1&#125;' igsr_Northern\ Han\ in\ China\ \(HGDP\)_undefined.tsv.tsv</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 1.因为使用ascp下载 所以只需要数据在ebi数据库中的位置就可以，保存为2_temp_id</span><br><span class="line">awk -F 'uk/' '&#123;print $2&#125;' 1_ftp_sites &gt; 2_temp_id</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 2.使用echo生成命令，并记录在3_run，方便以后查看哪里错了</span><br><span class="line">less 2_temp_id | while read line; </span><br><span class="line">do </span><br><span class="line">echo "ascp  -v -k 1 -l 200m -i \</span><br><span class="line">/Users/Yi-Liangwei/Applications/Aspera\ Connect.app/Contents/Resources/asperaweb_id_dsa.openssh \</span><br><span class="line">-Tr -Q -P33001 -L- fasp-g1k@fasp.1000genomes.ebi.ac.uk:"$&#123;line&#125; .; </span><br><span class="line">done &gt; 3_run </span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 3.下 载</span><br><span class="line">nohup sh 3_run</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 4.对于下载完成的cram文件 执行call snp环节</span><br><span class="line"><span class="meta">#</span> 在所有操作之前，如果没有下载基因组的索引文件</span><br><span class="line"><span class="meta">#</span> 可以使用下面两个命令生成</span><br><span class="line">samtools faidx Homo_sapiens_assembly38.fasta </span><br><span class="line"></span><br><span class="line">java -Xmx2g -jar ~/software/picard.jar CreateSequenceDictionary R=Homo_sapiens_assembly38.fasta O=Homo_sapiens_assembly38.dict</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 然后还需要准备三个文件，第一个是bam文件的prefix（CHS.info），第二个是rs位点的染色体和位置，</span><br><span class="line"><span class="meta">#</span> 第三个是根据第二个文件生成的bed文件rs_id.bed</span><br><span class="line"><span class="meta">#</span> 格式如下  CHS.info eg.</span><br><span class="line">HG00597</span><br><span class="line">HG00591</span><br><span class="line">HG00609</span><br><span class="line"><span class="meta">#</span> rs_id eg.</span><br><span class="line">rs12044597      1       1708801</span><br><span class="line">rs11185092      1       107886278</span><br><span class="line">rs10923724      1       119546842</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> rs_id.bed用下面的命令生成</span><br><span class="line">awk -v OFS='\t' '&#123;print "chr"$2,$3-1,$3+1,$1&#125;' rs_id   &gt; rs_id.bed</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 5.cram格式转bam格式去call snp</span><br><span class="line">less CHS.info | while read line; do </span><br><span class="line">samtools view  -T /data/gujiaqi/698_download/Homo_sapiens_assembly38.fasta $&#123;line&#125;.final.cram  -L rs_id.bed -M -b -o $&#123;line&#125;.m.1.bam; done</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 6.call snp and parse vcf file</span><br><span class="line"><span class="meta">#</span> 这时候还需要生成一个只包含rs编号的文件，两次的输入文件不一样的</span><br><span class="line">python bam2vcf.py -p CHS.info -rs rs_id</span><br><span class="line">python parse_vcf.py -p CHS.info -rs rs_number</span><br><span class="line"><span class="meta">#</span> 所有结果会写在rs.genotype 和 rs.frequency中</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 使用的call snp软件为bcftools，如果时间不紧的话可以用GATK，将bcftools那一行改成</span><br><span class="line"><span class="meta">#</span> java -XX:+UseParallelGC -XX:ParallelGCThreads=8 -Xmx10g -jar \</span><br><span class="line">~/software/GenomeAnalysisTK.jar -T HaplotypeCaller --genotyping_mode DISCOVERY -A \</span><br><span class="line">AlleleBalanceBySample -A DepthPerAlleleBySample -A DepthPerSampleHC \</span><br><span class="line">-A InbreedingCoeff -A MappingQualityZeroBySample -A StrandBiasBySample\</span><br><span class="line">-A Coverage -A FisherStrand -A HaplotypeScore -A MappingQualityRankSumTest \</span><br><span class="line">-A MappingQualityZero -A QualByDepth -A RMSMappingQuality -A ReadPosRankSumTest \</span><br><span class="line">-A VariantType -l INFO --emitRefConfidence GVCF \</span><br><span class="line">-rf BadCigar --variant_index_parameter 128000 --variant_index_type LINEAR \</span><br><span class="line">-R /data/gujiaqi/698_download/Homo_sapiens_assembly38.fasta -I HG00597.merged.bam\</span><br><span class="line">-o HG00597.merged.gatk.gvcf</span><br></pre></td></tr></table></figure>

<h3 id="Part-2"><a href="#Part-2" class="headerlink" title="Part 2."></a>Part 2.</h3><h5 id="对vcf进行计算前还需要准备一下"><a href="#对vcf进行计算前还需要准备一下" class="headerlink" title="对vcf进行计算前还需要准备一下"></a>对vcf进行计算前还需要准备一下</h5><figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 对filtered出来的vcf文件压缩并索引</span></span><br><span class="line">ls *vcf | <span class="keyword">while</span> read line; <span class="keyword">do</span> bgzip $&#123;line&#125; -@ <span class="number">8</span>;tabix $&#123;line&#125;.gz; done</span><br><span class="line"></span><br><span class="line"><span class="comment"># 合并为一个文件，这一步如果直接合并所有的话，顺序默认按照文件夹内从小到大的顺序</span></span><br><span class="line"><span class="comment"># 后续会再rehead，当然也可以一个Group一个Group的合并，然后再把合并的合并</span></span><br><span class="line">bcftools merge *gz -O z -o <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz</span><br><span class="line"></span><br><span class="line"><span class="comment"># 查看vcf是否包含1185个位点</span></span><br><span class="line">less <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz | awk <span class="string">'&#123;if ($0 !~ "#")print&#125;'</span> | wc -l</span><br><span class="line"><span class="comment"># 或者可以直接用plink把rs号给提取出来，结果会保存在 -o 指定的 rs_check.snplist中</span></span><br><span class="line"><span class="comment"># 再根据rs_check.snplist 和之前的rs_id看一下就可以了</span></span><br><span class="line">plink --write-snplist -vcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz -out rs_check</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 使用excel做一个all_sample.info 就把之前那些.info文件和在一起</span></span><br><span class="line"><span class="comment"># all_sample.info 内容如下</span></span><br><span class="line"><span class="comment"># HG00709	SouthernHan	698dataset	698dataset_SouthernHan_HG00709</span></span><br><span class="line"><span class="comment"># HGDP01296	NorthernHan	HGDP	HGDP_NorthernHan_HGDP01296</span></span><br><span class="line"><span class="comment"># HGDP00828	Japanese	HGDP	HGDP_Japanese_HGDP00828</span></span><br><span class="line"><span class="comment"># HGDP00774	Han in China	HGDP	HGDP_Han in China_HGDP00774</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 为了下一步 reheader通过bcftools提取所有样本的id</span></span><br><span class="line">bcftools query -l <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz ｜<span class="keyword">while</span> read line; <span class="keyword">do</span> grep $&#123;line&#125; all_sample.info ; done | awk -F <span class="string">'\t'</span> <span class="string">'&#123;print $4&#125;'</span> &gt; vcf.sample.reheader.list</span><br><span class="line">bcftools reheader <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz -s vcf.sample.reheader.list -o <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz</span><br></pre></td></tr></table></figure>

<h5 id="统计vcf文件中的sample和sites"><a href="#统计vcf文件中的sample和sites" class="headerlink" title="统计vcf文件中的sample和sites"></a>统计vcf文件中的sample和sites</h5><figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 统计vcf文件中的基本信息</span></span><br><span class="line"><span class="comment"># individual 的missing信息</span></span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --missing-indv  --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --missing-site --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --site-mean-depth  --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --site-quality --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --het --out vcf_check</span><br><span class="line"><span class="comment"># sites 的missing信息</span></span><br><span class="line"><span class="comment"># vcf_check.imiss vcf_check.lmiss vcf_check.ldepth.mean vcf_check.lqual  vcf_check.het</span></span><br><span class="line"><span class="comment"># 上面就是生成的结果</span></span><br><span class="line"><span class="comment"># awk '&#123;print $3&#125;' vcf_check.ldepth.mean | sort -g -r 可以看平均深度有位点depth的平均深度只有20左右</span></span><br><span class="line"><span class="comment"># awk -F '\t' '&#123;print $4&#125;'  vcf_check.imiss | sort | uniq -c 可以看多少样本有缺失位点，有30个样本都是有缺失的，但是最多也就只缺两个位点</span></span><br><span class="line"><span class="comment"># awk -F '\t' '&#123;print $5&#125;'  vcf_check.lmiss | sort | uniq -c 可以看每个位点有多少个缺失，有两个位点有10个样本以上的缺失</span></span><br><span class="line"><span class="comment"># 需要注意的是 vcf_check.lmiss 中会默认我们的物种是二倍体，所以Number of DATA的数量是sample*2 </span></span><br><span class="line"><span class="comment"># 同理在这个文件中的missing sites都会变为 2 倍</span></span><br></pre></td></tr></table></figure>

<h4 id="使用beagle填充缺失的基因型"><a href="#使用beagle填充缺失的基因型" class="headerlink" title="使用beagle填充缺失的基因型"></a>使用beagle填充缺失的基因型</h4><a id="more"></a> 

<p>##### </p>
<p>具体使用的python脚本如下</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># bam2vcf.py</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"><span class="keyword">import</span> argparse</span><br><span class="line"><span class="keyword">import</span> subprocess</span><br><span class="line"><span class="keyword">import</span> time</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">ap = argparse.ArgumentParser()</span><br><span class="line">ap.add_argument(<span class="string">"-p"</span>, <span class="string">"--prefix"</span>, required=<span class="literal">True</span>,</span><br><span class="line">	help=<span class="string">"the list of prefix of all sample"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-rs"</span>, <span class="string">"--rsid"</span>, required=<span class="literal">True</span>,</span><br><span class="line">	help=<span class="string">"the postion of each rs id, eg. chr1:100-101"</span>)</span><br><span class="line">args = vars(ap.parse_args())</span><br><span class="line"></span><br><span class="line"><span class="keyword">print</span> (<span class="string">"Start : &#123;&#125;"</span> ).format(time.ctime()) </span><br><span class="line">bam_file = []</span><br><span class="line">rs_check = []</span><br><span class="line">rs_id = &#123;&#125;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="keyword">with</span> open(args[<span class="string">'rsid'</span>]) <span class="keyword">as</span> rs:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> rs:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        good = <span class="string">'chr'</span>+str(lst1[<span class="number">1</span>])+<span class="string">'vs'</span>+str(lst1[<span class="number">2</span>].rstrip())</span><br><span class="line">        rs_check.append(good)</span><br><span class="line">        rs_id[good] = lst1[<span class="number">0</span>]</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="keyword">with</span> open(args[<span class="string">"prefix"</span>]) <span class="keyword">as</span> input_bam_file_prefix:</span><br><span class="line">    <span class="keyword">for</span> prefix <span class="keyword">in</span> input_bam_file_prefix:</span><br><span class="line">        prefix =prefix.rstrip()</span><br><span class="line">        new_dir = <span class="string">'mkdir &#123;&#125;'</span>.format(prefix)</span><br><span class="line">        subprocess.call(new_dir, shell = <span class="literal">True</span>)</span><br><span class="line">        merge_bam = <span class="string">'samtools merge -@ 16 &#123;&#125;/&#123;&#125;.merged.bam &#123;&#125;.*.bam'</span>.format(prefix,prefix,prefix)</span><br><span class="line">        subprocess.call(merge_bam, shell = <span class="literal">True</span>)</span><br><span class="line">        bam2vcf =  <span class="string">'bcftools mpileup -a AD,DP,SP &#123;&#125;/&#123;&#125;.merged.bam -f /data/gujiaqi/698_download/Homo_sapiens_assembly38.fasta | bcftools call -f GQ,GP -mO v -o &#123;&#125;/&#123;&#125;.merged.vcf'</span>.format(prefix,prefix,prefix,prefix)</span><br><span class="line">        subprocess.call(bam2vcf, shell = <span class="literal">True</span>)</span><br><span class="line">        f = open(<span class="string">'&#123;&#125;.filtered.vcf'</span>.format(prefix),<span class="string">'w'</span>)</span><br><span class="line">        <span class="keyword">with</span> open(<span class="string">'&#123;&#125;/&#123;&#125;.merged.vcf'</span>.format(prefix,prefix)) <span class="keyword">as</span> current_vcf_file:</span><br><span class="line">            <span class="keyword">for</span> row <span class="keyword">in</span> current_vcf_file:</span><br><span class="line">                <span class="comment"># wrtie filtered vcf</span></span><br><span class="line">                <span class="keyword">if</span> row.startswith(<span class="string">'#'</span>):</span><br><span class="line">                    f.write(row)</span><br><span class="line">                <span class="keyword">else</span>:</span><br><span class="line">                    lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">                    good = lst1[<span class="number">0</span>]+<span class="string">'vs'</span>+lst1[<span class="number">1</span>]</span><br><span class="line">                    <span class="keyword">if</span> good <span class="keyword">in</span> rs_check:</span><br><span class="line">                        lst1[<span class="number">2</span>] = rs_id[good]</span><br><span class="line">                        f.write(<span class="string">"\t"</span>.join(lst1))</span><br><span class="line">f.close()   </span><br><span class="line"><span class="keyword">print</span> (<span class="string">"END : &#123;&#125;"</span> ).format(time.ctime())</span><br></pre></td></tr></table></figure>

<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">### parse_vcf.py</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"><span class="keyword">import</span> argparse</span><br><span class="line"><span class="keyword">import</span> subprocess</span><br><span class="line"><span class="keyword">import</span> time</span><br><span class="line"><span class="keyword">import</span> re</span><br><span class="line"><span class="keyword">import</span> fileinput</span><br><span class="line"></span><br><span class="line">DP4_find = re.compile(<span class="string">r'DP4=(.*?);'</span>)</span><br><span class="line">        </span><br><span class="line"></span><br><span class="line">ap = argparse.ArgumentParser()</span><br><span class="line">ap.add_argument(<span class="string">"-p"</span>, <span class="string">"--prefix"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"the list of prefix of all sample"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-rs"</span>, <span class="string">"--rsid"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"the postion of each rs id, eg. chr1:100-101"</span>)</span><br><span class="line">args = vars(ap.parse_args())</span><br><span class="line"></span><br><span class="line"><span class="keyword">print</span> (<span class="string">"Start : &#123;&#125;"</span> ).format(time.ctime()) </span><br><span class="line"></span><br><span class="line">f1 = open(<span class="string">'rs.frequency'</span>,<span class="string">'w'</span>)</span><br><span class="line">f2 = open(<span class="string">'rs.genotype'</span>,<span class="string">'w'</span>)</span><br><span class="line">sample = []</span><br><span class="line">rs_id = [<span class="string">'Sample'</span>]</span><br><span class="line">c = <span class="number">0</span></span><br><span class="line">d = <span class="number">1</span></span><br><span class="line"><span class="keyword">with</span> open(args[<span class="string">"rsid"</span>]) <span class="keyword">as</span> input_rs:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_rs:</span><br><span class="line">        c += <span class="number">1</span></span><br><span class="line">        row =row.strip()</span><br><span class="line">        rs_id.append(row)</span><br><span class="line">        sample.append(row)</span><br><span class="line">f1.write(<span class="string">'\t'</span>.join(rs_id))</span><br><span class="line">f2.write(<span class="string">'\t'</span>.join(rs_id))</span><br><span class="line"><span class="keyword">with</span> open(args[<span class="string">'prefix'</span>]) <span class="keyword">as</span> input_prefix_file:</span><br><span class="line">    <span class="keyword">for</span> lines <span class="keyword">in</span> input_prefix_file:</span><br><span class="line">        lines = lines.strip()</span><br><span class="line">        freq_out = []</span><br><span class="line">        gt_out = []</span><br><span class="line">        GT_rs = &#123;&#125;</span><br><span class="line">        freq_rs = &#123;&#125;</span><br><span class="line">        <span class="keyword">with</span> open(<span class="string">"&#123;&#125;.filtered.vcf"</span>.format(lines)) <span class="keyword">as</span> input_vcf_file:</span><br><span class="line">            freq_out.append(lines)</span><br><span class="line">            gt_out.append(lines)</span><br><span class="line">            rs_id2 = []</span><br><span class="line">            <span class="keyword">for</span> row <span class="keyword">in</span> input_vcf_file:</span><br><span class="line">                <span class="keyword">if</span> row.startswith(<span class="string">'chr'</span>):</span><br><span class="line">                    lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">                    rs_id2.append(lst1[<span class="number">2</span>])</span><br><span class="line">                    DP4 = re.findall(DP4_find,lst1[<span class="number">7</span>])[<span class="number">0</span>]</span><br><span class="line">                    lst2 = DP4.split(<span class="string">','</span>)</span><br><span class="line">                    ref = int(lst2[<span class="number">0</span>]) + int(lst2[<span class="number">1</span>])</span><br><span class="line">                    alt = int(lst2[<span class="number">2</span>]) + int(lst2[<span class="number">3</span>])</span><br><span class="line">                    <span class="keyword">if</span> lst1[<span class="number">4</span>] == <span class="string">'.'</span>:</span><br><span class="line">                        freq_rs[lst1[<span class="number">2</span>]] = str(<span class="string">"&#123;&#125;:&#123;&#125;/0"</span>.format(lst1[<span class="number">3</span>],str(ref+alt)))</span><br><span class="line">                    <span class="keyword">elif</span> <span class="string">','</span> <span class="keyword">in</span> lst1[<span class="number">4</span>]:</span><br><span class="line">                        freq_rs[lst1[<span class="number">2</span>]] = str(<span class="string">"&#123;&#125;/&#123;&#125;/&#123;&#125;:&#123;&#125;/&#123;&#125;/&#123;&#125;"</span>.format(lst1[<span class="number">3</span>],lst1[<span class="number">4</span>].split(<span class="string">','</span>)[<span class="number">0</span>],lst1[<span class="number">4</span>].split(<span class="string">','</span>)[<span class="number">1</span>],lst1[<span class="number">9</span>].split(<span class="string">':'</span>)[<span class="number">4</span>].split(<span class="string">','</span>)[<span class="number">0</span>],lst1[<span class="number">9</span>].split(<span class="string">':'</span>)[<span class="number">4</span>].split(<span class="string">','</span>)[<span class="number">1</span>],lst1[<span class="number">9</span>].split(<span class="string">':'</span>)[<span class="number">4</span>].split(<span class="string">','</span>)[<span class="number">2</span>]))</span><br><span class="line">                    <span class="keyword">else</span>:</span><br><span class="line">                        freq_rs[lst1[<span class="number">2</span>]] = str(<span class="string">"&#123;&#125;/&#123;&#125;:&#123;&#125;/&#123;&#125;"</span>.format(lst1[<span class="number">3</span>],lst1[<span class="number">4</span>],ref,alt))</span><br><span class="line">                    <span class="comment"># write GT</span></span><br><span class="line">                    GT_type = lst1[<span class="number">9</span>][<span class="number">0</span>:<span class="number">3</span>]</span><br><span class="line">                    <span class="keyword">if</span> GT_type == <span class="string">'0/0'</span>:</span><br><span class="line">                        GT_rs[lst1[<span class="number">2</span>]] = str(lst1[<span class="number">3</span>])+<span class="string">"/"</span>+str(lst1[<span class="number">3</span>])</span><br><span class="line">                    <span class="keyword">elif</span> GT_type == <span class="string">'0/1'</span>:</span><br><span class="line">                        GT_rs[lst1[<span class="number">2</span>]] = str(lst1[<span class="number">3</span>])+<span class="string">"/"</span>+str(lst1[<span class="number">4</span>])</span><br><span class="line">                    <span class="keyword">elif</span> GT_type == <span class="string">'1/1'</span>:</span><br><span class="line">                        GT_rs[lst1[<span class="number">2</span>]] = str(lst1[<span class="number">4</span>])+<span class="string">"/"</span>+str(lst1[<span class="number">4</span>])</span><br><span class="line">                    <span class="keyword">elif</span> GT_type == <span class="string">'1/2'</span>:</span><br><span class="line">                        GT_rs[lst1[<span class="number">2</span>]] = str(lst1[<span class="number">3</span>])+<span class="string">"/"</span>+str(lst1[<span class="number">4</span>].replace(<span class="string">','</span>,<span class="string">'/'</span>))</span><br><span class="line">                    <span class="keyword">elif</span> GT_type == <span class="string">'./.'</span>:</span><br><span class="line">                        GT_rs[lst1[<span class="number">2</span>]] = <span class="string">'N'</span></span><br><span class="line">                    </span><br><span class="line">        <span class="keyword">for</span> i <span class="keyword">in</span> sample:</span><br><span class="line">            <span class="keyword">if</span> i <span class="keyword">in</span> rs_id2:</span><br><span class="line">                freq_out.append(freq_rs[i])</span><br><span class="line">                gt_out.append(GT_rs[i])</span><br><span class="line">            <span class="keyword">else</span>:</span><br><span class="line">                freq_out.append(<span class="string">"0/0"</span>)</span><br><span class="line">                gt_out.append(<span class="string">'N/N'</span>)</span><br><span class="line">        f1.write(<span class="string">'\n'</span>)</span><br><span class="line">        f1.write(<span class="string">'\t'</span>.join(freq_out))</span><br><span class="line">        f2.write(<span class="string">'\n'</span>)</span><br><span class="line">        f2.write(<span class="string">'\t'</span>.join(gt_out))</span><br><span class="line">        print(<span class="string">'&#123;&#125; DONE!'</span>.format(lines))</span><br><span class="line">f2.write(<span class="string">'\n'</span>)</span><br><span class="line">f1.write(<span class="string">'\n'</span>)</span><br><span class="line">f2.close()</span><br><span class="line">f1.close()</span><br><span class="line"></span><br><span class="line"><span class="comment"># creat a head</span></span><br><span class="line">pattern = re.compile(<span class="string">'.&#123;1&#125;'</span>)</span><br><span class="line">head = <span class="string">''</span></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> range(<span class="number">1</span>,c+<span class="number">1</span>):</span><br><span class="line">    stat = <span class="string">''</span></span><br><span class="line">    <span class="keyword">with</span> open(<span class="string">'rs.genotype'</span>) <span class="keyword">as</span> input_gt:</span><br><span class="line">        next(input_gt)</span><br><span class="line">        <span class="keyword">for</span> row <span class="keyword">in</span> input_gt:</span><br><span class="line">            lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">            stat += lst1[i]</span><br><span class="line">    final = <span class="string">''</span></span><br><span class="line">    <span class="keyword">if</span> stat.count(<span class="string">'A'</span>) &gt; <span class="number">0</span>:</span><br><span class="line">        final += <span class="string">'A'</span></span><br><span class="line">    <span class="keyword">if</span> stat.count(<span class="string">'T'</span>) &gt; <span class="number">0</span>:</span><br><span class="line">        final += <span class="string">'T'</span></span><br><span class="line">    <span class="keyword">if</span> stat.count(<span class="string">'C'</span>) &gt; <span class="number">0</span>:</span><br><span class="line">        final += <span class="string">'C'</span></span><br><span class="line">    <span class="keyword">if</span> stat.count(<span class="string">'G'</span>) &gt; <span class="number">0</span>:</span><br><span class="line">        final += <span class="string">'G'</span></span><br><span class="line">    head += <span class="string">'\t'</span> + <span class="string">'/'</span>.join(pattern.findall(final))</span><br><span class="line">head += <span class="string">'\n'</span></span><br><span class="line"><span class="keyword">with</span> open(<span class="string">'rs.genotype'</span>, <span class="string">'r+'</span>) <span class="keyword">as</span> f1, open(<span class="string">'rs.frequency'</span>, <span class="string">'r+'</span>) <span class="keyword">as</span> f2:</span><br><span class="line">    content = f1.read()        </span><br><span class="line">    f1.seek(<span class="number">0</span>, <span class="number">0</span>)</span><br><span class="line">    f1.write(head+content)</span><br><span class="line">    content = f2.read()        </span><br><span class="line">    f2.seek(<span class="number">0</span>, <span class="number">0</span>)</span><br><span class="line">    f2.write(head+content)</span><br><span class="line"><span class="keyword">print</span> (<span class="string">"END : &#123;&#125;"</span> ).format(time.ctime())</span><br></pre></td></tr></table></figure>

	  
	</div>

	<div>
  	<center>
	<div class="pagination">

    
    
    <a href="/2020/12/14/vcf2PCA/" type="button" class="btn btn-default"><i
                class="fa fa-arrow-circle-o-left"></i> Prev</a>
    

    <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
    
    <a href="/2020/12/06/NTFS硬盘挂载/" type="button" class="btn btn-default ">Next<i
                class="fa fa-arrow-circle-o-right"></i></a>
    

    
</div>

    </center>
	</div>
	
	<!-- comment -->
	
<section id="comment">
    <h2 class="title">Comments</h2>

    
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2020-12-08 
	</div>
	

	<!-- categories -->
    

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/1000-Genomes/">1000 Genomes<span>1</span></a></li>
    </ul>
	</div>
		

	<!-- toc -->
	<div class="meta-widget">
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2020 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
