<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 <div class="page-header logo">
  <h1>Keeping your word<span class="blink-fast">∎</span></h1>
</div>

<div class="row page">

	
	<div class="col-md-9">
	

		<div class="slogan">


		<i class="fa fa-heart blink-slow"></i>

		我们进化速度快到让你感动邪门？！

</div>    

		<div id="top_search"></div>
		<div class="mypage">
		
		<!-- title and entry -->
		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2021/04/26/GATK_WDL/" >GATK and WDL流处理</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2021-04-26  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h4 id="基础准备"><a href="#基础准备" class="headerlink" title="基础准备"></a>基础准备</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 下载安装</span></span></span><br><span class="line">wget https://github.com/broadinstitute/cromwell/releases/download/59/cromwell-59.jar</span><br><span class="line">wget https://github.com/broadinstitute/cromwell/releases/download/59/womtool-59.jar</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## cromwell用于run submit写好的wdl程序</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## womtool用于生成wdl的input的config文件</span></span></span><br></pre></td></tr></table></figure>

<p>主要参考以下github以及broad institute脚本</p>
<blockquote>
<p>git clone <a href="https://github.com/gatk-workflows/seq-format-conversion.git" target="_blank" rel="noopener">https://github.com/gatk-workflows/seq-format-conversion.git</a></p>
<p>git clone <a href="https://github.com/gatk-workflows/gatk4-data-processing.git" target="_blank" rel="noopener">https://github.com/gatk-workflows/gatk4-data-processing.git</a></p>
<p>git clone <a href="https://github.com/broadinstitute/wdl-runner.git" target="_blank" rel="noopener">https://github.com/broadinstitute/wdl-runner.git</a></p>
<p>git clone <a href="https://github.com/gatk-workflows/broad-prod-wgs-germline-snps-indels.git" target="_blank" rel="noopener">https://github.com/gatk-workflows/broad-prod-wgs-germline-snps-indels.git</a></p>
<p>git clone <a href="https://github.com/gatk-workflows/gatk4-germline-snps-indels.git" target="_blank" rel="noopener">https://github.com/gatk-workflows/gatk4-germline-snps-indels.git</a></p>
</blockquote>
<p>####代码部分</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br><span class="line">238</span><br><span class="line">239</span><br><span class="line">240</span><br><span class="line">241</span><br><span class="line">242</span><br><span class="line">243</span><br><span class="line">244</span><br><span class="line">245</span><br><span class="line">246</span><br><span class="line">247</span><br><span class="line">248</span><br><span class="line">249</span><br><span class="line">250</span><br><span class="line">251</span><br><span class="line">252</span><br><span class="line">253</span><br><span class="line">254</span><br><span class="line">255</span><br><span class="line">256</span><br><span class="line">257</span><br><span class="line">258</span><br><span class="line">259</span><br><span class="line">260</span><br><span class="line">261</span><br><span class="line">262</span><br><span class="line">263</span><br><span class="line">264</span><br><span class="line">265</span><br><span class="line">266</span><br><span class="line">267</span><br><span class="line">268</span><br><span class="line">269</span><br><span class="line">270</span><br><span class="line">271</span><br><span class="line">272</span><br><span class="line">273</span><br><span class="line">274</span><br><span class="line">275</span><br><span class="line">276</span><br><span class="line">277</span><br><span class="line">278</span><br><span class="line">279</span><br><span class="line">280</span><br><span class="line">281</span><br><span class="line">282</span><br><span class="line">283</span><br><span class="line">284</span><br><span class="line">285</span><br><span class="line">286</span><br><span class="line">287</span><br><span class="line">288</span><br><span class="line">289</span><br><span class="line">290</span><br><span class="line">291</span><br><span class="line">292</span><br><span class="line">293</span><br><span class="line">294</span><br><span class="line">295</span><br><span class="line">296</span><br><span class="line">297</span><br><span class="line">298</span><br><span class="line">299</span><br><span class="line">300</span><br><span class="line">301</span><br><span class="line">302</span><br><span class="line">303</span><br><span class="line">304</span><br><span class="line">305</span><br><span class="line">306</span><br><span class="line">307</span><br><span class="line">308</span><br><span class="line">309</span><br><span class="line">310</span><br><span class="line">311</span><br><span class="line">312</span><br><span class="line">313</span><br><span class="line">314</span><br><span class="line">315</span><br><span class="line">316</span><br><span class="line">317</span><br><span class="line">318</span><br><span class="line">319</span><br><span class="line">320</span><br><span class="line">321</span><br><span class="line">322</span><br><span class="line">323</span><br><span class="line">324</span><br><span class="line">325</span><br><span class="line">326</span><br><span class="line">327</span><br><span class="line">328</span><br><span class="line">329</span><br><span class="line">330</span><br><span class="line">331</span><br><span class="line">332</span><br><span class="line">333</span><br><span class="line">334</span><br><span class="line">335</span><br><span class="line">336</span><br><span class="line">337</span><br><span class="line">338</span><br><span class="line">339</span><br><span class="line">340</span><br><span class="line">341</span><br><span class="line">342</span><br><span class="line">343</span><br><span class="line">344</span><br><span class="line">345</span><br><span class="line">346</span><br><span class="line">347</span><br><span class="line">348</span><br><span class="line">349</span><br><span class="line">350</span><br><span class="line">351</span><br><span class="line">352</span><br><span class="line">353</span><br><span class="line">354</span><br><span class="line">355</span><br><span class="line">356</span><br><span class="line">357</span><br><span class="line">358</span><br><span class="line">359</span><br><span class="line">360</span><br><span class="line">361</span><br><span class="line">362</span><br><span class="line">363</span><br><span class="line">364</span><br><span class="line">365</span><br><span class="line">366</span><br><span class="line">367</span><br><span class="line">368</span><br><span class="line">369</span><br><span class="line">370</span><br><span class="line">371</span><br><span class="line">372</span><br><span class="line">373</span><br><span class="line">374</span><br><span class="line">375</span><br><span class="line">376</span><br><span class="line">377</span><br><span class="line">378</span><br><span class="line">379</span><br><span class="line">380</span><br><span class="line">381</span><br><span class="line">382</span><br><span class="line">383</span><br><span class="line">384</span><br><span class="line">385</span><br><span class="line">386</span><br><span class="line">387</span><br><span class="line">388</span><br><span class="line">389</span><br><span class="line">390</span><br><span class="line">391</span><br><span class="line">392</span><br><span class="line">393</span><br><span class="line">394</span><br><span class="line">395</span><br><span class="line">396</span><br><span class="line">397</span><br><span class="line">398</span><br><span class="line">399</span><br><span class="line">400</span><br><span class="line">401</span><br><span class="line">402</span><br><span class="line">403</span><br><span class="line">404</span><br><span class="line">405</span><br><span class="line">406</span><br><span class="line">407</span><br><span class="line">408</span><br><span class="line">409</span><br><span class="line">410</span><br><span class="line">411</span><br><span class="line">412</span><br><span class="line">413</span><br><span class="line">414</span><br><span class="line">415</span><br><span class="line">416</span><br><span class="line">417</span><br><span class="line">418</span><br><span class="line">419</span><br><span class="line">420</span><br><span class="line">421</span><br><span class="line">422</span><br><span class="line">423</span><br><span class="line">424</span><br><span class="line">425</span><br><span class="line">426</span><br><span class="line">427</span><br><span class="line">428</span><br><span class="line">429</span><br><span class="line">430</span><br><span class="line">431</span><br><span class="line">432</span><br><span class="line">433</span><br><span class="line">434</span><br><span class="line">435</span><br><span class="line">436</span><br><span class="line">437</span><br><span class="line">438</span><br><span class="line">439</span><br><span class="line">440</span><br><span class="line">441</span><br><span class="line">442</span><br><span class="line">443</span><br><span class="line">444</span><br><span class="line">445</span><br><span class="line">446</span><br><span class="line">447</span><br><span class="line">448</span><br><span class="line">449</span><br><span class="line">450</span><br><span class="line">451</span><br><span class="line">452</span><br><span class="line">453</span><br><span class="line">454</span><br><span class="line">455</span><br><span class="line">456</span><br><span class="line">457</span><br><span class="line">458</span><br><span class="line">459</span><br><span class="line">460</span><br><span class="line">461</span><br><span class="line">462</span><br><span class="line">463</span><br><span class="line">464</span><br><span class="line">465</span><br><span class="line">466</span><br><span class="line">467</span><br><span class="line">468</span><br><span class="line">469</span><br><span class="line">470</span><br><span class="line">471</span><br><span class="line">472</span><br><span class="line">473</span><br><span class="line">474</span><br><span class="line">475</span><br><span class="line">476</span><br><span class="line">477</span><br><span class="line">478</span><br><span class="line">479</span><br><span class="line">480</span><br><span class="line">481</span><br><span class="line">482</span><br><span class="line">483</span><br><span class="line">484</span><br><span class="line">485</span><br><span class="line">486</span><br><span class="line">487</span><br><span class="line">488</span><br><span class="line">489</span><br><span class="line">490</span><br><span class="line">491</span><br><span class="line">492</span><br><span class="line">493</span><br><span class="line">494</span><br><span class="line">495</span><br><span class="line">496</span><br><span class="line">497</span><br><span class="line">498</span><br><span class="line">499</span><br><span class="line">500</span><br><span class="line">501</span><br><span class="line">502</span><br><span class="line">503</span><br><span class="line">504</span><br><span class="line">505</span><br><span class="line">506</span><br><span class="line">507</span><br><span class="line">508</span><br><span class="line">509</span><br><span class="line">510</span><br><span class="line">511</span><br><span class="line">512</span><br><span class="line">513</span><br><span class="line">514</span><br><span class="line">515</span><br><span class="line">516</span><br><span class="line">517</span><br><span class="line">518</span><br><span class="line">519</span><br><span class="line">520</span><br><span class="line">521</span><br><span class="line">522</span><br><span class="line">523</span><br><span class="line">524</span><br><span class="line">525</span><br><span class="line">526</span><br><span class="line">527</span><br><span class="line">528</span><br><span class="line">529</span><br><span class="line">530</span><br><span class="line">531</span><br><span class="line">532</span><br><span class="line">533</span><br><span class="line">534</span><br><span class="line">535</span><br></pre></td><td class="code"><pre><span class="line">##############     version 0.1     ##############</span><br><span class="line">## This WDL converts paired FASTQ to gVCF</span><br><span class="line">##</span><br><span class="line">## Expectations :</span><br><span class="line">##</span><br><span class="line">##     All Sample will be analysed parallelize, caution to the computer performance</span><br><span class="line">##     All Paired Sample should renamed as SAMPLE_NAME_1.fastq.gz and SAMPLE_NAME_2.fastq.gz</span><br><span class="line">##     Volume mroe than triplicate is expected at current dictionary</span><br><span class="line">##</span><br><span class="line">## Dependency :</span><br><span class="line">##</span><br><span class="line">##     samtools</span><br><span class="line">##     trimmomatic</span><br><span class="line">##     bwa mem</span><br><span class="line">##     Picard</span><br><span class="line">##     GATK3.8</span><br><span class="line">##     MILLs, dbSNP, 1kGenome refernece vcf file could find in the follow links:</span><br><span class="line">##     &gt; https://best-practices-for-processing-hts-data.readthedocs.io/en/latest/index.html</span><br><span class="line">##     &gt; https://www.internationalgenome.org/data-portal/data-collection/30x-grch38</span><br><span class="line">##     Hg38 reference genome</span><br><span class="line">##</span><br><span class="line">## Pipeline :</span><br><span class="line">##</span><br><span class="line">##    1: Trimmomatic</span><br><span class="line">##    2: bwa mem</span><br><span class="line">##    3: picard fixmate</span><br><span class="line">##    4: picard sort sam</span><br><span class="line">##    5: picard mark duplicate</span><br><span class="line">##    6: gatk realign</span><br><span class="line">##    7: gatk BQSR</span><br><span class="line">##    8: gatk Haltopviewer call gvcf</span><br><span class="line">##    9: gatk JointGenotyp call merged vcf</span><br><span class="line">##</span><br><span class="line">## Inputs :</span><br><span class="line">##</span><br><span class="line">##     Run $java -jar womtools-version.jar inputs wgs.wdl to fill inputs json file</span><br><span class="line">##</span><br><span class="line">##</span><br><span class="line">## Contact with sunhaoyun1@163.com or sunhaoyun2@gmail.com</span><br><span class="line">##</span><br><span class="line">#################################################</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">############## WORKFLOW DEFINITION ##############</span><br><span class="line">workflow fq2gvcf &#123;</span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">  ##### 输入文本文件要求格式如下 #####：</span><br><span class="line">  ###     Sample1</span><br><span class="line">  ###     Sample2</span><br><span class="line">  ###     Sample3</span><br><span class="line">  ###     Sample4</span><br><span class="line">  ###     Sample5</span><br><span class="line">  ###     Sample6</span><br><span class="line">  ###     Sample7</span><br><span class="line">  ##### 所有输入文件将会并行计算，进行单独的gvcf calling以及最后的merge call #####</span><br><span class="line">  File inputSamplesFile</span><br><span class="line">  Array[Array[File]] inputSamples = read_tsv(inputSamplesFile)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">  ##### 准备所需要的软件 #####</span><br><span class="line">  File trimmomatic</span><br><span class="line">  File bwa</span><br><span class="line">  File gatk</span><br><span class="line">  File picard</span><br><span class="line">  File samtools</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">  ##### 准备需要的参考基因组及SNP信息 ######</span><br><span class="line">  ##### 此处直接使用GATK官网提供的以及谷歌云 #####</span><br><span class="line">  File reffasta</span><br><span class="line">  File refindex</span><br><span class="line">  File refdict</span><br><span class="line">  File indel</span><br><span class="line">  File indel_index</span><br><span class="line">  File mills</span><br><span class="line">  File mills_index</span><br><span class="line">  File db_snp</span><br><span class="line">  File db_snp_index</span><br><span class="line">  File tempdir</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">  ##### 设置Reads Group 以及任务名称 #####</span><br><span class="line">  String threadsnum</span><br><span class="line">  String library_name</span><br><span class="line">  String platform_unit</span><br><span class="line">  String sequencing_center</span><br><span class="line">  String run_date</span><br><span class="line">  File adaptor</span><br><span class="line">  String mission</span><br><span class="line">  </span><br><span class="line"></span><br><span class="line">  ##### 对每一行的样本执行如下命令 #####</span><br><span class="line">  scatter (sample in inputSamples) &#123;</span><br><span class="line">    </span><br><span class="line">      ##### 过滤低质量和接头 #####</span><br><span class="line">      call trim &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      Fq1=sample[0]+"_1.fastq.gz",</span><br><span class="line">      Fq2=sample[0]+"_2.fastq.gz",</span><br><span class="line">      TRIM=trimmomatic,</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      Adaptor=adaptor</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 比对至参考基因组 #####</span><br><span class="line">      call map2genome &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      BWA=bwa,</span><br><span class="line">      clean_Fq2=trim.clean_Fq1,</span><br><span class="line">      clean_Fq1=trim.clean_Fq2,</span><br><span class="line">      Samtools=samtools,</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,            RG="@RG\tID:"+sample[0]+"\tPL:"+platform_unit+"\tLB:"+library_name+"\tSM:"+sample[0]+"\tPU:"+sequencing_center,</span><br><span class="line">      Samtools=samtools</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 矫正双端测序的mate信息 #####</span><br><span class="line">      call fix_mate &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      Picard=picard,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      uBAM=map2genome.uBAM</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 对bam文件进行排序 #####</span><br><span class="line">      call sort_bam &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      Picard=picard,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      fBAM=fix_mate.fix_BAM</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 标记PCR中的重复片段 #####</span><br><span class="line">      call mark_dup &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      Picard=picard,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      sBAM=sort_bam.sort_BAM,</span><br><span class="line">      sBAM_index=sort_bam.sort_BAM_index</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 对INDEL区域进行重排以确定SNP #####</span><br><span class="line">      call realign_bam1 &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      GATK=gatk,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      INDEL=indel,</span><br><span class="line">      INDEL_index=indel_index,</span><br><span class="line">      dedupBAM=mark_dup.dedup_BAM,</span><br><span class="line">      dedupBAM_index=mark_dup.dedup_BAM_index,</span><br><span class="line"></span><br><span class="line">      &#125;</span><br><span class="line">      call realign_bam2 &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      GATK=gatk,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      INDEL=indel,</span><br><span class="line">      INDEL_index=indel_index,</span><br><span class="line">      realign_matrix = realign_bam1.realign_BAM_matrix,</span><br><span class="line">      dedupBAM=mark_dup.dedup_BAM,</span><br><span class="line">      dedupBAM_index=mark_dup.dedup_BAM_index</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 测序碱基质量值 #####</span><br><span class="line">      call bqsr1 &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      GATK=gatk,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      INDEL=indel,</span><br><span class="line">      INDEL_index=indel_index,</span><br><span class="line">      MILLs=mills,</span><br><span class="line">      MILLs_index=mills_index,</span><br><span class="line">      DB_SNP=db_snp,</span><br><span class="line">      DB_SNP_index=db_snp_index,</span><br><span class="line">      realignBAM=realign_bam2.realign_BAM,</span><br><span class="line">      realignBAM_index=realign_bam2.realign_BAM_index</span><br><span class="line">      &#125;</span><br><span class="line">      call bqsr2 &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      GATK=gatk,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      BQSR_BAM_matrix = bqsr1.BQSR_BAM_matrix,</span><br><span class="line">      realignBAM=realign_bam2.realign_BAM,</span><br><span class="line">      realignBAM_index=realign_bam2.realign_BAM_index</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### gVCF calling #####</span><br><span class="line">      call HaplotypeCaller&#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      BQSRBAM=bqsr2.BQSR_BAM,</span><br><span class="line">      BQSR_BAM_index=bqsr2.BQSR_BAM_index,</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      GATK=gatk,</span><br><span class="line">      TempDir=tempdir</span><br><span class="line">      &#125;</span><br><span class="line">  &#125;</span><br><span class="line"> </span><br><span class="line"></span><br><span class="line">      ##### 将得到的所有样本的gVCF合并进行joint calling #####</span><br><span class="line">      call GenotypeGVCFs&#123;</span><br><span class="line">      input:GATK=gatk,</span><br><span class="line">      RefFasta=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      Total_name=mission,</span><br><span class="line">      DB_SNP=db_snp,</span><br><span class="line">      DB_SNP_index=db_snp_index,</span><br><span class="line">      GVCFs=HaplotypeCaller.Haplotype_vcf</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task trim &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File TRIM</span><br><span class="line">  File Fq1</span><br><span class="line">  File Fq2</span><br><span class="line">  File TempDir</span><br><span class="line">  File Adaptor</span><br><span class="line">  String ThreadsNum</span><br><span class="line"></span><br><span class="line">  command &#123;</span><br><span class="line">    java -Djava.io.tmpdir=$&#123;TempDir&#125; -Xmx64g -jar $&#123;TRIM&#125; \</span><br><span class="line">        PE -phred33 \</span><br><span class="line">        -threads $&#123;ThreadsNum&#125; \</span><br><span class="line">        $&#123;Fq1&#125; $&#123;Fq2&#125; \</span><br><span class="line">        $&#123;SampleName&#125;.1f.pair.fq.gz $&#123;SampleName&#125;.1f.upair.fq.gz \</span><br><span class="line">        $&#123;SampleName&#125;.2r.pair.fq.gz $&#123;SampleName&#125;.2r.upair.fq.gz  \</span><br><span class="line">       ILLUMINACLIP:$&#123;Adaptor&#125;:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File clean_Fq1 = "$&#123;SampleName&#125;.1f.pair.fq.gz"</span><br><span class="line">    File clean_Fq2 = "$&#123;SampleName&#125;.2r.pair.fq.gz"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task map2genome &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File BWA</span><br><span class="line">  File clean_Fq1</span><br><span class="line">  File clean_Fq2</span><br><span class="line">  File Samtools</span><br><span class="line">  String RG</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  command &#123;</span><br><span class="line">    $&#123;BWA&#125; mem -Y -t $&#123;ThreadsNum&#125; -R '$&#123;RG&#125;' -M $&#123;RefGenome&#125; \</span><br><span class="line">        $&#123;clean_Fq1&#125; $&#123;clean_Fq2&#125;  | \</span><br><span class="line">        $&#123;Samtools&#125; view -Shb -o $&#123;SampleName&#125;.ubam -</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File uBAM = "$&#123;SampleName&#125;.ubam"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task fix_mate &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File Picard</span><br><span class="line">  File TempDir</span><br><span class="line">  File uBAM</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;Picard&#125; FixMateInformation MAX_RECORDS_IN_RAM=2000000 \</span><br><span class="line">        VALIDATION_STRINGENCY=SILENT ADD_MATE_CIGAR=True \</span><br><span class="line">        ASSUME_SORTED=true I=$&#123;uBAM&#125; O=$&#123;SampleName&#125;.fbam</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File fix_BAM = "$&#123;SampleName&#125;.fix_mate.bam"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task sort_bam &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File Picard</span><br><span class="line">  File TempDir</span><br><span class="line">  File fBAM</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;Picard&#125; SortSam MAX_RECORDS_IN_RAM=2000000 \</span><br><span class="line">        VALIDATION_STRINGENCY=SILENT SORT_ORDER=coordinate \</span><br><span class="line">        CREATE_INDEX=true  I=$&#123;fBAM&#125; O=$&#123;SampleName&#125;.sort.bam</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File sort_BAM = "$&#123;SampleName&#125;.sort.bam"</span><br><span class="line">    File sort_BAM_index = '$&#123;SampleName&#125;.sort.bai'</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task mark_dup &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File Picard</span><br><span class="line">  File TempDir</span><br><span class="line">  File sBAM</span><br><span class="line">  File sBAM_index</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;Picard&#125; MarkDuplicates MAX_RECORDS_IN_RAM=2000000 \</span><br><span class="line">        VALIDATION_STRINGENCY=SILENT M=$&#123;SampleName&#125;.matrix \</span><br><span class="line">        CREATE_INDEX=true  I=$&#123;sBAM&#125; O=$&#123;SampleName&#125;.dedup.bam</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File dedup_BAM = "$&#123;SampleName&#125;.dedup.bam"</span><br><span class="line">    File dedup_BAM_index = "$&#123;SampleName&#125;.dedup.bai"</span><br><span class="line">    File dedup_BAM_matrix = "$&#123;SampleName&#125;.matrix"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task realign_bam1 &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File GATK</span><br><span class="line">  File TempDir</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  File INDEL</span><br><span class="line">  File INDEL_index</span><br><span class="line">  File dedupBAM</span><br><span class="line">  File dedupBAM_index</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;GATK&#125; -T RealignerTargetCreator \</span><br><span class="line">        -R $&#123;RefGenome&#125; -I $&#123;dedupBAM&#125; -known $&#123;INDEL&#125; \</span><br><span class="line">        -o realignment_targets.list</span><br><span class="line">    &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">        File realign_BAM_matrix = "realignment_targets.list"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task realign_bam2 &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File GATK</span><br><span class="line">  File TempDir</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  File INDEL</span><br><span class="line">  File INDEL_index</span><br><span class="line">  File realign_matrix</span><br><span class="line">  File dedupBAM</span><br><span class="line">  File dedupBAM_index</span><br><span class="line">  command &#123;</span><br><span class="line">     java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;GATK&#125; -T IndelRealigner -R $&#123;RefGenome&#125; \</span><br><span class="line">        -I $&#123;dedupBAM&#125; -targetIntervals realignment_targets.list \</span><br><span class="line">        -known $&#123;INDEL&#125; CREATE_INDEX=true -o $&#123;SampleName&#125;.realign.bam</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File realign_BAM = "$&#123;SampleName&#125;.realign.bam"</span><br><span class="line">    File realign_BAM_index = "$&#123;SampleName&#125;.realign.bai"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task bqsr1 &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File GATK</span><br><span class="line">  File TempDir</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  File INDEL</span><br><span class="line">  File INDEL_index</span><br><span class="line">  File MILLs</span><br><span class="line">  File MILLs_index</span><br><span class="line">  File DB_SNP</span><br><span class="line">  File DB_SNP_index</span><br><span class="line">  File realignBAM</span><br><span class="line">  File realignBAM_index</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;GATK&#125; -T BaseRecalibrator \</span><br><span class="line">        --preserve_qscores_less_than 6 \</span><br><span class="line">        -nct $&#123;ThreadsNum&#125; -R $&#123;RefGenome&#125; -I $&#123;realignBAM&#125; \</span><br><span class="line">        -XL chrX -XL chrY -XL chrM \</span><br><span class="line">        -knownSites $&#123;INDEL&#125; \</span><br><span class="line">        -knownSites $&#123;MILLs&#125; \</span><br><span class="line">        -knownSites $&#123;DB_SNP&#125; \</span><br><span class="line">        -o recal_data.table   &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File BQSR_BAM_matrix = "$&#123;SampleName&#125;.recal_data.table"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task bqsr2 &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File GATK</span><br><span class="line">  File TempDir</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  File BQSR_BAM_matrix</span><br><span class="line">  File realignBAM</span><br><span class="line">  File realignBAM_index</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;GATK&#125; -T PrintReads -R $&#123;RefGenome&#125; \</span><br><span class="line">        --preserve_qscores_less_than 6 \</span><br><span class="line">        -nct $&#123;ThreadsNum&#125; -I $&#123;realignBAM&#125; \</span><br><span class="line">        -SQQ 10 -SQQ 20 -SQQ 30 -rf BadCigar \</span><br><span class="line">        -BQSR recal_data.table -o $&#123;SampleName&#125;.BQSR.bam\</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File BQSR_BAM = "$&#123;SampleName&#125;.BQSR.bam"</span><br><span class="line">    File BQSR_BAM_index = "$&#123;SampleName&#125;.BQSR.bai"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task HaplotypeCaller &#123;</span><br><span class="line">String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File GATK</span><br><span class="line">  File TempDir</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  File BQSRBAM</span><br><span class="line">  File BQSR_BAM_index</span><br><span class="line">  command &#123;</span><br><span class="line">        java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;GATK&#125; \</span><br><span class="line">        -T HaplotypeCaller --genotyping_mode DISCOVERY \</span><br><span class="line">        -A AlleleBalanceBySample -A DepthPerAlleleBySample \</span><br><span class="line">        -A DepthPerSampleHC -A InbreedingCoeff \</span><br><span class="line">        -A MappingQualityZeroBySample -A StrandBiasBySample \</span><br><span class="line">        -A Coverage -A FisherStrand -A HaplotypeScore \</span><br><span class="line">        -A MappingQualityRankSumTest -A MappingQualityZero \</span><br><span class="line">        -A QualByDepth -A RMSMappingQuality \</span><br><span class="line">        -A ReadPosRankSumTest -A VariantType -l INFO \</span><br><span class="line">        --emitRefConfidence GVCF -rf BadCigar \</span><br><span class="line">        --variant_index_parameter 128000 --variant_index_type LINEAR \</span><br><span class="line">        -R $&#123;RefGenome&#125; -I $&#123;BQSRBAM&#125; -o $&#123;SampleName&#125;.HaplotypeCaller.g.vcf</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File Haplotype_vcf = "$&#123;SampleName&#125;.HaplotypeCaller.g.vcf"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line">task GenotypeGVCFs &#123;</span><br><span class="line">  File GATK</span><br><span class="line">  File RefFasta</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  Array[File] GVCFs</span><br><span class="line">  String Total_name</span><br><span class="line">  File DB_SNP</span><br><span class="line">  File DB_SNP_index</span><br><span class="line"></span><br><span class="line">  command &#123;</span><br><span class="line">    java -jar $&#123;GATK&#125; \</span><br><span class="line">        -T GenotypeGVCFs \</span><br><span class="line">        -R $&#123;RefFasta&#125; \</span><br><span class="line">        -G StandardAnnotation \</span><br><span class="line">        -D $&#123;DB_SNP&#125; \</span><br><span class="line">        -V $&#123;sep=" -V " GVCFs&#125; \</span><br><span class="line">        -o $&#123;Total_name&#125;_rawVariants.vcf</span><br><span class="line">  &#125;</span><br><span class="line">  output &#123;</span><br><span class="line">    File rawVCF = "$&#123;Total_name&#125;_rawVariants.vcf"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

<h4 id="输入文件部分"><a href="#输入文件部分" class="headerlink" title="输入文件部分"></a>输入文件部分</h4><figure class="highlight json"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">&#123;</span><br><span class="line">  <span class="attr">"fq2gvcf.adaptor"</span>: <span class="string">"/home/sunhaoyun/software/adapters/adaptor.fa"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.inputSamplesFile"</span>: <span class="string">"/home/sunhaoyun/working/SGDP/samples_list.txt"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.picard"</span>: <span class="string">"/home/sunhaoyun/software/picard.jar"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.indel"</span>: <span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.known_indels.vcf.gz"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.indel_index"</span>: <span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.known_indels.vcf.gz.tbi"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.tempdir"</span>: <span class="string">"/data/sunhaoyun/temp"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.mission"</span>: <span class="string">"TEST"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.refindex"</span>: <span class="string">"/home/sunhaoyun/working/reference_genome/Homo_sapiens_assembly38.fai"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.refdict"</span>: <span class="string">"/home/sunhaoyun/working/reference_genome/Homo_sapiens_assembly38.dict"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.library_name"</span>: <span class="string">"TEST"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.platform_unit"</span>: <span class="string">"ILLMINA"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.sequencing_center"</span>: <span class="string">"JSNU"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.run_date"</span>: <span class="string">"20200309"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.reffasta"</span>: <span class="string">"/home/sunhaoyun/working/reference_genome/Homo_sapiens_assembly38.fasta"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.threadsnum"</span>: <span class="string">"24"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.gatk"</span>: <span class="string">"/home/sunhaoyun/software/GenomeAnalysisTK.jar"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.mills"</span>: <span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Mills_and_1000G_gold_standard.indels.hg38.vcf.gz"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.mills_index"</span>:<span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Mills_and_1000G_gold_standard.indels.hg38.vcf.gz.tbi"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.db_snp"</span>: <span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.dbsnp138.vcf.gz"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.db_snp_index"</span>: <span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.dbsnp138.vcf.gz.tbi"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.samtools"</span>: <span class="string">"/home/sunhaoyun/software/samtools/bin/samtools"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.bwa"</span>: <span class="string">"/usr/bin/bwa"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.trimmomatic"</span>: <span class="string">"/home/sunhaoyun/software/trimmomatic-0.39.jar"</span></span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">java -jar cromwell-59.jar run wgs.wdl -i wgs.inputs.json</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2021/04/26/GATK_WDL/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2021/04/13/othmcl/" >orthomcl鉴定直系同源基因</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2021-04-13  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>使用MySQL的时候可能会遇到的问题。</p>
<ol>
<li><p>由于Mysql默认要求复杂密码 当出现特殊字符时候 会导致orthomcl运行出错，所以可以先取消复杂密码设置</p>
   <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">SHOW VARIABLES LIKE &apos;validate_password%&apos;;</span><br><span class="line">+--------------------------------------+-------+  </span><br><span class="line">| Variable_name                        | Value |  </span><br><span class="line">+--------------------------------------+-------+  </span><br><span class="line">| validate_password_dictionary_file    |       |  </span><br><span class="line">| validate_password_length             | 8     |  </span><br><span class="line">| validate_password_mixed_case_count   | 1     |  </span><br><span class="line">| validate_password_number_count       | 3     |  </span><br><span class="line">| validate_password_policy             | LOW   |  </span><br><span class="line">| validate_password_special_char_count | 1     |  </span><br><span class="line">+--------------------------------------+-------+  </span><br><span class="line">set global validate_password_special_char_count=0;</span><br></pre></td></tr></table></figure>



</li>
</ol>
<ol start="2">
<li><p>在设置MySQL时候由于数据结构类型，可能导致超出缓存大小，有两种解决方法</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br></pre></td><td class="code"><pre><span class="line"># 报错信息如下:</span><br><span class="line"># DBD::mysql::st execute failed: The total number of locks exceeds the lock table size at /home/RNAseq/miniconda/bin/orthomclPairs line 709, &lt;F&gt; line 12.</span><br><span class="line">show variables like &quot;%_buffer%&quot;; #查看缓存</span><br><span class="line">+-------------------------------------+----------------+</span><br><span class="line">| Variable_name                       | Value          |</span><br><span class="line">+-------------------------------------+----------------+</span><br><span class="line">| bulk_insert_buffer_size             | 8388608        |</span><br><span class="line">| innodb_buffer_pool_chunk_size       | 134217728      |</span><br><span class="line">| innodb_buffer_pool_dump_at_shutdown | ON             |</span><br><span class="line">| innodb_buffer_pool_dump_now         | OFF            |</span><br><span class="line">| innodb_buffer_pool_dump_pct         | 25             |</span><br><span class="line">| innodb_buffer_pool_filename         | ib_buffer_pool |</span><br><span class="line">| innodb_buffer_pool_in_core_file     | ON             |</span><br><span class="line">| innodb_buffer_pool_instances        | 1              |</span><br><span class="line">| innodb_buffer_pool_load_abort       | OFF            |</span><br><span class="line">| innodb_buffer_pool_load_at_startup  | ON             |</span><br><span class="line">| innodb_buffer_pool_load_now         | OFF            |</span><br><span class="line">| innodb_buffer_pool_size             | 134217728      |</span><br><span class="line">| innodb_change_buffer_max_size       | 25             |</span><br><span class="line">| innodb_change_buffering             | all            |</span><br><span class="line">| innodb_log_buffer_size              | 16777216       |</span><br><span class="line">| innodb_sort_buffer_size             | 1048576        |</span><br><span class="line">| join_buffer_size                    | 262144         |</span><br><span class="line">| key_buffer_size                     | 8388608        |</span><br><span class="line">| myisam_sort_buffer_size             | 8388608        |</span><br><span class="line">| net_buffer_length                   | 16384          |</span><br><span class="line">| preload_buffer_size                 | 32768          |</span><br><span class="line">| read_buffer_size                    | 131072         |</span><br><span class="line">| read_rnd_buffer_size                | 262144         |</span><br><span class="line">| sort_buffer_size                    | 262144         |</span><br><span class="line">| sql_buffer_result                   | OFF            |</span><br><span class="line">+-------------------------------------+----------------+</span><br><span class="line"># 默认innodb_buffer_pool_size太小了</span><br><span class="line">SET GLOBAL innodb_buffer_pool_size=64*1024*1024*1024;</span><br></pre></td></tr></table></figure>



</li>
</ol>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">CREATE USER &apos;orthomcl&apos;@&apos;localhost&apos; IDENTIFIED BY &apos;orthomcl&apos;;</span><br><span class="line">SELECT TABLE_NAME, ENGINE FROM information_schema.TABLES where TABLE_SCHEMA = &apos;database&apos;;</span><br><span class="line">#如果不想更改缓存的话 把储存模式改为myisam</span><br><span class="line">GRANT ALL PRIVILEGES ON orthomcl.* TO &apos;orthomcl&apos;@&apos;localhost&apos; IDENTIFIED BY &apos;orthomcl&apos;;</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">还需要改一下输入pep的head用来适配orthomcl的兼容模式，并过滤小于50aa的氨基酸</span></span><br><span class="line">bioawk -c fastx '&#123; print "&gt;Ib|"$name; print $seq &#125;'Ibatatas.faa &gt; Ibatatas.fa</span><br><span class="line">seqkit seq -m 50 Ibatatas.fa &gt; Ibatatas.fasta</span><br></pre></td></tr></table></figure>

<p>找出1:1:1:1的直系同源基因</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">cat groups/groups.all.counts.txt | \</span><br><span class="line">awk &apos;&#123; if ($2 == 1 &amp;&amp; $3 == 1 &amp;&amp; $4 == 1 &amp;&amp; $5 == 1 &amp;&amp; $6 == 1) print $0 &#125;&apos; | \</span><br><span class="line">wc -l</span><br></pre></td></tr></table></figure>

<p>在两个物种以上的同源基因</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">cat groups/groups.all.counts.txt | \</span><br><span class="line">awk &apos;&#123; if (($2 == 1 || $2 == 0) &amp;&amp; ($3 == 1 || $3 == 0) &amp;&amp; ($4 == 1 || $4 == 0) &amp;&amp; \</span><br><span class="line">($5 == 1 || $5 == 0) &amp;&amp; ($6 == 1 || $6 == 0)) print $0 &#125;&apos; | \</span><br><span class="line">wc -l</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">cat groups/groups.all.counts.txt | \</span><br><span class="line">awk '&#123; if (($2 == 1 || $2 == 0) &amp;&amp; ($3 == 1 || $3 == 0) &amp;&amp; ($4 == 1 || $4 == 0) &amp;&amp; \</span><br><span class="line"><span class="meta">($</span><span class="bash">5 == 1 || <span class="variable">$5</span> == 0) &amp;&amp; (<span class="variable">$6</span> == 1 || <span class="variable">$6</span> == 0)) <span class="built_in">print</span> <span class="variable">$0</span> &#125;<span class="string">' | \</span></span></span><br><span class="line">awk '&#123; for (i=2; i&lt;=NF; i++) sum[i] += $i &#125; END &#123; for (i in sum) print i, sum[i] &#125;' | \</span><br><span class="line">sort -k1,1</span><br></pre></td></tr></table></figure>

<p>有多少个类群至少包含两个仅在一个物种中鉴定得到的</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">cat groups/groups.all.counts.txt | \</span><br><span class="line">awk '&#123; count=0; for (i=2; i&lt;=NF; i++) if ($i == 0) count += 1; if (count == NF - 2) print $0 &#125;' |  \</span><br><span class="line">awk '&#123; for (i=2; i&lt;=NF; i++) sum[i] += $i &#125; END &#123;for (i in sum) print i, sum[i] &#125;' | \</span><br><span class="line">sort -k1,1</span><br></pre></td></tr></table></figure>

<p>扩展模式</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">cat groups/groups.all.counts.txt | \</span><br><span class="line">awk '&#123; count=0; for (i=2; i&lt;=NF; i++) if ($i == 0) count += 1; if (count &lt; NF - 2) print $0 &#125;' |  \</span><br><span class="line">awk '&#123; for (i=2; i&lt;=NF; i++) if ($i &gt; 1) sum[i] += 1 &#125; END &#123; for (i in sum) print i, sum[i] - 1 &#125;' | \</span><br><span class="line">sort -k1,1</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">cat groups/groups.all.counts.txt | \</span><br><span class="line">awk '&#123; count=0; for (i=2; i&lt;=NF; i++) if ($i == 0) count += 1; if (count &lt; NF - 2) print $0 &#125;' |  \</span><br><span class="line">awk '&#123; for (i=2; i&lt;=NF; i++) if ($i == 1) sum[i] += 1 &#125; END &#123; for (i in sum) print i, sum[i] &#125;' | \</span><br><span class="line">sort -k1,1</span><br></pre></td></tr></table></figure>

<p>另外在附加使用Inparanoid鉴定Target物种和模式物种的同源关系</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 首先需要编辑perl脚本中调节参数，根据两个物种的亲缘关系远近程度选择合适的BLOSUM和PAM</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 随后</span></span><br><span class="line">nohup perl inparanoid.pl batatas.longest.pep.fa longest.at.pep.fa</span><br><span class="line"><span class="meta">#</span><span class="bash"> 得到的结果也比较清晰，Inparanoid会先将物种内部基因相互blast然后再种间blast</span></span><br><span class="line">table.batatas.longest.pep.fa-longest.at.pep.fa</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2021/04/13/othmcl/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2021/04/06/Inparanoid/" >Inparanoid</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2021-04-06  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h3 id="使用Inparanoid鉴定直系同源基因"><a href="#使用Inparanoid鉴定直系同源基因" class="headerlink" title="使用Inparanoid鉴定直系同源基因"></a>使用<a href="https://github.com/Junting98/Inparanoid" target="_blank" rel="noopener">Inparanoid</a>鉴定直系同源基因</h3><blockquote>
<p>依赖 :</p>
<p>perl :: XML</p>
<p>formatdb blast</p>
</blockquote>

	
	</div>
  <a type="button" href="/2021/04/06/Inparanoid/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2021/03/26/Python 卡方检验/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2021-03-26  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>Python t-test</p>
<p>首先获取位点可能最接近的省份</p>
<p>####需要满足如下规则：</p>
<p><strong><em>1185个位点在芯片上可以找到399个（根据位置和rs号）</em></strong></p>
<p>####程序编写</p>
<p>1.所有省份为分布有差异的前提下，就是假如江苏浙江两个省份399个点算下来自身就没有差异，那我们算的也没有差异。</p>
<p>2.399个位点中，最佳匹配的省份和第二匹配的省份两组数据t-test差异显著</p>
<p>3.复杂一些 设计模型 卡方检验</p>
<p>模型1:这个人在所有的省份分布上没有差异 （所有省份赋予同一概率）计算 得到 lnL1</p>
<p>模型2:这个人在前三个最佳匹配的省份上，分布上均有差异（在新疆、台湾、河北上具有不同的概率，在其余省份具有一个概率）计算得到 lnL2</p>
<p>模型2 <strong><em>vs.</em></strong> 模型1。 得到Chi- square 自由度为省份数 计算P值（得到 前三个省份是否足够显著突出于其余省份）</p>
<p>模型3:这个人在新疆分布上和背景一致，在台湾和河北具有突出的概率（台湾、河北、新疆+backgroud）</p>
<p>模型3 <strong><em>vs</em></strong>. 模型2 对新疆可以从backgroud中分离的假设检验得到Chi-square和P</p>
<p>模型4 <strong><em>vs</em></strong>. 模型2 对台湾可以从backgroud中分离的假设检验得到Chi-square和P</p>
<p>模型5 <strong><em>vs</em></strong>. 模型2 对河北可以从backgroud中分离的假设检验得到Chi-square和P</p>
<p>则当假设检验 2 <strong><em>vs.</em></strong> 1成立（P&lt;0.05） 且 3 <strong><em>vs.</em></strong> 2成立（P&lt;0.05 且 4 <strong><em>vs</em></strong>. 2不成立(P&gt;0.05) 且 5 <strong><em>vs</em></strong>.2 不成立(P&gt;0.05) </p>
<p>才能够判断此人为新疆人</p>

	
	</div>
  <a type="button" href="/2021/03/26/Python 卡方检验/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2021/02/14/CV记录/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2021-02-14  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>CV记录</p>
<p>子安武人：dio 罗兹瓦尔</p>
<p>茅野爱衣：　川本明里（明里姐）</p>

	
	</div>
  <a type="button" href="/2021/02/14/CV记录/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/29/bwa_picard_pipeline/" >二代重测序流程</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-29  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>首先使用迅雷下载sra文件 速度谁用谁知道</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"># 将ncbi 的 sra格式转化为fastq格式</span><br><span class="line">nohup fastq-dump --split-3 --gzip -O /home/sunhaoyun/data/ERR1347700.1 &amp;</span><br></pre></td></tr></table></figure>

<p>然后call就vans了</p>
<p>还有一点就是之前动物所的老师想要把60X变成20X的深度的reads，那边师兄给的方法是直接把Fq扔掉1/3，</p>
<p>现在学了个聪明点的方法，用rarefraction去除比较有可信度</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#################################</span></span><br><span class="line"><span class="comment">#bwa picard GATK pipeline</span></span><br><span class="line"><span class="comment">#@author: sunhaoyun </span></span><br><span class="line"><span class="comment">#@contact qq: 172293639</span></span><br><span class="line"><span class="comment"># STEP follow the 1000genome project high coverage sequencing wheel</span></span><br><span class="line"><span class="comment"># the specific prameter could check NYGC websites :)</span></span><br><span class="line"><span class="comment"># commond reference webites:</span></span><br><span class="line"><span class="comment"># https://best-practices-for-processing-hts-data.readthedocs.io/en/latest/index.html</span></span><br><span class="line"><span class="comment"># https://www.internationalgenome.org/data-portal/data-collection/30x-grch38</span></span><br><span class="line"><span class="comment"># reference genome could found in this google cloud sites:</span></span><br><span class="line"><span class="comment"># There will be about 3X Volume storage space bigger than Fq files</span></span><br><span class="line"><span class="comment">#1: Trimmomatic</span></span><br><span class="line"><span class="comment">#2: bwa mem</span></span><br><span class="line"><span class="comment">#3. picard fixmate</span></span><br><span class="line"><span class="comment">#4: picard sort sam</span></span><br><span class="line"><span class="comment">#5: picard mark duplicate</span></span><br><span class="line"><span class="comment">#6: gatk realign</span></span><br><span class="line"><span class="comment">#7: gatk BQSR</span></span><br><span class="line"><span class="comment">#8: gatk haltopviewer call gvcf</span></span><br><span class="line"><span class="comment">################################</span></span><br><span class="line"><span class="keyword">import</span> subprocess</span><br><span class="line"><span class="keyword">import</span> argparse</span><br><span class="line"><span class="keyword">import</span> time</span><br><span class="line"><span class="keyword">import</span> os.path</span><br><span class="line"><span class="keyword">from</span> pathlib <span class="keyword">import</span> Path</span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line"><span class="comment">################### SET SOFTWARE PATH HERE ################### </span></span><br><span class="line"></span><br><span class="line">adaptor_path = <span class="string">"/home/gujiaqi/software/adapters/adaptor.fa"</span></span><br><span class="line">trimmomatic_path = <span class="string">"/home/gujiaqi/software/trimmomatic-0.39.jar"</span></span><br><span class="line">bwa_path = <span class="string">"/usr/bin/bwa"</span> </span><br><span class="line">picard_path = <span class="string">"/home/gujiaqi/software/picard.jar"</span></span><br><span class="line">gatk_path = <span class="string">"/home/gujiaqi/software/GenomeAnalysisTK.jar"</span></span><br><span class="line">temp_path = <span class="string">"/data/gujiaqi/temp"</span></span><br><span class="line">db_snp = <span class="string">"/data/gujiaqi/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.dbsnp138.vcf.gz"</span></span><br><span class="line">known_indel = <span class="string">"/data/gujiaqi/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.known_indels.vcf.gz"</span></span><br><span class="line">mills = <span class="string">"/data/gujiaqi/reference_genome/resources_broad_hg38_v0_Mills_and_1000G_gold_standard.indels.hg38.vcf.gz"</span></span><br><span class="line"><span class="comment">###################    SOFTWARE CHECK   ######################</span></span><br><span class="line"><span class="keyword">if</span> <span class="keyword">not</span> Path(adaptor_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(temp_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(trimmomatic_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(bwa_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(picard_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(gatk_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(db_snp).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(known_indel).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(mills).exists():</span><br><span class="line">    print(<span class="string">"SOFTWARE SET FAILED, PLEASE CHECK AGAIN"</span>)</span><br><span class="line">    sys.exit()</span><br><span class="line">    </span><br><span class="line"></span><br><span class="line"></span><br><span class="line">ap = argparse.ArgumentParser()</span><br><span class="line">ap.add_argument(<span class="string">"-p"</span>, <span class="string">"--prefix"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"the  prefix of sample"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-f"</span>, <span class="string">"--refer"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"reference genome path"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-o"</span>, <span class="string">"--outpath"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"Out put file path"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-t"</span>, <span class="string">"--threads"</span>, required=<span class="literal">False</span>,</span><br><span class="line">    help=<span class="string">"process will be used"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-1"</span>, <span class="string">"--reads1"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"Input followed reads"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-2"</span>, <span class="string">"--reads2"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"Input reverse reads"</span>)</span><br><span class="line"></span><br><span class="line">args = vars(ap.parse_args())</span><br><span class="line"><span class="keyword">print</span> (<span class="string">"The Task id is &#123;&#125;"</span>.format(args[<span class="string">"prefix"</span>]))</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> args[<span class="string">"threads"</span>]:</span><br><span class="line">    cpu = args[<span class="string">"threads"</span>]</span><br><span class="line"><span class="keyword">else</span>:</span><br><span class="line">    cpu = <span class="number">8</span></span><br><span class="line"><span class="keyword">print</span> (<span class="string">"DATE STATUS: The Task Start at: &#123;&#125;"</span>.format(time.ctime())) </span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> <span class="keyword">not</span> Path(args[<span class="string">"prefix"</span>]).exists():</span><br><span class="line">    mk_work_dir = <span class="string">'mkdir &#123;&#125;'</span>.format(args[<span class="string">"prefix"</span>])</span><br><span class="line">    prefix = <span class="string">"&#123;&#125;"</span>.format(args[<span class="string">"prefix"</span>])</span><br><span class="line">    subprocess.call(mk_work_dir, shell=<span class="literal">True</span>)</span><br><span class="line">    print(<span class="string">"Work dictionary &#123;&#125; is created"</span>.format(args[<span class="string">"prefix"</span>]))</span><br><span class="line"><span class="keyword">else</span>:</span><br><span class="line">    prefix = <span class="string">"&#123;&#125;"</span>.format(args[<span class="string">"prefix"</span>])</span><br><span class="line">    print(<span class="string">"Work dictionary  &#123;&#125; is created"</span>.format(args[<span class="string">"prefix"</span>]))</span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 1 ####### </span></span><br><span class="line"><span class="comment">### Trimmomatic software should be downloaded ###</span></span><br><span class="line"><span class="comment">### You could modify the adaptor file and trimmomatic path bellow</span></span><br><span class="line"></span><br><span class="line"><span class="comment">### prepare well ###</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 1 ... filtering the fastq files \n\n\n"</span>)</span><br><span class="line">trim1_pa = <span class="string">"&#123;&#125;/&#123;&#125;_f1.trim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line">trim1_up = <span class="string">"&#123;&#125;/&#123;&#125;_f1.untrim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line">trim2_pa = <span class="string">"&#123;&#125;/&#123;&#125;_r2.trim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line">trim2_up = <span class="string">"&#123;&#125;/&#123;&#125;_r2.untrim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line"></span><br><span class="line">Trim_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -jar &#123;&#125; \</span></span><br><span class="line"><span class="string">PE -phred33 -threads &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; \</span></span><br><span class="line"><span class="string">ILLUMINACLIP:&#123;&#125;:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50"</span>\</span><br><span class="line">.format(cpu,trimmomatic_path,cpu,args[<span class="string">"reads1"</span>],args[<span class="string">"reads2"</span>]\</span><br><span class="line">,trim1_pa,trim1_up,trim2_pa,trim2_up,adaptor_path)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(Trim_cmd))</span><br><span class="line">subprocess.call(Trim_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: Trim Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 2 ####### </span></span><br><span class="line"><span class="comment">### bwa software should be downloaded ###</span></span><br><span class="line"><span class="comment">### You could modify the bwa path bellow</span></span><br><span class="line"><span class="comment">### Check for index file completely ###</span></span><br><span class="line">genome = <span class="string">"&#123;&#125;"</span>.format(args[<span class="string">"refer"</span>])</span><br><span class="line">bwa_index = <span class="string">"&#123;&#125;.amb"</span>.format(genome)</span><br><span class="line">genome_dict = <span class="string">'&#123;&#125;.dict'</span>.format(genome)</span><br><span class="line">fasta_index = <span class="string">'&#123;&#125;.fai'</span>.format(genome)</span><br><span class="line"><span class="keyword">if</span> <span class="keyword">not</span> Path(bwa_index).exists() <span class="keyword">or</span> <span class="keyword">not</span> Path(bwa_index).exists() <span class="keyword">or</span> <span class="keyword">not</span> Path(bwa_index).exists():</span><br><span class="line">    print(<span class="string">"some index file are missing, please offer bwa/samtools/picard index~"</span>)</span><br><span class="line">    print(<span class="string">"DATE STATUS: Failed at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line">    sys.exit()</span><br><span class="line"><span class="keyword">else</span>:</span><br><span class="line">    print(<span class="string">"index file are repared well"</span>)</span><br><span class="line">    RG = <span class="string">r"@RG\tID:&#123;&#125;\tPL:ILLUMINA\tLB:&#123;&#125;\tSM:&#123;&#125;\tPU:1"</span>.format(prefix,prefix,prefix)</span><br><span class="line">    print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 2 ... align the fastq files to &#123;&#125; \n\n\n"</span>.format(genome))</span><br><span class="line">    bwa_cmd = <span class="string">"&#123;&#125; mem -Y -t &#123;&#125; -R '&#123;&#125;' -M  &#123;&#125; &#123;&#125; &#123;&#125; | \</span></span><br><span class="line"><span class="string">    samtools view -Shb -o &#123;&#125;/&#123;&#125;.bam -"</span>\</span><br><span class="line">    .format(bwa_path,cpu,RG,genome,trim1_pa,trim2_pa,prefix,prefix)</span><br><span class="line">    print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(bwa_cmd))</span><br><span class="line">    subprocess.call(bwa_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">    print(<span class="string">"DATE STATUS: BWA MEM Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 3 ####### </span></span><br><span class="line"><span class="comment">#### picard fix mate information</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 3 ... Fix mate information in the BAM \n\n\n"</span>)</span><br><span class="line">fix_mate_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; FixMateInformation MAX_RECORDS_IN_RAM=2000000 \</span></span><br><span class="line"><span class="string">VALIDATION_STRINGENCY=SILENT ADD_MATE_CIGAR=True \</span></span><br><span class="line"><span class="string">ASSUME_SORTED=true I=&#123;&#125;/&#123;&#125;.bam O=&#123;&#125;/&#123;&#125;.fixmate.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,picard_path,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(fix_mate_cmd))</span><br><span class="line">subprocess.call(fix_mate_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: PICARD FIX MATE Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 4 ####### </span></span><br><span class="line"><span class="comment">#### picard sort bam</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 4 ... coordinate sort BAM \n\n\n"</span>)</span><br><span class="line">sort_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; SortSam MAX_RECORDS_IN_RAM=2000000 \</span></span><br><span class="line"><span class="string">VALIDATION_STRINGENCY=SILENT SORT_ORDER=coordinate \</span></span><br><span class="line"><span class="string">CREATE_INDEX=true I=&#123;&#125;/&#123;&#125;.fixmate.bam O=&#123;&#125;/&#123;&#125;.sorted.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,picard_path,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(sort_cmd))</span><br><span class="line">subprocess.call(sort_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: PICARD SORT SAM Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 5 ####### </span></span><br><span class="line"><span class="comment">#### picard mark duplication </span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 5 ... Mark PCR duplicates \n\n\n"</span>)</span><br><span class="line">mark_dump_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; MarkDuplicates MAX_RECORDS_IN_RAM=2000000 \</span></span><br><span class="line"><span class="string">VALIDATION_STRINGENCY=SILENT M=&#123;&#125;/&#123;&#125;.matrix \</span></span><br><span class="line"><span class="string">CREATE_INDEX=true I=&#123;&#125;/&#123;&#125;.sorted.bam O=&#123;&#125;/&#123;&#125;.dedup.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,picard_path,prefix,prefix,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(mark_dump_cmd))</span><br><span class="line">subprocess.call(mark_dump_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: PICARD MARK DUPLICATE Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 6 ####### </span></span><br><span class="line"><span class="comment">#### Realign indel intervels</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 6 ... Realign the indel intervels \n\n\n"</span>)</span><br><span class="line">realign_prepare = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T RealignerTargetCreator \</span></span><br><span class="line"><span class="string">-R &#123;&#125; -I &#123;&#125;/&#123;&#125;.dedup.bam -known &#123;&#125; \</span></span><br><span class="line"><span class="string">-o &#123;&#125;/realignment_targets.list"</span>.format(cpu,temp_path,gatk_path,genome,prefix,prefix,known_indel,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(realign_prepare))</span><br><span class="line">subprocess.call(realign_prepare, shell=<span class="literal">True</span>)</span><br><span class="line">realign_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T IndelRealigner -R &#123;&#125; \</span></span><br><span class="line"><span class="string">-I &#123;&#125;/&#123;&#125;.dedup.bam -targetIntervals &#123;&#125;/realignment_targets.list \</span></span><br><span class="line"><span class="string">-known &#123;&#125; -o &#123;&#125;/&#123;&#125;.realign.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,genome,prefix,prefix,prefix,known_indel,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(realign_cmd))</span><br><span class="line">subprocess.call(realign_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">index_cmd = <span class="string">"samtools index &#123;&#125;/&#123;&#125;.realign.bam"</span>.format(prefix,prefix)</span><br><span class="line">subprocess.call(index_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: GATK REALIGN  Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 7 ####### </span></span><br><span class="line"><span class="comment">#### Recalibrate base quality scores</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 7 ... Recalibrate base quality scores using known SNPs \n\n\n"</span>)</span><br><span class="line">recalibrate_prepare = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T BaseRecalibrator \</span></span><br><span class="line"><span class="string">--preserve_qscores_less_than 6 \</span></span><br><span class="line"><span class="string">-nct &#123;&#125; -R &#123;&#125; -I &#123;&#125;/&#123;&#125;.realign.bam \</span></span><br><span class="line"><span class="string">-XL chrX -XL chrY -XL chrM \</span></span><br><span class="line"><span class="string">-knownSites &#123;&#125; -knownSites &#123;&#125; -knownSites &#123;&#125; \</span></span><br><span class="line"><span class="string">-o &#123;&#125;/recal_data.table"</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,cpu,genome,prefix,prefix,mills,known_indel,db_snp,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(recalibrate_prepare))</span><br><span class="line">subprocess.call(recalibrate_prepare, shell=<span class="literal">True</span>)</span><br><span class="line">recalibrate_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T PrintReads \</span></span><br><span class="line"><span class="string">--disable_indel_quals \</span></span><br><span class="line"><span class="string">--preserve_qscores_less_than 6 \</span></span><br><span class="line"><span class="string">-nct &#123;&#125; -R &#123;&#125; -I &#123;&#125;/&#123;&#125;.realign.bam \</span></span><br><span class="line"><span class="string">-SQQ 10 -SQQ 20 -SQQ 30 -rf BadCigar \</span></span><br><span class="line"><span class="string">-BQSR &#123;&#125;/recal_data.table -o &#123;&#125;/&#123;&#125;.BQSR.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,cpu,genome,prefix,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(recalibrate_cmd))</span><br><span class="line">subprocess.call(recalibrate_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">index_cmd = <span class="string">"samtools index &#123;&#125;/&#123;&#125;.BQSR.bam"</span>.format(prefix,prefix)</span><br><span class="line">subprocess.call(index_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: GATK BQSR Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 8 ####### </span></span><br><span class="line"><span class="comment">#### Raw variant calls using HaplotypeCaller on single sample</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 8 ... CALL SNP !!! \n\n\n"</span>)</span><br><span class="line">callsnp_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; \</span></span><br><span class="line"><span class="string">-T HaplotypeCaller --genotyping_mode DISCOVERY \</span></span><br><span class="line"><span class="string">-A AlleleBalanceBySample -A DepthPerAlleleBySample \</span></span><br><span class="line"><span class="string">-A DepthPerSampleHC -A InbreedingCoeff \</span></span><br><span class="line"><span class="string">-A MappingQualityZeroBySample -A StrandBiasBySample \</span></span><br><span class="line"><span class="string">-A Coverage -A FisherStrand -A HaplotypeScore \</span></span><br><span class="line"><span class="string">-A MappingQualityRankSumTest -A MappingQualityZero \</span></span><br><span class="line"><span class="string">-A QualByDepth -A RMSMappingQuality \</span></span><br><span class="line"><span class="string">-A ReadPosRankSumTest -A VariantType -l INFO \</span></span><br><span class="line"><span class="string">--emitRefConfidence GVCF -rf BadCigar \</span></span><br><span class="line"><span class="string">--variant_index_parameter 128000 --variant_index_type LINEAR \</span></span><br><span class="line"><span class="string">-R &#123;&#125; -nct 4 -I &#123;&#125;/&#123;&#125;.BQSR.bam -o &#123;&#125;/&#123;&#125;.g.vcf "</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,genome,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(callsnp_cmd))</span><br><span class="line">subprocess.call(callsnp_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: GATK HaplotypeCaller Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line">print(<span class="string">"\n\n\n\t\tALl process has been finished, if you want check how many time consume \ you could use commond below"</span>)</span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tgrep 'DATE STATUS:' $log_file"</span>)</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/12/29/bwa_picard_pipeline/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/22/zhuanluzu/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">perl ~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/run_DE_analysis.pl --matrix final_reads_counts_reformat.txt --method DESeq2 --samples_file sample.list</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/12/22/zhuanluzu/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/22/grab_PGG/" >抓取PG的G网站上的各省份的输入的rs号的等位基因频率</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#@author:shy</span></span><br><span class="line"><span class="comment"># This script is used to grab PGG.han rs info</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="keyword">from</span> bs4 <span class="keyword">import</span> BeautifulSoup</span><br><span class="line"><span class="keyword">import</span> re</span><br><span class="line"><span class="keyword">import</span> requests, json</span><br><span class="line"><span class="keyword">import</span> urllib.request, urllib.error</span><br><span class="line"><span class="keyword">import</span> urllib</span><br><span class="line"><span class="keyword">import</span> xlwt</span><br><span class="line"><span class="keyword">import</span> sqlite3</span><br><span class="line"><span class="keyword">import</span> argparse</span><br><span class="line"><span class="keyword">import</span> time </span><br><span class="line"></span><br><span class="line"></span><br><span class="line">FIND_id = re.compile(<span class="string">r'id=(.*?)"&gt;'</span>)</span><br><span class="line">FIND_ref_alt = re.compile(<span class="string">r'-(.*?)-(.*?)&lt;'</span>)</span><br><span class="line">FIND_af = re.compile(<span class="string">r'&lt;td class="text-center"&gt;0.(.*?)&lt;/td&gt;\n&lt;td class="text-center"&gt;'</span>)</span><br><span class="line">FIND_final = re.compile(<span class="string">r'&lt;td class="text-center"&gt;(.*?)&lt;/td&gt;'</span>)</span><br><span class="line">FIND_all = re.compile(<span class="string">r'&lt;td class="text-center"&gt;(.*?)&lt;/td&gt;'</span>)</span><br><span class="line"></span><br><span class="line">ap = argparse.ArgumentParser()</span><br><span class="line">ap.add_argument(<span class="string">"-q"</span>, <span class="string">"--query"</span>, required=<span class="literal">True</span>,</span><br><span class="line">	help=<span class="string">"Input the rs id"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-p"</span>, <span class="string">"--provience"</span>, required=<span class="literal">True</span>,</span><br><span class="line">	help=<span class="string">"Input the Provience"</span>)</span><br><span class="line"></span><br><span class="line">args = vars(ap.parse_args())</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">main</span><span class="params">()</span>:</span></span><br><span class="line">    baseurl  = <span class="string">"https://www.hanchinesegenomes.org/HCGD/data/variant?term=&#123;&#125;"</span>.format(args[<span class="string">"query"</span>])</span><br><span class="line">    getData(baseurl)</span><br><span class="line">    </span><br><span class="line"></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">getData</span><span class="params">(baseurl)</span>:</span></span><br><span class="line">    html = askURL(baseurl)</span><br><span class="line">    soup = BeautifulSoup(html,<span class="string">"html.parser"</span>)</span><br><span class="line">    <span class="keyword">for</span> item <span class="keyword">in</span> soup.find_all(<span class="string">'div'</span>,class_=<span class="string">"table-responsive m-b-15"</span>):</span><br><span class="line">        item = str(item)</span><br><span class="line">        <span class="keyword">try</span>:</span><br><span class="line">            p_id = re.findall(FIND_id,item)[<span class="number">0</span>]</span><br><span class="line">            baseurl  = <span class="string">"https://www.hanchinesegenomes.org/HCGD/data/variant/detail?id=&#123;&#125;"</span>.format(p_id)</span><br><span class="line">            getData2(baseurl)</span><br><span class="line">        <span class="keyword">except</span>:</span><br><span class="line">            print(<span class="string">"&#123;&#125; is not in PGG database ~"</span>.format(args[<span class="string">"query"</span>]))</span><br><span class="line"></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">getData2</span><span class="params">(baseurl)</span>:</span></span><br><span class="line">    html = askURL(baseurl)</span><br><span class="line">    Provience = []</span><br><span class="line">    <span class="keyword">with</span> open(args[<span class="string">"provience"</span>]) <span class="keyword">as</span> input_p:</span><br><span class="line">        <span class="keyword">for</span> row <span class="keyword">in</span> input_p:</span><br><span class="line">            Provience.append(row.strip())</span><br><span class="line">    soup = BeautifulSoup(html,<span class="string">"html.parser"</span>)</span><br><span class="line">    <span class="keyword">for</span> item <span class="keyword">in</span> soup.find_all(<span class="string">'table'</span>,class_=<span class="string">"table table-custom-s table-condensed table-nowrap m-b-0"</span>):</span><br><span class="line">        item = str(item)</span><br><span class="line">        <span class="keyword">try</span>:</span><br><span class="line">            p_ref_alt = re.findall(FIND_ref_alt,item)[<span class="number">0</span>]</span><br><span class="line">            ref_alt = args[<span class="string">"query"</span>] + <span class="string">'\t'</span>+p_ref_alt[<span class="number">1</span>].replace(<span class="string">'-'</span>,<span class="string">'-&gt;'</span>)</span><br><span class="line">            af = re.findall(FIND_af,item)</span><br><span class="line">            all = re.findall(FIND_all,item)</span><br><span class="line">        <span class="keyword">except</span>:</span><br><span class="line">            <span class="keyword">continue</span></span><br><span class="line">        m = <span class="number">0</span></span><br><span class="line">        <span class="keyword">for</span> i <span class="keyword">in</span> all:</span><br><span class="line">            <span class="keyword">if</span> i <span class="keyword">in</span> Provience:</span><br><span class="line">                print(ref_alt+<span class="string">'\t'</span>+i+<span class="string">'\t'</span>+<span class="string">"0."</span>+af[m])</span><br><span class="line">                m += <span class="number">1</span></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">askURL</span><span class="params">(url)</span>:</span></span><br><span class="line">    <span class="comment"># simulate browser head info </span></span><br><span class="line">    head = &#123;</span><br><span class="line">        <span class="string">"User-Agent"</span>:<span class="string">"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36"</span></span><br><span class="line">        &#125;</span><br><span class="line">    requset = urllib.request.Request(url,headers=head)</span><br><span class="line">    html = <span class="string">""</span></span><br><span class="line">    <span class="keyword">try</span>:</span><br><span class="line">        response = urllib.request.urlopen(requset)</span><br><span class="line">        html = response.read().decode(<span class="string">"utf-8"</span>)</span><br><span class="line">    <span class="keyword">except</span> urllib.error.HTTPError <span class="keyword">as</span> e:</span><br><span class="line">        <span class="keyword">if</span> hasattr(e,<span class="string">"code"</span>):</span><br><span class="line">            print(e.code)</span><br><span class="line">        <span class="keyword">if</span> hasattr(e,<span class="string">"reason"</span>):</span><br><span class="line">            print(e.reason)</span><br><span class="line">    <span class="keyword">return</span> html</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> __name__ == <span class="string">"__main__"</span>:</span><br><span class="line">    main()</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/12/22/grab_PGG/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/22/可爱的人的喜好/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>最喜欢的导演</p>
<blockquote>
<p>新房昭之</p>
</blockquote>
<p>最喜欢的声优</p>
<blockquote>
<p>神谷浩史</p>
</blockquote>
<p>最喜欢的画师</p>
<blockquote>
<p>羽海野千花</p>
</blockquote>

	
	</div>
  <a type="button" href="/2020/12/22/可爱的人的喜好/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/08/1K_genome/" >主要记录千人基因组数据分析</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-08  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h3 id="Part-1"><a href="#Part-1" class="headerlink" title="Part 1."></a>Part 1.</h3><h4 id="数据下载"><a href="#数据下载" class="headerlink" title="数据下载"></a>数据下载</h4><blockquote>
<p>感觉再做一些可以去迅雷入职了</p>
</blockquote>
<p>首先下载alignment的list   </p>
<p>eg. igsr_Northern\ Han\ in\ China\ (HGDP)_undefined.tsv</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 0.提取第一列保存为1_ftp_sites</span></span><br><span class="line">awk '&#123;print $1&#125;' igsr_Northern\ Han\ in\ China\ \(HGDP\)_undefined.tsv.tsv</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 1.因为使用ascp下载 所以只需要数据在ebi数据库中的位置就可以，保存为2_temp_id</span></span><br><span class="line">awk -F 'uk/' '&#123;print $2&#125;' 1_ftp_sites &gt; 2_temp_id</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 2.使用<span class="built_in">echo</span>生成命令，并记录在3_run，方便以后查看哪里错了</span></span><br><span class="line">less 2_temp_id | while read line; </span><br><span class="line">do </span><br><span class="line">echo "ascp  -v -k 1 -l 200m -i \</span><br><span class="line">/Users/Yi-Liangwei/Applications/Aspera\ Connect.app/Contents/Resources/asperaweb_id_dsa.openssh \</span><br><span class="line">-Tr -Q -P33001 -L- fasp-g1k@fasp.1000genomes.ebi.ac.uk:"$&#123;line&#125; .; </span><br><span class="line">done &gt; 3_run </span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 3.下 载</span></span><br><span class="line">nohup sh 3_run</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 4.对于下载完成的cram文件 执行call snp环节</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 在所有操作之前，如果没有下载基因组的索引文件</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 可以使用下面两个命令生成</span></span><br><span class="line">samtools faidx Homo_sapiens_assembly38.fasta </span><br><span class="line"></span><br><span class="line">java -Xmx2g -jar ~/software/picard.jar CreateSequenceDictionary R=Homo_sapiens_assembly38.fasta O=Homo_sapiens_assembly38.dict</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 然后还需要准备三个文件，第一个是bam文件的prefix（CHS.info），第二个是rs位点的染色体和位置，</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 第三个是根据第二个文件生成的bed文件rs_id.bed</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 格式如下  CHS.info eg.</span></span><br><span class="line">HG00597</span><br><span class="line">HG00591</span><br><span class="line">HG00609</span><br><span class="line"><span class="meta">#</span><span class="bash"> rs_id eg.</span></span><br><span class="line">rs12044597      1       1708801</span><br><span class="line">rs11185092      1       107886278</span><br><span class="line">rs10923724      1       119546842</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> rs_id.bed用下面的命令生成</span></span><br><span class="line">awk -v OFS='\t' '&#123;print "chr"$2,$3-1,$3+1,$1&#125;' rs_id   &gt; rs_id.bed</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 5.cram格式转bam格式去call snp</span></span><br><span class="line">less CHS.info | while read line; do </span><br><span class="line">samtools view  -T /data/gujiaqi/698_download/Homo_sapiens_assembly38.fasta $&#123;line&#125;.final.cram  -L rs_id.bed -M -b -o $&#123;line&#125;.m.1.bam; done</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 6.call snp and parse vcf file</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 这时候还需要生成一个只包含rs编号的文件，两次的输入文件不一样的</span></span><br><span class="line">python bam2vcf.py -p CHS.info -rs rs_id</span><br><span class="line">python parse_vcf.py -p CHS.info -rs rs_number</span><br><span class="line"><span class="meta">#</span><span class="bash"> 所有结果会写在rs.genotype 和 rs.frequency中</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用的call snp软件为bcftools，如果时间不紧的话可以用GATK，将bcftools那一行改成</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> java -XX:+UseParallelGC -XX:ParallelGCThreads=8 -Xmx10g -jar \</span></span><br><span class="line">~/software/GenomeAnalysisTK.jar -T HaplotypeCaller --genotyping_mode DISCOVERY -A \</span><br><span class="line">AlleleBalanceBySample -A DepthPerAlleleBySample -A DepthPerSampleHC \</span><br><span class="line">-A InbreedingCoeff -A MappingQualityZeroBySample -A StrandBiasBySample\</span><br><span class="line">-A Coverage -A FisherStrand -A HaplotypeScore -A MappingQualityRankSumTest \</span><br><span class="line">-A MappingQualityZero -A QualByDepth -A RMSMappingQuality -A ReadPosRankSumTest \</span><br><span class="line">-A VariantType -l INFO --emitRefConfidence GVCF \</span><br><span class="line">-rf BadCigar --variant_index_parameter 128000 --variant_index_type LINEAR \</span><br><span class="line">-R /data/gujiaqi/698_download/Homo_sapiens_assembly38.fasta -I HG00597.merged.bam\</span><br><span class="line">-o HG00597.merged.gatk.gvcf</span><br></pre></td></tr></table></figure>

<h3 id="Part-2"><a href="#Part-2" class="headerlink" title="Part 2."></a>Part 2.</h3><h5 id="对vcf进行计算前还需要准备一下"><a href="#对vcf进行计算前还需要准备一下" class="headerlink" title="对vcf进行计算前还需要准备一下"></a>对vcf进行计算前还需要准备一下</h5><figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 对filtered出来的vcf文件压缩并索引</span></span><br><span class="line">ls *vcf | <span class="keyword">while</span> read line; <span class="keyword">do</span> bgzip $&#123;line&#125; -@ <span class="number">8</span>;tabix $&#123;line&#125;.gz; done</span><br><span class="line"></span><br><span class="line"><span class="comment"># 合并为一个文件，这一步如果直接合并所有的话，顺序默认按照文件夹内从小到大的顺序</span></span><br><span class="line"><span class="comment"># 后续会再rehead，当然也可以一个Group一个Group的合并，然后再把合并的合并</span></span><br><span class="line">bcftools merge *gz -O z -o <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz</span><br><span class="line"></span><br><span class="line"><span class="comment"># 查看vcf是否包含1185个位点</span></span><br><span class="line">less <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz | awk <span class="string">'&#123;if ($0 !~ "#")print&#125;'</span> | wc -l</span><br><span class="line"><span class="comment"># 或者可以直接用plink把rs号给提取出来，结果会保存在 -o 指定的 rs_check.snplist中</span></span><br><span class="line"><span class="comment"># 再根据rs_check.snplist 和之前的rs_id看一下就可以了</span></span><br><span class="line">plink --write-snplist -vcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz -out rs_check</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 使用excel做一个all_sample.info 就把之前那些.info文件和在一起</span></span><br><span class="line"><span class="comment"># all_sample.info 内容如下</span></span><br><span class="line"><span class="comment"># HG00709	SouthernHan	698dataset	698dataset_SouthernHan_HG00709</span></span><br><span class="line"><span class="comment"># HGDP01296	NorthernHan	HGDP	HGDP_NorthernHan_HGDP01296</span></span><br><span class="line"><span class="comment"># HGDP00828	Japanese	HGDP	HGDP_Japanese_HGDP00828</span></span><br><span class="line"><span class="comment"># HGDP00774	Han in China	HGDP	HGDP_Han in China_HGDP00774</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 为了下一步 reheader通过bcftools提取所有样本的id</span></span><br><span class="line">bcftools query -l <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz ｜<span class="keyword">while</span> read line; <span class="keyword">do</span> grep $&#123;line&#125; all_sample.info ; done | awk -F <span class="string">'\t'</span> <span class="string">'&#123;print $4&#125;'</span> &gt; vcf.sample.reheader.list</span><br><span class="line">bcftools reheader <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz -s vcf.sample.reheader.list -o <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz</span><br></pre></td></tr></table></figure>

<h5 id="统计vcf文件中的sample和sites"><a href="#统计vcf文件中的sample和sites" class="headerlink" title="统计vcf文件中的sample和sites"></a>统计vcf文件中的sample和sites</h5><figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 统计vcf文件中的基本信息</span></span><br><span class="line"><span class="comment"># individual 的missing信息</span></span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --missing-indv  --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --missing-site --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --site-mean-depth  --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --site-quality --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --het --out vcf_check</span><br><span class="line"><span class="comment"># sites 的missing信息</span></span><br><span class="line"><span class="comment"># vcf_check.imiss vcf_check.lmiss vcf_check.ldepth.mean vcf_check.lqual  vcf_check.het</span></span><br><span class="line"><span class="comment"># 上面就是生成的结果</span></span><br><span class="line"><span class="comment"># awk '&#123;print $3&#125;' vcf_check.ldepth.mean | sort -g -r 可以看平均深度有位点depth的平均深度只有20左右</span></span><br><span class="line"><span class="comment"># awk -F '\t' '&#123;print $4&#125;'  vcf_check.imiss | sort | uniq -c 可以看多少样本有缺失位点，有30个样本都是有缺失的，但是最多也就只缺两个位点</span></span><br><span class="line"><span class="comment"># awk -F '\t' '&#123;print $5&#125;'  vcf_check.lmiss | sort | uniq -c 可以看每个位点有多少个缺失，有两个位点有10个样本以上的缺失</span></span><br><span class="line"><span class="comment"># 需要注意的是 vcf_check.lmiss 中会默认我们的物种是二倍体，所以Number of DATA的数量是sample*2 </span></span><br><span class="line"><span class="comment"># 同理在这个文件中的missing sites都会变为 2 倍</span></span><br></pre></td></tr></table></figure>

<p>#####使用beagle填充缺失的基因型</p>
<figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">java -Xmx64g -jar ~/software/beagle.<span class="number">18</span>May20.d20.jar gt=<span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  out=<span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.beagle impute=true</span><br></pre></td></tr></table></figure>

<p>#####population statistic</p>
<p>PCA</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 使用plink计算协方差matrix</span></span><br><span class="line">plink --vcf 698_HGDP.1185.reheader.beagle.vcf.gz --make-bed --double-id --pca  5 --out pca.5</span><br></pre></td></tr></table></figure>

<p>Structure</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 用admixture生成cluster matrix </span></span><br><span class="line">for i in &#123;2..5&#125;; do  admixture --cv pca.5.bed $i &gt; log$&#123;i&#125;.out; done</span><br><span class="line"><span class="meta">#</span><span class="bash"> 检查一下admixture的CV值，CV越小越好，但是这个只是一个推测，K还是需要自己判断</span></span><br><span class="line">grep "CV" *out | awk '&#123;print $3,$4&#125;' | sed -e 's/(//;s/)//;s/://'</span><br><span class="line">rename pca.5. structure. *Q</span><br></pre></td></tr></table></figure>

<p>Fst</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 生成四个group的id的统计文件</span></span><br><span class="line">bcftools query -l 698_HGDP.1185.reheader.beagle.vcf.gz | grep 'Sou' &gt; SouthernHan.info</span><br><span class="line">bcftools query -l 698_HGDP.1185.reheader.beagle.vcf.gz | grep 'ni' &gt; HaninChina.info</span><br><span class="line">bcftools query -l 698_HGDP.1185.reheader.beagle.vcf.gz | grep 'Jap' &gt; Japanese.info</span><br><span class="line">bcftools query -l 698_HGDP.1185.reheader.beagle.vcf.gz | grep 'NorthernHan' &gt; NorthernHan.info</span><br></pre></td></tr></table></figure>

<p>然后转到R里面画这两个图</p>
<p>需要用到的包 tidyverse ggsci ggplot2 gridExtra pophelper</p>
<p>结果会生成两个图 PCA.pdf 和 Structure.pdf</p>
	
	</div>
  <a type="button" href="/2020/12/08/1K_genome/#more" class="btn btn-default more">Read More</a>
</div>

		

		</div>

		<!-- pagination -->
		<div>
  		<center>
		<div class="pagination">

   
    
           <a type="button" class="btn btn-default disabled"><i class="fa fa-arrow-circle-o-left"></i>Prev</a>
        

        <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
 
       <a href="/page/2/" type="button" class="btn btn-default ">Next<i class="fa fa-arrow-circle-o-right"></i></a>     
        

  
</div>

  		</center>
		</div>

		
		
	</div> <!-- col-md-9 -->

	
		<div class="col-md-3">
	<div id="sidebar">
	
			
  <div id="site_search">
   <div class="form-group">
    <input type="text" id="local-search-input" name="q" results="0" placeholder="Search" class="st-search-input st-default-search-input form-control"/>
   </div>  
  <div id="local-search-result"></div>
  </div>


		
			
		
			
	<div class="widget">
		<h4>Tag Cloud</h4>
		<ul class="tag_box inline list-unstyled">		
		
			<li><a href="/tags/Linux/">Linux<span>1</span></a></li>
		
			<li><a href="/tags/Python/">Python<span>1</span></a></li>
		
			<li><a href="/tags/Mysql-OrthoMCL/">Mysql, OrthoMCL<span>1</span></a></li>
		
			<li><a href="/tags/gene-duplication/">gene duplication<span>1</span></a></li>
		
			<li><a href="/tags/Markdown-maker/">Markdown, maker<span>2</span></a></li>
		
			<li><a href="/tags/grabber/">grabber<span>1</span></a></li>
		
			<li><a href="/tags/ncbi/">ncbi<span>1</span></a></li>
		
			<li><a href="/tags/Ortholog/">Ortholog<span>1</span></a></li>
		
			<li><a href="/tags/1000-Genomes/">1000 Genomes<span>1</span></a></li>
		
			<li><a href="/tags/HMM/">HMM<span>1</span></a></li>
		
			<li><a href="/tags/GiagaScience/">GiagaScience<span>1</span></a></li>
		
			<li><a href="/tags/NGS/">NGS<span>1</span></a></li>
		
			<li><a href="/tags/gene-family/">gene family<span>1</span></a></li>
		
			<li><a href="/tags/Ensembl/">Ensembl<span>2</span></a></li>
		
			<li><a href="/tags/KaKscalculator-R/">KaKscalculator,R<span>1</span></a></li>
		
			<li><a href="/tags/variants-calling/">variants calling<span>1</span></a></li>
		
			<li><a href="/tags/Phylogeny/">Phylogeny<span>1</span></a></li>
		
			<li><a href="/tags/OrthoFinder/">OrthoFinder<span>1</span></a></li>
		
			<li><a href="/tags/NCBI/">NCBI<span>1</span></a></li>
		
			<li><a href="/tags/ortholog/">ortholog<span>2</span></a></li>
		
		 
		</ul>
	</div>


		
			
<div class="widget">
  <h4>Recent Posts</h4>
  <ul class="entry list-unstyled">
    
      <li>
        <a href="/2021/04/26/GATK_WDL/" ><i class="fa fa-file-o"></i>GATK and WDL流处理</a>
      </li>
    
      <li>
        <a href="/2021/04/13/othmcl/" ><i class="fa fa-file-o"></i>orthomcl鉴定直系同源基因</a>
      </li>
    
      <li>
        <a href="/2021/04/06/Inparanoid/" ><i class="fa fa-file-o"></i>Inparanoid</a>
      </li>
    
      <li>
        <a href="/2021/03/26/Python 卡方检验/" ><i class="fa fa-file-o"></i></a>
      </li>
    
      <li>
        <a href="/2021/02/14/CV记录/" ><i class="fa fa-file-o"></i></a>
      </li>
    
  </ul>
</div>

		
			
<div class="widget">
	<h4>Links</h4>
	<ul class="blogroll list-unstyled">
	
		<li><i class="fa fa-github"></i><a href="https://github.com/goodlucksun" title="All Github repository." target="_blank"]);">My github</a></li>
	
	</ul>
</div>


		
	</div> <!-- sidebar -->
</div> <!-- col-md-3 -->

	
	
</div> <!-- row-fluid -->
	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2021 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
