<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 <div class="page-header logo">
  <h1>Keeping your word<span class="blink-fast">∎</span></h1>
</div>

<div class="row page">

	
	<div class="col-md-9">
	

		<div class="slogan">


		<i class="fa fa-heart blink-slow"></i>

		我们进化速度快到让你感动邪门？！

</div>    

		<div id="top_search"></div>
		<div class="mypage">
		
		<!-- title and entry -->
		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/04/22/Nature 配色风格pal_npg/" >ggsci自动生成适合学术期刊的配色</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-04-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>最近做了很多图被说不好看。想到ggsci可以自动给分组添加颜色，就想看看他可不可以把颜色的十六进制码输出出来，然后发现是有这个函数的。</p>
<p>ggsci提供以下颜色</p>
<blockquote>
<p>Name    Scales    Palette Types    Palette Generator</p>
<p>NPG    scale_color_npg() scale_fill_npg()    “nrc”    pal_npg()</p>
<p>AAAS    scale_color_aaas() scale_fill_aaas()    “default”    pal_aaas()</p>
<p>NEJM    scale_color_nejm() scale_fill_nejm()    “default”    pal_nejm()</p>
<p>Lancet    scale_color_lancet() scale_fill_lancet()    “lanonc”    pal_lancet()</p>
<p>JAMA    scale_color_jama() scale_fill_jama()    “default”    pal_jama()</p>
<p>JCO    scale_color_jco() scale_fill_jco()    “default”    pal_jco()</p>
<p>UCSCGB    scale_color_ucscgb() scale_fill_ucscgb()    “default”    pal_ucscgb()</p>
<p>D3    scale_color_d3()</p>
</blockquote>
	
	</div>
  <a type="button" href="/2020/04/22/Nature 配色风格pal_npg/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/03/09/identify/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-03-09  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">hmmsearch -o batatas.hmmsearch.log --tblout batatas.hmmsearch.tblout SBP.hmm  longest.renamehead.transcript.fasta.transdecoder.pep</span><br><span class="line"></span><br><span class="line">awk '&#123;if ($1!~/^#/)print $1&#125;' batatas.hmmsearch.tblout &gt; batatas.hmmsearch.seqid</span><br><span class="line"></span><br><span class="line">makeblastdb -in  longest.renamehead.transcript.fasta.transdecoder.pep -title batatas -dbtype prot -parse_seqids -out batatas</span><br><span class="line"></span><br><span class="line">blastp -db batatas -query atspl.pep.fa -evalue 1e-3 -outfmt 6 &gt; batatas.blastp</span><br><span class="line"></span><br><span class="line">awk '&#123;print $2&#125;' batatas.blastp | sort | uniq &gt; batatas.blastp.seqid</span><br><span class="line"></span><br><span class="line">cat batatas*seqid | sort | uniq &gt; batatas.spl.seqid</span><br><span class="line"></span><br><span class="line">perl extract.pl batatas.spl.seqid batatas.pep.fa &gt; batatas.spl.fa</span><br><span class="line"></span><br><span class="line">grep '&gt;' trifida.spl.long.fa &gt; trifida.spl.seqid</span><br><span class="line"></span><br><span class="line">cat batatas.smart.remove batatas.spl.seqid| sort | uniq -u &gt; batatas.smartSBP.seqid</span><br><span class="line"></span><br><span class="line">perl extract.pl batatas.smartSBP.seqid batatas.pep.fa &gt; batatas.spl.smart.fa</span><br><span class="line">	</span><br><span class="line">blastp -query batatas.spl.smart.fa -subject trifida.spl.smart.fa -outfmt 6 -max_target_seqs 1 &gt; b2t.log</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">less batatas.spl.seqid | while read line; do grep $&#123;line&#125; transcript.fasta.transdecoder.gff3 | awk -v OFS="\t" '&#123; if ($3 == "mRNA") print $1, $4, $5 &#125;' &gt;&gt; blast_result.bed; done</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/03/09/identify/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/03/09/文献记录/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-03-09  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="Genome-wide-identification-and-characterization-phylogenetic-comparison-and-expression-profiles-of-SPL-transcription-factor-family-in-B-juncea-Cruciferae"><a href="#Genome-wide-identification-and-characterization-phylogenetic-comparison-and-expression-profiles-of-SPL-transcription-factor-family-in-B-juncea-Cruciferae" class="headerlink" title="Genome-wide identification and characterization, phylogenetic comparison and expression profiles of SPL transcription factor family in B. juncea (Cruciferae)"></a>Genome-wide identification and characterization, phylogenetic comparison and expression profiles of <em>SPL</em> transcription factor family in <em>B</em>. <em>juncea</em> (Cruciferae)</h1><p>Sixteen SBP-LIKE (SPL) genes were grouped into eight clades following their conserved SBP domain in Arabidopsis [1]. Of those, 10 SPL genes belong to five clades are targeted by miR156 [2, 3], which might highly conserved and acted as important regulators in vegetative phase change in plants [4, 5].</p>
<p>此外，SPL基因在不同植物物种的多种植物发育过程中起着重要的调节作用，如植物营养相变[6号、雄性育性[7号、赤霉素生物合成[8号、植物结构[9号和植物对胁迫的响应[9号。然而，与其他植物物种相比，由于缺乏基因组信息，SPL基因的全基因组分析在灯心草中几乎没有。</p>

	
	</div>
  <a type="button" href="/2020/03/09/文献记录/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/27/寻找新的marker/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-27  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h3 id="寻找新的marker"><a href="#寻找新的marker" class="headerlink" title="寻找新的marker"></a>寻找新的marker</h3><p>大概意思就是要找到位于内含子区域的，比较保守的marker（类似CNEEs？）。考虑目前手头上有的数据，17个组装过了的<em>Prunella</em>的基因组，老板的意思是每个基因组文件makeblastdb，然后用<em>strophiata</em>的注释得到的intron去库里面blast，然后再对得到的intron划分500bp的window去看有哪些位置是特别特别保守的。</p>
<p>张老师的意思是，不管组装不组装，直接用bam去看，因为每个bam都是bwa mem到<em>strophiata</em>的，如果我们把每个bam的文件都按照<em>strophiata</em>的位置信息替换成相应序列，（其实我感觉应该先filter 掉indel），然后直接用<em>strophiata</em>的gff里面的intron位置去提出来，就能得到序列了。然后再去划分500bp去看 ，效果是一样的。我就是不太确定这么做的INDEL区域如果在intron和exon的边界会对结果有多大的差异。</p>
<p>但是我感觉 intron本身的进化速率就要高一些，在这个里面找conserve的不知道有什么生物学意义。</p>
<h3 id="流程"><a href="#流程" class="headerlink" title="流程"></a>流程</h3><p>####提取一致性序列，一定要使用很老版本的samtools，新版本只能通过vcf提取consensus序列</p>
<p>samtools mpileup -uf my_reference.fna my_file.bam | bcftools view -cg - | vcfutils.pl vcf2fq &gt; my_consensus.fq</p>
<p>fq2fa</p>
<p>####统计genome长度，划分region，统计差异度</p>
<p>这个就自己写脚本看就行</p>
<h5 id="统计差异度的脚本1。分为16个状态，从0到1"><a href="#统计差异度的脚本1。分为16个状态，从0到1" class="headerlink" title="统计差异度的脚本1。分为16个状态，从0到1"></a>统计差异度的脚本1。分为16个状态，从0到1</h5><p>得到的16个序列一起比较a[1] = b[1] …….h[1] : parttern =1</p>
<p>感觉这个是写起来最麻烦，因为每个位点要做16*15*14*13*12*11…..个判断，但是可以直接把图画出来，不过还是运算太大，放弃</p>
<h5 id="统计差异度的脚本2-。只统计pair-wise的差异度，再找所有pair-wise都为0或者1的，就是最保守的"><a href="#统计差异度的脚本2-。只统计pair-wise的差异度，再找所有pair-wise都为0或者1的，就是最保守的" class="headerlink" title="统计差异度的脚本2 。只统计pair-wise的差异度，再找所有pair-wise都为0或者1的，就是最保守的"></a>统计差异度的脚本2 。只统计pair-wise的差异度，再找所有pair-wise都为0或者1的，就是最保守的</h5><p> 统计差异度的脚本3</p>
<p>直接去alignment</p>
<p>![image-20191227161628708](/Users/sunhaoyun/Library/Application Support/typora-user-images/image-20191227161628708.png)</p>
<table>
<thead>
<tr>
<th>name</th>
<th>Both pairs mapped</th>
</tr>
</thead>
<tbody><tr>
<td>Prunella_collaris_S26913_per</td>
<td>95.6%</td>
</tr>
<tr>
<td>Mike</td>
<td>32</td>
</tr>
</tbody></table>

	
	</div>
  <a type="button" href="/2019/12/27/寻找新的marker/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/17/1217liuliang/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-17  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>1217</p>

	
	</div>
  <a type="button" href="/2019/12/17/1217liuliang/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/15/mysql -u root -p/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-15  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>mysql -u root -p</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"># 创建本地登录的账户</span><br><span class="line">create user &apos;mysql&apos;@&apos;localhost&apos; identified by &apos;111111&apos;;</span><br><span class="line"></span><br><span class="line"># 创建远程可登录的账户</span><br><span class="line">create user &apos;mysql&apos;@&apos;%&apos; identified by &apos;111111&apos;;</span><br><span class="line"></span><br><span class="line"># 用户创建完成后，刷新授权</span><br><span class="line">flush privileges;</span><br><span class="line"></span><br><span class="line"># 创建一个新的数据库</span><br><span class="line">create database orthomcl DEFAULT CHARSET utf8 COLLATE utf8_general_ci;</span><br><span class="line"></span><br><span class="line"># 授权</span><br><span class="line">GRANT ALL PRIVILEGES ON orthomcl.* TO &apos;orthomcl&apos;@&apos;localhost&apos;   IDENTIFIED BY &apos;111111&apos;;</span><br><span class="line">GRANT ALL PRIVILEGES ON orthomcl.* TO &apos;orthomcl&apos;@&apos;%&apos;   IDENTIFIED BY &apos;111111&apos;;</span><br><span class="line">flush privileges;</span><br><span class="line"></span><br><span class="line"># 查看结果</span><br><span class="line">show databases;</span><br></pre></td></tr></table></figure>

<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">perl scripts/orthomcl-setup-database.pl --user orthomcl --password 111111 --host 127.0.0.1 --database orthomcl --outfile configure_outfile.conf [--no-create-database]</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">perl scripts/orthomcl-pipeline.pl -i ../../data/orthomcl/ -o ../../data/orthomcl_out -m configure_outfile.conf --nocompliant</span><br></pre></td></tr></table></figure>

<p>Warning: directory “../../data/orthomcl_out” already exists, are you sure you want to store data here [Y]? Y</p>
<p>Starting OrthoMCL pipeline on: Sun Dec 15 11:12:16 2019</p>
<p>Git commit: 08bd2baf43db6e35afb3e22e95dce8a3f740729c</p>
<p>=Stage 1: Validate Files =</p>
<p>Validating Pcollaris_S26913_per_pep.fasta … 6276 sequences</p>
<p>Validating Pocularis_EAK479_pep.fasta … 6224 sequences</p>
<p>Validating Prubeculoides_QHH0839_pep.fasta … 6704 sequences</p>
<p>Validating Pimmaculata_XZ14285_pep.fasta … 5791 sequences</p>
<p>Validating Pkoslowi_26923_pep.fasta … 11783 sequences</p>
<p>Validating Pcollaris_PA20150614-Sichuan_pep.fasta … 8328 sequences</p>
<p>Validating Prubida_2S_63614_pep.fasta … 12875 sequences</p>
<p>Validating Pstrophita_assemble_pep.fasta … 20581 sequences</p>
<p>Validating Pcollaris_130562_Jon_Italy_pep.fasta … 6855 sequences</p>
<p>Validating Patrogularis_141230_pep.fasta … 10880 sequences</p>
<p>Validating Pmodularis_117541_pep.fasta … 8978 sequences</p>
<p>Validating Pmontanella_m80200_pep.fasta … 8534 sequences</p>
<p>Validating Pcollaris_S1318_taiwan_pep.fasta … 14461 sequences</p>
<p>Validating Pocularis_MR1290_pep.fasta … 6774 sequences</p>
<p>Validating Phimalayana_66660_pep.fasta … 6852 sequences</p>
<p>Validating Phimalayana_57883_pep.fasta … 7552 sequences</p>
<p>Validating Pfulvescens_QHH0845_pep.fasta … 9746 sequences</p>
<p>Validating Gallus_gallus_pep.fasta … 28444 sequences</p>
<p>Validated 18 files</p>
<p>Stage 1 took 0.17 minutes </p>
<p>=Stage 2: Validate Database=</p>
<p>Warning: some tables exist already in database dbi:mysql:orthomcl_new:127.0.0.1:mysql_local_infile, user=root, database_name=orthomcl_new. Do you want to remove (y/n)? y</p>
<p>Executing: ‘drop database orthomcl_new’</p>
<p>Executing: ‘create database orthomcl_new’</p>
<p>Successfully removed old database entries</p>
<p>Stage 2 took 0.02 minutes </p>
<p>=Stage 3: Load OrthoMCL Database Schema=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclInstallSchema “/home/sunhaoyun/software/orthomcl-pipeline/configure_outfile.conf” “/home/sunhaoyun/data/orthomcl_out/log/orthomclSchema.log” 1&gt;/home/sunhaoyun/data/orthomcl_out/log/3.loadschema.stdout.log 2&gt;/home/sunhaoyun/data/orthomcl_out/log/3.loadschema.stderr.log</p>
<p>Stage 3 took 0.00 minutes </p>
<p>=Stage 4: Adjust Fasta=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_S26913_per_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_S26913_per_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pocularis_EAK479_pep “/home/sunhaoyun/data/orthomcl/Pocularis_EAK479_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Prubeculoides_QHH0839_pep “/home/sunhaoyun/data/orthomcl/Prubeculoides_QHH0839_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pimmaculata_XZ14285_pep “/home/sunhaoyun/data/orthomcl/Pimmaculata_XZ14285_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pkoslowi_26923_pep “/home/sunhaoyun/data/orthomcl/Pkoslowi_26923_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_PA20150614-Sichuan_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_PA20150614-Sichuan_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Prubida_2S_63614_pep “/home/sunhaoyun/data/orthomcl/Prubida_2S_63614_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pstrophita_assemble_pep “/home/sunhaoyun/data/orthomcl/Pstrophita_assemble_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_130562_Jon_Italy_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_130562_Jon_Italy_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Patrogularis_141230_pep “/home/sunhaoyun/data/orthomcl/Patrogularis_141230_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pmodularis_117541_pep “/home/sunhaoyun/data/orthomcl/Pmodularis_117541_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pmontanella_m80200_pep “/home/sunhaoyun/data/orthomcl/Pmontanella_m80200_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_S1318_taiwan_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_S1318_taiwan_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pocularis_MR1290_pep “/home/sunhaoyun/data/orthomcl/Pocularis_MR1290_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Phimalayana_66660_pep “/home/sunhaoyun/data/orthomcl/Phimalayana_66660_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Phimalayana_57883_pep “/home/sunhaoyun/data/orthomcl/Phimalayana_57883_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pfulvescens_QHH0845_pep “/home/sunhaoyun/data/orthomcl/Pfulvescens_QHH0845_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Gallus_gallus_pep “/home/sunhaoyun/data/orthomcl/Gallus_gallus_pep.fasta” 1</p>
<p>Stage 4 took 0.08 minutes </p>
<p>=Stage 5: Filter Fasta=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclFilterFasta “/home/sunhaoyun/data/orthomcl_out/compliant_fasta” 10 20</p>
<p>Stage 5 took 0.12 minutes </p>
<p>=Stage 6: Split Fasta=</p>
<p>splitting /home/sunhaoyun/data/orthomcl_out/blast_dir/goodProteins.fasta into 4 pieces</p>
<p>jStage 6 took 0.42 minutes </p>
<p>=Stage 7: Format Database=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/blast-2.2.26/bin/formatdb -i “/home/sunhaoyun/data/orthomcl_out/blast_dir/goodProteins.fasta” -p “T” -l “/home/sunhaoyun/data/orthomcl_out/log/formatdb.log” 1&gt;/home/sunhaoyun/data/orthomcl_out/log/7.format-stdout.log 2&gt;/home/sunhaoyun/data/orthomcl_out/log/7.format-stderr.log</p>

	
	</div>
  <a type="button" href="/2019/12/15/mysql -u root -p/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/06/基因组组装评价/" >Genome assemble statistic</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-06  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>##基因组组装评价</p>
<p>真的有很多的脚本可以去算这个数，但是，每个软件算出来的数值都不太一样，有的差距很小，有的差距很大，还有把N50 L50搞反的，真的，五花八门，也没有查到文献中推荐哪个好，好像大家都是自己算自己的。</p>
<blockquote>
<blockquote>
<p>N50</p>
<blockquote>
<p>the <strong><em>N50</em></strong> is defined as the sequence length of the shortest contig at 50% of the total genome length.</p>
</blockquote>
<p>L50</p>
<blockquote>
<p>Given a set of contigs, each with its own length, the <strong><em>L50</em></strong> count is defined as the smallest number of contigs whose length sum makes up half of genome size. From the example above the L50=3.</p>
</blockquote>
<p>N90</p>
<blockquote>
<p>The <strong><em>N90 statistic</em></strong> is less than or equal to the <em>N50</em> statistic; it is the length for which the collection of all contigs of that length or longer contains at least 90% of the sum of the lengths of all contigs, </p>
</blockquote>
<p>NG50</p>
<blockquote>
<p>Note that <em>N50</em> is calculated in the context of the assembly size rather than the genome size</p>
</blockquote>
<p>D50</p>
<blockquote>
<p>The <em>D50</em> statistic is the lowest value <em>d</em> for which the sum of the lengths of the largest <em>d</em> lengths is at least 50% of the sum of all of the length</p>
</blockquote>
</blockquote>
<p>统计方法还有很多，参考<a href="https://en.wikipedia.org/wiki/N50%2C_L50%2C_and_related_statistics" target="_blank" rel="noopener">wiki</a></p>
</blockquote>
	
	</div>
  <a type="button" href="/2019/12/06/基因组组装评价/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/04/shimei/" >建一颗ML树</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-04  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h4 id="流程大概就是"><a href="#流程大概就是" class="headerlink" title="流程大概就是"></a>流程大概就是</h4><p>刘亮老师讲的那个差不多</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 安装mafft</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 下载mafft，因为mafft的安装包是.rpm结尾的压缩包，rpm包如果不用管理员安装的话会提示权限错误，但是咱这脑</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 子就别用管理员了，万一删了啥不该删的。rpm2cpio这个工具可以把rpm包在当前目录解压缩，使用方法为</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> rpm2cpio XXX.rpm | cpio -div 解压出来的结果会默认先放在usr里面</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> wget https://mafft.cbrc.jp/alignment/software/mafft-7.450-gcc_fc6.x86_64.rpm</span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 由于mac没有rpm2cpio, linux跳过brew install  rpm2cpio这句</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> brew install  rpm2cpio</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> rpm2cpio mafft-7.450-gcc_fc6.x86_64.rpm | cpio -div</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 解压完可以看到出现了一个usr目录，usr里的bin放的是执行文件，mafft要求所有所有执行文件都在环境变量里，</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 所以下一步先添加环境变量，就是快捷方式，首先要查看当前目录在哪儿,<span class="built_in">pwd</span> 获取当前目录,假如说现在目录 </span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 在/home/zhengyuno/softwares/mafft</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">cd</span> usr</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">cd</span> bin</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">pwd</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash">/home/zhengyuno/softwares/mafft/usr/bin</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> vim ~/.bash_profile</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 滚到最下面的空白处把<span class="built_in">pwd</span>的结果加上一些符号粘贴进去，vim编辑器输入时候，摁i是进入insert模式，输入完了摁</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> esc退出insert模式，然后摁 :wq 保存并退出 （左下角会显示状态）。输入的时候一定要记得前后的PATH要不然</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 就啥快捷方式都没了，别问我为啥知道，试过好多次了T.T</span></span><br><span class="line"></span><br><span class="line">export PATH=/home/zhengyuno/softwares/mafft/usr/bin:$PATH</span><br><span class="line"><span class="meta">#</span><span class="bash"> 输入mafft出来这个交互式页面就ok了</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> mafft</span></span><br><span class="line"></span><br><span class="line">---------------------------------------------------------------------</span><br><span class="line"></span><br><span class="line">   MAFFT v7.450 (2019/Aug/23)</span><br><span class="line"></span><br><span class="line">        MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)</span><br><span class="line">        https://mafft.cbrc.jp/alignment/software/</span><br><span class="line">---------------------------------------------------------------------</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">Input file? (FASTA format; Folder=/Users/sunhaoyun/temp_dir/usr/bin)</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> ctrl + c 中断当前程序，然后我们就不用交互式的模式了，直接用一行的<span class="built_in">command</span> line模式就行</span></span><br></pre></td></tr></table></figure>

<p>这里推荐看<a href="http://www.chenlianfu.com/?p=2214" target="_blank" rel="noopener">陈连福老师的博客</a> </p>
<p>以及<a href="https://mafft.cbrc.jp/alignment/software/manual/manual.html" target="_blank" rel="noopener">官方文档</a>，当然我这种比较懒的官方的manual都没看完过</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## mafft的使用 首先选一个算法，看一下算法有很多，有的适合差不多长的，有的适合比较近缘的，不会选，</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 用auto，看一下output选项，因为下一步过滤要fasta格式，他默认也是fasta,不用改</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> mafft --auto cds_ENSG00000000419.fasta &gt; cds419_mafft_out.fa</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">运行完了屏幕上的信息可以保存一下</span></span><br></pre></td></tr></table></figure>
	
	</div>
  <a type="button" href="/2019/12/04/shimei/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/10/01/re/" >李笑来老板的re part</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-10-01  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="正则表达式"><a href="#正则表达式" class="headerlink" title="正则表达式"></a>正则表达式</h1><p>正则表达式本质上是个独立的语言，短小却格外强悍 —— 乃至于，如果你竟然没学会它的话，你的之前学的编程技能干脆与残疾无异。</p>
<p>Wikipedia 上对正则表达式的说明如下：</p>
<blockquote>
<p><strong>正则表达式</strong>（英语：Regular Expression，在代码中常简写为 regex、regexp 或 RE），又称<em>正规表示式</em>、<em>正规表示法</em>、<em>正规运算式</em>、<em>规则运算式</em>、<em>常规表示法</em>，是计算机科学的一个概念。正则表达式使用单个字符串来描述、匹配一系列符合某个句法规则的字符串。在很多文本编辑器里，正则表达式通常被用来检索、替换那些符合某个模式的文本。许多程序设计语言都支持利用正则表达式进行字符串操作。例如，在 Perl 中就内建了一个功能强大的正则表达式引擎。正则表达式这个概念最初是由 Unix 中的工具软件（例如 sed 和 grep）普及开的。</p>
</blockquote>
	
	</div>
  <a type="button" href="/2019/10/01/re/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/09/12/SVDQUARTETS AND ASTRAL ACTIVITY/" >evomics 课程</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-09-12  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<blockquote>
<p><a href="http://evomics.org/learning/population-and-speciation-genomics/2018-population-and-speciation-genomics/svdquartets-astral-activity/" target="_blank" rel="noopener">http://evomics.org/learning/population-and-speciation-genomics/2018-population-and-speciation-genomics/svdquartets-astral-activity/</a></p>
</blockquote>
<p>下载实验数据</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">cd /Users/sunhaoyun/Downloads/sample_data</span><br><span class="line">wget phylosolutions.com/tutorials/wpsg2018/svdquartets_tutorial.zip</span><br><span class="line">cd svdquartets_tutorial</span><br></pre></td></tr></table></figure>

<p>从Liu和Edwards（2009）使用的一个例子开始，注意由于“不完全谱系分选”现象导致个别基因座的基因树与物种树发生冲突</p>
<p>首先用交互模式，对Liu的数据进行重复，其中-L参数为打开一个log文件储存所有输出结果</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">paup anomaly_zone.nex -L anomaly.log</span><br></pre></td></tr></table></figure>

<p>从串联最大似然分析开始，使用logdet距离进行nj分析</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">paup&gt; dset distance=logdet;</span><br><span class="line">paup&gt; nj;</span><br><span class="line">paup&gt; automodel;</span><br></pre></td></tr></table></figure>

<p>automodel将评估一组56个模型，选择最小化AIC的模型。 完成后，模型参数将设置为其最大似然估计值，我们可以将其作为最大似然搜索的起点，随后查看树</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">paup&gt; set criterion=likelihood;</span><br><span class="line">paup&gt; hsearch;</span><br><span class="line">paup&gt; describe/plot=p;</span><br></pre></td></tr></table></figure>

<p>记住这棵树的拓扑关系所暗示的分类群之间的关系，然后对对相同的数据进行svdquartets分析</p>
<p>可选参数：</p>
<p>quartets exhaustively (“evalquartets=all”)</p>
<p>sample quartets randomly (“evalquartets=random nquartets=n”)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">paup&gt; svdq evalq=all bootstrap;</span><br></pre></td></tr></table></figure>

<p>该树从理论是与所有5个推断的四重关系兼容，但串联分析的ML方法估计树不正确。 记下两个内部节点的引导比例，以便稍后讨论。</p>
<p>博主写了小脚本可以用以快速生成命令，其中！开头代表对外部命令行执行命令</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">paup&gt; !cat  astral_prep.py</span><br><span class="line">paup&gt; !python astral_prep.py &gt; run_astral.nex</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2019/09/12/SVDQUARTETS AND ASTRAL ACTIVITY/#more" class="btn btn-default more">Read More</a>
</div>

		

		</div>

		<!-- pagination -->
		<div>
  		<center>
		<div class="pagination">

   
    
           <a type="button" class="btn btn-default disabled"><i class="fa fa-arrow-circle-o-left"></i>Prev</a>
        

        <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
 
       <a href="/page/2/" type="button" class="btn btn-default ">Next<i class="fa fa-arrow-circle-o-right"></i></a>     
        

  
</div>

  		</center>
		</div>

		
		
	</div> <!-- col-md-9 -->

	
		<div class="col-md-3">
	<div id="sidebar">
	
			
  <div id="site_search">
   <div class="form-group">
    <input type="text" id="local-search-input" name="q" results="0" placeholder="Search" class="st-search-input st-default-search-input form-control"/>
   </div>  
  <div id="local-search-result"></div>
  </div>


		
			
		
			
	<div class="widget">
		<h4>Tag Cloud</h4>
		<ul class="tag_box inline list-unstyled">		
		
			<li><a href="/tags/evolution/">evolution<span>1</span></a></li>
		
			<li><a href="/tags/Markdown-maker/">Markdown, maker<span>2</span></a></li>
		
			<li><a href="/tags/geome/">geome<span>1</span></a></li>
		
			<li><a href="/tags/HMM/">HMM<span>1</span></a></li>
		
			<li><a href="/tags/Mac/">Mac<span>1</span></a></li>
		
			<li><a href="/tags/Phylogeny/">Phylogeny<span>1</span></a></li>
		
			<li><a href="/tags/Python/">Python<span>1</span></a></li>
		
		 
		</ul>
	</div>


		
			
<div class="widget">
  <h4>Recent Posts</h4>
  <ul class="entry list-unstyled">
    
      <li>
        <a href="/2020/04/22/Nature 配色风格pal_npg/" ><i class="fa fa-file-o"></i>ggsci自动生成适合学术期刊的配色</a>
      </li>
    
      <li>
        <a href="/2020/03/09/identify/" ><i class="fa fa-file-o"></i></a>
      </li>
    
      <li>
        <a href="/2020/03/09/文献记录/" ><i class="fa fa-file-o"></i></a>
      </li>
    
      <li>
        <a href="/2019/12/27/寻找新的marker/" ><i class="fa fa-file-o"></i></a>
      </li>
    
      <li>
        <a href="/2019/12/17/1217liuliang/" ><i class="fa fa-file-o"></i></a>
      </li>
    
  </ul>
</div>

		
			
<div class="widget">
	<h4>Links</h4>
	<ul class="blogroll list-unstyled">
	
		<li><i class="fa fa-github"></i><a href="https://github.com/goodlucksun" title="All Github repository." target="_blank"]);">My github</a></li>
	
	</ul>
</div>


		
	</div> <!-- sidebar -->
</div> <!-- col-md-3 -->

	
	
</div> <!-- row-fluid -->
	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2020 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
