<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 <div class="page-header logo">
  <h1>Keeping your word<span class="blink-fast">∎</span></h1>
</div>

<div class="row page">

	
	<div class="col-md-9">
	

		<div class="slogan">


		<i class="fa fa-heart blink-slow"></i>

		我们进化速度快到让你感动邪门？！

</div>    

		<div id="top_search"></div>
		<div class="mypage">
		
		<!-- title and entry -->
		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/22/详细注释的OrthoFinder/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="ortholog-identification"><a href="#ortholog-identification" class="headerlink" title="ortholog identification"></a>ortholog identification</h1><h2 id="filter-Prunella-strophiata"><a href="#filter-Prunella-strophiata" class="headerlink" title="filter Prunella strophiata"></a>filter Prunella strophiata</h2><h3 id="in-order-to-keep-gene-id-format-consistantly-use-awk-and-python-scripts-and-gffread-and-transeq-for-refomating"><a href="#in-order-to-keep-gene-id-format-consistantly-use-awk-and-python-scripts-and-gffread-and-transeq-for-refomating" class="headerlink" title="in order to keep gene id format consistantly, use awk and python scripts and gffread and transeq for refomating"></a>in order to keep gene id format consistantly, use awk and python scripts and gffread and transeq for refomating</h3><h3 id="1-gene-and-mRNA-will-cause-redirect-error-remove-first"><a href="#1-gene-and-mRNA-will-cause-redirect-error-remove-first" class="headerlink" title="1. gene and mRNA will cause redirect error, remove first"></a>1. gene and mRNA will cause redirect error, remove first</h3><h4 id="因为原始的Prunella-strophiata的注释文件中，gene-id-的格式如下"><a href="#因为原始的Prunella-strophiata的注释文件中，gene-id-的格式如下" class="headerlink" title="因为原始的Prunella strophiata的注释文件中，gene id 的格式如下"></a>因为原始的Prunella strophiata的注释文件中，gene id 的格式如下</h4><blockquote>
<p>contig1000      EVM     gene    23218   23970   .       +       .       ID=evm.TU.contig1000.2;Name=EVM%20prediction%20contig1000.2<br>contig1000      EVM     mRNA    23218   23970   .       +       .       ID=evm.model.contig1000.2;Parent=evm.TU.contig1000.2;Name=EVM%20predicti<br>on%20contig1000.2<br>contig1000      EVM     exon    23218   23970   .       +       .       ID=evm.model.contig1000.2.exon1;Parent=evm.model.contig1000.2<br>contig1000      EVM     CDS     23218   23970   .       +       0       ID=cds.evm.model.contig1000.2;Parent=evm.model.contig1000.2</p>
</blockquote>
<h4 id="但是为了在最终结果的-Orthogroups-中能清楚的看出来group中包含哪些物种，所以把注释文件中的-gene-id-替换为物种名加序号的格式"><a href="#但是为了在最终结果的-Orthogroups-中能清楚的看出来group中包含哪些物种，所以把注释文件中的-gene-id-替换为物种名加序号的格式" class="headerlink" title="但是为了在最终结果的 Orthogroups 中能清楚的看出来group中包含哪些物种，所以把注释文件中的 gene id 替换为物种名加序号的格式"></a>但是为了在最终结果的 Orthogroups 中能清楚的看出来group中包含哪些物种，所以把注释文件中的 gene id 替换为物种名加序号的格式</h4><h4 id="如Prunella-strophiata中的基因，需要修改命名为-Prunella-strophiata-best-assembled-1-Prunella-strophiata-best-assembled-2"><a href="#如Prunella-strophiata中的基因，需要修改命名为-Prunella-strophiata-best-assembled-1-Prunella-strophiata-best-assembled-2" class="headerlink" title="如Prunella strophiata中的基因，需要修改命名为 Prunella_strophiata_best_assembled_1, Prunella_strophiata_best_assembled_2"></a>如Prunella strophiata中的基因，需要修改命名为 Prunella_strophiata_best_assembled_1, Prunella_strophiata_best_assembled_2</h4><h4 id="因为gene和的信息内容是一致的，使用下面的命令先去除包含mrna的行"><a href="#因为gene和的信息内容是一致的，使用下面的命令先去除包含mrna的行" class="headerlink" title="因为gene和的信息内容是一致的，使用下面的命令先去除包含mrna的行"></a>因为gene和的信息内容是一致的，使用下面的命令先去除包含mrna的行</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">awk '&#123;if($3 != "mRNA")print $0&#125;' Prunella_strophiata_assemble.evm.gff3  &gt; Prunella_strophiata_assemble.remove_mRNA.gff3</span><br></pre></td></tr></table></figure>

<h3 id="2-replace-gene-id"><a href="#2-replace-gene-id" class="headerlink" title="2. replace gene id"></a>2. replace gene id</h3><h4 id="使用-reformat-py2-脚本去替换原始的gff3文件中的-gene-id"><a href="#使用-reformat-py2-脚本去替换原始的gff3文件中的-gene-id" class="headerlink" title="使用 reformat.py2 脚本去替换原始的gff3文件中的 gene id"></a>使用 reformat.py2 脚本去替换原始的gff3文件中的 gene id</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python reformat.py2 Prunella_strophiata_assemble.remove_mRNA.gff3 &gt;  Prunella_strophiata.best_assembled.gff3</span><br></pre></td></tr></table></figure>

<h4 id="修改后的注释文件如下-差别只在最后一列，这样输出还有一个目的是为了gffread软件的读取"><a href="#修改后的注释文件如下-差别只在最后一列，这样输出还有一个目的是为了gffread软件的读取" class="headerlink" title="修改后的注释文件如下,差别只在最后一列，这样输出还有一个目的是为了gffread软件的读取"></a>修改后的注释文件如下,差别只在最后一列，这样输出还有一个目的是为了gffread软件的读取</h4><blockquote>
<p>contig1000      EVM     gene    23218   23970   .       +       .       ID=Prunella_strophita_best_assembled_1<br>contig1000      EVM     exon    23218   23970   .       +       .       Parent=Prunella_strophita_best_assembled_1<br>contig1000      EVM     CDS     23218   23970   .       +       0       Parent=Prunella_strophita_best_assembled_1</p>
</blockquote>
<h3 id="3-extract-cds-sequence"><a href="#3-extract-cds-sequence" class="headerlink" title="3. extract cds sequence"></a>3. extract cds sequence</h3><h4 id="使用-gffread-软件，输入修改过的gff文件和基因组序列，提取出其中的cds序列，gffread在提取的时候会识别第七列的正负链信息取反向互补"><a href="#使用-gffread-软件，输入修改过的gff文件和基因组序列，提取出其中的cds序列，gffread在提取的时候会识别第七列的正负链信息取反向互补" class="headerlink" title="使用 gffread 软件，输入修改过的gff文件和基因组序列，提取出其中的cds序列，gffread在提取的时候会识别第七列的正负链信息取反向互补"></a>使用 gffread 软件，输入修改过的gff文件和基因组序列，提取出其中的cds序列，gffread在提取的时候会识别第七列的正负链信息取反向互补</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gffread Prunella_strophiata.best_assembled.gff3 -g Prunella_strophiata_assemble.fa -x Prunella_strophiata.best_assembled.cds.fa</span><br></pre></td></tr></table></figure>

<h3 id="4-cds2protein"><a href="#4-cds2protein" class="headerlink" title="4. cds2protein"></a>4. cds2protein</h3><h4 id="使用EMBOSS数据包中的-transeq-程序对上一步提取出来的cds序列进行翻译"><a href="#使用EMBOSS数据包中的-transeq-程序对上一步提取出来的cds序列进行翻译" class="headerlink" title="使用EMBOSS数据包中的 transeq 程序对上一步提取出来的cds序列进行翻译"></a>使用EMBOSS数据包中的 transeq 程序对上一步提取出来的cds序列进行翻译</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">transeq -sequence Prunella_strophiata.best_assembled.cds.fa -outseq Prunella_strophaita.best_assembled.pep.fa</span><br></pre></td></tr></table></figure>

<h3 id="5-remove-the-info-added-by-transeq"><a href="#5-remove-the-info-added-by-transeq" class="headerlink" title="5. remove the info added by transeq"></a>5. remove the info added by transeq</h3><h4 id="因为EMBOSS-中的-transeq-软件会对-gene-id-进行修改，将Prunella-strophiata-best-assembled-1改为了Prunella-strophiata-best-assembled-1-1"><a href="#因为EMBOSS-中的-transeq-软件会对-gene-id-进行修改，将Prunella-strophiata-best-assembled-1改为了Prunella-strophiata-best-assembled-1-1" class="headerlink" title="因为EMBOSS 中的 transeq 软件会对 gene id 进行修改，将Prunella_strophiata_best_assembled_1改为了Prunella_strophiata_best_assembled_1_1"></a>因为EMBOSS 中的 transeq 软件会对 gene id 进行修改，将Prunella_strophiata_best_assembled_1改为了Prunella_strophiata_best_assembled_1_1</h4><h4 id="所以使用sed命令去除掉gene-id末尾的-1-同时就得到了最终的蛋白文件"><a href="#所以使用sed命令去除掉gene-id末尾的-1-同时就得到了最终的蛋白文件" class="headerlink" title="所以使用sed命令去除掉gene id末尾的_1 同时就得到了最终的蛋白文件"></a>所以使用sed命令去除掉gene id末尾的_1 同时就得到了最终的蛋白文件</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">sed -i 's/\(.*\)_1\(.*\)/\1\2/g' Prunella_strophiata.best_assembled.pep.fa</span><br></pre></td></tr></table></figure>

<h2 id="filter-Du’s-file"><a href="#filter-Du’s-file" class="headerlink" title="filter Du’s file"></a>filter Du’s file</h2><h4 id="杜老师的16个岩鹨基因组的原始注释文件中即没有包含cds的信息，也没有包含gene-id的信息"><a href="#杜老师的16个岩鹨基因组的原始注释文件中即没有包含cds的信息，也没有包含gene-id的信息" class="headerlink" title="杜老师的16个岩鹨基因组的原始注释文件中即没有包含cds的信息，也没有包含gene id的信息"></a>杜老师的16个岩鹨基因组的原始注释文件中即没有包含cds的信息，也没有包含gene id的信息</h4><h4 id="原始注释如下"><a href="#原始注释如下" class="headerlink" title="原始注释如下"></a>原始注释如下</h4><blockquote>
<p>C26941125       exonerate:protein2genome:local  gene    1346    2626    1492    -       .       gene_id 1 ; sequence evm.model.contig1003.2 ; gene_orientation + ; identity 99.30 ; similarity 99.30<br>C26941125       exonerate:protein2genome:local  exon    2490    2626    .       -       .       insertions 0 ; deletions 0 ; identity 95.56 ; similarity 95.56<br>C26941125       exonerate:protein2genome:local  intron  2378    2489    .       -       .       intron_id 1<br>C26941125       exonerate:protein2genome:local  exon    2235    2377    .       -       .       insertions 0 ; deletions 1 ; identity 100.00 ; similarity 100.00 ; frameshifts 2<br>C26941125       exonerate:protein2genome:local  intron  2148    2234    .       -       .       intron_id 2<br>C26941125       exonerate:protein2genome:local  exon    2116    2147    .       -       .       insertions 0 ; deletions 0 ; identity 100.00 ; similarity 100.00<br>C26941125       exonerate:protein2genome:local  intron  1897    2115    .       -       .       intron_id 3<br>C26941125       exonerate:protein2genome:local  exon    1346    1896    .       -       .       insertions 0 ; deletions 0 ; identity 100.00 ; similarity 100.00<br>scaffold175008  exonerate:protein2genome:local  gene    1423    2438    1202    +       .       gene_id 1 ; sequence evm.model.contig1003.1 ; gene_orientation + ; identity 89.45 ; similarity 89.84<br>scaffold175008  exonerate:protein2genome:local  exon    1423    2014    .       +       .       insertions 0 ; deletions 0 ; identity 86.80 ; similarity 86.80<br>scaffold175008  exonerate:protein2genome:local  intron  2015    2259    .       +       .       intron_id 1<br>scaffold175008  exonerate:protein2genome:local  exon    2260    2438    .       +       .       insertions 0 ; deletions 0 ; identity 98.31 ; similarity 100.00</p>
</blockquote>
<h4 id="所以首先要做的是先添加gene-id的信息，然后再根据注释文件中的exon的位置信息，提取出所有转录本"><a href="#所以首先要做的是先添加gene-id的信息，然后再根据注释文件中的exon的位置信息，提取出所有转录本" class="headerlink" title="所以首先要做的是先添加gene id的信息，然后再根据注释文件中的exon的位置信息，提取出所有转录本"></a>所以首先要做的是先添加gene id的信息，然后再根据注释文件中的exon的位置信息，提取出所有转录本</h4><h4 id="再用transdecoder软件，对所有的转录本去预测ORF，得到最长ORF"><a href="#再用transdecoder软件，对所有的转录本去预测ORF，得到最长ORF" class="headerlink" title="再用transdecoder软件，对所有的转录本去预测ORF，得到最长ORF"></a>再用transdecoder软件，对所有的转录本去预测ORF，得到最长ORF</h4><h3 id="1-get-cdna-file-and-transdecoder-commands"><a href="#1-get-cdna-file-and-transdecoder-commands" class="headerlink" title="1. get cdna file and transdecoder commands"></a>1. get cdna file and transdecoder commands</h3><h3 id="This-step-aims-at-reformat-16-genome-annotation-of-Du-by-adding-gene-id"><a href="#This-step-aims-at-reformat-16-genome-annotation-of-Du-by-adding-gene-id" class="headerlink" title="This step aims at reformat 16 genome annotation of Du by adding gene id"></a>This step aims at reformat 16 genome annotation of Du by adding gene id</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">ls *gff3 | sed 's/.gff3//g' | while read line;</span><br><span class="line">do python reformat.py $&#123;line&#125;.gff3  &gt; $&#123;line&#125;.refomat.gff3;</span><br><span class="line">gffread $&#123;line&#125;.refomat.gff3 -g $&#123;line&#125;.fa  -w $&#123;line&#125;.cdna.fa;</span><br><span class="line">echo "TransDecoder.LongOrfs -t "$&#123;line&#125;".cdna.fa &amp;" &gt;&gt; run_TransDecoder.LongOrfs.sh</span><br><span class="line">echo "TransDecoder.Predict -t "$&#123;line&#125;".cdna.fa &amp;" &gt;&gt; run_TransDecoder.Predict.sh</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<h4 id="在上面的循环中，执行了4个循环的命令，在下面分四个说明"><a href="#在上面的循环中，执行了4个循环的命令，在下面分四个说明" class="headerlink" title="在上面的循环中，执行了4个循环的命令，在下面分四个说明"></a>在上面的循环中，执行了4个循环的命令，在下面分四个说明</h4><h4 id="1-使用reformat-py脚本修改gene-id的格式"><a href="#1-使用reformat-py脚本修改gene-id的格式" class="headerlink" title="1. 使用reformat.py脚本修改gene id的格式"></a>1. 使用reformat.py脚本修改gene id的格式</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python reformat.py $&#123;line&#125;.gff3  &gt; $&#123;line&#125;.refomat.gff3;</span><br></pre></td></tr></table></figure>

<h4 id="reformat-py-脚本如下，各部分代表的意义与上面的-reformat-py2-类似"><a href="#reformat-py-脚本如下，各部分代表的意义与上面的-reformat-py2-类似" class="headerlink" title="reformat.py 脚本如下，各部分代表的意义与上面的 reformat.py2 类似"></a>reformat.py 脚本如下，各部分代表的意义与上面的 reformat.py2 类似</h4><h4 id="修改后的gff文件格式如下，差别只在最后一列"><a href="#修改后的gff文件格式如下，差别只在最后一列" class="headerlink" title="修改后的gff文件格式如下，差别只在最后一列"></a>修改后的gff文件格式如下，差别只在最后一列</h4><blockquote>
<p>C26941125       exonerate:protein2genome:local  gene    1346    2626    1492    -       .       ID=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  exon    2490    2626    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  intron  2378    2489    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  exon    2235    2377    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  intron  2148    2234    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  exon    2116    2147    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  intron  1897    2115    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  exon    1346    1896    .       -       .       Parent=Prunella_atrogularis_141230_1</p>
</blockquote>
<h4 id="2-使用gffread提取exons组成的cdna序列"><a href="#2-使用gffread提取exons组成的cdna序列" class="headerlink" title="2.使用gffread提取exons组成的cdna序列"></a>2.使用gffread提取exons组成的cdna序列</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gffread $&#123;line&#125;.refomat.gff3 -g $&#123;line&#125;.fa  -w $&#123;line&#125;.cdna.fa;</span><br></pre></td></tr></table></figure>

<h4 id="3-与-4-的命令的目的就是生成TransDecoder软件所诉需要的命令行，即软件名字-cdna序列名字"><a href="#3-与-4-的命令的目的就是生成TransDecoder软件所诉需要的命令行，即软件名字-cdna序列名字" class="headerlink" title="3. 与 4. 的命令的目的就是生成TransDecoder软件所诉需要的命令行，即软件名字+cdna序列名字"></a>3. 与 4. 的命令的目的就是生成TransDecoder软件所诉需要的命令行，即软件名字+cdna序列名字</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">echo &quot;TransDecoder.LongOrfs -t &quot;$&#123;line&#125;&quot;.cdna.fa &amp;&quot; &gt;&gt; run_TransDecoder.LongOrfs.sh</span><br><span class="line">echo &quot;TransDecoder.Predict -t &quot;$&#123;line&#125;&quot;.cdna.fa &amp;&quot; &gt;&gt; run_TransDecoder.Predict.sh</span><br></pre></td></tr></table></figure>

<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">### 2. use TransDecoder to predict ORF region</span><br><span class="line">#### 运行上一步保存的命令行，TransDecoder使用cdna序列预测cds序列</span><br><span class="line">nohup sh run_TransDecoder.LongOrfs.sh &gt; run_TransDecoder.LongOrfs.log</span><br><span class="line">nohup sh run_TransDecoder.Predict.sh &gt; run_TransDecoder.Predict.log</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">### 3. generate longest transcripts and remove duplicated sequence in pep file</span><br><span class="line">```shell</span><br><span class="line">ls *.transdecoder.pep | while read line</span><br><span class="line">do</span><br><span class="line">/usr/bin/perl retrive_longest_splicing_variant.pl $&#123;line&#125; longest.$&#123;line&#125;</span><br><span class="line">/usr/bin/perl rm_dup_fasta.pl longest.$&#123;line&#125; temp.longest.$&#123;line&#125;</span><br><span class="line">awk -F &apos;#&apos; &apos;&#123;if($0 ~&quot;^&gt;&quot;)&#123;print $1&#125;else&#123;print $0&#125;&#125;&apos; temp.$&#123;line&#125; &gt;   final.$&#123;line&#125;</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<h4 id="在上面的循环中，执行了3个循环的命令，在下面分三个说明"><a href="#在上面的循环中，执行了3个循环的命令，在下面分三个说明" class="headerlink" title="在上面的循环中，执行了3个循环的命令，在下面分三个说明"></a>在上面的循环中，执行了3个循环的命令，在下面分三个说明</h4><h4 id="1-对TransDecoder的结果中的蛋白文件进行提取最长转录本"><a href="#1-对TransDecoder的结果中的蛋白文件进行提取最长转录本" class="headerlink" title="1. 对TransDecoder的结果中的蛋白文件进行提取最长转录本"></a>1. 对TransDecoder的结果中的蛋白文件进行提取最长转录本</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">/usr/bin/perl retrive_longest_splicing_variant.pl $&#123;line&#125; longest.$&#123;line&#125;</span><br></pre></td></tr></table></figure>

<h4 id="2-对TransDecoder的结果中的蛋白文件进行去重复"><a href="#2-对TransDecoder的结果中的蛋白文件进行去重复" class="headerlink" title="2. 对TransDecoder的结果中的蛋白文件进行去重复"></a>2. 对TransDecoder的结果中的蛋白文件进行去重复</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">/usr/bin/perl rm_dup_fasta.pl longest.$&#123;line&#125; temp.longest.$&#123;line&#125;</span><br></pre></td></tr></table></figure>

<h4 id="3-对蛋白文件的gene-id进行修改，将冗余的部分去掉"><a href="#3-对蛋白文件的gene-id进行修改，将冗余的部分去掉" class="headerlink" title="3. 对蛋白文件的gene id进行修改，将冗余的部分去掉"></a>3. 对蛋白文件的gene id进行修改，将冗余的部分去掉</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">awk -F '#' '&#123;if($0 ~"^&gt;")&#123;print $1&#125;else&#123;print $0&#125;&#125;' temp.$&#123;line&#125; &gt;   final.$&#123;line&#125;</span><br></pre></td></tr></table></figure>

<h2 id="filter-BGI-file"><a href="#filter-BGI-file" class="headerlink" title="filter BGI file"></a>filter BGI file</h2><h3 id="因为BGI的gff注释文件中，已经有了原始的基因名，所以先提取出原始的基因名，再添加一个对应的基因名，再将原始的注释文件替换"><a href="#因为BGI的gff注释文件中，已经有了原始的基因名，所以先提取出原始的基因名，再添加一个对应的基因名，再将原始的注释文件替换" class="headerlink" title="因为BGI的gff注释文件中，已经有了原始的基因名，所以先提取出原始的基因名，再添加一个对应的基因名，再将原始的注释文件替换"></a>因为BGI的gff注释文件中，已经有了原始的基因名，所以先提取出原始的基因名，再添加一个对应的基因名，再将原始的注释文件替换</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">ls *BGI*  | while read line</span><br><span class="line">do</span><br><span class="line">grep '&gt;' $&#123;line&#125;.pep | sed 's/&gt;//g' &gt; $&#123;line&#125;.modify.seqid</span><br><span class="line">awk '&#123;if($3 =="gene")&#123;a=index($0,"~~");b=index($0,";");print substr($0,a+2,b-a-2)&#125;&#125;' &gt; $&#123;line&#125;.origin.seqid</span><br><span class="line">paste -d '\t' $&#123;line&#125;.origin.seqid $&#123;line&#125;.modify.seqid &gt; $&#123;line&#125;_coress.seqid</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line">python  replace_gff_gene_id.py Prunella_fulvescens_BGI_coress.seqid Prunella_fulvescens.homolog.gff &gt; Prunella_fulvescens.modify.gff3</span><br><span class="line">python  replace_gff_gene_id.py Prunella_himalayana_BGI_coress.seqid Prunella_himalayana.homolog.gff &gt;  Prunella_himalayana.modify.gff3</span><br></pre></td></tr></table></figure>

<h2 id="download-Gallus-genome-from-Ensembl-database-and-remove-reduant-transcripts"><a href="#download-Gallus-genome-from-Ensembl-database-and-remove-reduant-transcripts" class="headerlink" title="download Gallus genome from Ensembl database and remove reduant transcripts"></a>download Gallus genome from Ensembl database and remove reduant transcripts</h2><h3 id="1-下载人和鸡的蛋白序列"><a href="#1-下载人和鸡的蛋白序列" class="headerlink" title="1. 下载人和鸡的蛋白序列"></a>1. 下载人和鸡的蛋白序列</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.ensembl.org/pub/release-100/fasta/gallus_gallus/pep/Gallus_gallus.GRCg6a.pep.all.fa.gz</span><br><span class="line">gunzip Gallus_gallus.GRCg6a.pep.all.fa.gz</span><br><span class="line"></span><br><span class="line">wget ftp://ftp.ensembl.org/pub/release-100/fasta/homo_sapiens/pep/Homo_sapiens.GRCh38.pep.all.fa.gz</span><br><span class="line">gunzip Homo_sapiens.GRCh38.pep.all.fa.gz</span><br></pre></td></tr></table></figure>

<h3 id="2-过滤掉同一gene-id中短的mrna序列"><a href="#2-过滤掉同一gene-id中短的mrna序列" class="headerlink" title="2. 过滤掉同一gene id中短的mrna序列"></a>2. 过滤掉同一gene id中短的mrna序列</h3><h3 id="retrive-longest-transcripts-from-above-protein-fasta-file"><a href="#retrive-longest-transcripts-from-above-protein-fasta-file" class="headerlink" title="retrive longest transcripts from above protein fasta file"></a>retrive longest transcripts from above protein fasta file</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">ls *fa | while read line</span><br><span class="line">do</span><br><span class="line">python ~/software/OrthoFinder/tools/primary_transcript.py $&#123;line&#125;</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<h1 id="CORE-PROGRAM"><a href="#CORE-PROGRAM" class="headerlink" title="CORE PROGRAM"></a>CORE PROGRAM</h1><h2 id="run-OrthoFinder-and-get-the-Orthogroups"><a href="#run-OrthoFinder-and-get-the-Orthogroups" class="headerlink" title="run OrthoFinder and get the Orthogroups"></a>run OrthoFinder and get the Orthogroups</h2><h3 id="将上述所有蛋白序列文件作为输入去预测Orthogroups"><a href="#将上述所有蛋白序列文件作为输入去预测Orthogroups" class="headerlink" title="将上述所有蛋白序列文件作为输入去预测Orthogroups"></a>将上述所有蛋白序列文件作为输入去预测Orthogroups</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">~/software/OrthoFinder/orthofinder -f pep -S diamond</span><br></pre></td></tr></table></figure>

<h2 id="published-info"><a href="#published-info" class="headerlink" title="published info"></a>published info</h2><h3 id="OrthoFinder发表在GB的文章，引用次数一年内已经有69次"><a href="#OrthoFinder发表在GB的文章，引用次数一年内已经有69次" class="headerlink" title="OrthoFinder发表在GB的文章，引用次数一年内已经有69次"></a>OrthoFinder发表在GB的文章，引用次数一年内已经有69次</h3><blockquote>
<p>Emms D.M. &amp; Kelly S. OrthoFinder: phylogenetic orthology inference for comparative<br>genomics (2019), Genome Biology 20:238</p>
</blockquote>
<h2 id="use-longest-transcripts-gene-ids-filter-gff-file-and-get-intron-sequences"><a href="#use-longest-transcripts-gene-ids-filter-gff-file-and-get-intron-sequences" class="headerlink" title="use longest transcripts gene ids filter gff file and get intron sequences"></a>use longest transcripts gene ids filter gff file and get intron sequences</h2><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">less *.seqid |sed 's/seqid//g'| while read line</span><br><span class="line">do cat $&#123;line&#125; | while read row</span><br><span class="line">grep -w $&#123;row&#125; $&#123;line&#125;.gff3</span><br><span class="line">done &gt; $&#123;line&#125;.reformat.gff3</span><br><span class="line">/usr/bin/perl extract_intron_from_gff3_based_on_exon.pl $&#123;line&#125;.fa $&#123;line&#125;.reformat.gff3 $&#123;line&#125;.intron.fa</span><br><span class="line">done</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/05/22/详细注释的OrthoFinder/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/15/KaKs_calculator/" >KaKs 计算进化速率</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-15  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>KaKs calculator</p>
<h1 id="首先这里还需要再运行一次pal2nal-nogap版本的"><a href="#首先这里还需要再运行一次pal2nal-nogap版本的" class="headerlink" title="首先这里还需要再运行一次pal2nal nogap版本的"></a>首先这里还需要再运行一次pal2nal nogap版本的</h1><h1 id="因为paml内设-clean-data-会将序列里的gap去掉"><a href="#因为paml内设-clean-data-会将序列里的gap去掉" class="headerlink" title="因为paml内设 clean data  会将序列里的gap去掉"></a>因为paml内设 clean data  会将序列里的gap去掉</h1><h1 id="但是KaKS-calculator他在运行的时候，如果序列碱基不是三的倍数会报错，序列中有gap也会报错"><a href="#但是KaKS-calculator他在运行的时候，如果序列碱基不是三的倍数会报错，序列中有gap也会报错" class="headerlink" title="但是KaKS_calculator他在运行的时候，如果序列碱基不是三的倍数会报错，序列中有gap也会报错"></a>但是KaKS_calculator他在运行的时候，如果序列碱基不是三的倍数会报错，序列中有gap也会报错</h1><h1 id="所以"><a href="#所以" class="headerlink" title="所以"></a>所以</h1><p>perl ~/Downloads/softwares/pal2nal.v14/pal2nal.pl 4fanshu.spl.muscle.pep.fa  4fanshu.spl_id.wid70.cds.fa –nogap &gt; t1</p>
<h1 id="将clustal格式文件改为AXT格式，即将每一对序列pairwise输出，因为一共有104个序列所以结果就是C-2-5356对"><a href="#将clustal格式文件改为AXT格式，即将每一对序列pairwise输出，因为一共有104个序列所以结果就是C-2-5356对" class="headerlink" title="将clustal格式文件改为AXT格式，即将每一对序列pairwise输出，因为一共有104个序列所以结果就是C 2  5356对"></a>将clustal格式文件改为AXT格式，即将每一对序列pairwise输出，因为一共有104个序列所以结果就是C 2  5356对</h1><pre><code>104</code></pre><p>AXTConvertor t1 t2 </p>
<h1 id="这里用的估算kaks的方法是平均法-Model-Averaging-on-a-set-of-candidate-models，在一篇frontiersin和BMC-genomics里发现都是这个方法，"><a href="#这里用的估算kaks的方法是平均法-Model-Averaging-on-a-set-of-candidate-models，在一篇frontiersin和BMC-genomics里发现都是这个方法，" class="headerlink" title="这里用的估算kaks的方法是平均法 Model Averaging on a set of candidate models，在一篇frontiersin和BMC genomics里发现都是这个方法，"></a>这里用的估算kaks的方法是平均法 Model Averaging on a set of candidate models，在一篇frontiersin和BMC genomics里发现都是这个方法，</h1><h1 id="但是在苹果的那个里面他用的是ML的方法，ml方法速度很快很快，同时omega值比MA方法的更为发散，但是在这两种方法的结果当中，都没有找到正选择的基因对"><a href="#但是在苹果的那个里面他用的是ML的方法，ml方法速度很快很快，同时omega值比MA方法的更为发散，但是在这两种方法的结果当中，都没有找到正选择的基因对" class="headerlink" title="但是在苹果的那个里面他用的是ML的方法，ml方法速度很快很快，同时omega值比MA方法的更为发散，但是在这两种方法的结果当中，都没有找到正选择的基因对"></a>但是在苹果的那个里面他用的是ML的方法，ml方法速度很快很快，同时omega值比MA方法的更为发散，但是在这两种方法的结果当中，都没有找到正选择的基因对</h1><p>KaKS_calculator -i t2 -o t3 -m MA<br>#</p>
<h1 id="整理过程-⬇️"><a href="#整理过程-⬇️" class="headerlink" title="整理过程 ⬇️"></a>整理过程 ⬇️</h1><h1 id="首先提取id-和-Ka-Ks"><a href="#首先提取id-和-Ka-Ks" class="headerlink" title="首先提取id 和 Ka/Ks"></a>首先提取id 和 Ka/Ks</h1><p>awk ‘{print $1,$5}’ t3  &gt; t3_filter</p>
<h1 id="将id列中的两个id分为两列"><a href="#将id列中的两个id分为两列" class="headerlink" title="将id列中的两个id分为两列"></a>将id列中的两个id分为两列</h1><p>awk -v OFS=’\t’ -F ‘&amp;’ ‘{print $1,$2}’ t3_filter &gt; t3_fltered</p>
<h1 id="用ggtree提取一下树文件中的-Tips-label-生成8个-tips-label文件"><a href="#用ggtree提取一下树文件中的-Tips-label-生成8个-tips-label文件" class="headerlink" title="用ggtree提取一下树文件中的 Tips label 生成8个 tips label文件"></a>用ggtree提取一下树文件中的 Tips label 生成8个 tips label文件</h1><pre><code>library(ggtree)
tree &lt;- read.newick(&quot;4fanshu.spl.muscle.gb.pep.meg.nwk&quot;)
tree_label &lt;- tree[[&quot;tip.label&quot;]]
clade1 &lt;- tree_label[1:31]
clade2 &lt;- tree_label[32:40]
clade3 &lt;- tree_label[41:46]
clade4 &lt;- tree_label[47:59]
clade5 &lt;- tree_label[60:75]
clade6 &lt;- tree_label[76:83]
clade7 &lt;- tree_label[84:87]
clade8 &lt;- tree_label[88:104]

write.table(tree_label,file=&quot;t4.label&quot;,
            quote = FALSE,row.names = FALSE,col.names = FALSE)

for(i  in 1:8){
  val_name = paste(&quot;clade&quot;,i,sep = &quot;&quot;)
  output_name = paste(&quot;t4.&quot;,val_name,&quot;.label&quot;,sep = &quot;&quot;)
  clade_id = eval(parse(text = paste(&quot;em &lt;- rbind(&quot;, val_name, &quot;)&quot;, sep = &quot;&quot;)))
  write.table(clade_id,file=output_name,sep = &apos;\n&apos;,
              quote = FALSE,row.names = FALSE,col.names = FALSE)
}

```R_script


# 对每一个label文件 都把对应的ka ks文件中的基因对提取出来，并且是两个基因都在的
ls *clade* | while read line
do
python sort_kaks.py  ${line} t3_filtered &gt; ${line}.t5
done

#对每一个clade生成对角矩阵</code></pre>
	
	</div>
  <a type="button" href="/2020/05/15/KaKs_calculator/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/15/PAML/" >PAML and MCScanX-transposed</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-15  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>通常师姐她们做的PAML选择压力检验 是为了分析一个基因，或者多个基因串联，构建的系统发育树中，收选择的物种是谁，在不同的物种当中是否具有不同的选择方向。</p>
<p>意思也就是在这篇文章中Molecular Evolution of the TET Gene Family in Mammals，TET有三个直系同源基因，然后把这三个基因对应的20个物种分别建树，先用sites model中的模型M0（假设比对中所有密码子位点都为一个ω）看每个基因的进化速率</p>
<p>Selection pressure causes differentiation of the SPL gene family in the Juglandaceae</p>
<p>这一篇用branch sites model的</p>
<p>branch sites model 是相对于集合了branch model和sites model的，因为他允许了在model中sites的omega值不同的同时也允许了branch的omega也不相同。即反馈到ctl文件中就是 Model =2 Nsite =2，代表了branch model中的foreground ： background 和Sites Model中的M2 model W0 = 0 ;W1 = 1; W2 &gt; 1</p>
<p>对于ModelA和ModelA null 的比较，Null 在检测中强制fix omega =1，就代表着backgroud的sites均处于negative selection,.foreground中的sites均处于neutral selection 而在modelA中 foreground 的sites全部处于positive selection </p>
<p>但是但是但是 PAML这个软件的目的就是检测正选择，而当输入的序列过于相似的时候，很少量的正选择的位点就很难反馈到模型上，十分容易造成正选择模型和凌假设模型差异不显著</p>
<p>NOTE:NG86 assumes no transition-transversion rate difference and no codon usage bias.</p>
<p>####1.WGD</p>
<p>Whole-genome duplications (WGDs) have occurred in thelineages of plants [1], animals [2,3] and fungi [4,5], with possibleconsequences including evolution of novel or modified genefunctions [6,7,8,9], and/or provision of ‘‘buffer capacity’’ [10,11]or  genetic  redundancy  that  increases  genetic  robustness[12,13,14,15,16,17].  Genome  duplication  may  also  increaseopportunities for nonreciprocal recombination [18,19,20], permit-ting or causing duplicated genes to evolve in concert for a period oftime. Rapid DNA loss and restructuring of low-copy DNA[21,22,23,24], retrotransposon activation [25,26,27] and epigeneticchanges [28,29,30,31,32,33] following WGD may further providematerials for evolutionary change</p>
<p>全基因组重复，基因组复制也可能增加不可逆重组的机会[18，19，20]，允许或导致复制的基因在一段时间内协同进化</p>
<p>####2.Tandem and Proximal</p>
<p>Tandem duplicates  are  consecutive  in  the  genome  while  proximal duplicates are near one another but separated by a few gene</p>
<p>These two gene duplication modes are presumed to arise throughunequal crossing over [36] or localized transposon activitie</p>
<p>####3.Dispersed</p>
<p>Dispersed duplicates are neither adjacent to each other in thegenome nor within homeologous chromosome segments </p>
<p>####4.Transposed</p>
<p>DNA  transposons  such  aspackmules  (rice)  [39],  helitrons  (maize)  [40],  and  CACTAelements (sorghum) [27] may relocate duplicated genes or genesegments to new chromosomal positions (referred to as DNA basedtransposed duplication). RNA based transposed duplication, oftenreferred to as retrotransposition, typically creates a single-exonretrocopy from a multi-exon parental gene, by reverse transcrip-tion of a spliced messenger RNA. It is presumed that the retrocopyduplicates only the transcribed sequence of the parental gene,detached from the parental promoter. The new retrogene is oftendeposited in a novel chromosomal environment with new (i.e. non-ancestral) neighboring genes and, having lost its native promoter,is only likely to survive as a functional gene if a new promoter is acquired [41,42]</p>
<p>NOTE！</p>
<p>A transposed duplicate pairs must be meet the following criteria: one gene existed in its ancestral locus, and the other was located in a non-ancestral locus (<a href="https://www.frontiersin.org/articles/10.3389/fpls.2018.00161/full#B87" target="_blank" rel="noopener">Wang Y. et al., 2011</a>).</p>
<p>NOTE！</p>
<p>If a pair of transposed duplicated genes comprised an ancestral gene with more than two exons and a novel transposed copy without an intron, then this pair was inferred to be derived from RNA-based transposition (retro transposition).</p>

	
	</div>
  <a type="button" href="/2020/05/15/PAML/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/15/OrthoFinde/" >OrthoFinder 寻找同源基因</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-15  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="ortholog-identification"><a href="#ortholog-identification" class="headerlink" title="ortholog identification"></a>ortholog identification</h1><h2 id="in-order-to-keep-gene-id-format-consistantly-use-awk-and-python-and-gffread-and-transeq-for-refomating"><a href="#in-order-to-keep-gene-id-format-consistantly-use-awk-and-python-and-gffread-and-transeq-for-refomating" class="headerlink" title="in order to keep gene id format consistantly,use awk and python and gffread and transeq for refomating"></a>in order to keep gene id format consistantly,use awk and python and gffread and transeq for refomating</h2><h3 id="base-pipeline"><a href="#base-pipeline" class="headerlink" title="base pipeline"></a>base pipeline</h3><h3 id="1-gene-and-mRNA-will-cause-redirect-error-remove-first"><a href="#1-gene-and-mRNA-will-cause-redirect-error-remove-first" class="headerlink" title="1. gene and mRNA will cause redirect error, remove first"></a>1. gene and mRNA will cause redirect error, remove first</h3><p>awk ‘{if($3 != “mRNA”)print $0}’ Prunella_strophita_assemble.evm.gff3  &gt; Prunella_strophita_assemble.remove_mRNA.gff3</p>
<h3 id="2-replace-gene-id"><a href="#2-replace-gene-id" class="headerlink" title="2. replace gene id"></a>2. replace gene id</h3><p>python reformat.py2 Prunella_strophita_assemble.remove_mRNA.gff3 &gt;  Prunella_strophita.best_assembled.gff3</p>
<h3 id="3-extract-cds-sequence"><a href="#3-extract-cds-sequence" class="headerlink" title="3. extract cds sequence"></a>3. extract cds sequence</h3><p>gffread Prunella_strophita.best_assembled.gff3 -g Prunella_strophita_assemble.fa -x Prunella_strophita.best_assembled.cds.fa</p>
<h3 id="4-cds2protein"><a href="#4-cds2protein" class="headerlink" title="4. cds2protein"></a>4. cds2protein</h3><p>transeq -sequence Prunella_strophita.best_assembled.cds.fa -outseq Prunella_strophita.best_assembled.pep.fa</p>
<h3 id="5-remove-the-info-added-by-transeq"><a href="#5-remove-the-info-added-by-transeq" class="headerlink" title="5. remove the info added by transeq"></a>5. remove the info added by transeq</h3><p>sed -i ‘s/(.<em>)_1(.</em>)/\1\2/g’ Prunella_strophita.best_assembled.pep.fa</p>
<h2 id="shell-pipeline"><a href="#shell-pipeline" class="headerlink" title="shell pipeline"></a>shell pipeline</h2><p>ls *gff3 | sed ‘s/.gff3//g’ | while read line;<br>do python reformat.py ${line}.gff3  &gt; ${line}.refomat.gff3;<br>gffread ${line}.refomat.gff3 -g ${line}.fa  -w ${line}.cdna.fa;<br>echo “TransDecoder.LongOrfs -t “${line}”.cdna.fa &amp;” &gt;&gt; run_TransDecoder.LongOrfs.sh<br>echo “TransDecoder.Predict -t “${line}”.cdna.fa &amp;” &gt;&gt; run_TransDecoder.Predict.sh<br>done</p>
<h1 id="use-TransDecoder-to-predict-ORF-region"><a href="#use-TransDecoder-to-predict-ORF-region" class="headerlink" title="use TransDecoder to predict ORF region"></a>use TransDecoder to predict ORF region</h1><p>nohup sh run_TransDecoder.LongOrfs.sh &gt; run_TransDecoder.LongOrfs.log<br>nohup sh run_TransDecoder.Predict.sh &gt; run_TransDecoder.Predict.log</p>
<h1 id="generate-longest-transcripts-and-remove-duplicated-sequence-in-pep-file"><a href="#generate-longest-transcripts-and-remove-duplicated-sequence-in-pep-file" class="headerlink" title="generate longest transcripts and remove duplicated sequence in pep file"></a>generate longest transcripts and remove duplicated sequence in pep file</h1><p>ls *.transdecoder.pep | while read line<br>do<br>/usr/bin/perl retrive_longest_splicing_variant.pl ${line} longest.${line}<br>/usr/bin/perl rm_dup_fasta.pl longest.${line} temp.longest.${line}<br>awk -F ‘#’ ‘{if($0 ~”^&gt;”){print $1}else{print $0}}’ temp.${line} &gt;   final.${line}<br>done </p>
<h1 id="rename-file"><a href="#rename-file" class="headerlink" title="rename file"></a>rename file</h1><p>rename .cdna.fa.transdecoder.pep .pep.fa <em>.cdna.fa.transdecoder.pep<br>rename final.longest.P P final.longest.Prunella_</em></p>
<h2 id="download-Gallus-genome-from-Ensembl-database-and-remove-reduant-transcripts"><a href="#download-Gallus-genome-from-Ensembl-database-and-remove-reduant-transcripts" class="headerlink" title="download Gallus genome from Ensembl database and remove reduant transcripts"></a>download Gallus genome from Ensembl database and remove reduant transcripts</h2><p>wget <a href="ftp://ftp.ensembl.org/pub/release-100/fasta/gallus_gallus/pep/Gallus_gallus.GRCg6a.pep.all.fa.gz" target="_blank" rel="noopener">ftp://ftp.ensembl.org/pub/release-100/fasta/gallus_gallus/pep/Gallus_gallus.GRCg6a.pep.all.fa.gz</a><br>gunzip Gallus_gallus.GRCg6a.pep.all.fa.gz</p>
<p>python ~/software/OrthoFinder/tools/primary_transcript.py Gallus_gallus.GRCg6a.pep.all.fa<br>mv /home/sunhaoyun/data/Prunella_genome/primary_transcripts/Gallus_gallus.GRCg6a.pep.all.fa Gallus.pep.fa</p>
<h2 id="run-OrthoFinder"><a href="#run-OrthoFinder" class="headerlink" title="run OrthoFinder"></a>run OrthoFinder</h2><p>~/software/OrthoFinder/orthofinder -f pep -S diamond</p>
<h1 id="analyse-result"><a href="#analyse-result" class="headerlink" title="analyse result"></a>analyse result</h1><h3 id="1-overview-how-many-genes-were-assigned-to-orthogroups"><a href="#1-overview-how-many-genes-were-assigned-to-orthogroups" class="headerlink" title="1. overview how many genes were assigned to orthogroups"></a>1. overview how many genes were assigned to orthogroups</h3><h4 id="PATH"><a href="#PATH" class="headerlink" title="PATH"></a>PATH</h4><p>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/Prunella_sp19_with_Gallus/Comparative_Genomics_Statistics/Statistics_Overall.tsv</p>
<h3 id="2-check-the-percentages-on-a-per-species-basis-here"><a href="#2-check-the-percentages-on-a-per-species-basis-here" class="headerlink" title="2. check the percentages on a per species basis here"></a>2. check the percentages on a per species basis here</h3><h4 id="PATH-1"><a href="#PATH-1" class="headerlink" title="PATH"></a>PATH</h4><p>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/OrthoFinder/Prunella_sp19_with_Gallus/CComparative_Genomics_Statistics/Statistics_PerSpecies.tsv</p>
<h3 id="3-species-tree"><a href="#3-species-tree" class="headerlink" title="3. species tree"></a>3. species tree</h3><p>####PATH<br>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/OrthoFinder/Prunella_sp19_with_Gallus/Species_Tree/SpeciesTree_rooted.txt</p>
<h3 id="NOTE"><a href="#NOTE" class="headerlink" title="NOTE"></a>NOTE</h3><h3 id="This-speceis-tree-may-have-some-error"><a href="#This-speceis-tree-may-have-some-error" class="headerlink" title="This speceis tree may have some error"></a>This speceis tree may have some error</h3><h3 id="If-the-species-tree-is-not-correct-then-this-will-not-impact-the-orthogroup-inference"><a href="#If-the-species-tree-is-not-correct-then-this-will-not-impact-the-orthogroup-inference" class="headerlink" title="If the species tree is not correct then this will not impact the orthogroup inference,"></a>If the species tree is not correct then this will not impact the orthogroup inference,</h3><h3 id="but-it-might-affect-the-orthologue-inference-in-some-of-the-gene-trees-which-have-gene-duplication-events"><a href="#but-it-might-affect-the-orthologue-inference-in-some-of-the-gene-trees-which-have-gene-duplication-events" class="headerlink" title="but it might affect the orthologue inference in some of the gene trees which have gene duplication events."></a>but it might affect the orthologue inference in some of the gene trees which have gene duplication events.</h3><h3 id="4-find-out-Gallus-orthologues-are-in-the-other-speceis"><a href="#4-find-out-Gallus-orthologues-are-in-the-other-speceis" class="headerlink" title="4. find out Gallus orthologues are in  the other speceis"></a>4. find out Gallus orthologues are in  the other speceis</h3><h4 id="such-as-methyltransferase-like-21C-gene-METTL21C-plays-an-important-role-in-chicken-muscle-development"><a href="#such-as-methyltransferase-like-21C-gene-METTL21C-plays-an-important-role-in-chicken-muscle-development" class="headerlink" title="such as methyltransferase-like 21C gene (METTL21C) plays an important role in chicken muscle development"></a>such as methyltransferase-like 21C gene (METTL21C) plays an important role in chicken muscle development</h4><h4 id="GgMETTL21C1-corresponding-gene-id-is-ENSGALG00000001790-6"><a href="#GgMETTL21C1-corresponding-gene-id-is-ENSGALG00000001790-6" class="headerlink" title="GgMETTL21C1 corresponding gene id is ENSGALG00000001790.6"></a>GgMETTL21C1 corresponding gene id is ENSGALG00000001790.6</h4><h4 id="PATH-2"><a href="#PATH-2" class="headerlink" title="PATH"></a>PATH</h4><p>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/OrthoFinder/Prunella_sp19_with_Gallus/Orthologues/Orthologues_Gallus.pep/</p>
<p>ls *tsv | while read line; do grep ‘ENSGALG00000001790.6’ ${line}; done</p>
<h3 id="5-Orthogroups-comparisons-across-a-clade-of-species"><a href="#5-Orthogroups-comparisons-across-a-clade-of-species" class="headerlink" title="5. Orthogroups comparisons across a clade of species"></a>5. Orthogroups comparisons across a clade of species</h3><h4 id="overview-file-PATH"><a href="#overview-file-PATH" class="headerlink" title="overview file PATH"></a>overview file PATH</h4><p>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/OrthoFinder/Prunella_sp19_with_Gallus/Orthogroups/Orthogroups.tsv</p>
<h4 id="For-each-orthogroup-there-is-a-FASTA-file-in"><a href="#For-each-orthogroup-there-is-a-FASTA-file-in" class="headerlink" title="For each orthogroup there is a FASTA file in"></a>For each orthogroup there is a FASTA file in</h4><p>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/OrthoFinder/Prunella_sp19_with_Gallus/Orthogroup_Sequences</p>
<h4 id="which-contains-the-sequences-for-the-genes-in-that-orthogroup"><a href="#which-contains-the-sequences-for-the-genes-in-that-orthogroup" class="headerlink" title="which contains the sequences for the genes in that orthogroup."></a>which contains the sequences for the genes in that orthogroup.</h4><h2 id="Other-scipts"><a href="#Other-scipts" class="headerlink" title="Other scipts"></a>Other scipts</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"># reformat.py</span><br><span class="line">import sys</span><br><span class="line">i=1</span><br><span class="line">with open(sys.argv[1]) as input_gff:</span><br><span class="line">    sp_name = input_gff.name.replace(&apos;.gff3&apos;,&apos;&apos;)</span><br><span class="line">    for row in input_gff:</span><br><span class="line">        lst1 = row.split(&apos;\t&apos;)</span><br><span class="line">        if &quot;gene_id&quot; in lst1[8]:</span><br><span class="line">            lst1[8] = &quot;ID=&quot;+sp_name+&quot;_&quot;+str(i)</span><br><span class="line">            new_list=&apos;\t&apos;.join(lst1)</span><br><span class="line">            print(new_list)</span><br><span class="line">            i+=1</span><br><span class="line">        else:</span><br><span class="line">            lst1[8] = &quot;Parent=&quot;+sp_name+&quot;_&quot;+str(i-1)</span><br><span class="line">            new_list=&apos;\t&apos;.join(lst1)</span><br><span class="line">            print(new_list)</span><br><span class="line">```python</span><br></pre></td></tr></table></figure>

<p>#reformat.py2<br>import sys<br>i=1<br>with open(sys.argv[1]) as input_gff:<br>    sp_name = input_gff.name.replace(‘assemble.remove_mRNA.gff3’,’best_assembled’)<br>    for row in input_gff:<br>        lst1 = row.split(‘\t’)<br>        if “TU” in lst1[8]:<br>            lst1[8] = “ID=”+sp_name+”<em>“+str(i)<br>            new_list=’\t’.join(lst1)<br>            print(new_list)<br>            i+=1<br>        else:<br>            lst1[8] = “Parent=”+sp_name+”</em>“+str(i-1)<br>            new_list=’\t’.join(lst1)<br>            print(new_list)</p>
<pre><code class="python"></code></pre>

	
	</div>
  <a type="button" href="/2020/05/15/OrthoFinde/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/04/22/Nature 配色风格pal_npg/" >ggsci自动生成适合学术期刊的配色</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-04-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>最近做了很多图被说不好看。想到ggsci可以自动给分组添加颜色，就想看看他可不可以把颜色的十六进制码输出出来，然后发现是有这个函数的。</p>
<p>ggsci提供以下颜色</p>
<blockquote>
<p>Name    Scales    Palette Types    Palette Generator</p>
<p>NPG    scale_color_npg() scale_fill_npg()    “nrc”    pal_npg()</p>
<p>AAAS    scale_color_aaas() scale_fill_aaas()    “default”    pal_aaas()</p>
<p>NEJM    scale_color_nejm() scale_fill_nejm()    “default”    pal_nejm()</p>
<p>Lancet    scale_color_lancet() scale_fill_lancet()    “lanonc”    pal_lancet()</p>
<p>JAMA    scale_color_jama() scale_fill_jama()    “default”    pal_jama()</p>
<p>JCO    scale_color_jco() scale_fill_jco()    “default”    pal_jco()</p>
<p>UCSCGB    scale_color_ucscgb() scale_fill_ucscgb()    “default”    pal_ucscgb()</p>
<p>D3    scale_color_d3()</p>
</blockquote>
	
	</div>
  <a type="button" href="/2020/04/22/Nature 配色风格pal_npg/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/03/09/identify/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-03-09  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">hmmsearch -o batatas.hmmsearch.log --tblout batatas.hmmsearch.tblout SBP.hmm  longest.renamehead.transcript.fasta.transdecoder.pep</span><br><span class="line"></span><br><span class="line">awk '&#123;if ($1!~/^#/)print $1&#125;' batatas.hmmsearch.tblout &gt; batatas.hmmsearch.seqid</span><br><span class="line"></span><br><span class="line">makeblastdb -in  longest.renamehead.transcript.fasta.transdecoder.pep -title batatas -dbtype prot -parse_seqids -out batatas</span><br><span class="line"></span><br><span class="line">blastp -db batatas -query atspl.pep.fa -evalue 1e-3 -outfmt 6 &gt; batatas.blastp</span><br><span class="line"></span><br><span class="line">awk '&#123;print $2&#125;' batatas.blastp | sort | uniq &gt; batatas.blastp.seqid</span><br><span class="line"></span><br><span class="line">cat batatas*seqid | sort | uniq &gt; batatas.spl.seqid</span><br><span class="line"></span><br><span class="line">perl extract.pl batatas.spl.seqid batatas.pep.fa &gt; batatas.spl.fa</span><br><span class="line"></span><br><span class="line">grep '&gt;' trifida.spl.long.fa &gt; trifida.spl.seqid</span><br><span class="line"></span><br><span class="line">cat batatas.smart.remove batatas.spl.seqid| sort | uniq -u &gt; batatas.smartSBP.seqid</span><br><span class="line"></span><br><span class="line">perl extract.pl batatas.smartSBP.seqid batatas.pep.fa &gt; batatas.spl.smart.fa</span><br><span class="line">	</span><br><span class="line">blastp -query batatas.spl.smart.fa -subject trifida.spl.smart.fa -outfmt 6 -max_target_seqs 1 &gt; b2t.log</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">less batatas.spl.seqid | while read line; do grep $&#123;line&#125; transcript.fasta.transdecoder.gff3 | awk -v OFS="\t" '&#123; if ($3 == "mRNA") print $1, $4, $5 &#125;' &gt;&gt; blast_result.bed; done</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/03/09/identify/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/21/mysql/" >OrthoMCL进行同源基因比对</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-21  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="以root账户登陆mysql"><a href="#以root账户登陆mysql" class="headerlink" title="以root账户登陆mysql"></a>以root账户登陆mysql</h1><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">mysql -u root -p</span><br></pre></td></tr></table></figure>

<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"># 创建本地登录的账户</span><br><span class="line">create user &apos;mysql&apos;@&apos;localhost&apos; identified by &apos;111111&apos;;</span><br><span class="line"></span><br><span class="line"># 创建远程可登录的账户</span><br><span class="line">create user &apos;mysql&apos;@&apos;%&apos; identified by &apos;111111&apos;;</span><br><span class="line"></span><br><span class="line"># 用户创建完成后，刷新授权</span><br><span class="line">flush privileges;</span><br><span class="line"></span><br><span class="line"># 创建一个新的数据库</span><br><span class="line">create database orthomcl DEFAULT CHARSET utf8 COLLATE utf8_general_ci;</span><br><span class="line"></span><br><span class="line"># 授权</span><br><span class="line">GRANT ALL PRIVILEGES ON orthomcl.* TO &apos;orthomcl&apos;@&apos;localhost&apos;   IDENTIFIED BY &apos;111111&apos;;</span><br><span class="line">GRANT ALL PRIVILEGES ON orthomcl.* TO &apos;orthomcl&apos;@&apos;%&apos;   IDENTIFIED BY &apos;111111&apos;;</span><br><span class="line">flush privileges;</span><br><span class="line"></span><br><span class="line"># 查看结果</span><br><span class="line">show databases;</span><br></pre></td></tr></table></figure>

<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">perl scripts/orthomcl-setup-database.pl --user orthomcl --password 111111 --host 127.0.0.1 --database orthomcl --outfile configure_outfile.conf [--no-create-database]</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">perl scripts/orthomcl-pipeline.pl -i ../../data/orthomcl/ -o ../../data/orthomcl_out -m configure_outfile.conf --nocompliant</span><br></pre></td></tr></table></figure>

<p>Warning: directory “../../data/orthomcl_out” already exists, are you sure you want to store data here [Y]? Y</p>
<p>Starting OrthoMCL pipeline on: Sun Dec 15 11:12:16 2019</p>
<p>Git commit: 08bd2baf43db6e35afb3e22e95dce8a3f740729c</p>
<p>=Stage 1: Validate Files =</p>
<p>Validating Pcollaris_S26913_per_pep.fasta … 6276 sequences</p>
<p>Validating Pocularis_EAK479_pep.fasta … 6224 sequences</p>
<p>Validating Prubeculoides_QHH0839_pep.fasta … 6704 sequences</p>
<p>Validating Pimmaculata_XZ14285_pep.fasta … 5791 sequences</p>
<p>Validating Pkoslowi_26923_pep.fasta … 11783 sequences</p>
<p>Validating Pcollaris_PA20150614-Sichuan_pep.fasta … 8328 sequences</p>
<p>Validating Prubida_2S_63614_pep.fasta … 12875 sequences</p>
<p>Validating Pstrophita_assemble_pep.fasta … 20581 sequences</p>
<p>Validating Pcollaris_130562_Jon_Italy_pep.fasta … 6855 sequences</p>
<p>Validating Patrogularis_141230_pep.fasta … 10880 sequences</p>
<p>Validating Pmodularis_117541_pep.fasta … 8978 sequences</p>
<p>Validating Pmontanella_m80200_pep.fasta … 8534 sequences</p>
<p>Validating Pcollaris_S1318_taiwan_pep.fasta … 14461 sequences</p>
<p>Validating Pocularis_MR1290_pep.fasta … 6774 sequences</p>
<p>Validating Phimalayana_66660_pep.fasta … 6852 sequences</p>
<p>Validating Phimalayana_57883_pep.fasta … 7552 sequences</p>
<p>Validating Pfulvescens_QHH0845_pep.fasta … 9746 sequences</p>
<p>Validating Gallus_gallus_pep.fasta … 28444 sequences</p>
<p>Validated 18 files</p>
<p>Stage 1 took 0.17 minutes </p>
<p>=Stage 2: Validate Database=</p>
<p>Warning: some tables exist already in database dbi:mysql:orthomcl_new:127.0.0.1:mysql_local_infile, user=root, database_name=orthomcl_new. Do you want to remove (y/n)? y</p>
<p>Executing: ‘drop database orthomcl_new’</p>
<p>Executing: ‘create database orthomcl_new’</p>
<p>Successfully removed old database entries</p>
<p>Stage 2 took 0.02 minutes </p>
<p>=Stage 3: Load OrthoMCL Database Schema=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclInstallSchema “/home/sunhaoyun/software/orthomcl-pipeline/configure_outfile.conf” “/home/sunhaoyun/data/orthomcl_out/log/orthomclSchema.log” 1&gt;/home/sunhaoyun/data/orthomcl_out/log/3.loadschema.stdout.log 2&gt;/home/sunhaoyun/data/orthomcl_out/log/3.loadschema.stderr.log</p>
<p>Stage 3 took 0.00 minutes </p>
<p>=Stage 4: Adjust Fasta=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_S26913_per_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_S26913_per_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pocularis_EAK479_pep “/home/sunhaoyun/data/orthomcl/Pocularis_EAK479_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Prubeculoides_QHH0839_pep “/home/sunhaoyun/data/orthomcl/Prubeculoides_QHH0839_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pimmaculata_XZ14285_pep “/home/sunhaoyun/data/orthomcl/Pimmaculata_XZ14285_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pkoslowi_26923_pep “/home/sunhaoyun/data/orthomcl/Pkoslowi_26923_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_PA20150614-Sichuan_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_PA20150614-Sichuan_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Prubida_2S_63614_pep “/home/sunhaoyun/data/orthomcl/Prubida_2S_63614_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pstrophita_assemble_pep “/home/sunhaoyun/data/orthomcl/Pstrophita_assemble_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_130562_Jon_Italy_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_130562_Jon_Italy_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Patrogularis_141230_pep “/home/sunhaoyun/data/orthomcl/Patrogularis_141230_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pmodularis_117541_pep “/home/sunhaoyun/data/orthomcl/Pmodularis_117541_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pmontanella_m80200_pep “/home/sunhaoyun/data/orthomcl/Pmontanella_m80200_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_S1318_taiwan_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_S1318_taiwan_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pocularis_MR1290_pep “/home/sunhaoyun/data/orthomcl/Pocularis_MR1290_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Phimalayana_66660_pep “/home/sunhaoyun/data/orthomcl/Phimalayana_66660_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Phimalayana_57883_pep “/home/sunhaoyun/data/orthomcl/Phimalayana_57883_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pfulvescens_QHH0845_pep “/home/sunhaoyun/data/orthomcl/Pfulvescens_QHH0845_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Gallus_gallus_pep “/home/sunhaoyun/data/orthomcl/Gallus_gallus_pep.fasta” 1</p>
<p>Stage 4 took 0.08 minutes </p>
<p>=Stage 5: Filter Fasta=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclFilterFasta “/home/sunhaoyun/data/orthomcl_out/compliant_fasta” 10 20</p>
<p>Stage 5 took 0.12 minutes </p>
<p>=Stage 6: Split Fasta=</p>
<p>splitting /home/sunhaoyun/data/orthomcl_out/blast_dir/goodProteins.fasta into 4 pieces</p>
<p>jStage 6 took 0.42 minutes </p>
<p>=Stage 7: Format Database=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/blast-2.2.26/bin/formatdb -i “/home/sunhaoyun/data/orthomcl_out/blast_dir/goodProteins.fasta” -p “T” -l “/home/sunhaoyun/data/orthomcl_out/log/formatdb.log” 1&gt;/home/sunhaoyun/data/orthomcl_out/log/7.format-stdout.log 2&gt;/home/sunhaoyun/data/orthomcl_out/log/7.format-stderr.log</p>

	
	</div>
  <a type="button" href="/2019/12/21/mysql/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/06/基因组组装评价/" >Genome assemble statistic</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-06  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>##基因组组装评价</p>
<p>真的有很多的脚本可以去算这个数，但是，每个软件算出来的数值都不太一样，有的差距很小，有的差距很大，还有把N50 L50搞反的，真的，五花八门，也没有查到文献中推荐哪个好，好像大家都是自己算自己的。</p>
<blockquote>
<blockquote>
<p>N50</p>
<blockquote>
<p>the <strong><em>N50</em></strong> is defined as the sequence length of the shortest contig at 50% of the total genome length.</p>
</blockquote>
<p>L50</p>
<blockquote>
<p>Given a set of contigs, each with its own length, the <strong><em>L50</em></strong> count is defined as the smallest number of contigs whose length sum makes up half of genome size. From the example above the L50=3.</p>
</blockquote>
<p>N90</p>
<blockquote>
<p>The <strong><em>N90 statistic</em></strong> is less than or equal to the <em>N50</em> statistic; it is the length for which the collection of all contigs of that length or longer contains at least 90% of the sum of the lengths of all contigs, </p>
</blockquote>
<p>NG50</p>
<blockquote>
<p>Note that <em>N50</em> is calculated in the context of the assembly size rather than the genome size</p>
</blockquote>
<p>D50</p>
<blockquote>
<p>The <em>D50</em> statistic is the lowest value <em>d</em> for which the sum of the lengths of the largest <em>d</em> lengths is at least 50% of the sum of all of the length</p>
</blockquote>
</blockquote>
<p>统计方法还有很多，参考<a href="https://en.wikipedia.org/wiki/N50%2C_L50%2C_and_related_statistics" target="_blank" rel="noopener">wiki</a></p>
</blockquote>
	
	</div>
  <a type="button" href="/2019/12/06/基因组组装评价/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/04/shimei/" >建一颗ML树</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-04  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h4 id="流程大概就是"><a href="#流程大概就是" class="headerlink" title="流程大概就是"></a>流程大概就是</h4><p>刘亮老师讲的那个差不多</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 安装mafft</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 下载mafft，因为mafft的安装包是.rpm结尾的压缩包，rpm包如果不用管理员安装的话会提示权限错误，但是咱这脑</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 子就别用管理员了，万一删了啥不该删的。rpm2cpio这个工具可以把rpm包在当前目录解压缩，使用方法为</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> rpm2cpio XXX.rpm | cpio -div 解压出来的结果会默认先放在usr里面</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> wget https://mafft.cbrc.jp/alignment/software/mafft-7.450-gcc_fc6.x86_64.rpm</span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 由于mac没有rpm2cpio, linux跳过brew install  rpm2cpio这句</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> brew install  rpm2cpio</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> rpm2cpio mafft-7.450-gcc_fc6.x86_64.rpm | cpio -div</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 解压完可以看到出现了一个usr目录，usr里的bin放的是执行文件，mafft要求所有所有执行文件都在环境变量里，</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 所以下一步先添加环境变量，就是快捷方式，首先要查看当前目录在哪儿,<span class="built_in">pwd</span> 获取当前目录,假如说现在目录 </span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 在/home/zhengyuno/softwares/mafft</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">cd</span> usr</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">cd</span> bin</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">pwd</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash">/home/zhengyuno/softwares/mafft/usr/bin</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> vim ~/.bash_profile</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 滚到最下面的空白处把<span class="built_in">pwd</span>的结果加上一些符号粘贴进去，vim编辑器输入时候，摁i是进入insert模式，输入完了摁</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> esc退出insert模式，然后摁 :wq 保存并退出 （左下角会显示状态）。输入的时候一定要记得前后的PATH要不然</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 就啥快捷方式都没了，别问我为啥知道，试过好多次了T.T</span></span><br><span class="line"></span><br><span class="line">export PATH=/home/zhengyuno/softwares/mafft/usr/bin:$PATH</span><br><span class="line"><span class="meta">#</span><span class="bash"> 输入mafft出来这个交互式页面就ok了</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> mafft</span></span><br><span class="line"></span><br><span class="line">---------------------------------------------------------------------</span><br><span class="line"></span><br><span class="line">   MAFFT v7.450 (2019/Aug/23)</span><br><span class="line"></span><br><span class="line">        MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)</span><br><span class="line">        https://mafft.cbrc.jp/alignment/software/</span><br><span class="line">---------------------------------------------------------------------</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">Input file? (FASTA format; Folder=/Users/sunhaoyun/temp_dir/usr/bin)</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> ctrl + c 中断当前程序，然后我们就不用交互式的模式了，直接用一行的<span class="built_in">command</span> line模式就行</span></span><br></pre></td></tr></table></figure>

<p>这里推荐看<a href="http://www.chenlianfu.com/?p=2214" target="_blank" rel="noopener">陈连福老师的博客</a> </p>
<p>以及<a href="https://mafft.cbrc.jp/alignment/software/manual/manual.html" target="_blank" rel="noopener">官方文档</a>，当然我这种比较懒的官方的manual都没看完过</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## mafft的使用 首先选一个算法，看一下算法有很多，有的适合差不多长的，有的适合比较近缘的，不会选，</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 用auto，看一下output选项，因为下一步过滤要fasta格式，他默认也是fasta,不用改</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> mafft --auto cds_ENSG00000000419.fasta &gt; cds419_mafft_out.fa</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">运行完了屏幕上的信息可以保存一下</span></span><br></pre></td></tr></table></figure>
	
	</div>
  <a type="button" href="/2019/12/04/shimei/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/10/01/re/" >李笑来老板的re part</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-10-01  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="正则表达式"><a href="#正则表达式" class="headerlink" title="正则表达式"></a>正则表达式</h1><p>正则表达式本质上是个独立的语言，短小却格外强悍 —— 乃至于，如果你竟然没学会它的话，你的之前学的编程技能干脆与残疾无异。</p>
<p>Wikipedia 上对正则表达式的说明如下：</p>
<blockquote>
<p><strong>正则表达式</strong>（英语：Regular Expression，在代码中常简写为 regex、regexp 或 RE），又称<em>正规表示式</em>、<em>正规表示法</em>、<em>正规运算式</em>、<em>规则运算式</em>、<em>常规表示法</em>，是计算机科学的一个概念。正则表达式使用单个字符串来描述、匹配一系列符合某个句法规则的字符串。在很多文本编辑器里，正则表达式通常被用来检索、替换那些符合某个模式的文本。许多程序设计语言都支持利用正则表达式进行字符串操作。例如，在 Perl 中就内建了一个功能强大的正则表达式引擎。正则表达式这个概念最初是由 Unix 中的工具软件（例如 sed 和 grep）普及开的。</p>
</blockquote>
	
	</div>
  <a type="button" href="/2019/10/01/re/#more" class="btn btn-default more">Read More</a>
</div>

		

		</div>

		<!-- pagination -->
		<div>
  		<center>
		<div class="pagination">

   
    
           <a type="button" class="btn btn-default disabled"><i class="fa fa-arrow-circle-o-left"></i>Prev</a>
        

        <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
 
       <a href="/page/2/" type="button" class="btn btn-default ">Next<i class="fa fa-arrow-circle-o-right"></i></a>     
        

  
</div>

  		</center>
		</div>

		
		
	</div> <!-- col-md-9 -->

	
		<div class="col-md-3">
	<div id="sidebar">
	
			
  <div id="site_search">
   <div class="form-group">
    <input type="text" id="local-search-input" name="q" results="0" placeholder="Search" class="st-search-input st-default-search-input form-control"/>
   </div>  
  <div id="local-search-result"></div>
  </div>


		
			
		
			
	<div class="widget">
		<h4>Tag Cloud</h4>
		<ul class="tag_box inline list-unstyled">		
		
			<li><a href="/tags/OrthoFinder/">OrthoFinder<span>1</span></a></li>
		
			<li><a href="/tags/geome/">geome<span>1</span></a></li>
		
			<li><a href="/tags/KaKscalculator-R/">KaKscalculator,R<span>1</span></a></li>
		
			<li><a href="/tags/Phylogeny/">Phylogeny<span>1</span></a></li>
		
			<li><a href="/tags/Python/">Python<span>1</span></a></li>
		
			<li><a href="/tags/gene-duplication/">gene duplication<span>1</span></a></li>
		
			<li><a href="/tags/HMM/">HMM<span>1</span></a></li>
		
			<li><a href="/tags/Mysql-OrthoMCL/">Mysql, OrthoMCL<span>1</span></a></li>
		
			<li><a href="/tags/Markdown-maker/">Markdown, maker<span>2</span></a></li>
		
		 
		</ul>
	</div>


		
			
<div class="widget">
  <h4>Recent Posts</h4>
  <ul class="entry list-unstyled">
    
      <li>
        <a href="/2020/05/22/详细注释的OrthoFinder/" ><i class="fa fa-file-o"></i></a>
      </li>
    
      <li>
        <a href="/2020/05/15/KaKs_calculator/" ><i class="fa fa-file-o"></i>KaKs 计算进化速率</a>
      </li>
    
      <li>
        <a href="/2020/05/15/PAML/" ><i class="fa fa-file-o"></i>PAML and MCScanX-transposed</a>
      </li>
    
      <li>
        <a href="/2020/05/15/OrthoFinde/" ><i class="fa fa-file-o"></i>OrthoFinder 寻找同源基因</a>
      </li>
    
      <li>
        <a href="/2020/04/22/Nature 配色风格pal_npg/" ><i class="fa fa-file-o"></i>ggsci自动生成适合学术期刊的配色</a>
      </li>
    
  </ul>
</div>

		
			
<div class="widget">
	<h4>Links</h4>
	<ul class="blogroll list-unstyled">
	
		<li><i class="fa fa-github"></i><a href="https://github.com/goodlucksun" title="All Github repository." target="_blank"]);">My github</a></li>
	
	</ul>
</div>


		
	</div> <!-- sidebar -->
</div> <!-- col-md-3 -->

	
	
</div> <!-- row-fluid -->
	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2020 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
