<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 <div class="page-header logo">
  <h1>Keeping your word<span class="blink-fast">∎</span></h1>
</div>

<div class="row page">

	
	<div class="col-md-9">
	

		<div class="slogan">


		<i class="fa fa-heart blink-slow"></i>

		Déjà vu.

</div>    

		<div id="top_search"></div>
		<div class="mypage">
		
		<!-- title and entry -->
		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/08/29/HMM_example/" >HMM 隐式马尔可夫模型</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-08-29  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h2 id="Hidden-Markov-Model-隐含马尔可夫模型和-Viterbi-algorithm-维特比算法示例"><a href="#Hidden-Markov-Model-隐含马尔可夫模型和-Viterbi-algorithm-维特比算法示例" class="headerlink" title="Hidden Markov Model 隐含马尔可夫模型和 Viterbi algorithm 维特比算法示例"></a>Hidden Markov Model 隐含马尔可夫模型和 Viterbi algorithm 维特比算法示例</h2><p>基本概念：首先它得是一个统计模型，通过已有的样本来预测马尔可夫过程中的隐含参数。</p>
<p>可见状态链：观测得到的样本组成的数据集</p>
<p>隐含状态链：样本背后的隐含的数据集</p>
<p>转换概率 transition probability：可随意设定转换概率，根据不同的转换概率会得到不同的HMM模型</p>
<p>输出概率 emission probability：从隐含状态转换为可见状态时的概率，也可自由设定</p>
<p>提到马尔可夫链，基本上指的都是隐含状态链，因为在隐含状态链之间存在转换概率。我们在求马尔可夫过程的时候，可以分为以下情况：</p>
<p>​                知道<strong><em>隐含状态数量</em></strong>，知道<strong><em>转换概率</em></strong>，根据<strong><em>可见状态链</em></strong>，求<strong><em>隐含状态链</em></strong>。这是最简单的解码问题，有两种解决方法，两种都对，但结果的意义不一样。一种解法，通过最大似然直接对链的整体找到最可能的解，另一种通过对每一个单次事件概率。两种的差异个人觉得在于看中整体与局部。</p>
<p>​                知道<strong><em>隐含状态数量</em></strong>，知道<strong><em>转换概率</em></strong>，根据<strong><em>可见状态链</em></strong>，求<strong><em>发生可见状态链的概率</em></strong>。这种问题是用来检测数据是否真的与模型吻合。</p>
<p>​                知道<strong><em>隐含状态数量</em></strong>，不知道<strong><em>转换概率</em></strong>，根据<strong><em>可见状态链</em></strong>，求<strong><em>转换概率</em></strong>。这种为最常见的情况。此处用到Viterbi algorithm 算法。直接以下面的晴雨天为例子</p>
<blockquote>
<p> 说两句废话，答主认为呢，要了解一个算法，要做到以下两点：会其意，知其形。答主回答的，其实主要是第一点。但是这一点呢，恰恰是最重要，而且很多书上不会讲的。正如你在追一个姑娘，姑娘对你说“你什么都没做错！”你要是只看姑娘的表达形式呢，认为自己什么都没做错，显然就理解错了。你要理会姑娘的意思，“你赶紧给我道歉！”这样当你看到对应的表达形式呢，赶紧认错，跪地求饶就对了。数学也是一样，你要是不理解意思，光看公式，往往一头雾水。不过呢，数学的表达顶多也就是晦涩了点，姑娘的表达呢，有的时候就完全和本意相反。所以答主一直认为理解姑娘比理解数学难多了。</p>
</blockquote>
<p>HMM（隐马尔可夫模型）是用来描述隐含未知参数的统计模型，举一个经典的例子：一个东京的朋友每天根据天气{下雨，天晴}决定当天的活动{公园散步,购物,清理房间}中的一种，我每天只能在twitter上看到她发的推“啊，我前天公园散步、昨天购物、今天清理房间了！”，那么我可以根据她发的推特推断东京这三天的天气。在这个例子里，显状态是活动，隐状态是天气</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># -*- coding:utf-8 -*-</span></span><br><span class="line"><span class="comment"># Filename: viterbi.py</span></span><br><span class="line"><span class="comment"># Author：hankcs</span></span><br><span class="line"><span class="comment"># Date: 2014-05-13 下午8:51</span></span><br><span class="line"></span><br><span class="line">states = (<span class="string">'Rainy'</span>, <span class="string">'Sunny'</span>)</span><br><span class="line"></span><br><span class="line">observations = (<span class="string">'walk'</span>, <span class="string">'shop'</span>, <span class="string">'clean'</span>)</span><br><span class="line"></span><br><span class="line">start_probability = &#123;<span class="string">'Rainy'</span>: <span class="number">0.6</span>, <span class="string">'Sunny'</span>: <span class="number">0.4</span>&#125;</span><br><span class="line"></span><br><span class="line">transition_probability = &#123;</span><br><span class="line">    <span class="string">'Rainy'</span>: &#123;<span class="string">'Rainy'</span>: <span class="number">0.7</span>, <span class="string">'Sunny'</span>: <span class="number">0.3</span>&#125;,</span><br><span class="line">    <span class="string">'Sunny'</span>: &#123;<span class="string">'Rainy'</span>: <span class="number">0.4</span>, <span class="string">'Sunny'</span>: <span class="number">0.6</span>&#125;,</span><br><span class="line">    &#125;</span><br><span class="line"></span><br><span class="line">emission_probability = &#123;</span><br><span class="line">    <span class="string">'Rainy'</span>: &#123;<span class="string">'walk'</span>: <span class="number">0.1</span>, <span class="string">'shop'</span>: <span class="number">0.4</span>, <span class="string">'clean'</span>: <span class="number">0.5</span>&#125;,</span><br><span class="line">    <span class="string">'Sunny'</span>: &#123;<span class="string">'walk'</span>: <span class="number">0.6</span>, <span class="string">'shop'</span>: <span class="number">0.3</span>, <span class="string">'clean'</span>: <span class="number">0.1</span>&#125;,</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 打印路径概率表</span></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">print_dptable</span><span class="params">(V)</span>:</span></span><br><span class="line">    print(<span class="string">"    "</span>),</span><br><span class="line">    <span class="keyword">for</span> i <span class="keyword">in</span> range(len(V)):</span><br><span class="line">        print(<span class="string">"%7d"</span> % i),</span><br><span class="line">    <span class="keyword">print</span></span><br><span class="line"></span><br><span class="line">    <span class="keyword">for</span> y <span class="keyword">in</span> V[<span class="number">0</span>].keys():</span><br><span class="line">        print(<span class="string">"%.5s: "</span> % y),</span><br><span class="line">        <span class="keyword">for</span> t <span class="keyword">in</span> range(len(V)):</span><br><span class="line">            print(<span class="string">"%.7s"</span> % (<span class="string">"%f"</span> % V[t][y])),</span><br><span class="line">        <span class="keyword">print</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">viterbi</span><span class="params">(obs, states, start_p, trans_p, emit_p)</span>:</span></span><br><span class="line">    <span class="string">"""</span></span><br><span class="line"><span class="string">    # 任何一个HMM都可以通过下列五元组来描述</span></span><br><span class="line"><span class="string">    :param obs:观测序列</span></span><br><span class="line"><span class="string">    :param states:隐状态</span></span><br><span class="line"><span class="string">    :param start_p:初始概率（隐状态）</span></span><br><span class="line"><span class="string">    :param trans_p:转移概率（隐状态）</span></span><br><span class="line"><span class="string">    :param emit_p: 发射概率 （隐状态表现为显状态的概率）</span></span><br><span class="line"><span class="string">    :return:</span></span><br><span class="line"><span class="string">    """</span></span><br><span class="line">    <span class="comment"># 路径概率表 V[时间][隐状态] = 概率</span></span><br><span class="line">    V = [&#123;&#125;]</span><br><span class="line">    <span class="comment"># 一个中间变量，代表当前状态是哪个隐状态</span></span><br><span class="line">    path = &#123;&#125;</span><br><span class="line"></span><br><span class="line">    <span class="comment"># 初始化初始状态 (t == 0)</span></span><br><span class="line">    print(<span class="string">'Day 1(walk): '</span>)</span><br><span class="line">    <span class="keyword">for</span> y <span class="keyword">in</span> states:</span><br><span class="line">        V[<span class="number">0</span>][y] = start_p[y] * emit_p[y][obs[<span class="number">0</span>]]</span><br><span class="line">        path[y] = [y]</span><br><span class="line">        <span class="keyword">print</span> y, V[<span class="number">0</span>][y], <span class="string">'='</span>, start_p[y], <span class="string">'*'</span>, emit_p[y][obs[<span class="number">0</span>]]</span><br><span class="line">    print(<span class="string">'_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _'</span>)</span><br><span class="line">    <span class="keyword">print</span><span class="string">'V: '</span>, V</span><br><span class="line">    print(<span class="string">'__________________________________________'</span>)</span><br><span class="line">    <span class="comment"># 对 t &gt; 0 跑一遍维特比算法</span></span><br><span class="line">    <span class="keyword">for</span> t <span class="keyword">in</span> range(<span class="number">1</span>, len(obs)):</span><br><span class="line">        V.append(&#123;&#125;)</span><br><span class="line">        newpath = &#123;&#125;</span><br><span class="line"></span><br><span class="line">        <span class="keyword">print</span> <span class="string">'Day'</span>, t+<span class="number">1</span>, <span class="string">'('</span>+obs[t]+<span class="string">')'</span></span><br><span class="line">        <span class="keyword">for</span> y <span class="keyword">in</span> states:</span><br><span class="line">            <span class="comment"># 概率 隐状态 =    前状态是y0的概率 * y0转移到y的概率 * y表现为当前状态的概率</span></span><br><span class="line">            <span class="comment"># (prob, state) = max([(V[t - 1][y0] * trans_p[y0][y] * emit_p[y][obs[t]], y0) for y0 in states])</span></span><br><span class="line">            tmpList = []</span><br><span class="line">            <span class="keyword">print</span></span><br><span class="line">            <span class="keyword">for</span> y0 <span class="keyword">in</span> states:</span><br><span class="line">                tmpList.append((V[t - <span class="number">1</span>][y0] * trans_p[y0][y] * \</span><br><span class="line">                emit_p[y][obs[t]], y0))</span><br><span class="line"></span><br><span class="line">                <span class="keyword">print</span> <span class="string">'當天'</span>+y+<span class="string">': '</span>, <span class="string">"%.7s"</span> % (V[t - <span class="number">1</span>][y0] * trans_p[y0][y] * \</span><br><span class="line">emit_p[y][obs[t]]),</span><br><span class="line">                <span class="keyword">print</span> <span class="string">'='</span>,</span><br><span class="line">                <span class="keyword">print</span> V[t - <span class="number">1</span>][y0], <span class="string">'(前一天 '</span>+y0+<span class="string">')'</span>, <span class="string">'*'</span>,</span><br><span class="line">                <span class="keyword">print</span> trans_p[y0][y], <span class="string">'('</span>+y0+<span class="string">' to '</span>+y+<span class="string">')'</span>, <span class="string">'*'</span>,</span><br><span class="line">                <span class="keyword">print</span> emit_p[y][obs[t]], <span class="string">'('</span>+y, obs[t]+<span class="string">')'</span></span><br><span class="line">            <span class="keyword">print</span> <span class="string">'當天'</span>+y+<span class="string">'  最大的是: '</span>, max(tmpList)[<span class="number">0</span>]</span><br><span class="line">            (prob, state) = max(tmpList)</span><br><span class="line"></span><br><span class="line">            <span class="comment"># 记录最大概率</span></span><br><span class="line">            V[t][y] = prob</span><br><span class="line">            <span class="comment"># 记录路径</span></span><br><span class="line">            newpath[y] = path[state] + [y]</span><br><span class="line"></span><br><span class="line">        <span class="comment"># 不需要保留旧路径</span></span><br><span class="line">        path = newpath</span><br><span class="line">        print(<span class="string">'__________________________________________'</span>)</span><br><span class="line"></span><br><span class="line">    print_dptable(V)</span><br><span class="line">    (prob, state) = max([(V[len(obs) - <span class="number">1</span>][y], y) <span class="keyword">for</span> y <span class="keyword">in</span> states])</span><br><span class="line">    <span class="keyword">return</span> (prob, path[state])</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">example</span><span class="params">()</span>:</span></span><br><span class="line">    <span class="keyword">return</span> viterbi(observations,</span><br><span class="line">                   states,</span><br><span class="line">                   start_probability,</span><br><span class="line">                   transition_probability,</span><br><span class="line">                   emission_probability)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">print(example())</span><br><span class="line"></span><br><span class="line">print(<span class="string">'\n訣竅: 從最後一天推算，\n1. Day3 機率最大的那條路徑\n2. 其Day2 機率最大那條路徑\n3. 的最可能Day1'</span>)</span><br></pre></td></tr></table></figure>

<p>所以说 逐渐增加长度，每增加一次长度，之后再重新算一遍在这个长度下最后一个位置的转换概率，计算到最后一位时，就知道了每个位置的最大概率。</p>
<p>结果：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br></pre></td><td class="code"><pre><span class="line">Day 1(walk): </span><br><span class="line">Rainy 0.06 = 0.6 * 0.1</span><br><span class="line">Sunny 0.24 = 0.4 * 0.6</span><br><span class="line">_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _</span><br><span class="line">V:  [&#123;&apos;Rainy&apos;: 0.06, &apos;Sunny&apos;: 0.24&#125;]</span><br><span class="line">__________________________________________</span><br><span class="line">Day 2 (shop)</span><br><span class="line"></span><br><span class="line">當天Rainy:  0.0168 = 0.06 (前一天 Rainy) * 0.7 (Rainy to Rainy) * 0.4 (Rainy shop)</span><br><span class="line">當天Rainy:  0.0384 = 0.24 (前一天 Sunny) * 0.4 (Sunny to Rainy) * 0.4 (Rainy shop)</span><br><span class="line">當天Rainy  最大的是:  0.0384</span><br><span class="line"></span><br><span class="line">當天Sunny:  0.0054 = 0.06 (前一天 Rainy) * 0.3 (Rainy to Sunny) * 0.3 (Sunny shop)</span><br><span class="line">當天Sunny:  0.0432 = 0.24 (前一天 Sunny) * 0.6 (Sunny to Sunny) * 0.3 (Sunny shop)</span><br><span class="line">當天Sunny  最大的是:  0.0432</span><br><span class="line">__________________________________________</span><br><span class="line">Day 3 (clean)</span><br><span class="line"></span><br><span class="line">當天Rainy:  0.01344 = 0.0384 (前一天 Rainy) * 0.7 (Rainy to Rainy) * 0.5 (Rainy clean)</span><br><span class="line">當天Rainy:  0.00864 = 0.0432 (前一天 Sunny) * 0.4 (Sunny to Rainy) * 0.5 (Rainy clean)</span><br><span class="line">當天Rainy  最大的是:  0.01344</span><br><span class="line"></span><br><span class="line">當天Sunny:  0.00115 = 0.0384 (前一天 Rainy) * 0.3 (Rainy to Sunny) * 0.1 (Sunny clean)</span><br><span class="line">當天Sunny:  0.00259 = 0.0432 (前一天 Sunny) * 0.6 (Sunny to Sunny) * 0.1 (Sunny clean)</span><br><span class="line">當天Sunny  最大的是:  0.002592</span><br><span class="line">__________________________________________</span><br><span class="line">           0       1       2</span><br><span class="line">Rainy:  0.06000 0.03840 0.01344</span><br><span class="line">Sunny:  0.24000 0.04320 0.00259</span><br><span class="line">(0.01344, [&apos;Sunny&apos;, &apos;Rainy&apos;, &apos;Rainy&apos;])</span><br><span class="line"></span><br><span class="line">訣竅: 從最後一天推算，</span><br><span class="line">1. Day3 機率最大的那條路徑</span><br><span class="line">2. 其Day2 機率最大那條路徑</span><br><span class="line">3. 的最可能Day1</span><br></pre></td></tr></table></figure>

<blockquote>
<p><a href="https://gist.github.com/yesseecity" target="_blank" rel="noopener">https://gist.github.com/yesseecity</a> 将步骤列出</p>
<p><a href="http://www.hankcs.com/nlp/hmm-and-segmentation-tagging-named-entity-recognition.html" target="_blank" rel="noopener">http://www.hankcs.com/nlp/hmm-and-segmentation-tagging-named-entity-recognition.html</a> 一个根据天气和推特的例子</p>
<p><a href="https://cloud.tencent.com/developer/article/1066699" target="_blank" rel="noopener">https://cloud.tencent.com/developer/article/1066699</a> 马尔可夫模型简单理解</p>
</blockquote>

	
	</div>
  <a type="button" href="/2019/08/29/HMM_example/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/08/21/MAKER相关操作/" >MAKER相关操作</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-08-21  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>####maker pipeline</p>
<p>1.Initial MAKER Analysis</p>
<p>2.Training Gene Prediction (snap, augustus) </p>
<p>3.Round 2 MAKER Analysis</p>
<p>4.Training Gene Prediction (snap, augustus)</p>
<p>5.Round 3 MAKER Analysis</p>
<p>6.Create Maker standard gene set file</p>
<p>7.PASA </p>
<h5 id="1-Initial-MAKER-Analysis"><a href="#1-Initial-MAKER-Analysis" class="headerlink" title="1.Initial MAKER Analysis"></a>1.Initial MAKER Analysis</h5><p>#创建第一次运行maker需要的config文件 </p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">maker -CTL</span><br></pre></td></tr></table></figure>

<p>#查看修改opts文件，运行maker</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"># set of ESTs or assembled mRNA-seq in fasta format 三代转录数据</span><br><span class="line"></span><br><span class="line">est=/home/Data/avianbase/Prunella_strophita_BWA_INDEX/polished.hq.fasta</span><br><span class="line"></span><br><span class="line"># protein sequence file in fasta format (i.e. from mutiple oransisms) 从ensembl数据库中下载的human mus chicken Tae 数据 cat合并</span><br><span class="line"> </span><br><span class="line">protein=/home/Data/avianbase/Prunella_strophita_BWA_INDEX/ensembl/all/merge_pep.fa </span><br><span class="line"></span><br><span class="line"># 首次运行maker将est2 genome 与protein2genome 设置为1 ，即从头预测</span><br><span class="line"></span><br><span class="line">est2genome=1 </span><br><span class="line">protein2genome=1</span><br><span class="line"></span><br><span class="line"># 设置完成后执行</span><br><span class="line"></span><br><span class="line">nohup mpiexec -n 40 maker &gt; run_maker.log</span><br></pre></td></tr></table></figure>

<hr>
<p>#maker产生的结果</p>
<p>​    -Prunella_strophita_assemble_master_datastore_index.log 记录了每一个contig 比对的情况</p>
<p>使用maker的 gff3_merge     ### fasta_merge ###  对log文件进行注释信息的提取，在得到的 Prunella_strophita_assemble.maker.gff 文件中，包含了est , protein,  repeat masker 以及 maker 的注释结果</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#################</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># generate_gff.sh</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#################</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># sed -i 's/rndN/-/g' generate_gff.sh</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 每一轮用sed替换rndN信息</span></span><br><span class="line">gff3_merge -s -d Prunella_strophita_assemble_master_datastore_index.log &gt; Prunella_strophita_assemble.maker.gff</span><br><span class="line">fasta_merge -d Prunella_strophita_assemble_master_datastore_index.log</span><br><span class="line">gff3_merge -n -s -d Prunella_strophita_assemble_master_datastore_index.log &gt; Prunella_strophita_assemble.maker.noseq.gff</span><br><span class="line"><span class="meta">#</span><span class="bash"> est alignments</span></span><br><span class="line">awk '&#123; if ($2 == "est2genome") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt; Prunella_rndN.maker.est2genome.gff</span><br><span class="line"><span class="meta">#</span><span class="bash"> protein alignments</span></span><br><span class="line">awk '&#123; if ($2 == "protein2genome") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt;  Prunella_rndN.maker.protein2genome.gff</span><br><span class="line"><span class="meta">#</span><span class="bash"> repeat alignments</span></span><br><span class="line">awk '&#123; if ($2 ~ "repeat") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt;  Prunella_rndN.maker.repeats.gff</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> current dir </span></span><br><span class="line"><span class="meta">#</span><span class="bash"> /home/maker/20190718_version_maker/round1/Prunella_strophita_assemble.maker.output</span></span><br><span class="line"></span><br><span class="line">sed -i 's/rndN/rnd1/g' generate_gff.sh</span><br><span class="line">sh generate_gff.sh</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 对产生的gff统计基因数量与平均长度</span></span><br><span class="line">cat Prunella_strophita_assemble.maker.noseq.gff | awk '&#123; if ($3 == "gene") print $0 &#125;' | awk '&#123; sum += ($5 - $4) &#125; END &#123; print NR, sum / NR &#125;'</span><br><span class="line"></span><br><span class="line">sh gene_count.sh</span><br><span class="line">round1 result: 19711 7584.01</span><br></pre></td></tr></table></figure>

<h5 id="2-Training-Gene-Prediction"><a href="#2-Training-Gene-Prediction" class="headerlink" title="2.Training Gene Prediction"></a>2.Training Gene Prediction</h5><p>#将     snap.sh     busco.sh     gene_count.sh     cp到结果文件夹</p>
<p>通过训练基因模型再次注释提升预测的准确性</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##################</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## snap 基因结构预测</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##################</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## snap.sh</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##################</span></span></span><br><span class="line"></span><br><span class="line">mkdir snap</span><br><span class="line">mkdir snap/round1</span><br><span class="line">cd snap/round1</span><br><span class="line">maker2zff -x 0.25 -l 50 -d ../../Prunella_strophita_assemble_master_datastore_index.log</span><br><span class="line">rename genome Prunella_rnd1.zff.length50_aed0.25 genome*</span><br><span class="line">fathom Prunella_rnd1.zff.length50_aed0.25.ann Prunella_rnd1.zff.length50_aed0.25.dna -gene-stats &gt; gene-stats.log 2&gt;&amp;1</span><br><span class="line">fathom  Prunella_rnd1.zff.length50_aed0.25.ann Prunella_rnd1.zff.length50_aed0.25.dna -validate &gt; validate.log 2&gt;&amp;1</span><br><span class="line">fathom  Prunella_rnd1.zff.length50_aed0.25.ann Prunella_rnd1.zff.length50_aed0.25.dna -categorize 1000 &gt; categorize.log 2&gt;&amp;1</span><br><span class="line">fathom uni.ann uni.dna -export 1000 -plus &gt; uni-plus.log 2&gt;&amp;1</span><br><span class="line">mkdir params</span><br><span class="line">cd params</span><br><span class="line">forge ../export.ann ../export.dna &gt; ../forge.log 2&gt;&amp;1</span><br><span class="line">cd ..</span><br><span class="line">hmm-assembler.pl Prunella_rnd1.zff.length50_aed0.25 params &gt; Prunella_rnd1.zff.length50_aed0.25.hmm</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###################</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## BUSCO 基因结构预测</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###################</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">首先为了预测环境的准确性，将第一轮maker运行结果中的gff文件添加首末端1000bp的长度，可能会导致某些注释的mrna超过scaffold的边界，从而会导致bedtools报错，可以统计前后mrna的数量用以计算，而且此项影响较小。</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## run_BUSCO.sh</span></span></span><br><span class="line">awk -v OFS="\t" '&#123; if ($3 == "mRNA") print $1, $4, $5 &#125;' Prunella_strophita_assemble.maker.noseq.gff | \</span><br><span class="line">  awk -v OFS="\t" '&#123; if ($2 &lt; 1000) print $1, "0", $3+1000; else print $1, $2-1000, $3+1000 &#125;' | \</span><br><span class="line">  bedtools getfasta -fi /home/Data/avianbase/Prunella_strophita_INDEX/Prunella_strophita_assemble.fa  -bed - -fo Prunella_rnd1.all.maker.transcripts1000.fasta</span><br><span class="line"></span><br><span class="line">mkdir busco</span><br><span class="line">cd busco</span><br><span class="line">BUSCO.py -i Prunella_rnd1.all.maker.transcripts1000.fasta -o Prunella_rnd1 -l /home/maker/software/maker/exe/busco/ave_odb9/ -m genome -c 8 --long -sp human -z --augustus_prameters '--progress=true'</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> busco 的结果需要提取并修改为 augustus 物种进行下一轮的预测</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># BUSCO_rndN_number  oringin</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># species_name       target</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###############</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## generate_species.sh</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###############</span></span></span><br><span class="line"></span><br><span class="line">rename BUSCO_rndN_number species_name *</span><br><span class="line">sed -i 's/BUSCO_rndN_number/species_name/g' species_name_parameters.cfg</span><br><span class="line">sed -i 's/BUSCO_rndN_number/species_name/g' species_name_parameters.cfg.orig1</span><br><span class="line">mkdir ~/software/maker/exe/augustus/config/species/species_name</span><br><span class="line">cp species_name* ~/software/maker/exe/augustus/config/species/species_name</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> current dir /home/maker/20190718_version_maker/round1/Prunella_strophita_assemble.maker.output</span></span><br><span class="line">sh snap.sh</span><br><span class="line">sh run_busco.sh</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">cd /home/maker/20190718_version_maker/round/Prunella_strophita_assemble.maker.output/busco/run_0726_rnd1/augustus_output/retraining_parameters</span><br><span class="line">sed -i 's/rndN/rnd1/g' generate_species.sh</span><br><span class="line">sed -i 's/species_name/Prunella_chiecken_rnd1/g' generate_species.sh</span><br><span class="line">sh generate_species.sh</span><br></pre></td></tr></table></figure>

<p>#####3.Round 2 MAKER Analysis</p>
<p>将第一轮生成的gff结果中 est2genome.gff 和 protein2genome.gff 和 repeats.gff 以及 snap预测的基因模型中 hmm 文件输入第二轮的maker_opts.ctl</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line">cd /home/maker/maker_0720/round2</span><br><span class="line"></span><br><span class="line">maker -CTL</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 将maker_opts文件中的 snap 与 Augustus 预测文件添加，将 est2genome.gff, protein2genome.gff,  repeats.gff 输入，将est2genome protein2genome 设置为0</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">est_gff=/home/maker/20190704_version_maker/rnd2_data/Prunella_rnd1.all.maker.est2genome.gff   <span class="comment">#aligned ESTs or mRNA-seq from an external GFF3 file</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">genome=/home/Data/avianbase/Prunella_strophita_BWA_INDEX/Prunella_strophita_assemble.fa </span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">protein_gff=/home/maker/20190704_version_maker/rnd2_data/Prunella_rnd1.all.maker.protein2genome.gff  <span class="comment">#aligned protein homology evidence from an external GFF3 file</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">rm_gff=/home/maker/20190704_version_maker/rnd2_data/Prunella_rnd1.all.maker.repeats.gff <span class="comment">#pre-identified repeat elements from an external GFF3 file</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">snaphmm=/home/maker/20190704_version_maker/rnd2_data/Prunella_rnd1.zff.length50_aed0.25.hmm <span class="comment">#SNAP HMM file</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">augustus_species=Prunell_strophita <span class="comment">#Augustus gene prediction species model</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">max_dna_len=300000 <span class="comment">#length for dividing up contigs into chunks (increases/decreases memory usage)</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">split_hit=20000 <span class="comment">#length for the splitting of hits (expected max intron size for evidence alignments)</span></span></span><br><span class="line"></span><br><span class="line">nohup mpiexec -n 40 maker &gt; run_maker.log</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#################</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># generate_gff.sh</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#################</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># sed -i 's/rndN/-/g' generate_gff.sh</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 每一轮用sed替换rndN信息</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 因为预测结果中使用基因结构预测</span></span><br><span class="line"></span><br><span class="line">gff3_merge -s -d Prunella_strophita_assemble_master_datastore_index.log &gt; Prunella_strophita_assemble.maker.gff</span><br><span class="line">fasta_merge -d Prunella_strophita_assemble_master_datastore_index.log</span><br><span class="line">gff3_merge -n -s -d Prunella_strophita_assemble_master_datastore_index.log &gt; Prunella_strophita_assemble.maker.noseq.gff</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> est alignments</span></span><br><span class="line">awk '&#123; if ($2 ~ "est2genome") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt; Prunella_rndN.maker.est2genome.gff</span><br><span class="line"><span class="meta">#</span><span class="bash"> protein alignments</span></span><br><span class="line">awk '&#123; if ($2 ~ "protein2genome") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt;  Prunella_rndN.maker.protein2genome.gff</span><br><span class="line"><span class="meta">#</span><span class="bash"> repeat alignments</span></span><br><span class="line">awk '&#123; if ($2 ~ "repeat") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt;  Prunella_rndN.maker.repeats.gff</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">sed -i 's/rndN/rnd2/g' generate_gff.sh</span><br><span class="line">sh generate_gff.sh</span><br><span class="line"></span><br><span class="line">sh gene_count.sh</span><br><span class="line">18227 13821.9</span><br><span class="line"><span class="meta">#</span><span class="bash"> 由于第一轮是根据已有信息预测，长度在第二轮得到了修正</span></span><br></pre></td></tr></table></figure>

<h5 id="4-Training-Gene-Prediction"><a href="#4-Training-Gene-Prediction" class="headerlink" title="4.Training Gene Prediction"></a>4.Training Gene Prediction</h5><p># 与第二步相同, 将 snap.sh     busco.sh     gene_count.sh     cp到结果文件夹</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">sed -i 's/rnd1/rnd2/g' *.sh</span><br><span class="line">sed -i 's/round1/round2/g' snap.sh</span><br><span class="line">sed -i 's/00rnd1/01_rnd2/g' run_BUSCO.sh</span><br><span class="line">sh snap.sh</span><br><span class="line">sh run_BUSCO.sh</span><br><span class="line">cd busco/augustus_output/retraining_paramters</span><br><span class="line">sh generate_species.sh</span><br></pre></td></tr></table></figure>

<p>#####5.Round 3 MAKER analysis</p>
<p>将第二轮maker结果输入maker_opts.ctl 重复第三步</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">vim maker_opts.ctl</span><br><span class="line">nohup mpiexec -n 40 maker &gt; run_maker.log</span><br><span class="line">sh generate_gff.sh</span><br></pre></td></tr></table></figure>

<p>#####6.Create Maker standard gene set file</p>
<p># 对第三轮的maker 结果</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">awk -v OFS="\t" '&#123; if ($3 == "mRNA") print $1, $4, $5 &#125;' /home/maker/20190704_version_maker/02_rnd3/Prunella_strophita_assemble.output/Prunella_strophita_assemble.maker.noseq.gff | \</span><br><span class="line">  bedtools getfasta -fi /home/Data/avianbase/Prunella_strophita_INDEX/Prunella_strophita_assemble.fa -bed - -fo Prunella_rnd3.all.maker.fasta</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">mkdir busco</span><br><span class="line">cd busco</span><br><span class="line"><span class="meta">#</span><span class="bash"> 不再再添加前后1000bp的片段，同时运行busco时将model改为 transcriptome </span></span><br><span class="line">run_BUSCO.py -i Prunella_rnd3.all.maker.transcripts.fasta -o Prunella_rnd3 -l /home/maker/software/busco-master/aves_odb9/  -m transcriptome -c 8 -sp Prunella_chicken_rnd2 -z --augustus_parameters='--progress=true'</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 查看busco的完整性检查结果</span></span><br><span class="line">less run_Prunella_rnd3/short_summary_Prunella_rnd3.txt</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> gne count</span></span><br><span class="line">sh gene_count.sh</span><br><span class="line">18148 13917.2</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 提取结果中maker注释的gff部分</span></span><br><span class="line">perl get_maker_gff.pl Prunella_strophita_assemble.maker.noseq.gff Prunella_strophita_assemble.maker.only.final.noseq.gff</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 替换gff中的ID</span></span><br><span class="line">maker_map_ids --prefix Prunella --justify 5 Prunella_strophita_assemble.maker.noseq.gff &gt;</span><br><span class="line">Prunella_rnd3.maker.name.map</span><br><span class="line"></span><br><span class="line">map_gff_ids Prunella_rnd3.maker.name.map Prunella_strophita_assemble.maker.only.final.noseq.gff</span><br></pre></td></tr></table></figure>

<table>
<thead>
<tr>
<th>Gene number</th>
<th>Gene average length</th>
<th>Info</th>
</tr>
</thead>
<tbody><tr>
<td>19711</td>
<td>7584.01</td>
<td>Round 1</td>
</tr>
<tr>
<td>18227</td>
<td>13821.9</td>
<td>Round 2</td>
</tr>
<tr>
<td>18148</td>
<td>13917.2</td>
<td>Round 3</td>
</tr>
</tbody></table>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">grep 'three_prime_UTR' Prunella_strophita_assemble.maker.only.final.noseq.gff | wc -l</span><br><span class="line">2752</span><br><span class="line">grep 'five_prime_UTR' Prunella_strophita_assemble.maker.only.final.noseq.gff | wc -l</span><br><span class="line">2494</span><br></pre></td></tr></table></figure>

<h5 id="7-PASA-prediect-utr-region"><a href="#7-PASA-prediect-utr-region" class="headerlink" title="7.PASA prediect utr region"></a>7.PASA prediect utr region</h5><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> gmap cds2gff</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> clean cds 去除低质量序列</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> /data/maker/PASA_20/data</span></span><br><span class="line">/home/maker/software/PASApipeline/bin/seqclean XZ15142.hq.fasta</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> custom alignments gff  首先对基因组index然后比对生成 cds 和 gene 模式gff</span></span><br><span class="line">/home/maker/software/PASApipeline/scripts/process_GMAP_alignments_gff3_chimeras_ok.pl --genome Prunella_strophita_assemble.fa  --transcripts XZ15142.hq.fasta.clean --CPU 2 -N 2  -I 500000  &gt; gmap.spliced_alignments.gff3</span><br><span class="line"></span><br><span class="line">gmap -D . -d Prunella_strophita_assemble.fa.gmap -f 2 -n 2 -x 50 -t 2 -B 5  --max-intronlength-middle=500000 --max-intronlength-ends=500000 XZ15142.hq.fasta.clean</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> Running Alignment Assembly 将基因组与cds信息比对</span></span><br><span class="line">/home/maker/software/PASApipeline/Launch_PASA_pipeline.pl -c sqlite_conf/alignAssembly.config -C -r -R -g data/Prunella_strophita_assemble.fa -t data/XZ15142.hq.fasta.clean -T -u data/XZ15142.hq.fasta --ALIGNERS blat --CPU 2 -N 2 --IMPORT_CUSTOM_ALIGNMENTS_GFF3 data/gmap.spliced_alignments.gff3</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> Building comprehensive transcriptome database 将结果输入到sqlite数据库</span></span><br><span class="line">~/software/PASApipeline/scripts/build_comprehensive_transcriptome.dbi -c sqlite_conf/alignAssembly.config -t data/XZ15142.hq.fasta.clean</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> First round, using pre-existing gene structure annotations 对 gff进行update</span></span><br><span class="line">  ~/software/PASApipeline/Launch_PASA_pipeline.pl -c sqlite_conf/annotCompare.config -g data/Prunella_strophita_assemble.fa -t data/XZ15142.hq.fasta.clean -A -L --annots data/gmap_gene.gff3 --CPU 2</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> Second round</span></span><br><span class="line">~/software/PASApipeline/Launch_PASA_pipeline.pl -c sqlite_conf/annotCompare.config -g data/Prunella_strophita_assemble.fa -t data/XZ15142.hq.fasta.clean -A -L --annots  --CPU 2</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2019/08/21/MAKER相关操作/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/08/13/hello-world/" >Hello World</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-08-13  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>Welcome to <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a>! This is your very first post. Check <a href="https://hexo.io/docs/" target="_blank" rel="noopener">documentation</a> for more info. If you get any problems when using Hexo, you can find the answer in <a href="https://hexo.io/docs/troubleshooting.html" target="_blank" rel="noopener">troubleshooting</a> or you can ask me on <a href="https://github.com/hexojs/hexo/issues" target="_blank" rel="noopener">GitHub</a>.</p>
<h2 id="Quick-Start"><a href="#Quick-Start" class="headerlink" title="Quick Start"></a>Quick Start</h2><h3 id="Create-a-new-post"><a href="#Create-a-new-post" class="headerlink" title="Create a new post"></a>Create a new post</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo new <span class="string">"My New Post"</span></span><br></pre></td></tr></table></figure>

<p>More info: <a href="https://hexo.io/docs/writing.html" target="_blank" rel="noopener">Writing</a></p>
<h3 id="Run-server"><a href="#Run-server" class="headerlink" title="Run server"></a>Run server</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo server</span><br></pre></td></tr></table></figure>

<p>More info: <a href="https://hexo.io/docs/server.html" target="_blank" rel="noopener">Server</a></p>
<h3 id="Generate-static-files"><a href="#Generate-static-files" class="headerlink" title="Generate static files"></a>Generate static files</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo generate</span><br></pre></td></tr></table></figure>

<p>More info: <a href="https://hexo.io/docs/generating.html" target="_blank" rel="noopener">Generating</a></p>
<h3 id="Deploy-to-remote-sites"><a href="#Deploy-to-remote-sites" class="headerlink" title="Deploy to remote sites"></a>Deploy to remote sites</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo deploy</span><br></pre></td></tr></table></figure>

<p>More info: <a href="https://hexo.io/docs/deployment.html" target="_blank" rel="noopener">Deployment</a></p>

	
	</div>
  <a type="button" href="/2019/08/13/hello-world/#more" class="btn btn-default more">Read More</a>
</div>

		

		</div>

		<!-- pagination -->
		<div>
  		<center>
		<div class="pagination">

   
</div>

  		</center>
		</div>

		
		
	</div> <!-- col-md-9 -->

	
		<div class="col-md-3">
	<div id="sidebar">
	
			
  <div id="site_search">
   <div class="form-group">
    <input type="text" id="local-search-input" name="q" results="0" placeholder="Search" class="st-search-input st-default-search-input form-control"/>
   </div>  
  <div id="local-search-result"></div>
  </div>


		
			
		
			
	<div class="widget">
		<h4>Tag Cloud</h4>
		<ul class="tag_box inline list-unstyled">		
		
			<li><a href="/tags/Markdown-maker/">Markdown, maker<span>1</span></a></li>
		
			<li><a href="/tags/HMM/">HMM<span>1</span></a></li>
		
		 
		</ul>
	</div>


		
			
<div class="widget">
  <h4>Recent Posts</h4>
  <ul class="entry list-unstyled">
    
      <li>
        <a href="/2019/08/29/HMM_example/" ><i class="fa fa-file-o"></i>HMM 隐式马尔可夫模型</a>
      </li>
    
      <li>
        <a href="/2019/08/21/MAKER相关操作/" ><i class="fa fa-file-o"></i>MAKER相关操作</a>
      </li>
    
      <li>
        <a href="/2019/08/13/hello-world/" ><i class="fa fa-file-o"></i>Hello World</a>
      </li>
    
  </ul>
</div>

		
			
<div class="widget">
	<h4>Links</h4>
	<ul class="blogroll list-unstyled">
	
		<li><i class="fa fa-github"></i><a href="https://github.com/goodlucksun" title="All Github repository." target="_blank"]);">My github</a></li>
	
	</ul>
</div>


		
	</div> <!-- sidebar -->
</div> <!-- col-md-3 -->

	
	
</div> <!-- row-fluid -->
	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2019 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
