<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 <div class="page-header logo">
  <h1>Keeping your word<span class="blink-fast">∎</span></h1>
</div>

<div class="row page">

	
	<div class="col-md-9">
	

		<div class="slogan">


		<i class="fa fa-heart blink-slow"></i>

		我们进化速度快到让你感动邪门？！

</div>    

		<div id="top_search"></div>
		<div class="mypage">
		
		<!-- title and entry -->
		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/06/22/KaKs_calculator/" >KaKs 计算进化速率_update</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-06-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h3 id="首先我们还是需要用蛋白的比对文件，去匹配cds的比对文件"><a href="#首先我们还是需要用蛋白的比对文件，去匹配cds的比对文件" class="headerlink" title="首先我们还是需要用蛋白的比对文件，去匹配cds的比对文件"></a>首先我们还是需要用蛋白的比对文件，去匹配cds的比对文件</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"># rename.6merge.ms.pep 为蛋白全长的比对文件,在使用kaks的时候需要nogap命令，不然还是会报错</span><br><span class="line">perl ~/Downloads/softwares/pal2nal.v14/pal2nal.pl rename.6merge.ms.pep rename.6merge.cds -nogap &gt; rename.6merge.ms.pal2nal.clu</span><br><span class="line"></span><br><span class="line"># 然后转换为成对的基因对的格式</span><br><span class="line">AXTConvertor rename.6merge.ms.pal2nal.clu rename.6merge.ms.pal2nal.AXT</span><br><span class="line"></span><br><span class="line"># 然后再是用kaks calculator对 axt格式的文件 输出其kaks等信息 ，-m选择model为YN，如果选择model为MA的话大概需要花费20行个小时， 但是如果使用YN的话只需要十几秒，暂时没有找到有什么具体的差异，算出来的方式有一点差异。MA的话是如果 有三种异常值，一个是50</span><br><span class="line">一个是NA 一个是 0.0001，具体解释如下：如果只有一个非同一突变的位点那就是0.0001，如果只有一个同义突变的位点 那就是50，如果啥都没有，完全一样那就是NA。当模型选为YN的话，有两种异常值 -0。和 就不再计算NA或者50的位点。。同时有许多对基因</span><br><span class="line"># 发现不同的模型之间差距还是蛮大的。有可能有两倍的差距</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">KaKs_Calculator -i rename.6merge.ms.pal2nal.AXT -o rename.6merge.ms.pal2nal.kaks -m YN -c 11</span><br><span class="line">awk -F &apos;\t&apos; -v OFS=&apos;\t&apos; &apos;&#123;print $1,$5&#125;&apos; rename.6merge.ms.pal2nal.kaks  | awk -F &apos;\&amp;&apos; -v OFS=&apos;\t&apos; &apos;&#123;print $1,$2&#125;&apos;</span><br></pre></td></tr></table></figure>

<p>Table 2. Methods Incorporated in KaKs_Calculator</p>
<table>
<thead>
<tr>
<th>Approximate method</th>
<th></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody><tr>
<td>Method</td>
<td>Mutation model<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn1" target="_blank" rel="noopener">#1</a></td>
<td>Reference</td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>Step 1</td>
<td>Step 2</td>
<td>Step 3</td>
<td></td>
</tr>
<tr>
<td>NG</td>
<td>JC</td>
<td>JC</td>
<td>JC</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib26" target="_blank" rel="noopener"><em>26</em></a></td>
</tr>
<tr>
<td>LWL</td>
<td>JC</td>
<td>K2P</td>
<td>K2P</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib28" target="_blank" rel="noopener"><em>28</em></a></td>
</tr>
<tr>
<td>MLWL</td>
<td>K2P</td>
<td>K2P</td>
<td>K2P</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib30" target="_blank" rel="noopener"><em>30</em></a></td>
</tr>
<tr>
<td>LPB</td>
<td>—<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn3" target="_blank" rel="noopener">*</a></td>
<td>—<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn3" target="_blank" rel="noopener">*</a></td>
<td>K2P</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib25" target="_blank" rel="noopener">25.</a>, <a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib29" target="_blank" rel="noopener">29.</a></td>
</tr>
<tr>
<td>MLPB</td>
<td>—<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn3" target="_blank" rel="noopener">*</a></td>
<td>—<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn3" target="_blank" rel="noopener">*</a></td>
<td>K2P</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib30" target="_blank" rel="noopener"><em>30</em></a></td>
</tr>
<tr>
<td>YN</td>
<td>HKY</td>
<td>HKY</td>
<td>HKY</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib27" target="_blank" rel="noopener"><em>27</em></a></td>
</tr>
<tr>
<td>MYN</td>
<td>TN</td>
<td>TN</td>
<td>TN</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib31" target="_blank" rel="noopener"><em>31</em></a></td>
</tr>
<tr>
<td>Maximum likelihood method</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Method</td>
<td>Mutation model<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn2" target="_blank" rel="noopener">#2</a></td>
<td>Reference</td>
<td></td>
<td></td>
</tr>
<tr>
<td>GY</td>
<td>HKY</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib19" target="_blank" rel="noopener">19.</a>, <a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib20" target="_blank" rel="noopener">20.</a></td>
<td></td>
<td></td>
</tr>
<tr>
<td>MS</td>
<td>a model that has the smallest AICC among 14 candidate models</td>
<td>Model-selected method proposed in this study</td>
<td></td>
<td></td>
</tr>
<tr>
<td>MA</td>
<td>a model that averages parameters across 14 candidate models</td>
<td>Model-averaged method proposed in this study</td>
<td></td>
<td></td>
</tr>
</tbody></table>
<p>#1. The approximate method involves three basic steps: Step 1: counting the numbers of synonymous and nonsynonymous sites; Step 2: calculating the numbers of synonymous and nonsynonymous substitutions; Step 3: correcting for multiple substitutions.</p>
<p>#2. The maximum likelihood method uses the probability theory to finish the three steps in one go <a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib4" target="_blank" rel="noopener">(<em>4</em>)</a>.</p>
<p>*No specific definition of synonymous and nonsynonymous sites or substitutions.</p>
<h4 id="有个想法-计算每个物种对之间的kaks比值"><a href="#有个想法-计算每个物种对之间的kaks比值" class="headerlink" title="有个想法    计算每个物种对之间的kaks比值"></a>有个想法    计算每个物种对之间的kaks比值</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line">less Ipomoea.id | while read line</span><br><span class="line">do</span><br><span class="line">echo &quot;python get_target_kaks.py -r rename.6merge.ms.pal2nal.clu -1 At -2 &quot;$&#123;line&#125;&quot; &gt; At&quot;$&#123;line&#125;&quot;.clu&quot;</span><br><span class="line">done    </span><br><span class="line"></span><br><span class="line">python get_target_kaks.py -r rename.6merge.ms.pal2nal.clu -1 Ib -2 Sly &gt; IbSly.clu</span><br><span class="line">python get_target_kaks.py -r rename.6merge.ms.pal2nal.clu -1 Ib -2 Inil &gt; IbInil.clu</span><br><span class="line">python get_target_kaks.py -r rename.6merge.ms.pal2nal.clu -1 Ib -2 Itb &gt; IbItb.clu</span><br><span class="line">python get_target_kaks.py -r rename.6merge.ms.pal2nal.clu -1 Ib -2 Itf &gt; IbItf.clu</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">#计算所有甘薯属和番茄的关系，计算所有番薯属和拟南芥的关系</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># 给他们加上clustal的标识head</span><br><span class="line">ls S*clu | while read line                      </span><br><span class="line">do</span><br><span class="line">cat head $&#123;line&#125; &gt; $&#123;line&#125;.ok</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># 转换为AXT格式</span><br><span class="line">ls *clu.ok  | sed &apos;s/.clu.ok//g&apos; |while read line</span><br><span class="line">do</span><br><span class="line">AXTConvertor $&#123;line&#125;.clu.ok $&#123;line&#125;.axt</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line"># 生成kaks文件</span><br><span class="line">ls *axt| sed &apos;s/.axt//g&apos;  | while read line; do  echo &quot;KaKs_Calculator -i &quot;$&#123;line&#125;&quot;.axt -o &quot;$&#123;line&#125;&quot;.kaks -m MA -c 11 &amp;&quot;; done</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/06/22/KaKs_calculator/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/06/16/WGCNA/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-06-16  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>##RSEM</p>
<h3 id="1-Get-matrix-of-gene-expression-FPKM-or-TPM-or-reads-count"><a href="#1-Get-matrix-of-gene-expression-FPKM-or-TPM-or-reads-count" class="headerlink" title="1. Get matrix of gene expression (FPKM or TPM),or reads count"></a>1. Get matrix of gene expression (FPKM or TPM),or reads count</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"> </span><br><span class="line"><span class="meta"> #</span><span class="bash">  rsem-generate-data-matrix is a perl script, be sure of the prameter inside is <span class="built_in">set</span> as getting counts or FPKM or TPM</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> vim pl file to make target col as 6 <span class="keyword">for</span> FPKM</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> vim pl file to make target col as 5 <span class="keyword">for</span> TPM</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> vim pl file to make target col as 4 <span class="keyword">for</span> counts</span></span><br><span class="line"><span class="meta"> #</span><span class="bash"><span class="comment"># rsem-generate-data-matrix PATH</span></span></span><br><span class="line"> rsem-generate-data-matrix results_file/S*gene* &gt; 30.counts.matrix</span><br><span class="line"> </span><br><span class="line"><span class="meta"> #</span><span class="bash"><span class="comment"># remove reduant information</span></span></span><br><span class="line"> sed -i 's/results_file\///g' 30.counts.matrix </span><br><span class="line"> sed -i 's/.genes.results//g' 30.counts.matrix</span><br><span class="line"> sed -i 's/"//g' 30.counts.matrix </span><br><span class="line"> awk '&#123;if( $0 !~"^L")print $0&#125;' 30.counts.matrix &gt; 30.count.matrix </span><br><span class="line"> </span><br><span class="line"></span><br><span class="line"><span class="meta"> #</span><span class="bash"><span class="comment"># vim rsem-generate-data-matrix to get </span></span></span><br><span class="line"> vim ~/software/RSEM-1.3.2/rsem-generate-data-matrix</span><br><span class="line"> rsem-generate-data-matrix S*gene* &gt; 39.FPKM.matrix</span><br><span class="line"> awk '&#123;if( $0 !~"^L")print $0&#125;' 30.FPKM.matrix &gt; 30.count.matrix</span><br></pre></td></tr></table></figure>

<p>###2. Primary filter</p>
<p>#####filter condition: counts &gt; 10 in 90 percent samples </p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># filter by counts &gt;10 in 90 percent samples</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line">ab = &#123;&#125;</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> count_matrix:</span><br><span class="line">    head_include = count_matrix.readline().split(<span class="string">'\t'</span>)</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> count_matrix:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        m = <span class="number">0</span></span><br><span class="line">        <span class="keyword">for</span> i <span class="keyword">in</span> range(len(lst1)<span class="number">-1</span>):</span><br><span class="line">            <span class="keyword">if</span> float(lst1[i+<span class="number">1</span>].strip()) &gt; <span class="number">10.0</span>:</span><br><span class="line">                m += <span class="number">1</span></span><br><span class="line">        <span class="keyword">if</span> m &gt; int(<span class="number">0.9</span>*len(head_include)):</span><br><span class="line">            ab[lst1[<span class="number">0</span>]] = <span class="string">"yes"</span></span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            ab[lst1[<span class="number">0</span>]] = <span class="string">"bad"</span></span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">2</span>]) <span class="keyword">as</span> fpkm_matrix:</span><br><span class="line">    print(fpkm_matrix.readline().rstrip())</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> fpkm_matrix:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        <span class="keyword">if</span> ab[lst1[<span class="number">0</span>]] == <span class="string">"yes"</span>:</span><br><span class="line">            print(row.strip())</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> use reads counts filter FPKM matrix</span></span><br><span class="line">python f.py  30.counts.matrix 30.FPKM.noLG.matrix  &gt; 30.FPKM.noLG.filtered.matrix</span><br></pre></td></tr></table></figure>

<h3 id="3-trinity"><a href="#3-trinity" class="headerlink" title="3.trinity"></a>3.trinity</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/PtR --matrix 32.filter.noLG.FPKM.matrix --samples 32samples.txt --log2 --min_rowSums 10 --compare_replicates</span><br><span class="line">~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/PtR --matrix  32.filter.noLG.FPKM.matrix --min_rowSums 10 -s 32samples.txt --log2 --CPM --sample_cor_matrix</span><br><span class="line">~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/PtR --matrix  32.filter.noLG.FPKM.matrix -s 32samples.txt --min_rowSums 10 --log2 --CPM --center_rows   --prin_comp 8</span><br></pre></td></tr></table></figure>

<p>##WGCNA</p>
<h3 id="How-many-samples-does-WGCNA-need"><a href="#How-many-samples-does-WGCNA-need" class="headerlink" title="How many samples does WGCNA need ?"></a>How many samples does WGCNA need ?</h3><p>20 samples</p>
<p>If at all possible, one should <strong>have</strong> at least 20 <strong>samples</strong>; as with <strong>any</strong> analysis methods, more <strong>samples</strong> usually lead to more robust and refined results</p>
<h3 id="If-replicated-samples-could-use-as-individuals"><a href="#If-replicated-samples-could-use-as-individuals" class="headerlink" title="If replicated samples could use as individuals?"></a>If replicated samples could use as individuals?</h3><p>There is a biostar question about that</p>
<blockquote>
<p>Q: Hi, in WGCNA manual I learn that the minimum sample size for WGCNA is 15. If I have 10 different treatment, each treatment has three replicates, do I have 30 samples or just 10 samples?</p>
<p>A: The value of 15 seems arbitrary to me - I am not sure that it is a hard threshold. If the developer of WGCNA went by the Aztec system of numbers, maybe he would have said that 20 is the minimum (Aztecs had 20-base system of numbers).</p>
</blockquote>
<p>This answer get 3 stars include me :)</p>
<blockquote>
<p><a href="https://www.biostars.org/p/342696/" target="_blank" rel="noopener">https://www.biostars.org/p/342696/</a></p>
</blockquote>
<p>###Does gene set need normalized?</p>
<p>Dec 24, 2017 - These <em>normalization</em> methods make a big difference if one wants to compare expression of gene A to expression of gene B; but <em>WGCNA</em> calculates correlations for which gene-wise scaling factors make no difference. (Sample-wise scaling factors of course do, so samples do <em>need</em> to be <em>normalized</em>.)</p>
<h3 id="Whether-should-I-filter-gene-sets"><a href="#Whether-should-I-filter-gene-sets" class="headerlink" title="Whether  should I filter gene sets"></a>Whether  should I filter gene sets</h3><blockquote>
<p>offical:</p>
<p>We <strong>do</strong> not recommend <strong>filtering genes</strong> by differential expression. <strong>WGCNA</strong> is designed to be an unsupervised analysis method that clusters <strong>genes</strong> based on their expression profiles.</p>
</blockquote>
<p>But it really need a lot memory, 16G PC is not available for gene sets large than 50k</p>
<h3 id="Answear-two-forementioned-question"><a href="#Answear-two-forementioned-question" class="headerlink" title="Answear two forementioned question :"></a>Answear two forementioned question :</h3><p>Can WGCNA be used to analyze RNA-Seq data?</p>
<p>Yes. As far as WGCNA is concerned, working with (properly normalized) RNA-seq data isn’t really any different from working with (properly normalized) microarray data.</p>
<p><strong>We suggest removing features whose counts are consistently low</strong> (for example, removing all features that have a count of less than say 10 in more than 90% of the samples) because such low-expressed features tend to reflect noise and correlations based on counts that are mostly zero aren’t really meaningful. The actual thresholds should be based on experimental design, sequencing depth and sample counts.</p>
<p><strong>We then recommend a variance-stabilizing transformation. For example,</strong> <strong>package DESeq2 implements the function</strong> <strong>varianceStabilizingTransformation which we have found useful, but one</strong> <strong>could also start with normalized counts (or RPKM/FPKM data) and</strong> <strong>log-transform them using log2(x+1).</strong> For highly expressed features, the differences between full variance stabilization and a simple log transformation are small.</p>
<p><strong>Whether one uses RPKM, FPKM, or simply normalized counts doesn’t make</strong> <strong>a whole lot of difference for WGCNA analysis as long as all samples</strong> <strong>were processed the same way.</strong> These normalization methods make a big difference if one wants to compare expression of gene A to expression of gene B; but WGCNA calculates correlations for which gene-wise scaling factors make no difference. (Sample-wise scaling factors of course do, so samples do need to be normalized.)</p>
<p>If data come from different batches, we recommend to check for batch effects and, if needed, adjust for them. We use ComBat for batch effect removal but other methods should also work.</p>
<p>Finally, we usually check quantile scatterplots to make sure there are no systematic shifts between samples; if sample quantiles show correlations (which they usually do), quantile normalization can be used to remove this effect.</p>
<blockquote>
<p><a href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html" target="_blank" rel="noopener">OFFICIAL FAQ</a></p>
</blockquote>
<h5 id="Final-I-decided"><a href="#Final-I-decided" class="headerlink" title="Final I decided:"></a>Final I decided:</h5><p>The authors do recommend to remove the “noisier” genes, either by mean expression or variance, but they also recommmend to not filter by differential expression.</p>
<ol>
<li>Use reads count to filter: if reads count of one gene in 90% samples is less than <strong><em>10</em></strong> , remove it (i got 20902 genes,15 SPL genes included finally)    </li>
</ol>
<h2 id="WGCNA-Process"><a href="#WGCNA-Process" class="headerlink" title="WGCNA Process"></a>WGCNA Process</h2><h4 id="Samples-30-samples-during-5-period-of-batatas-devolopment"><a href="#Samples-30-samples-during-5-period-of-batatas-devolopment" class="headerlink" title="Samples: 30 samples during 5 period of batatas devolopment"></a>Samples: 30 samples during 5 period of batatas devolopment</h4><p>10TR 20TR 30FR 30SR 40FR 40SR 50FR 50SR</p>
<p>Only after 30 days, we could recognize SR based on the diameter &gt; 2.5mm (and DAT10TR DAT20TR is FR).</p>
<p>each period has 4 rep</p>
<p>go</p>
<h4 id="Target-Use-WGCNA-distinguishes-which-genes-have-the-same-express-pattern-and-find-the-gene-most-correlated-to-SPL-genes-in-the-same-module"><a href="#Target-Use-WGCNA-distinguishes-which-genes-have-the-same-express-pattern-and-find-the-gene-most-correlated-to-SPL-genes-in-the-same-module" class="headerlink" title="Target: Use WGCNA distinguishes which genes have the same express pattern and find the gene most correlated to SPL genes in the same module"></a>Target: Use WGCNA distinguishes which genes have the same express pattern and find the gene most correlated to <em>SPL</em> genes in the same module</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(WGCNA)</span><br><span class="line">allowWGCNAThreads(<span class="number">12</span>)</span><br><span class="line">options(stringsAsFactors = <span class="literal">FALSE</span>)</span><br><span class="line">setwd(<span class="string">"~/DATA/WGCNA/"</span>)</span><br><span class="line">expro=read.table(<span class="string">'rename.30.FPKM.noLG.filtered.matrix'</span>,row.names = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># filter most 25% variant genes</span></span><br><span class="line"><span class="comment">#m.vars=apply(expro,1,var)</span></span><br><span class="line"><span class="comment">#expro.upper=expro[which(m.vars&gt;quantile(m.vars, probs = seq(0, 1, 0.25))[4]),]</span></span><br><span class="line"><span class="comment">#dim(expro.upper)</span></span><br><span class="line"><span class="comment">#datExpr=as.data.frame(t(expro.upper));</span></span><br><span class="line"><span class="comment">#nGenes = ncol(datExpr)</span></span><br><span class="line"><span class="comment">#nSamples = nrow(datExpr)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># do not perform filter</span></span><br><span class="line">datExpr=as.data.frame(t(expro));</span><br><span class="line">nGenes = ncol(datExpr)</span><br><span class="line">nSamples = nrow(datExpr)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## check outlier</span></span><br><span class="line">gsg = goodSamplesGenes(datExpr, verbose = <span class="number">3</span>);</span><br><span class="line">gsg$allOK</span><br><span class="line">sampleTree = hclust(dist(datExpr), method = <span class="string">"average"</span>)</span><br><span class="line">plot(sampleTree, main = <span class="string">"Sample clustering to detect outliers"</span></span><br><span class="line">     , sub=<span class="string">""</span>, xlab=<span class="string">""</span>)</span><br><span class="line"><span class="comment"># save(datExpr, file = "FPKM-01-dataInput.RData")</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## if outlier exsit, then remove gsg False genes using this function </span></span><br><span class="line"><span class="comment">#if (!gsg$allOK)&#123;</span></span><br><span class="line"><span class="comment">## Optionally, print the gene and sample names that were removed:</span></span><br><span class="line"><span class="comment">#if (sum(!gsg$goodGenes)&gt;0) </span></span><br><span class="line"><span class="comment"># printFlush(paste("Removing genes:", </span></span><br><span class="line"><span class="comment"># paste(names(datExpr)[!gsg$goodGenes], collapse = ",")));</span></span><br><span class="line"><span class="comment">#if (sum(!gsg$goodSamples)&gt;0) </span></span><br><span class="line"><span class="comment"># printFlush(paste("Removing samples:", </span></span><br><span class="line"><span class="comment"># paste(rownames(datExpr)[!gsg$goodSamples], collapse = ",")));</span></span><br><span class="line"><span class="comment"># Remove the offending genes and samples from the data:</span></span><br><span class="line"><span class="comment">#datExpr = datExpr[gsg$goodSamples, gsg$goodGenes]</span></span><br><span class="line"><span class="comment">#&#125;</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## Log2 fold change</span></span><br><span class="line"><span class="comment">#datExpr = log2(datExpr+1)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## try different soft threshold,from 1 to 25</span></span><br><span class="line">powers = c(seq(<span class="number">1</span>,<span class="number">25</span>,by = <span class="number">1</span>))</span><br><span class="line">sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = <span class="number">5</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">## plot the 25 value</span></span><br><span class="line">par(mfrow = c(<span class="number">1</span>,<span class="number">2</span>));</span><br><span class="line">cex1 = <span class="number">0.8</span>;</span><br><span class="line">plot(sft$fitIndices[,<span class="number">1</span>], -sign(sft$fitIndices[,<span class="number">3</span>])*sft$fitIndices[,<span class="number">2</span>],</span><br><span class="line">     xlab=<span class="string">"Soft Threshold (power)"</span>,ylab=<span class="string">"Scale Free Topology Model Fit,signed R^2"</span>,type=<span class="string">"n"</span>,</span><br><span class="line">     main = paste(<span class="string">"Scale independence"</span>));</span><br><span class="line">text(sft$fitIndices[,<span class="number">1</span>], -sign(sft$fitIndices[,<span class="number">3</span>])*sft$fitIndices[,<span class="number">2</span>],</span><br><span class="line">     labels=powers,cex=cex1,col=<span class="string">"red"</span>);</span><br><span class="line">abline(h=<span class="number">0.80</span>,col=<span class="string">"red"</span>)</span><br><span class="line">plot(sft$fitIndices[,<span class="number">1</span>], sft$fitIndices[,<span class="number">5</span>],</span><br><span class="line">     xlab=<span class="string">"Soft Threshold (power)"</span>,ylab=<span class="string">"Mean Connectivity"</span>, type=<span class="string">"n"</span>,</span><br><span class="line">     main = paste(<span class="string">"Mean connectivity"</span>))</span><br><span class="line">text(sft$fitIndices[,<span class="number">1</span>], sft$fitIndices[,<span class="number">5</span>], labels=powers, cex=cex1,col=<span class="string">"red"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">#### get an estimate power value(Closest to 0.9)</span></span><br><span class="line">sft$powerEstimate</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## One-step network construction and module detection##</span></span><br><span class="line">cor = WGCNA::cor</span><br><span class="line">net = blockwiseModules(</span><br><span class="line">  <span class="comment"># input data</span></span><br><span class="line">  datExpr, </span><br><span class="line">  <span class="comment"># calculate correlation coefficient</span></span><br><span class="line">  power = sft$powerEstimate, </span><br><span class="line">  corType = <span class="string">"pearson"</span>,</span><br><span class="line">  networkType = <span class="string">"unsigned"</span>,</span><br><span class="line">  maxBlockSize = <span class="number">6000</span>,</span><br><span class="line">  TOMType = <span class="string">"unsigned"</span>,</span><br><span class="line">  deepSplit = <span class="number">2</span>,</span><br><span class="line">  minModuleSize = <span class="number">30</span>,</span><br><span class="line">  reassignThreshold = <span class="number">0</span>,</span><br><span class="line">  <span class="comment"># distance among modules less than 0.15, otherwise correlation among module more than 0.85 will be merged</span></span><br><span class="line">  mergeCutHeight = <span class="number">0.15</span>,</span><br><span class="line">  numericLabels = <span class="literal">TRUE</span>, </span><br><span class="line">  pamRespectsDendro = <span class="literal">FALSE</span>,)</span><br><span class="line"><span class="comment">## show how many modules identified</span></span><br><span class="line">table(net$colors)</span><br><span class="line"><span class="comment">## plot results</span></span><br><span class="line"><span class="comment">## Convert labels to colors for plotting</span></span><br><span class="line">mergedColors = labels2colors(net$colors)</span><br><span class="line"><span class="comment">## Plot the dendrogram and the module colors underneath</span></span><br><span class="line">plotDendroAndColors(net$dendrograms[[<span class="number">1</span>]], mergedColors[net$blockGenes[[<span class="number">1</span>]]],<span class="string">"Module colors"</span>,</span><br><span class="line">                    dendroLabels = <span class="literal">FALSE</span>, hang = <span class="number">0.03</span>,</span><br><span class="line">                    addGuide = <span class="literal">TRUE</span>, guideHang = <span class="number">0.05</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## get all edge file and node file</span></span><br><span class="line"></span><br><span class="line">moduleLabels = net$colors</span><br><span class="line">moduleColors = labels2colors(net$colors)</span><br><span class="line">modProbes = names(datExpr)</span><br><span class="line">modules = <span class="string">"all_without_normalize"</span></span><br><span class="line">TOM = TOMsimilarityFromExpr(datExpr, power = sft$powerEstimate);</span><br><span class="line">cyt = exportNetworkToCytoscape(TOM,</span><br><span class="line">                               edgeFile = paste(<span class="string">"CytoscapeInput-edges-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               nodeFile = paste(<span class="string">"CytoscapeInput-nodes-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               weighted = <span class="literal">TRUE</span>,</span><br><span class="line">                               threshold = <span class="number">0.02</span>,</span><br><span class="line">                               nodeNames = modProbes,</span><br><span class="line">                               nodeAttr = moduleColors)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Or retrieve specific module edges</span></span><br><span class="line">modules = c(<span class="string">"turquoise"</span>)</span><br><span class="line"><span class="comment"># 要可视化的基因</span></span><br><span class="line">probes = colnames(datExpr)</span><br><span class="line">inModule = is.finite(match(moduleColors, modules));</span><br><span class="line">modProbes = probes[inModule]</span><br><span class="line"><span class="comment"># 候选基因的 TOM</span></span><br><span class="line">modTOM = TOM[inModule, inModule]</span><br><span class="line"></span><br><span class="line">dimnames(modTOM) = list(modProbes, modProbes)</span><br><span class="line"><span class="comment"># Export the network into edge and node list files Cytoscape can read</span></span><br><span class="line">cyt = exportNetworkToCytoscape(modTOM,</span><br><span class="line">                               edgeFile = paste(<span class="string">"CytoscapeInput-edges-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               nodeFile = paste(<span class="string">"CytoscapeInput-nodes-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               weighted = <span class="literal">TRUE</span>,</span><br><span class="line">                               threshold = <span class="number">0.02</span>,</span><br><span class="line">                               nodeNames = modProbes,</span><br><span class="line">                               nodeAttr = moduleColors[inModule])</span><br><span class="line"><span class="comment">########</span></span><br><span class="line"><span class="comment"># If our target genes are not clustered in one module as expected</span></span><br><span class="line"><span class="comment"># may be reconstruct net in more sensitive parameter</span></span><br><span class="line"><span class="comment">#######</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># get module</span></span><br><span class="line">MEs0 = moduleEigengenes(datExpr, moduleColors)$eigengenes</span><br><span class="line">MEs = orderMEs(MEs0)</span><br><span class="line">modNames = substring(names(MEs), <span class="number">3</span>)</span><br><span class="line">geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = <span class="string">"p"</span>))</span><br><span class="line">MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), </span><br><span class="line">                                          nSamples))</span><br><span class="line">names(geneModuleMembership) = paste(<span class="string">"MM"</span>, modNames, sep=<span class="string">""</span>)</span><br><span class="line">names(MMPvalue) = paste(<span class="string">"p.MM"</span>, modNames, sep=<span class="string">""</span>)</span><br><span class="line"></span><br><span class="line">write.csv(MMPvalue,file = <span class="string">"MMPvalue.csv"</span>,quote=<span class="literal">F</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">HubGenes &lt;- chooseTopHubInEachModule(datExpr,modules)</span><br><span class="line">write.table (HubGenes,file = <span class="string">"HubGenes_of_turquoise_module.txt"</span>,quote=<span class="literal">F</span>,sep=<span class="string">'\t'</span>)</span><br></pre></td></tr></table></figure>

<h3 id="SPL-gene-analysis"><a href="#SPL-gene-analysis" class="headerlink" title="SPL gene analysis"></a>SPL gene analysis</h3><h4 id="get-target-gene-color-and-id"><a href="#get-target-gene-color-and-id" class="headerlink" title="get target gene color and id"></a>get target gene color and id</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">get top gene based on weight</span></span><br><span class="line">less  spl.ori.id | while read line; do grep -w $&#123;line&#125;  CytoscapeInput-nodes-all_without_normalize.txt; done &gt; spl.color</span><br><span class="line">mkdir result</span><br><span class="line">awk '&#123;print "grep -w \""$1"\" 0.05.CytoscapeInput-edges-all.txt &gt; result/"$1"."$3".result"&#125;' spl.color &gt; go</span><br><span class="line">sh go</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> direction among genes <span class="keyword">in</span> sample color are lacking of direction, cuz we constrcut net use <span class="string">"unsigned"</span> method</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> therefore we could sort the edge file by 3rd column</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> either we should filter those replicated genes <span class="keyword">in</span> 1st 2nd column </span></span><br><span class="line">ls *result.2 | sed 's/.result.2//g' | while read line;do python sort.py $&#123;line&#125;.result &gt; $&#123;line&#125;.id.2;done</span><br></pre></td></tr></table></figure>

<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## sort.py </span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line">i = <span class="number">0</span></span><br><span class="line">lst3 = []</span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">sort_list</span><span class="params">(one_list)</span>:</span></span><br><span class="line">  <span class="keyword">return</span> list(set(one_list))</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_data:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_data:</span><br><span class="line">        i += <span class="number">1</span></span><br><span class="line">file_len = i</span><br><span class="line">sort_file = <span class="string">''</span>.join(sorted(open(sys.argv[<span class="number">1</span>]), key=<span class="keyword">lambda</span> s: s.split()[<span class="number">2</span>],reverse=<span class="number">1</span>))</span><br><span class="line">lst1 = sort_file.strip().split(<span class="string">'\n'</span>)</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> range(int(file_len*<span class="number">0.1</span>)):</span><br><span class="line">    lst2 = lst1[i].split(<span class="string">'\t'</span>)</span><br><span class="line">    lst3.append(lst2[<span class="number">0</span>])</span><br><span class="line">    lst3.append(lst2[<span class="number">1</span>])</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> range(len(sort_list(lst3))):</span><br><span class="line">    print(sort_list(lst3)[i<span class="number">-1</span>])</span><br></pre></td></tr></table></figure>

<h4 id="get-gene-based-on-FPKM-cor"><a href="#get-gene-based-on-FPKM-cor" class="headerlink" title="get gene based on FPKM cor"></a>get gene based on FPKM cor</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> only analyze gene <span class="keyword">in</span> turquoise</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> get gene based on FPKM cor</span></span><br><span class="line">less  turquoise.gene.id | while read line  </span><br><span class="line">do</span><br><span class="line">grep -w $&#123;line&#125; ../../9_RSEM/88/88.FPKM.noLGfiltered.matrix</span><br><span class="line">done &gt; turquoise.gene.FPKM.matrix</span><br></pre></td></tr></table></figure>

<h4 id="get-correlation-ship-between-spl-gene-and-turquoise-gene"><a href="#get-correlation-ship-between-spl-gene-and-turquoise-gene" class="headerlink" title="get correlation ship between spl gene and turquoise gene"></a>get correlation ship between spl gene and turquoise gene</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(circlize)</span><br><span class="line"><span class="keyword">library</span>(corrplot)</span><br><span class="line"><span class="keyword">library</span>(pheatmap)</span><br><span class="line">setwd(<span class="string">"~/data/gene_family/downstream/10_GO_KEGG"</span>)</span><br><span class="line"></span><br><span class="line">gene &lt;- read.delim(<span class="string">'hubgene/turquoise.gene.FPKM.matrix'</span>, row.names = <span class="number">1</span>,header = <span class="literal">T</span>, sep = <span class="string">'\t'</span>)</span><br><span class="line">gene &lt;- t(gene) </span><br><span class="line">gene &lt;- log2(gene+<span class="number">1</span>) </span><br><span class="line">gene_cor &lt;- cor(gene, method = <span class="string">'pearson'</span>)</span><br><span class="line">df &lt;- as.data.frame(gene_cor)</span><br><span class="line"></span><br><span class="line">spl_turquoise &lt;- df[,which(names(df)%<span class="keyword">in</span>%c(<span class="string">"g34536"</span>,<span class="string">"g46868"</span>,<span class="string">"g54228"</span>,<span class="string">"g31999"</span>,<span class="string">"g44906"</span>))]</span><br><span class="line">gene_cor &lt;- as.matrix(spl_turquoise)</span><br><span class="line"></span><br><span class="line">col &lt;- colorRampPalette(c(<span class="string">"#BB4444"</span>, <span class="string">"#EE9988"</span>, <span class="string">"#FFFFFF"</span>, <span class="string">"#77AADD"</span>, <span class="string">"#4477AA"</span>))</span><br><span class="line">pheatmap(spl_turquoise,</span><br><span class="line">         labels_row = <span class="literal">NULL</span>,</span><br><span class="line">         angle_col = <span class="string">"45"</span>,</span><br><span class="line">         cluster_row = <span class="literal">T</span>,</span><br><span class="line">         )</span><br><span class="line">  </span><br><span class="line">write.csv(spl_turquoise, file = <span class="string">"5spl_turquoise.gene.cor.matrix"</span>,quote = <span class="literal">F</span>)</span><br></pre></td></tr></table></figure>

<h2 id="eggNOG-mapper-and-ClusterProfile"><a href="#eggNOG-mapper-and-ClusterProfile" class="headerlink" title="eggNOG mapper and ClusterProfile"></a>eggNOG mapper and ClusterProfile</h2><p>we first submit our whole protome to <a href="http://eggnog-mapper.embl.de/" target="_blank" rel="noopener">eggNOG website</a>,  against 33090 database representing plants.</p>
<p>Then we use R script to create a R package named </p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> we should modify some cloumn name and necessary content both <span class="keyword">in</span> these two file</span></span><br><span class="line">query_seqs.fa.emapper.annotations</span><br><span class="line"></span><br><span class="line">nohup Rscript  makeOrgPackageFromEmapper.R 1&gt;make.log 2&gt;err.log</span><br></pre></td></tr></table></figure>

<h4 id="makeOrgPackageFromEmapper-R"><a href="#makeOrgPackageFromEmapper-R" class="headerlink" title="makeOrgPackageFromEmapper.R"></a>makeOrgPackageFromEmapper.R</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(tidyverse)</span><br><span class="line"><span class="keyword">library</span>(stringr)</span><br><span class="line"><span class="keyword">library</span>(KEGGREST)</span><br><span class="line"><span class="keyword">library</span>(AnnotationForge)</span><br><span class="line"></span><br><span class="line"><span class="comment">#' Title</span></span><br><span class="line"><span class="comment">#'</span></span><br><span class="line"><span class="comment">#' @param f_emapper_anno eggnog-mapper annotation result</span></span><br><span class="line"><span class="comment">#' @param author Who is the creator of this package? like "xxx &lt;xxx@xxx.xx&gt;"</span></span><br><span class="line"><span class="comment">#' @param tax_id The Taxonomy ID that represents your organism. (NCBI has a nice online browser for finding the one you need)</span></span><br><span class="line"><span class="comment">#' @param genus Single string indicating the genus</span></span><br><span class="line"><span class="comment">#' @param species Single string indicating the species</span></span><br><span class="line"><span class="comment">#'</span></span><br><span class="line"><span class="comment">#' @return OrgDb name</span></span><br><span class="line"><span class="comment">#' @export</span></span><br><span class="line"><span class="comment">#'</span></span><br><span class="line"><span class="comment">#' @examples</span></span><br><span class="line">makeOrgPackageFromEmapper &lt;- <span class="keyword">function</span>(f_emapper_anno, </span><br><span class="line">                                      author, </span><br><span class="line">                                      tax_id = <span class="string">"0"</span>, </span><br><span class="line">                                      genus = <span class="string">"default"</span>, </span><br><span class="line">                                      species = <span class="string">"default"</span>) &#123;</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># f_emapper_anno = "input/sesame.emapper.annotations"</span></span><br><span class="line">  <span class="comment"># author = "zhangxudong &lt;zhangxudong@genek.tv&gt;"</span></span><br><span class="line">  <span class="comment"># tax_id = "4182"</span></span><br><span class="line">  <span class="comment"># genus = "Sesamum"</span></span><br><span class="line">  <span class="comment"># species = "indicum"</span></span><br><span class="line">  </span><br><span class="line">  <span class="comment"># read emapper result</span></span><br><span class="line">  emapper &lt;- read_delim(f_emapper_anno,</span><br><span class="line">                        <span class="string">"\t"</span>, escape_double = <span class="literal">FALSE</span>, trim_ws = <span class="literal">TRUE</span>)</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract gene name from emapper</span></span><br><span class="line">  gene_info &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(GID = query_name, GENENAME = `eggNOG annot`) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract go annotation from emapper</span></span><br><span class="line">  gos &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(query_name, GO_terms) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  gene2go = data.frame(GID = character(),</span><br><span class="line">                       GO = character(),</span><br><span class="line">                       EVIDENCE = character())</span><br><span class="line">  </span><br><span class="line">  <span class="keyword">for</span> (row <span class="keyword">in</span> <span class="number">1</span>:nrow(gos)) &#123;</span><br><span class="line">    the_gid &lt;- gos[row, <span class="string">"query_name"</span>][[<span class="number">1</span>]]</span><br><span class="line">    the_gos &lt;- str_split(gos[row,<span class="string">"GO_terms"</span>], <span class="string">","</span>, simplify = <span class="literal">FALSE</span>)[[<span class="number">1</span>]]</span><br><span class="line">    </span><br><span class="line">    df_temp &lt;- data_frame(GID = rep(the_gid, length(the_gos)),</span><br><span class="line">                          GO = the_gos,</span><br><span class="line">                          EVIDENCE = rep(<span class="string">"IEA"</span>, length(the_gos)))</span><br><span class="line">    gene2go &lt;- rbind(gene2go, df_temp)</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract kegg pathway annotation from emapper ----------------------------</span></span><br><span class="line">  kos &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(query_name, KEGG_KOs) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  gene2ko = data.frame(GID = character(),</span><br><span class="line">                       Ko = character())</span><br><span class="line">  </span><br><span class="line">  df_temp &lt;- list()</span><br><span class="line">  <span class="keyword">for</span> (row <span class="keyword">in</span> <span class="number">1</span>:nrow(kos)) &#123;</span><br><span class="line">    the_gid &lt;- kos[row, <span class="string">"query_name"</span>][[<span class="number">1</span>]]</span><br><span class="line">    the_kos &lt;- str_split(kos[row,<span class="string">"KEGG_KOs"</span>], <span class="string">","</span>, simplify = <span class="literal">FALSE</span>)[[<span class="number">1</span>]]</span><br><span class="line">    </span><br><span class="line">    df_temp[[row]] &lt;- data_frame(GID = rep(the_gid, length(the_kos)),</span><br><span class="line">                                 Ko = the_kos)</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  gene2ko &lt;- bind_rows(df_temp)</span><br><span class="line">  </span><br><span class="line">  load(file = <span class="string">"kegg_info.RData"</span>)</span><br><span class="line">  gene2pathway &lt;- gene2ko %&gt;% left_join(ko2pathway, by = <span class="string">"Ko"</span>) %&gt;%</span><br><span class="line">    left_join(pathway2name, by = <span class="string">"Pathway"</span>) %&gt;%</span><br><span class="line">    dplyr::select(GID, Ko, Pathway, Pathway_Name, Pathway_Class, Pathway_Subclass) %&gt;%</span><br><span class="line">    distinct() %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract COG annotation from emapper -------------------------------------</span></span><br><span class="line">  cog_info &lt;- read_delim(<span class="string">"cog_funclass.tab"</span>, </span><br><span class="line">                         <span class="string">"\t"</span>, escape_double = <span class="literal">FALSE</span>, trim_ws = <span class="literal">TRUE</span>)</span><br><span class="line">  </span><br><span class="line">  cogs &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(query_name, COG = `COG cat`) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  gene2cog = data.frame(GID = character(),</span><br><span class="line">                        COG = character())</span><br><span class="line">  </span><br><span class="line">  df_temp &lt;- list()</span><br><span class="line">  <span class="keyword">for</span> (row <span class="keyword">in</span> <span class="number">1</span>:nrow(cogs)) &#123;</span><br><span class="line">    the_gid &lt;- cogs[row, <span class="string">"query_name"</span>][[<span class="number">1</span>]]</span><br><span class="line">    the_cogs &lt;- str_trim(str_split(cogs[row,<span class="string">"COG"</span>], <span class="string">","</span>, simplify = <span class="literal">FALSE</span>)[[<span class="number">1</span>]])</span><br><span class="line">    </span><br><span class="line">    df_temp[[row]] &lt;- data_frame(GID = rep(the_gid, length(the_cogs)),</span><br><span class="line">                                 COG = the_cogs)</span><br><span class="line">  &#125;</span><br><span class="line">  gene2cog &lt;- bind_rows(df_temp)</span><br><span class="line">  </span><br><span class="line">  gene2cog &lt;- gene2cog %&gt;% left_join(cog_info, by = <span class="string">"COG"</span>)</span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">  <span class="comment"># make OrgDb</span></span><br><span class="line">  makeOrgPackage(gene_info=gene_info,</span><br><span class="line">                 go=gene2go,</span><br><span class="line">                 <span class="comment">#ko=gene2ko,</span></span><br><span class="line">                 pathway=gene2pathway,</span><br><span class="line">                 cog=gene2cog,</span><br><span class="line">                 version=<span class="string">"0.0.2"</span>,</span><br><span class="line">                 maintainer=author,</span><br><span class="line">                 author=author,</span><br><span class="line">                 outputDir = <span class="string">"."</span>,</span><br><span class="line">                 tax_id=tax_id,</span><br><span class="line">                 genus=genus,</span><br><span class="line">                 species=species,</span><br><span class="line">                 goTable=<span class="string">"go"</span>)</span><br><span class="line">  </span><br><span class="line">  my_orgdb &lt;- str_c(<span class="string">"org."</span>, str_to_upper(str_sub(genus, <span class="number">1</span>, <span class="number">1</span>)) , species, <span class="string">".eg.db"</span>, sep = <span class="string">""</span>)</span><br><span class="line">  <span class="keyword">return</span>(my_orgdb)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">my_orgdb &lt;- makeOrgPackageFromEmapper(<span class="string">"query_seqs.fa.emapper.annotations"</span>, </span><br><span class="line">                                      <span class="string">"shy &lt;2020180495@jsnu.edu.cn&gt;"</span>, </span><br><span class="line">                                      tax_id = <span class="string">"4120"</span>, </span><br><span class="line">                                      genus = <span class="string">"Ipomoea"</span>, </span><br><span class="line">                                      species = <span class="string">"batatas"</span>)</span><br></pre></td></tr></table></figure>

<h3 id="And-i-also-use-kobas-enrichment-methods-but-it-still-queued"><a href="#And-i-also-use-kobas-enrichment-methods-but-it-still-queued" class="headerlink" title="And i also use kobas enrichment methods, but it still queued"></a>And i also use kobas enrichment methods, but it still queued</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"># ipomoea nil</span><br><span class="line">http://kobas.cbi.pku.edu.cn/kobas3/retrieve/?taskid=8ee9e38a368d4ce98ea359f629b0ca38</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># sl</span><br><span class="line"></span><br><span class="line">page:http://kobas.cbi.pku.edu.cn/kobas3/retrieve/?taskid=48bed844140b48a7a4e76e8aca4fa16a</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># enrichment nil </span><br><span class="line"></span><br><span class="line">http://kobas.cbi.pku.edu.cn/kobas3/retrieve/?taskid=bd863ad74ea74fa9adc91b4bfe080520</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">#enrichment sl</span><br></pre></td></tr></table></figure>

<h4 id="There-is-a-very-funny-bug"><a href="#There-is-a-very-funny-bug" class="headerlink" title="There is a very funny bug"></a>There is a very funny bug</h4><p>when i first visit kobas webwites, I find they only have two species included in protein blast tags (human and mouse), after I pee back I found many items loaded…. So I could use Ipomoea nil as reference species…</p>
<h4 id="GO-and-KEGG-annotate-by-ClusterProfile"><a href="#GO-and-KEGG-annotate-by-ClusterProfile" class="headerlink" title="GO and KEGG annotate by ClusterProfile"></a>GO and KEGG annotate by ClusterProfile</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(readr)</span><br><span class="line"><span class="keyword">library</span>(clusterProfiler)</span><br><span class="line"><span class="keyword">library</span>(DOSE)</span><br><span class="line"><span class="keyword">library</span>(purrr)</span><br><span class="line"><span class="keyword">library</span>(tidyverse)</span><br><span class="line"><span class="keyword">library</span>(org.Ibatatas.eg.db)</span><br><span class="line">setwd(<span class="string">"~/data/gene_family/downstream/10_GO_KEGG/"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 安装构建的R包</span></span><br><span class="line">my_orgdb &lt;- <span class="string">"org.Ibatatas.eg.db/"</span></span><br><span class="line"><span class="keyword">if</span> (requireNamespace(my_orgdb, quietly = <span class="literal">TRUE</span>))</span><br><span class="line">  remove.packages(my_orgdb)</span><br><span class="line">install.packages(my_orgdb, repos = <span class="literal">NULL</span>, type = <span class="string">"source"</span>)</span><br><span class="line"><span class="keyword">library</span>(org.Ibatatas.eg.db)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">load(<span class="string">"kegg_info.RData"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">gene_list &lt;- read.table(<span class="string">"hubgene/CytoscapeInput-edges-turquoise.top10.id"</span>, header = <span class="literal">FALSE</span>)</span><br><span class="line">gene_list &lt;- as.character(gene_list$V1)</span><br><span class="line"><span class="comment">#gene_list2 &lt;- read.table("../39/30/gene_filter/final/top10_percent.g34536.brown.f", header = FALSE)</span></span><br><span class="line"><span class="comment">#gene_list2 &lt;- as.character(gene_list2$V1)</span></span><br><span class="line"></span><br><span class="line">ego &lt;- enrichGO(gene          = gene_list, <span class="comment">#差异基因 vector</span></span><br><span class="line">                keyType       = <span class="string">"GID"</span>,  <span class="comment">#差异基因的 ID 类型，需要是 OrgDb 支持的</span></span><br><span class="line">                OrgDb         = org.Ibatatas.eg.db, <span class="comment">#对应的OrgDb</span></span><br><span class="line">                ont           = <span class="string">"BP"</span>, <span class="comment">#GO 分类名称，CC BP MF </span></span><br><span class="line">                pvalueCutoff  = <span class="number">0.05</span>, <span class="comment">#Pvalue 阈值</span></span><br><span class="line">                qvalueCutoff  = <span class="number">0.05</span>, <span class="comment">#qvalue 阈值</span></span><br><span class="line">                pAdjustMethod = <span class="string">"BH"</span>, <span class="comment">#Pvalue 矫正方法 </span></span><br><span class="line">                readable      = <span class="literal">FALSE</span>) <span class="comment">#TRUE 则展示SYMBOL，FALSE 则展示原来的ID</span></span><br><span class="line"><span class="comment">#将 ego 对象转换为dataframe，新版本可以用as.data.frame(ego)</span></span><br><span class="line">ego_results&lt;-as.data.frame(ego)</span><br><span class="line">barplot(ego, showCategory=<span class="number">20</span>, x = <span class="string">"GeneRatio"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">pathway2gene &lt;- AnnotationDbi::select(org.Ibatatas.eg.db, </span><br><span class="line">                                      keys = keys(org.Ibatatas.eg.db), </span><br><span class="line">                                      columns = c(<span class="string">"Pathway"</span>)) %&gt;%</span><br><span class="line">  na.omit() %&gt;%</span><br><span class="line">  dplyr::select(Pathway, GID)</span><br><span class="line"></span><br><span class="line">ekp &lt;- enricher(gene_list, </span><br><span class="line">                TERM2GENE = pathway2gene, </span><br><span class="line">                TERM2NAME = pathway2name, </span><br><span class="line">                pvalueCutoff = <span class="number">0.05</span>, </span><br><span class="line">                qvalueCutoff = <span class="number">0.2</span>,</span><br><span class="line">                pAdjustMethod = <span class="string">"BH"</span>,</span><br><span class="line">                minGSSize = <span class="number">1</span>)</span><br><span class="line">ekp_results &lt;- as.data.frame(ekp)</span><br><span class="line"></span><br><span class="line">barplot(ekp, showCategory=<span class="number">20</span>, x = <span class="string">"GeneRatio"</span>)</span><br><span class="line"></span><br><span class="line">dotplot(ekp)</span><br><span class="line"></span><br><span class="line">emapplot(ekp)</span><br></pre></td></tr></table></figure>

<h3 id="use-CytoScape-construct-network-plot"><a href="#use-CytoScape-construct-network-plot" class="headerlink" title="use CytoScape construct network plot"></a>use CytoScape construct network plot</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p><img src="file:///Users/sunhaoyun/Library/Application%20Support/typora-user-images/image-20200620111509583.png?lastModify=1592810480" alt="image-20200620111509583"></p>
<h3 id="Hub-based-analysis"><a href="#Hub-based-analysis" class="headerlink" title="Hub-based analysis"></a>Hub-based analysis</h3><blockquote>
<p><a href="https://www.karger.com/Article/Fulltext/495205" target="_blank" rel="noopener">Weighted Gene Correlation Network Analysis (WGCNA) Detected Loss of MAGI2 Promotes Chronic Kidney Disease (CKD) by Podocyte Damage</a></p>
</blockquote>
<p>Highly connected hub nodes are central to the network’s architecture [<a href="https://www.karger.com/Article/Fulltext/495205#ref6" target="_blank" rel="noopener">6</a>] and studies suggested that genes more centralized in the network are more likely to be key drivers to respective cellular functions compared with peripheral genes [<a href="https://www.karger.com/Article/Fulltext/495205#ref26" target="_blank" rel="noopener">26</a>]. Therefore, more important nodes can be obtained by identifying hub nodes. In this study, the top 5% of nodes with highest connectivity were defined as hub genes in each module [<a href="https://www.karger.com/Article/Fulltext/495205#ref27" target="_blank" rel="noopener">27</a>]. Fig. <a href="https://www.karger.com/Article/Fulltext/495205#f03" target="_blank" rel="noopener">3</a>A-D are the networks of hub genes in the brown, yellow, green and red modules, only displaying connections with weight (w) above threshold values of 0.3, 0.15, 0.05, and 0.05, respectively. Clearly, hub genes (red nodes) exhibited high connectivity with neighboring genes whose functions are consistent with GO and KEGG findings. Taking the brown module as an example, which showed functional enrichment in metabolic and oxidation-reduction processes, some hub genes have been also reported to participate in similar processes, e.g. the SLC13 Family (SLC13A3) [<a href="https://www.karger.com/Article/Fulltext/495205#ref28" target="_blank" rel="noopener">28</a>, <a href="https://www.karger.com/Article/Fulltext/495205#ref29" target="_blank" rel="noopener">29</a>], SLC22 Family (SLC22A6) [<a href="https://www.karger.com/Article/Fulltext/495205#ref30" target="_blank" rel="noopener">30</a>] or MSRA that plays a protective role in the progression of UUO-induced kidney fibrosis via suppression of fibrotic responses caused by oxidative stress [<a href="https://www.karger.com/Article/Fulltext/495205#ref31" target="_blank" rel="noopener">31</a>]. Similarly, some hub genes in the yellow module have been identified as pro-inflammatory factors in a variety of human diseases, including CTSS and ITGB2 [<a href="https://www.karger.com/Article/Fulltext/495205#ref32" target="_blank" rel="noopener">32</a>], consistent with the BP term or KEGG pathway enriched in the yellow module. Meanwhile, some well-known inflammatory factors also had high connectivity to these hub genes, e.g. IL1B, CXCR4 and CASP1. These observations further supported the notion that proposed roles for hub genes of unknown functions may be inferred from clusters of genes similarly expressed across many biological conditions.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">awk &apos;&#123;if ($1 ~ &quot;^g&quot;)print $1&quot;.t1&quot;&#125;&apos; 0.1.CytoscapeInput-nodes-turquoise.txt  &gt; 0.1.gene.id</span><br><span class="line">awk &apos;&#123;if ($1 ~ &quot;^g&quot;)print $1&quot;.t1&quot;&#125;&apos; 0.05.CytoscapeInput-nodes-turquoise.txt  &gt; 0.05.gene.id</span><br><span class="line">awk &apos;&#123;if ($1 ~ &quot;^g&quot;)print $1&quot;.t1&quot;&#125;&apos; 0.02.CytoscapeInput-nodes-turquoise.txt  &gt; 0.02.gene.id</span><br></pre></td></tr></table></figure>

<p>![image-20200701084031314](/Users/sunhaoyun/Library/Application Support/typora-user-images/image-20200701084031314.png)</p>
<p>![image-20200701085601228](/Users/sunhaoyun/Library/Application Support/typora-user-images/image-20200701085601228.png)</p>
<p>To identify the key modules co-expressed with FT, we calculated themodule-trait association and ranked genes by their correlation with the FPKM value of PmFT.Finally,top 50 candidate genes (R2 &gt; 0.6) co-expressed with PmFTand 11 putative FTinteracting factors inArabidopsisflowering pathway </p>
<p>by “WGCNA” R package (version 1.4.9) [<a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-018-1526-8#ref-CR55" target="_blank" rel="noopener">55</a>], the parameters were set as follows: The power was 9; the minModuleSize was 30 and the cutHeight was 0.25. Finally, we visualized the sub-network using Cytoscape (version 3.4.0) program [<a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-018-1526-8#ref-CR56" target="_blank" rel="noopener">56</a>]. An additional investigation of the putative functions of <em>GhSK</em> genes and their co-expression genes was based on Gene Ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) enrichment analyses.</p>

	
	</div>
  <a type="button" href="/2020/06/16/WGCNA/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/28/wget_ncbi/" >下载鸟类的基因组2) ncbi</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-28  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h3 id="1-NCBI-的数据比较麻烦"><a href="#1-NCBI-的数据比较麻烦" class="headerlink" title="1. NCBI 的数据比较麻烦"></a>1. NCBI 的数据比较麻烦</h3><p>NCBI中的assemble分为两个数据库 </p>
<p>1个是GCA数据库（genebank） 1个是GCF数据库（refgSeq）</p>
<p>![image-20200528163802536](/Users/sunhaoyun/Library/Application Support/typora-user-images/image-20200528163802536.png)</p>
<p>两个库在数据上是有重叠的，且GCF优于GCA，所以使用时尽量先使用GCF中有的数据GCF的</p>
<p>不过一般很少使用ncbi的数据</p>
<p>A RefSeq (GCF) genome assembly represents an NCBI-derived copy of a submitted GenBank (GCA) assembly. RefSeq (GCF) assembly records are maintained by NCBI. In some cases the RefSeq (GCF) assembly may not be completely identical to the GenBank (GCA) assembly because NCBI staff may (1) remove short sequences or reported contaminants from the assembly or (2) add non-nuclear genome sequences (for example, mitochondrial or chloroplast genomes) to the assembly. All RefSeq (GCF) genome assemblies include annotation. In the majority of cases, this annotation is generated by the NCBI <a href="https://www.ncbi.nlm.nih.gov/genome/annotation_prok/" target="_blank" rel="noopener">prokaryotic</a> or <a href="https://www.ncbi.nlm.nih.gov/genome/annotation_euk/process/" target="_blank" rel="noopener">eukaryotic</a> genome annotation pipelines. In some cases, annotation is provided by the assembly submitter.</p>
<h3 id="2-需要准备的文件"><a href="#2-需要准备的文件" class="headerlink" title="2.  需要准备的文件"></a>2.  需要准备的文件</h3><p>#### </p>
<p><a href="https://www.ncbi.nlm.nih.gov/genome/?term=Aves" target="_blank" rel="noopener">NCBI ave</a></p>
<p>在搜索结果中能看到320个物种的相关assembly信息。</p>
<p>其中has annotation的117个</p>
<p>那就先把这117个下载下来</p>
<h4 id="分析ftp的构成"><a href="#分析ftp的构成" class="headerlink" title="分析ftp的构成"></a>分析ftp的构成</h4><blockquote>
<p><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/247/815/GCA_000247815.2_FicAlb1.5" target="_blank" rel="noopener">ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/247/815/GCA_000247815.2_FicAlb1.5</a></p>
</blockquote>
<p>GCA_000247815.2</p>
<p>发现地址中解构了GCA id 或者GCF id。同时还添加了 FicAlb1.5 物种的版本名称（在后面发现这个pattern其实也可能没有规律）</p>
<p>写了个爬虫去爬ncbi 但总是time out 不知道学校的网有什么鬼。而且他这个带着物种的版本名称的页面 竟然没有index</p>
<p>所以 把这些结果全选复制粘贴下来。。</p>
<p>保存为list1</p>
<p>格式如下</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line">1.</span><br><span class="line">FicAlb1.5</span><br><span class="line"></span><br><span class="line">Organism: Ficedula albicollis (Collared flycatcher)</span><br><span class="line">Sex: male</span><br><span class="line">Submitter: Uppsala University</span><br><span class="line">Date: 2013/06/27</span><br><span class="line">Assembly level: Chromosome</span><br><span class="line">Genome representation: full</span><br><span class="line">RefSeq category: representative genome</span><br><span class="line">GenBank assembly accession: GCA_000247815.2 (latest)</span><br><span class="line">RefSeq assembly accession: GCF_000247815.1 (latest)</span><br><span class="line">IDs: 37801 [UID] 716688 [GenBank] 726138 [RefSeq]</span><br><span class="line">Item in clipboard</span><br><span class="line"></span><br><span class="line">Select item 1116181</span><br><span class="line">2.</span><br><span class="line">NumMel1.0</span><br><span class="line"></span><br><span class="line">Organism: Numida meleagris (helmeted guineafowl)</span><br><span class="line">Infraspecific name: Breed: g44 Domestic line</span><br><span class="line">Sex: male</span><br><span class="line">Submitter: INRA</span><br><span class="line">Date: 2017/06/01</span><br><span class="line">Assembly level: Chromosome</span><br><span class="line">Genome representation: full</span><br><span class="line">RefSeq category: representative genome</span><br><span class="line">GenBank assembly accession: GCA_002078875.2 (latest)</span><br><span class="line">RefSeq assembly accession: GCF_002078875.1 (latest)</span><br><span class="line">IDs: 1116181 [UID] 4541828 [GenBank] 4550868 [RefSeq]</span><br><span class="line">Item in clipboard</span><br></pre></td></tr></table></figure>

<p>对这个文件进行过滤 至提取出来有用的信息。。比如包含GCF id的就提取 name 和 GCF id 不包含GCF id的就提取name 和GCA id</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"># 提取GCF id</span><br><span class="line">grep &apos;GCF&apos; list | awk &apos;&#123;print $4&#125;&apos; &gt; fore.name</span><br><span class="line"># 提取序号行和序号行的下一行</span><br><span class="line">grep &apos;^[0-9]&apos; list -A 1 &gt; temp1</span><br><span class="line"># 提取字母开头的行</span><br><span class="line">awk &apos;&#123;if ($0 ~ &quot;^[a-Z]&quot;)print&#125;&apos;  temp1 &gt; midlle.name</span><br><span class="line"># 生成 GCF id对应物种partern的文件 final name</span><br><span class="line">paste -d &apos;\t&apos; fore.name final.list</span><br><span class="line"># 格式如下</span><br><span class="line">GCA_000247815.2 FicAlb1.5</span><br><span class="line">GCA_002078875.2 NumMel1.0</span><br></pre></td></tr></table></figure>

<p>然后重新排列组合这些pattern 就可以得到 wget的运行文件了</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> sys,re</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">GCF = <span class="string">"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF"</span></span><br><span class="line">genome = <span class="string">"fna.gz"</span></span><br><span class="line">annotate =  <span class="string">"gff.gz"</span></span><br><span class="line">cds = <span class="string">"cds*"</span></span><br><span class="line">protein = <span class="string">"protein.faa.gz"</span></span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> tfile:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> tfile:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        head = lst1[<span class="number">0</span>][<span class="number">4</span>:<span class="number">7</span>]</span><br><span class="line">        middle = lst1[<span class="number">0</span>][<span class="number">7</span>:<span class="number">10</span>]</span><br><span class="line">        tail = lst1[<span class="number">0</span>][<span class="number">10</span>:<span class="number">13</span>]</span><br><span class="line">        f1 = lst1[<span class="number">0</span>]</span><br><span class="line">        r2 = lst1[<span class="number">1</span>]</span><br><span class="line">        genome_ftp = <span class="string">"wget -c "</span>+GCF+<span class="string">'/'</span>+head+<span class="string">'/'</span>+middle+<span class="string">'/'</span>+tail+<span class="string">'/'</span>+f1+<span class="string">'_'</span>+r2.strip()+<span class="string">'/*'</span>+genome</span><br><span class="line">        annotate_ftp = <span class="string">"wget -c "</span>+GCF+<span class="string">'/'</span>+head+<span class="string">'/'</span>+middle+<span class="string">'/'</span>+tail+<span class="string">'/'</span>+f1+<span class="string">'_'</span>+r2.strip()+<span class="string">'/*'</span>+annotate</span><br><span class="line">        cds_ftp = <span class="string">"wget -c "</span>+GCF+<span class="string">'/'</span>+head+<span class="string">'/'</span>+middle+<span class="string">'/'</span>+tail+<span class="string">'/'</span>+f1+<span class="string">'_'</span>+r2.strip()+<span class="string">'/*'</span>+cds</span><br><span class="line">        protein_ftp = <span class="string">"wget -c "</span>+GCF+<span class="string">'/'</span>+head+<span class="string">'/'</span>+middle+<span class="string">'/'</span>+tail+<span class="string">'/'</span>+f1+<span class="string">'_'</span>+r2.strip()+<span class="string">'/*'</span>+protein</span><br><span class="line">        print(genome_ftp)</span><br><span class="line">        print(annotate_ftp)</span><br><span class="line">        print(cds_ftp)</span><br><span class="line">        print(protein_ftp)</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/05/28/wget_ncbi/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/28/wget_giga/" >下载鸟类的基因组2) giga</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-28  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>###1. 首先是张郭杰老师他们的48个鸟类基因组，他们全都放在一个project文件夹下了 十分方便</p>
<p><a href="http://gigadb.org/dataset/101000" target="_blank" rel="noopener">project 地址</a></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/100001_101000/101000/gene_and_repeat_annotation/coding_sequences/cds/*.cds.gz</span><br><span class="line"></span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/100001_101000/101000/assembly/*.fa.gz</span><br><span class="line"></span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/100001_101000/101000/gene_and_repeat_annotation/gff/*gff.gz</span><br><span class="line"></span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/100001_101000/101000/gene_and_repeat_annotation/coding_sequences/pep/*pep.gz</span><br></pre></td></tr></table></figure>

<h3 id="2-然后是有五个天堂鸟的基因组"><a href="#2-然后是有五个天堂鸟的基因组" class="headerlink" title="2. 然后是有五个天堂鸟的基因组"></a>2. 然后是有五个天堂鸟的基因组</h3><p>只需要根据最后网址提供的doi循环输出就好</p>
<p><a href="http://gigadb.org/dataset/100521" target="_blank" rel="noopener">project 地址</a></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102157/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102158/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102159/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102160/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102161/*</span><br></pre></td></tr></table></figure>

<h3 id="3-然后是19个企鹅的基因组"><a href="#3-然后是19个企鹅的基因组" class="headerlink" title="3. 然后是19个企鹅的基因组"></a>3. 然后是19个企鹅的基因组</h3><p><a href="http://gigadb.org/dataset/100649" target="_blank" rel="noopener">project地址</a></p>
<p>同样只需要修改doi</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102164/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102165/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102166/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102167/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102168/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102169/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102170/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102171/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102172/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102173/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102174/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102175/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102176/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102177/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102178/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102179/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102180/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102181/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102182/*</span><br></pre></td></tr></table></figure>

<h4 id="所有giga的项目都支持md5sum十分方便"><a href="#所有giga的项目都支持md5sum十分方便" class="headerlink" title="所有giga的项目都支持md5sum十分方便"></a>所有giga的项目都支持md5sum十分方便</h4>
	
	</div>
  <a type="button" href="/2020/05/28/wget_giga/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/25/wget_Ensembl/" >下载鸟类的基因组1) Ensembl</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-25  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="如何从ENSEMBL数据库-下载里面所有鸟类的-基因组-蛋白序列-cds序列-和gff3注释文件"><a href="#如何从ENSEMBL数据库-下载里面所有鸟类的-基因组-蛋白序列-cds序列-和gff3注释文件" class="headerlink" title="如何从ENSEMBL数据库 下载里面所有鸟类的 基因组 蛋白序列 cds序列 和gff3注释文件"></a>如何从ENSEMBL数据库 下载里面所有鸟类的 基因组 蛋白序列 cds序列 和gff3注释文件</h1><p>去年下过一次 但是当时没有存下过程<br>首先进入Ensmbl数据库下载 all Species list (首页给出的Sp list 只有动物的)<br>然后根据图片和common name挑选出哪些是鸟类，并把前两列存在一个新的txt文件中</p>
<p>38bird.ensembl.list.txt</p>
<p>然后观察ftp网址发现，只需要有scientific name就可以下载，但是所有字母都为小写，且string之间有下划线</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">sed -i &apos;&apos; &apos;s/ /_/g&apos; 38bird.ensembl.list.txt</span><br><span class="line">awk &apos;&#123;print $2&#125;&apos; 38bird.ensembl.list.txt &gt; specific_name.list</span><br><span class="line">cat scientific_name.list | tr  &apos;[A-Z]&apos; &apos;[a-z]&apos; &gt; xiaoxie.scientific.list</span><br></pre></td></tr></table></figure>

<p>然后就可以根据 xiaoxie.scientific.list 循环输出wget下载命令,用wget -c 可以续传，因为下载速度实在太慢了</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">less xiaoxie.scientific.list | while read line</span><br><span class="line">do</span><br><span class="line">echo &apos;wget -c ftp://ftp.ensembl.org/pub/release-100/fasta/&apos;$&#123;line&#125;&apos;/cds/*.cds.all.fa.gz&apos;</span><br><span class="line">done &gt; download_cds.sh</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">less xiaoxie.scientific.list | while read line</span><br><span class="line">do</span><br><span class="line">echo &apos;wget -c  ftp://ftp.ensembl.org/pub/release-100/gff3/&apos;$&#123;line&#125;&apos;/*.100.gff3.gz&apos;</span><br><span class="line">done &gt; download_gff3.sh</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">less xiaoxie.scientific.list | while read line</span><br><span class="line">do</span><br><span class="line">echo &apos;wget -c  ftp://ftp.ensembl.org/pub/release-100/fasta/&apos;$&#123;line&#125;&apos;/pep/*.100.protein.gz&apos;</span><br><span class="line">done &gt; download_protein.sh</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">less xiaoxie.scientific.list | while read line</span><br><span class="line">do</span><br><span class="line">echo &apos;wget -c ftp://ftp.ensembl.org/pub/release-100/fasta/&apos;$&#123;line&#125;&apos;/dna/*.dna_sm.toplevel.fa.gz&apos;</span><br><span class="line">done &gt; download_genome.sh</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">nohup sh *.sh &gt; run_wget.log</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/05/25/wget_Ensembl/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/25/Ensembl_filter/" >过滤Ensembl中数据的冗余信息</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-25  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h2 id="Ensembl-格式介绍"><a href="#Ensembl-格式介绍" class="headerlink" title="Ensembl 格式介绍"></a>Ensembl 格式介绍</h2><p>u1s1 真的佩服Emsembl数据库的开发团队，十分想问候他们怎么做的这么详细的</p>
<hr>
<p>sudo killall coreaudiod</p>
<p>Ensembl 数据库格式十分简洁明了，每一个基因都有对应的 gene id, 每一个转录本有对应的<br>trasncript id, 每一个蛋白有对应的 protein id, 而且这三个id之间。。没有规律</p>
<p>比如 这是鸡中的一个蛋白序列的简化stable的信息</p>
<blockquote>
<p>ENSGALG00000009622    ENSGALT00000032268     ENSGALP00000031632<br>其实在真实的序列中，也不是上面的样子而是如下这样包含了stableeversion的</p>
<blockquote>
<p>ENSGALP00000073247.1 pep chromosome:GRCg6a:15:8238927:8239208:1<br>gene:ENSGALG00000049010.1 transcript:ENSGALT00000103863.1<br>gene_biotype:IG_V_gene transcript_biotype:IG_V_gene</p>
</blockquote>
</blockquote>
<p>在biomart中 除了stable id 还有很多label .. awesome～</p>
<h3 id="我想要得到蛋白序列中同一个gene-id中最长的蛋白序列"><a href="#我想要得到蛋白序列中同一个gene-id中最长的蛋白序列" class="headerlink" title="我想要得到蛋白序列中同一个gene id中最长的蛋白序列"></a>我想要得到蛋白序列中同一个gene id中最长的蛋白序列</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$PATH:OrthoFinder/tools/primary_transcript.py</span><br></pre></td></tr></table></figure>

<p>orthofinder 中已经给出了怎么提取最长转录本的python脚本。并且最长的转录本的id被替换为gene id输出</p>
<p>提蛋白真的很方便，但是，老师还需要对应基因的intron序列<br>intron序列就需要gff文件中的exon信息和mRNA信息</p>
<p>怎么通过带着gene id的最长转录本去过滤gff文件呢。</p>
<blockquote>
<p>经过一系列的测试。。比如带gene id的蛋白序列blastp回去找到transcript id。但是blast的算法导致 multi match的情况不好去除</p>
</blockquote>
<p>所以还是得写个脚本</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"><span class="keyword">import</span> re</span><br><span class="line"><span class="keyword">from</span> operator <span class="keyword">import</span> itemgetter</span><br><span class="line"></span><br><span class="line">lst = []</span><br><span class="line">Sequence = &#123;&#125;</span><br><span class="line"></span><br><span class="line">transcript2gene = &#123;&#125;</span><br><span class="line">pep2gene = &#123;&#125;</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_emsembl_fasta:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_emsembl_fasta:</span><br><span class="line">        <span class="keyword">if</span> row.startswith(<span class="string">'&gt;'</span>):</span><br><span class="line">            lst = row.split(<span class="string">" "</span>)</span><br><span class="line">            pep_id = lst[<span class="number">0</span>].replace(<span class="string">'&gt;'</span>,<span class="string">''</span>)</span><br><span class="line">            gene_id = lst[<span class="number">3</span>].replace(<span class="string">'gene:'</span>,<span class="string">''</span>)</span><br><span class="line">            transcript_id = lst[<span class="number">4</span>].replace(<span class="string">'transcript:'</span>,<span class="string">''</span>)</span><br><span class="line">            Sequence[transcript_id] = <span class="string">''</span></span><br><span class="line">            transcript2gene[transcript_id] = gene_id</span><br><span class="line">            pep2gene[transcript_id] = pep_id</span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            Sequence[transcript_id] += row.strip()</span><br><span class="line"><span class="keyword">with</span> open(<span class="string">"t1.temp"</span>,<span class="string">'w'</span>) <span class="keyword">as</span> Out:</span><br><span class="line">    <span class="keyword">for</span> i <span class="keyword">in</span> Sequence.keys():</span><br><span class="line">        Out.write(i+<span class="string">"\t"</span>+str(len(Sequence[i]))+<span class="string">"\t"</span>+transcript2gene[i]+<span class="string">'\t'</span>+pep2gene[i]+<span class="string">"\n"</span>)</span><br><span class="line"></span><br><span class="line"><span class="keyword">from</span> operator <span class="keyword">import</span> itemgetter</span><br><span class="line">gene_id_len=open(<span class="string">"t1.temp"</span>,<span class="string">"r"</span>)</span><br><span class="line">table = []</span><br><span class="line"><span class="keyword">for</span> line <span class="keyword">in</span> gene_id_len:</span><br><span class="line">    col = line.strip().split(<span class="string">"\t"</span>)</span><br><span class="line">    col[<span class="number">0</span>] = str(col[<span class="number">0</span>])</span><br><span class="line">    col[<span class="number">1</span>] = int(col[<span class="number">1</span>])</span><br><span class="line">    col[<span class="number">2</span>] = str(col[<span class="number">2</span>])</span><br><span class="line">    col[<span class="number">3</span>] = str(col[<span class="number">3</span>])</span><br><span class="line">    table.append(col)</span><br><span class="line"><span class="comment">#print(table)</span></span><br><span class="line">table_sorted = sorted(table, key=itemgetter(<span class="number">2</span>, <span class="number">1</span>),reverse=<span class="literal">True</span>)</span><br><span class="line"><span class="comment">#print(table_sorted)</span></span><br><span class="line">output_file = open(<span class="string">"t2.temp"</span>,<span class="string">"w"</span>)</span><br><span class="line"><span class="keyword">for</span> row <span class="keyword">in</span> table_sorted:</span><br><span class="line">    row = [str(x) <span class="keyword">for</span> x <span class="keyword">in</span> row]</span><br><span class="line">    output_file.write(<span class="string">"\t"</span>.join(row) + <span class="string">'\n'</span>)</span><br><span class="line">output_file.close()</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">input_file=open(<span class="string">"t2.temp"</span>,<span class="string">"r"</span>)</span><br><span class="line">dict2=&#123;&#125;</span><br><span class="line"><span class="keyword">for</span> line <span class="keyword">in</span> input_file.readlines():</span><br><span class="line">    col = line.strip().split(<span class="string">"\t"</span>)</span><br><span class="line">    col[<span class="number">0</span>] = str(col[<span class="number">0</span>])</span><br><span class="line">    col[<span class="number">1</span>] = int(col[<span class="number">1</span>])</span><br><span class="line">    col[<span class="number">2</span>] = str(col[<span class="number">2</span>])</span><br><span class="line">    col[<span class="number">3</span>] = str(col[<span class="number">3</span>])</span><br><span class="line">    <span class="keyword">if</span> col[<span class="number">2</span>] <span class="keyword">not</span> <span class="keyword">in</span> dict2:</span><br><span class="line">        dict2[col[<span class="number">2</span>]]=col[<span class="number">0</span>]</span><br><span class="line">    <span class="keyword">else</span>:</span><br><span class="line">        dict2[col[<span class="number">2</span>]]+= <span class="string">'\t'</span>+col[<span class="number">0</span>]</span><br><span class="line"><span class="comment">#print(dict2)</span></span><br><span class="line"><span class="comment">#print(dict2.values())</span></span><br><span class="line">list_values=list(dict2.values())</span><br><span class="line"><span class="comment">#print(list_values)</span></span><br><span class="line">result_file = open(<span class="string">"final.fa"</span>,<span class="string">"w"</span>)</span><br><span class="line"><span class="keyword">for</span> line <span class="keyword">in</span> list_values:</span><br><span class="line">    col = line.strip().split(<span class="string">"\t"</span>)</span><br><span class="line">    col[<span class="number">0</span>] = str(col[<span class="number">0</span>])</span><br><span class="line">    <span class="comment">#print(col[0])</span></span><br><span class="line"></span><br><span class="line">    result_file.write(col[<span class="number">0</span>]+<span class="string">"\n"</span>)</span><br><span class="line">result_file.close()</span><br></pre></td></tr></table></figure>

<p>专为Ensembl设计，结果会保存最长转录本的 transcript id， gene id， pep id</p>
<p>然后再过滤gff</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">import sys</span><br><span class="line"></span><br><span class="line">ab = &#123;&#125;</span><br><span class="line">with open(sys.argv[1]) as  rename_file:</span><br><span class="line">    for row in rename_file:</span><br><span class="line">        lst1 = row.strip().split(&apos;\t&apos;)</span><br><span class="line">        ab[lst1[0]] = lst1[1]</span><br><span class="line"></span><br><span class="line">with open(sys.argv[2]) as gff_file:</span><br><span class="line">    for row in gff_file:</span><br><span class="line">	for key,value in ab.items():</span><br><span class="line">       	    if &quot;ID=gene:&quot;+key in row:</span><br><span class="line">                print(row.strip())</span><br><span class="line">            elif value in row:</span><br><span class="line">                print(row.strip())</span><br></pre></td></tr></table></figure>

<p>这样就得到了只包含最长转录本的gff文件</p>
<p>再用张老师的提取intron的脚本就可以得到intron文件</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#!/usr/bin/perl -w</span></span><br><span class="line"><span class="keyword">use</span> warnings;</span><br><span class="line"><span class="keyword">use</span> strict;</span><br><span class="line"><span class="keyword">use</span> Set::IntSpan;</span><br><span class="line"><span class="keyword">use</span> Bio::DB::Fasta;</span><br><span class="line"></span><br><span class="line"><span class="comment"># date, 2016-12-30, zhanglei</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">my</span> $ref       = $ARGV[<span class="number">0</span>];</span><br><span class="line"><span class="keyword">my</span> $gene2pep  = $ARGV[<span class="number">1</span>];</span><br><span class="line"><span class="keyword">my</span> $gene2tran = $ARGV[<span class="number">2</span>];</span><br><span class="line"><span class="keyword">my</span> $gtf       = $ARGV[<span class="number">3</span>];</span><br><span class="line"><span class="keyword">my</span> $out       = $ARGV[<span class="number">4</span>];</span><br><span class="line"><span class="keyword">die</span> <span class="string">"Usage: $0 ref.fa gene2pep gene2transcript file.gff3 out.fa</span></span><br><span class="line"><span class="string">This script is used to extract intron from the gff3 file(NOT GTF file!)\n"</span> <span class="keyword">if</span> @ARGV &lt; <span class="number">5</span>;</span><br><span class="line"></span><br><span class="line"><span class="keyword">print</span> <span class="string">"Loading the ID file\t"</span>;</span><br><span class="line"><span class="keyword">my</span> %gene2peps;</span><br><span class="line"><span class="keyword">my</span> %trans2gene;</span><br><span class="line"><span class="keyword">open</span> IN, <span class="string">"&lt;$gene2pep"</span>;</span><br><span class="line"><span class="keyword">while</span> (&lt;IN&gt;) &#123;</span><br><span class="line">	<span class="keyword">chomp</span>;</span><br><span class="line">	<span class="keyword">my</span> @array = (<span class="keyword">split</span> <span class="regexp">/\s+/</span>);</span><br><span class="line">	$gene2peps&#123;$array[<span class="number">0</span>]&#125; = $array[<span class="number">1</span>];</span><br><span class="line">&#125;</span><br><span class="line"><span class="keyword">close</span> IN;</span><br><span class="line"></span><br><span class="line"><span class="keyword">open</span> IN, <span class="string">"&lt;$gene2tran"</span>;</span><br><span class="line"><span class="keyword">while</span> (&lt;IN&gt;) &#123;</span><br><span class="line">	<span class="keyword">chomp</span>;</span><br><span class="line">	<span class="keyword">my</span> @array = (<span class="keyword">split</span> <span class="regexp">/\s+/</span>);</span><br><span class="line">	$trans2gene&#123;$array[<span class="number">1</span>]&#125; = $array[<span class="number">0</span>];</span><br><span class="line">&#125;</span><br><span class="line"><span class="keyword">close</span> IN;</span><br><span class="line"></span><br><span class="line"><span class="keyword">print</span> <span class="string">"Extracting gff3 file\t"</span>;</span><br><span class="line"><span class="comment"># extract gtf</span></span><br><span class="line"><span class="keyword">open</span> IN, <span class="string">"&lt;$gtf"</span>;</span><br><span class="line"><span class="keyword">my</span> @lines;</span><br><span class="line"><span class="keyword">while</span> (&lt;IN&gt;) &#123;</span><br><span class="line">	<span class="keyword">my</span> @array = (<span class="keyword">split</span> <span class="regexp">/\s+/</span>, $_);</span><br><span class="line">	<span class="keyword">if</span> ($array[<span class="number">2</span>] eq <span class="string">'mRNA'</span> || $array[<span class="number">2</span>] eq <span class="string">'exon'</span>) &#123;</span><br><span class="line">		<span class="keyword">push</span> @lines, $_;</span><br><span class="line">	&#125;</span><br><span class="line">&#125;</span><br><span class="line"><span class="keyword">close</span> IN;</span><br><span class="line"></span><br><span class="line"><span class="keyword">print</span> <span class="string">"done\n"</span>;</span><br><span class="line"><span class="keyword">print</span> <span class="string">"Formatdb $ref\t"</span>;</span><br><span class="line"><span class="comment"># sequence DB</span></span><br><span class="line"><span class="keyword">my</span> $ref_db = Bio::DB::Fasta -&gt; new($ref);</span><br><span class="line"><span class="keyword">print</span> <span class="string">"done\n"</span>;</span><br><span class="line"><span class="keyword">print</span> <span class="string">"Get gene id from gff3\t"</span>;</span><br><span class="line"><span class="comment"># load the gff file</span></span><br><span class="line"><span class="keyword">open</span> OUT, <span class="string">"&gt;$out"</span>;</span><br><span class="line"><span class="keyword">my</span> %hash;</span><br><span class="line"><span class="keyword">foreach</span> (@lines) &#123;</span><br><span class="line">	<span class="keyword">chomp</span>;</span><br><span class="line">	<span class="keyword">next</span> <span class="keyword">if</span> /^<span class="comment">#/;</span></span><br><span class="line">	<span class="keyword">next</span> <span class="keyword">if</span> /^\<span class="keyword">s</span>+<span class="regexp">/;</span></span><br><span class="line"><span class="regexp">	my @cols = (split /</span>\<span class="keyword">s</span>+<span class="regexp">/, $_);</span></span><br><span class="line"><span class="regexp">	my $gene_id = my $trans_id = my $pep_id;</span></span><br><span class="line"><span class="regexp">	if ($cols[2] eq 'mRNA') &#123;</span></span><br><span class="line"><span class="regexp">		($gene_id) = (/</span>Parent\=gene\:(\w+);<span class="regexp">/);   #Parent=gene:ENSGALG00000053455</span></span><br><span class="line"><span class="regexp">		($trans_id) = (/</span>ID\=transcript\:(\w+);<span class="regexp">/);</span></span><br><span class="line"><span class="regexp">#		print "mRNA $gene_id $trans_id\n";</span></span><br><span class="line"><span class="regexp">	&#125;</span></span><br><span class="line"><span class="regexp">	elsif ($cols[2] eq 'exon') &#123;</span></span><br><span class="line"><span class="regexp">		($trans_id) = (/</span>Parent\=transcript\:(\w+);<span class="regexp">/);  #Parent=transcript:</span></span><br><span class="line"><span class="regexp">		$gene_id = $trans2gene&#123;$trans_id&#125;;</span></span><br><span class="line"><span class="regexp">#		print "exon $gene_id $trans_id\n";</span></span><br><span class="line"><span class="regexp">	&#125;</span></span><br><span class="line"><span class="regexp">	$pep_id = $gene2peps&#123;$gene_id&#125;;</span></span><br><span class="line"><span class="regexp"></span></span><br><span class="line"><span class="regexp">	next unless $gene_id;</span></span><br><span class="line"><span class="regexp">	# chr, strand, gene start and end</span></span><br><span class="line"><span class="regexp">	# chr, strand, gene start and end</span></span><br><span class="line"><span class="regexp">	if ($cols[2] eq 'mRNA' ) &#123;</span></span><br><span class="line"><span class="regexp">		$hash&#123;$gene_id&#125;&#123;chr&#125; = $cols[0];</span></span><br><span class="line"><span class="regexp">		$hash&#123;$gene_id&#125;&#123;strand&#125; = $cols[6];</span></span><br><span class="line"><span class="regexp">		$hash&#123;$gene_id&#125;&#123;start&#125; = $cols[3];</span></span><br><span class="line"><span class="regexp">		$hash&#123;$gene_id&#125;&#123;end&#125; = $cols[4];</span></span><br><span class="line"><span class="regexp">		$hash&#123;$gene_id&#125;&#123;ex_num&#125; = 0;</span></span><br><span class="line"><span class="regexp">		$hash&#123;$gene_id&#125;&#123;pep_id&#125; = $pep_id;</span></span><br><span class="line"><span class="regexp">		$hash&#123;$gene_id&#125;&#123;trans_id&#125; = $trans_id;</span></span><br><span class="line"><span class="regexp">		$hash&#123;$gene_id&#125;&#123;exon&#125; = new Set::IntSpan;</span></span><br><span class="line"><span class="regexp">	&#125;</span></span><br><span class="line"><span class="regexp">	# exon number</span></span><br><span class="line"><span class="regexp">	if ($cols[2] eq 'exon' &amp;&amp; exists $hash&#123;$gene_id&#125;) &#123;</span></span><br><span class="line"><span class="regexp">		my ( $start, $end ) = @cols[3, 4];</span></span><br><span class="line"><span class="regexp">		$hash&#123;$gene_id&#125;&#123;exon&#125;-&gt;U("$start-$end");</span></span><br><span class="line"><span class="regexp">	&#125;</span></span><br><span class="line"><span class="regexp">&#125;</span></span><br><span class="line"><span class="regexp">print "done\n";</span></span><br><span class="line"><span class="regexp">print "Extract introns from $ref\t";</span></span><br><span class="line"><span class="regexp"># get the exon sequence</span></span><br><span class="line"><span class="regexp">foreach my $gene_id (sort keys %hash) &#123;</span></span><br><span class="line"><span class="regexp">	my $chr = $hash&#123;$gene_id&#125;&#123;chr&#125;;</span></span><br><span class="line"><span class="regexp">	my $strand = $hash&#123;$gene_id&#125;&#123;strand&#125;;</span></span><br><span class="line"><span class="regexp">	my $start = $hash&#123;$gene_id&#125;&#123;start&#125;;</span></span><br><span class="line"><span class="regexp">	my $end   = $hash&#123;$gene_id&#125;&#123;end&#125;;</span></span><br><span class="line"><span class="regexp">	my $pep_id = $hash&#123;$gene_id&#125;&#123;pep_id&#125;;</span></span><br><span class="line"><span class="regexp">	my $trans_id = $hash&#123;$gene_id&#125;&#123;trans_id&#125;;</span></span><br><span class="line"><span class="regexp"></span></span><br><span class="line"><span class="regexp">	my $all_set = new Set::IntSpan "$start-$end";</span></span><br><span class="line"><span class="regexp">	my $exon_set = $hash&#123;$gene_id&#125;&#123;exon&#125;;</span></span><br><span class="line"><span class="regexp">	my $intron_set = diff $all_set $exon_set;</span></span><br><span class="line"><span class="regexp">	my @array_intron = sets $intron_set;</span></span><br><span class="line"><span class="regexp"></span></span><br><span class="line"><span class="regexp">	my $seq;</span></span><br><span class="line"><span class="regexp">	# upstream and downstream, 0 bp</span></span><br><span class="line"><span class="regexp">#	my $min = min $set;</span></span><br><span class="line"><span class="regexp">#	$min = $min - 1;</span></span><br><span class="line"><span class="regexp">#	my $s_min = 1 + $min - 0;</span></span><br><span class="line"><span class="regexp">#	my $max = max $set;</span></span><br><span class="line"><span class="regexp">#	$max = $max + 1;</span></span><br><span class="line"><span class="regexp">#	my $e_max = $max - 1 + 0;</span></span><br><span class="line"><span class="regexp">#	my $upstream = $ref_db-&gt;seq($chr, $s_min =&gt; $min);</span></span><br><span class="line"><span class="regexp">#	my $downstream = $ref_db-&gt;seq($chr, $max =&gt; $e_max);</span></span><br><span class="line"><span class="regexp">#	$downstream = $upstream = "";</span></span><br><span class="line"><span class="regexp">#   $seq = "$upstream";</span></span><br><span class="line"><span class="regexp">	# exon region</span></span><br><span class="line"><span class="regexp"></span></span><br><span class="line"><span class="regexp">	@array_intron = reverse @array_intron if $strand eq '-';</span></span><br><span class="line"><span class="regexp">	foreach my $intron (@array_intron) &#123;</span></span><br><span class="line"><span class="regexp">		my ($start, $end) = (split /</span>\-<span class="regexp">/, $intron);</span></span><br><span class="line"><span class="regexp">		$end = $start unless defined $end;</span></span><br><span class="line"><span class="regexp">		my $seq = $ref_db-&gt;seq($chr, $start =&gt; $end);</span></span><br><span class="line"><span class="regexp">		$seq = reverse_complement($seq) if $strand eq '-';</span></span><br><span class="line"><span class="regexp">		my $new_strand;</span></span><br><span class="line"><span class="regexp">		if ($strand eq '-') &#123;</span></span><br><span class="line"><span class="regexp">			$new_strand = "minus_strand";</span></span><br><span class="line"><span class="regexp">		&#125;</span></span><br><span class="line"><span class="regexp">		elsif ($strand eq '+') &#123;</span></span><br><span class="line"><span class="regexp">			$new_strand = "positive_strand";</span></span><br><span class="line"><span class="regexp">		&#125;</span></span><br><span class="line"><span class="regexp">		else &#123;</span></span><br><span class="line"><span class="regexp">			print "Warning $gene_id $strand not defined\n";</span></span><br><span class="line"><span class="regexp">		&#125;</span></span><br><span class="line"><span class="regexp">		print OUT "&gt;Gene_ID:$gene_id#Transcript_ID:$trans_id#Protein_ID:$pep_id#intron#$new_strand#$chr\:$start-$end\n$seq\n";</span></span><br><span class="line"><span class="regexp">	&#125;</span></span><br><span class="line"><span class="regexp">&#125;</span></span><br><span class="line"><span class="regexp">close OUT;</span></span><br><span class="line"><span class="regexp">print "done\n";</span></span><br><span class="line"><span class="regexp">### sub routines ###</span></span><br><span class="line"><span class="regexp">sub reverse_complement &#123;</span></span><br><span class="line"><span class="regexp">        my $dna = shift;</span></span><br><span class="line"><span class="regexp"></span></span><br><span class="line"><span class="regexp">	# reverse the DNA sequence</span></span><br><span class="line"><span class="regexp">        my $revcomp = reverse($dna);</span></span><br><span class="line"><span class="regexp"></span></span><br><span class="line"><span class="regexp">	# complement the reversed DNA sequence</span></span><br><span class="line"><span class="regexp">        $revcomp =~ tr/</span>ACGTacgt/TGCAtgca/;</span><br><span class="line">        <span class="keyword">return</span> $revcomp;</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

<p>cds也就比较好提了。毕竟各个id都有了</p>

	
	</div>
  <a type="button" href="/2020/05/25/Ensembl_filter/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/15/详细注释的OrthoFinder/" >详细一点的Orthofinder，与之前英文的没区别</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-15  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="ortholog-identification"><a href="#ortholog-identification" class="headerlink" title="ortholog identification"></a>ortholog identification</h1><h2 id="filter-Prunella-strophiata"><a href="#filter-Prunella-strophiata" class="headerlink" title="filter Prunella strophiata"></a>filter Prunella strophiata</h2><h3 id="in-order-to-keep-gene-id-format-consistantly-use-awk-and-python-scripts-and-gffread-and-transeq-for-refomating"><a href="#in-order-to-keep-gene-id-format-consistantly-use-awk-and-python-scripts-and-gffread-and-transeq-for-refomating" class="headerlink" title="in order to keep gene id format consistantly, use awk and python scripts and gffread and transeq for refomating"></a>in order to keep gene id format consistantly, use awk and python scripts and gffread and transeq for refomating</h3><h3 id="1-gene-and-mRNA-will-cause-redirect-error-remove-first"><a href="#1-gene-and-mRNA-will-cause-redirect-error-remove-first" class="headerlink" title="1. gene and mRNA will cause redirect error, remove first"></a>1. gene and mRNA will cause redirect error, remove first</h3><h4 id="因为原始的Prunella-strophiata的注释文件中，gene-id-的格式如下"><a href="#因为原始的Prunella-strophiata的注释文件中，gene-id-的格式如下" class="headerlink" title="因为原始的Prunella strophiata的注释文件中，gene id 的格式如下"></a>因为原始的Prunella strophiata的注释文件中，gene id 的格式如下</h4><blockquote>
<p>contig1000      EVM     gene    23218   23970   .       +       .       ID=evm.TU.contig1000.2;Name=EVM%20prediction%20contig1000.2<br>contig1000      EVM     mRNA    23218   23970   .       +       .       ID=evm.model.contig1000.2;Parent=evm.TU.contig1000.2;Name=EVM%20predicti<br>on%20contig1000.2<br>contig1000      EVM     exon    23218   23970   .       +       .       ID=evm.model.contig1000.2.exon1;Parent=evm.model.contig1000.2<br>contig1000      EVM     CDS     23218   23970   .       +       0       ID=cds.evm.model.contig1000.2;Parent=evm.model.contig1000.2</p>
</blockquote>
<h4 id="但是为了在最终结果的-Orthogroups-中能清楚的看出来group中包含哪些物种，所以把注释文件中的-gene-id-替换为物种名加序号的格式"><a href="#但是为了在最终结果的-Orthogroups-中能清楚的看出来group中包含哪些物种，所以把注释文件中的-gene-id-替换为物种名加序号的格式" class="headerlink" title="但是为了在最终结果的 Orthogroups 中能清楚的看出来group中包含哪些物种，所以把注释文件中的 gene id 替换为物种名加序号的格式"></a>但是为了在最终结果的 Orthogroups 中能清楚的看出来group中包含哪些物种，所以把注释文件中的 gene id 替换为物种名加序号的格式</h4><h4 id="如Prunella-strophiata中的基因，需要修改命名为-Prunella-strophiata-best-assembled-1-Prunella-strophiata-best-assembled-2"><a href="#如Prunella-strophiata中的基因，需要修改命名为-Prunella-strophiata-best-assembled-1-Prunella-strophiata-best-assembled-2" class="headerlink" title="如Prunella strophiata中的基因，需要修改命名为 Prunella_strophiata_best_assembled_1, Prunella_strophiata_best_assembled_2"></a>如Prunella strophiata中的基因，需要修改命名为 Prunella_strophiata_best_assembled_1, Prunella_strophiata_best_assembled_2</h4><h4 id="因为gene和的信息内容是一致的，使用下面的命令先去除包含mrna的行"><a href="#因为gene和的信息内容是一致的，使用下面的命令先去除包含mrna的行" class="headerlink" title="因为gene和的信息内容是一致的，使用下面的命令先去除包含mrna的行"></a>因为gene和的信息内容是一致的，使用下面的命令先去除包含mrna的行</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">awk '&#123;if($3 != "mRNA")print $0&#125;' Prunella_strophiata_assemble.evm.gff3  &gt; Prunella_strophiata_assemble.remove_mRNA.gff3</span><br></pre></td></tr></table></figure>

<h3 id="2-replace-gene-id"><a href="#2-replace-gene-id" class="headerlink" title="2. replace gene id"></a>2. replace gene id</h3><h4 id="使用-reformat-py2-脚本去替换原始的gff3文件中的-gene-id"><a href="#使用-reformat-py2-脚本去替换原始的gff3文件中的-gene-id" class="headerlink" title="使用 reformat.py2 脚本去替换原始的gff3文件中的 gene id"></a>使用 reformat.py2 脚本去替换原始的gff3文件中的 gene id</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python reformat.py2 Prunella_strophiata_assemble.remove_mRNA.gff3 &gt;  Prunella_strophiata.best_assembled.gff3</span><br></pre></td></tr></table></figure>

<h4 id="修改后的注释文件如下-差别只在最后一列，这样输出还有一个目的是为了gffread软件的读取"><a href="#修改后的注释文件如下-差别只在最后一列，这样输出还有一个目的是为了gffread软件的读取" class="headerlink" title="修改后的注释文件如下,差别只在最后一列，这样输出还有一个目的是为了gffread软件的读取"></a>修改后的注释文件如下,差别只在最后一列，这样输出还有一个目的是为了gffread软件的读取</h4><blockquote>
<p>contig1000      EVM     gene    23218   23970   .       +       .       ID=Prunella_strophita_best_assembled_1<br>contig1000      EVM     exon    23218   23970   .       +       .       Parent=Prunella_strophita_best_assembled_1<br>contig1000      EVM     CDS     23218   23970   .       +       0       Parent=Prunella_strophita_best_assembled_1</p>
</blockquote>
<h3 id="3-extract-cds-sequence"><a href="#3-extract-cds-sequence" class="headerlink" title="3. extract cds sequence"></a>3. extract cds sequence</h3><h4 id="使用-gffread-软件，输入修改过的gff文件和基因组序列，提取出其中的cds序列，gffread在提取的时候会识别第七列的正负链信息取反向互补"><a href="#使用-gffread-软件，输入修改过的gff文件和基因组序列，提取出其中的cds序列，gffread在提取的时候会识别第七列的正负链信息取反向互补" class="headerlink" title="使用 gffread 软件，输入修改过的gff文件和基因组序列，提取出其中的cds序列，gffread在提取的时候会识别第七列的正负链信息取反向互补"></a>使用 gffread 软件，输入修改过的gff文件和基因组序列，提取出其中的cds序列，gffread在提取的时候会识别第七列的正负链信息取反向互补</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gffread Prunella_strophiata.best_assembled.gff3 -g Prunella_strophiata_assemble.fa -x Prunella_strophiata.best_assembled.cds.fa</span><br></pre></td></tr></table></figure>

<h3 id="4-cds2protein"><a href="#4-cds2protein" class="headerlink" title="4. cds2protein"></a>4. cds2protein</h3><h4 id="使用EMBOSS数据包中的-transeq-程序对上一步提取出来的cds序列进行翻译"><a href="#使用EMBOSS数据包中的-transeq-程序对上一步提取出来的cds序列进行翻译" class="headerlink" title="使用EMBOSS数据包中的 transeq 程序对上一步提取出来的cds序列进行翻译"></a>使用EMBOSS数据包中的 transeq 程序对上一步提取出来的cds序列进行翻译</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">transeq -sequence Prunella_strophiata.best_assembled.cds.fa -outseq Prunella_strophaita.best_assembled.pep.fa</span><br></pre></td></tr></table></figure>

<h3 id="5-remove-the-info-added-by-transeq"><a href="#5-remove-the-info-added-by-transeq" class="headerlink" title="5. remove the info added by transeq"></a>5. remove the info added by transeq</h3><h4 id="因为EMBOSS-中的-transeq-软件会对-gene-id-进行修改，将Prunella-strophiata-best-assembled-1改为了Prunella-strophiata-best-assembled-1-1"><a href="#因为EMBOSS-中的-transeq-软件会对-gene-id-进行修改，将Prunella-strophiata-best-assembled-1改为了Prunella-strophiata-best-assembled-1-1" class="headerlink" title="因为EMBOSS 中的 transeq 软件会对 gene id 进行修改，将Prunella_strophiata_best_assembled_1改为了Prunella_strophiata_best_assembled_1_1"></a>因为EMBOSS 中的 transeq 软件会对 gene id 进行修改，将Prunella_strophiata_best_assembled_1改为了Prunella_strophiata_best_assembled_1_1</h4><h4 id="所以使用sed命令去除掉gene-id末尾的-1-同时就得到了最终的蛋白文件"><a href="#所以使用sed命令去除掉gene-id末尾的-1-同时就得到了最终的蛋白文件" class="headerlink" title="所以使用sed命令去除掉gene id末尾的_1 同时就得到了最终的蛋白文件"></a>所以使用sed命令去除掉gene id末尾的_1 同时就得到了最终的蛋白文件</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">sed -i 's/\(.*\)_1\(.*\)/\1\2/g' Prunella_strophiata.best_assembled.pep.fa</span><br></pre></td></tr></table></figure>

<h2 id="filter-Du’s-file"><a href="#filter-Du’s-file" class="headerlink" title="filter Du’s file"></a>filter Du’s file</h2><h4 id="杜老师的16个岩鹨基因组的原始注释文件中即没有包含cds的信息，也没有包含gene-id的信息"><a href="#杜老师的16个岩鹨基因组的原始注释文件中即没有包含cds的信息，也没有包含gene-id的信息" class="headerlink" title="杜老师的16个岩鹨基因组的原始注释文件中即没有包含cds的信息，也没有包含gene id的信息"></a>杜老师的16个岩鹨基因组的原始注释文件中即没有包含cds的信息，也没有包含gene id的信息</h4><h4 id="原始注释如下"><a href="#原始注释如下" class="headerlink" title="原始注释如下"></a>原始注释如下</h4><blockquote>
<p>C26941125       exonerate:protein2genome:local  gene    1346    2626    1492    -       .       gene_id 1 ; sequence evm.model.contig1003.2 ; gene_orientation + ; identity 99.30 ; similarity 99.30<br>C26941125       exonerate:protein2genome:local  exon    2490    2626    .       -       .       insertions 0 ; deletions 0 ; identity 95.56 ; similarity 95.56<br>C26941125       exonerate:protein2genome:local  intron  2378    2489    .       -       .       intron_id 1<br>C26941125       exonerate:protein2genome:local  exon    2235    2377    .       -       .       insertions 0 ; deletions 1 ; identity 100.00 ; similarity 100.00 ; frameshifts 2<br>C26941125       exonerate:protein2genome:local  intron  2148    2234    .       -       .       intron_id 2<br>C26941125       exonerate:protein2genome:local  exon    2116    2147    .       -       .       insertions 0 ; deletions 0 ; identity 100.00 ; similarity 100.00<br>C26941125       exonerate:protein2genome:local  intron  1897    2115    .       -       .       intron_id 3<br>C26941125       exonerate:protein2genome:local  exon    1346    1896    .       -       .       insertions 0 ; deletions 0 ; identity 100.00 ; similarity 100.00<br>scaffold175008  exonerate:protein2genome:local  gene    1423    2438    1202    +       .       gene_id 1 ; sequence evm.model.contig1003.1 ; gene_orientation + ; identity 89.45 ; similarity 89.84<br>scaffold175008  exonerate:protein2genome:local  exon    1423    2014    .       +       .       insertions 0 ; deletions 0 ; identity 86.80 ; similarity 86.80<br>scaffold175008  exonerate:protein2genome:local  intron  2015    2259    .       +       .       intron_id 1<br>scaffold175008  exonerate:protein2genome:local  exon    2260    2438    .       +       .       insertions 0 ; deletions 0 ; identity 98.31 ; similarity 100.00</p>
</blockquote>
<h4 id="所以首先要做的是先添加gene-id的信息，然后再根据注释文件中的exon的位置信息，提取出所有转录本"><a href="#所以首先要做的是先添加gene-id的信息，然后再根据注释文件中的exon的位置信息，提取出所有转录本" class="headerlink" title="所以首先要做的是先添加gene id的信息，然后再根据注释文件中的exon的位置信息，提取出所有转录本"></a>所以首先要做的是先添加gene id的信息，然后再根据注释文件中的exon的位置信息，提取出所有转录本</h4><h4 id="再用transdecoder软件，对所有的转录本去预测ORF，得到最长ORF"><a href="#再用transdecoder软件，对所有的转录本去预测ORF，得到最长ORF" class="headerlink" title="再用transdecoder软件，对所有的转录本去预测ORF，得到最长ORF"></a>再用transdecoder软件，对所有的转录本去预测ORF，得到最长ORF</h4><h3 id="1-get-cdna-file-and-transdecoder-commands"><a href="#1-get-cdna-file-and-transdecoder-commands" class="headerlink" title="1. get cdna file and transdecoder commands"></a>1. get cdna file and transdecoder commands</h3><h3 id="This-step-aims-at-reformat-16-genome-annotation-of-Du-by-adding-gene-id"><a href="#This-step-aims-at-reformat-16-genome-annotation-of-Du-by-adding-gene-id" class="headerlink" title="This step aims at reformat 16 genome annotation of Du by adding gene id"></a>This step aims at reformat 16 genome annotation of Du by adding gene id</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">ls *gff3 | sed 's/.gff3//g' | while read line;</span><br><span class="line">do python reformat.py $&#123;line&#125;.gff3  &gt; $&#123;line&#125;.refomat.gff3;</span><br><span class="line">gffread $&#123;line&#125;.refomat.gff3 -g $&#123;line&#125;.fa  -w $&#123;line&#125;.cdna.fa;</span><br><span class="line">echo "TransDecoder.LongOrfs -t "$&#123;line&#125;".cdna.fa &amp;" &gt;&gt; run_TransDecoder.LongOrfs.sh</span><br><span class="line">echo "TransDecoder.Predict -t "$&#123;line&#125;".cdna.fa &amp;" &gt;&gt; run_TransDecoder.Predict.sh</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<h4 id="在上面的循环中，执行了4个循环的命令，在下面分四个说明"><a href="#在上面的循环中，执行了4个循环的命令，在下面分四个说明" class="headerlink" title="在上面的循环中，执行了4个循环的命令，在下面分四个说明"></a>在上面的循环中，执行了4个循环的命令，在下面分四个说明</h4><h4 id="1-使用reformat-py脚本修改gene-id的格式"><a href="#1-使用reformat-py脚本修改gene-id的格式" class="headerlink" title="1. 使用reformat.py脚本修改gene id的格式"></a>1. 使用reformat.py脚本修改gene id的格式</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python reformat.py $&#123;line&#125;.gff3  &gt; $&#123;line&#125;.refomat.gff3;</span><br></pre></td></tr></table></figure>

<h4 id="reformat-py-脚本如下，各部分代表的意义与上面的-reformat-py2-类似"><a href="#reformat-py-脚本如下，各部分代表的意义与上面的-reformat-py2-类似" class="headerlink" title="reformat.py 脚本如下，各部分代表的意义与上面的 reformat.py2 类似"></a>reformat.py 脚本如下，各部分代表的意义与上面的 reformat.py2 类似</h4><h4 id="修改后的gff文件格式如下，差别只在最后一列"><a href="#修改后的gff文件格式如下，差别只在最后一列" class="headerlink" title="修改后的gff文件格式如下，差别只在最后一列"></a>修改后的gff文件格式如下，差别只在最后一列</h4><blockquote>
<p>C26941125       exonerate:protein2genome:local  gene    1346    2626    1492    -       .       ID=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  exon    2490    2626    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  intron  2378    2489    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  exon    2235    2377    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  intron  2148    2234    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  exon    2116    2147    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  intron  1897    2115    .       -       .       Parent=Prunella_atrogularis_141230_1<br>C26941125       exonerate:protein2genome:local  exon    1346    1896    .       -       .       Parent=Prunella_atrogularis_141230_1</p>
</blockquote>
<h4 id="2-使用gffread提取exons组成的cdna序列"><a href="#2-使用gffread提取exons组成的cdna序列" class="headerlink" title="2.使用gffread提取exons组成的cdna序列"></a>2.使用gffread提取exons组成的cdna序列</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gffread $&#123;line&#125;.refomat.gff3 -g $&#123;line&#125;.fa  -w $&#123;line&#125;.cdna.fa;</span><br></pre></td></tr></table></figure>

<h4 id="3-与-4-的命令的目的就是生成TransDecoder软件所诉需要的命令行，即软件名字-cdna序列名字"><a href="#3-与-4-的命令的目的就是生成TransDecoder软件所诉需要的命令行，即软件名字-cdna序列名字" class="headerlink" title="3. 与 4. 的命令的目的就是生成TransDecoder软件所诉需要的命令行，即软件名字+cdna序列名字"></a>3. 与 4. 的命令的目的就是生成TransDecoder软件所诉需要的命令行，即软件名字+cdna序列名字</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">echo &quot;TransDecoder.LongOrfs -t &quot;$&#123;line&#125;&quot;.cdna.fa &amp;&quot; &gt;&gt; run_TransDecoder.LongOrfs.sh</span><br><span class="line">echo &quot;TransDecoder.Predict -t &quot;$&#123;line&#125;&quot;.cdna.fa &amp;&quot; &gt;&gt; run_TransDecoder.Predict.sh</span><br></pre></td></tr></table></figure>

<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">### 2. use TransDecoder to predict ORF region</span><br><span class="line">#### 运行上一步保存的命令行，TransDecoder使用cdna序列预测cds序列</span><br><span class="line">nohup sh run_TransDecoder.LongOrfs.sh &gt; run_TransDecoder.LongOrfs.log</span><br><span class="line">nohup sh run_TransDecoder.Predict.sh &gt; run_TransDecoder.Predict.log</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">### 3. generate longest transcripts and remove duplicated sequence in pep file</span><br><span class="line">```shell</span><br><span class="line">ls *.transdecoder.pep | while read line</span><br><span class="line">do</span><br><span class="line">/usr/bin/perl retrive_longest_splicing_variant.pl $&#123;line&#125; longest.$&#123;line&#125;</span><br><span class="line">/usr/bin/perl rm_dup_fasta.pl longest.$&#123;line&#125; temp.longest.$&#123;line&#125;</span><br><span class="line">awk -F &apos;#&apos; &apos;&#123;if($0 ~&quot;^&gt;&quot;)&#123;print $1&#125;else&#123;print $0&#125;&#125;&apos; temp.$&#123;line&#125; &gt;   final.$&#123;line&#125;</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<h4 id="在上面的循环中，执行了3个循环的命令，在下面分三个说明"><a href="#在上面的循环中，执行了3个循环的命令，在下面分三个说明" class="headerlink" title="在上面的循环中，执行了3个循环的命令，在下面分三个说明"></a>在上面的循环中，执行了3个循环的命令，在下面分三个说明</h4><h4 id="1-对TransDecoder的结果中的蛋白文件进行提取最长转录本"><a href="#1-对TransDecoder的结果中的蛋白文件进行提取最长转录本" class="headerlink" title="1. 对TransDecoder的结果中的蛋白文件进行提取最长转录本"></a>1. 对TransDecoder的结果中的蛋白文件进行提取最长转录本</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">/usr/bin/perl retrive_longest_splicing_variant.pl $&#123;line&#125; longest.$&#123;line&#125;</span><br></pre></td></tr></table></figure>

<h4 id="2-对TransDecoder的结果中的蛋白文件进行去重复"><a href="#2-对TransDecoder的结果中的蛋白文件进行去重复" class="headerlink" title="2. 对TransDecoder的结果中的蛋白文件进行去重复"></a>2. 对TransDecoder的结果中的蛋白文件进行去重复</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">/usr/bin/perl rm_dup_fasta.pl longest.$&#123;line&#125; temp.longest.$&#123;line&#125;</span><br></pre></td></tr></table></figure>

<h4 id="3-对蛋白文件的gene-id进行修改，将冗余的部分去掉"><a href="#3-对蛋白文件的gene-id进行修改，将冗余的部分去掉" class="headerlink" title="3. 对蛋白文件的gene id进行修改，将冗余的部分去掉"></a>3. 对蛋白文件的gene id进行修改，将冗余的部分去掉</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">awk -F '#' '&#123;if($0 ~"^&gt;")&#123;print $1&#125;else&#123;print $0&#125;&#125;' temp.$&#123;line&#125; &gt;   final.$&#123;line&#125;</span><br></pre></td></tr></table></figure>

<h2 id="filter-BGI-file"><a href="#filter-BGI-file" class="headerlink" title="filter BGI file"></a>filter BGI file</h2><h3 id="因为BGI的gff注释文件中，已经有了原始的基因名，所以先提取出原始的基因名，再添加一个对应的基因名，再将原始的注释文件替换"><a href="#因为BGI的gff注释文件中，已经有了原始的基因名，所以先提取出原始的基因名，再添加一个对应的基因名，再将原始的注释文件替换" class="headerlink" title="因为BGI的gff注释文件中，已经有了原始的基因名，所以先提取出原始的基因名，再添加一个对应的基因名，再将原始的注释文件替换"></a>因为BGI的gff注释文件中，已经有了原始的基因名，所以先提取出原始的基因名，再添加一个对应的基因名，再将原始的注释文件替换</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">ls *BGI*  | while read line</span><br><span class="line">do</span><br><span class="line">grep '&gt;' $&#123;line&#125;.pep | sed 's/&gt;//g' &gt; $&#123;line&#125;.modify.seqid</span><br><span class="line">awk '&#123;if($3 =="gene")&#123;a=index($0,"~~");b=index($0,";");print substr($0,a+2,b-a-2)&#125;&#125;' &gt; $&#123;line&#125;.origin.seqid</span><br><span class="line">paste -d '\t' $&#123;line&#125;.origin.seqid $&#123;line&#125;.modify.seqid &gt; $&#123;line&#125;_coress.seqid</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line">python  replace_gff_gene_id.py Prunella_fulvescens_BGI_coress.seqid Prunella_fulvescens.homolog.gff &gt; Prunella_fulvescens.modify.gff3</span><br><span class="line">python  replace_gff_gene_id.py Prunella_himalayana_BGI_coress.seqid Prunella_himalayana.homolog.gff &gt;  Prunella_himalayana.modify.gff3</span><br></pre></td></tr></table></figure>

<h2 id="download-Gallus-genome-from-Ensembl-database-and-remove-reduant-transcripts"><a href="#download-Gallus-genome-from-Ensembl-database-and-remove-reduant-transcripts" class="headerlink" title="download Gallus genome from Ensembl database and remove reduant transcripts"></a>download Gallus genome from Ensembl database and remove reduant transcripts</h2><h3 id="1-下载人和鸡的蛋白序列"><a href="#1-下载人和鸡的蛋白序列" class="headerlink" title="1. 下载人和鸡的蛋白序列"></a>1. 下载人和鸡的蛋白序列</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.ensembl.org/pub/release-100/fasta/gallus_gallus/pep/Gallus_gallus.GRCg6a.pep.all.fa.gz</span><br><span class="line">gunzip Gallus_gallus.GRCg6a.pep.all.fa.gz</span><br><span class="line"></span><br><span class="line">wget ftp://ftp.ensembl.org/pub/release-100/fasta/homo_sapiens/pep/Homo_sapiens.GRCh38.pep.all.fa.gz</span><br><span class="line">gunzip Homo_sapiens.GRCh38.pep.all.fa.gz</span><br></pre></td></tr></table></figure>

<h3 id="2-过滤掉同一gene-id中短的mrna序列"><a href="#2-过滤掉同一gene-id中短的mrna序列" class="headerlink" title="2. 过滤掉同一gene id中短的mrna序列"></a>2. 过滤掉同一gene id中短的mrna序列</h3><h3 id="retrive-longest-transcripts-from-above-protein-fasta-file"><a href="#retrive-longest-transcripts-from-above-protein-fasta-file" class="headerlink" title="retrive longest transcripts from above protein fasta file"></a>retrive longest transcripts from above protein fasta file</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">ls *fa | while read line</span><br><span class="line">do</span><br><span class="line">python ~/software/OrthoFinder/tools/primary_transcript.py $&#123;line&#125;</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<h1 id="CORE-PROGRAM"><a href="#CORE-PROGRAM" class="headerlink" title="CORE PROGRAM"></a>CORE PROGRAM</h1><h2 id="run-OrthoFinder-and-get-the-Orthogroups"><a href="#run-OrthoFinder-and-get-the-Orthogroups" class="headerlink" title="run OrthoFinder and get the Orthogroups"></a>run OrthoFinder and get the Orthogroups</h2><h3 id="将上述所有蛋白序列文件作为输入去预测Orthogroups"><a href="#将上述所有蛋白序列文件作为输入去预测Orthogroups" class="headerlink" title="将上述所有蛋白序列文件作为输入去预测Orthogroups"></a>将上述所有蛋白序列文件作为输入去预测Orthogroups</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">~/software/OrthoFinder/orthofinder -f pep -S diamond</span><br></pre></td></tr></table></figure>

<h2 id="published-info"><a href="#published-info" class="headerlink" title="published info"></a>published info</h2><h3 id="OrthoFinder发表在GB的文章，引用次数一年内已经有69次"><a href="#OrthoFinder发表在GB的文章，引用次数一年内已经有69次" class="headerlink" title="OrthoFinder发表在GB的文章，引用次数一年内已经有69次"></a>OrthoFinder发表在GB的文章，引用次数一年内已经有69次</h3><blockquote>
<p>Emms D.M. &amp; Kelly S. OrthoFinder: phylogenetic orthology inference for comparative<br>genomics (2019), Genome Biology 20:238</p>
</blockquote>
<h2 id="use-longest-transcripts-gene-ids-filter-gff-file-and-get-intron-sequences"><a href="#use-longest-transcripts-gene-ids-filter-gff-file-and-get-intron-sequences" class="headerlink" title="use longest transcripts gene ids filter gff file and get intron sequences"></a>use longest transcripts gene ids filter gff file and get intron sequences</h2><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">less *.seqid |sed 's/seqid//g'| while read line</span><br><span class="line">do cat $&#123;line&#125; | while read row</span><br><span class="line">grep -w $&#123;row&#125; $&#123;line&#125;.gff3</span><br><span class="line">done &gt; $&#123;line&#125;.reformat.gff3</span><br><span class="line">/usr/bin/perl extract_intron_from_gff3_based_on_exon.pl $&#123;line&#125;.fa $&#123;line&#125;.reformat.gff3 $&#123;line&#125;.intron.fa</span><br><span class="line">done</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/05/15/详细注释的OrthoFinder/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/15/PAML/" >PAML and MCScanX-transposed</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-15  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>通常师姐她们做的PAML选择压力检验 是为了分析一个基因，或者多个基因串联，构建的系统发育树中，收选择的物种是谁，在不同的物种当中是否具有不同的选择方向。</p>
<p>意思也就是在这篇文章中Molecular Evolution of the TET Gene Family in Mammals，TET有三个直系同源基因，然后把这三个基因对应的20个物种分别建树，先用sites model中的模型M0（假设比对中所有密码子位点都为一个ω）看每个基因的进化速率</p>
<p>Selection pressure causes differentiation of the SPL gene family in the Juglandaceae</p>
<p>这一篇用branch sites model的</p>
<p>branch sites model 是相对于集合了branch model和sites model的，因为他允许了在model中sites的omega值不同的同时也允许了branch的omega也不相同。即反馈到ctl文件中就是 Model =2 Nsite =2，代表了branch model中的foreground ： background 和Sites Model中的M2 model W0 = 0 ;W1 = 1; W2 &gt; 1</p>
<p>对于ModelA和ModelA null 的比较，Null 在检测中强制fix omega =1，就代表着backgroud的sites均处于negative selection,.foreground中的sites均处于neutral selection 而在modelA中 foreground 的sites全部处于positive selection </p>
<p>但是但是但是 PAML这个软件的目的就是检测正选择，而当输入的序列过于相似的时候，很少量的正选择的位点就很难反馈到模型上，十分容易造成正选择模型和凌假设模型差异不显著</p>
<p>NOTE:NG86 assumes no transition-transversion rate difference and no codon usage bias.</p>
<p>####1.WGD</p>
<p>Whole-genome duplications (WGDs) have occurred in thelineages of plants [1], animals [2,3] and fungi [4,5], with possibleconsequences including evolution of novel or modified genefunctions [6,7,8,9], and/or provision of ‘‘buffer capacity’’ [10,11]or  genetic  redundancy  that  increases  genetic  robustness[12,13,14,15,16,17].  Genome  duplication  may  also  increaseopportunities for nonreciprocal recombination [18,19,20], permit-ting or causing duplicated genes to evolve in concert for a period oftime. Rapid DNA loss and restructuring of low-copy DNA[21,22,23,24], retrotransposon activation [25,26,27] and epigeneticchanges [28,29,30,31,32,33] following WGD may further providematerials for evolutionary change</p>
<p>全基因组重复，基因组复制也可能增加不可逆重组的机会[18，19，20]，允许或导致复制的基因在一段时间内协同进化</p>
<p>####2.Tandem and Proximal</p>
<p>Tandem duplicates  are  consecutive  in  the  genome  while  proximal duplicates are near one another but separated by a few gene</p>
<p>These two gene duplication modes are presumed to arise throughunequal crossing over [36] or localized transposon activitie</p>
<p>####3.Dispersed</p>
<p>Dispersed duplicates are neither adjacent to each other in thegenome nor within homeologous chromosome segments </p>
<p>####4.Transposed</p>
<p>DNA  transposons  such  aspackmules  (rice)  [39],  helitrons  (maize)  [40],  and  CACTAelements (sorghum) [27] may relocate duplicated genes or genesegments to new chromosomal positions (referred to as DNA basedtransposed duplication). RNA based transposed duplication, oftenreferred to as retrotransposition, typically creates a single-exonretrocopy from a multi-exon parental gene, by reverse transcrip-tion of a spliced messenger RNA. It is presumed that the retrocopyduplicates only the transcribed sequence of the parental gene,detached from the parental promoter. The new retrogene is oftendeposited in a novel chromosomal environment with new (i.e. non-ancestral) neighboring genes and, having lost its native promoter,is only likely to survive as a functional gene if a new promoter is acquired [41,42]</p>
<p>NOTE！</p>
<p>A transposed duplicate pairs must be meet the following criteria: one gene existed in its ancestral locus, and the other was located in a non-ancestral locus (<a href="https://www.frontiersin.org/articles/10.3389/fpls.2018.00161/full#B87" target="_blank" rel="noopener">Wang Y. et al., 2011</a>).</p>
<p>NOTE！</p>
<p>If a pair of transposed duplicated genes comprised an ancestral gene with more than two exons and a novel transposed copy without an intron, then this pair was inferred to be derived from RNA-based transposition (retro transposition).</p>

	
	</div>
  <a type="button" href="/2020/05/15/PAML/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/15/OrthoFinde/" >OrthoFinder 寻找同源基因</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-15  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="ortholog-identification"><a href="#ortholog-identification" class="headerlink" title="ortholog identification"></a>ortholog identification</h1><h2 id="in-order-to-keep-gene-id-format-consistantly-use-awk-and-python-and-gffread-and-transeq-for-refomating"><a href="#in-order-to-keep-gene-id-format-consistantly-use-awk-and-python-and-gffread-and-transeq-for-refomating" class="headerlink" title="in order to keep gene id format consistantly,use awk and python and gffread and transeq for refomating"></a>in order to keep gene id format consistantly,use awk and python and gffread and transeq for refomating</h2><h3 id="base-pipeline"><a href="#base-pipeline" class="headerlink" title="base pipeline"></a>base pipeline</h3><h3 id="1-gene-and-mRNA-will-cause-redirect-error-remove-first"><a href="#1-gene-and-mRNA-will-cause-redirect-error-remove-first" class="headerlink" title="1. gene and mRNA will cause redirect error, remove first"></a>1. gene and mRNA will cause redirect error, remove first</h3><p>awk ‘{if($3 != “mRNA”)print $0}’ Prunella_strophita_assemble.evm.gff3  &gt; Prunella_strophita_assemble.remove_mRNA.gff3</p>
<h3 id="2-replace-gene-id"><a href="#2-replace-gene-id" class="headerlink" title="2. replace gene id"></a>2. replace gene id</h3><p>python reformat.py2 Prunella_strophita_assemble.remove_mRNA.gff3 &gt;  Prunella_strophita.best_assembled.gff3</p>
<h3 id="3-extract-cds-sequence"><a href="#3-extract-cds-sequence" class="headerlink" title="3. extract cds sequence"></a>3. extract cds sequence</h3><p>gffread Prunella_strophita.best_assembled.gff3 -g Prunella_strophita_assemble.fa -x Prunella_strophita.best_assembled.cds.fa</p>
<h3 id="4-cds2protein"><a href="#4-cds2protein" class="headerlink" title="4. cds2protein"></a>4. cds2protein</h3><p>transeq -sequence Prunella_strophita.best_assembled.cds.fa -outseq Prunella_strophita.best_assembled.pep.fa</p>
<h3 id="5-remove-the-info-added-by-transeq"><a href="#5-remove-the-info-added-by-transeq" class="headerlink" title="5. remove the info added by transeq"></a>5. remove the info added by transeq</h3><p>sed -i ‘s/(.<em>)_1(.</em>)/\1\2/g’ Prunella_strophita.best_assembled.pep.fa</p>
<h2 id="shell-pipeline"><a href="#shell-pipeline" class="headerlink" title="shell pipeline"></a>shell pipeline</h2><p>ls *gff3 | sed ‘s/.gff3//g’ | while read line;<br>do python reformat.py ${line}.gff3  &gt; ${line}.refomat.gff3;<br>gffread ${line}.refomat.gff3 -g ${line}.fa  -w ${line}.cdna.fa;<br>echo “TransDecoder.LongOrfs -t “${line}”.cdna.fa &amp;” &gt;&gt; run_TransDecoder.LongOrfs.sh<br>echo “TransDecoder.Predict -t “${line}”.cdna.fa &amp;” &gt;&gt; run_TransDecoder.Predict.sh<br>done</p>
<h1 id="use-TransDecoder-to-predict-ORF-region"><a href="#use-TransDecoder-to-predict-ORF-region" class="headerlink" title="use TransDecoder to predict ORF region"></a>use TransDecoder to predict ORF region</h1><p>nohup sh run_TransDecoder.LongOrfs.sh &gt; run_TransDecoder.LongOrfs.log<br>nohup sh run_TransDecoder.Predict.sh &gt; run_TransDecoder.Predict.log</p>
<h1 id="generate-longest-transcripts-and-remove-duplicated-sequence-in-pep-file"><a href="#generate-longest-transcripts-and-remove-duplicated-sequence-in-pep-file" class="headerlink" title="generate longest transcripts and remove duplicated sequence in pep file"></a>generate longest transcripts and remove duplicated sequence in pep file</h1><p>ls *.transdecoder.pep | while read line<br>do<br>/usr/bin/perl retrive_longest_splicing_variant.pl ${line} longest.${line}<br>/usr/bin/perl rm_dup_fasta.pl longest.${line} temp.longest.${line}<br>awk -F ‘#’ ‘{if($0 ~”^&gt;”){print $1}else{print $0}}’ temp.${line} &gt;   final.${line}<br>done </p>
<h1 id="rename-file"><a href="#rename-file" class="headerlink" title="rename file"></a>rename file</h1><p>rename .cdna.fa.transdecoder.pep .pep.fa <em>.cdna.fa.transdecoder.pep<br>rename final.longest.P P final.longest.Prunella_</em></p>
<h2 id="download-Gallus-genome-from-Ensembl-database-and-remove-reduant-transcripts"><a href="#download-Gallus-genome-from-Ensembl-database-and-remove-reduant-transcripts" class="headerlink" title="download Gallus genome from Ensembl database and remove reduant transcripts"></a>download Gallus genome from Ensembl database and remove reduant transcripts</h2><p>wget <a href="ftp://ftp.ensembl.org/pub/release-100/fasta/gallus_gallus/pep/Gallus_gallus.GRCg6a.pep.all.fa.gz" target="_blank" rel="noopener">ftp://ftp.ensembl.org/pub/release-100/fasta/gallus_gallus/pep/Gallus_gallus.GRCg6a.pep.all.fa.gz</a><br>gunzip Gallus_gallus.GRCg6a.pep.all.fa.gz</p>
<p>python ~/software/OrthoFinder/tools/primary_transcript.py Gallus_gallus.GRCg6a.pep.all.fa<br>mv /home/sunhaoyun/data/Prunella_genome/primary_transcripts/Gallus_gallus.GRCg6a.pep.all.fa Gallus.pep.fa</p>
<h2 id="run-OrthoFinder"><a href="#run-OrthoFinder" class="headerlink" title="run OrthoFinder"></a>run OrthoFinder</h2><p>~/software/OrthoFinder/orthofinder -f pep -S diamond</p>
<h1 id="analyse-result"><a href="#analyse-result" class="headerlink" title="analyse result"></a>analyse result</h1><h3 id="1-overview-how-many-genes-were-assigned-to-orthogroups"><a href="#1-overview-how-many-genes-were-assigned-to-orthogroups" class="headerlink" title="1. overview how many genes were assigned to orthogroups"></a>1. overview how many genes were assigned to orthogroups</h3><h4 id="PATH"><a href="#PATH" class="headerlink" title="PATH"></a>PATH</h4><p>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/Prunella_sp19_with_Gallus/Comparative_Genomics_Statistics/Statistics_Overall.tsv</p>
<h3 id="2-check-the-percentages-on-a-per-species-basis-here"><a href="#2-check-the-percentages-on-a-per-species-basis-here" class="headerlink" title="2. check the percentages on a per species basis here"></a>2. check the percentages on a per species basis here</h3><h4 id="PATH-1"><a href="#PATH-1" class="headerlink" title="PATH"></a>PATH</h4><p>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/OrthoFinder/Prunella_sp19_with_Gallus/CComparative_Genomics_Statistics/Statistics_PerSpecies.tsv</p>
<h3 id="3-species-tree"><a href="#3-species-tree" class="headerlink" title="3. species tree"></a>3. species tree</h3><p>####PATH<br>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/OrthoFinder/Prunella_sp19_with_Gallus/Species_Tree/SpeciesTree_rooted.txt</p>
<h3 id="NOTE"><a href="#NOTE" class="headerlink" title="NOTE"></a>NOTE</h3><h3 id="This-speceis-tree-may-have-some-error"><a href="#This-speceis-tree-may-have-some-error" class="headerlink" title="This speceis tree may have some error"></a>This speceis tree may have some error</h3><h3 id="If-the-species-tree-is-not-correct-then-this-will-not-impact-the-orthogroup-inference"><a href="#If-the-species-tree-is-not-correct-then-this-will-not-impact-the-orthogroup-inference" class="headerlink" title="If the species tree is not correct then this will not impact the orthogroup inference,"></a>If the species tree is not correct then this will not impact the orthogroup inference,</h3><h3 id="but-it-might-affect-the-orthologue-inference-in-some-of-the-gene-trees-which-have-gene-duplication-events"><a href="#but-it-might-affect-the-orthologue-inference-in-some-of-the-gene-trees-which-have-gene-duplication-events" class="headerlink" title="but it might affect the orthologue inference in some of the gene trees which have gene duplication events."></a>but it might affect the orthologue inference in some of the gene trees which have gene duplication events.</h3><h3 id="4-find-out-Gallus-orthologues-are-in-the-other-speceis"><a href="#4-find-out-Gallus-orthologues-are-in-the-other-speceis" class="headerlink" title="4. find out Gallus orthologues are in  the other speceis"></a>4. find out Gallus orthologues are in  the other speceis</h3><h4 id="such-as-methyltransferase-like-21C-gene-METTL21C-plays-an-important-role-in-chicken-muscle-development"><a href="#such-as-methyltransferase-like-21C-gene-METTL21C-plays-an-important-role-in-chicken-muscle-development" class="headerlink" title="such as methyltransferase-like 21C gene (METTL21C) plays an important role in chicken muscle development"></a>such as methyltransferase-like 21C gene (METTL21C) plays an important role in chicken muscle development</h4><h4 id="GgMETTL21C1-corresponding-gene-id-is-ENSGALG00000001790-6"><a href="#GgMETTL21C1-corresponding-gene-id-is-ENSGALG00000001790-6" class="headerlink" title="GgMETTL21C1 corresponding gene id is ENSGALG00000001790.6"></a>GgMETTL21C1 corresponding gene id is ENSGALG00000001790.6</h4><h4 id="PATH-2"><a href="#PATH-2" class="headerlink" title="PATH"></a>PATH</h4><p>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/OrthoFinder/Prunella_sp19_with_Gallus/Orthologues/Orthologues_Gallus.pep/</p>
<p>ls *tsv | while read line; do grep ‘ENSGALG00000001790.6’ ${line}; done</p>
<h3 id="5-Orthogroups-comparisons-across-a-clade-of-species"><a href="#5-Orthogroups-comparisons-across-a-clade-of-species" class="headerlink" title="5. Orthogroups comparisons across a clade of species"></a>5. Orthogroups comparisons across a clade of species</h3><h4 id="overview-file-PATH"><a href="#overview-file-PATH" class="headerlink" title="overview file PATH"></a>overview file PATH</h4><p>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/OrthoFinder/Prunella_sp19_with_Gallus/Orthogroups/Orthogroups.tsv</p>
<h4 id="For-each-orthogroup-there-is-a-FASTA-file-in"><a href="#For-each-orthogroup-there-is-a-FASTA-file-in" class="headerlink" title="For each orthogroup there is a FASTA file in"></a>For each orthogroup there is a FASTA file in</h4><p>/home/public/Prunella/Prunella_genome/Sunhaoyun_genome/OrthoFinder/Prunella_sp19_with_Gallus/Orthogroup_Sequences</p>
<h4 id="which-contains-the-sequences-for-the-genes-in-that-orthogroup"><a href="#which-contains-the-sequences-for-the-genes-in-that-orthogroup" class="headerlink" title="which contains the sequences for the genes in that orthogroup."></a>which contains the sequences for the genes in that orthogroup.</h4><h2 id="Other-scipts"><a href="#Other-scipts" class="headerlink" title="Other scipts"></a>Other scipts</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"># reformat.py</span><br><span class="line">import sys</span><br><span class="line">i=1</span><br><span class="line">with open(sys.argv[1]) as input_gff:</span><br><span class="line">    sp_name = input_gff.name.replace(&apos;.gff3&apos;,&apos;&apos;)</span><br><span class="line">    for row in input_gff:</span><br><span class="line">        lst1 = row.split(&apos;\t&apos;)</span><br><span class="line">        if &quot;gene_id&quot; in lst1[8]:</span><br><span class="line">            lst1[8] = &quot;ID=&quot;+sp_name+&quot;_&quot;+str(i)</span><br><span class="line">            new_list=&apos;\t&apos;.join(lst1)</span><br><span class="line">            print(new_list)</span><br><span class="line">            i+=1</span><br><span class="line">        else:</span><br><span class="line">            lst1[8] = &quot;Parent=&quot;+sp_name+&quot;_&quot;+str(i-1)</span><br><span class="line">            new_list=&apos;\t&apos;.join(lst1)</span><br><span class="line">            print(new_list)</span><br><span class="line">```python</span><br></pre></td></tr></table></figure>

<p>#reformat.py2<br>import sys<br>i=1<br>with open(sys.argv[1]) as input_gff:<br>    sp_name = input_gff.name.replace(‘assemble.remove_mRNA.gff3’,’best_assembled’)<br>    for row in input_gff:<br>        lst1 = row.split(‘\t’)<br>        if “TU” in lst1[8]:<br>            lst1[8] = “ID=”+sp_name+”<em>“+str(i)<br>            new_list=’\t’.join(lst1)<br>            print(new_list)<br>            i+=1<br>        else:<br>            lst1[8] = “Parent=”+sp_name+”</em>“+str(i-1)<br>            new_list=’\t’.join(lst1)<br>            print(new_list)</p>
<pre><code class="python"></code></pre>

	
	</div>
  <a type="button" href="/2020/05/15/OrthoFinde/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/04/29/identify/" >Blast + Hmmsearch 鉴定一个基因家族</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-04-29  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">hmmsearch -o batatas.hmmsearch.log --tblout batatas.hmmsearch.tblout SBP.hmm  longest.renamehead.transcript.fasta.transdecoder.pep</span><br><span class="line"></span><br><span class="line">awk '&#123;if ($1!~/^#/)print $1&#125;' batatas.hmmsearch.tblout &gt; batatas.hmmsearch.seqid</span><br><span class="line"></span><br><span class="line">makeblastdb -in  longest.renamehead.transcript.fasta.transdecoder.pep -title batatas -dbtype prot -parse_seqids -out batatas</span><br><span class="line"></span><br><span class="line">blastp -db batatas -query atspl.pep.fa -evalue 1e-3 -outfmt 6 &gt; batatas.blastp</span><br><span class="line"></span><br><span class="line">awk '&#123;print $2&#125;' batatas.blastp | sort | uniq &gt; batatas.blastp.seqid</span><br><span class="line"></span><br><span class="line">cat batatas*seqid | sort | uniq &gt; batatas.spl.seqid</span><br><span class="line"></span><br><span class="line">perl extract.pl batatas.spl.seqid batatas.pep.fa &gt; batatas.spl.fa</span><br><span class="line"></span><br><span class="line">grep '&gt;' trifida.spl.long.fa &gt; trifida.spl.seqid</span><br><span class="line"></span><br><span class="line">cat batatas.smart.remove batatas.spl.seqid| sort | uniq -u &gt; batatas.smartSBP.seqid</span><br><span class="line"></span><br><span class="line">perl extract.pl batatas.smartSBP.seqid batatas.pep.fa &gt; batatas.spl.smart.fa</span><br><span class="line">	</span><br><span class="line">blastp -query batatas.spl.smart.fa -subject trifida.spl.smart.fa -outfmt 6 -max_target_seqs 1 &gt; b2t.log</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">less batatas.spl.seqid | while read line; do grep $&#123;line&#125; transcript.fasta.transdecoder.gff3 | awk -v OFS="\t" '&#123; if ($3 == "mRNA") print $1, $4, $5 &#125;' &gt;&gt; blast_result.bed; done</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/04/29/identify/#more" class="btn btn-default more">Read More</a>
</div>

		

		</div>

		<!-- pagination -->
		<div>
  		<center>
		<div class="pagination">

   
    
           <a type="button" class="btn btn-default disabled"><i class="fa fa-arrow-circle-o-left"></i>Prev</a>
        

        <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
 
       <a href="/page/2/" type="button" class="btn btn-default ">Next<i class="fa fa-arrow-circle-o-right"></i></a>     
        

  
</div>

  		</center>
		</div>

		
		
	</div> <!-- col-md-9 -->

	
		<div class="col-md-3">
	<div id="sidebar">
	
			
  <div id="site_search">
   <div class="form-group">
    <input type="text" id="local-search-input" name="q" results="0" placeholder="Search" class="st-search-input st-default-search-input form-control"/>
   </div>  
  <div id="local-search-result"></div>
  </div>


		
			
		
			
	<div class="widget">
		<h4>Tag Cloud</h4>
		<ul class="tag_box inline list-unstyled">		
		
			<li><a href="/tags/geome/">geome<span>1</span></a></li>
		
			<li><a href="/tags/Mysql-OrthoMCL/">Mysql, OrthoMCL<span>1</span></a></li>
		
			<li><a href="/tags/NCBI/">NCBI<span>1</span></a></li>
		
			<li><a href="/tags/KaKscalculator-R/">KaKscalculator,R<span>1</span></a></li>
		
			<li><a href="/tags/Python/">Python<span>1</span></a></li>
		
			<li><a href="/tags/Markdown-maker/">Markdown, maker<span>2</span></a></li>
		
			<li><a href="/tags/Phylogeny/">Phylogeny<span>1</span></a></li>
		
			<li><a href="/tags/OrthoFinder/">OrthoFinder<span>1</span></a></li>
		
			<li><a href="/tags/Ortholog/">Ortholog<span>1</span></a></li>
		
			<li><a href="/tags/Ensembl/">Ensembl<span>2</span></a></li>
		
			<li><a href="/tags/gene-duplication/">gene duplication<span>1</span></a></li>
		
			<li><a href="/tags/GiagaScience/">GiagaScience<span>1</span></a></li>
		
			<li><a href="/tags/HMM/">HMM<span>1</span></a></li>
		
			<li><a href="/tags/gene-family/">gene family<span>1</span></a></li>
		
		 
		</ul>
	</div>


		
			
<div class="widget">
  <h4>Recent Posts</h4>
  <ul class="entry list-unstyled">
    
      <li>
        <a href="/2020/06/22/KaKs_calculator/" ><i class="fa fa-file-o"></i>KaKs 计算进化速率_update</a>
      </li>
    
      <li>
        <a href="/2020/06/16/WGCNA/" ><i class="fa fa-file-o"></i></a>
      </li>
    
      <li>
        <a href="/2020/05/28/wget_ncbi/" ><i class="fa fa-file-o"></i>下载鸟类的基因组2) ncbi</a>
      </li>
    
      <li>
        <a href="/2020/05/28/wget_giga/" ><i class="fa fa-file-o"></i>下载鸟类的基因组2) giga</a>
      </li>
    
      <li>
        <a href="/2020/05/25/wget_Ensembl/" ><i class="fa fa-file-o"></i>下载鸟类的基因组1) Ensembl</a>
      </li>
    
  </ul>
</div>

		
			
<div class="widget">
	<h4>Links</h4>
	<ul class="blogroll list-unstyled">
	
		<li><i class="fa fa-github"></i><a href="https://github.com/goodlucksun" title="All Github repository." target="_blank"]);">My github</a></li>
	
	</ul>
</div>


		
	</div> <!-- sidebar -->
</div> <!-- col-md-3 -->

	
	
</div> <!-- row-fluid -->
	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2020 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
