<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 <div class="page-header logo">
  <h1>Keeping your word<span class="blink-fast">∎</span></h1>
</div>

<div class="row page">

	
	<div class="col-md-9">
	

		<div class="slogan">


		<i class="fa fa-heart blink-slow"></i>

		我们进化速度快到让你感动邪门？！

</div>    

		<div id="top_search"></div>
		<div class="mypage">
		
		<!-- title and entry -->
		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/29/bwa_picard_pipeline/" >二代重测序流程</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-29  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>首先使用迅雷下载sra文件 速度谁用谁知道</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"># 将ncbi 的 sra格式转化为fastq格式</span><br><span class="line">nohup fastq-dump --split-3 --gzip -O /home/sunhaoyun/data/ERR1347700.1 &amp;</span><br></pre></td></tr></table></figure>

<p>然后call就vans了</p>
<p>还有一点就是之前动物所的老师想要把60X变成20X的深度的reads，那边师兄给的方法是直接把Fq扔掉1/3，</p>
<p>现在学了个聪明点的方法，用rarefraction去除比较有可信度</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#################################</span></span><br><span class="line"><span class="comment">#bwa picard GATK pipeline</span></span><br><span class="line"><span class="comment">#@author: sunhaoyun </span></span><br><span class="line"><span class="comment">#@contact qq: 172293639</span></span><br><span class="line"><span class="comment"># STEP follow the 1000genome project high coverage sequencing wheel</span></span><br><span class="line"><span class="comment"># the specific prameter could check NYGC websites :)</span></span><br><span class="line"><span class="comment"># commond reference webites:</span></span><br><span class="line"><span class="comment"># https://best-practices-for-processing-hts-data.readthedocs.io/en/latest/index.html</span></span><br><span class="line"><span class="comment"># https://www.internationalgenome.org/data-portal/data-collection/30x-grch38</span></span><br><span class="line"><span class="comment"># reference genome could found in this google cloud sites:</span></span><br><span class="line"><span class="comment"># There will be about 3X Volume storage space bigger than Fq files</span></span><br><span class="line"><span class="comment">#1: Trimmomatic</span></span><br><span class="line"><span class="comment">#2: bwa mem</span></span><br><span class="line"><span class="comment">#3. picard fixmate</span></span><br><span class="line"><span class="comment">#4: picard sort sam</span></span><br><span class="line"><span class="comment">#5: picard mark duplicate</span></span><br><span class="line"><span class="comment">#6: gatk realign</span></span><br><span class="line"><span class="comment">#7: gatk BQSR</span></span><br><span class="line"><span class="comment">#8: gatk haltopviewer call gvcf</span></span><br><span class="line"><span class="comment">################################</span></span><br><span class="line"><span class="keyword">import</span> subprocess</span><br><span class="line"><span class="keyword">import</span> argparse</span><br><span class="line"><span class="keyword">import</span> time</span><br><span class="line"><span class="keyword">import</span> os.path</span><br><span class="line"><span class="keyword">from</span> pathlib <span class="keyword">import</span> Path</span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line"><span class="comment">################### SET SOFTWARE PATH HERE ################### </span></span><br><span class="line"></span><br><span class="line">adaptor_path = <span class="string">"/home/gujiaqi/software/adapters/adaptor.fa"</span></span><br><span class="line">trimmomatic_path = <span class="string">"/home/gujiaqi/software/trimmomatic-0.39.jar"</span></span><br><span class="line">bwa_path = <span class="string">"/usr/bin/bwa"</span> </span><br><span class="line">picard_path = <span class="string">"/home/gujiaqi/software/picard.jar"</span></span><br><span class="line">gatk_path = <span class="string">"/home/gujiaqi/software/GenomeAnalysisTK.jar"</span></span><br><span class="line">temp_path = <span class="string">"/data/gujiaqi/temp"</span></span><br><span class="line">db_snp = <span class="string">"/data/gujiaqi/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.dbsnp138.vcf.gz"</span></span><br><span class="line">known_indel = <span class="string">"/data/gujiaqi/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.known_indels.vcf.gz"</span></span><br><span class="line">mills = <span class="string">"/data/gujiaqi/reference_genome/resources_broad_hg38_v0_Mills_and_1000G_gold_standard.indels.hg38.vcf.gz"</span></span><br><span class="line"><span class="comment">###################    SOFTWARE CHECK   ######################</span></span><br><span class="line"><span class="keyword">if</span> <span class="keyword">not</span> Path(adaptor_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(temp_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(trimmomatic_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(bwa_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(picard_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(gatk_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(db_snp).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(known_indel).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(mills).exists():</span><br><span class="line">    print(<span class="string">"SOFTWARE SET FAILED, PLEASE CHECK AGAIN"</span>)</span><br><span class="line">    sys.exit()</span><br><span class="line">    </span><br><span class="line"></span><br><span class="line"></span><br><span class="line">ap = argparse.ArgumentParser()</span><br><span class="line">ap.add_argument(<span class="string">"-p"</span>, <span class="string">"--prefix"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"the  prefix of sample"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-f"</span>, <span class="string">"--refer"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"reference genome path"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-o"</span>, <span class="string">"--outpath"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"Out put file path"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-t"</span>, <span class="string">"--threads"</span>, required=<span class="literal">False</span>,</span><br><span class="line">    help=<span class="string">"process will be used"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-1"</span>, <span class="string">"--reads1"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"Input followed reads"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-2"</span>, <span class="string">"--reads2"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"Input reverse reads"</span>)</span><br><span class="line"></span><br><span class="line">args = vars(ap.parse_args())</span><br><span class="line"><span class="keyword">print</span> (<span class="string">"The Task id is &#123;&#125;"</span>.format(args[<span class="string">"prefix"</span>]))</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> args[<span class="string">"threads"</span>]:</span><br><span class="line">    cpu = args[<span class="string">"threads"</span>]</span><br><span class="line"><span class="keyword">else</span>:</span><br><span class="line">    cpu = <span class="number">8</span></span><br><span class="line"><span class="keyword">print</span> (<span class="string">"DATE STATUS: The Task Start at: &#123;&#125;"</span>.format(time.ctime())) </span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> <span class="keyword">not</span> Path(args[<span class="string">"prefix"</span>]).exists():</span><br><span class="line">    mk_work_dir = <span class="string">'mkdir &#123;&#125;'</span>.format(args[<span class="string">"prefix"</span>])</span><br><span class="line">    prefix = <span class="string">"&#123;&#125;"</span>.format(args[<span class="string">"prefix"</span>])</span><br><span class="line">    subprocess.call(mk_work_dir, shell=<span class="literal">True</span>)</span><br><span class="line">    print(<span class="string">"Work dictionary &#123;&#125; is created"</span>.format(args[<span class="string">"prefix"</span>]))</span><br><span class="line"><span class="keyword">else</span>:</span><br><span class="line">    prefix = <span class="string">"&#123;&#125;"</span>.format(args[<span class="string">"prefix"</span>])</span><br><span class="line">    print(<span class="string">"Work dictionary  &#123;&#125; is created"</span>.format(args[<span class="string">"prefix"</span>]))</span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 1 ####### </span></span><br><span class="line"><span class="comment">### Trimmomatic software should be downloaded ###</span></span><br><span class="line"><span class="comment">### You could modify the adaptor file and trimmomatic path bellow</span></span><br><span class="line"></span><br><span class="line"><span class="comment">### prepare well ###</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 1 ... filtering the fastq files \n\n\n"</span>)</span><br><span class="line">trim1_pa = <span class="string">"&#123;&#125;/&#123;&#125;_f1.trim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line">trim1_up = <span class="string">"&#123;&#125;/&#123;&#125;_f1.untrim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line">trim2_pa = <span class="string">"&#123;&#125;/&#123;&#125;_r2.trim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line">trim2_up = <span class="string">"&#123;&#125;/&#123;&#125;_r2.untrim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line"></span><br><span class="line">Trim_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -jar &#123;&#125; \</span></span><br><span class="line"><span class="string">PE -phred33 -threads &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; \</span></span><br><span class="line"><span class="string">ILLUMINACLIP:&#123;&#125;:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50"</span>\</span><br><span class="line">.format(cpu,trimmomatic_path,cpu,args[<span class="string">"reads1"</span>],args[<span class="string">"reads2"</span>]\</span><br><span class="line">,trim1_pa,trim1_up,trim2_pa,trim2_up,adaptor_path)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(Trim_cmd))</span><br><span class="line">subprocess.call(Trim_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: Trim Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 2 ####### </span></span><br><span class="line"><span class="comment">### bwa software should be downloaded ###</span></span><br><span class="line"><span class="comment">### You could modify the bwa path bellow</span></span><br><span class="line"><span class="comment">### Check for index file completely ###</span></span><br><span class="line">genome = <span class="string">"&#123;&#125;"</span>.format(args[<span class="string">"refer"</span>])</span><br><span class="line">bwa_index = <span class="string">"&#123;&#125;.amb"</span>.format(genome)</span><br><span class="line">genome_dict = <span class="string">'&#123;&#125;.dict'</span>.format(genome)</span><br><span class="line">fasta_index = <span class="string">'&#123;&#125;.fai'</span>.format(genome)</span><br><span class="line"><span class="keyword">if</span> <span class="keyword">not</span> Path(bwa_index).exists() <span class="keyword">or</span> <span class="keyword">not</span> Path(bwa_index).exists() <span class="keyword">or</span> <span class="keyword">not</span> Path(bwa_index).exists():</span><br><span class="line">    print(<span class="string">"some index file are missing, please offer bwa/samtools/picard index~"</span>)</span><br><span class="line">    print(<span class="string">"DATE STATUS: Failed at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line">    sys.exit()</span><br><span class="line"><span class="keyword">else</span>:</span><br><span class="line">    print(<span class="string">"index file are repared well"</span>)</span><br><span class="line">    RG = <span class="string">r"@RG\tID:&#123;&#125;\tPL:ILLUMINA\tLB:&#123;&#125;\tSM:&#123;&#125;\tPU:1"</span>.format(prefix,prefix,prefix)</span><br><span class="line">    print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 2 ... align the fastq files to &#123;&#125; \n\n\n"</span>.format(genome))</span><br><span class="line">    bwa_cmd = <span class="string">"&#123;&#125; mem -Y -t &#123;&#125; -R '&#123;&#125;' -M  &#123;&#125; &#123;&#125; &#123;&#125; | \</span></span><br><span class="line"><span class="string">    samtools view -Shb -o &#123;&#125;/&#123;&#125;.bam -"</span>\</span><br><span class="line">    .format(bwa_path,cpu,RG,genome,trim1_pa,trim2_pa,prefix,prefix)</span><br><span class="line">    print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(bwa_cmd))</span><br><span class="line">    subprocess.call(bwa_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">    print(<span class="string">"DATE STATUS: BWA MEM Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 3 ####### </span></span><br><span class="line"><span class="comment">#### picard fix mate information</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 3 ... Fix mate information in the BAM \n\n\n"</span>)</span><br><span class="line">fix_mate_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; FixMateInformation MAX_RECORDS_IN_RAM=2000000 \</span></span><br><span class="line"><span class="string">VALIDATION_STRINGENCY=SILENT ADD_MATE_CIGAR=True \</span></span><br><span class="line"><span class="string">ASSUME_SORTED=true I=&#123;&#125;/&#123;&#125;.bam O=&#123;&#125;/&#123;&#125;.fixmate.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,picard_path,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(fix_mate_cmd))</span><br><span class="line">subprocess.call(fix_mate_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: PICARD FIX MATE Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 4 ####### </span></span><br><span class="line"><span class="comment">#### picard sort bam</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 4 ... coordinate sort BAM \n\n\n"</span>)</span><br><span class="line">sort_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; SortSam MAX_RECORDS_IN_RAM=2000000 \</span></span><br><span class="line"><span class="string">VALIDATION_STRINGENCY=SILENT SORT_ORDER=coordinate \</span></span><br><span class="line"><span class="string">CREATE_INDEX=true I=&#123;&#125;/&#123;&#125;.fixmate.bam O=&#123;&#125;/&#123;&#125;.sorted.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,picard_path,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(sort_cmd))</span><br><span class="line">subprocess.call(sort_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: PICARD SORT SAM Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 5 ####### </span></span><br><span class="line"><span class="comment">#### picard mark duplication </span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 5 ... Mark PCR duplicates \n\n\n"</span>)</span><br><span class="line">mark_dump_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; MarkDuplicates MAX_RECORDS_IN_RAM=2000000 \</span></span><br><span class="line"><span class="string">VALIDATION_STRINGENCY=SILENT M=&#123;&#125;/&#123;&#125;.matrix \</span></span><br><span class="line"><span class="string">CREATE_INDEX=true I=&#123;&#125;/&#123;&#125;.sorted.bam O=&#123;&#125;/&#123;&#125;.dedup.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,picard_path,prefix,prefix,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(mark_dump_cmd))</span><br><span class="line">subprocess.call(mark_dump_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: PICARD MARK DUPLICATE Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 6 ####### </span></span><br><span class="line"><span class="comment">#### Realign indel intervels</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 6 ... Realign the indel intervels \n\n\n"</span>)</span><br><span class="line">realign_prepare = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T RealignerTargetCreator \</span></span><br><span class="line"><span class="string">-R &#123;&#125; -I &#123;&#125;/&#123;&#125;.dedup.bam -known &#123;&#125; \</span></span><br><span class="line"><span class="string">-o &#123;&#125;/realignment_targets.list"</span>.format(cpu,temp_path,gatk_path,genome,prefix,prefix,known_indel,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(realign_prepare))</span><br><span class="line">subprocess.call(realign_prepare, shell=<span class="literal">True</span>)</span><br><span class="line">realign_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T IndelRealigner -R &#123;&#125; \</span></span><br><span class="line"><span class="string">-I &#123;&#125;/&#123;&#125;.dedup.bam -targetIntervals &#123;&#125;/realignment_targets.list \</span></span><br><span class="line"><span class="string">-known &#123;&#125; -o &#123;&#125;/&#123;&#125;.realign.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,genome,prefix,prefix,prefix,known_indel,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(realign_cmd))</span><br><span class="line">subprocess.call(realign_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">index_cmd = <span class="string">"samtools index &#123;&#125;/&#123;&#125;.realign.bam"</span>.format(prefix,prefix)</span><br><span class="line">subprocess.call(index_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: GATK REALIGN  Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 7 ####### </span></span><br><span class="line"><span class="comment">#### Recalibrate base quality scores</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 7 ... Recalibrate base quality scores using known SNPs \n\n\n"</span>)</span><br><span class="line">recalibrate_prepare = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T BaseRecalibrator \</span></span><br><span class="line"><span class="string">--preserve_qscores_less_than 6 \</span></span><br><span class="line"><span class="string">-nct &#123;&#125; -R &#123;&#125; -I &#123;&#125;/&#123;&#125;.realign.bam \</span></span><br><span class="line"><span class="string">-XL chrX -XL chrY -XL chrM \</span></span><br><span class="line"><span class="string">-knownSites &#123;&#125; -knownSites &#123;&#125; -knownSites &#123;&#125; \</span></span><br><span class="line"><span class="string">-o &#123;&#125;/recal_data.table"</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,cpu,genome,prefix,prefix,mills,known_indel,db_snp,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(recalibrate_prepare))</span><br><span class="line">subprocess.call(recalibrate_prepare, shell=<span class="literal">True</span>)</span><br><span class="line">recalibrate_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T PrintReads \</span></span><br><span class="line"><span class="string">--disable_indel_quals \</span></span><br><span class="line"><span class="string">--preserve_qscores_less_than 6 \</span></span><br><span class="line"><span class="string">-nct &#123;&#125; -R &#123;&#125; -I &#123;&#125;/&#123;&#125;.realign.bam \</span></span><br><span class="line"><span class="string">-SQQ 10 -SQQ 20 -SQQ 30 -rf BadCigar \</span></span><br><span class="line"><span class="string">-BQSR &#123;&#125;/recal_data.table -o &#123;&#125;/&#123;&#125;.BQSR.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,cpu,genome,prefix,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(recalibrate_cmd))</span><br><span class="line">subprocess.call(recalibrate_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">index_cmd = <span class="string">"samtools index &#123;&#125;/&#123;&#125;.BQSR.bam"</span>.format(prefix,prefix)</span><br><span class="line">subprocess.call(index_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: GATK BQSR Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 8 ####### </span></span><br><span class="line"><span class="comment">#### Raw variant calls using HaplotypeCaller on single sample</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 8 ... CALL SNP !!! \n\n\n"</span>)</span><br><span class="line">callsnp_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; \</span></span><br><span class="line"><span class="string">-T HaplotypeCaller --genotyping_mode DISCOVERY \</span></span><br><span class="line"><span class="string">-A AlleleBalanceBySample -A DepthPerAlleleBySample \</span></span><br><span class="line"><span class="string">-A DepthPerSampleHC -A InbreedingCoeff \</span></span><br><span class="line"><span class="string">-A MappingQualityZeroBySample -A StrandBiasBySample \</span></span><br><span class="line"><span class="string">-A Coverage -A FisherStrand -A HaplotypeScore \</span></span><br><span class="line"><span class="string">-A MappingQualityRankSumTest -A MappingQualityZero \</span></span><br><span class="line"><span class="string">-A QualByDepth -A RMSMappingQuality \</span></span><br><span class="line"><span class="string">-A ReadPosRankSumTest -A VariantType -l INFO \</span></span><br><span class="line"><span class="string">--emitRefConfidence GVCF -rf BadCigar \</span></span><br><span class="line"><span class="string">--variant_index_parameter 128000 --variant_index_type LINEAR \</span></span><br><span class="line"><span class="string">-R &#123;&#125; -nct 4 -I &#123;&#125;/&#123;&#125;.BQSR.bam -o &#123;&#125;/&#123;&#125;.g.vcf "</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,genome,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(callsnp_cmd))</span><br><span class="line">subprocess.call(callsnp_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: GATK HaplotypeCaller Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line">print(<span class="string">"\n\n\n\t\tALl process has been finished, if you want check how many time consume \ you could use commond below"</span>)</span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tgrep 'DATE STATUS:' $log_file"</span>)</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/12/29/bwa_picard_pipeline/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/22/zhuanluzu/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">perl ~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/run_DE_analysis.pl --matrix final_reads_counts_reformat.txt --method DESeq2 --samples_file sample.list</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/12/22/zhuanluzu/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/22/grab_PGG/" >抓取PG的G网站上的各省份的输入的rs号的等位基因频率</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#@author:shy</span></span><br><span class="line"><span class="comment"># This script is used to grab PGG.han rs info</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="keyword">from</span> bs4 <span class="keyword">import</span> BeautifulSoup</span><br><span class="line"><span class="keyword">import</span> re</span><br><span class="line"><span class="keyword">import</span> requests, json</span><br><span class="line"><span class="keyword">import</span> urllib.request, urllib.error</span><br><span class="line"><span class="keyword">import</span> urllib</span><br><span class="line"><span class="keyword">import</span> xlwt</span><br><span class="line"><span class="keyword">import</span> sqlite3</span><br><span class="line"><span class="keyword">import</span> argparse</span><br><span class="line"><span class="keyword">import</span> time </span><br><span class="line"></span><br><span class="line"></span><br><span class="line">FIND_id = re.compile(<span class="string">r'id=(.*?)"&gt;'</span>)</span><br><span class="line">FIND_ref_alt = re.compile(<span class="string">r'-(.*?)-(.*?)&lt;'</span>)</span><br><span class="line">FIND_af = re.compile(<span class="string">r'&lt;td class="text-center"&gt;0.(.*?)&lt;/td&gt;\n&lt;td class="text-center"&gt;'</span>)</span><br><span class="line">FIND_final = re.compile(<span class="string">r'&lt;td class="text-center"&gt;(.*?)&lt;/td&gt;'</span>)</span><br><span class="line">FIND_all = re.compile(<span class="string">r'&lt;td class="text-center"&gt;(.*?)&lt;/td&gt;'</span>)</span><br><span class="line"></span><br><span class="line">ap = argparse.ArgumentParser()</span><br><span class="line">ap.add_argument(<span class="string">"-q"</span>, <span class="string">"--query"</span>, required=<span class="literal">True</span>,</span><br><span class="line">	help=<span class="string">"Input the rs id"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-p"</span>, <span class="string">"--provience"</span>, required=<span class="literal">True</span>,</span><br><span class="line">	help=<span class="string">"Input the Provience"</span>)</span><br><span class="line"></span><br><span class="line">args = vars(ap.parse_args())</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">main</span><span class="params">()</span>:</span></span><br><span class="line">    baseurl  = <span class="string">"https://www.hanchinesegenomes.org/HCGD/data/variant?term=&#123;&#125;"</span>.format(args[<span class="string">"query"</span>])</span><br><span class="line">    getData(baseurl)</span><br><span class="line">    </span><br><span class="line"></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">getData</span><span class="params">(baseurl)</span>:</span></span><br><span class="line">    html = askURL(baseurl)</span><br><span class="line">    soup = BeautifulSoup(html,<span class="string">"html.parser"</span>)</span><br><span class="line">    <span class="keyword">for</span> item <span class="keyword">in</span> soup.find_all(<span class="string">'div'</span>,class_=<span class="string">"table-responsive m-b-15"</span>):</span><br><span class="line">        item = str(item)</span><br><span class="line">        <span class="keyword">try</span>:</span><br><span class="line">            p_id = re.findall(FIND_id,item)[<span class="number">0</span>]</span><br><span class="line">            baseurl  = <span class="string">"https://www.hanchinesegenomes.org/HCGD/data/variant/detail?id=&#123;&#125;"</span>.format(p_id)</span><br><span class="line">            getData2(baseurl)</span><br><span class="line">        <span class="keyword">except</span>:</span><br><span class="line">            print(<span class="string">"&#123;&#125; is not in PGG database ~"</span>.format(args[<span class="string">"query"</span>]))</span><br><span class="line"></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">getData2</span><span class="params">(baseurl)</span>:</span></span><br><span class="line">    html = askURL(baseurl)</span><br><span class="line">    Provience = []</span><br><span class="line">    <span class="keyword">with</span> open(args[<span class="string">"provience"</span>]) <span class="keyword">as</span> input_p:</span><br><span class="line">        <span class="keyword">for</span> row <span class="keyword">in</span> input_p:</span><br><span class="line">            Provience.append(row.strip())</span><br><span class="line">    soup = BeautifulSoup(html,<span class="string">"html.parser"</span>)</span><br><span class="line">    <span class="keyword">for</span> item <span class="keyword">in</span> soup.find_all(<span class="string">'table'</span>,class_=<span class="string">"table table-custom-s table-condensed table-nowrap m-b-0"</span>):</span><br><span class="line">        item = str(item)</span><br><span class="line">        <span class="keyword">try</span>:</span><br><span class="line">            p_ref_alt = re.findall(FIND_ref_alt,item)[<span class="number">0</span>]</span><br><span class="line">            ref_alt = args[<span class="string">"query"</span>] + <span class="string">'\t'</span>+p_ref_alt[<span class="number">1</span>].replace(<span class="string">'-'</span>,<span class="string">'-&gt;'</span>)</span><br><span class="line">            af = re.findall(FIND_af,item)</span><br><span class="line">            all = re.findall(FIND_all,item)</span><br><span class="line">        <span class="keyword">except</span>:</span><br><span class="line">            <span class="keyword">continue</span></span><br><span class="line">        m = <span class="number">0</span></span><br><span class="line">        <span class="keyword">for</span> i <span class="keyword">in</span> all:</span><br><span class="line">            <span class="keyword">if</span> i <span class="keyword">in</span> Provience:</span><br><span class="line">                print(ref_alt+<span class="string">'\t'</span>+i+<span class="string">'\t'</span>+<span class="string">"0."</span>+af[m])</span><br><span class="line">                m += <span class="number">1</span></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">askURL</span><span class="params">(url)</span>:</span></span><br><span class="line">    <span class="comment"># simulate browser head info </span></span><br><span class="line">    head = &#123;</span><br><span class="line">        <span class="string">"User-Agent"</span>:<span class="string">"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36"</span></span><br><span class="line">        &#125;</span><br><span class="line">    requset = urllib.request.Request(url,headers=head)</span><br><span class="line">    html = <span class="string">""</span></span><br><span class="line">    <span class="keyword">try</span>:</span><br><span class="line">        response = urllib.request.urlopen(requset)</span><br><span class="line">        html = response.read().decode(<span class="string">"utf-8"</span>)</span><br><span class="line">    <span class="keyword">except</span> urllib.error.HTTPError <span class="keyword">as</span> e:</span><br><span class="line">        <span class="keyword">if</span> hasattr(e,<span class="string">"code"</span>):</span><br><span class="line">            print(e.code)</span><br><span class="line">        <span class="keyword">if</span> hasattr(e,<span class="string">"reason"</span>):</span><br><span class="line">            print(e.reason)</span><br><span class="line">    <span class="keyword">return</span> html</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> __name__ == <span class="string">"__main__"</span>:</span><br><span class="line">    main()</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/12/22/grab_PGG/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/22/可爱的人的喜好/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>最喜欢的导演</p>
<blockquote>
<p>新房昭之</p>
</blockquote>
<p>最喜欢的声优</p>
<blockquote>
<p>神谷浩史</p>
</blockquote>
<p>最喜欢的画师</p>
<blockquote>
<p>羽海野千花</p>
</blockquote>

	
	</div>
  <a type="button" href="/2020/12/22/可爱的人的喜好/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/08/1K_genome/" >主要记录千人基因组数据分析</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-08  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h3 id="Part-1"><a href="#Part-1" class="headerlink" title="Part 1."></a>Part 1.</h3><h4 id="数据下载"><a href="#数据下载" class="headerlink" title="数据下载"></a>数据下载</h4><blockquote>
<p>感觉再做一些可以去迅雷入职了</p>
</blockquote>
<p>首先下载alignment的list   </p>
<p>eg. igsr_Northern\ Han\ in\ China\ (HGDP)_undefined.tsv</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 0.提取第一列保存为1_ftp_sites</span></span><br><span class="line">awk '&#123;print $1&#125;' igsr_Northern\ Han\ in\ China\ \(HGDP\)_undefined.tsv.tsv</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 1.因为使用ascp下载 所以只需要数据在ebi数据库中的位置就可以，保存为2_temp_id</span></span><br><span class="line">awk -F 'uk/' '&#123;print $2&#125;' 1_ftp_sites &gt; 2_temp_id</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 2.使用<span class="built_in">echo</span>生成命令，并记录在3_run，方便以后查看哪里错了</span></span><br><span class="line">less 2_temp_id | while read line; </span><br><span class="line">do </span><br><span class="line">echo "ascp  -v -k 1 -l 200m -i \</span><br><span class="line">/Users/Yi-Liangwei/Applications/Aspera\ Connect.app/Contents/Resources/asperaweb_id_dsa.openssh \</span><br><span class="line">-Tr -Q -P33001 -L- fasp-g1k@fasp.1000genomes.ebi.ac.uk:"$&#123;line&#125; .; </span><br><span class="line">done &gt; 3_run </span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 3.下 载</span></span><br><span class="line">nohup sh 3_run</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 4.对于下载完成的cram文件 执行call snp环节</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 在所有操作之前，如果没有下载基因组的索引文件</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 可以使用下面两个命令生成</span></span><br><span class="line">samtools faidx Homo_sapiens_assembly38.fasta </span><br><span class="line"></span><br><span class="line">java -Xmx2g -jar ~/software/picard.jar CreateSequenceDictionary R=Homo_sapiens_assembly38.fasta O=Homo_sapiens_assembly38.dict</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 然后还需要准备三个文件，第一个是bam文件的prefix（CHS.info），第二个是rs位点的染色体和位置，</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 第三个是根据第二个文件生成的bed文件rs_id.bed</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 格式如下  CHS.info eg.</span></span><br><span class="line">HG00597</span><br><span class="line">HG00591</span><br><span class="line">HG00609</span><br><span class="line"><span class="meta">#</span><span class="bash"> rs_id eg.</span></span><br><span class="line">rs12044597      1       1708801</span><br><span class="line">rs11185092      1       107886278</span><br><span class="line">rs10923724      1       119546842</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> rs_id.bed用下面的命令生成</span></span><br><span class="line">awk -v OFS='\t' '&#123;print "chr"$2,$3-1,$3+1,$1&#125;' rs_id   &gt; rs_id.bed</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 5.cram格式转bam格式去call snp</span></span><br><span class="line">less CHS.info | while read line; do </span><br><span class="line">samtools view  -T /data/gujiaqi/698_download/Homo_sapiens_assembly38.fasta $&#123;line&#125;.final.cram  -L rs_id.bed -M -b -o $&#123;line&#125;.m.1.bam; done</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 6.call snp and parse vcf file</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 这时候还需要生成一个只包含rs编号的文件，两次的输入文件不一样的</span></span><br><span class="line">python bam2vcf.py -p CHS.info -rs rs_id</span><br><span class="line">python parse_vcf.py -p CHS.info -rs rs_number</span><br><span class="line"><span class="meta">#</span><span class="bash"> 所有结果会写在rs.genotype 和 rs.frequency中</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用的call snp软件为bcftools，如果时间不紧的话可以用GATK，将bcftools那一行改成</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> java -XX:+UseParallelGC -XX:ParallelGCThreads=8 -Xmx10g -jar \</span></span><br><span class="line">~/software/GenomeAnalysisTK.jar -T HaplotypeCaller --genotyping_mode DISCOVERY -A \</span><br><span class="line">AlleleBalanceBySample -A DepthPerAlleleBySample -A DepthPerSampleHC \</span><br><span class="line">-A InbreedingCoeff -A MappingQualityZeroBySample -A StrandBiasBySample\</span><br><span class="line">-A Coverage -A FisherStrand -A HaplotypeScore -A MappingQualityRankSumTest \</span><br><span class="line">-A MappingQualityZero -A QualByDepth -A RMSMappingQuality -A ReadPosRankSumTest \</span><br><span class="line">-A VariantType -l INFO --emitRefConfidence GVCF \</span><br><span class="line">-rf BadCigar --variant_index_parameter 128000 --variant_index_type LINEAR \</span><br><span class="line">-R /data/gujiaqi/698_download/Homo_sapiens_assembly38.fasta -I HG00597.merged.bam\</span><br><span class="line">-o HG00597.merged.gatk.gvcf</span><br></pre></td></tr></table></figure>

<h3 id="Part-2"><a href="#Part-2" class="headerlink" title="Part 2."></a>Part 2.</h3><h5 id="对vcf进行计算前还需要准备一下"><a href="#对vcf进行计算前还需要准备一下" class="headerlink" title="对vcf进行计算前还需要准备一下"></a>对vcf进行计算前还需要准备一下</h5><figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 对filtered出来的vcf文件压缩并索引</span></span><br><span class="line">ls *vcf | <span class="keyword">while</span> read line; <span class="keyword">do</span> bgzip $&#123;line&#125; -@ <span class="number">8</span>;tabix $&#123;line&#125;.gz; done</span><br><span class="line"></span><br><span class="line"><span class="comment"># 合并为一个文件，这一步如果直接合并所有的话，顺序默认按照文件夹内从小到大的顺序</span></span><br><span class="line"><span class="comment"># 后续会再rehead，当然也可以一个Group一个Group的合并，然后再把合并的合并</span></span><br><span class="line">bcftools merge *gz -O z -o <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz</span><br><span class="line"></span><br><span class="line"><span class="comment"># 查看vcf是否包含1185个位点</span></span><br><span class="line">less <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz | awk <span class="string">'&#123;if ($0 !~ "#")print&#125;'</span> | wc -l</span><br><span class="line"><span class="comment"># 或者可以直接用plink把rs号给提取出来，结果会保存在 -o 指定的 rs_check.snplist中</span></span><br><span class="line"><span class="comment"># 再根据rs_check.snplist 和之前的rs_id看一下就可以了</span></span><br><span class="line">plink --write-snplist -vcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz -out rs_check</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 使用excel做一个all_sample.info 就把之前那些.info文件和在一起</span></span><br><span class="line"><span class="comment"># all_sample.info 内容如下</span></span><br><span class="line"><span class="comment"># HG00709	SouthernHan	698dataset	698dataset_SouthernHan_HG00709</span></span><br><span class="line"><span class="comment"># HGDP01296	NorthernHan	HGDP	HGDP_NorthernHan_HGDP01296</span></span><br><span class="line"><span class="comment"># HGDP00828	Japanese	HGDP	HGDP_Japanese_HGDP00828</span></span><br><span class="line"><span class="comment"># HGDP00774	Han in China	HGDP	HGDP_Han in China_HGDP00774</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 为了下一步 reheader通过bcftools提取所有样本的id</span></span><br><span class="line">bcftools query -l <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz ｜<span class="keyword">while</span> read line; <span class="keyword">do</span> grep $&#123;line&#125; all_sample.info ; done | awk -F <span class="string">'\t'</span> <span class="string">'&#123;print $4&#125;'</span> &gt; vcf.sample.reheader.list</span><br><span class="line">bcftools reheader <span class="number">698</span>_HGDP.<span class="number">1185</span>.vcf.gz -s vcf.sample.reheader.list -o <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz</span><br></pre></td></tr></table></figure>

<h5 id="统计vcf文件中的sample和sites"><a href="#统计vcf文件中的sample和sites" class="headerlink" title="统计vcf文件中的sample和sites"></a>统计vcf文件中的sample和sites</h5><figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 统计vcf文件中的基本信息</span></span><br><span class="line"><span class="comment"># individual 的missing信息</span></span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --missing-indv  --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --missing-site --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --site-mean-depth  --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --site-quality --out vcf_check</span><br><span class="line">vcftools --gzvcf <span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  --het --out vcf_check</span><br><span class="line"><span class="comment"># sites 的missing信息</span></span><br><span class="line"><span class="comment"># vcf_check.imiss vcf_check.lmiss vcf_check.ldepth.mean vcf_check.lqual  vcf_check.het</span></span><br><span class="line"><span class="comment"># 上面就是生成的结果</span></span><br><span class="line"><span class="comment"># awk '&#123;print $3&#125;' vcf_check.ldepth.mean | sort -g -r 可以看平均深度有位点depth的平均深度只有20左右</span></span><br><span class="line"><span class="comment"># awk -F '\t' '&#123;print $4&#125;'  vcf_check.imiss | sort | uniq -c 可以看多少样本有缺失位点，有30个样本都是有缺失的，但是最多也就只缺两个位点</span></span><br><span class="line"><span class="comment"># awk -F '\t' '&#123;print $5&#125;'  vcf_check.lmiss | sort | uniq -c 可以看每个位点有多少个缺失，有两个位点有10个样本以上的缺失</span></span><br><span class="line"><span class="comment"># 需要注意的是 vcf_check.lmiss 中会默认我们的物种是二倍体，所以Number of DATA的数量是sample*2 </span></span><br><span class="line"><span class="comment"># 同理在这个文件中的missing sites都会变为 2 倍</span></span><br></pre></td></tr></table></figure>

<p>#####使用beagle填充缺失的基因型</p>
<figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">java -Xmx64g -jar ~/software/beagle.<span class="number">18</span>May20.d20.jar gt=<span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.vcf.gz  out=<span class="number">698</span>_HGDP.<span class="number">1185</span>.reheader.beagle impute=true</span><br></pre></td></tr></table></figure>

<p>#####population statistic</p>
<p>PCA</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 使用plink计算协方差matrix</span></span><br><span class="line">plink --vcf 698_HGDP.1185.reheader.beagle.vcf.gz --make-bed --double-id --pca  5 --out pca.5</span><br></pre></td></tr></table></figure>

<p>Structure</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 用admixture生成cluster matrix </span></span><br><span class="line">for i in &#123;2..5&#125;; do  admixture --cv pca.5.bed $i &gt; log$&#123;i&#125;.out; done</span><br><span class="line"><span class="meta">#</span><span class="bash"> 检查一下admixture的CV值，CV越小越好，但是这个只是一个推测，K还是需要自己判断</span></span><br><span class="line">grep "CV" *out | awk '&#123;print $3,$4&#125;' | sed -e 's/(//;s/)//;s/://'</span><br><span class="line">rename pca.5. structure. *Q</span><br></pre></td></tr></table></figure>

<p>Fst</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 生成四个group的id的统计文件</span></span><br><span class="line">bcftools query -l 698_HGDP.1185.reheader.beagle.vcf.gz | grep 'Sou' &gt; SouthernHan.info</span><br><span class="line">bcftools query -l 698_HGDP.1185.reheader.beagle.vcf.gz | grep 'ni' &gt; HaninChina.info</span><br><span class="line">bcftools query -l 698_HGDP.1185.reheader.beagle.vcf.gz | grep 'Jap' &gt; Japanese.info</span><br><span class="line">bcftools query -l 698_HGDP.1185.reheader.beagle.vcf.gz | grep 'NorthernHan' &gt; NorthernHan.info</span><br></pre></td></tr></table></figure>

<p>然后转到R里面画这两个图</p>
<p>需要用到的包 tidyverse ggsci ggplot2 gridExtra pophelper</p>
<p>结果会生成两个图 PCA.pdf 和 Structure.pdf</p>
	
	</div>
  <a type="button" href="/2020/12/08/1K_genome/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/12/06/NTFS硬盘挂载/" >挂载硬盘至服务器</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-12-06  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>挂载的意思就是。硬盘本身插在接口上是有一个物理位置的。位于 /dev 文件夹内</p>
<p>在dev文件夹中 sda sdb sdc 就代表了三个硬盘，/dev是不可以直接查看里面的内容的</p>
<p>我们平常访问他们里面的东西需要把他从/dev中挂载到一个方便的目录中。</p>
<p>比如windows系统里 /dev/sda1被挂载在C:     就是C盘</p>
<p>在linux中通常习惯为挂载在/mnt目录中</p>
<p>通常使用的硬盘格式有 NTFS exFAT   EXT？FAT..</p>
<p>现在买回来的硬盘通常为NTFS</p>
<p>但是linux自身并不支持NTFS格式的硬盘，但有相关工具ntfs-3g可以挂载ntfs</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">mount -t ntfs /dev/sdf /mnt/temp</span><br><span class="line">mount -t ntfs /dev/sdg /mnt/temp2</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/12/06/NTFS硬盘挂载/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/07/02/mscanX/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-07-02  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br><span class="line">blastp -query longest.nil.pep.fa -db nil.pep -max_target_seqs 5 -outfmt 6 -num_threads 10 -out nil.blast &amp;</span><br><span class="line">blastp -query longest.trifida.pep.fa -db trifida.pep -max_target_seqs 5 -outfmt 6 -num_threads 10 -out trifida.blast &amp;</span><br><span class="line">blastp -query longest.triloba.pep.fa -db triloba.pep -max_target_seqs 5 -outfmt 6 -num_threads 10 -out triloba.blast &amp;</span><br></pre></td></tr></table></figure>

<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><span class="line">perl MCScanX-transposed.pl -x 6 -i data -t batatas -c trifida,triloba,nil,sly,Os,at -o result/batatas6</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">perl MCScanX-transposed.pl -x 5 -i data -t trifida -c triloba,nil,sly,Os,at -o result/trifida5</span><br><span class="line">perl MCScanX-transposed.pl -x 4 -i data -t triloba -c nil,sly,Os,at -o result/triloba4</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">perl MCScanX-transposed.pl -x 3 -i data -t nil -c sly,at,Os -o result/nil3</span><br><span class="line">perl MCScanX-transposed.pl -x 3 -i data -t trifida -c sly,at,Os -o result/trifida3</span><br><span class="line">perl MCScanX-transposed.pl -x 3 -i data -t triloba -c sly,at,Os -o result/triloba3</span><br><span class="line">perl MCScanX-transposed.pl  -x 3 -i data -t batatas -c sly,at,Os -o result/batatas3</span><br><span class="line">perl detect_dup_modes_for_a_family.pl -i ItbSPL.id -d result/triloba3/triloba -o triloba3.ItbSPL</span><br><span class="line">perl detect_dup_modes_for_a_family.pl -i ItfSPL.id -d result/trifida3/trifida -o trifida3.ItfSPL</span><br><span class="line">perl detect_dup_modes_for_a_family.pl -i InilSPL.id -d result/nil3/nil -o nil3.InilSPL</span><br><span class="line">perl detect_dup_modes_for_a_family.pl -i IbSPL.id -d result/batatas3/batatas -o batatas3.IbSPL</span><br><span class="line">ls *SPL | while read line; do python rep.py SPL.replace.id $&#123;line&#125; &gt; $&#123;line&#125;.rename; done</span><br><span class="line">java annotate_tree_with_dup_modes -t InilSPL.nwk  -s nil3.InilSPL.rename -o nil3.InilSPL.png -x 800 -y 1600</span><br><span class="line">java annotate_tree_with_dup_modes -t ItfSPL.nwk  -s trifida3.ItfSPL.rename -o trifida3.ItfSPL.png -x 800 -y 1600</span><br><span class="line">java annotate_tree_with_dup_modes -t ItbSPL.nwk  -s triloba3.ItbSPL.rename -o triloba3.ItbSPL.png -x 800 -y 1600</span><br><span class="line">java annotate_tree_with_dup_modes -t IbSPL.nwk  -s batatas3.IbSPL.rename -o batatas3.IbSPL.png -x 800 -y 1600</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">java annotate_tree_with_dup_modes -t batatas.spl.29.muscle.gb.nwk -s result/batatas6.dup.re -o result/batatas6.png -x 800 -y 1600</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/07/02/mscanX/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/06/22/KaKs_calculator/" >KaKs 计算进化速率_update</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-06-22  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h3 id="首先我们还是需要用蛋白的比对文件，去匹配cds的比对文件"><a href="#首先我们还是需要用蛋白的比对文件，去匹配cds的比对文件" class="headerlink" title="首先我们还是需要用蛋白的比对文件，去匹配cds的比对文件"></a>首先我们还是需要用蛋白的比对文件，去匹配cds的比对文件</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"># rename.6merge.ms.pep 为蛋白全长的比对文件,在使用kaks的时候需要nogap命令，不然还是会报错</span><br><span class="line">perl ~/Downloads/softwares/pal2nal.v14/pal2nal.pl rename.6merge.ms.pep rename.6merge.cds -nogap &gt; rename.6merge.ms.pal2nal.clu</span><br><span class="line"></span><br><span class="line"># 然后转换为成对的基因对的格式</span><br><span class="line">AXTConvertor rename.6merge.ms.pal2nal.clu rename.6merge.ms.pal2nal.AXT</span><br><span class="line"></span><br><span class="line"># 然后再是用kaks calculator对 axt格式的文件 输出其kaks等信息 ，-m选择model为YN，如果选择model为MA的话大概需要花费20行个小时， 但是如果使用YN的话只需要十几秒，暂时没有找到有什么具体的差异，算出来的方式有一点差异。MA的话是如果 有三种异常值，一个是50</span><br><span class="line">一个是NA 一个是 0.0001，具体解释如下：如果只有一个非同一突变的位点那就是0.0001，如果只有一个同义突变的位点 那就是50，如果啥都没有，完全一样那就是NA。当模型选为YN的话，有两种异常值 -0。和 就不再计算NA或者50的位点。。同时有许多对基因</span><br><span class="line"># 发现不同的模型之间差距还是蛮大的。有可能有两倍的差距</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">KaKs_Calculator -i rename.6merge.ms.pal2nal.AXT -o rename.6merge.ms.pal2nal.kaks -m YN -c 11</span><br><span class="line">awk -F &apos;\t&apos; -v OFS=&apos;\t&apos; &apos;&#123;print $1,$5&#125;&apos; rename.6merge.ms.pal2nal.kaks  | awk -F &apos;\&amp;&apos; -v OFS=&apos;\t&apos; &apos;&#123;print $1,$2&#125;&apos;</span><br></pre></td></tr></table></figure>

<p>Table 2. Methods Incorporated in KaKs_Calculator</p>
<table>
<thead>
<tr>
<th>Approximate method</th>
<th></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody><tr>
<td>Method</td>
<td>Mutation model<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn1" target="_blank" rel="noopener">#1</a></td>
<td>Reference</td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>Step 1</td>
<td>Step 2</td>
<td>Step 3</td>
<td></td>
</tr>
<tr>
<td>NG</td>
<td>JC</td>
<td>JC</td>
<td>JC</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib26" target="_blank" rel="noopener"><em>26</em></a></td>
</tr>
<tr>
<td>LWL</td>
<td>JC</td>
<td>K2P</td>
<td>K2P</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib28" target="_blank" rel="noopener"><em>28</em></a></td>
</tr>
<tr>
<td>MLWL</td>
<td>K2P</td>
<td>K2P</td>
<td>K2P</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib30" target="_blank" rel="noopener"><em>30</em></a></td>
</tr>
<tr>
<td>LPB</td>
<td>—<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn3" target="_blank" rel="noopener">*</a></td>
<td>—<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn3" target="_blank" rel="noopener">*</a></td>
<td>K2P</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib25" target="_blank" rel="noopener">25.</a>, <a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib29" target="_blank" rel="noopener">29.</a></td>
</tr>
<tr>
<td>MLPB</td>
<td>—<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn3" target="_blank" rel="noopener">*</a></td>
<td>—<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn3" target="_blank" rel="noopener">*</a></td>
<td>K2P</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib30" target="_blank" rel="noopener"><em>30</em></a></td>
</tr>
<tr>
<td>YN</td>
<td>HKY</td>
<td>HKY</td>
<td>HKY</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib27" target="_blank" rel="noopener"><em>27</em></a></td>
</tr>
<tr>
<td>MYN</td>
<td>TN</td>
<td>TN</td>
<td>TN</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib31" target="_blank" rel="noopener"><em>31</em></a></td>
</tr>
<tr>
<td>Maximum likelihood method</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Method</td>
<td>Mutation model<a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#tbl2fn2" target="_blank" rel="noopener">#2</a></td>
<td>Reference</td>
<td></td>
<td></td>
</tr>
<tr>
<td>GY</td>
<td>HKY</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib19" target="_blank" rel="noopener">19.</a>, <a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib20" target="_blank" rel="noopener">20.</a></td>
<td></td>
<td></td>
</tr>
<tr>
<td>MS</td>
<td>a model that has the smallest AICC among 14 candidate models</td>
<td>Model-selected method proposed in this study</td>
<td></td>
<td></td>
</tr>
<tr>
<td>MA</td>
<td>a model that averages parameters across 14 candidate models</td>
<td>Model-averaged method proposed in this study</td>
<td></td>
<td></td>
</tr>
</tbody></table>
<p>#1. The approximate method involves three basic steps: Step 1: counting the numbers of synonymous and nonsynonymous sites; Step 2: calculating the numbers of synonymous and nonsynonymous substitutions; Step 3: correcting for multiple substitutions.</p>
<p>#2. The maximum likelihood method uses the probability theory to finish the three steps in one go <a href="https://www.sciencedirect.com/science/article/pii/S1672022907600072?via%3Dihub#bib4" target="_blank" rel="noopener">(<em>4</em>)</a>.</p>
<p>*No specific definition of synonymous and nonsynonymous sites or substitutions.</p>
<h4 id="有个想法-计算每个物种对之间的kaks比值"><a href="#有个想法-计算每个物种对之间的kaks比值" class="headerlink" title="有个想法    计算每个物种对之间的kaks比值"></a>有个想法    计算每个物种对之间的kaks比值</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line">less Ipomoea.id | while read line</span><br><span class="line">do</span><br><span class="line">echo &quot;python get_target_kaks.py -r rename.6merge.ms.pal2nal.clu -1 At -2 &quot;$&#123;line&#125;&quot; &gt; At&quot;$&#123;line&#125;&quot;.clu&quot;</span><br><span class="line">done    </span><br><span class="line"></span><br><span class="line">python get_target_kaks.py -r rename.6merge.ms.pal2nal.clu -1 Ib -2 Sly &gt; IbSly.clu</span><br><span class="line">python get_target_kaks.py -r rename.6merge.ms.pal2nal.clu -1 Ib -2 Inil &gt; IbInil.clu</span><br><span class="line">python get_target_kaks.py -r rename.6merge.ms.pal2nal.clu -1 Ib -2 Itb &gt; IbItb.clu</span><br><span class="line">python get_target_kaks.py -r rename.6merge.ms.pal2nal.clu -1 Ib -2 Itf &gt; IbItf.clu</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">#计算所有甘薯属和番茄的关系，计算所有番薯属和拟南芥的关系</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># 给他们加上clustal的标识head</span><br><span class="line">ls S*clu | while read line                      </span><br><span class="line">do</span><br><span class="line">cat head $&#123;line&#125; &gt; $&#123;line&#125;.ok</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># 转换为AXT格式</span><br><span class="line">ls *clu.ok  | sed &apos;s/.clu.ok//g&apos; |while read line</span><br><span class="line">do</span><br><span class="line">AXTConvertor $&#123;line&#125;.clu.ok $&#123;line&#125;.axt</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line"># 生成kaks文件</span><br><span class="line">ls *axt| sed &apos;s/.axt//g&apos;  | while read line; do  echo &quot;KaKs_Calculator -i &quot;$&#123;line&#125;&quot;.axt -o &quot;$&#123;line&#125;&quot;.kaks -m MA -c 11 &amp;&quot;; done</span><br></pre></td></tr></table></figure>


	
	</div>
  <a type="button" href="/2020/06/22/KaKs_calculator/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/06/16/WGCNA/" ></a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-06-16  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>##RSEM</p>
<h3 id="1-Get-matrix-of-gene-expression-FPKM-or-TPM-or-reads-count"><a href="#1-Get-matrix-of-gene-expression-FPKM-or-TPM-or-reads-count" class="headerlink" title="1. Get matrix of gene expression (FPKM or TPM),or reads count"></a>1. Get matrix of gene expression (FPKM or TPM),or reads count</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"> </span><br><span class="line"><span class="meta"> #</span><span class="bash">  rsem-generate-data-matrix is a perl script, be sure of the prameter inside is <span class="built_in">set</span> as getting counts or FPKM or TPM</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> vim pl file to make target col as 6 <span class="keyword">for</span> FPKM</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> vim pl file to make target col as 5 <span class="keyword">for</span> TPM</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> vim pl file to make target col as 4 <span class="keyword">for</span> counts</span></span><br><span class="line"><span class="meta"> #</span><span class="bash"><span class="comment"># rsem-generate-data-matrix PATH</span></span></span><br><span class="line"> rsem-generate-data-matrix results_file/S*gene* &gt; 30.counts.matrix</span><br><span class="line"> </span><br><span class="line"><span class="meta"> #</span><span class="bash"><span class="comment"># remove reduant information</span></span></span><br><span class="line"> sed -i 's/results_file\///g' 30.counts.matrix </span><br><span class="line"> sed -i 's/.genes.results//g' 30.counts.matrix</span><br><span class="line"> sed -i 's/"//g' 30.counts.matrix </span><br><span class="line"> awk '&#123;if( $0 !~"^L")print $0&#125;' 30.counts.matrix &gt; 30.count.matrix </span><br><span class="line"> </span><br><span class="line"></span><br><span class="line"><span class="meta"> #</span><span class="bash"><span class="comment"># vim rsem-generate-data-matrix to get </span></span></span><br><span class="line"> vim ~/software/RSEM-1.3.2/rsem-generate-data-matrix</span><br><span class="line"> rsem-generate-data-matrix S*gene* &gt; 39.FPKM.matrix</span><br><span class="line"> awk '&#123;if( $0 !~"^L")print $0&#125;' 30.FPKM.matrix &gt; 30.count.matrix</span><br></pre></td></tr></table></figure>

<p>###2. Primary filter</p>
<p>#####filter condition: counts &gt; 10 in 90 percent samples </p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># filter by counts &gt;10 in 90 percent samples</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line">ab = &#123;&#125;</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> count_matrix:</span><br><span class="line">    head_include = count_matrix.readline().split(<span class="string">'\t'</span>)</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> count_matrix:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        m = <span class="number">0</span></span><br><span class="line">        <span class="keyword">for</span> i <span class="keyword">in</span> range(len(lst1)<span class="number">-1</span>):</span><br><span class="line">            <span class="keyword">if</span> float(lst1[i+<span class="number">1</span>].strip()) &gt; <span class="number">10.0</span>:</span><br><span class="line">                m += <span class="number">1</span></span><br><span class="line">        <span class="keyword">if</span> m &gt; int(<span class="number">0.9</span>*len(head_include)):</span><br><span class="line">            ab[lst1[<span class="number">0</span>]] = <span class="string">"yes"</span></span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            ab[lst1[<span class="number">0</span>]] = <span class="string">"bad"</span></span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">2</span>]) <span class="keyword">as</span> fpkm_matrix:</span><br><span class="line">    print(fpkm_matrix.readline().rstrip())</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> fpkm_matrix:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        <span class="keyword">if</span> ab[lst1[<span class="number">0</span>]] == <span class="string">"yes"</span>:</span><br><span class="line">            print(row.strip())</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> use reads counts filter FPKM matrix</span></span><br><span class="line">python f.py  30.counts.matrix 30.FPKM.noLG.matrix  &gt; 30.FPKM.noLG.filtered.matrix</span><br></pre></td></tr></table></figure>

<h3 id="3-trinity"><a href="#3-trinity" class="headerlink" title="3.trinity"></a>3.trinity</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/PtR --matrix 32.filter.noLG.FPKM.matrix --samples 32samples.txt --log2 --min_rowSums 10 --compare_replicates</span><br><span class="line">~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/PtR --matrix  32.filter.noLG.FPKM.matrix --min_rowSums 10 -s 32samples.txt --log2 --CPM --sample_cor_matrix</span><br><span class="line">~/software/trinityrnaseq-v2.9.1/Analysis/DifferentialExpression/PtR --matrix  32.filter.noLG.FPKM.matrix -s 32samples.txt --min_rowSums 10 --log2 --CPM --center_rows   --prin_comp 8</span><br></pre></td></tr></table></figure>

<p>##WGCNA</p>
<h3 id="How-many-samples-does-WGCNA-need"><a href="#How-many-samples-does-WGCNA-need" class="headerlink" title="How many samples does WGCNA need ?"></a>How many samples does WGCNA need ?</h3><p>20 samples</p>
<p>If at all possible, one should <strong>have</strong> at least 20 <strong>samples</strong>; as with <strong>any</strong> analysis methods, more <strong>samples</strong> usually lead to more robust and refined results</p>
<h3 id="If-replicated-samples-could-use-as-individuals"><a href="#If-replicated-samples-could-use-as-individuals" class="headerlink" title="If replicated samples could use as individuals?"></a>If replicated samples could use as individuals?</h3><p>There is a biostar question about that</p>
<blockquote>
<p>Q: Hi, in WGCNA manual I learn that the minimum sample size for WGCNA is 15. If I have 10 different treatment, each treatment has three replicates, do I have 30 samples or just 10 samples?</p>
<p>A: The value of 15 seems arbitrary to me - I am not sure that it is a hard threshold. If the developer of WGCNA went by the Aztec system of numbers, maybe he would have said that 20 is the minimum (Aztecs had 20-base system of numbers).</p>
</blockquote>
<p>This answer get 3 stars include me :)</p>
<blockquote>
<p><a href="https://www.biostars.org/p/342696/" target="_blank" rel="noopener">https://www.biostars.org/p/342696/</a></p>
</blockquote>
<p>###Does gene set need normalized?</p>
<p>Dec 24, 2017 - These <em>normalization</em> methods make a big difference if one wants to compare expression of gene A to expression of gene B; but <em>WGCNA</em> calculates correlations for which gene-wise scaling factors make no difference. (Sample-wise scaling factors of course do, so samples do <em>need</em> to be <em>normalized</em>.)</p>
<h3 id="Whether-should-I-filter-gene-sets"><a href="#Whether-should-I-filter-gene-sets" class="headerlink" title="Whether  should I filter gene sets"></a>Whether  should I filter gene sets</h3><blockquote>
<p>offical:</p>
<p>We <strong>do</strong> not recommend <strong>filtering genes</strong> by differential expression. <strong>WGCNA</strong> is designed to be an unsupervised analysis method that clusters <strong>genes</strong> based on their expression profiles.</p>
</blockquote>
<p>But it really need a lot memory, 16G PC is not available for gene sets large than 50k</p>
<h3 id="Answear-two-forementioned-question"><a href="#Answear-two-forementioned-question" class="headerlink" title="Answear two forementioned question :"></a>Answear two forementioned question :</h3><p>Can WGCNA be used to analyze RNA-Seq data?</p>
<p>Yes. As far as WGCNA is concerned, working with (properly normalized) RNA-seq data isn’t really any different from working with (properly normalized) microarray data.</p>
<p><strong>We suggest removing features whose counts are consistently low</strong> (for example, removing all features that have a count of less than say 10 in more than 90% of the samples) because such low-expressed features tend to reflect noise and correlations based on counts that are mostly zero aren’t really meaningful. The actual thresholds should be based on experimental design, sequencing depth and sample counts.</p>
<p><strong>We then recommend a variance-stabilizing transformation. For example,</strong> <strong>package DESeq2 implements the function</strong> <strong>varianceStabilizingTransformation which we have found useful, but one</strong> <strong>could also start with normalized counts (or RPKM/FPKM data) and</strong> <strong>log-transform them using log2(x+1).</strong> For highly expressed features, the differences between full variance stabilization and a simple log transformation are small.</p>
<p><strong>Whether one uses RPKM, FPKM, or simply normalized counts doesn’t make</strong> <strong>a whole lot of difference for WGCNA analysis as long as all samples</strong> <strong>were processed the same way.</strong> These normalization methods make a big difference if one wants to compare expression of gene A to expression of gene B; but WGCNA calculates correlations for which gene-wise scaling factors make no difference. (Sample-wise scaling factors of course do, so samples do need to be normalized.)</p>
<p>If data come from different batches, we recommend to check for batch effects and, if needed, adjust for them. We use ComBat for batch effect removal but other methods should also work.</p>
<p>Finally, we usually check quantile scatterplots to make sure there are no systematic shifts between samples; if sample quantiles show correlations (which they usually do), quantile normalization can be used to remove this effect.</p>
<blockquote>
<p><a href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html" target="_blank" rel="noopener">OFFICIAL FAQ</a></p>
</blockquote>
<h5 id="Final-I-decided"><a href="#Final-I-decided" class="headerlink" title="Final I decided:"></a>Final I decided:</h5><p>The authors do recommend to remove the “noisier” genes, either by mean expression or variance, but they also recommmend to not filter by differential expression.</p>
<ol>
<li>Use reads count to filter: if reads count of one gene in 90% samples is less than <strong><em>10</em></strong> , remove it (i got 20902 genes,15 SPL genes included finally)    </li>
</ol>
<h2 id="WGCNA-Process"><a href="#WGCNA-Process" class="headerlink" title="WGCNA Process"></a>WGCNA Process</h2><h4 id="Samples-30-samples-during-5-period-of-batatas-devolopment"><a href="#Samples-30-samples-during-5-period-of-batatas-devolopment" class="headerlink" title="Samples: 30 samples during 5 period of batatas devolopment"></a>Samples: 30 samples during 5 period of batatas devolopment</h4><p>10TR 20TR 30FR 30SR 40FR 40SR 50FR 50SR</p>
<p>Only after 30 days, we could recognize SR based on the diameter &gt; 2.5mm (and DAT10TR DAT20TR is FR).</p>
<p>each period has 4 rep</p>
<p>go</p>
<h4 id="Target-Use-WGCNA-distinguishes-which-genes-have-the-same-express-pattern-and-find-the-gene-most-correlated-to-SPL-genes-in-the-same-module"><a href="#Target-Use-WGCNA-distinguishes-which-genes-have-the-same-express-pattern-and-find-the-gene-most-correlated-to-SPL-genes-in-the-same-module" class="headerlink" title="Target: Use WGCNA distinguishes which genes have the same express pattern and find the gene most correlated to SPL genes in the same module"></a>Target: Use WGCNA distinguishes which genes have the same express pattern and find the gene most correlated to <em>SPL</em> genes in the same module</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(WGCNA)</span><br><span class="line">allowWGCNAThreads(<span class="number">12</span>)</span><br><span class="line">options(stringsAsFactors = <span class="literal">FALSE</span>)</span><br><span class="line">setwd(<span class="string">"~/DATA/WGCNA/"</span>)</span><br><span class="line">expro=read.table(<span class="string">'rename.30.FPKM.noLG.filtered.matrix'</span>,row.names = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># filter most 25% variant genes</span></span><br><span class="line"><span class="comment">#m.vars=apply(expro,1,var)</span></span><br><span class="line"><span class="comment">#expro.upper=expro[which(m.vars&gt;quantile(m.vars, probs = seq(0, 1, 0.25))[4]),]</span></span><br><span class="line"><span class="comment">#dim(expro.upper)</span></span><br><span class="line"><span class="comment">#datExpr=as.data.frame(t(expro.upper));</span></span><br><span class="line"><span class="comment">#nGenes = ncol(datExpr)</span></span><br><span class="line"><span class="comment">#nSamples = nrow(datExpr)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># do not perform filter</span></span><br><span class="line">datExpr=as.data.frame(t(expro));</span><br><span class="line">nGenes = ncol(datExpr)</span><br><span class="line">nSamples = nrow(datExpr)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## check outlier</span></span><br><span class="line">gsg = goodSamplesGenes(datExpr, verbose = <span class="number">3</span>);</span><br><span class="line">gsg$allOK</span><br><span class="line">sampleTree = hclust(dist(datExpr), method = <span class="string">"average"</span>)</span><br><span class="line">plot(sampleTree, main = <span class="string">"Sample clustering to detect outliers"</span></span><br><span class="line">     , sub=<span class="string">""</span>, xlab=<span class="string">""</span>)</span><br><span class="line"><span class="comment"># save(datExpr, file = "FPKM-01-dataInput.RData")</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## if outlier exsit, then remove gsg False genes using this function </span></span><br><span class="line"><span class="comment">#if (!gsg$allOK)&#123;</span></span><br><span class="line"><span class="comment">## Optionally, print the gene and sample names that were removed:</span></span><br><span class="line"><span class="comment">#if (sum(!gsg$goodGenes)&gt;0) </span></span><br><span class="line"><span class="comment"># printFlush(paste("Removing genes:", </span></span><br><span class="line"><span class="comment"># paste(names(datExpr)[!gsg$goodGenes], collapse = ",")));</span></span><br><span class="line"><span class="comment">#if (sum(!gsg$goodSamples)&gt;0) </span></span><br><span class="line"><span class="comment"># printFlush(paste("Removing samples:", </span></span><br><span class="line"><span class="comment"># paste(rownames(datExpr)[!gsg$goodSamples], collapse = ",")));</span></span><br><span class="line"><span class="comment"># Remove the offending genes and samples from the data:</span></span><br><span class="line"><span class="comment">#datExpr = datExpr[gsg$goodSamples, gsg$goodGenes]</span></span><br><span class="line"><span class="comment">#&#125;</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## Log2 fold change</span></span><br><span class="line"><span class="comment">#datExpr = log2(datExpr+1)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## try different soft threshold,from 1 to 25</span></span><br><span class="line">powers = c(seq(<span class="number">1</span>,<span class="number">25</span>,by = <span class="number">1</span>))</span><br><span class="line">sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = <span class="number">5</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">## plot the 25 value</span></span><br><span class="line">par(mfrow = c(<span class="number">1</span>,<span class="number">2</span>));</span><br><span class="line">cex1 = <span class="number">0.8</span>;</span><br><span class="line">plot(sft$fitIndices[,<span class="number">1</span>], -sign(sft$fitIndices[,<span class="number">3</span>])*sft$fitIndices[,<span class="number">2</span>],</span><br><span class="line">     xlab=<span class="string">"Soft Threshold (power)"</span>,ylab=<span class="string">"Scale Free Topology Model Fit,signed R^2"</span>,type=<span class="string">"n"</span>,</span><br><span class="line">     main = paste(<span class="string">"Scale independence"</span>));</span><br><span class="line">text(sft$fitIndices[,<span class="number">1</span>], -sign(sft$fitIndices[,<span class="number">3</span>])*sft$fitIndices[,<span class="number">2</span>],</span><br><span class="line">     labels=powers,cex=cex1,col=<span class="string">"red"</span>);</span><br><span class="line">abline(h=<span class="number">0.80</span>,col=<span class="string">"red"</span>)</span><br><span class="line">plot(sft$fitIndices[,<span class="number">1</span>], sft$fitIndices[,<span class="number">5</span>],</span><br><span class="line">     xlab=<span class="string">"Soft Threshold (power)"</span>,ylab=<span class="string">"Mean Connectivity"</span>, type=<span class="string">"n"</span>,</span><br><span class="line">     main = paste(<span class="string">"Mean connectivity"</span>))</span><br><span class="line">text(sft$fitIndices[,<span class="number">1</span>], sft$fitIndices[,<span class="number">5</span>], labels=powers, cex=cex1,col=<span class="string">"red"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">#### get an estimate power value(Closest to 0.9)</span></span><br><span class="line">sft$powerEstimate</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## One-step network construction and module detection##</span></span><br><span class="line">cor = WGCNA::cor</span><br><span class="line">net = blockwiseModules(</span><br><span class="line">  <span class="comment"># input data</span></span><br><span class="line">  datExpr, </span><br><span class="line">  <span class="comment"># calculate correlation coefficient</span></span><br><span class="line">  power = sft$powerEstimate, </span><br><span class="line">  corType = <span class="string">"pearson"</span>,</span><br><span class="line">  networkType = <span class="string">"unsigned"</span>,</span><br><span class="line">  maxBlockSize = <span class="number">6000</span>,</span><br><span class="line">  TOMType = <span class="string">"unsigned"</span>,</span><br><span class="line">  deepSplit = <span class="number">2</span>,</span><br><span class="line">  minModuleSize = <span class="number">30</span>,</span><br><span class="line">  reassignThreshold = <span class="number">0</span>,</span><br><span class="line">  <span class="comment"># distance among modules less than 0.15, otherwise correlation among module more than 0.85 will be merged</span></span><br><span class="line">  mergeCutHeight = <span class="number">0.15</span>,</span><br><span class="line">  numericLabels = <span class="literal">TRUE</span>, </span><br><span class="line">  pamRespectsDendro = <span class="literal">FALSE</span>,)</span><br><span class="line"><span class="comment">## show how many modules identified</span></span><br><span class="line">table(net$colors)</span><br><span class="line"><span class="comment">## plot results</span></span><br><span class="line"><span class="comment">## Convert labels to colors for plotting</span></span><br><span class="line">mergedColors = labels2colors(net$colors)</span><br><span class="line"><span class="comment">## Plot the dendrogram and the module colors underneath</span></span><br><span class="line">plotDendroAndColors(net$dendrograms[[<span class="number">1</span>]], mergedColors[net$blockGenes[[<span class="number">1</span>]]],<span class="string">"Module colors"</span>,</span><br><span class="line">                    dendroLabels = <span class="literal">FALSE</span>, hang = <span class="number">0.03</span>,</span><br><span class="line">                    addGuide = <span class="literal">TRUE</span>, guideHang = <span class="number">0.05</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## get all edge file and node file</span></span><br><span class="line"></span><br><span class="line">moduleLabels = net$colors</span><br><span class="line">moduleColors = labels2colors(net$colors)</span><br><span class="line">modProbes = names(datExpr)</span><br><span class="line">modules = <span class="string">"all_without_normalize"</span></span><br><span class="line">TOM = TOMsimilarityFromExpr(datExpr, power = sft$powerEstimate);</span><br><span class="line">cyt = exportNetworkToCytoscape(TOM,</span><br><span class="line">                               edgeFile = paste(<span class="string">"CytoscapeInput-edges-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               nodeFile = paste(<span class="string">"CytoscapeInput-nodes-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               weighted = <span class="literal">TRUE</span>,</span><br><span class="line">                               threshold = <span class="number">0.02</span>,</span><br><span class="line">                               nodeNames = modProbes,</span><br><span class="line">                               nodeAttr = moduleColors)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Or retrieve specific module edges</span></span><br><span class="line">modules = c(<span class="string">"turquoise"</span>)</span><br><span class="line"><span class="comment"># 要可视化的基因</span></span><br><span class="line">probes = colnames(datExpr)</span><br><span class="line">inModule = is.finite(match(moduleColors, modules));</span><br><span class="line">modProbes = probes[inModule]</span><br><span class="line"><span class="comment"># 候选基因的 TOM</span></span><br><span class="line">modTOM = TOM[inModule, inModule]</span><br><span class="line"></span><br><span class="line">dimnames(modTOM) = list(modProbes, modProbes)</span><br><span class="line"><span class="comment"># Export the network into edge and node list files Cytoscape can read</span></span><br><span class="line">cyt = exportNetworkToCytoscape(modTOM,</span><br><span class="line">                               edgeFile = paste(<span class="string">"CytoscapeInput-edges-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               nodeFile = paste(<span class="string">"CytoscapeInput-nodes-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               weighted = <span class="literal">TRUE</span>,</span><br><span class="line">                               threshold = <span class="number">0.02</span>,</span><br><span class="line">                               nodeNames = modProbes,</span><br><span class="line">                               nodeAttr = moduleColors[inModule])</span><br><span class="line"><span class="comment">########</span></span><br><span class="line"><span class="comment"># If our target genes are not clustered in one module as expected</span></span><br><span class="line"><span class="comment"># may be reconstruct net in more sensitive parameter</span></span><br><span class="line"><span class="comment">#######</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># get module</span></span><br><span class="line">MEs0 = moduleEigengenes(datExpr, moduleColors)$eigengenes</span><br><span class="line">MEs = orderMEs(MEs0)</span><br><span class="line">modNames = substring(names(MEs), <span class="number">3</span>)</span><br><span class="line">geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = <span class="string">"p"</span>))</span><br><span class="line">MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), </span><br><span class="line">                                          nSamples))</span><br><span class="line">names(geneModuleMembership) = paste(<span class="string">"MM"</span>, modNames, sep=<span class="string">""</span>)</span><br><span class="line">names(MMPvalue) = paste(<span class="string">"p.MM"</span>, modNames, sep=<span class="string">""</span>)</span><br><span class="line"></span><br><span class="line">write.csv(MMPvalue,file = <span class="string">"MMPvalue.csv"</span>,quote=<span class="literal">F</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">HubGenes &lt;- chooseTopHubInEachModule(datExpr,modules)</span><br><span class="line">write.table (HubGenes,file = <span class="string">"HubGenes_of_turquoise_module.txt"</span>,quote=<span class="literal">F</span>,sep=<span class="string">'\t'</span>)</span><br></pre></td></tr></table></figure>

<h3 id="SPL-gene-analysis"><a href="#SPL-gene-analysis" class="headerlink" title="SPL gene analysis"></a>SPL gene analysis</h3><h4 id="get-target-gene-color-and-id"><a href="#get-target-gene-color-and-id" class="headerlink" title="get target gene color and id"></a>get target gene color and id</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">get top gene based on weight</span></span><br><span class="line">less  spl.ori.id | while read line; do grep -w $&#123;line&#125;  CytoscapeInput-nodes-all_without_normalize.txt; done &gt; spl.color</span><br><span class="line">mkdir result</span><br><span class="line">awk '&#123;print "grep -w \""$1"\" 0.05.CytoscapeInput-edges-all.txt &gt; result/"$1"."$3".result"&#125;' spl.color &gt; go</span><br><span class="line">sh go</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> direction among genes <span class="keyword">in</span> sample color are lacking of direction, cuz we constrcut net use <span class="string">"unsigned"</span> method</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> therefore we could sort the edge file by 3rd column</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> either we should filter those replicated genes <span class="keyword">in</span> 1st 2nd column </span></span><br><span class="line">ls *result.2 | sed 's/.result.2//g' | while read line;do python sort.py $&#123;line&#125;.result &gt; $&#123;line&#125;.id.2;done</span><br></pre></td></tr></table></figure>

<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## sort.py </span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line">i = <span class="number">0</span></span><br><span class="line">lst3 = []</span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">sort_list</span><span class="params">(one_list)</span>:</span></span><br><span class="line">  <span class="keyword">return</span> list(set(one_list))</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_data:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_data:</span><br><span class="line">        i += <span class="number">1</span></span><br><span class="line">file_len = i</span><br><span class="line">sort_file = <span class="string">''</span>.join(sorted(open(sys.argv[<span class="number">1</span>]), key=<span class="keyword">lambda</span> s: s.split()[<span class="number">2</span>],reverse=<span class="number">1</span>))</span><br><span class="line">lst1 = sort_file.strip().split(<span class="string">'\n'</span>)</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> range(int(file_len*<span class="number">0.1</span>)):</span><br><span class="line">    lst2 = lst1[i].split(<span class="string">'\t'</span>)</span><br><span class="line">    lst3.append(lst2[<span class="number">0</span>])</span><br><span class="line">    lst3.append(lst2[<span class="number">1</span>])</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> range(len(sort_list(lst3))):</span><br><span class="line">    print(sort_list(lst3)[i<span class="number">-1</span>])</span><br></pre></td></tr></table></figure>

<h4 id="get-gene-based-on-FPKM-cor"><a href="#get-gene-based-on-FPKM-cor" class="headerlink" title="get gene based on FPKM cor"></a>get gene based on FPKM cor</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> only analyze gene <span class="keyword">in</span> turquoise</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> get gene based on FPKM cor</span></span><br><span class="line">less  turquoise.gene.id | while read line  </span><br><span class="line">do</span><br><span class="line">grep -w $&#123;line&#125; ../../9_RSEM/88/88.FPKM.noLGfiltered.matrix</span><br><span class="line">done &gt; turquoise.gene.FPKM.matrix</span><br></pre></td></tr></table></figure>

<h4 id="get-correlation-ship-between-spl-gene-and-turquoise-gene"><a href="#get-correlation-ship-between-spl-gene-and-turquoise-gene" class="headerlink" title="get correlation ship between spl gene and turquoise gene"></a>get correlation ship between spl gene and turquoise gene</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(circlize)</span><br><span class="line"><span class="keyword">library</span>(corrplot)</span><br><span class="line"><span class="keyword">library</span>(pheatmap)</span><br><span class="line">setwd(<span class="string">"~/data/gene_family/downstream/10_GO_KEGG"</span>)</span><br><span class="line"></span><br><span class="line">gene &lt;- read.delim(<span class="string">'hubgene/turquoise.gene.FPKM.matrix'</span>, row.names = <span class="number">1</span>,header = <span class="literal">T</span>, sep = <span class="string">'\t'</span>)</span><br><span class="line">gene &lt;- t(gene) </span><br><span class="line">gene &lt;- log2(gene+<span class="number">1</span>) </span><br><span class="line">gene_cor &lt;- cor(gene, method = <span class="string">'pearson'</span>)</span><br><span class="line">df &lt;- as.data.frame(gene_cor)</span><br><span class="line"></span><br><span class="line">spl_turquoise &lt;- df[,which(names(df)%<span class="keyword">in</span>%c(<span class="string">"g34536"</span>,<span class="string">"g46868"</span>,<span class="string">"g54228"</span>,<span class="string">"g31999"</span>,<span class="string">"g44906"</span>))]</span><br><span class="line">gene_cor &lt;- as.matrix(spl_turquoise)</span><br><span class="line"></span><br><span class="line">col &lt;- colorRampPalette(c(<span class="string">"#BB4444"</span>, <span class="string">"#EE9988"</span>, <span class="string">"#FFFFFF"</span>, <span class="string">"#77AADD"</span>, <span class="string">"#4477AA"</span>))</span><br><span class="line">pheatmap(spl_turquoise,</span><br><span class="line">         labels_row = <span class="literal">NULL</span>,</span><br><span class="line">         angle_col = <span class="string">"45"</span>,</span><br><span class="line">         cluster_row = <span class="literal">T</span>,</span><br><span class="line">         )</span><br><span class="line">  </span><br><span class="line">write.csv(spl_turquoise, file = <span class="string">"5spl_turquoise.gene.cor.matrix"</span>,quote = <span class="literal">F</span>)</span><br></pre></td></tr></table></figure>

<h2 id="eggNOG-mapper-and-ClusterProfile"><a href="#eggNOG-mapper-and-ClusterProfile" class="headerlink" title="eggNOG mapper and ClusterProfile"></a>eggNOG mapper and ClusterProfile</h2><p>we first submit our whole protome to <a href="http://eggnog-mapper.embl.de/" target="_blank" rel="noopener">eggNOG website</a>,  against 33090 database representing plants.</p>
<p>Then we use R script to create a R package named </p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> we should modify some cloumn name and necessary content both <span class="keyword">in</span> these two file</span></span><br><span class="line">query_seqs.fa.emapper.annotations</span><br><span class="line"></span><br><span class="line">nohup Rscript  makeOrgPackageFromEmapper.R 1&gt;make.log 2&gt;err.log</span><br></pre></td></tr></table></figure>

<h4 id="makeOrgPackageFromEmapper-R"><a href="#makeOrgPackageFromEmapper-R" class="headerlink" title="makeOrgPackageFromEmapper.R"></a>makeOrgPackageFromEmapper.R</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(tidyverse)</span><br><span class="line"><span class="keyword">library</span>(stringr)</span><br><span class="line"><span class="keyword">library</span>(KEGGREST)</span><br><span class="line"><span class="keyword">library</span>(AnnotationForge)</span><br><span class="line"></span><br><span class="line"><span class="comment">#' Title</span></span><br><span class="line"><span class="comment">#'</span></span><br><span class="line"><span class="comment">#' @param f_emapper_anno eggnog-mapper annotation result</span></span><br><span class="line"><span class="comment">#' @param author Who is the creator of this package? like "xxx &lt;xxx@xxx.xx&gt;"</span></span><br><span class="line"><span class="comment">#' @param tax_id The Taxonomy ID that represents your organism. (NCBI has a nice online browser for finding the one you need)</span></span><br><span class="line"><span class="comment">#' @param genus Single string indicating the genus</span></span><br><span class="line"><span class="comment">#' @param species Single string indicating the species</span></span><br><span class="line"><span class="comment">#'</span></span><br><span class="line"><span class="comment">#' @return OrgDb name</span></span><br><span class="line"><span class="comment">#' @export</span></span><br><span class="line"><span class="comment">#'</span></span><br><span class="line"><span class="comment">#' @examples</span></span><br><span class="line">makeOrgPackageFromEmapper &lt;- <span class="keyword">function</span>(f_emapper_anno, </span><br><span class="line">                                      author, </span><br><span class="line">                                      tax_id = <span class="string">"0"</span>, </span><br><span class="line">                                      genus = <span class="string">"default"</span>, </span><br><span class="line">                                      species = <span class="string">"default"</span>) &#123;</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># f_emapper_anno = "input/sesame.emapper.annotations"</span></span><br><span class="line">  <span class="comment"># author = "zhangxudong &lt;zhangxudong@genek.tv&gt;"</span></span><br><span class="line">  <span class="comment"># tax_id = "4182"</span></span><br><span class="line">  <span class="comment"># genus = "Sesamum"</span></span><br><span class="line">  <span class="comment"># species = "indicum"</span></span><br><span class="line">  </span><br><span class="line">  <span class="comment"># read emapper result</span></span><br><span class="line">  emapper &lt;- read_delim(f_emapper_anno,</span><br><span class="line">                        <span class="string">"\t"</span>, escape_double = <span class="literal">FALSE</span>, trim_ws = <span class="literal">TRUE</span>)</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract gene name from emapper</span></span><br><span class="line">  gene_info &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(GID = query_name, GENENAME = `eggNOG annot`) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract go annotation from emapper</span></span><br><span class="line">  gos &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(query_name, GO_terms) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  gene2go = data.frame(GID = character(),</span><br><span class="line">                       GO = character(),</span><br><span class="line">                       EVIDENCE = character())</span><br><span class="line">  </span><br><span class="line">  <span class="keyword">for</span> (row <span class="keyword">in</span> <span class="number">1</span>:nrow(gos)) &#123;</span><br><span class="line">    the_gid &lt;- gos[row, <span class="string">"query_name"</span>][[<span class="number">1</span>]]</span><br><span class="line">    the_gos &lt;- str_split(gos[row,<span class="string">"GO_terms"</span>], <span class="string">","</span>, simplify = <span class="literal">FALSE</span>)[[<span class="number">1</span>]]</span><br><span class="line">    </span><br><span class="line">    df_temp &lt;- data_frame(GID = rep(the_gid, length(the_gos)),</span><br><span class="line">                          GO = the_gos,</span><br><span class="line">                          EVIDENCE = rep(<span class="string">"IEA"</span>, length(the_gos)))</span><br><span class="line">    gene2go &lt;- rbind(gene2go, df_temp)</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract kegg pathway annotation from emapper ----------------------------</span></span><br><span class="line">  kos &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(query_name, KEGG_KOs) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  gene2ko = data.frame(GID = character(),</span><br><span class="line">                       Ko = character())</span><br><span class="line">  </span><br><span class="line">  df_temp &lt;- list()</span><br><span class="line">  <span class="keyword">for</span> (row <span class="keyword">in</span> <span class="number">1</span>:nrow(kos)) &#123;</span><br><span class="line">    the_gid &lt;- kos[row, <span class="string">"query_name"</span>][[<span class="number">1</span>]]</span><br><span class="line">    the_kos &lt;- str_split(kos[row,<span class="string">"KEGG_KOs"</span>], <span class="string">","</span>, simplify = <span class="literal">FALSE</span>)[[<span class="number">1</span>]]</span><br><span class="line">    </span><br><span class="line">    df_temp[[row]] &lt;- data_frame(GID = rep(the_gid, length(the_kos)),</span><br><span class="line">                                 Ko = the_kos)</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  gene2ko &lt;- bind_rows(df_temp)</span><br><span class="line">  </span><br><span class="line">  load(file = <span class="string">"kegg_info.RData"</span>)</span><br><span class="line">  gene2pathway &lt;- gene2ko %&gt;% left_join(ko2pathway, by = <span class="string">"Ko"</span>) %&gt;%</span><br><span class="line">    left_join(pathway2name, by = <span class="string">"Pathway"</span>) %&gt;%</span><br><span class="line">    dplyr::select(GID, Ko, Pathway, Pathway_Name, Pathway_Class, Pathway_Subclass) %&gt;%</span><br><span class="line">    distinct() %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  <span class="comment"># extract COG annotation from emapper -------------------------------------</span></span><br><span class="line">  cog_info &lt;- read_delim(<span class="string">"cog_funclass.tab"</span>, </span><br><span class="line">                         <span class="string">"\t"</span>, escape_double = <span class="literal">FALSE</span>, trim_ws = <span class="literal">TRUE</span>)</span><br><span class="line">  </span><br><span class="line">  cogs &lt;- emapper %&gt;%</span><br><span class="line">    dplyr::select(query_name, COG = `COG cat`) %&gt;%</span><br><span class="line">    na.omit()</span><br><span class="line">  </span><br><span class="line">  gene2cog = data.frame(GID = character(),</span><br><span class="line">                        COG = character())</span><br><span class="line">  </span><br><span class="line">  df_temp &lt;- list()</span><br><span class="line">  <span class="keyword">for</span> (row <span class="keyword">in</span> <span class="number">1</span>:nrow(cogs)) &#123;</span><br><span class="line">    the_gid &lt;- cogs[row, <span class="string">"query_name"</span>][[<span class="number">1</span>]]</span><br><span class="line">    the_cogs &lt;- str_trim(str_split(cogs[row,<span class="string">"COG"</span>], <span class="string">","</span>, simplify = <span class="literal">FALSE</span>)[[<span class="number">1</span>]])</span><br><span class="line">    </span><br><span class="line">    df_temp[[row]] &lt;- data_frame(GID = rep(the_gid, length(the_cogs)),</span><br><span class="line">                                 COG = the_cogs)</span><br><span class="line">  &#125;</span><br><span class="line">  gene2cog &lt;- bind_rows(df_temp)</span><br><span class="line">  </span><br><span class="line">  gene2cog &lt;- gene2cog %&gt;% left_join(cog_info, by = <span class="string">"COG"</span>)</span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">  <span class="comment"># make OrgDb</span></span><br><span class="line">  makeOrgPackage(gene_info=gene_info,</span><br><span class="line">                 go=gene2go,</span><br><span class="line">                 <span class="comment">#ko=gene2ko,</span></span><br><span class="line">                 pathway=gene2pathway,</span><br><span class="line">                 cog=gene2cog,</span><br><span class="line">                 version=<span class="string">"0.0.2"</span>,</span><br><span class="line">                 maintainer=author,</span><br><span class="line">                 author=author,</span><br><span class="line">                 outputDir = <span class="string">"."</span>,</span><br><span class="line">                 tax_id=tax_id,</span><br><span class="line">                 genus=genus,</span><br><span class="line">                 species=species,</span><br><span class="line">                 goTable=<span class="string">"go"</span>)</span><br><span class="line">  </span><br><span class="line">  my_orgdb &lt;- str_c(<span class="string">"org."</span>, str_to_upper(str_sub(genus, <span class="number">1</span>, <span class="number">1</span>)) , species, <span class="string">".eg.db"</span>, sep = <span class="string">""</span>)</span><br><span class="line">  <span class="keyword">return</span>(my_orgdb)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">my_orgdb &lt;- makeOrgPackageFromEmapper(<span class="string">"query_seqs.fa.emapper.annotations"</span>, </span><br><span class="line">                                      <span class="string">"shy &lt;2020180495@jsnu.edu.cn&gt;"</span>, </span><br><span class="line">                                      tax_id = <span class="string">"4120"</span>, </span><br><span class="line">                                      genus = <span class="string">"Ipomoea"</span>, </span><br><span class="line">                                      species = <span class="string">"batatas"</span>)</span><br></pre></td></tr></table></figure>

<h3 id="And-i-also-use-kobas-enrichment-methods-but-it-still-queued"><a href="#And-i-also-use-kobas-enrichment-methods-but-it-still-queued" class="headerlink" title="And i also use kobas enrichment methods, but it still queued"></a>And i also use kobas enrichment methods, but it still queued</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"># ipomoea nil</span><br><span class="line">http://kobas.cbi.pku.edu.cn/kobas3/retrieve/?taskid=8ee9e38a368d4ce98ea359f629b0ca38</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># sl</span><br><span class="line"></span><br><span class="line">page:http://kobas.cbi.pku.edu.cn/kobas3/retrieve/?taskid=48bed844140b48a7a4e76e8aca4fa16a</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># enrichment nil </span><br><span class="line"></span><br><span class="line">http://kobas.cbi.pku.edu.cn/kobas3/retrieve/?taskid=bd863ad74ea74fa9adc91b4bfe080520</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">#enrichment sl</span><br></pre></td></tr></table></figure>

<h4 id="There-is-a-very-funny-bug"><a href="#There-is-a-very-funny-bug" class="headerlink" title="There is a very funny bug"></a>There is a very funny bug</h4><p>when i first visit kobas webwites, I find they only have two species included in protein blast tags (human and mouse), after I pee back I found many items loaded…. So I could use Ipomoea nil as reference species…</p>
<h4 id="GO-and-KEGG-annotate-by-ClusterProfile"><a href="#GO-and-KEGG-annotate-by-ClusterProfile" class="headerlink" title="GO and KEGG annotate by ClusterProfile"></a>GO and KEGG annotate by ClusterProfile</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(readr)</span><br><span class="line"><span class="keyword">library</span>(clusterProfiler)</span><br><span class="line"><span class="keyword">library</span>(DOSE)</span><br><span class="line"><span class="keyword">library</span>(purrr)</span><br><span class="line"><span class="keyword">library</span>(tidyverse)</span><br><span class="line"><span class="keyword">library</span>(org.Ibatatas.eg.db)</span><br><span class="line">setwd(<span class="string">"~/data/gene_family/downstream/10_GO_KEGG/"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 安装构建的R包</span></span><br><span class="line">my_orgdb &lt;- <span class="string">"org.Ibatatas.eg.db/"</span></span><br><span class="line"><span class="keyword">if</span> (requireNamespace(my_orgdb, quietly = <span class="literal">TRUE</span>))</span><br><span class="line">  remove.packages(my_orgdb)</span><br><span class="line">install.packages(my_orgdb, repos = <span class="literal">NULL</span>, type = <span class="string">"source"</span>)</span><br><span class="line"><span class="keyword">library</span>(org.Ibatatas.eg.db)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">load(<span class="string">"kegg_info.RData"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">gene_list &lt;- read.table(<span class="string">"hubgene/CytoscapeInput-edges-turquoise.top10.id"</span>, header = <span class="literal">FALSE</span>)</span><br><span class="line">gene_list &lt;- as.character(gene_list$V1)</span><br><span class="line"><span class="comment">#gene_list2 &lt;- read.table("../39/30/gene_filter/final/top10_percent.g34536.brown.f", header = FALSE)</span></span><br><span class="line"><span class="comment">#gene_list2 &lt;- as.character(gene_list2$V1)</span></span><br><span class="line"></span><br><span class="line">ego &lt;- enrichGO(gene          = gene_list, <span class="comment">#差异基因 vector</span></span><br><span class="line">                keyType       = <span class="string">"GID"</span>,  <span class="comment">#差异基因的 ID 类型，需要是 OrgDb 支持的</span></span><br><span class="line">                OrgDb         = org.Ibatatas.eg.db, <span class="comment">#对应的OrgDb</span></span><br><span class="line">                ont           = <span class="string">"BP"</span>, <span class="comment">#GO 分类名称，CC BP MF </span></span><br><span class="line">                pvalueCutoff  = <span class="number">0.05</span>, <span class="comment">#Pvalue 阈值</span></span><br><span class="line">                qvalueCutoff  = <span class="number">0.05</span>, <span class="comment">#qvalue 阈值</span></span><br><span class="line">                pAdjustMethod = <span class="string">"BH"</span>, <span class="comment">#Pvalue 矫正方法 </span></span><br><span class="line">                readable      = <span class="literal">FALSE</span>) <span class="comment">#TRUE 则展示SYMBOL，FALSE 则展示原来的ID</span></span><br><span class="line"><span class="comment">#将 ego 对象转换为dataframe，新版本可以用as.data.frame(ego)</span></span><br><span class="line">ego_results&lt;-as.data.frame(ego)</span><br><span class="line">barplot(ego, showCategory=<span class="number">20</span>, x = <span class="string">"GeneRatio"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">pathway2gene &lt;- AnnotationDbi::select(org.Ibatatas.eg.db, </span><br><span class="line">                                      keys = keys(org.Ibatatas.eg.db), </span><br><span class="line">                                      columns = c(<span class="string">"Pathway"</span>)) %&gt;%</span><br><span class="line">  na.omit() %&gt;%</span><br><span class="line">  dplyr::select(Pathway, GID)</span><br><span class="line"></span><br><span class="line">ekp &lt;- enricher(gene_list, </span><br><span class="line">                TERM2GENE = pathway2gene, </span><br><span class="line">                TERM2NAME = pathway2name, </span><br><span class="line">                pvalueCutoff = <span class="number">0.05</span>, </span><br><span class="line">                qvalueCutoff = <span class="number">0.2</span>,</span><br><span class="line">                pAdjustMethod = <span class="string">"BH"</span>,</span><br><span class="line">                minGSSize = <span class="number">1</span>)</span><br><span class="line">ekp_results &lt;- as.data.frame(ekp)</span><br><span class="line"></span><br><span class="line">barplot(ekp, showCategory=<span class="number">20</span>, x = <span class="string">"GeneRatio"</span>)</span><br><span class="line"></span><br><span class="line">dotplot(ekp)</span><br><span class="line"></span><br><span class="line">emapplot(ekp)</span><br></pre></td></tr></table></figure>

<h3 id="use-CytoScape-construct-network-plot"><a href="#use-CytoScape-construct-network-plot" class="headerlink" title="use CytoScape construct network plot"></a>use CytoScape construct network plot</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p><img src="file:///Users/sunhaoyun/Library/Application%20Support/typora-user-images/image-20200620111509583.png?lastModify=1592810480" alt="image-20200620111509583"></p>
<h3 id="Hub-based-analysis"><a href="#Hub-based-analysis" class="headerlink" title="Hub-based analysis"></a>Hub-based analysis</h3><blockquote>
<p><a href="https://www.karger.com/Article/Fulltext/495205" target="_blank" rel="noopener">Weighted Gene Correlation Network Analysis (WGCNA) Detected Loss of MAGI2 Promotes Chronic Kidney Disease (CKD) by Podocyte Damage</a></p>
</blockquote>
<p>Highly connected hub nodes are central to the network’s architecture [<a href="https://www.karger.com/Article/Fulltext/495205#ref6" target="_blank" rel="noopener">6</a>] and studies suggested that genes more centralized in the network are more likely to be key drivers to respective cellular functions compared with peripheral genes [<a href="https://www.karger.com/Article/Fulltext/495205#ref26" target="_blank" rel="noopener">26</a>]. Therefore, more important nodes can be obtained by identifying hub nodes. In this study, the top 5% of nodes with highest connectivity were defined as hub genes in each module [<a href="https://www.karger.com/Article/Fulltext/495205#ref27" target="_blank" rel="noopener">27</a>]. Fig. <a href="https://www.karger.com/Article/Fulltext/495205#f03" target="_blank" rel="noopener">3</a>A-D are the networks of hub genes in the brown, yellow, green and red modules, only displaying connections with weight (w) above threshold values of 0.3, 0.15, 0.05, and 0.05, respectively. Clearly, hub genes (red nodes) exhibited high connectivity with neighboring genes whose functions are consistent with GO and KEGG findings. Taking the brown module as an example, which showed functional enrichment in metabolic and oxidation-reduction processes, some hub genes have been also reported to participate in similar processes, e.g. the SLC13 Family (SLC13A3) [<a href="https://www.karger.com/Article/Fulltext/495205#ref28" target="_blank" rel="noopener">28</a>, <a href="https://www.karger.com/Article/Fulltext/495205#ref29" target="_blank" rel="noopener">29</a>], SLC22 Family (SLC22A6) [<a href="https://www.karger.com/Article/Fulltext/495205#ref30" target="_blank" rel="noopener">30</a>] or MSRA that plays a protective role in the progression of UUO-induced kidney fibrosis via suppression of fibrotic responses caused by oxidative stress [<a href="https://www.karger.com/Article/Fulltext/495205#ref31" target="_blank" rel="noopener">31</a>]. Similarly, some hub genes in the yellow module have been identified as pro-inflammatory factors in a variety of human diseases, including CTSS and ITGB2 [<a href="https://www.karger.com/Article/Fulltext/495205#ref32" target="_blank" rel="noopener">32</a>], consistent with the BP term or KEGG pathway enriched in the yellow module. Meanwhile, some well-known inflammatory factors also had high connectivity to these hub genes, e.g. IL1B, CXCR4 and CASP1. These observations further supported the notion that proposed roles for hub genes of unknown functions may be inferred from clusters of genes similarly expressed across many biological conditions.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">awk &apos;&#123;if ($1 ~ &quot;^g&quot;)print $1&quot;.t1&quot;&#125;&apos; 0.1.CytoscapeInput-nodes-turquoise.txt  &gt; 0.1.gene.id</span><br><span class="line">awk &apos;&#123;if ($1 ~ &quot;^g&quot;)print $1&quot;.t1&quot;&#125;&apos; 0.05.CytoscapeInput-nodes-turquoise.txt  &gt; 0.05.gene.id</span><br><span class="line">awk &apos;&#123;if ($1 ~ &quot;^g&quot;)print $1&quot;.t1&quot;&#125;&apos; 0.02.CytoscapeInput-nodes-turquoise.txt  &gt; 0.02.gene.id</span><br></pre></td></tr></table></figure>

<p>![image-20200701084031314](/Users/sunhaoyun/Library/Application Support/typora-user-images/image-20200701084031314.png)</p>
<p>![image-20200701085601228](/Users/sunhaoyun/Library/Application Support/typora-user-images/image-20200701085601228.png)</p>
<p>To identify the key modules co-expressed with FT, we calculated themodule-trait association and ranked genes by their correlation with the FPKM value of PmFT.Finally,top 50 candidate genes (R2 &gt; 0.6) co-expressed with PmFTand 11 putative FTinteracting factors inArabidopsisflowering pathway </p>
<p>by “WGCNA” R package (version 1.4.9) [<a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-018-1526-8#ref-CR55" target="_blank" rel="noopener">55</a>], the parameters were set as follows: The power was 9; the minModuleSize was 30 and the cutHeight was 0.25. Finally, we visualized the sub-network using Cytoscape (version 3.4.0) program [<a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-018-1526-8#ref-CR56" target="_blank" rel="noopener">56</a>]. An additional investigation of the putative functions of <em>GhSK</em> genes and their co-expression genes was based on Gene Ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) enrichment analyses.</p>

	
	</div>
  <a type="button" href="/2020/06/16/WGCNA/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2020/05/28/wget_giga/" >下载鸟类的基因组2) giga</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2020-05-28  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>###1. 首先是张郭杰老师他们的48个鸟类基因组，他们全都放在一个project文件夹下了 十分方便</p>
<p><a href="http://gigadb.org/dataset/101000" target="_blank" rel="noopener">project 地址</a></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/100001_101000/101000/gene_and_repeat_annotation/coding_sequences/cds/*.cds.gz</span><br><span class="line"></span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/100001_101000/101000/assembly/*.fa.gz</span><br><span class="line"></span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/100001_101000/101000/gene_and_repeat_annotation/gff/*gff.gz</span><br><span class="line"></span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/100001_101000/101000/gene_and_repeat_annotation/coding_sequences/pep/*pep.gz</span><br></pre></td></tr></table></figure>

<h3 id="2-然后是有五个天堂鸟的基因组"><a href="#2-然后是有五个天堂鸟的基因组" class="headerlink" title="2. 然后是有五个天堂鸟的基因组"></a>2. 然后是有五个天堂鸟的基因组</h3><p>只需要根据最后网址提供的doi循环输出就好</p>
<p><a href="http://gigadb.org/dataset/100521" target="_blank" rel="noopener">project 地址</a></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102157/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102158/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102159/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102160/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102161/*</span><br></pre></td></tr></table></figure>

<h3 id="3-然后是19个企鹅的基因组"><a href="#3-然后是19个企鹅的基因组" class="headerlink" title="3. 然后是19个企鹅的基因组"></a>3. 然后是19个企鹅的基因组</h3><p><a href="http://gigadb.org/dataset/100649" target="_blank" rel="noopener">project地址</a></p>
<p>同样只需要修改doi</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102164/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102165/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102166/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102167/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102168/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102169/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102170/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102171/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102172/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102173/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102174/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102175/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102176/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102177/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102178/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102179/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102180/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102181/*</span><br><span class="line">wget -c ftp://parrot.genomics.cn/gigadb/pub/10.5524/102001_103000/102182/*</span><br></pre></td></tr></table></figure>

<h4 id="所有giga的项目都支持md5sum十分方便"><a href="#所有giga的项目都支持md5sum十分方便" class="headerlink" title="所有giga的项目都支持md5sum十分方便"></a>所有giga的项目都支持md5sum十分方便</h4>
	
	</div>
  <a type="button" href="/2020/05/28/wget_giga/#more" class="btn btn-default more">Read More</a>
</div>

		

		</div>

		<!-- pagination -->
		<div>
  		<center>
		<div class="pagination">

   
    
           <a type="button" class="btn btn-default disabled"><i class="fa fa-arrow-circle-o-left"></i>Prev</a>
        

        <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
 
       <a href="/page/2/" type="button" class="btn btn-default ">Next<i class="fa fa-arrow-circle-o-right"></i></a>     
        

  
</div>

  		</center>
		</div>

		
		
	</div> <!-- col-md-9 -->

	
		<div class="col-md-3">
	<div id="sidebar">
	
			
  <div id="site_search">
   <div class="form-group">
    <input type="text" id="local-search-input" name="q" results="0" placeholder="Search" class="st-search-input st-default-search-input form-control"/>
   </div>  
  <div id="local-search-result"></div>
  </div>


		
			
		
			
	<div class="widget">
		<h4>Tag Cloud</h4>
		<ul class="tag_box inline list-unstyled">		
		
			<li><a href="/tags/Markdown-maker/">Markdown, maker<span>2</span></a></li>
		
			<li><a href="/tags/Python/">Python<span>1</span></a></li>
		
			<li><a href="/tags/Linux/">Linux<span>1</span></a></li>
		
			<li><a href="/tags/OrthoFinder/">OrthoFinder<span>1</span></a></li>
		
			<li><a href="/tags/NCBI/">NCBI<span>1</span></a></li>
		
			<li><a href="/tags/gene-family/">gene family<span>1</span></a></li>
		
			<li><a href="/tags/Mysql-OrthoMCL/">Mysql, OrthoMCL<span>1</span></a></li>
		
			<li><a href="/tags/Phylogeny/">Phylogeny<span>1</span></a></li>
		
			<li><a href="/tags/gene-duplication/">gene duplication<span>1</span></a></li>
		
			<li><a href="/tags/KaKscalculator-R/">KaKscalculator,R<span>1</span></a></li>
		
			<li><a href="/tags/ncbi/">ncbi<span>1</span></a></li>
		
			<li><a href="/tags/HMM/">HMM<span>1</span></a></li>
		
			<li><a href="/tags/Ortholog/">Ortholog<span>1</span></a></li>
		
			<li><a href="/tags/GiagaScience/">GiagaScience<span>1</span></a></li>
		
			<li><a href="/tags/Ensembl/">Ensembl<span>2</span></a></li>
		
			<li><a href="/tags/1000-Genomes/">1000 Genomes<span>1</span></a></li>
		
			<li><a href="/tags/NGS/">NGS<span>1</span></a></li>
		
			<li><a href="/tags/grabber/">grabber<span>1</span></a></li>
		
		 
		</ul>
	</div>


		
			
<div class="widget">
  <h4>Recent Posts</h4>
  <ul class="entry list-unstyled">
    
      <li>
        <a href="/2020/12/29/bwa_picard_pipeline/" ><i class="fa fa-file-o"></i>二代重测序流程</a>
      </li>
    
      <li>
        <a href="/2020/12/22/zhuanluzu/" ><i class="fa fa-file-o"></i></a>
      </li>
    
      <li>
        <a href="/2020/12/22/grab_PGG/" ><i class="fa fa-file-o"></i>抓取PG的G网站上的各省份的输入的rs号的等位基因频率</a>
      </li>
    
      <li>
        <a href="/2020/12/22/可爱的人的喜好/" ><i class="fa fa-file-o"></i></a>
      </li>
    
      <li>
        <a href="/2020/12/08/1K_genome/" ><i class="fa fa-file-o"></i>主要记录千人基因组数据分析</a>
      </li>
    
  </ul>
</div>

		
			
<div class="widget">
	<h4>Links</h4>
	<ul class="blogroll list-unstyled">
	
		<li><i class="fa fa-github"></i><a href="https://github.com/goodlucksun" title="All Github repository." target="_blank"]);">My github</a></li>
	
	</ul>
</div>


		
	</div> <!-- sidebar -->
</div> <!-- col-md-3 -->

	
	
</div> <!-- row-fluid -->
	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2020 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
