<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>下载鸟类的基因组2) ncbi | Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="下载鸟类的基因组2) ncbi">
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header">
			<h1> 下载鸟类的基因组2) ncbi</h1>
		</div>
	



<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <h3 id="1-NCBI-的数据比较麻烦"><a href="#1-NCBI-的数据比较麻烦" class="headerlink" title="1. NCBI 的数据比较麻烦"></a>1. NCBI 的数据比较麻烦</h3><p>NCBI中的assemble分为两个数据库 </p>
<p>1个是GCA数据库（genebank） 1个是GCF数据库（refgSeq）</p>
<p>![image-20200528163802536](/Users/sunhaoyun/Library/Application Support/typora-user-images/image-20200528163802536.png)</p>
<p>两个库在数据上是有重叠的，且GCF优于GCA，所以使用时尽量先使用GCF中有的数据GCF的</p>
<p>不过一般很少使用ncbi的数据</p>
<p>A RefSeq (GCF) genome assembly represents an NCBI-derived copy of a submitted GenBank (GCA) assembly. RefSeq (GCF) assembly records are maintained by NCBI. In some cases the RefSeq (GCF) assembly may not be completely identical to the GenBank (GCA) assembly because NCBI staff may (1) remove short sequences or reported contaminants from the assembly or (2) add non-nuclear genome sequences (for example, mitochondrial or chloroplast genomes) to the assembly. All RefSeq (GCF) genome assemblies include annotation. In the majority of cases, this annotation is generated by the NCBI <a href="https://www.ncbi.nlm.nih.gov/genome/annotation_prok/" target="_blank" rel="noopener">prokaryotic</a> or <a href="https://www.ncbi.nlm.nih.gov/genome/annotation_euk/process/" target="_blank" rel="noopener">eukaryotic</a> genome annotation pipelines. In some cases, annotation is provided by the assembly submitter.</p>
<h3 id="2-需要准备的文件"><a href="#2-需要准备的文件" class="headerlink" title="2.  需要准备的文件"></a>2.  需要准备的文件</h3><p>#### </p>
<p><a href="https://www.ncbi.nlm.nih.gov/genome/?term=Aves" target="_blank" rel="noopener">NCBI ave</a></p>
<p>在搜索结果中能看到320个物种的相关assembly信息。</p>
<p>其中has annotation的117个</p>
<p>那就先把这117个下载下来</p>
<h4 id="分析ftp的构成"><a href="#分析ftp的构成" class="headerlink" title="分析ftp的构成"></a>分析ftp的构成</h4><blockquote>
<p><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/247/815/GCA_000247815.2_FicAlb1.5" target="_blank" rel="noopener">ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/247/815/GCA_000247815.2_FicAlb1.5</a></p>
</blockquote>
<p>GCA_000247815.2</p>
<p>发现地址中解构了GCA id 或者GCF id。同时还添加了 FicAlb1.5 物种的版本名称（在后面发现这个pattern其实也可能没有规律）</p>
<p>写了个爬虫去爬ncbi 但总是time out 不知道学校的网有什么鬼。而且他这个带着物种的版本名称的页面 竟然没有index</p>
<p>所以 把这些结果全选复制粘贴下来。。</p>
<p>保存为list1</p>
<p>格式如下</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line">1.</span><br><span class="line">FicAlb1.5</span><br><span class="line"></span><br><span class="line">Organism: Ficedula albicollis (Collared flycatcher)</span><br><span class="line">Sex: male</span><br><span class="line">Submitter: Uppsala University</span><br><span class="line">Date: 2013/06/27</span><br><span class="line">Assembly level: Chromosome</span><br><span class="line">Genome representation: full</span><br><span class="line">RefSeq category: representative genome</span><br><span class="line">GenBank assembly accession: GCA_000247815.2 (latest)</span><br><span class="line">RefSeq assembly accession: GCF_000247815.1 (latest)</span><br><span class="line">IDs: 37801 [UID] 716688 [GenBank] 726138 [RefSeq]</span><br><span class="line">Item in clipboard</span><br><span class="line"></span><br><span class="line">Select item 1116181</span><br><span class="line">2.</span><br><span class="line">NumMel1.0</span><br><span class="line"></span><br><span class="line">Organism: Numida meleagris (helmeted guineafowl)</span><br><span class="line">Infraspecific name: Breed: g44 Domestic line</span><br><span class="line">Sex: male</span><br><span class="line">Submitter: INRA</span><br><span class="line">Date: 2017/06/01</span><br><span class="line">Assembly level: Chromosome</span><br><span class="line">Genome representation: full</span><br><span class="line">RefSeq category: representative genome</span><br><span class="line">GenBank assembly accession: GCA_002078875.2 (latest)</span><br><span class="line">RefSeq assembly accession: GCF_002078875.1 (latest)</span><br><span class="line">IDs: 1116181 [UID] 4541828 [GenBank] 4550868 [RefSeq]</span><br><span class="line">Item in clipboard</span><br></pre></td></tr></table></figure>

<p>对这个文件进行过滤 至提取出来有用的信息。。比如包含GCF id的就提取 name 和 GCF id 不包含GCF id的就提取name 和GCA id</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"># 提取GCF id</span><br><span class="line">grep &apos;GCF&apos; list | awk &apos;&#123;print $4&#125;&apos; &gt; fore.name</span><br><span class="line"># 提取序号行和序号行的下一行</span><br><span class="line">grep &apos;^[0-9]&apos; list -A 1 &gt; temp1</span><br><span class="line"># 提取字母开头的行</span><br><span class="line">awk &apos;&#123;if ($0 ~ &quot;^[a-Z]&quot;)print&#125;&apos;  temp1 &gt; midlle.name</span><br><span class="line"># 生成 GCF id对应物种partern的文件 final name</span><br><span class="line">paste -d &apos;\t&apos; fore.name final.list</span><br><span class="line"># 格式如下</span><br><span class="line">GCA_000247815.2 FicAlb1.5</span><br><span class="line">GCA_002078875.2 NumMel1.0</span><br></pre></td></tr></table></figure>

<p>然后重新排列组合这些pattern 就可以得到 wget的运行文件了</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> sys,re</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">GCF = <span class="string">"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF"</span></span><br><span class="line">genome = <span class="string">"fna.gz"</span></span><br><span class="line">annotate =  <span class="string">"gff.gz"</span></span><br><span class="line">cds = <span class="string">"cds*"</span></span><br><span class="line">protein = <span class="string">"protein.faa.gz"</span></span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> tfile:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> tfile:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        head = lst1[<span class="number">0</span>][<span class="number">4</span>:<span class="number">7</span>]</span><br><span class="line">        middle = lst1[<span class="number">0</span>][<span class="number">7</span>:<span class="number">10</span>]</span><br><span class="line">        tail = lst1[<span class="number">0</span>][<span class="number">10</span>:<span class="number">13</span>]</span><br><span class="line">        f1 = lst1[<span class="number">0</span>]</span><br><span class="line">        r2 = lst1[<span class="number">1</span>]</span><br><span class="line">        genome_ftp = <span class="string">"wget -c "</span>+GCF+<span class="string">'/'</span>+head+<span class="string">'/'</span>+middle+<span class="string">'/'</span>+tail+<span class="string">'/'</span>+f1+<span class="string">'_'</span>+r2.strip()+<span class="string">'/*'</span>+genome</span><br><span class="line">        annotate_ftp = <span class="string">"wget -c "</span>+GCF+<span class="string">'/'</span>+head+<span class="string">'/'</span>+middle+<span class="string">'/'</span>+tail+<span class="string">'/'</span>+f1+<span class="string">'_'</span>+r2.strip()+<span class="string">'/*'</span>+annotate</span><br><span class="line">        cds_ftp = <span class="string">"wget -c "</span>+GCF+<span class="string">'/'</span>+head+<span class="string">'/'</span>+middle+<span class="string">'/'</span>+tail+<span class="string">'/'</span>+f1+<span class="string">'_'</span>+r2.strip()+<span class="string">'/*'</span>+cds</span><br><span class="line">        protein_ftp = <span class="string">"wget -c "</span>+GCF+<span class="string">'/'</span>+head+<span class="string">'/'</span>+middle+<span class="string">'/'</span>+tail+<span class="string">'/'</span>+f1+<span class="string">'_'</span>+r2.strip()+<span class="string">'/*'</span>+protein</span><br><span class="line">        print(genome_ftp)</span><br><span class="line">        print(annotate_ftp)</span><br><span class="line">        print(cds_ftp)</span><br><span class="line">        print(protein_ftp)</span><br></pre></td></tr></table></figure>

	  
	</div>

	<div>
  	<center>
	<div class="pagination">

    
    
    <a href="/2020/06/16/WGCNA/" type="button" class="btn btn-default"><i
                class="fa fa-arrow-circle-o-left"></i> Prev</a>
    

    <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
    
    <a href="/2020/05/28/wget_giga/" type="button" class="btn btn-default ">Next<i
                class="fa fa-arrow-circle-o-right"></i></a>
    

    
</div>

    </center>
	</div>
	
	<!-- comment -->
	
<section id="comment">
    <h2 class="title">Comments</h2>

    
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2020-05-28 
	</div>
	

	<!-- categories -->
    

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/NCBI/">NCBI<span>1</span></a></li>
    </ul>
	</div>
		

	<!-- toc -->
	<div class="meta-widget">
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2020 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
