<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>Genome assemble statistic | Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="Genome assemble statistic">
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header">
			<h1> Genome assemble statistic</h1>
		</div>
	



<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <p>##基因组组装评价</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">git clone https://github.com/sanger-pathogens/assembly-stats.git</span><br><span class="line">mkdir build</span><br><span class="line">cd build</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 非root安装</span></span></span><br><span class="line">cmake -DINSTALL_DIR:PATH=/home/qu/sunhy/sunhy/softwares/assemblystat ..</span><br><span class="line">make &amp;&amp; make install</span><br><span class="line">cp ../assemblystat/* ../bin</span><br><span class="line">source ~/sunhy/source_shy</span><br><span class="line"></span><br><span class="line">cd ~/sunhy/sunhy/data/prunella/genome/genome/</span><br><span class="line">ls *.fa | while read line;do assembly-stats $&#123;line&#125; &gt;&gt; assembly.stat.total; done</span><br></pre></td></tr></table></figure>

<p>发现如果用这个脚本的话，格式转换起来嗨得自己写脚本，换别的软件试试</p>
<p>用了quast的方法。发现也不是很好，尤其是其得到的统计图，好难看</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python /home/qu/sunhy/sunhy/softwares/quast-5.0.2/quast.py *.fa</span><br></pre></td></tr></table></figure>

<p>最后还是用的bbmap的脚本，statswrapper.sh</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> bbmap的输入也比较有意思，先用tr把换行符换为逗号</span></span><br><span class="line">ls *.fa | tr "\n" "," </span><br><span class="line"></span><br><span class="line">sh /home/qu/sunhy/sunhy/softwares/bbmap/statswrapper.sh in=Prunella_atrogularis_141230.fa,Prunella_collaris_130562_J_Italy.fa,Prunella_collaris_PA20150614-Sichuan.fa,Prunella_collaris_S1318_taiwan.fa,Prunella_collaris_S26913_per.fa,Prunella_fulvescens_QHH0845.fa,Prunella_himalayana_57883.fa,Prunella_himalayana_66660.fa,Prunella_immaculata_XZ14285.fa,Prunella_koslowi_26923.fa,Prunella_modularis_117541.fa,Prunella_montanella_m80200.fa,Prunella_ocularis_EAK479.fa,Prunella_ocularis_MR1290.fa,Prunella_rubeculoides_QHH0839.fa,Prunella_rubida_2S_63614.fa,Prunella_strophita_assemble.fa &gt; bbmap.stats.txt</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 统计cds intron exon数量和平均长度</span></span><br><span class="line">ls *.gff3 | while read line</span><br><span class="line">do</span><br><span class="line">echo $&#123;line&#125; &gt;&gt; cei</span><br><span class="line">cat $&#123;line&#125;| awk '&#123; if ($3 == "gene") print $0 &#125;' | awk '&#123; sum += ($5 - $4) &#125; END &#123; print NR, sum / NR &#125;' &gt;&gt; cei</span><br><span class="line">cat $&#123;line&#125;| awk '&#123; if ($3 == "exon") print $0 &#125;' | awk '&#123; sum += ($5 - $4) &#125; END &#123; print NR, sum / NR &#125;' &gt;&gt; cei</span><br><span class="line">cat $&#123;line&#125;| awk '&#123; if ($3 == "intron") print $0 &#125;' | awk '&#123; sum += ($5 - $4) &#125; END &#123; print NR, sum / NR &#125;'&gt;&gt;cei</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>整理到excel表格里</p>
<p>###提取序列</p>
<p>因为有正链和反链，然后又收到命令可能要把exon和intron提出来，写了个比较蠢的python的方法，然后在本地测试和服务器上结果不太一样，也懒得改了，这个也是能用的，主要不一样在row.split(‘\t’)，感觉是本机不能识别换行符的原因？</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line">input_gff = sys.argv[<span class="number">1</span>]</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="keyword">with</span> open(input_gff, <span class="string">'r'</span>) <span class="keyword">as</span> intf:</span><br><span class="line">    cds_out = open(str(intf.name) + <span class="string">'.fo.cds'</span>, <span class="string">'w'</span>)</span><br><span class="line">    exon_out = open(str(intf.name) + <span class="string">'.fo.exon'</span>, <span class="string">'w'</span>)</span><br><span class="line">    intron_out = open(str(intf.name) + <span class="string">'.fo.intron'</span>, <span class="string">'w'</span>)</span><br><span class="line">    cds_re_out = open(str(intf.name) + <span class="string">'.re.cds'</span>, <span class="string">'w'</span>)</span><br><span class="line">    exon_re_out = open(str(intf.name) + <span class="string">'.re.exon'</span>, <span class="string">'w'</span>)</span><br><span class="line">    intron_re_out = open(str(intf.name) + <span class="string">'.re.intron'</span>, <span class="string">'w'</span>)</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> intf:</span><br><span class="line">        row = row.strip()</span><br><span class="line">        parttern = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        parttern = [x <span class="keyword">for</span> x <span class="keyword">in</span> parttern <span class="keyword">if</span> x != <span class="string">''</span>]</span><br><span class="line">        parttern2 = str(parttern[<span class="number">6</span>])</span><br><span class="line">        parttern2 = parttern2.strip()</span><br><span class="line">        <span class="keyword">if</span> parttern[<span class="number">2</span>] == <span class="string">'gene'</span> <span class="keyword">and</span> parttern2 == <span class="string">'+'</span>:</span><br><span class="line">            cds_out.write(str(parttern[<span class="number">0</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">3</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">4</span>])+<span class="string">'\n'</span>)</span><br><span class="line">        <span class="keyword">elif</span> parttern[<span class="number">2</span>] == <span class="string">'gene'</span> <span class="keyword">and</span> parttern2 == <span class="string">'-'</span>:</span><br><span class="line">            cds_re_out.write(str(parttern[<span class="number">0</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">3</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">4</span>])+<span class="string">'\n'</span>)</span><br><span class="line">        <span class="keyword">elif</span> parttern[<span class="number">2</span>] == <span class="string">'exon'</span> <span class="keyword">and</span> parttern2 == <span class="string">'+'</span>:</span><br><span class="line">            exon_out.write(str(parttern[<span class="number">0</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">3</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">4</span>])+<span class="string">'\n'</span>)</span><br><span class="line">        <span class="keyword">elif</span> parttern[<span class="number">2</span>] == <span class="string">'exon'</span> <span class="keyword">and</span> parttern2 == <span class="string">'-'</span>:   </span><br><span class="line">            exon_re_out.write(str(parttern[<span class="number">0</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">3</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">4</span>])+<span class="string">'\n'</span>)</span><br><span class="line">        <span class="keyword">elif</span> parttern[<span class="number">2</span>] == <span class="string">'intron'</span> <span class="keyword">and</span> parttern2 == <span class="string">'+'</span>:</span><br><span class="line">            intron_out.write(str(parttern[<span class="number">0</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">3</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">4</span>])+<span class="string">'\n'</span>)</span><br><span class="line">        <span class="keyword">elif</span> parttern[<span class="number">2</span>] == <span class="string">'intron'</span> <span class="keyword">and</span> parttern2 == <span class="string">'-'</span>:</span><br><span class="line">            intron_re_out.write(str(parttern[<span class="number">0</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">3</span>])+<span class="string">'\t'</span>+str(parttern[<span class="number">4</span>])+<span class="string">'\n'</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">cds_out.close()</span><br><span class="line">exon_out.close()</span><br><span class="line">intron_out.close()</span><br><span class="line">cds_re_out.close()</span><br><span class="line">exon_re_out.close()</span><br><span class="line">intron_re_out.close()</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 先做一个只有物种前缀的list</span></span><br><span class="line">ls ../genome/*.fa &gt;list</span><br><span class="line">sed -i 's/..\/genome//g' list </span><br><span class="line">sed -i 's/.fa//g' list </span><br><span class="line"><span class="meta">#</span><span class="bash"> 最后用bedtools getfasta 提取序列</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> follow 链的</span></span><br><span class="line">cat list | while read line; do bedtools getfasta -fi ../genome/$&#123;line&#125;.fa -bed cds/$&#123;line&#125;.gff3.fo.cds -fo cds_fa/$&#123;line&#125;.cds.fa; done</span><br><span class="line"><span class="meta">#</span><span class="bash"> reverse 链的</span></span><br><span class="line">cat list | while read line; do bedtools getfasta -fi ../genome/$&#123;line&#125;.fa -bed cds/$&#123;line&#125;.gff3.re.cds -fo cds_re_fa/$&#123;line&#125;.cds.re.fa; done</span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用EMBOSS里的revseq</span></span><br><span class="line">ls *.fa | while read line; do revseq $&#123;line&#125; -outseq $&#123;line&#125;.finish; done</span><br><span class="line"><span class="meta">#</span><span class="bash"> 合并正链和反链</span></span><br><span class="line">cat list | while read line; do cat cds_fa/$&#123;line&#125;.cds.fa cds_re_fa/$&#123;line&#125;.cds.fa.finish &gt; cds_final/$&#123;line&#125;.cds.merge.fa ; done</span><br></pre></td></tr></table></figure>

	  
	</div>

	<div>
  	<center>
	<div class="pagination">

    
    
    <a type="button" class="btn btn-default disabled"><i class="fa fa-arrow-circle-o-left"></i>Prev</a>
    

    <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
    
    <a href="/2019/12/04/shimei/" type="button" class="btn btn-default ">Next<i
                class="fa fa-arrow-circle-o-right"></i></a>
    

    
</div>

    </center>
	</div>
	
	<!-- comment -->
	
<section id="comment">
    <h2 class="title">Comments</h2>

    
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2019-12-06 
	</div>
	

	<!-- categories -->
    

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/geome/">geome<span>1</span></a></li>
    </ul>
	</div>
		

	<!-- toc -->
	<div class="meta-widget">
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2019 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
