<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>Easyfig and pyani | Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="Easyfig and pyani">
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header">
			<h1> Easyfig and pyani</h1>
		</div>
	



<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <h5 id="使用gffread根据gff文件和基因组提取出phb和VCM的基因序列："><a href="#使用gffread根据gff文件和基因组提取出phb和VCM的基因序列：" class="headerlink" title="使用gffread根据gff文件和基因组提取出phb和VCM的基因序列："></a>使用gffread根据gff文件和基因组提取出phb和VCM的基因序列：</h5><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">gffread phb.gff3 -g phb.chr.fa- x phb.cds.fa #在这个之前对phb进行了一点小修改，主要是gff文件格式的问题，phb.gff3中缺少CDS的位置信息，然后因为gene和CDS的位置是一摸一样的，所以用小脚本加上去了</span><br><span class="line">gffread VCM.gff3 -g VCM.chr.fa -x VCM.cds.fa</span><br><span class="line">然后使用transeq获得蛋白序列</span><br><span class="line">transeq -sequence phb.cds.fa -outseq phb.pep.fa</span><br><span class="line">transeq -sequence VCM.cds.fa -outseq VCM.pep.fa</span><br></pre></td></tr></table></figure>

<h5 id="为了获取-gbk结尾的genebank注释文件，拿到CDS文件后，首先使用eggnogmapper网站对phb物种的基因进行注释，他会将上传的基因序列对比到eggnogmapper的已知蛋白数据库，找到最同源的序列。"><a href="#为了获取-gbk结尾的genebank注释文件，拿到CDS文件后，首先使用eggnogmapper网站对phb物种的基因进行注释，他会将上传的基因序列对比到eggnogmapper的已知蛋白数据库，找到最同源的序列。" class="headerlink" title="为了获取.gbk结尾的genebank注释文件，拿到CDS文件后，首先使用eggnogmapper网站对phb物种的基因进行注释，他会将上传的基因序列对比到eggnogmapper的已知蛋白数据库，找到最同源的序列。"></a>为了获取.gbk结尾的genebank注释文件，拿到CDS文件后，首先使用<a href="http://eggnog-mapper.embl.de/" target="_blank" rel="noopener">eggnogmapper</a>网站对phb物种的基因进行注释，他会将上传的基因序列对比到eggnogmapper的已知蛋白数据库，找到最同源的序列。</h5><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">下载网站中注释的结果文件保存为phb.annotation 和VCM.annotation(因为他给的结果文件很多，下载tab分割的那个)</span><br></pre></td></tr></table></figure>

<h5 id="然后使用emapper2gbk软件根据gff、genome、protein和annotation文件生成-gbk结尾的文件。"><a href="#然后使用emapper2gbk软件根据gff、genome、protein和annotation文件生成-gbk结尾的文件。" class="headerlink" title="然后使用emapper2gbk软件根据gff、genome、protein和annotation文件生成.gbk结尾的文件。"></a>然后使用emapper2gbk软件根据gff、genome、protein和annotation文件生成.gbk结尾的文件。</h5><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">emapper2gbk genomes -fn phb.chr.fa -g phb.gff3 -o phb.gbk -fp phb.pep.fa -a phb.annotations -n "Escherichia coli"</span><br><span class="line">emapper2gbk genomes -fn VCM.chr.fa -g VCM.gff3 -o VCM.gbk -fp VCM.pep.fa -a VCM.annotations -n "Escherichia coli"</span><br><span class="line"><span class="meta">#</span><span class="bash"> 这样就得到了VCM.gbk和 phb.gbk</span></span><br></pre></td></tr></table></figure>

<p>做以上这些也主要是因为easyfig需要输入.gbk结尾的基因注释，如果什么其他方法更快捷的话更好一点。为啥使用emapper2gbk是因为他制作的.gbk文件比较标准，如果输入easyfig的文件不标准的话是没办法生成那个箭头，也就是基因的位置的。</p>
<h5 id="然后制作blast文件"><a href="#然后制作blast文件" class="headerlink" title="然后制作blast文件"></a>然后制作blast文件</h5><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">生成本地比对数据库，数据库类性指定为nucl 核苷酸序列</span></span><br><span class="line">makeblastdb -in VCM.chr.fa -dbtype nucl</span><br><span class="line"><span class="meta">#</span><span class="bash">将phb的基因组序列直接比对到VCM的基因组序列，其中evalue 1e-3，输出格式为6 tab格式，只输出最佳的1个匹配</span></span><br><span class="line">blastn -query phb.chr.fa -db VCM.chr.fa -outfmt 6  -num_alignments 1 -evalue 1e-3 &gt; phb_as_query.blast</span><br></pre></td></tr></table></figure>

<h5 id="pyani"><a href="#pyani" class="headerlink" title="pyani"></a>pyani</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c anaconda perl</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-app-cpanminus</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-hash-merge</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-parallel-forkmanager</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-scalar-util-numeric</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-yaml</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-class-data-inheritable</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-exception-class</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-test-pod</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c anaconda biopython</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-file-which # skip if you are not comparing to reference annotation</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-mce</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-threaded</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-list-util</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda perl-math-utils</span><br><span class="line">/Volumes/Windows/研究生期间文件/data/help/yuyicheng/miniconda/condabin/conda install -c bioconda cdbtools</span><br></pre></td></tr></table></figure>

	  
	</div>

	<div>
  	<center>
	<div class="pagination">

    
    
    <a href="/2021/07/29/pip提示warning/" type="button" class="btn btn-default"><i
                class="fa fa-arrow-circle-o-left"></i> Prev</a>
    

    <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
    
    <a href="/2021/07/12/Braker/" type="button" class="btn btn-default ">Next<i
                class="fa fa-arrow-circle-o-right"></i></a>
    

    
</div>

    </center>
	</div>
	
	<!-- comment -->
	
<section id="comment">
    <h2 class="title">Comments</h2>

    
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2021-07-15 
	</div>
	

	<!-- categories -->
    

	<!-- tags -->
		

	<!-- toc -->
	<div class="meta-widget">
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2021 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
