<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>MAKER相关操作 | Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="MAKER相关操作">
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header">
			<h1> MAKER相关操作</h1>
		</div>
	



<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <p>####maker pipeline</p>
<p>1.Initial MAKER Analysis</p>
<p>2.Training Gene Prediction (snap, augustus) </p>
<p>3.Round 2 MAKER Analysis</p>
<p>4.Training Gene Prediction (snap, augustus)</p>
<p>5.Round 3 MAKER Analysis</p>
<p>6.Create Maker standard gene set file</p>
<p>7.PASA </p>
<h5 id="1-Initial-MAKER-Analysis"><a href="#1-Initial-MAKER-Analysis" class="headerlink" title="1.Initial MAKER Analysis"></a>1.Initial MAKER Analysis</h5><p>#创建第一次运行maker需要的config文件 </p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">maker -CTL</span><br></pre></td></tr></table></figure>

<p>#查看修改opts文件，运行maker</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"># set of ESTs or assembled mRNA-seq in fasta format 三代转录数据</span><br><span class="line"></span><br><span class="line">est=/home/Data/avianbase/Prunella_strophita_BWA_INDEX/polished.hq.fasta</span><br><span class="line"></span><br><span class="line"># protein sequence file in fasta format (i.e. from mutiple oransisms) 从ensembl数据库中下载的human mus chicken Tae 数据 cat合并</span><br><span class="line"> </span><br><span class="line">protein=/home/Data/avianbase/Prunella_strophita_BWA_INDEX/ensembl/all/merge_pep.fa </span><br><span class="line"></span><br><span class="line"># 首次运行maker将est2 genome 与protein2genome 设置为1 ，即从头预测</span><br><span class="line"></span><br><span class="line">est2genome=1 </span><br><span class="line">protein2genome=1</span><br><span class="line"></span><br><span class="line"># 设置完成后执行</span><br><span class="line"></span><br><span class="line">nohup mpiexec -n 40 maker &gt; run_maker.log</span><br></pre></td></tr></table></figure>

<hr>
<p>#maker产生的结果</p>
<p>​    -Prunella_strophita_assemble_master_datastore_index.log 记录了每一个contig 比对的情况</p>
<p>使用maker的 gff3_merge     ### fasta_merge ###  对log文件进行注释信息的提取，在得到的 Prunella_strophita_assemble.maker.gff 文件中，包含了est , protein,  repeat masker 以及 maker 的注释结果</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span>#################</span><br><span class="line"><span class="meta">#</span># generate_gff.sh</span><br><span class="line"><span class="meta">#</span>#################</span><br><span class="line"><span class="meta">#</span># sed -i 's/rndN/-/g' generate_gff.sh</span><br><span class="line"><span class="meta">#</span> 每一轮用sed替换rndN信息</span><br><span class="line">gff3_merge -s -d Prunella_strophita_assemble_master_datastore_index.log &gt; Prunella_strophita_assemble.maker.gff</span><br><span class="line">fasta_merge -d Prunella_strophita_assemble_master_datastore_index.log</span><br><span class="line">gff3_merge -n -s -d Prunella_strophita_assemble_master_datastore_index.log &gt; Prunella_strophita_assemble.maker.noseq.gff</span><br><span class="line"><span class="meta">#</span> est alignments</span><br><span class="line">awk '&#123; if ($2 == "est2genome") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt; Prunella_rndN.maker.est2genome.gff</span><br><span class="line"><span class="meta">#</span> protein alignments</span><br><span class="line">awk '&#123; if ($2 == "protein2genome") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt;  Prunella_rndN.maker.protein2genome.gff</span><br><span class="line"><span class="meta">#</span> repeat alignments</span><br><span class="line">awk '&#123; if ($2 ~ "repeat") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt;  Prunella_rndN.maker.repeats.gff</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> current dir </span><br><span class="line"><span class="meta">#</span> /home/maker/20190718_version_maker/round1/Prunella_strophita_assemble.maker.output</span><br><span class="line"></span><br><span class="line">sed -i 's/rndN/rnd1/g' generate_gff.sh</span><br><span class="line">sh generate_gff.sh</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 对产生的gff统计基因数量与平均长度</span><br><span class="line">cat Prunella_strophita_assemble.maker.noseq.gff | awk '&#123; if ($3 == "gene") print $0 &#125;' | awk '&#123; sum += ($5 - $4) &#125; END &#123; print NR, sum / NR &#125;'</span><br><span class="line"></span><br><span class="line">sh gene_count.sh</span><br><span class="line">round1 result: 19711 7584.01</span><br></pre></td></tr></table></figure>

<h5 id="2-Training-Gene-Prediction"><a href="#2-Training-Gene-Prediction" class="headerlink" title="2.Training Gene Prediction"></a>2.Training Gene Prediction</h5><p>#将     snap.sh     busco.sh     gene_count.sh     cp到结果文件夹</p>
<p>通过训练基因模型再次注释提升预测的准确性</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span>##################</span><br><span class="line"><span class="meta">#</span>## snap 基因结构预测</span><br><span class="line"><span class="meta">#</span>##################</span><br><span class="line"><span class="meta">#</span>## snap.sh</span><br><span class="line"><span class="meta">#</span>##################</span><br><span class="line"></span><br><span class="line">mkdir snap</span><br><span class="line">mkdir snap/round1</span><br><span class="line">cd snap/round1</span><br><span class="line">maker2zff -x 0.25 -l 50 -d ../../Prunella_strophita_assemble_master_datastore_index.log</span><br><span class="line">rename genome Prunella_rnd1.zff.length50_aed0.25 genome*</span><br><span class="line">fathom Prunella_rnd1.zff.length50_aed0.25.ann Prunella_rnd1.zff.length50_aed0.25.dna -gene-stats &gt; gene-stats.log 2&gt;&amp;1</span><br><span class="line">fathom  Prunella_rnd1.zff.length50_aed0.25.ann Prunella_rnd1.zff.length50_aed0.25.dna -validate &gt; validate.log 2&gt;&amp;1</span><br><span class="line">fathom  Prunella_rnd1.zff.length50_aed0.25.ann Prunella_rnd1.zff.length50_aed0.25.dna -categorize 1000 &gt; categorize.log 2&gt;&amp;1</span><br><span class="line">fathom uni.ann uni.dna -export 1000 -plus &gt; uni-plus.log 2&gt;&amp;1</span><br><span class="line">mkdir params</span><br><span class="line">cd params</span><br><span class="line">forge ../export.ann ../export.dna &gt; ../forge.log 2&gt;&amp;1</span><br><span class="line">cd ..</span><br><span class="line">hmm-assembler.pl Prunella_rnd1.zff.length50_aed0.25 params &gt; Prunella_rnd1.zff.length50_aed0.25.hmm</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>###################</span><br><span class="line"><span class="meta">#</span>## BUSCO 基因结构预测</span><br><span class="line"><span class="meta">#</span>###################</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>首先为了预测环境的准确性，将第一轮maker运行结果中的gff文件添加首末端1000bp的长度，可能会导致某些注释的mrna超过scaffold的边界，从而会导致bedtools报错，可以统计前后mrna的数量用以计算，而且此项影响较小。</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>## run_BUSCO.sh</span><br><span class="line">awk -v OFS="\t" '&#123; if ($3 == "mRNA") print $1, $4, $5 &#125;' Prunella_strophita_assemble.maker.noseq.gff | \</span><br><span class="line">  awk -v OFS="\t" '&#123; if ($2 &lt; 1000) print $1, "0", $3+1000; else print $1, $2-1000, $3+1000 &#125;' | \</span><br><span class="line">  bedtools getfasta -fi /home/Data/avianbase/Prunella_strophita_INDEX/Prunella_strophita_assemble.fa  -bed - -fo Prunella_rnd1.all.maker.transcripts1000.fasta</span><br><span class="line"></span><br><span class="line">mkdir busco</span><br><span class="line">cd busco</span><br><span class="line">BUSCO.py -i Prunella_rnd1.all.maker.transcripts1000.fasta -o Prunella_rnd1 -l /home/maker/software/maker/exe/busco/ave_odb9/ -m genome -c 8 --long -sp human -z --augustus_prameters '--progress=true'</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> busco 的结果需要提取并修改为 augustus 物种进行下一轮的预测</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span># BUSCO_rndN_number  oringin</span><br><span class="line"><span class="meta">#</span># species_name       target</span><br><span class="line"><span class="meta">#</span>###############</span><br><span class="line"><span class="meta">#</span>## generate_species.sh</span><br><span class="line"><span class="meta">#</span>###############</span><br><span class="line"></span><br><span class="line">rename BUSCO_rndN_number species_name *</span><br><span class="line">sed -i 's/BUSCO_rndN_number/species_name/g' species_name_parameters.cfg</span><br><span class="line">sed -i 's/BUSCO_rndN_number/species_name/g' species_name_parameters.cfg.orig1</span><br><span class="line">mkdir ~/software/maker/exe/augustus/config/species/species_name</span><br><span class="line">cp species_name* ~/software/maker/exe/augustus/config/species/species_name</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> current dir /home/maker/20190718_version_maker/round1/Prunella_strophita_assemble.maker.output</span><br><span class="line">sh snap.sh</span><br><span class="line">sh run_busco.sh</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">cd /home/maker/20190718_version_maker/round/Prunella_strophita_assemble.maker.output/busco/run_0726_rnd1/augustus_output/retraining_parameters</span><br><span class="line">sed -i 's/rndN/rnd1/g' generate_species.sh</span><br><span class="line">sed -i 's/species_name/Prunella_chiecken_rnd1/g' generate_species.sh</span><br><span class="line">sh generate_species.sh</span><br></pre></td></tr></table></figure>

<p>#####3.Round 2 MAKER Analysis</p>
<p>将第一轮生成的gff结果中 est2genome.gff 和 protein2genome.gff 和 repeats.gff 以及 snap预测的基因模型中 hmm 文件输入第二轮的maker_opts.ctl</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line">cd /home/maker/maker_0720/round2</span><br><span class="line"></span><br><span class="line">maker -CTL</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 将maker_opts文件中的 snap 与 Augustus 预测文件添加，将 est2genome.gff, protein2genome.gff,  repeats.gff 输入，将est2genome protein2genome 设置为0</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>est_gff=/home/maker/20190704_version_maker/rnd2_data/Prunella_rnd1.all.maker.est2genome.gff   #aligned ESTs or mRNA-seq from an external GFF3 file</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>genome=/home/Data/avianbase/Prunella_strophita_BWA_INDEX/Prunella_strophita_assemble.fa </span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>protein_gff=/home/maker/20190704_version_maker/rnd2_data/Prunella_rnd1.all.maker.protein2genome.gff  #aligned protein homology evidence from an external GFF3 file</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>rm_gff=/home/maker/20190704_version_maker/rnd2_data/Prunella_rnd1.all.maker.repeats.gff #pre-identified repeat elements from an external GFF3 file</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>snaphmm=/home/maker/20190704_version_maker/rnd2_data/Prunella_rnd1.zff.length50_aed0.25.hmm #SNAP HMM file</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>augustus_species=Prunell_strophita #Augustus gene prediction species model</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>max_dna_len=300000 #length for dividing up contigs into chunks (increases/decreases memory usage)</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>split_hit=20000 #length for the splitting of hits (expected max intron size for evidence alignments)</span><br><span class="line"></span><br><span class="line">nohup mpiexec -n 40 maker &gt; run_maker.log</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span>#################</span><br><span class="line"><span class="meta">#</span># generate_gff.sh</span><br><span class="line"><span class="meta">#</span>#################</span><br><span class="line"><span class="meta">#</span># sed -i 's/rndN/-/g' generate_gff.sh</span><br><span class="line"><span class="meta">#</span> 每一轮用sed替换rndN信息</span><br><span class="line"><span class="meta">#</span> 因为预测结果中使用基因结构预测</span><br><span class="line"></span><br><span class="line">gff3_merge -s -d Prunella_strophita_assemble_master_datastore_index.log &gt; Prunella_strophita_assemble.maker.gff</span><br><span class="line">fasta_merge -d Prunella_strophita_assemble_master_datastore_index.log</span><br><span class="line">gff3_merge -n -s -d Prunella_strophita_assemble_master_datastore_index.log &gt; Prunella_strophita_assemble.maker.noseq.gff</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> est alignments</span><br><span class="line">awk '&#123; if ($2 ~ "est2genome") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt; Prunella_rndN.maker.est2genome.gff</span><br><span class="line"><span class="meta">#</span> protein alignments</span><br><span class="line">awk '&#123; if ($2 ~ "protein2genome") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt;  Prunella_rndN.maker.protein2genome.gff</span><br><span class="line"><span class="meta">#</span> repeat alignments</span><br><span class="line">awk '&#123; if ($2 ~ "repeat") print $0 &#125;' Prunella_strophita_assemble.maker.noseq.gff &gt;  Prunella_rndN.maker.repeats.gff</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">sed -i 's/rndN/rnd2/g' generate_gff.sh</span><br><span class="line">sh generate_gff.sh</span><br><span class="line"></span><br><span class="line">sh gene_count.sh</span><br><span class="line">18227 13821.9</span><br><span class="line"><span class="meta">#</span> 由于第一轮是根据已有信息预测，长度在第二轮得到了修正</span><br></pre></td></tr></table></figure>

<h5 id="4-Training-Gene-Prediction"><a href="#4-Training-Gene-Prediction" class="headerlink" title="4.Training Gene Prediction"></a>4.Training Gene Prediction</h5><p># 与第二步相同, 将 snap.sh     busco.sh     gene_count.sh     cp到结果文件夹</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">sed -i 's/rnd1/rnd2/g' *.sh</span><br><span class="line">sed -i 's/round1/round2/g' snap.sh</span><br><span class="line">sed -i 's/00rnd1/01_rnd2/g' run_BUSCO.sh</span><br><span class="line">sh snap.sh</span><br><span class="line">sh run_BUSCO.sh</span><br><span class="line">cd busco/augustus_output/retraining_paramters</span><br><span class="line">sh generate_species.sh</span><br></pre></td></tr></table></figure>

<p>#####5.Round 3 MAKER analysis</p>
<p>将第二轮maker结果输入maker_opts.ctl 重复第三步</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">vim maker_opts.ctl</span><br><span class="line">nohup mpiexec -n 40 maker &gt; run_maker.log</span><br><span class="line">sh generate_gff.sh</span><br></pre></td></tr></table></figure>

<p>#####6.Create Maker standard gene set file</p>
<p># 对第三轮的maker 结果</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">awk -v OFS="\t" '&#123; if ($3 == "mRNA") print $1, $4, $5 &#125;' /home/maker/20190704_version_maker/02_rnd3/Prunella_strophita_assemble.output/Prunella_strophita_assemble.maker.noseq.gff | \</span><br><span class="line">  bedtools getfasta -fi /home/Data/avianbase/Prunella_strophita_INDEX/Prunella_strophita_assemble.fa -bed - -fo Prunella_rnd3.all.maker.fasta</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">mkdir busco</span><br><span class="line">cd busco</span><br><span class="line"><span class="meta">#</span> 不再再添加前后1000bp的片段，同时运行busco时将model改为 transcriptome </span><br><span class="line">run_BUSCO.py -i Prunella_rnd3.all.maker.transcripts.fasta -o Prunella_rnd3 -l /home/maker/software/busco-master/aves_odb9/  -m transcriptome -c 8 -sp Prunella_chicken_rnd2 -z --augustus_parameters='--progress=true'</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 查看busco的完整性检查结果</span><br><span class="line">less run_Prunella_rnd3/short_summary_Prunella_rnd3.txt</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> gne count</span><br><span class="line">sh gene_count.sh</span><br><span class="line">18148 13917.2</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 提取结果中maker注释的gff部分</span><br><span class="line">perl get_maker_gff.pl Prunella_strophita_assemble.maker.noseq.gff Prunella_strophita_assemble.maker.only.final.noseq.gff</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> 替换gff中的ID</span><br><span class="line">maker_map_ids --prefix Prunella --justify 5 Prunella_strophita_assemble.maker.noseq.gff &gt;</span><br><span class="line">Prunella_rnd3.maker.name.map</span><br><span class="line"></span><br><span class="line">map_gff_ids Prunella_rnd3.maker.name.map Prunella_strophita_assemble.maker.only.final.noseq.gff</span><br></pre></td></tr></table></figure>

<table>
<thead>
<tr>
<th>Gene number</th>
<th>Gene average length</th>
<th>Info</th>
</tr>
</thead>
<tbody><tr>
<td>19711</td>
<td>7584.01</td>
<td>Round 1</td>
</tr>
<tr>
<td>18227</td>
<td>13821.9</td>
<td>Round 2</td>
</tr>
<tr>
<td>18148</td>
<td>13917.2</td>
<td>Round 3</td>
</tr>
</tbody></table>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">grep 'three_prime_UTR' Prunella_strophita_assemble.maker.only.final.noseq.gff | wc -l</span><br><span class="line">2752</span><br><span class="line">grep 'five_prime_UTR' Prunella_strophita_assemble.maker.only.final.noseq.gff | wc -l</span><br><span class="line">2494</span><br></pre></td></tr></table></figure>

<h5 id="7-PASA-prediect-utr-region"><a href="#7-PASA-prediect-utr-region" class="headerlink" title="7.PASA prediect utr region"></a>7.PASA prediect utr region</h5><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> gmap cds2gff</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> clean cds 去除低质量序列</span><br><span class="line"><span class="meta">#</span> /data/maker/PASA_20/data</span><br><span class="line">/home/maker/software/PASApipeline/bin/seqclean XZ15142.hq.fasta</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> custom alignments gff  首先对基因组index然后比对生成 cds 和 gene 模式gff</span><br><span class="line">/home/maker/software/PASApipeline/scripts/process_GMAP_alignments_gff3_chimeras_ok.pl --genome Prunella_strophita_assemble.fa  --transcripts XZ15142.hq.fasta.clean --CPU 2 -N 2  -I 500000  &gt; gmap.spliced_alignments.gff3</span><br><span class="line"></span><br><span class="line">gmap -D . -d Prunella_strophita_assemble.fa.gmap -f 2 -n 2 -x 50 -t 2 -B 5  --max-intronlength-middle=500000 --max-intronlength-ends=500000 XZ15142.hq.fasta.clean</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> Running Alignment Assembly 将基因组与cds信息比对</span><br><span class="line">/home/maker/software/PASApipeline/Launch_PASA_pipeline.pl -c sqlite_conf/alignAssembly.config -C -r -R -g data/Prunella_strophita_assemble.fa -t data/XZ15142.hq.fasta.clean -T -u data/XZ15142.hq.fasta --ALIGNERS blat --CPU 2 -N 2 --IMPORT_CUSTOM_ALIGNMENTS_GFF3 data/gmap.spliced_alignments.gff3</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> Building comprehensive transcriptome database 将结果输入到sqlite数据库</span><br><span class="line">~/software/PASApipeline/scripts/build_comprehensive_transcriptome.dbi -c sqlite_conf/alignAssembly.config -t data/XZ15142.hq.fasta.clean</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> First round, using pre-existing gene structure annotations 对 gff进行update</span><br><span class="line">  ~/software/PASApipeline/Launch_PASA_pipeline.pl -c sqlite_conf/annotCompare.config -g data/Prunella_strophita_assemble.fa -t data/XZ15142.hq.fasta.clean -A -L --annots data/gmap_gene.gff3 --CPU 2</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span> Second round</span><br><span class="line">~/software/PASApipeline/Launch_PASA_pipeline.pl -c sqlite_conf/annotCompare.config -g data/Prunella_strophita_assemble.fa -t data/XZ15142.hq.fasta.clean -A -L --annots  --CPU 2</span><br></pre></td></tr></table></figure>

	  
	</div>

	<div>
  	<center>
	<div class="pagination">

    
    
    <a type="button" class="btn btn-default disabled"><i class="fa fa-arrow-circle-o-left"></i>Prev</a>
    

    <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
    
    <a href="/2019/08/13/hello-world/" type="button" class="btn btn-default ">Next<i
                class="fa fa-arrow-circle-o-right"></i></a>
    

    
</div>

    </center>
	</div>
	
	<!-- comment -->
	
<section id="comment">
    <h2 class="title">Comments</h2>

    
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2019-08-21 
	</div>
	

	<!-- categories -->
    

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/Markdown-maker/">Markdown maker<span>1</span></a></li>
    </ul>
	</div>
		

	<!-- toc -->
	<div class="meta-widget">
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2019 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
