<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>GATK and WDL流处理 | Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="GATK and WDL流处理">
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header">
			<h1> GATK and WDL流处理</h1>
		</div>
	



<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <h4 id="基础准备"><a href="#基础准备" class="headerlink" title="基础准备"></a>基础准备</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 下载安装</span></span></span><br><span class="line">wget https://github.com/broadinstitute/cromwell/releases/download/59/cromwell-59.jar</span><br><span class="line">wget https://github.com/broadinstitute/cromwell/releases/download/59/womtool-59.jar</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## cromwell用于run submit写好的wdl程序</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## womtool用于生成wdl的input的config文件</span></span></span><br></pre></td></tr></table></figure>

<p>主要参考以下github以及broad institute脚本</p>
<blockquote>
<p>git clone <a href="https://github.com/gatk-workflows/seq-format-conversion.git" target="_blank" rel="noopener">https://github.com/gatk-workflows/seq-format-conversion.git</a></p>
<p>git clone <a href="https://github.com/gatk-workflows/gatk4-data-processing.git" target="_blank" rel="noopener">https://github.com/gatk-workflows/gatk4-data-processing.git</a></p>
<p>git clone <a href="https://github.com/broadinstitute/wdl-runner.git" target="_blank" rel="noopener">https://github.com/broadinstitute/wdl-runner.git</a></p>
<p>git clone <a href="https://github.com/gatk-workflows/broad-prod-wgs-germline-snps-indels.git" target="_blank" rel="noopener">https://github.com/gatk-workflows/broad-prod-wgs-germline-snps-indels.git</a></p>
<p>git clone <a href="https://github.com/gatk-workflows/gatk4-germline-snps-indels.git" target="_blank" rel="noopener">https://github.com/gatk-workflows/gatk4-germline-snps-indels.git</a></p>
</blockquote>
<p>####代码部分</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br><span class="line">238</span><br><span class="line">239</span><br><span class="line">240</span><br><span class="line">241</span><br><span class="line">242</span><br><span class="line">243</span><br><span class="line">244</span><br><span class="line">245</span><br><span class="line">246</span><br><span class="line">247</span><br><span class="line">248</span><br><span class="line">249</span><br><span class="line">250</span><br><span class="line">251</span><br><span class="line">252</span><br><span class="line">253</span><br><span class="line">254</span><br><span class="line">255</span><br><span class="line">256</span><br><span class="line">257</span><br><span class="line">258</span><br><span class="line">259</span><br><span class="line">260</span><br><span class="line">261</span><br><span class="line">262</span><br><span class="line">263</span><br><span class="line">264</span><br><span class="line">265</span><br><span class="line">266</span><br><span class="line">267</span><br><span class="line">268</span><br><span class="line">269</span><br><span class="line">270</span><br><span class="line">271</span><br><span class="line">272</span><br><span class="line">273</span><br><span class="line">274</span><br><span class="line">275</span><br><span class="line">276</span><br><span class="line">277</span><br><span class="line">278</span><br><span class="line">279</span><br><span class="line">280</span><br><span class="line">281</span><br><span class="line">282</span><br><span class="line">283</span><br><span class="line">284</span><br><span class="line">285</span><br><span class="line">286</span><br><span class="line">287</span><br><span class="line">288</span><br><span class="line">289</span><br><span class="line">290</span><br><span class="line">291</span><br><span class="line">292</span><br><span class="line">293</span><br><span class="line">294</span><br><span class="line">295</span><br><span class="line">296</span><br><span class="line">297</span><br><span class="line">298</span><br><span class="line">299</span><br><span class="line">300</span><br><span class="line">301</span><br><span class="line">302</span><br><span class="line">303</span><br><span class="line">304</span><br><span class="line">305</span><br><span class="line">306</span><br><span class="line">307</span><br><span class="line">308</span><br><span class="line">309</span><br><span class="line">310</span><br><span class="line">311</span><br><span class="line">312</span><br><span class="line">313</span><br><span class="line">314</span><br><span class="line">315</span><br><span class="line">316</span><br><span class="line">317</span><br><span class="line">318</span><br><span class="line">319</span><br><span class="line">320</span><br><span class="line">321</span><br><span class="line">322</span><br><span class="line">323</span><br><span class="line">324</span><br><span class="line">325</span><br><span class="line">326</span><br><span class="line">327</span><br><span class="line">328</span><br><span class="line">329</span><br><span class="line">330</span><br><span class="line">331</span><br><span class="line">332</span><br><span class="line">333</span><br><span class="line">334</span><br><span class="line">335</span><br><span class="line">336</span><br><span class="line">337</span><br><span class="line">338</span><br><span class="line">339</span><br><span class="line">340</span><br><span class="line">341</span><br><span class="line">342</span><br><span class="line">343</span><br><span class="line">344</span><br><span class="line">345</span><br><span class="line">346</span><br><span class="line">347</span><br><span class="line">348</span><br><span class="line">349</span><br><span class="line">350</span><br><span class="line">351</span><br><span class="line">352</span><br><span class="line">353</span><br><span class="line">354</span><br><span class="line">355</span><br><span class="line">356</span><br><span class="line">357</span><br><span class="line">358</span><br><span class="line">359</span><br><span class="line">360</span><br><span class="line">361</span><br><span class="line">362</span><br><span class="line">363</span><br><span class="line">364</span><br><span class="line">365</span><br><span class="line">366</span><br><span class="line">367</span><br><span class="line">368</span><br><span class="line">369</span><br><span class="line">370</span><br><span class="line">371</span><br><span class="line">372</span><br><span class="line">373</span><br><span class="line">374</span><br><span class="line">375</span><br><span class="line">376</span><br><span class="line">377</span><br><span class="line">378</span><br><span class="line">379</span><br><span class="line">380</span><br><span class="line">381</span><br><span class="line">382</span><br><span class="line">383</span><br><span class="line">384</span><br><span class="line">385</span><br><span class="line">386</span><br><span class="line">387</span><br><span class="line">388</span><br><span class="line">389</span><br><span class="line">390</span><br><span class="line">391</span><br><span class="line">392</span><br><span class="line">393</span><br><span class="line">394</span><br><span class="line">395</span><br><span class="line">396</span><br><span class="line">397</span><br><span class="line">398</span><br><span class="line">399</span><br><span class="line">400</span><br><span class="line">401</span><br><span class="line">402</span><br><span class="line">403</span><br><span class="line">404</span><br><span class="line">405</span><br><span class="line">406</span><br><span class="line">407</span><br><span class="line">408</span><br><span class="line">409</span><br><span class="line">410</span><br><span class="line">411</span><br><span class="line">412</span><br><span class="line">413</span><br><span class="line">414</span><br><span class="line">415</span><br><span class="line">416</span><br><span class="line">417</span><br><span class="line">418</span><br><span class="line">419</span><br><span class="line">420</span><br><span class="line">421</span><br><span class="line">422</span><br><span class="line">423</span><br><span class="line">424</span><br><span class="line">425</span><br><span class="line">426</span><br><span class="line">427</span><br><span class="line">428</span><br><span class="line">429</span><br><span class="line">430</span><br><span class="line">431</span><br><span class="line">432</span><br><span class="line">433</span><br><span class="line">434</span><br><span class="line">435</span><br><span class="line">436</span><br><span class="line">437</span><br><span class="line">438</span><br><span class="line">439</span><br><span class="line">440</span><br><span class="line">441</span><br><span class="line">442</span><br><span class="line">443</span><br><span class="line">444</span><br><span class="line">445</span><br><span class="line">446</span><br><span class="line">447</span><br><span class="line">448</span><br><span class="line">449</span><br><span class="line">450</span><br><span class="line">451</span><br><span class="line">452</span><br><span class="line">453</span><br><span class="line">454</span><br><span class="line">455</span><br><span class="line">456</span><br><span class="line">457</span><br><span class="line">458</span><br><span class="line">459</span><br><span class="line">460</span><br><span class="line">461</span><br><span class="line">462</span><br><span class="line">463</span><br><span class="line">464</span><br><span class="line">465</span><br><span class="line">466</span><br><span class="line">467</span><br><span class="line">468</span><br><span class="line">469</span><br><span class="line">470</span><br><span class="line">471</span><br><span class="line">472</span><br><span class="line">473</span><br><span class="line">474</span><br><span class="line">475</span><br><span class="line">476</span><br><span class="line">477</span><br><span class="line">478</span><br><span class="line">479</span><br><span class="line">480</span><br><span class="line">481</span><br><span class="line">482</span><br><span class="line">483</span><br><span class="line">484</span><br><span class="line">485</span><br><span class="line">486</span><br><span class="line">487</span><br><span class="line">488</span><br><span class="line">489</span><br><span class="line">490</span><br><span class="line">491</span><br><span class="line">492</span><br><span class="line">493</span><br><span class="line">494</span><br><span class="line">495</span><br><span class="line">496</span><br><span class="line">497</span><br><span class="line">498</span><br><span class="line">499</span><br><span class="line">500</span><br><span class="line">501</span><br><span class="line">502</span><br><span class="line">503</span><br><span class="line">504</span><br><span class="line">505</span><br><span class="line">506</span><br><span class="line">507</span><br><span class="line">508</span><br><span class="line">509</span><br><span class="line">510</span><br><span class="line">511</span><br><span class="line">512</span><br><span class="line">513</span><br><span class="line">514</span><br><span class="line">515</span><br><span class="line">516</span><br><span class="line">517</span><br><span class="line">518</span><br><span class="line">519</span><br><span class="line">520</span><br><span class="line">521</span><br><span class="line">522</span><br><span class="line">523</span><br><span class="line">524</span><br><span class="line">525</span><br><span class="line">526</span><br><span class="line">527</span><br><span class="line">528</span><br><span class="line">529</span><br><span class="line">530</span><br><span class="line">531</span><br><span class="line">532</span><br><span class="line">533</span><br><span class="line">534</span><br><span class="line">535</span><br></pre></td><td class="code"><pre><span class="line">##############     version 0.1     ##############</span><br><span class="line">## This WDL converts paired FASTQ to gVCF</span><br><span class="line">##</span><br><span class="line">## Expectations :</span><br><span class="line">##</span><br><span class="line">##     All Sample will be analysed parallelize, caution to the computer performance</span><br><span class="line">##     All Paired Sample should renamed as SAMPLE_NAME_1.fastq.gz and SAMPLE_NAME_2.fastq.gz</span><br><span class="line">##     Volume mroe than triplicate is expected at current dictionary</span><br><span class="line">##</span><br><span class="line">## Dependency :</span><br><span class="line">##</span><br><span class="line">##     samtools</span><br><span class="line">##     trimmomatic</span><br><span class="line">##     bwa mem</span><br><span class="line">##     Picard</span><br><span class="line">##     GATK3.8</span><br><span class="line">##     MILLs, dbSNP, 1kGenome refernece vcf file could find in the follow links:</span><br><span class="line">##     &gt; https://best-practices-for-processing-hts-data.readthedocs.io/en/latest/index.html</span><br><span class="line">##     &gt; https://www.internationalgenome.org/data-portal/data-collection/30x-grch38</span><br><span class="line">##     Hg38 reference genome</span><br><span class="line">##</span><br><span class="line">## Pipeline :</span><br><span class="line">##</span><br><span class="line">##    1: Trimmomatic</span><br><span class="line">##    2: bwa mem</span><br><span class="line">##    3: picard fixmate</span><br><span class="line">##    4: picard sort sam</span><br><span class="line">##    5: picard mark duplicate</span><br><span class="line">##    6: gatk realign</span><br><span class="line">##    7: gatk BQSR</span><br><span class="line">##    8: gatk Haltopviewer call gvcf</span><br><span class="line">##    9: gatk JointGenotyp call merged vcf</span><br><span class="line">##</span><br><span class="line">## Inputs :</span><br><span class="line">##</span><br><span class="line">##     Run $java -jar womtools-version.jar inputs wgs.wdl to fill inputs json file</span><br><span class="line">##</span><br><span class="line">##</span><br><span class="line">## Contact with sunhaoyun1@163.com or sunhaoyun2@gmail.com</span><br><span class="line">##</span><br><span class="line">#################################################</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">############## WORKFLOW DEFINITION ##############</span><br><span class="line">workflow fq2gvcf &#123;</span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">  ##### 输入文本文件要求格式如下 #####：</span><br><span class="line">  ###     Sample1</span><br><span class="line">  ###     Sample2</span><br><span class="line">  ###     Sample3</span><br><span class="line">  ###     Sample4</span><br><span class="line">  ###     Sample5</span><br><span class="line">  ###     Sample6</span><br><span class="line">  ###     Sample7</span><br><span class="line">  ##### 所有输入文件将会并行计算，进行单独的gvcf calling以及最后的merge call #####</span><br><span class="line">  File inputSamplesFile</span><br><span class="line">  Array[Array[File]] inputSamples = read_tsv(inputSamplesFile)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">  ##### 准备所需要的软件 #####</span><br><span class="line">  File trimmomatic</span><br><span class="line">  File bwa</span><br><span class="line">  File gatk</span><br><span class="line">  File picard</span><br><span class="line">  File samtools</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">  ##### 准备需要的参考基因组及SNP信息 ######</span><br><span class="line">  ##### 此处直接使用GATK官网提供的以及谷歌云 #####</span><br><span class="line">  File reffasta</span><br><span class="line">  File refindex</span><br><span class="line">  File refdict</span><br><span class="line">  File indel</span><br><span class="line">  File indel_index</span><br><span class="line">  File mills</span><br><span class="line">  File mills_index</span><br><span class="line">  File db_snp</span><br><span class="line">  File db_snp_index</span><br><span class="line">  File tempdir</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">  ##### 设置Reads Group 以及任务名称 #####</span><br><span class="line">  String threadsnum</span><br><span class="line">  String library_name</span><br><span class="line">  String platform_unit</span><br><span class="line">  String sequencing_center</span><br><span class="line">  String run_date</span><br><span class="line">  File adaptor</span><br><span class="line">  String mission</span><br><span class="line">  </span><br><span class="line"></span><br><span class="line">  ##### 对每一行的样本执行如下命令 #####</span><br><span class="line">  scatter (sample in inputSamples) &#123;</span><br><span class="line">    </span><br><span class="line">      ##### 过滤低质量和接头 #####</span><br><span class="line">      call trim &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      Fq1=sample[0]+"_1.fastq.gz",</span><br><span class="line">      Fq2=sample[0]+"_2.fastq.gz",</span><br><span class="line">      TRIM=trimmomatic,</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      Adaptor=adaptor</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 比对至参考基因组 #####</span><br><span class="line">      call map2genome &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      BWA=bwa,</span><br><span class="line">      clean_Fq2=trim.clean_Fq1,</span><br><span class="line">      clean_Fq1=trim.clean_Fq2,</span><br><span class="line">      Samtools=samtools,</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,            RG="@RG\tID:"+sample[0]+"\tPL:"+platform_unit+"\tLB:"+library_name+"\tSM:"+sample[0]+"\tPU:"+sequencing_center,</span><br><span class="line">      Samtools=samtools</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 矫正双端测序的mate信息 #####</span><br><span class="line">      call fix_mate &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      Picard=picard,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      uBAM=map2genome.uBAM</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 对bam文件进行排序 #####</span><br><span class="line">      call sort_bam &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      Picard=picard,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      fBAM=fix_mate.fix_BAM</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 标记PCR中的重复片段 #####</span><br><span class="line">      call mark_dup &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      Picard=picard,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      sBAM=sort_bam.sort_BAM,</span><br><span class="line">      sBAM_index=sort_bam.sort_BAM_index</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 对INDEL区域进行重排以确定SNP #####</span><br><span class="line">      call realign_bam1 &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      GATK=gatk,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      INDEL=indel,</span><br><span class="line">      INDEL_index=indel_index,</span><br><span class="line">      dedupBAM=mark_dup.dedup_BAM,</span><br><span class="line">      dedupBAM_index=mark_dup.dedup_BAM_index,</span><br><span class="line"></span><br><span class="line">      &#125;</span><br><span class="line">      call realign_bam2 &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      GATK=gatk,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      INDEL=indel,</span><br><span class="line">      INDEL_index=indel_index,</span><br><span class="line">      realign_matrix = realign_bam1.realign_BAM_matrix,</span><br><span class="line">      dedupBAM=mark_dup.dedup_BAM,</span><br><span class="line">      dedupBAM_index=mark_dup.dedup_BAM_index</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### 测序碱基质量值 #####</span><br><span class="line">      call bqsr1 &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      GATK=gatk,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      INDEL=indel,</span><br><span class="line">      INDEL_index=indel_index,</span><br><span class="line">      MILLs=mills,</span><br><span class="line">      MILLs_index=mills_index,</span><br><span class="line">      DB_SNP=db_snp,</span><br><span class="line">      DB_SNP_index=db_snp_index,</span><br><span class="line">      realignBAM=realign_bam2.realign_BAM,</span><br><span class="line">      realignBAM_index=realign_bam2.realign_BAM_index</span><br><span class="line">      &#125;</span><br><span class="line">      call bqsr2 &#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      GATK=gatk,</span><br><span class="line">      TempDir=tempdir,</span><br><span class="line">      BQSR_BAM_matrix = bqsr1.BQSR_BAM_matrix,</span><br><span class="line">      realignBAM=realign_bam2.realign_BAM,</span><br><span class="line">      realignBAM_index=realign_bam2.realign_BAM_index</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line">      ##### gVCF calling #####</span><br><span class="line">      call HaplotypeCaller&#123;</span><br><span class="line">      input:SampleName=sample[0],</span><br><span class="line">      BQSRBAM=bqsr2.BQSR_BAM,</span><br><span class="line">      BQSR_BAM_index=bqsr2.BQSR_BAM_index,</span><br><span class="line">      ThreadsNum=threadsnum,</span><br><span class="line">      RefGenome=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      GATK=gatk,</span><br><span class="line">      TempDir=tempdir</span><br><span class="line">      &#125;</span><br><span class="line">  &#125;</span><br><span class="line"> </span><br><span class="line"></span><br><span class="line">      ##### 将得到的所有样本的gVCF合并进行joint calling #####</span><br><span class="line">      call GenotypeGVCFs&#123;</span><br><span class="line">      input:GATK=gatk,</span><br><span class="line">      RefFasta=reffasta,</span><br><span class="line">      RefIndex=refindex,</span><br><span class="line">      RefDict=refdict,</span><br><span class="line">      Total_name=mission,</span><br><span class="line">      DB_SNP=db_snp,</span><br><span class="line">      DB_SNP_index=db_snp_index,</span><br><span class="line">      GVCFs=HaplotypeCaller.Haplotype_vcf</span><br><span class="line">      &#125;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task trim &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File TRIM</span><br><span class="line">  File Fq1</span><br><span class="line">  File Fq2</span><br><span class="line">  File TempDir</span><br><span class="line">  File Adaptor</span><br><span class="line">  String ThreadsNum</span><br><span class="line"></span><br><span class="line">  command &#123;</span><br><span class="line">    java -Djava.io.tmpdir=$&#123;TempDir&#125; -Xmx64g -jar $&#123;TRIM&#125; \</span><br><span class="line">        PE -phred33 \</span><br><span class="line">        -threads $&#123;ThreadsNum&#125; \</span><br><span class="line">        $&#123;Fq1&#125; $&#123;Fq2&#125; \</span><br><span class="line">        $&#123;SampleName&#125;.1f.pair.fq.gz $&#123;SampleName&#125;.1f.upair.fq.gz \</span><br><span class="line">        $&#123;SampleName&#125;.2r.pair.fq.gz $&#123;SampleName&#125;.2r.upair.fq.gz  \</span><br><span class="line">       ILLUMINACLIP:$&#123;Adaptor&#125;:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File clean_Fq1 = "$&#123;SampleName&#125;.1f.pair.fq.gz"</span><br><span class="line">    File clean_Fq2 = "$&#123;SampleName&#125;.2r.pair.fq.gz"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task map2genome &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File BWA</span><br><span class="line">  File clean_Fq1</span><br><span class="line">  File clean_Fq2</span><br><span class="line">  File Samtools</span><br><span class="line">  String RG</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  command &#123;</span><br><span class="line">    $&#123;BWA&#125; mem -Y -t $&#123;ThreadsNum&#125; -R '$&#123;RG&#125;' -M $&#123;RefGenome&#125; \</span><br><span class="line">        $&#123;clean_Fq1&#125; $&#123;clean_Fq2&#125;  | \</span><br><span class="line">        $&#123;Samtools&#125; view -Shb -o $&#123;SampleName&#125;.ubam -</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File uBAM = "$&#123;SampleName&#125;.ubam"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task fix_mate &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File Picard</span><br><span class="line">  File TempDir</span><br><span class="line">  File uBAM</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;Picard&#125; FixMateInformation MAX_RECORDS_IN_RAM=2000000 \</span><br><span class="line">        VALIDATION_STRINGENCY=SILENT ADD_MATE_CIGAR=True \</span><br><span class="line">        ASSUME_SORTED=true I=$&#123;uBAM&#125; O=$&#123;SampleName&#125;.fbam</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File fix_BAM = "$&#123;SampleName&#125;.fix_mate.bam"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task sort_bam &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File Picard</span><br><span class="line">  File TempDir</span><br><span class="line">  File fBAM</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;Picard&#125; SortSam MAX_RECORDS_IN_RAM=2000000 \</span><br><span class="line">        VALIDATION_STRINGENCY=SILENT SORT_ORDER=coordinate \</span><br><span class="line">        CREATE_INDEX=true  I=$&#123;fBAM&#125; O=$&#123;SampleName&#125;.sort.bam</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File sort_BAM = "$&#123;SampleName&#125;.sort.bam"</span><br><span class="line">    File sort_BAM_index = '$&#123;SampleName&#125;.sort.bai'</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task mark_dup &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File Picard</span><br><span class="line">  File TempDir</span><br><span class="line">  File sBAM</span><br><span class="line">  File sBAM_index</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;Picard&#125; MarkDuplicates MAX_RECORDS_IN_RAM=2000000 \</span><br><span class="line">        VALIDATION_STRINGENCY=SILENT M=$&#123;SampleName&#125;.matrix \</span><br><span class="line">        CREATE_INDEX=true  I=$&#123;sBAM&#125; O=$&#123;SampleName&#125;.dedup.bam</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File dedup_BAM = "$&#123;SampleName&#125;.dedup.bam"</span><br><span class="line">    File dedup_BAM_index = "$&#123;SampleName&#125;.dedup.bai"</span><br><span class="line">    File dedup_BAM_matrix = "$&#123;SampleName&#125;.matrix"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task realign_bam1 &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File GATK</span><br><span class="line">  File TempDir</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  File INDEL</span><br><span class="line">  File INDEL_index</span><br><span class="line">  File dedupBAM</span><br><span class="line">  File dedupBAM_index</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;GATK&#125; -T RealignerTargetCreator \</span><br><span class="line">        -R $&#123;RefGenome&#125; -I $&#123;dedupBAM&#125; -known $&#123;INDEL&#125; \</span><br><span class="line">        -o realignment_targets.list</span><br><span class="line">    &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">        File realign_BAM_matrix = "realignment_targets.list"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task realign_bam2 &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File GATK</span><br><span class="line">  File TempDir</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  File INDEL</span><br><span class="line">  File INDEL_index</span><br><span class="line">  File realign_matrix</span><br><span class="line">  File dedupBAM</span><br><span class="line">  File dedupBAM_index</span><br><span class="line">  command &#123;</span><br><span class="line">     java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;GATK&#125; -T IndelRealigner -R $&#123;RefGenome&#125; \</span><br><span class="line">        -I $&#123;dedupBAM&#125; -targetIntervals realignment_targets.list \</span><br><span class="line">        -known $&#123;INDEL&#125; CREATE_INDEX=true -o $&#123;SampleName&#125;.realign.bam</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File realign_BAM = "$&#123;SampleName&#125;.realign.bam"</span><br><span class="line">    File realign_BAM_index = "$&#123;SampleName&#125;.realign.bai"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task bqsr1 &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File GATK</span><br><span class="line">  File TempDir</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  File INDEL</span><br><span class="line">  File INDEL_index</span><br><span class="line">  File MILLs</span><br><span class="line">  File MILLs_index</span><br><span class="line">  File DB_SNP</span><br><span class="line">  File DB_SNP_index</span><br><span class="line">  File realignBAM</span><br><span class="line">  File realignBAM_index</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;GATK&#125; -T BaseRecalibrator \</span><br><span class="line">        --preserve_qscores_less_than 6 \</span><br><span class="line">        -nct $&#123;ThreadsNum&#125; -R $&#123;RefGenome&#125; -I $&#123;realignBAM&#125; \</span><br><span class="line">        -XL chrX -XL chrY -XL chrM \</span><br><span class="line">        -knownSites $&#123;INDEL&#125; \</span><br><span class="line">        -knownSites $&#123;MILLs&#125; \</span><br><span class="line">        -knownSites $&#123;DB_SNP&#125; \</span><br><span class="line">        -o recal_data.table   &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File BQSR_BAM_matrix = "$&#123;SampleName&#125;.recal_data.table"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task bqsr2 &#123;</span><br><span class="line"></span><br><span class="line">  String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File GATK</span><br><span class="line">  File TempDir</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  File BQSR_BAM_matrix</span><br><span class="line">  File realignBAM</span><br><span class="line">  File realignBAM_index</span><br><span class="line">  command &#123;</span><br><span class="line">    java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;GATK&#125; -T PrintReads -R $&#123;RefGenome&#125; \</span><br><span class="line">        --preserve_qscores_less_than 6 \</span><br><span class="line">        -nct $&#123;ThreadsNum&#125; -I $&#123;realignBAM&#125; \</span><br><span class="line">        -SQQ 10 -SQQ 20 -SQQ 30 -rf BadCigar \</span><br><span class="line">        -BQSR recal_data.table -o $&#123;SampleName&#125;.BQSR.bam\</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File BQSR_BAM = "$&#123;SampleName&#125;.BQSR.bam"</span><br><span class="line">    File BQSR_BAM_index = "$&#123;SampleName&#125;.BQSR.bai"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">task HaplotypeCaller &#123;</span><br><span class="line">String SampleName</span><br><span class="line">  File RefGenome</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  File GATK</span><br><span class="line">  File TempDir</span><br><span class="line">  String ThreadsNum</span><br><span class="line">  File BQSRBAM</span><br><span class="line">  File BQSR_BAM_index</span><br><span class="line">  command &#123;</span><br><span class="line">        java -XX:+UseParallelGC -XX:ParallelGCThreads=$&#123;ThreadsNum&#125; -Djava.io.tmpdir=$&#123;TempDir&#125; \</span><br><span class="line">        -jar $&#123;GATK&#125; \</span><br><span class="line">        -T HaplotypeCaller --genotyping_mode DISCOVERY \</span><br><span class="line">        -A AlleleBalanceBySample -A DepthPerAlleleBySample \</span><br><span class="line">        -A DepthPerSampleHC -A InbreedingCoeff \</span><br><span class="line">        -A MappingQualityZeroBySample -A StrandBiasBySample \</span><br><span class="line">        -A Coverage -A FisherStrand -A HaplotypeScore \</span><br><span class="line">        -A MappingQualityRankSumTest -A MappingQualityZero \</span><br><span class="line">        -A QualByDepth -A RMSMappingQuality \</span><br><span class="line">        -A ReadPosRankSumTest -A VariantType -l INFO \</span><br><span class="line">        --emitRefConfidence GVCF -rf BadCigar \</span><br><span class="line">        --variant_index_parameter 128000 --variant_index_type LINEAR \</span><br><span class="line">        -R $&#123;RefGenome&#125; -I $&#123;BQSRBAM&#125; -o $&#123;SampleName&#125;.HaplotypeCaller.g.vcf</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  output &#123;</span><br><span class="line">    File Haplotype_vcf = "$&#123;SampleName&#125;.HaplotypeCaller.g.vcf"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line">task GenotypeGVCFs &#123;</span><br><span class="line">  File GATK</span><br><span class="line">  File RefFasta</span><br><span class="line">  File RefIndex</span><br><span class="line">  File RefDict</span><br><span class="line">  Array[File] GVCFs</span><br><span class="line">  String Total_name</span><br><span class="line">  File DB_SNP</span><br><span class="line">  File DB_SNP_index</span><br><span class="line"></span><br><span class="line">  command &#123;</span><br><span class="line">    java -jar $&#123;GATK&#125; \</span><br><span class="line">        -T GenotypeGVCFs \</span><br><span class="line">        -R $&#123;RefFasta&#125; \</span><br><span class="line">        -G StandardAnnotation \</span><br><span class="line">        -D $&#123;DB_SNP&#125; \</span><br><span class="line">        -V $&#123;sep=" -V " GVCFs&#125; \</span><br><span class="line">        -o $&#123;Total_name&#125;_rawVariants.vcf</span><br><span class="line">  &#125;</span><br><span class="line">  output &#123;</span><br><span class="line">    File rawVCF = "$&#123;Total_name&#125;_rawVariants.vcf"</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

<h4 id="输入文件部分"><a href="#输入文件部分" class="headerlink" title="输入文件部分"></a>输入文件部分</h4><figure class="highlight json"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">&#123;</span><br><span class="line">  <span class="attr">"fq2gvcf.adaptor"</span>: <span class="string">"/home/sunhaoyun/software/adapters/adaptor.fa"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.inputSamplesFile"</span>: <span class="string">"/home/sunhaoyun/working/SGDP/samples_list.txt"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.picard"</span>: <span class="string">"/home/sunhaoyun/software/picard.jar"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.indel"</span>: <span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.known_indels.vcf.gz"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.indel_index"</span>: <span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.known_indels.vcf.gz.tbi"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.tempdir"</span>: <span class="string">"/data/sunhaoyun/temp"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.mission"</span>: <span class="string">"TEST"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.refindex"</span>: <span class="string">"/home/sunhaoyun/working/reference_genome/Homo_sapiens_assembly38.fai"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.refdict"</span>: <span class="string">"/home/sunhaoyun/working/reference_genome/Homo_sapiens_assembly38.dict"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.library_name"</span>: <span class="string">"TEST"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.platform_unit"</span>: <span class="string">"ILLMINA"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.sequencing_center"</span>: <span class="string">"JSNU"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.run_date"</span>: <span class="string">"20200309"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.reffasta"</span>: <span class="string">"/home/sunhaoyun/working/reference_genome/Homo_sapiens_assembly38.fasta"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.threadsnum"</span>: <span class="string">"24"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.gatk"</span>: <span class="string">"/home/sunhaoyun/software/GenomeAnalysisTK.jar"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.mills"</span>: <span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Mills_and_1000G_gold_standard.indels.hg38.vcf.gz"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.mills_index"</span>:<span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Mills_and_1000G_gold_standard.indels.hg38.vcf.gz.tbi"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.db_snp"</span>: <span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.dbsnp138.vcf.gz"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.db_snp_index"</span>: <span class="string">"/data/sunhaoyun/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.dbsnp138.vcf.gz.tbi"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.samtools"</span>: <span class="string">"/home/sunhaoyun/software/samtools/bin/samtools"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.bwa"</span>: <span class="string">"/usr/bin/bwa"</span>,</span><br><span class="line">  <span class="attr">"fq2gvcf.trimmomatic"</span>: <span class="string">"/home/sunhaoyun/software/trimmomatic-0.39.jar"</span></span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">java -jar cromwell-59.jar run wgs.wdl -i wgs.inputs.json</span><br></pre></td></tr></table></figure>

	  
	</div>

	<div>
  	<center>
	<div class="pagination">

    
    
    <a type="button" class="btn btn-default disabled"><i class="fa fa-arrow-circle-o-left"></i>Prev</a>
    

    <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
    
    <a href="/2021/04/06/Inparanoid/" type="button" class="btn btn-default ">Next<i
                class="fa fa-arrow-circle-o-right"></i></a>
    

    
</div>

    </center>
	</div>
	
	<!-- comment -->
	
<section id="comment">
    <h2 class="title">Comments</h2>

    
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2021-04-26 
	</div>
	

	<!-- categories -->
    

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/variants-calling/">variants calling<span>1</span></a></li>
    </ul>
	</div>
		

	<!-- toc -->
	<div class="meta-widget">
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2021 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
