<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>Page 2 | Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 <div class="page-header logo">
  <h1>Keeping your word<span class="blink-fast">∎</span></h1>
</div>

<div class="row page">

	
	<div class="col-md-9">
	

		<div class="slogan">


		<i class="fa fa-heart blink-slow"></i>

		我们进化速度快到让你感动邪门？！

</div>    

		<div id="top_search"></div>
		<div class="mypage">
		
		<!-- title and entry -->
		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/21/mysql/" >OrthoMCL进行同源基因比对</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-21  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="以root账户登陆mysql"><a href="#以root账户登陆mysql" class="headerlink" title="以root账户登陆mysql"></a>以root账户登陆mysql</h1><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">mysql -u root -p</span><br></pre></td></tr></table></figure>

<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"># 创建本地登录的账户</span><br><span class="line">create user &apos;mysql&apos;@&apos;localhost&apos; identified by &apos;111111&apos;;</span><br><span class="line"></span><br><span class="line"># 创建远程可登录的账户</span><br><span class="line">create user &apos;mysql&apos;@&apos;%&apos; identified by &apos;111111&apos;;</span><br><span class="line"></span><br><span class="line"># 用户创建完成后，刷新授权</span><br><span class="line">flush privileges;</span><br><span class="line"></span><br><span class="line"># 创建一个新的数据库</span><br><span class="line">create database orthomcl DEFAULT CHARSET utf8 COLLATE utf8_general_ci;</span><br><span class="line"></span><br><span class="line"># 授权</span><br><span class="line">GRANT ALL PRIVILEGES ON orthomcl.* TO &apos;orthomcl&apos;@&apos;localhost&apos;   IDENTIFIED BY &apos;111111&apos;;</span><br><span class="line">GRANT ALL PRIVILEGES ON orthomcl.* TO &apos;orthomcl&apos;@&apos;%&apos;   IDENTIFIED BY &apos;111111&apos;;</span><br><span class="line">flush privileges;</span><br><span class="line"></span><br><span class="line"># 查看结果</span><br><span class="line">show databases;</span><br></pre></td></tr></table></figure>

<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">perl scripts/orthomcl-setup-database.pl --user orthomcl --password 111111 --host 127.0.0.1 --database orthomcl --outfile configure_outfile.conf [--no-create-database]</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">perl scripts/orthomcl-pipeline.pl -i ../../data/orthomcl/ -o ../../data/orthomcl_out -m configure_outfile.conf --nocompliant</span><br></pre></td></tr></table></figure>

<p>Warning: directory “../../data/orthomcl_out” already exists, are you sure you want to store data here [Y]? Y</p>
<p>Starting OrthoMCL pipeline on: Sun Dec 15 11:12:16 2019</p>
<p>Git commit: 08bd2baf43db6e35afb3e22e95dce8a3f740729c</p>
<p>=Stage 1: Validate Files =</p>
<p>Validating Pcollaris_S26913_per_pep.fasta … 6276 sequences</p>
<p>Validating Pocularis_EAK479_pep.fasta … 6224 sequences</p>
<p>Validating Prubeculoides_QHH0839_pep.fasta … 6704 sequences</p>
<p>Validating Pimmaculata_XZ14285_pep.fasta … 5791 sequences</p>
<p>Validating Pkoslowi_26923_pep.fasta … 11783 sequences</p>
<p>Validating Pcollaris_PA20150614-Sichuan_pep.fasta … 8328 sequences</p>
<p>Validating Prubida_2S_63614_pep.fasta … 12875 sequences</p>
<p>Validating Pstrophita_assemble_pep.fasta … 20581 sequences</p>
<p>Validating Pcollaris_130562_Jon_Italy_pep.fasta … 6855 sequences</p>
<p>Validating Patrogularis_141230_pep.fasta … 10880 sequences</p>
<p>Validating Pmodularis_117541_pep.fasta … 8978 sequences</p>
<p>Validating Pmontanella_m80200_pep.fasta … 8534 sequences</p>
<p>Validating Pcollaris_S1318_taiwan_pep.fasta … 14461 sequences</p>
<p>Validating Pocularis_MR1290_pep.fasta … 6774 sequences</p>
<p>Validating Phimalayana_66660_pep.fasta … 6852 sequences</p>
<p>Validating Phimalayana_57883_pep.fasta … 7552 sequences</p>
<p>Validating Pfulvescens_QHH0845_pep.fasta … 9746 sequences</p>
<p>Validating Gallus_gallus_pep.fasta … 28444 sequences</p>
<p>Validated 18 files</p>
<p>Stage 1 took 0.17 minutes </p>
<p>=Stage 2: Validate Database=</p>
<p>Warning: some tables exist already in database dbi:mysql:orthomcl_new:127.0.0.1:mysql_local_infile, user=root, database_name=orthomcl_new. Do you want to remove (y/n)? y</p>
<p>Executing: ‘drop database orthomcl_new’</p>
<p>Executing: ‘create database orthomcl_new’</p>
<p>Successfully removed old database entries</p>
<p>Stage 2 took 0.02 minutes </p>
<p>=Stage 3: Load OrthoMCL Database Schema=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclInstallSchema “/home/sunhaoyun/software/orthomcl-pipeline/configure_outfile.conf” “/home/sunhaoyun/data/orthomcl_out/log/orthomclSchema.log” 1&gt;/home/sunhaoyun/data/orthomcl_out/log/3.loadschema.stdout.log 2&gt;/home/sunhaoyun/data/orthomcl_out/log/3.loadschema.stderr.log</p>
<p>Stage 3 took 0.00 minutes </p>
<p>=Stage 4: Adjust Fasta=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_S26913_per_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_S26913_per_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pocularis_EAK479_pep “/home/sunhaoyun/data/orthomcl/Pocularis_EAK479_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Prubeculoides_QHH0839_pep “/home/sunhaoyun/data/orthomcl/Prubeculoides_QHH0839_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pimmaculata_XZ14285_pep “/home/sunhaoyun/data/orthomcl/Pimmaculata_XZ14285_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pkoslowi_26923_pep “/home/sunhaoyun/data/orthomcl/Pkoslowi_26923_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_PA20150614-Sichuan_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_PA20150614-Sichuan_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Prubida_2S_63614_pep “/home/sunhaoyun/data/orthomcl/Prubida_2S_63614_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pstrophita_assemble_pep “/home/sunhaoyun/data/orthomcl/Pstrophita_assemble_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_130562_Jon_Italy_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_130562_Jon_Italy_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Patrogularis_141230_pep “/home/sunhaoyun/data/orthomcl/Patrogularis_141230_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pmodularis_117541_pep “/home/sunhaoyun/data/orthomcl/Pmodularis_117541_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pmontanella_m80200_pep “/home/sunhaoyun/data/orthomcl/Pmontanella_m80200_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pcollaris_S1318_taiwan_pep “/home/sunhaoyun/data/orthomcl/Pcollaris_S1318_taiwan_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pocularis_MR1290_pep “/home/sunhaoyun/data/orthomcl/Pocularis_MR1290_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Phimalayana_66660_pep “/home/sunhaoyun/data/orthomcl/Phimalayana_66660_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Phimalayana_57883_pep “/home/sunhaoyun/data/orthomcl/Phimalayana_57883_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Pfulvescens_QHH0845_pep “/home/sunhaoyun/data/orthomcl/Pfulvescens_QHH0845_pep.fasta” 1</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclAdjustFasta Gallus_gallus_pep “/home/sunhaoyun/data/orthomcl/Gallus_gallus_pep.fasta” 1</p>
<p>Stage 4 took 0.08 minutes </p>
<p>=Stage 5: Filter Fasta=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/orthomclSoftware-v2.0.9/bin/orthomclFilterFasta “/home/sunhaoyun/data/orthomcl_out/compliant_fasta” 10 20</p>
<p>Stage 5 took 0.12 minutes </p>
<p>=Stage 6: Split Fasta=</p>
<p>splitting /home/sunhaoyun/data/orthomcl_out/blast_dir/goodProteins.fasta into 4 pieces</p>
<p>jStage 6 took 0.42 minutes </p>
<p>=Stage 7: Format Database=</p>
<p>/home/sunhaoyun/software/orthomcl-pipeline/blast-2.2.26/bin/formatdb -i “/home/sunhaoyun/data/orthomcl_out/blast_dir/goodProteins.fasta” -p “T” -l “/home/sunhaoyun/data/orthomcl_out/log/formatdb.log” 1&gt;/home/sunhaoyun/data/orthomcl_out/log/7.format-stdout.log 2&gt;/home/sunhaoyun/data/orthomcl_out/log/7.format-stderr.log</p>

	
	</div>
  <a type="button" href="/2019/12/21/mysql/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/06/基因组组装评价/" >Genome assemble statistic</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-06  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>##基因组组装评价</p>
<p>真的有很多的脚本可以去算这个数，但是，每个软件算出来的数值都不太一样，有的差距很小，有的差距很大，还有把N50 L50搞反的，真的，五花八门，也没有查到文献中推荐哪个好，好像大家都是自己算自己的。</p>
<blockquote>
<blockquote>
<p>N50</p>
<blockquote>
<p>the <strong><em>N50</em></strong> is defined as the sequence length of the shortest contig at 50% of the total genome length.</p>
</blockquote>
<p>L50</p>
<blockquote>
<p>Given a set of contigs, each with its own length, the <strong><em>L50</em></strong> count is defined as the smallest number of contigs whose length sum makes up half of genome size. From the example above the L50=3.</p>
</blockquote>
<p>N90</p>
<blockquote>
<p>The <strong><em>N90 statistic</em></strong> is less than or equal to the <em>N50</em> statistic; it is the length for which the collection of all contigs of that length or longer contains at least 90% of the sum of the lengths of all contigs, </p>
</blockquote>
<p>NG50</p>
<blockquote>
<p>Note that <em>N50</em> is calculated in the context of the assembly size rather than the genome size</p>
</blockquote>
<p>D50</p>
<blockquote>
<p>The <em>D50</em> statistic is the lowest value <em>d</em> for which the sum of the lengths of the largest <em>d</em> lengths is at least 50% of the sum of all of the length</p>
</blockquote>
</blockquote>
<p>统计方法还有很多，参考<a href="https://en.wikipedia.org/wiki/N50%2C_L50%2C_and_related_statistics" target="_blank" rel="noopener">wiki</a></p>
</blockquote>
	
	</div>
  <a type="button" href="/2019/12/06/基因组组装评价/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/12/04/shimei/" >建一颗ML树</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-12-04  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h4 id="流程大概就是"><a href="#流程大概就是" class="headerlink" title="流程大概就是"></a>流程大概就是</h4><p>刘亮老师讲的那个差不多</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 安装mafft</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 下载mafft，因为mafft的安装包是.rpm结尾的压缩包，rpm包如果不用管理员安装的话会提示权限错误，但是咱这脑</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 子就别用管理员了，万一删了啥不该删的。rpm2cpio这个工具可以把rpm包在当前目录解压缩，使用方法为</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> rpm2cpio XXX.rpm | cpio -div 解压出来的结果会默认先放在usr里面</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> wget https://mafft.cbrc.jp/alignment/software/mafft-7.450-gcc_fc6.x86_64.rpm</span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 由于mac没有rpm2cpio, linux跳过brew install  rpm2cpio这句</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> brew install  rpm2cpio</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> rpm2cpio mafft-7.450-gcc_fc6.x86_64.rpm | cpio -div</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 解压完可以看到出现了一个usr目录，usr里的bin放的是执行文件，mafft要求所有所有执行文件都在环境变量里，</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 所以下一步先添加环境变量，就是快捷方式，首先要查看当前目录在哪儿,<span class="built_in">pwd</span> 获取当前目录,假如说现在目录 </span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 在/home/zhengyuno/softwares/mafft</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">cd</span> usr</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">cd</span> bin</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">pwd</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash">/home/zhengyuno/softwares/mafft/usr/bin</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> vim ~/.bash_profile</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 滚到最下面的空白处把<span class="built_in">pwd</span>的结果加上一些符号粘贴进去，vim编辑器输入时候，摁i是进入insert模式，输入完了摁</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> esc退出insert模式，然后摁 :wq 保存并退出 （左下角会显示状态）。输入的时候一定要记得前后的PATH要不然</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 就啥快捷方式都没了，别问我为啥知道，试过好多次了T.T</span></span><br><span class="line"></span><br><span class="line">export PATH=/home/zhengyuno/softwares/mafft/usr/bin:$PATH</span><br><span class="line"><span class="meta">#</span><span class="bash"> 输入mafft出来这个交互式页面就ok了</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> mafft</span></span><br><span class="line"></span><br><span class="line">---------------------------------------------------------------------</span><br><span class="line"></span><br><span class="line">   MAFFT v7.450 (2019/Aug/23)</span><br><span class="line"></span><br><span class="line">        MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)</span><br><span class="line">        https://mafft.cbrc.jp/alignment/software/</span><br><span class="line">---------------------------------------------------------------------</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">Input file? (FASTA format; Folder=/Users/sunhaoyun/temp_dir/usr/bin)</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> ctrl + c 中断当前程序，然后我们就不用交互式的模式了，直接用一行的<span class="built_in">command</span> line模式就行</span></span><br></pre></td></tr></table></figure>

<p>这里推荐看<a href="http://www.chenlianfu.com/?p=2214" target="_blank" rel="noopener">陈连福老师的博客</a> </p>
<p>以及<a href="https://mafft.cbrc.jp/alignment/software/manual/manual.html" target="_blank" rel="noopener">官方文档</a>，当然我这种比较懒的官方的manual都没看完过</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## mafft的使用 首先选一个算法，看一下算法有很多，有的适合差不多长的，有的适合比较近缘的，不会选，</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 用auto，看一下output选项，因为下一步过滤要fasta格式，他默认也是fasta,不用改</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> mafft --auto cds_ENSG00000000419.fasta &gt; cds419_mafft_out.fa</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">运行完了屏幕上的信息可以保存一下</span></span><br></pre></td></tr></table></figure>
	
	</div>
  <a type="button" href="/2019/12/04/shimei/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/10/01/re/" >李笑来老板的re part</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-10-01  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h1 id="正则表达式"><a href="#正则表达式" class="headerlink" title="正则表达式"></a>正则表达式</h1><p>正则表达式本质上是个独立的语言，短小却格外强悍 —— 乃至于，如果你竟然没学会它的话，你的之前学的编程技能干脆与残疾无异。</p>
<p>Wikipedia 上对正则表达式的说明如下：</p>
<blockquote>
<p><strong>正则表达式</strong>（英语：Regular Expression，在代码中常简写为 regex、regexp 或 RE），又称<em>正规表示式</em>、<em>正规表示法</em>、<em>正规运算式</em>、<em>规则运算式</em>、<em>常规表示法</em>，是计算机科学的一个概念。正则表达式使用单个字符串来描述、匹配一系列符合某个句法规则的字符串。在很多文本编辑器里，正则表达式通常被用来检索、替换那些符合某个模式的文本。许多程序设计语言都支持利用正则表达式进行字符串操作。例如，在 Perl 中就内建了一个功能强大的正则表达式引擎。正则表达式这个概念最初是由 Unix 中的工具软件（例如 sed 和 grep）普及开的。</p>
</blockquote>
	
	</div>
  <a type="button" href="/2019/10/01/re/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/08/29/HMM_example/" >HMM 隐式马尔可夫模型</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-08-29  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<h2 id="Hidden-Markov-Model-隐含马尔可夫模型和-Viterbi-algorithm-维特比算法示例"><a href="#Hidden-Markov-Model-隐含马尔可夫模型和-Viterbi-algorithm-维特比算法示例" class="headerlink" title="Hidden Markov Model 隐含马尔可夫模型和 Viterbi algorithm 维特比算法示例"></a>Hidden Markov Model 隐含马尔可夫模型和 Viterbi algorithm 维特比算法示例</h2><p>基本概念：首先它得是一个统计模型，通过已有的样本来预测马尔可夫过程中的隐含参数。</p>
<p>可见状态链：观测得到的样本组成的数据集</p>
<p>隐含状态链：样本背后的隐含的数据集</p>
<p>转换概率 transition probability：可随意设定转换概率，根据不同的转换概率会得到不同的HMM模型</p>
<p>输出概率 emission probability：从隐含状态转换为可见状态时的概率，也可自由设定</p>
	
	</div>
  <a type="button" href="/2019/08/29/HMM_example/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/08/21/MAKER相关操作/" >MAKER相关操作</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-08-21  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>####maker pipeline</p>
<p>1.Initial MAKER Analysis</p>
<p>2.Training Gene Prediction (snap, augustus) </p>
<p>3.Round 2 MAKER Analysis</p>
<p>4.Training Gene Prediction (snap, augustus)</p>
<p>5.Round 3 MAKER Analysis</p>
<p>6.Create Maker standard gene set file</p>
<p>7.PASA </p>
	
	</div>
  <a type="button" href="/2019/08/21/MAKER相关操作/#more" class="btn btn-default more">Read More</a>
</div>

		
			
	
	<!-- display as entry -->
<div class="row">
	<div class="col-md-8">
		<h3 class="title">
			<a href="/2019/08/13/hello-world/" >Hello World</a>
		</h3>
		</div>
	<div class="col-md-4">
		<div class="date">post @ 2019-08-13  </div>
		</div>
	</div>
	


			<div class="entry">
  <div class="row">
	
	
		<p>Welcome to <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a>! This is your very first post. Check <a href="https://hexo.io/docs/" target="_blank" rel="noopener">documentation</a> for more info. If you get any problems when using Hexo, you can find the answer in <a href="https://hexo.io/docs/troubleshooting.html" target="_blank" rel="noopener">troubleshooting</a> or you can ask me on <a href="https://github.com/hexojs/hexo/issues" target="_blank" rel="noopener">GitHub</a>.</p>
<h2 id="Quick-Start"><a href="#Quick-Start" class="headerlink" title="Quick Start"></a>Quick Start</h2><h3 id="Create-a-new-post"><a href="#Create-a-new-post" class="headerlink" title="Create a new post"></a>Create a new post</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo new <span class="string">"My New Post"</span></span><br></pre></td></tr></table></figure>

<p>More info: <a href="https://hexo.io/docs/writing.html" target="_blank" rel="noopener">Writing</a></p>
<h3 id="Run-server"><a href="#Run-server" class="headerlink" title="Run server"></a>Run server</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo server</span><br></pre></td></tr></table></figure>

<p>More info: <a href="https://hexo.io/docs/server.html" target="_blank" rel="noopener">Server</a></p>
<h3 id="Generate-static-files"><a href="#Generate-static-files" class="headerlink" title="Generate static files"></a>Generate static files</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo generate</span><br></pre></td></tr></table></figure>

<p>More info: <a href="https://hexo.io/docs/generating.html" target="_blank" rel="noopener">Generating</a></p>
<h3 id="Deploy-to-remote-sites"><a href="#Deploy-to-remote-sites" class="headerlink" title="Deploy to remote sites"></a>Deploy to remote sites</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo deploy</span><br></pre></td></tr></table></figure>

<p>More info: <a href="https://hexo.io/docs/deployment.html" target="_blank" rel="noopener">Deployment</a></p>

	
	</div>
  <a type="button" href="/2019/08/13/hello-world/#more" class="btn btn-default more">Read More</a>
</div>

		

		</div>

		<!-- pagination -->
		<div>
  		<center>
		<div class="pagination">

   
    
     <a href="/" type="button" class="btn btn-default"><i class="fa fa-arrow-circle-o-left"></i> Prev</a>
      

        <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
 
          <a type="button" class="btn btn-default disabled">Next<i class="fa fa-arrow-circle-o-right"></i></a>
        

  
</div>

  		</center>
		</div>

		
		
	</div> <!-- col-md-9 -->

	
		<div class="col-md-3">
	<div id="sidebar">
	
			
  <div id="site_search">
   <div class="form-group">
    <input type="text" id="local-search-input" name="q" results="0" placeholder="Search" class="st-search-input st-default-search-input form-control"/>
   </div>  
  <div id="local-search-result"></div>
  </div>


		
			
		
			
	<div class="widget">
		<h4>Tag Cloud</h4>
		<ul class="tag_box inline list-unstyled">		
		
			<li><a href="/tags/OrthoFinder/">OrthoFinder<span>1</span></a></li>
		
			<li><a href="/tags/gene-family/">gene family<span>1</span></a></li>
		
			<li><a href="/tags/KaKscalculator-R/">KaKscalculator,R<span>1</span></a></li>
		
			<li><a href="/tags/Python/">Python<span>1</span></a></li>
		
			<li><a href="/tags/geome/">geome<span>1</span></a></li>
		
			<li><a href="/tags/Markdown-maker/">Markdown, maker<span>2</span></a></li>
		
			<li><a href="/tags/Mysql-OrthoMCL/">Mysql, OrthoMCL<span>1</span></a></li>
		
			<li><a href="/tags/GiagaScience/">GiagaScience<span>1</span></a></li>
		
			<li><a href="/tags/gene-duplication/">gene duplication<span>1</span></a></li>
		
			<li><a href="/tags/HMM/">HMM<span>1</span></a></li>
		
			<li><a href="/tags/NCBI/">NCBI<span>1</span></a></li>
		
			<li><a href="/tags/Ortholog/">Ortholog<span>1</span></a></li>
		
			<li><a href="/tags/Phylogeny/">Phylogeny<span>1</span></a></li>
		
			<li><a href="/tags/Ensembl/">Ensembl<span>2</span></a></li>
		
		 
		</ul>
	</div>


		
			
<div class="widget">
  <h4>Recent Posts</h4>
  <ul class="entry list-unstyled">
    
      <li>
        <a href="/2020/05/28/wget_giga/" ><i class="fa fa-file-o"></i>下载鸟类的基因组2) giga</a>
      </li>
    
      <li>
        <a href="/2020/05/28/wget_ncbi/" ><i class="fa fa-file-o"></i>下载鸟类的基因组2) ncbi</a>
      </li>
    
      <li>
        <a href="/2020/05/25/Ensembl_filter/" ><i class="fa fa-file-o"></i>过滤Ensembl中数据的冗余信息</a>
      </li>
    
      <li>
        <a href="/2020/05/25/wget_Ensembl/" ><i class="fa fa-file-o"></i>下载鸟类的基因组1) Ensembl</a>
      </li>
    
      <li>
        <a href="/2020/05/15/详细注释的OrthoFinder/" ><i class="fa fa-file-o"></i>详细一点的Orthofinder，与之前英文的没区别</a>
      </li>
    
  </ul>
</div>

		
			
<div class="widget">
	<h4>Links</h4>
	<ul class="blogroll list-unstyled">
	
		<li><i class="fa fa-github"></i><a href="https://github.com/goodlucksun" title="All Github repository." target="_blank"]);">My github</a></li>
	
	</ul>
</div>


		
	</div> <!-- sidebar -->
</div> <!-- col-md-3 -->

	
	
</div> <!-- row-fluid -->
	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2020 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
