<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>建一颗ML树 | Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="建一颗ML树">
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header">
			<h1> 建一颗ML树</h1>
		</div>
	



<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <h4 id="流程大概就是"><a href="#流程大概就是" class="headerlink" title="流程大概就是"></a>流程大概就是</h4><p>刘亮老师讲的那个差不多</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 安装mafft</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 下载mafft，因为mafft的安装包是.rpm结尾的压缩包，rpm包如果不用管理员安装的话会提示权限错误，但是咱这脑子就别用管理员了，万一删了啥不该删的。rpm2cpio这个工具可以把rpm包在当前目录解压缩，使用方法为 rpm2cpio XXX.rpm | cpio -div 解压出来的结果会默认先放在usr里面</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> wget https://mafft.cbrc.jp/alignment/software/mafft-7.450-gcc_fc6.x86_64.rpm</span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 由于mac没有rpm2cpio, linux跳过brew install  rpm2cpio这句</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> brew install  rpm2cpio</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> rpm2cpio mafft-7.450-gcc_fc6.x86_64.rpm | cpio -div</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 解压完可以看到出现了一个usr目录，usr里的bin放的是执行文件，mafft要求所有所有执行文件都在环境变量里，所以下一步先添加环境变量，就是快捷方式，首先要查看当前目录在哪儿,<span class="built_in">pwd</span> 获取当前目录,假如说现在目录在/home/zhengyuno/softwares/mafft</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">cd</span> usr</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">cd</span> bin</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">pwd</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash">/home/zhengyuno/softwares/usr/bin</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> vim ~/.bash_profile</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 滚到最下面的空白处把<span class="built_in">pwd</span>的结果加上一些符号粘贴进去，vim编辑器输入时候，摁i是进入insert模式，输入完了摁esc退出insert模式，然后摁 :wq 保存并退出 （左下角会显示状态）。输入的时候一定要记得前后的PATH要不然就啥快捷方式都没了，别问我为啥知道，试过好多次了T.T</span></span><br><span class="line"></span><br><span class="line">export PATH=/home/zhengyuno/softwares/mafft/usr/bin:PATH</span><br><span class="line"><span class="meta">#</span><span class="bash"> 输入mafft出来这个交互式页面就ok了</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> mafft</span></span><br><span class="line"></span><br><span class="line">---------------------------------------------------------------------</span><br><span class="line"></span><br><span class="line">   MAFFT v7.450 (2019/Aug/23)</span><br><span class="line"></span><br><span class="line">        MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)</span><br><span class="line">        https://mafft.cbrc.jp/alignment/software/</span><br><span class="line">---------------------------------------------------------------------</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">Input file? (FASTA format; Folder=/Users/sunhaoyun/temp_dir/usr/bin)</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> ctrl + c 中断当前程序，然后我们就不用交互式的模式了，直接用一行的<span class="built_in">command</span> line模式就行</span></span><br></pre></td></tr></table></figure>

<p>这里推荐看<a href="http://www.chenlianfu.com/?p=2214" target="_blank" rel="noopener">陈连福老师的博客</a> </p>
<p>以及<a href="https://mafft.cbrc.jp/alignment/software/manual/manual.html" target="_blank" rel="noopener">官方文档</a>，当然我这种比较懒的官方的manual都没看完过</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## mafft的使用 首先选一个算法，看一下算法有很多，有的适合差不多长的，有的适合比较近缘的，不会选，用auto，看一下output选项，因为下一步过滤要fasta格式，他默认也是fasta,不用改</span></span></span><br><span class="line"><span class="meta">$</span><span class="bash"> mafft --auto cds_ENSG00000000419.fasta &gt; cds419_mafft_out.fa</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">运行完了屏幕上的信息可以保存一下</span></span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 安装G-block</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> gblock安装很简单，回到softwares目录. Gblock是tar z 压缩文件 用tar 的xz命令就行</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> wget http://molevol.cmima.csic.es/castresana/Gblocks/Gblocks_Linux64_0.91b.tar.Z</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> tar zxvf Gblocks_Linux64_0.91b.tar.Z</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">cd</span> Gblocks_0.91b</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> ./Gblocks</span></span><br><span class="line">******************************************************</span><br><span class="line">                    GBLOCKS 0.91b                     </span><br><span class="line">SELECTION OF CONSERVED BLOCKS FROM MULTIPLE ALIGNMENTS</span><br><span class="line">        FOR THEIR USE IN PHYLOGENETIC ANALYSIS        </span><br><span class="line">******************************************************</span><br><span class="line"></span><br><span class="line">o. Open File</span><br><span class="line"></span><br><span class="line">b. Block Parameters</span><br><span class="line"></span><br><span class="line">s. Saving Options</span><br><span class="line"></span><br><span class="line">g. (Get Blocks)</span><br><span class="line"></span><br><span class="line">q. Quit</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">Your Choice: </span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 应该可以运行 也是交互式的 ctrl + c 退出就行,也添加一下变量吧，</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> vim ~/.bash_profile</span></span><br><span class="line">export PATH=/home/zhengyuno/softwares/Gblocks_0.91b:PATH</span><br></pre></td></tr></table></figure>

<p>Gblocks 的<a href="http://molevol.cmima.csic.es/castresana/Gblocks/Gblocks_documentation.html" target="_blank" rel="noopener">官方文档</a></p>
<p>这个还是得看一下</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## -t=选择氨基酸还是核酸 -b1几几这些他的默认要求都比较高，所以如果使用默认的在亲缘关系较远的物种，就会什么都不剩下，所以可以选择不过滤或者手动把b12345都设置的低一些，比如我这个有90个物种，如果都是默认参数在所有物种里都保守的话，就没有保守的可以算进来了，主要还是看研究的问题是什么</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">$</span><span class="bash"> Gblocks out.fa -b1=10 -b2=10 -b3=6 -b4=10 -b5=h </span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 默认是原来的名字加-gb结尾的fa文件</span></span></span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 安装raxml</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 其实现在raxml-ng 出来了稳定版，更推荐使用这个新的，因为raxml-ng的wiki写的真的特别好，一步一步的。不过当时刘亮老师教的raxml现在用的也很多,就还用raxml把，回到softwares</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> raxml 用git <span class="built_in">clone</span>安装,下好了进去直接编译就好啦,一般编译多线程的版本，就是用起来更快的版本PTHREADS,两个都编译好了，因为有时候数据量小的话也不用那么多cpu</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> git <span class="built_in">clone</span> https://github.com/stamatak/standard-RAxML.git</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> <span class="built_in">cd</span> standard-RAxML</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> make -f Makefile.PTHREADS.gcc</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> make -f Makefile.gcc</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> standard-RAxML</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> vim ~/.bash_profile</span></span><br><span class="line">export PATH=/home/zhengyuno/softwares/standard-RAxML/:$PATH</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> Raxml 需要的输入文件是phylip，而Gblocks的输出文件是fasta格式，所以就用raxml里面的小工具转换一下格式。这一步时候注意目录</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> sh /home/zhengyuno/softwares/standard-RAxML/usefulScripts/convertFasta2Phylip.sh out.fa-gb &gt; out.phylip</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 运行raxml -<span class="comment">#设置100次bootstrap，-p -b 设置seed，其实raxml-ng的那个说明里面写的更清楚，按照他那个跑一遍会更好</span></span></span><br><span class="line"><span class="meta">$</span><span class="bash"> raxmlHPC -s out.phylip -n out -mGTRGAMMA -p 666 -<span class="comment">#100 -b 666 </span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 然后一个ML法构建的基因树就做好了</span></span><br></pre></td></tr></table></figure>

<p><a href="https://github.com/stamatak/standard-RAxML" target="_blank" rel="noopener">raxml文档</a></p>
<p><a href="https://github.com/amkozlov/raxml-ng/wiki" target="_blank" rel="noopener">raxml-ng文档</a></p>
	  
	</div>

	<div>
  	<center>
	<div class="pagination">

    
    
    <a type="button" class="btn btn-default disabled"><i class="fa fa-arrow-circle-o-left"></i>Prev</a>
    

    <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
    
    <a href="/2019/10/01/re/" type="button" class="btn btn-default ">Next<i
                class="fa fa-arrow-circle-o-right"></i></a>
    

    
</div>

    </center>
	</div>
	
	<!-- comment -->
	
<section id="comment">
    <h2 class="title">Comments</h2>

    
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2019-12-04 
	</div>
	

	<!-- categories -->
    

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/Phylogeny/">Phylogeny<span>1</span></a></li>
    </ul>
	</div>
		

	<!-- toc -->
	<div class="meta-widget">
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2019 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
