<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>原始数据的整理 | Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="原始数据的整理">
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header">
			<h1> 原始数据的整理</h1>
		</div>
	



<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <p>[toc]</p>
<p>####1. SPL基因在四个甘薯属物种中的鉴定</p>
<p>Identification of <em>SPL</em> genes in four <em>Ipomoea</em> genus species</p>
<p>工作目录 ./1.identification</p>
<a id="more"></a>

<p>#####1.1 甘薯属基因组的下载</p>
<blockquote>
<p><em>Ipomoea nil</em> </p>
<blockquote>
<p>genome:<a href="http://viewer.shigen.info/asagao/data/Asagao_1.2.fa.gz" target="_blank" rel="noopener">http://viewer.shigen.info/asagao/data/Asagao_1.2.fa.gz</a></p>
<p>gff:<a href="http://viewer.shigen.info/asagao/data/Asagao_1.2.gff.gz" target="_blank" rel="noopener">http://viewer.shigen.info/asagao/data/Asagao_1.2.gff.gz</a></p>
</blockquote>
<p><em>Ipomoea triloba</em></p>
<blockquote>
<p>genome:<a href="http://sweetpotato.plantbiology.msu.edu/download_v3/NSP323_triloba_chr_v3.fa.zip" target="_blank" rel="noopener">http://sweetpotato.plantbiology.msu.edu/download_v3/NSP323_triloba_chr_v3.fa.zip</a></p>
<p>gff:<a href="http://sweetpotato.plantbiology.msu.edu/download_v3/NSP323_triloba_v3.hc.gene_models.gff3.zip" target="_blank" rel="noopener">http://sweetpotato.plantbiology.msu.edu/download_v3/NSP323_triloba_v3.hc.gene_models.gff3.zip</a></p>
</blockquote>
<p><em>Ipomoea trifida</em></p>
<blockquote>
<p>genome:<a href="http://sweetpotato.plantbiology.msu.edu/download_v3/NSP306_trifida_chr_v3.fa.zip" target="_blank" rel="noopener">http://sweetpotato.plantbiology.msu.edu/download_v3/NSP306_trifida_chr_v3.fa.zip</a></p>
<p>gff:<a href="http://sweetpotato.plantbiology.msu.edu/download_v3/NSP306_trifida_v3.hc.gene_models.gff3.zip" target="_blank" rel="noopener">http://sweetpotato.plantbiology.msu.edu/download_v3/NSP306_trifida_v3.hc.gene_models.gff3.zip</a></p>
</blockquote>
<p><em>Ipomoea batatas</em></p>
<blockquote>
<p>genome:<a href="https://ipomoea-genome.org/pasi3.fa" target="_blank" rel="noopener">https://ipomoea-genome.org/pasi3.fa</a></p>
<p>gff:<a href="https://ipomoea-genome.org/pasi3.clean.gff3" target="_blank" rel="noopener">https://ipomoea-genome.org/pasi3.clean.gff3</a></p>
</blockquote>
</blockquote>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### 获取CDS和PEP序列 ######</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### 同时进行改名字，缩写为Ib Itf Itb In</span></span></span><br><span class="line">gffread Asagao_1.2.gff -g Asagao_1.2.fa -x Asagao_1.2.cds.fa</span><br><span class="line">transeq -sequence Asagao_1.2.cds.fa -outseq Asagao_1.2.pep.fa</span><br><span class="line">sed -i '' 's/_1$//g' Asagao_1.2.pep.fa</span><br><span class="line">rename 's/Asagao_1.2/In/g' Asagao_1.2.*</span><br><span class="line">gffread NSP323_triloba_v3.hc.gene_models.gff3 -g NSP323_triloba_chr_v3.fa -x NSP323_triloba_chr_v3.cds.fa</span><br><span class="line">transeq -sequence NSP323_triloba_chr_v3.cds.fa  -outseq NSP323_triloba_chr_v3.pep.fa</span><br><span class="line">sed -i '' 's/_1$//g' NSP323_triloba_chr_v3.pep.fa</span><br><span class="line">rename 's/NSP323_triloba_chr_v3/Itb/g' NSP323_triloba_chr_v3.*</span><br><span class="line">gffread NSP306_trifida_v3.hc.gene_models.gff3 -g NSP306_trifida_chr_v3.fa -x NSP306_trifida_chr_v3.cds.fa</span><br><span class="line">transeq -sequence NSP306_trifida_chr_v3.cds.fa  -outseq NSP306_trifida_chr_v3.pep.fa</span><br><span class="line">sed -i '' 's/_1$//g' NSP306_trifida_chr_v3.pep.fa</span><br><span class="line">rename 's/NSP306_trifida_chr_v3./Itf/g' NSP306_trifida_chr_v3.*</span><br><span class="line">gffread pasi3.clean.gff3 -g pasi3.fa -x pasi3.cds.fa</span><br><span class="line">transeq -sequence pasi3.cds.fa -outseq pasi3.pep.fa</span><br><span class="line">sed -i '' 's/_1$//g' pasi3.pep.fa</span><br><span class="line">rename 's/pasi3/Ib/g' pasi3.*</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### 转录本数量 #####</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### Ib.pep.fa ####</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#####  64295  ######</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### In.pep.fa ####</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#####  45365   #####</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### Itb.pep.fa ###</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#####  47091   #####</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### Itf.pep.fa ###</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#####  44158   #####</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### 获取最长转录本 ######</span></span></span><br><span class="line">perl ../0_scripts/retrive_longest_splicing_variant.pl Ib.pep.fa Ib.pep.longest.fa</span><br><span class="line">perl ../0_scripts/retrive_longest_splicing_variant.pl Itf.pep.fa Itf.pep.longest.fa</span><br><span class="line">perl ../0_scripts/retrive_longest_splicing_variant.pl Itb.pep.fa Itb.pep.longest.fa</span><br><span class="line">perl ../0_scripts/retrive_longest_splicing_variant.pl In.pep.fa In.pep.longest.fa</span><br><span class="line">perl ../0_scripts/retrive_longest_splicing_variant.pl Ib.cds.fa Ib.cds.longest.fa</span><br><span class="line">perl ../0_scripts/retrive_longest_splicing_variant.pl Itf.cds.fa Itf.cds.longest.fa</span><br><span class="line">perl ../0_scripts/retrive_longest_splicing_variant.pl Itb.cds.fa Itb.cds.longest.fa</span><br><span class="line">perl ../0_scripts/retrive_longest_splicing_variant.pl In.cds.fa In.cds.longest.fa</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### 最长转录本数量 ########</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###  Ib.pep.longest.fa ###</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###  64295 ###############</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###  In.pep.longest.fa ###</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###  42783 ###############</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###  Itb.pep.longest.fa ##</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###  31426 ###############</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###  Itf.pep.longest.fa ##</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###  32301 ###############</span></span></span><br></pre></td></tr></table></figure>

<p>0_scripts/retrive_longest_splicing_variant.pl</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#!/usr/bin/perl -w</span></span><br><span class="line">use strict;</span><br><span class="line">use Bio::SeqIO;</span><br><span class="line">use Bio::Perl;</span><br><span class="line">use Bio::Seq;</span><br><span class="line">use Bio::PrimarySeq;</span><br><span class="line"></span><br><span class="line"><span class="comment"># zhanglei, 2018-10-11</span></span><br><span class="line"></span><br><span class="line">my $pep = $ARGV[<span class="number">0</span>];  <span class="comment"># pep fasta file</span></span><br><span class="line">my $out = $ARGV[<span class="number">1</span>];  <span class="comment"># outputfile</span></span><br><span class="line">die <span class="string">"Usage: $0 pep.fasta longest.pep.fasta\n"</span> <span class="keyword">if</span> @ARGV &lt; <span class="number">2</span>;</span><br><span class="line"></span><br><span class="line"><span class="comment">#</span></span><br><span class="line">my %hash;</span><br><span class="line">my $protein_seq = Bio::SeqIO -&gt; new(-file =&gt; "$pep", -format =&gt; "fasta");</span><br><span class="line">while(my $seq_pp = $protein_seq -&gt; next_seq)&#123;</span><br><span class="line">	my $seq_id = $seq_pp -&gt; display_id;</span><br><span class="line">	my $seq = $seq_pp -&gt; seq;</span><br><span class="line">	my @array = (split /\./, $seq_id);</span><br><span class="line">	my $gene_id = $array[<span class="number">0</span>];</span><br><span class="line">	$hash&#123;$gene_id&#125;&#123;$seq_id&#125; = $seq;</span><br><span class="line">&#125;</span><br><span class="line"><span class="comment">#</span></span><br><span class="line">open OUT, <span class="string">"&gt;$out.length_info"</span>;</span><br><span class="line">my %longest;</span><br><span class="line">foreach my $gene_id (keys %hash)&#123;</span><br><span class="line">	my @seqs = keys %&#123;$hash&#123;$gene_id&#125;&#125;;</span><br><span class="line">	<span class="keyword">if</span> (scalar @seqs == <span class="number">1</span>)&#123;</span><br><span class="line">		$longest&#123;$gene_id&#125; = $seqs[<span class="number">0</span>];</span><br><span class="line">		my $length = length($hash&#123;$gene_id&#125;&#123;$seqs[<span class="number">0</span>]&#125;);</span><br><span class="line">		<span class="keyword">print</span> OUT <span class="string">"$gene_id\t$seqs[0]\t$length\n"</span>;</span><br><span class="line">	&#125;</span><br><span class="line">	elsif (scalar @seqs &gt; <span class="number">1</span>) &#123;</span><br><span class="line">		my $max_id = $seqs[<span class="number">0</span>];</span><br><span class="line">		my $max_length = length($hash&#123;$gene_id&#125;&#123;$seqs[<span class="number">0</span>]&#125;);</span><br><span class="line">		<span class="keyword">print</span> OUT <span class="string">"$gene_id\t$seqs[0]\t$max_length\n"</span>;</span><br><span class="line">		shift @seqs;</span><br><span class="line">		foreach my $seq (@seqs) &#123;</span><br><span class="line">			my $length = length($hash&#123;$gene_id&#125;&#123;$seq&#125;);</span><br><span class="line">			<span class="keyword">print</span> OUT <span class="string">"$gene_id\t$seq\t$length\n"</span>;</span><br><span class="line">			<span class="keyword">if</span> ($length &gt; $max_length) &#123;</span><br><span class="line">				$max_id = $seq;</span><br><span class="line">				$max_length = $length;</span><br><span class="line">			&#125;</span><br><span class="line">		&#125;</span><br><span class="line">		$longest&#123;$gene_id&#125; = $max_id;</span><br><span class="line">	&#125;</span><br><span class="line">&#125;</span><br><span class="line">close OUT;</span><br><span class="line"><span class="comment">#</span></span><br><span class="line">my $protein_io = Bio::SeqIO -&gt; new(-file =&gt; "&gt;$out", -format =&gt; "fasta");</span><br><span class="line">foreach my $gene_id (sort keys %longest) &#123;</span><br><span class="line">	my $transcript_id = $longest&#123;$gene_id&#125;;</span><br><span class="line">	my $seq           = $hash&#123;$gene_id&#125;&#123;$transcript_id&#125;;</span><br><span class="line">	my $length        = length($seq);</span><br><span class="line">	my $id            = <span class="string">"$transcript_id"</span>;</span><br><span class="line">	my $seqobj        = Bio::Seq -&gt; new (-display_id =&gt; $id, -seq =&gt; $seq);</span><br><span class="line">	$protein_io-&gt;write_seq($seqobj);</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

<h5 id="1-2-拟南芥SPL基因和pfam中的SBP结构"><a href="#1-2-拟南芥SPL基因和pfam中的SBP结构" class="headerlink" title="1.2 拟南芥SPL基因和pfam中的SBP结构"></a>1.2 拟南芥<em>SPL</em>基因和pfam中的SBP结构</h5><blockquote>
<p><em>Arabidopsis</em> <em>SPL</em> genes 参考:Cardon G, Hohmann S, Klein J, Nettesheim K, Saedler H, Huijser P. (1999) <a href="http://arabidopsis.org/servlets/TairObject?type=publication&id=1087" target="_blank" rel="noopener">Molecular characterisation of the Arabidopsis SBP-box genes</a> Gene 237:91-104</p>
<table>
<thead>
<tr>
<th>Protein Name</th>
<th>Genomic_LOC</th>
</tr>
</thead>
<tbody><tr>
<td>AtSPL1</td>
<td>At2g47070</td>
</tr>
<tr>
<td>AtSPL2</td>
<td>At5g43270</td>
</tr>
<tr>
<td>AtSPL3</td>
<td>At2g33810</td>
</tr>
<tr>
<td>AtSPL4</td>
<td>At1g53160</td>
</tr>
<tr>
<td>AtSPL5</td>
<td>At3g15270</td>
</tr>
<tr>
<td>AtSPL6</td>
<td>At1g69170</td>
</tr>
<tr>
<td>AtSPL7</td>
<td>At5g18830</td>
</tr>
<tr>
<td>AtSPL8</td>
<td>At1g02065</td>
</tr>
<tr>
<td>AtSPL9</td>
<td>At2g42200</td>
</tr>
<tr>
<td>AtSPL10</td>
<td>At1g27370</td>
</tr>
<tr>
<td>AtSPL11</td>
<td>At1g27360</td>
</tr>
<tr>
<td>AtSPL12</td>
<td>At3g60030</td>
</tr>
<tr>
<td>AtSPL13</td>
<td>At5g50570</td>
</tr>
<tr>
<td>AtSPL14</td>
<td>At1g20980</td>
</tr>
<tr>
<td>AtSPL15</td>
<td>At3g57920</td>
</tr>
<tr>
<td>AtSPL16</td>
<td>At1g76580</td>
</tr>
</tbody></table>
<p>下载以上SPL蛋白序列保存为 At.spl.pep.fa</p>
</blockquote>
<blockquote>
<p>SBP (PF03110) HMM model:<a href="http://pfam.xfam.org/family/PF03110/hmm" target="_blank" rel="noopener">http://pfam.xfam.org/family/PF03110/hmm</a></p>
<p>保存为SBP.hmm</p>
</blockquote>
<p>#####1.3 SPL基因的鉴定</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">for i in &#123;Ib,Itf,Itb,In&#125;</span><br><span class="line">do</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## hmmsearch 根据pfam鉴定潜在SPL基因</span></span></span><br><span class="line">hmmsearch -o $&#123;i&#125;.hmmsearch.log --tblout $&#123;i&#125;.hmmsearch.tblout  ../1.2_At_and_pfam/SBP.hmm  ../1.1_Ipomoea_genome/$&#123;i&#125;.pep.longest.fa</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 提取hmmsearch鉴定出的id</span></span></span><br><span class="line">awk '&#123;if ($1!~/^#/)print $1&#125;' $&#123;i&#125;.hmmsearch.tblout &gt; $&#123;i&#125;.hmmsearch.seqid</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 构建blastp的文库</span></span></span><br><span class="line">makeblastdb -in ../1.1_Ipomoea_genome/$&#123;i&#125;.pep.longest.fa  -title batatas -dbtype prot -parse_seqids -out $&#123;i&#125;</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##  blastp 根据拟南芥鉴定潜在SPL基因</span></span></span><br><span class="line">blastp -db $&#123;i&#125; -query ../1.2_At_and_pfam/At.spl.pep.fa -evalue 1e-3 -outfmt 6 &gt; $&#123;i&#125;.blastp</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##  提取blastp鉴定出的id </span></span></span><br><span class="line">awk '&#123;print $2&#125;' $&#123;i&#125;.blastp | sort | uniq &gt; $&#123;i&#125;.blastp.seqid</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##  合并hmmseach和blastp鉴定出的id </span></span></span><br><span class="line">cat $&#123;i&#125;.hmmsearch.seqid $&#123;i&#125;.blastp.seqid | sort | uniq &gt; $&#123;i&#125;.spl.candidate.seqid</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##  提取候选spl基因</span></span></span><br><span class="line">python3 ../0_scripts/get_seq.py $&#123;i&#125;.spl.candidate.seqid ../1.1_Ipomoea_genome/$&#123;i&#125;.pep.longest.fa &gt; $&#123;i&#125;.spl.candidate.fa</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>0_scripts/get_seq.py</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#!/usr/local/bin/python3</span></span><br><span class="line"><span class="keyword">from</span> Bio <span class="keyword">import</span> SeqIO</span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line">wanted = [line.strip() <span class="keyword">for</span> line <span class="keyword">in</span> open(sys.argv[<span class="number">1</span>])]</span><br><span class="line">seqiter = SeqIO.parse(open(sys.argv[<span class="number">2</span>]), <span class="string">'fasta'</span>)</span><br><span class="line">SeqIO.write((seq <span class="keyword">for</span> seq <span class="keyword">in</span> seqiter <span class="keyword">if</span> seq.id <span class="keyword">in</span> wanted), sys.stdout, <span class="string">"fasta"</span>)</span><br></pre></td></tr></table></figure>

<p>#####1.4 SBP 结构域的鉴定</p>
<p>上传至SMART和Scanprosite鉴定SBP结构域</p>
<blockquote>
<p>SMART websites: <a href="http://smart.embl.de/" target="_blank" rel="noopener">http://smart.embl.de/</a></p>
<blockquote>
<p>进入smart后选择setup选项，勾选</p>
<p>“Outlier homologues and homologues of known structure” checked by default</p>
<p>“PFAM domains” checked by default</p>
<p>“signal peptides” checked by default</p>
<p>“internal repeats” checked by default</p>
<p>“intrinsic protein disorder” checked by default</p>
</blockquote>
<p>Scanprosite websites: <a href="https://prosite.expasy.org/scanprosite/" target="_blank" rel="noopener">https://prosite.expasy.org/scanprosite/</a></p>
</blockquote>
<p>将鉴定得到序列逐一输入，检查是否有缺少保守结构域的候选<em>SPL</em>基因</p>
<p>由于I. Batatas 基因组组装质量较低，使用transdecoder 和I. Trifida 基因结构对其进行矫正：</p>
<table>
<thead>
<tr>
<th></th>
<th>Original</th>
<th>Manually</th>
</tr>
</thead>
<tbody><tr>
<td></td>
<td>g39905.t1</td>
<td>g39905.t1</td>
</tr>
<tr>
<td>transcript_s</td>
<td>12671777</td>
<td>12671777</td>
</tr>
<tr>
<td>transcript_e</td>
<td>12677342</td>
<td>12677483</td>
</tr>
<tr>
<td>exon1_s</td>
<td>12671777</td>
<td>12671777</td>
</tr>
<tr>
<td>exon1_e</td>
<td>12671976</td>
<td>12671976</td>
</tr>
<tr>
<td>exon2_s</td>
<td>12673273</td>
<td><strong><em>12673111</em></strong></td>
</tr>
<tr>
<td>exon2_e</td>
<td>12673864</td>
<td>12673864</td>
</tr>
<tr>
<td>exon3_s</td>
<td>12674075</td>
<td>12674075</td>
</tr>
<tr>
<td>exon3_e</td>
<td>12674802</td>
<td>12674802</td>
</tr>
<tr>
<td>exon4_s</td>
<td>12675339</td>
<td>12675339</td>
</tr>
<tr>
<td>exon4_e</td>
<td>12675454</td>
<td>12675475</td>
</tr>
<tr>
<td>exon5_s</td>
<td>12675882</td>
<td>12675882</td>
</tr>
<tr>
<td>exon5_e</td>
<td>12676118</td>
<td>12676118</td>
</tr>
<tr>
<td>exon6_s</td>
<td>12676881</td>
<td>12676881</td>
</tr>
<tr>
<td>exon6_e</td>
<td>12677342</td>
<td>12677342</td>
</tr>
</tbody></table>
<table>
<thead>
<tr>
<th></th>
<th>Original</th>
<th>Manually</th>
</tr>
</thead>
<tbody><tr>
<td></td>
<td>g13567.t1</td>
<td>g13567.t1</td>
</tr>
<tr>
<td>transcript_e</td>
<td>5277025</td>
<td>5277025</td>
</tr>
<tr>
<td>transcript_s</td>
<td>5281566</td>
<td>5281652</td>
</tr>
<tr>
<td>exon1_e</td>
<td>5281316</td>
<td>5281316</td>
</tr>
<tr>
<td>exon1_s</td>
<td>5281566</td>
<td><strong><em>5281652</em></strong></td>
</tr>
<tr>
<td>exon2_e</td>
<td>5279657</td>
<td>5279657</td>
</tr>
<tr>
<td>exon2_s</td>
<td>5280371</td>
<td>5280371</td>
</tr>
<tr>
<td>exon3_e</td>
<td>5279374</td>
<td>5279374</td>
</tr>
<tr>
<td>exon3_s</td>
<td>5279510</td>
<td>5279510</td>
</tr>
<tr>
<td>exon4_e</td>
<td>5278907</td>
<td>5278907</td>
</tr>
<tr>
<td>exon4_s</td>
<td>5279167</td>
<td>5279167</td>
</tr>
<tr>
<td>exon5_e</td>
<td>5278214</td>
<td><strong><em>5278106</em></strong></td>
</tr>
<tr>
<td>exon5_s</td>
<td>5278653</td>
<td>5277404</td>
</tr>
<tr>
<td>exon6_e</td>
<td>5277025</td>
<td></td>
</tr>
<tr>
<td>exon6_s</td>
<td>5277404</td>
<td></td>
</tr>
</tbody></table>
<table>
<thead>
<tr>
<th></th>
<th>Original</th>
<th>Manually</th>
<th></th>
<th>Original</th>
<th>Manually</th>
</tr>
</thead>
<tbody><tr>
<td></td>
<td>g53175.t1</td>
<td>g53175.t1</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_e</td>
<td>14675855</td>
<td>14675855</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_s</td>
<td>14679645</td>
<td>14679645</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>exon1_e</td>
<td>14678787</td>
<td>14678821</td>
<td>cds1_e</td>
<td>14678787</td>
<td>14678821</td>
</tr>
<tr>
<td>exon1_s</td>
<td>14679645</td>
<td>14679645</td>
<td>cds1_s</td>
<td>14679418</td>
<td>14679418</td>
</tr>
<tr>
<td>exon2_e</td>
<td>14678082</td>
<td>14678082</td>
<td>cds2_e</td>
<td>14678082</td>
<td>14678082</td>
</tr>
<tr>
<td>exon2_s</td>
<td>14678218</td>
<td>14678218</td>
<td>cds2_s</td>
<td>14678218</td>
<td>14678218</td>
</tr>
<tr>
<td>exon3_e</td>
<td>14676968</td>
<td>14676895</td>
<td>cds3_e</td>
<td>14676968</td>
<td>14676898</td>
</tr>
<tr>
<td>exon3_s</td>
<td>14677743</td>
<td>14677743</td>
<td>cds3_s</td>
<td>14677743</td>
<td>14677743</td>
</tr>
<tr>
<td>exon4_e</td>
<td>14676518</td>
<td>14675855</td>
<td>cds4_e</td>
<td>14676518</td>
<td></td>
</tr>
<tr>
<td>exon4_s</td>
<td>14676623</td>
<td>14676332</td>
<td>cds4_s</td>
<td>14676623</td>
<td></td>
</tr>
<tr>
<td>exon5_e</td>
<td>14675855</td>
<td></td>
<td>cds5_e</td>
<td>14676079</td>
<td></td>
</tr>
<tr>
<td>exon5_s</td>
<td>14676275</td>
<td></td>
<td>cds5_s</td>
<td>14676275</td>
<td></td>
</tr>
</tbody></table>
<table>
<thead>
<tr>
<th></th>
<th>Original</th>
<th>Manually</th>
<th></th>
<th>Original</th>
<th>Manually</th>
</tr>
</thead>
<tbody><tr>
<td></td>
<td>g13053.t1</td>
<td>g13053.t1</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_s</td>
<td>1987008</td>
<td>1987008</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_e</td>
<td>1990526</td>
<td>1990526</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>exon1_s</td>
<td>1987008</td>
<td>1987008</td>
<td>cds1_s</td>
<td>1987735</td>
<td>1987735</td>
</tr>
<tr>
<td>exon1_e</td>
<td>1987210</td>
<td>1987210</td>
<td>cds1_e</td>
<td>1988585</td>
<td>1988585</td>
</tr>
<tr>
<td>exon2_s</td>
<td>1987359</td>
<td>1987359</td>
<td>cds2_s</td>
<td>1988856</td>
<td><strong><em>1988936</em></strong></td>
</tr>
<tr>
<td>exon2_e</td>
<td>1988585</td>
<td>1988585</td>
<td>cds2_e</td>
<td>1988935</td>
<td><strong><em>1989040</em></strong></td>
</tr>
<tr>
<td>exon3_s</td>
<td>1988856</td>
<td><strong><em>1988936</em></strong></td>
<td>cds3_s</td>
<td>1989263</td>
<td><strong><em>1989263</em></strong></td>
</tr>
<tr>
<td>exon3_e</td>
<td>1988935</td>
<td><strong><em>1989040</em></strong></td>
<td>cds3_e</td>
<td>1989777</td>
<td><strong><em>1989914</em></strong></td>
</tr>
<tr>
<td>exon4_s</td>
<td>1989263</td>
<td>1989263</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>exon4_e</td>
<td>1990526</td>
<td>1990526</td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody></table>
<table>
<thead>
<tr>
<th></th>
<th>Original</th>
<th>Manually</th>
<th></th>
<th>Original</th>
<th>Manually</th>
</tr>
</thead>
<tbody><tr>
<td></td>
<td>g47387.t1</td>
<td>g47387.t1</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_e</td>
<td>3502531</td>
<td>3502531</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_s</td>
<td>3506412</td>
<td>3506412</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>exon1_e</td>
<td>3506352</td>
<td><strong><em>3504159</em></strong></td>
<td>cds1_e</td>
<td>3504440</td>
<td><strong><em>3504159</em></strong></td>
</tr>
<tr>
<td>exon1_s</td>
<td>3506412</td>
<td><strong><em>3504674</em></strong></td>
<td>cds1_s</td>
<td>3504672</td>
<td><strong><em>3504672</em></strong></td>
</tr>
<tr>
<td>exon2_e</td>
<td>3504440</td>
<td>3503962</td>
<td>cds2_e</td>
<td>3504159</td>
<td>3503962</td>
</tr>
<tr>
<td>exon2_s</td>
<td>3504674</td>
<td>3504065</td>
<td>cds2_s</td>
<td>3504275</td>
<td>3504065</td>
</tr>
<tr>
<td>exon3_e</td>
<td>3503962</td>
<td>3503083</td>
<td>cds3_e</td>
<td>3503962</td>
<td>3503083</td>
</tr>
<tr>
<td>exon3_s</td>
<td>3504065</td>
<td>3503095</td>
<td>cds3_s</td>
<td>3504065</td>
<td>3503095</td>
</tr>
<tr>
<td>exon4_e</td>
<td>3503083</td>
<td>3502531</td>
<td>cds4_e</td>
<td>3503083</td>
<td></td>
</tr>
<tr>
<td>exon4_s</td>
<td>3503095</td>
<td>3502988</td>
<td>cds4_s</td>
<td>3503095</td>
<td></td>
</tr>
<tr>
<td>exon5_e</td>
<td>3502531</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>exon5_s</td>
<td>3502988</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody></table>
<table>
<thead>
<tr>
<th></th>
<th>Original</th>
<th>Manually</th>
<th></th>
<th>Original</th>
<th>Manually</th>
</tr>
</thead>
<tbody><tr>
<td></td>
<td>g54457.t1</td>
<td>g54457.t1</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_s</td>
<td>23998829</td>
<td>23996558</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_e</td>
<td>24003634</td>
<td>24003634</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>exon1_s</td>
<td>23998829</td>
<td><strong><em>23996558</em></strong></td>
<td>cds1_s</td>
<td>23998967</td>
<td>23998768</td>
</tr>
<tr>
<td>exon1_e</td>
<td>23999115</td>
<td><strong><em>23996829</em></strong></td>
<td>cds1_e</td>
<td>23999115</td>
<td>23999115</td>
</tr>
<tr>
<td>exon2_s</td>
<td>24000638</td>
<td><strong><em>23998098</em></strong></td>
<td>cds2_s</td>
<td>24000638</td>
<td>24000638</td>
</tr>
<tr>
<td>exon2_e</td>
<td>24000847</td>
<td><strong><em>23998168</em></strong></td>
<td>cds2_e</td>
<td>24000847</td>
<td>24000847</td>
</tr>
<tr>
<td>exon3_s</td>
<td>24003181</td>
<td>23998768</td>
<td>cds3_s</td>
<td>24003181</td>
<td>24003181</td>
</tr>
<tr>
<td>exon3_e</td>
<td>24003634</td>
<td>23999115</td>
<td>cds3_e</td>
<td>24003466</td>
<td>24003466</td>
</tr>
<tr>
<td>exon4_s</td>
<td></td>
<td>24000638</td>
<td>cds4_s</td>
<td></td>
<td><strong><em>23996558</em></strong></td>
</tr>
<tr>
<td>exon4_e</td>
<td></td>
<td>24000847</td>
<td>cds4_e</td>
<td></td>
<td><strong><em>23996829</em></strong></td>
</tr>
<tr>
<td>exon5_s</td>
<td></td>
<td>24003181</td>
<td>cds5_s</td>
<td></td>
<td><strong><em>23998098</em></strong></td>
</tr>
<tr>
<td>exon5_e</td>
<td></td>
<td>24003634</td>
<td>cds5_e</td>
<td></td>
<td><strong><em>23998168</em></strong></td>
</tr>
</tbody></table>
<table>
<thead>
<tr>
<th></th>
<th>Original</th>
<th>Manually</th>
<th></th>
<th>Original</th>
<th>Manually</th>
</tr>
</thead>
<tbody><tr>
<td></td>
<td>g46868.t1</td>
<td>g46868.t1</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_s</td>
<td>40109775</td>
<td>40109775</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_e</td>
<td>40115670</td>
<td>40116440</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>exon1_s</td>
<td>40109775</td>
<td>40109775</td>
<td>cds1_s</td>
<td>40111076</td>
<td>40111043</td>
</tr>
<tr>
<td>exon1_e</td>
<td>40110039</td>
<td><strong><em>40110698</em></strong></td>
<td>cds1_e</td>
<td>40111641</td>
<td>40111641</td>
</tr>
<tr>
<td>exon2_s</td>
<td>40111006</td>
<td><strong><em>40110721</em></strong></td>
<td>cds2_s</td>
<td>40111808</td>
<td>40111808</td>
</tr>
<tr>
<td>exon2_e</td>
<td>40111641</td>
<td><strong><em>40110735</em></strong></td>
<td>cds2_e</td>
<td>40112001</td>
<td>40112001</td>
</tr>
<tr>
<td>exon3_s</td>
<td>40111808</td>
<td>40111006</td>
<td>cds3_s</td>
<td>40112106</td>
<td>40112106</td>
</tr>
<tr>
<td>exon3_e</td>
<td>40112001</td>
<td>40111641</td>
<td>cds3_e</td>
<td>40112950</td>
<td>40112950</td>
</tr>
<tr>
<td>exon4_s</td>
<td>40112106</td>
<td>40111808</td>
<td>cds4_s</td>
<td>40113075</td>
<td>40113075</td>
</tr>
<tr>
<td>exon4_e</td>
<td>40112950</td>
<td>40112001</td>
<td>cds4_e</td>
<td>40113149</td>
<td>40113149</td>
</tr>
<tr>
<td>exon5_s</td>
<td>40113075</td>
<td>40112106</td>
<td>cds5_s</td>
<td>40113238</td>
<td>40113238</td>
</tr>
<tr>
<td>exon5_e</td>
<td>40113149</td>
<td>40112950</td>
<td>cds5_e</td>
<td>40113345</td>
<td>40113345</td>
</tr>
<tr>
<td>exon6_s</td>
<td>40113238</td>
<td>40113075</td>
<td>cds6_s</td>
<td>40113437</td>
<td>40113437</td>
</tr>
<tr>
<td>exon6_e</td>
<td>40113345</td>
<td>40113149</td>
<td>cds6_e</td>
<td>40113554</td>
<td>40113554</td>
</tr>
<tr>
<td>exon7_s</td>
<td>40113437</td>
<td>40113238</td>
<td>cds7_s</td>
<td>40113778</td>
<td>40113778</td>
</tr>
<tr>
<td>exon7_e</td>
<td>40113554</td>
<td>40113345</td>
<td>cds7_e</td>
<td>40113913</td>
<td>40113913</td>
</tr>
<tr>
<td>exon8_s</td>
<td>40113778</td>
<td>40113437</td>
<td>cds8_s</td>
<td>40114006</td>
<td>40114006</td>
</tr>
<tr>
<td>exon8_e</td>
<td>40113913</td>
<td>40113554</td>
<td>cds8_e</td>
<td>40114150</td>
<td>40114150</td>
</tr>
<tr>
<td>exon9_s</td>
<td>40114006</td>
<td>40113778</td>
<td>cds9_s</td>
<td>40114236</td>
<td>40114236</td>
</tr>
<tr>
<td>exon9_e</td>
<td>40114150</td>
<td>40113913</td>
<td>cds9_e</td>
<td>40114306</td>
<td><strong><em>40114835</em></strong></td>
</tr>
<tr>
<td>exon10_s</td>
<td>40114236</td>
<td>40114006</td>
<td>cds10_s</td>
<td>40114391</td>
<td>40115161</td>
</tr>
<tr>
<td>exon10_e</td>
<td>40114306</td>
<td>40114150</td>
<td>cds10_e</td>
<td>40114835</td>
<td>40115418</td>
</tr>
<tr>
<td>exon11_s</td>
<td>40114391</td>
<td>40114236</td>
<td>cds11_s</td>
<td>40115161</td>
<td></td>
</tr>
<tr>
<td>exon11_e</td>
<td>40114835</td>
<td>40114835</td>
<td>cds11_e</td>
<td>40115418</td>
<td></td>
</tr>
<tr>
<td>exon12_s</td>
<td>40115161</td>
<td>40115161</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>exon12_e</td>
<td>40115670</td>
<td><strong><em>40116440</em></strong></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody></table>
<table>
<thead>
<tr>
<th></th>
<th>Original</th>
<th>Manually</th>
<th></th>
<th>Original</th>
<th>Manually</th>
</tr>
</thead>
<tbody><tr>
<td></td>
<td>g44906.t1</td>
<td>g44906.t1</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_s</td>
<td>25648359</td>
<td>25648359</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>transcript_e</td>
<td>25656553</td>
<td>25656553</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>exon1_e</td>
<td>25648359</td>
<td><strong><em>25648359</em></strong></td>
<td>cds1_e</td>
<td>25648430</td>
<td></td>
</tr>
<tr>
<td>exon1_s</td>
<td>25648359</td>
<td><strong><em>25648868</em></strong></td>
<td>cds1_s</td>
<td>25648868</td>
<td></td>
</tr>
<tr>
<td>exon2_e</td>
<td>25648359</td>
<td></td>
<td>cds2_e</td>
<td>25649015</td>
<td></td>
</tr>
<tr>
<td>exon2_s</td>
<td>25648868</td>
<td></td>
<td>cds2_s</td>
<td>25649146</td>
<td></td>
</tr>
<tr>
<td>exon3_e</td>
<td>25649015</td>
<td>25649015</td>
<td>cds3_e</td>
<td>25649222</td>
<td></td>
</tr>
<tr>
<td>exon3_s</td>
<td>25649146</td>
<td>25649146</td>
<td>cds3_s</td>
<td>25649592</td>
<td></td>
</tr>
<tr>
<td>exon4_e</td>
<td>25649222</td>
<td>25649222</td>
<td>cds4_e</td>
<td>25649698</td>
<td></td>
</tr>
<tr>
<td>exon4_s</td>
<td>25649592</td>
<td>25649592</td>
<td>cds4_s</td>
<td>25649753</td>
<td></td>
</tr>
<tr>
<td>exon5_e</td>
<td>25649698</td>
<td>25649698</td>
<td>cds5_e</td>
<td>25649856</td>
<td></td>
</tr>
<tr>
<td>exon5_s</td>
<td>25649753</td>
<td>25649753</td>
<td>cds5_s</td>
<td>25650485</td>
<td></td>
</tr>
<tr>
<td>exon6_e</td>
<td>25649856</td>
<td>25649856</td>
<td>cds6_e</td>
<td>25651099</td>
<td></td>
</tr>
<tr>
<td>exon6_s</td>
<td>25650485</td>
<td>25650485</td>
<td>cds6_s</td>
<td>25651330</td>
<td></td>
</tr>
<tr>
<td>exon7_e</td>
<td>25651099</td>
<td>25651099</td>
<td>cds7_e</td>
<td>25651852</td>
<td><strong><em>25651317</em></strong></td>
</tr>
<tr>
<td>exon7_s</td>
<td>25651330</td>
<td><strong><em>25651397</em></strong></td>
<td>cds7_s</td>
<td>25652168</td>
<td><strong><em>25651397</em></strong></td>
</tr>
<tr>
<td>exon8_e</td>
<td>25651852</td>
<td><strong><em>25651794</em></strong></td>
<td>cds8_e</td>
<td>25652270</td>
<td><strong><em>25651794</em></strong></td>
</tr>
<tr>
<td>exon8_s</td>
<td>25652168</td>
<td></td>
<td>cds8_s</td>
<td>25652348</td>
<td>25652348</td>
</tr>
<tr>
<td>exon9_e</td>
<td>25652270</td>
<td></td>
<td>cds9_e</td>
<td>25652554</td>
<td>25652554</td>
</tr>
<tr>
<td>exon9_s</td>
<td>25652348</td>
<td>25652348</td>
<td>cds9_s</td>
<td>25652710</td>
<td>25652710</td>
</tr>
<tr>
<td>exon10_e</td>
<td>25652554</td>
<td>25652554</td>
<td>cds10_e</td>
<td>25653056</td>
<td>25653056</td>
</tr>
<tr>
<td>exon10_s</td>
<td>25652710</td>
<td>25652710</td>
<td>cds10_s</td>
<td>25653191</td>
<td>25653191</td>
</tr>
<tr>
<td>exon11_e</td>
<td>25653056</td>
<td>25653056</td>
<td>cds11_e</td>
<td>25653308</td>
<td>25653308</td>
</tr>
<tr>
<td>exon11_s</td>
<td>25653191</td>
<td>25653191</td>
<td>cds11_s</td>
<td>25653425</td>
<td>25653425</td>
</tr>
<tr>
<td>exon12_e</td>
<td>25653308</td>
<td>25653308</td>
<td>cds12_e</td>
<td>25653728</td>
<td>25653728</td>
</tr>
<tr>
<td>exon12_s</td>
<td>25653425</td>
<td>25653425</td>
<td>cds12_s</td>
<td>25653835</td>
<td>25653835</td>
</tr>
<tr>
<td>exon13_e</td>
<td>25653728</td>
<td>25653728</td>
<td>cds13_e</td>
<td>25653940</td>
<td>25653940</td>
</tr>
<tr>
<td>exon13_s</td>
<td>25653835</td>
<td>25653835</td>
<td>cds13_s</td>
<td>25654014</td>
<td>25654014</td>
</tr>
<tr>
<td>exon14_e</td>
<td>25653940</td>
<td>25653940</td>
<td>cds14_e</td>
<td>25654122</td>
<td>25654122</td>
</tr>
<tr>
<td>exon14_s</td>
<td>25654014</td>
<td>25654014</td>
<td>cds14_s</td>
<td>25654345</td>
<td>25654345</td>
</tr>
<tr>
<td>exon15_e</td>
<td>25654122</td>
<td>25654122</td>
<td>cds15_e</td>
<td>25654973</td>
<td>25654973</td>
</tr>
<tr>
<td>exon15_s</td>
<td>25654345</td>
<td>25654345</td>
<td>cds15_s</td>
<td>25654973</td>
<td><strong><em>25655145</em></strong></td>
</tr>
<tr>
<td>exon16_e</td>
<td>25654973</td>
<td><strong><em>25654973</em></strong></td>
<td>cds16_e</td>
<td>25655201</td>
<td></td>
</tr>
<tr>
<td>exon16_s</td>
<td>25655116</td>
<td><strong><em>25655145</em></strong></td>
<td>cds16_s</td>
<td>25655268</td>
<td></td>
</tr>
<tr>
<td>exon17_e</td>
<td>25655201</td>
<td></td>
<td>cds17_e</td>
<td>25655840</td>
<td>25655840</td>
</tr>
<tr>
<td>exon17_s</td>
<td>25655268</td>
<td></td>
<td>cds17_s</td>
<td>25656399</td>
<td>25656399</td>
</tr>
<tr>
<td>exon18_e</td>
<td>25655840</td>
<td>25655840</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>exon18_s</td>
<td>25656553</td>
<td>25656553</td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody></table>
<p>最终共计筛选出105个具有保守结构域的SPL基因，其locus_id 保存为</p>
<p><strong><em>Ipomoea.spl.id</em></strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 提取gff ###</span></span></span><br><span class="line">less Ipomoea.spl.id | while read line</span><br><span class="line">do</span><br><span class="line">grep -w $&#123;line&#125; ../1.1_Ipomoea_genome/*gff3</span><br><span class="line">done &gt; 4.spl.gff3</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 整理gff 提取cds pep</span></span></span><br><span class="line">gffread 4.spl.gff3 &gt; 4.gffread.final.gff</span><br><span class="line">python3 ../0_scripts/get_seq.py Ipomoea.spl.id2 ../1.3_blast_and_hmmer/Ib.spl.candidate.fa &gt; Ib.spl.fa</span><br><span class="line">python3 ../0_scripts/get_seq.py Ipomoea.spl.id2 ../1.3_blast_and_hmmer/Itf.spl.candidate.fa &gt; Itf.spl.fa</span><br><span class="line">python3 ../0_scripts/get_seq.py Ipomoea.spl.id2 ../1.3_blast_and_hmmer/Itb.spl.candidate.fa &gt; Itb.spl.fa</span><br><span class="line">python3 ../0_scripts/get_seq.py Ipomoea.spl.id2 ../1.3_blast_and_hmmer/In.spl.candidate.fa &gt; In.spl.fa</span><br><span class="line">cat *.spl.fa &gt; 4fanshu.spl_id.wid70.pep.fa</span><br><span class="line">sed -i '' 's/.t[0-9]//g' 4fanshu.spl_id.wid70.pep.fa</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 合并所有 ###</span></span></span><br><span class="line">python3 ../0_scripts/get_seq.py Ipomoea.spl.id2 ../1.1_Ipomoea_genome/Ib.cds.fa  &gt; Ib.cds.spl.fa</span><br><span class="line">python3 ../0_scripts/get_seq.py Ipomoea.spl.id2 ../1.1_Ipomoea_genome/Itf.cds.fa  &gt; Itf.cds.spl.fa</span><br><span class="line">python3 ../0_scripts/get_seq.py Ipomoea.spl.id2 ../1.1_Ipomoea_genome/Itb.cds.fa  &gt; Itb.cds.spl.fa</span><br><span class="line">python3 ../0_scripts/get_seq.py Ipomoea.spl.id2 ../1.1_Ipomoea_genome/In.cds.fa  &gt; In.cds.spl.fa</span><br><span class="line">cat *.cds.spl.fa &gt; 4fanshu.spl_id.wid70.cds.fa</span><br><span class="line">sed -i '' 's/.t[0-9]//g' 4fanshu.spl_id.wid70.cds.fa</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 替换 LOC 为 gene id ###</span></span></span><br><span class="line">rep_col1_to_col2.py rep.id 4fanshu.spl_id.wid70.cds.fa &gt; t </span><br><span class="line">mv t 4fanshu.spl_id.wid70.cds.fa</span><br><span class="line">rep_col1_to_col2.py rep.id 4fanshu.spl_id.wid70.pep.fa &gt; t</span><br><span class="line">mv t 4fanshu.spl_id.wid70.pep.fa</span><br><span class="line">rep_col1_to_col2.py rep.id 4.gffread.final.gff &gt; t</span><br><span class="line">mv t 4.gffread.final.gff</span><br></pre></td></tr></table></figure>

<p>0_scripts/rep_col1_to_col2.py </p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#!/usr/local/bin/python3</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line">ab = &#123;&#125;</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> rename_file:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> rename_file:</span><br><span class="line">        lst1 = row.strip().split(<span class="string">'\t'</span>)</span><br><span class="line">        ab[lst1[<span class="number">0</span>]] = lst1[<span class="number">1</span>]</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">2</span>]) <span class="keyword">as</span> target_file:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> target_file:</span><br><span class="line">        <span class="keyword">for</span> key,value <span class="keyword">in</span> ab.items():</span><br><span class="line">            row = row.replace(key,value)</span><br><span class="line">        print(row.strip())</span><br></pre></td></tr></table></figure>

<p>根据染色体命名为：</p>
<table>
<thead>
<tr>
<th>IbSPL1</th>
<th>g4106</th>
<th>g4106.t1</th>
</tr>
</thead>
<tbody><tr>
<td>IbSPL2</td>
<td>g4116</td>
<td>g4116.t1</td>
</tr>
<tr>
<td>IbSPL3</td>
<td>g4185</td>
<td>g4185.t1</td>
</tr>
<tr>
<td>IbSPL4</td>
<td>g5559</td>
<td>g5559.t1</td>
</tr>
<tr>
<td>IbSPL5</td>
<td>g9134</td>
<td>g9134.t1</td>
</tr>
<tr>
<td>IbSPL6</td>
<td>g12291</td>
<td>g12291.t1</td>
</tr>
<tr>
<td>IbSPL7</td>
<td>g13053</td>
<td>g13053.t1</td>
</tr>
<tr>
<td>IbSPL8</td>
<td>g13546</td>
<td>g13546.t1</td>
</tr>
<tr>
<td>IbSPL9</td>
<td>g13567</td>
<td>g13567.t1</td>
</tr>
<tr>
<td>IbSPL10</td>
<td>g13581</td>
<td>g13581.t1</td>
</tr>
<tr>
<td>IbSPL11</td>
<td>g24050</td>
<td>g24050.t1</td>
</tr>
<tr>
<td>IbSPL12</td>
<td>g24965</td>
<td>g24965.t1</td>
</tr>
<tr>
<td>IbSPL13</td>
<td>g30425</td>
<td>g30425.t1</td>
</tr>
<tr>
<td>IbSPL14</td>
<td>g30820</td>
<td>g30820.t1</td>
</tr>
<tr>
<td>IbSPL15</td>
<td>g31985</td>
<td>g31985.t1</td>
</tr>
<tr>
<td>IbSPL16</td>
<td>g31999</td>
<td>g31999.t1</td>
</tr>
<tr>
<td>IbSPL17</td>
<td>g34536</td>
<td>g34536.t1</td>
</tr>
<tr>
<td>IbSPL18</td>
<td>g39905</td>
<td>g39905.t1</td>
</tr>
<tr>
<td>IbSPL19</td>
<td>g39953</td>
<td>g39953.t1</td>
</tr>
<tr>
<td>IbSPL20</td>
<td>g44906</td>
<td>g44906.t1</td>
</tr>
<tr>
<td>IbSPL21</td>
<td>g46868</td>
<td>g46868.t1</td>
</tr>
<tr>
<td>IbSPL22</td>
<td>g47383</td>
<td>g47383.t1</td>
</tr>
<tr>
<td>IbSPL23</td>
<td>g47384</td>
<td>g47384.t1</td>
</tr>
<tr>
<td>IbSPL24</td>
<td>g47386</td>
<td>g47386.t1</td>
</tr>
<tr>
<td>IbSPL25</td>
<td>g47387</td>
<td>g47387.t1</td>
</tr>
<tr>
<td>IbSPL26</td>
<td>g53175</td>
<td>g53175.t1</td>
</tr>
<tr>
<td>IbSPL27</td>
<td>g53210</td>
<td>g53210.t1</td>
</tr>
<tr>
<td>IbSPL28</td>
<td>g54228</td>
<td>g54228.t1</td>
</tr>
<tr>
<td>IbSPL29</td>
<td>g54457</td>
<td>g54457.t1</td>
</tr>
<tr>
<td>ItfSPL1</td>
<td>itf01g00180</td>
<td>itf01g00180.t1</td>
</tr>
<tr>
<td>ItfSPL2</td>
<td>itf01g13650</td>
<td>itf01g13650.t1</td>
</tr>
<tr>
<td>ItfSPL3</td>
<td>itf01g24210</td>
<td>itf01g24210.t1</td>
</tr>
<tr>
<td>ItfSPL4</td>
<td>itf02g02590</td>
<td>itf02g02590.t2</td>
</tr>
<tr>
<td>ItfSPL5</td>
<td>itf02g12500</td>
<td>itf02g12500.t1</td>
</tr>
<tr>
<td>ItfSPL6</td>
<td>itf02g15160</td>
<td>itf02g15160.t1</td>
</tr>
<tr>
<td>ItfSPL7</td>
<td>itf04g02690</td>
<td>itf04g02690.t1</td>
</tr>
<tr>
<td>ItfSPL8</td>
<td>itf04g23550</td>
<td>itf04g23550.t4</td>
</tr>
<tr>
<td>ItfSPL9</td>
<td>itf05g01040</td>
<td>itf05g01040.t2</td>
</tr>
<tr>
<td>ItfSPL10</td>
<td>itf05g01080</td>
<td>itf05g01080.t2</td>
</tr>
<tr>
<td>ItfSPL11</td>
<td>itf07g21130</td>
<td>itf07g21130.t1</td>
</tr>
<tr>
<td>ItfSPL12</td>
<td>itf07g21140</td>
<td>itf07g21140.t1</td>
</tr>
<tr>
<td>ItfSPL13</td>
<td>itf07g21150</td>
<td>itf07g21150.t1</td>
</tr>
<tr>
<td>ItfSPL14</td>
<td>itf07g21160</td>
<td>itf07g21160.t1</td>
</tr>
<tr>
<td>ItfSPL15</td>
<td>itf08g18710</td>
<td>itf08g18710.t2</td>
</tr>
<tr>
<td>ItfSPL16</td>
<td>itf09g22050</td>
<td>itf09g22050.t1</td>
</tr>
<tr>
<td>ItfSPL17</td>
<td>itf11g01270</td>
<td>itf11g01270.t1</td>
</tr>
<tr>
<td>ItfSPL18</td>
<td>itf11g04360</td>
<td>itf11g04360.t1</td>
</tr>
<tr>
<td>ItfSPL19</td>
<td>itf11g12600</td>
<td>itf11g12600.t1</td>
</tr>
<tr>
<td>ItfSPL20</td>
<td>itf13g17160</td>
<td>itf13g17160.t8</td>
</tr>
<tr>
<td>ItfSPL21</td>
<td>itf13g17290</td>
<td>itf13g17290.t1</td>
</tr>
<tr>
<td>ItfSPL22</td>
<td>itf13g17520</td>
<td>itf13g17520.t1</td>
</tr>
<tr>
<td>ItfSPL23</td>
<td>itf13g21690</td>
<td>itf13g21690.t1</td>
</tr>
<tr>
<td>ItfSPL24</td>
<td>itf14g02560</td>
<td>itf14g02560.t3</td>
</tr>
<tr>
<td>ItfSPL25</td>
<td>itf15g02920</td>
<td>itf15g02920.t1</td>
</tr>
<tr>
<td>ItfSPL26</td>
<td>itf15g09570</td>
<td>itf15g09570.t1</td>
</tr>
<tr>
<td>ItfSPL27</td>
<td>itf00g25120</td>
<td>itf00g25120.t1</td>
</tr>
<tr>
<td>ItbSPL1</td>
<td>itb01g00120</td>
<td>itb01g00120.t1</td>
</tr>
<tr>
<td>ItbSPL2</td>
<td>itb01g14170</td>
<td>itb01g14170.t1</td>
</tr>
<tr>
<td>ItbSPL3</td>
<td>itb01g24030</td>
<td>itb01g24030.t1</td>
</tr>
<tr>
<td>ItbSPL4</td>
<td>itb02g07930</td>
<td>itb02g07930.t1</td>
</tr>
<tr>
<td>ItbSPL5</td>
<td>itb02g10600</td>
<td>itb02g10600.t1</td>
</tr>
<tr>
<td>ItbSPL6</td>
<td>itb02g17560</td>
<td>itb02g17560.t2</td>
</tr>
<tr>
<td>ItbSPL7</td>
<td>itb04g02760</td>
<td>itb04g02760.t1</td>
</tr>
<tr>
<td>ItbSPL8</td>
<td>itb04g23020</td>
<td>itb04g23020.t2</td>
</tr>
<tr>
<td>ItbSPL9</td>
<td>itb05g00470</td>
<td>itb05g00470.t3</td>
</tr>
<tr>
<td>ItbSPL10</td>
<td>itb05g00530</td>
<td>itb05g00530.t2</td>
</tr>
<tr>
<td>ItbSPL11</td>
<td>itb07g21890</td>
<td>itb07g21890.t1</td>
</tr>
<tr>
<td>ItbSPL12</td>
<td>itb07g21900</td>
<td>itb07g21900.t1</td>
</tr>
<tr>
<td>ItbSPL13</td>
<td>itb07g21950</td>
<td>itb07g21950.t1</td>
</tr>
<tr>
<td>ItbSPL14</td>
<td>itb08g12350</td>
<td>itb08g12350.t8</td>
</tr>
<tr>
<td>ItbSPL15</td>
<td>itb09g24600</td>
<td>itb09g24600.t1</td>
</tr>
<tr>
<td>ItbSPL16</td>
<td>itb10g20010</td>
<td>itb10g20010.t1</td>
</tr>
<tr>
<td>ItbSPL17</td>
<td>itb11g01250</td>
<td>itb11g01250.t1</td>
</tr>
<tr>
<td>ItbSPL18</td>
<td>itb11g04440</td>
<td>itb11g04440.t1</td>
</tr>
<tr>
<td>ItbSPL19</td>
<td>itb11g13890</td>
<td>itb11g13890.t1</td>
</tr>
<tr>
<td>ItbSPL20</td>
<td>itb13g20550</td>
<td>itb13g20550.t2</td>
</tr>
<tr>
<td>ItbSPL21</td>
<td>itb13g20670</td>
<td>itb13g20670.t2</td>
</tr>
<tr>
<td>ItbSPL22</td>
<td>itb13g20870</td>
<td>itb13g20870.t2</td>
</tr>
<tr>
<td>ItbSPL23</td>
<td>itb13g25270</td>
<td>itb13g25270.t1</td>
</tr>
<tr>
<td>ItbSPL24</td>
<td>itb14g02730</td>
<td>itb14g02730.t2</td>
</tr>
<tr>
<td>ItbSPL25</td>
<td>itb15g03190</td>
<td>itb15g03190.t1</td>
</tr>
<tr>
<td>ItbSPL26</td>
<td>itb15g09990</td>
<td>itb15g09990.t1</td>
</tr>
<tr>
<td>InSPL1</td>
<td>INIL03g17457</td>
<td>INIL03g17457.t1</td>
</tr>
<tr>
<td>InSPL2</td>
<td>INIL03g17694</td>
<td>INIL03g17694.t1</td>
</tr>
<tr>
<td>InSPL3</td>
<td>INIL03g22477</td>
<td>INIL03g22477.t1</td>
</tr>
<tr>
<td>InSPL4</td>
<td>INIL07g34397</td>
<td>INIL07g34397.t1</td>
</tr>
<tr>
<td>InSPL5</td>
<td>INIL08g38542</td>
<td>INIL08g38542.t1</td>
</tr>
<tr>
<td>InSPL6</td>
<td>INIL08g38543</td>
<td>INIL08g38543.t1</td>
</tr>
<tr>
<td>InSPL7</td>
<td>INIL08g38544</td>
<td>INIL08g38544.t1</td>
</tr>
<tr>
<td>InSPL8</td>
<td>INIL09g30067</td>
<td>INIL09g30067.t1</td>
</tr>
<tr>
<td>InSPL9</td>
<td>INIL09g30368</td>
<td>INIL09g30368.t1</td>
</tr>
<tr>
<td>InSPL10</td>
<td>INIL09g35951</td>
<td>INIL09g35951.t1</td>
</tr>
<tr>
<td>InSPL11</td>
<td>INIL10g12396</td>
<td>INIL10g12396.t1</td>
</tr>
<tr>
<td>InSPL12</td>
<td>INIL10g13059</td>
<td>INIL10g13059.t1</td>
</tr>
<tr>
<td>InSPL13</td>
<td>INIL11g22430</td>
<td>INIL11g22430.t1</td>
</tr>
<tr>
<td>InSPL14</td>
<td>INIL11g22436</td>
<td>INIL11g22436.t1</td>
</tr>
<tr>
<td>InSPL15</td>
<td>INIL12g01351</td>
<td>INIL12g01351.t1</td>
</tr>
<tr>
<td>InSPL16</td>
<td>INIL12g03042</td>
<td>INIL12g03042.t1</td>
</tr>
<tr>
<td>InSPL20</td>
<td>INIL13g08254</td>
<td>INIL13g08254.t1</td>
</tr>
<tr>
<td>InSPL19</td>
<td>INIL13g07848</td>
<td>INIL13g07848.t1</td>
</tr>
<tr>
<td>InSPL18</td>
<td>INIL13g07826</td>
<td>INIL13g07826.t1</td>
</tr>
<tr>
<td>InSPL17</td>
<td>INIL13g07816</td>
<td>INIL13g07816.t1</td>
</tr>
<tr>
<td>InSPL21</td>
<td>INIL14g03598</td>
<td>INIL14g03598.t1</td>
</tr>
<tr>
<td>InSPL22</td>
<td>INIL14g28336</td>
<td>INIL14g28336.t1</td>
</tr>
<tr>
<td>InSPL23</td>
<td>INIL15g27410</td>
<td>INIL15g27410.t1</td>
</tr>
</tbody></table>
<p>结构域范围文件保存为 </p>
<p><strong><em>SMART_result</em></strong></p>
<p>并从中提取SBP结构域 以及Deadc结构域，ANK结构域位置，保存为 </p>
<p><strong><em>spb.bed</em></strong> </p>
<p><strong><em>deadc.bed</em></strong> </p>
<p><strong><em>ank.bed</em></strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### 使用 bedtools getfasta 获取结构域的序列 ###################</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##### 由于bedtools 会从0开始计算，所以将结构域起始位置左移一位 ######</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##</span></span></span><br><span class="line">for i in &#123;sbp,deadc,ank&#125;</span><br><span class="line">do</span><br><span class="line"> bedtools getfasta -fi cc -bed $&#123;i&#125;.bed | awk -F ':' '&#123;print $1&#125;' &gt; domain.$&#123;i&#125;.fa</span><br><span class="line">mafft domain.$&#123;i&#125;.fa &gt; domain.$&#123;i&#125;.mafft.fa</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>使用GeneDoc进行保守序列的绘制</p>
<blockquote>
<p>File</p>
<blockquote>
<p>Import</p>
<blockquote>
<p>done</p>
</blockquote>
</blockquote>
<p>Project</p>
<blockquote>
<p>Configure</p>
<blockquote>
<p>No consensus</p>
</blockquote>
<p>Edit sequences List</p>
</blockquote>
</blockquote>
<p>得到结构域图ANK.pdf, sap.pdf,deac.pdf</p>
<p>#####1.5 基础的生物信息学分析</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">输入文件前去除蛋白文件中的*号</span></span><br><span class="line">sed -i 's/\*//g' 4fanshu.spl_id.wid70.pep.fa</span><br><span class="line"><span class="meta">#</span><span class="bash">得到甘薯SPL蛋白的长度，分子量大小和等电点</span></span><br><span class="line">perl ../0_scripts/stat_protein_fa.pl 4fanshu.spl_id.wid70.pep.fa 4fanshu.spl_id.wid70.pep.stat.txt</span><br></pre></td></tr></table></figure>

<p>0_scripts/stat_protein_fa.pl </p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#stat_protein_fa.pl </span></span><br><span class="line"><span class="comment">#北京组学生物科技有限公司</span></span><br><span class="line"><span class="comment">#email: huangls@biomics.com.cn</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">die</span> <span class="string">"perl $0 &lt;in&gt;  &lt;out&gt;"</span> <span class="keyword">unless</span>(@ARGV==<span class="number">2</span>);</span><br><span class="line"><span class="keyword">use</span> Bio::SeqIO;</span><br><span class="line"><span class="keyword">use</span> Bio::Seq;</span><br><span class="line"><span class="keyword">use</span> Bio::Tools::SeqStats;</span><br><span class="line"><span class="keyword">use</span> Bio::Tools::pICalculator;</span><br><span class="line"><span class="keyword">use</span> Data::Dumper;</span><br><span class="line"><span class="comment">#读入序列</span></span><br><span class="line"><span class="keyword">my</span> $in = Bio::SeqIO-&gt;new(</span><br><span class="line">        <span class="string">-file   =&gt;</span> <span class="string">"$ARGV[0]"</span>,</span><br><span class="line">        <span class="string">-format =&gt;</span> <span class="string">'Fasta'</span></span><br><span class="line">);</span><br><span class="line"></span><br><span class="line"><span class="keyword">open</span> OUT,<span class="string">"&gt;$ARGV[1]"</span> <span class="keyword">or</span> <span class="keyword">die</span> <span class="string">"$!"</span>;</span><br><span class="line"><span class="keyword">print</span> OUT <span class="string">"#ID\tlength\tMV(Da)\tpI\n"</span>;</span><br><span class="line"><span class="keyword">my</span> $calc = Bio::Tools::pICalculator-&gt;new(<span class="string">-places =&gt;</span> <span class="number">2</span>,<span class="string">-pKset =&gt;</span> <span class="string">'EMBOSS'</span>);</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">#逐条读取序列</span></span><br><span class="line"><span class="keyword">while</span> ( <span class="keyword">my</span> $seq = $in-&gt;next_se<span class="string">q()</span> ) &#123;</span><br><span class="line">        <span class="keyword">my</span> ( $id, $sequence, $desc ) = ( $seq-&gt;id, $seq-&gt;seq, $seq-&gt;desc );</span><br><span class="line">        <span class="keyword">my</span> $weight = Bio::Tools::SeqStats -&gt;get_mol_wt($seq);</span><br><span class="line">        $calc-&gt;se<span class="string">q($seq)</span>;</span><br><span class="line">    <span class="keyword">my</span> $iep = $calc-&gt;iep;</span><br><span class="line">    <span class="keyword">print</span> OUT <span class="keyword">sprintf</span>(<span class="string">"%s\t%s\t%s\t%s\n"</span>,</span><br><span class="line">                  $seq-&gt;id,</span><br><span class="line">                  $seq-&gt;<span class="keyword">length</span>,</span><br><span class="line">                  <span class="string">"$weight-&gt;[0]"</span>,</span><br><span class="line">                  $iep);</span><br><span class="line">&#125;</span><br><span class="line">$in-&gt;<span class="keyword">close</span>();</span><br><span class="line"><span class="keyword">close</span>(OUT);</span><br></pre></td></tr></table></figure>

<p>对 <strong><em>4fanshu.spl_id.wid70.pep.stat.txt</em></strong> 进行一些修改，主要是调整顺序，并把四个物种赋值添加<em>Ipomoea</em>，得到如下文件</p>
<p><strong><em>boxplot2.txt</em></strong></p>
<table>
<thead>
<tr>
<th></th>
<th>Accesion</th>
<th>length</th>
<th>MW</th>
<th>pI</th>
</tr>
</thead>
<tbody><tr>
<td>1</td>
<td>I. batatas</td>
<td>433</td>
<td>47803.9</td>
<td>8.61</td>
</tr>
<tr>
<td>2</td>
<td>I. batatas</td>
<td>267</td>
<td>30053.2</td>
<td>9.6</td>
</tr>
<tr>
<td>3</td>
<td>I. batatas</td>
<td>283</td>
<td>31455.4</td>
<td>8.5</td>
</tr>
<tr>
<td>4</td>
<td>I. batatas</td>
<td>996</td>
<td>110531.5</td>
<td>6.02</td>
</tr>
<tr>
<td>5</td>
<td>I. batatas</td>
<td>378</td>
<td>41909</td>
<td>9.18</td>
</tr>
<tr>
<td>6</td>
<td>I. batatas</td>
<td>469</td>
<td>52515.8</td>
<td>6.08</td>
</tr>
<tr>
<td>7</td>
<td>I. batatas</td>
<td>535</td>
<td>59565.6</td>
<td>6.11</td>
</tr>
<tr>
<td>8</td>
<td>I. batatas</td>
<td>988</td>
<td>109146</td>
<td>7.15</td>
</tr>
<tr>
<td>9</td>
<td>I. batatas</td>
<td>436</td>
<td>47589.8</td>
<td>8.08</td>
</tr>
<tr>
<td>10</td>
<td>I. batatas</td>
<td>337</td>
<td>36798.6</td>
<td>8.81</td>
</tr>
<tr>
<td>11</td>
<td>I. batatas</td>
<td>1141</td>
<td>126512.3</td>
<td>7.89</td>
</tr>
<tr>
<td>12</td>
<td>I. batatas</td>
<td>384</td>
<td>40488.6</td>
<td>8.73</td>
</tr>
<tr>
<td>13</td>
<td>I. batatas</td>
<td>338</td>
<td>36247.3</td>
<td>7.65</td>
</tr>
<tr>
<td>14</td>
<td>I. batatas</td>
<td>150</td>
<td>16919.8</td>
<td>9.72</td>
</tr>
<tr>
<td>15</td>
<td>I. batatas</td>
<td>487</td>
<td>53421.5</td>
<td>8.36</td>
</tr>
<tr>
<td>16</td>
<td>I. batatas</td>
<td>540</td>
<td>58946.9</td>
<td>8.37</td>
</tr>
<tr>
<td>17</td>
<td>I. batatas</td>
<td>139</td>
<td>15812.3</td>
<td>7.47</td>
</tr>
<tr>
<td>18</td>
<td>I. batatas</td>
<td>427</td>
<td>46308.2</td>
<td>9.02</td>
</tr>
<tr>
<td>19</td>
<td>I. batatas</td>
<td>557</td>
<td>60932</td>
<td>7.83</td>
</tr>
<tr>
<td>20</td>
<td>I. batatas</td>
<td>728</td>
<td>81298.5</td>
<td>6.37</td>
</tr>
<tr>
<td>21</td>
<td>I. batatas</td>
<td>1025</td>
<td>114509.7</td>
<td>6.94</td>
</tr>
<tr>
<td>22</td>
<td>I. batatas</td>
<td>436</td>
<td>49272.7</td>
<td>8.05</td>
</tr>
<tr>
<td>23</td>
<td>I. batatas</td>
<td>345</td>
<td>39232.8</td>
<td>6.73</td>
</tr>
<tr>
<td>24</td>
<td>I. batatas</td>
<td>366</td>
<td>42242.1</td>
<td>7.09</td>
</tr>
<tr>
<td>25</td>
<td>I. batatas</td>
<td>318</td>
<td>36222.5</td>
<td>8.64</td>
</tr>
<tr>
<td>26</td>
<td>I. batatas</td>
<td>527</td>
<td>56321.2</td>
<td>8.65</td>
</tr>
<tr>
<td>27</td>
<td>I. batatas</td>
<td>486</td>
<td>51655.1</td>
<td>8.73</td>
</tr>
<tr>
<td>28</td>
<td>I. batatas</td>
<td>165</td>
<td>18243.2</td>
<td>7.45</td>
</tr>
<tr>
<td>29</td>
<td>I. batatas</td>
<td>398</td>
<td>43952.3</td>
<td>10.06</td>
</tr>
<tr>
<td>30</td>
<td>I. trifida</td>
<td>1084</td>
<td>120646.7</td>
<td>7.02</td>
</tr>
<tr>
<td>31</td>
<td>I. trifida</td>
<td>729</td>
<td>81431.7</td>
<td>6.31</td>
</tr>
<tr>
<td>32</td>
<td>I. trifida</td>
<td>345</td>
<td>37016.9</td>
<td>8.87</td>
</tr>
<tr>
<td>33</td>
<td>I. trifida</td>
<td>533</td>
<td>56781.5</td>
<td>8.31</td>
</tr>
<tr>
<td>34</td>
<td>I. trifida</td>
<td>318</td>
<td>35783.3</td>
<td>10.53</td>
</tr>
<tr>
<td>35</td>
<td>I. trifida</td>
<td>195</td>
<td>21376.7</td>
<td>7.71</td>
</tr>
<tr>
<td>36</td>
<td>I. trifida</td>
<td>309</td>
<td>33695.8</td>
<td>8.78</td>
</tr>
<tr>
<td>37</td>
<td>I. trifida</td>
<td>1024</td>
<td>113522.6</td>
<td>5.57</td>
</tr>
<tr>
<td>38</td>
<td>I. trifida</td>
<td>311</td>
<td>34912.3</td>
<td>8.31</td>
</tr>
<tr>
<td>39</td>
<td>I. trifida</td>
<td>461</td>
<td>50483.5</td>
<td>8.48</td>
</tr>
<tr>
<td>40</td>
<td>I. trifida</td>
<td>348</td>
<td>39889.2</td>
<td>7.89</td>
</tr>
<tr>
<td>41</td>
<td>I. trifida</td>
<td>347</td>
<td>39802.4</td>
<td>8.33</td>
</tr>
<tr>
<td>42</td>
<td>I. trifida</td>
<td>351</td>
<td>40192.8</td>
<td>8.19</td>
</tr>
<tr>
<td>43</td>
<td>I. trifida</td>
<td>332</td>
<td>38020.3</td>
<td>5.4</td>
</tr>
<tr>
<td>44</td>
<td>I. trifida</td>
<td>453</td>
<td>49214.5</td>
<td>8.54</td>
</tr>
<tr>
<td>45</td>
<td>I. trifida</td>
<td>266</td>
<td>29341</td>
<td>9.35</td>
</tr>
<tr>
<td>46</td>
<td>I. trifida</td>
<td>341</td>
<td>36493.6</td>
<td>7.65</td>
</tr>
<tr>
<td>47</td>
<td>I. trifida</td>
<td>141</td>
<td>15914.6</td>
<td>9.46</td>
</tr>
<tr>
<td>48</td>
<td>I. trifida</td>
<td>539</td>
<td>58887</td>
<td>8.5</td>
</tr>
<tr>
<td>49</td>
<td>I. trifida</td>
<td>347</td>
<td>37870.8</td>
<td>8.98</td>
</tr>
<tr>
<td>50</td>
<td>I. trifida</td>
<td>440</td>
<td>48019.3</td>
<td>8.02</td>
</tr>
<tr>
<td>51</td>
<td>I. trifida</td>
<td>1004</td>
<td>111095.9</td>
<td>7.54</td>
</tr>
<tr>
<td>52</td>
<td>I. trifida</td>
<td>535</td>
<td>59589.6</td>
<td>6.11</td>
</tr>
<tr>
<td>53</td>
<td>I. trifida</td>
<td>497</td>
<td>55453.1</td>
<td>6.08</td>
</tr>
<tr>
<td>54</td>
<td>I. trifida</td>
<td>384</td>
<td>40437.4</td>
<td>8.73</td>
</tr>
<tr>
<td>55</td>
<td>I. trifida</td>
<td>1066</td>
<td>117775.5</td>
<td>8</td>
</tr>
<tr>
<td>56</td>
<td>I. trifida</td>
<td>139</td>
<td>15812.3</td>
<td>7.47</td>
</tr>
<tr>
<td>57</td>
<td>I. triloba</td>
<td>1083</td>
<td>120682.8</td>
<td>7.18</td>
</tr>
<tr>
<td>58</td>
<td>I. triloba</td>
<td>795</td>
<td>88681.8</td>
<td>6.25</td>
</tr>
<tr>
<td>59</td>
<td>I. triloba</td>
<td>345</td>
<td>37030.9</td>
<td>8.87</td>
</tr>
<tr>
<td>60</td>
<td>I. triloba</td>
<td>179</td>
<td>19922.1</td>
<td>10.22</td>
</tr>
<tr>
<td>61</td>
<td>I. triloba</td>
<td>193</td>
<td>21220.5</td>
<td>7.71</td>
</tr>
<tr>
<td>62</td>
<td>I. triloba</td>
<td>531</td>
<td>56552.3</td>
<td>8.43</td>
</tr>
<tr>
<td>63</td>
<td>I. triloba</td>
<td>307</td>
<td>33604.7</td>
<td>8.78</td>
</tr>
<tr>
<td>64</td>
<td>I. triloba</td>
<td>1024</td>
<td>113449.6</td>
<td>5.87</td>
</tr>
<tr>
<td>65</td>
<td>I. triloba</td>
<td>312</td>
<td>35008.4</td>
<td>8.31</td>
</tr>
<tr>
<td>66</td>
<td>I. triloba</td>
<td>461</td>
<td>50516.5</td>
<td>8.48</td>
</tr>
<tr>
<td>67</td>
<td>I. triloba</td>
<td>344</td>
<td>39186.5</td>
<td>7.92</td>
</tr>
<tr>
<td>68</td>
<td>I. triloba</td>
<td>289</td>
<td>33202.6</td>
<td>9.72</td>
</tr>
<tr>
<td>69</td>
<td>I. triloba</td>
<td>445</td>
<td>49217.7</td>
<td>5.93</td>
</tr>
<tr>
<td>70</td>
<td>I. triloba</td>
<td>450</td>
<td>49109.4</td>
<td>8.6</td>
</tr>
<tr>
<td>71</td>
<td>I. triloba</td>
<td>256</td>
<td>28236.7</td>
<td>8.85</td>
</tr>
<tr>
<td>72</td>
<td>I. triloba</td>
<td>139</td>
<td>15812.3</td>
<td>7.47</td>
</tr>
<tr>
<td>73</td>
<td>I. triloba</td>
<td>339</td>
<td>36362.5</td>
<td>7.65</td>
</tr>
<tr>
<td>74</td>
<td>I. triloba</td>
<td>140</td>
<td>15813.5</td>
<td>9.46</td>
</tr>
<tr>
<td>75</td>
<td>I. triloba</td>
<td>540</td>
<td>59083.1</td>
<td>8.62</td>
</tr>
<tr>
<td>76</td>
<td>I. triloba</td>
<td>342</td>
<td>37394.3</td>
<td>8.98</td>
</tr>
<tr>
<td>77</td>
<td>I. triloba</td>
<td>443</td>
<td>48381.8</td>
<td>8.38</td>
</tr>
<tr>
<td>78</td>
<td>I. triloba</td>
<td>1002</td>
<td>111088</td>
<td>7.68</td>
</tr>
<tr>
<td>79</td>
<td>I. triloba</td>
<td>535</td>
<td>59619.7</td>
<td>6.11</td>
</tr>
<tr>
<td>80</td>
<td>I. triloba</td>
<td>493</td>
<td>55006.7</td>
<td>6.03</td>
</tr>
<tr>
<td>81</td>
<td>I. triloba</td>
<td>380</td>
<td>40123.2</td>
<td>8.73</td>
</tr>
<tr>
<td>82</td>
<td>I. triloba</td>
<td>1060</td>
<td>117205.1</td>
<td>7.98</td>
</tr>
<tr>
<td>83</td>
<td>I. nil</td>
<td>151</td>
<td>16239.2</td>
<td>9.97</td>
</tr>
<tr>
<td>84</td>
<td>I. nil</td>
<td>184</td>
<td>20468.8</td>
<td>10.17</td>
</tr>
<tr>
<td>85</td>
<td>I. nil</td>
<td>460</td>
<td>50446.7</td>
<td>8.61</td>
</tr>
<tr>
<td>86</td>
<td>I. nil</td>
<td>460</td>
<td>51531</td>
<td>5.82</td>
</tr>
<tr>
<td>87</td>
<td>I. nil</td>
<td>287</td>
<td>32309</td>
<td>7.95</td>
</tr>
<tr>
<td>88</td>
<td>I. nil</td>
<td>358</td>
<td>40922.4</td>
<td>6.62</td>
</tr>
<tr>
<td>89</td>
<td>I. nil</td>
<td>289</td>
<td>32614.7</td>
<td>8.83</td>
</tr>
<tr>
<td>90</td>
<td>I. nil</td>
<td>343</td>
<td>36709.8</td>
<td>7.69</td>
</tr>
<tr>
<td>91</td>
<td>I. nil</td>
<td>103</td>
<td>12010.4</td>
<td>9.24</td>
</tr>
<tr>
<td>92</td>
<td>I. nil</td>
<td>527</td>
<td>57790.8</td>
<td>8.87</td>
</tr>
<tr>
<td>93</td>
<td>I. nil</td>
<td>371</td>
<td>39136.1</td>
<td>8.66</td>
</tr>
<tr>
<td>94</td>
<td>I. nil</td>
<td>1057</td>
<td>116589.4</td>
<td>7.77</td>
</tr>
<tr>
<td>95</td>
<td>I. nil</td>
<td>465</td>
<td>51098.1</td>
<td>8.41</td>
</tr>
<tr>
<td>96</td>
<td>I. nil</td>
<td>301</td>
<td>33835.1</td>
<td>8.68</td>
</tr>
<tr>
<td>97</td>
<td>I. nil</td>
<td>189</td>
<td>21206.4</td>
<td>10.55</td>
</tr>
<tr>
<td>98</td>
<td>I. nil</td>
<td>1022</td>
<td>113412.4</td>
<td>5.82</td>
</tr>
<tr>
<td>99</td>
<td>I. nil</td>
<td>309</td>
<td>34090.6</td>
<td>8.94</td>
</tr>
<tr>
<td>100</td>
<td>I. nil</td>
<td>404</td>
<td>44384.2</td>
<td>8.73</td>
</tr>
<tr>
<td>101</td>
<td>I. nil</td>
<td>993</td>
<td>110136.7</td>
<td>7.23</td>
</tr>
<tr>
<td>102</td>
<td>I. nil</td>
<td>512</td>
<td>57476</td>
<td>5.87</td>
</tr>
<tr>
<td>103</td>
<td>I. nil</td>
<td>1077</td>
<td>119855.8</td>
<td>7.34</td>
</tr>
<tr>
<td>104</td>
<td>I. nil</td>
<td>795</td>
<td>88747</td>
<td>5.84</td>
</tr>
<tr>
<td>105</td>
<td>I. nil</td>
<td>449</td>
<td>49071.2</td>
<td>7.89</td>
</tr>
<tr>
<td>106</td>
<td>Ipomoea</td>
<td>433</td>
<td>47803.9</td>
<td>8.61</td>
</tr>
<tr>
<td>107</td>
<td>Ipomoea</td>
<td>267</td>
<td>30053.2</td>
<td>9.6</td>
</tr>
<tr>
<td>108</td>
<td>Ipomoea</td>
<td>283</td>
<td>31455.4</td>
<td>8.5</td>
</tr>
<tr>
<td>109</td>
<td>Ipomoea</td>
<td>996</td>
<td>110531.5</td>
<td>6.02</td>
</tr>
<tr>
<td>110</td>
<td>Ipomoea</td>
<td>378</td>
<td>41909</td>
<td>9.18</td>
</tr>
<tr>
<td>111</td>
<td>Ipomoea</td>
<td>469</td>
<td>52515.8</td>
<td>6.08</td>
</tr>
<tr>
<td>112</td>
<td>Ipomoea</td>
<td>535</td>
<td>59565.6</td>
<td>6.11</td>
</tr>
<tr>
<td>113</td>
<td>Ipomoea</td>
<td>988</td>
<td>109146</td>
<td>7.15</td>
</tr>
<tr>
<td>114</td>
<td>Ipomoea</td>
<td>436</td>
<td>47589.8</td>
<td>8.08</td>
</tr>
<tr>
<td>115</td>
<td>Ipomoea</td>
<td>337</td>
<td>36798.6</td>
<td>8.81</td>
</tr>
<tr>
<td>116</td>
<td>Ipomoea</td>
<td>1141</td>
<td>126512.3</td>
<td>7.89</td>
</tr>
<tr>
<td>117</td>
<td>Ipomoea</td>
<td>384</td>
<td>40488.6</td>
<td>8.73</td>
</tr>
<tr>
<td>118</td>
<td>Ipomoea</td>
<td>338</td>
<td>36247.3</td>
<td>7.65</td>
</tr>
<tr>
<td>119</td>
<td>Ipomoea</td>
<td>150</td>
<td>16919.8</td>
<td>9.72</td>
</tr>
<tr>
<td>120</td>
<td>Ipomoea</td>
<td>487</td>
<td>53421.5</td>
<td>8.36</td>
</tr>
<tr>
<td>121</td>
<td>Ipomoea</td>
<td>540</td>
<td>58946.9</td>
<td>8.37</td>
</tr>
<tr>
<td>122</td>
<td>Ipomoea</td>
<td>139</td>
<td>15812.3</td>
<td>7.47</td>
</tr>
<tr>
<td>123</td>
<td>Ipomoea</td>
<td>427</td>
<td>46308.2</td>
<td>9.02</td>
</tr>
<tr>
<td>124</td>
<td>Ipomoea</td>
<td>557</td>
<td>60932</td>
<td>7.83</td>
</tr>
<tr>
<td>125</td>
<td>Ipomoea</td>
<td>728</td>
<td>81298.5</td>
<td>6.37</td>
</tr>
<tr>
<td>126</td>
<td>Ipomoea</td>
<td>1025</td>
<td>114509.7</td>
<td>6.94</td>
</tr>
<tr>
<td>127</td>
<td>Ipomoea</td>
<td>436</td>
<td>49272.7</td>
<td>8.05</td>
</tr>
<tr>
<td>128</td>
<td>Ipomoea</td>
<td>345</td>
<td>39232.8</td>
<td>6.73</td>
</tr>
<tr>
<td>129</td>
<td>Ipomoea</td>
<td>366</td>
<td>42242.1</td>
<td>7.09</td>
</tr>
<tr>
<td>130</td>
<td>Ipomoea</td>
<td>318</td>
<td>36222.5</td>
<td>8.64</td>
</tr>
<tr>
<td>131</td>
<td>Ipomoea</td>
<td>527</td>
<td>56321.2</td>
<td>8.65</td>
</tr>
<tr>
<td>132</td>
<td>Ipomoea</td>
<td>486</td>
<td>51655.1</td>
<td>8.73</td>
</tr>
<tr>
<td>133</td>
<td>Ipomoea</td>
<td>165</td>
<td>18243.2</td>
<td>7.45</td>
</tr>
<tr>
<td>134</td>
<td>Ipomoea</td>
<td>398</td>
<td>43952.3</td>
<td>10.06</td>
</tr>
<tr>
<td>135</td>
<td>Ipomoea</td>
<td>1084</td>
<td>120646.7</td>
<td>7.02</td>
</tr>
<tr>
<td>136</td>
<td>Ipomoea</td>
<td>729</td>
<td>81431.7</td>
<td>6.31</td>
</tr>
<tr>
<td>137</td>
<td>Ipomoea</td>
<td>345</td>
<td>37016.9</td>
<td>8.87</td>
</tr>
<tr>
<td>138</td>
<td>Ipomoea</td>
<td>533</td>
<td>56781.5</td>
<td>8.31</td>
</tr>
<tr>
<td>139</td>
<td>Ipomoea</td>
<td>318</td>
<td>35783.3</td>
<td>10.53</td>
</tr>
<tr>
<td>140</td>
<td>Ipomoea</td>
<td>195</td>
<td>21376.7</td>
<td>7.71</td>
</tr>
<tr>
<td>141</td>
<td>Ipomoea</td>
<td>309</td>
<td>33695.8</td>
<td>8.78</td>
</tr>
<tr>
<td>142</td>
<td>Ipomoea</td>
<td>1024</td>
<td>113522.6</td>
<td>5.57</td>
</tr>
<tr>
<td>143</td>
<td>Ipomoea</td>
<td>311</td>
<td>34912.3</td>
<td>8.31</td>
</tr>
<tr>
<td>144</td>
<td>Ipomoea</td>
<td>461</td>
<td>50483.5</td>
<td>8.48</td>
</tr>
<tr>
<td>145</td>
<td>Ipomoea</td>
<td>348</td>
<td>39889.2</td>
<td>7.89</td>
</tr>
<tr>
<td>146</td>
<td>Ipomoea</td>
<td>347</td>
<td>39802.4</td>
<td>8.33</td>
</tr>
<tr>
<td>147</td>
<td>Ipomoea</td>
<td>351</td>
<td>40192.8</td>
<td>8.19</td>
</tr>
<tr>
<td>148</td>
<td>Ipomoea</td>
<td>332</td>
<td>38020.3</td>
<td>5.4</td>
</tr>
<tr>
<td>149</td>
<td>Ipomoea</td>
<td>453</td>
<td>49214.5</td>
<td>8.54</td>
</tr>
<tr>
<td>150</td>
<td>Ipomoea</td>
<td>266</td>
<td>29341</td>
<td>9.35</td>
</tr>
<tr>
<td>151</td>
<td>Ipomoea</td>
<td>341</td>
<td>36493.6</td>
<td>7.65</td>
</tr>
<tr>
<td>152</td>
<td>Ipomoea</td>
<td>141</td>
<td>15914.6</td>
<td>9.46</td>
</tr>
<tr>
<td>153</td>
<td>Ipomoea</td>
<td>539</td>
<td>58887</td>
<td>8.5</td>
</tr>
<tr>
<td>154</td>
<td>Ipomoea</td>
<td>347</td>
<td>37870.8</td>
<td>8.98</td>
</tr>
<tr>
<td>155</td>
<td>Ipomoea</td>
<td>440</td>
<td>48019.3</td>
<td>8.02</td>
</tr>
<tr>
<td>156</td>
<td>Ipomoea</td>
<td>1004</td>
<td>111095.9</td>
<td>7.54</td>
</tr>
<tr>
<td>157</td>
<td>Ipomoea</td>
<td>535</td>
<td>59589.6</td>
<td>6.11</td>
</tr>
<tr>
<td>158</td>
<td>Ipomoea</td>
<td>497</td>
<td>55453.1</td>
<td>6.08</td>
</tr>
<tr>
<td>159</td>
<td>Ipomoea</td>
<td>384</td>
<td>40437.4</td>
<td>8.73</td>
</tr>
<tr>
<td>160</td>
<td>Ipomoea</td>
<td>1066</td>
<td>117775.5</td>
<td>8</td>
</tr>
<tr>
<td>161</td>
<td>Ipomoea</td>
<td>139</td>
<td>15812.3</td>
<td>7.47</td>
</tr>
<tr>
<td>162</td>
<td>Ipomoea</td>
<td>1083</td>
<td>120682.8</td>
<td>7.18</td>
</tr>
<tr>
<td>163</td>
<td>Ipomoea</td>
<td>795</td>
<td>88681.8</td>
<td>6.25</td>
</tr>
<tr>
<td>164</td>
<td>Ipomoea</td>
<td>345</td>
<td>37030.9</td>
<td>8.87</td>
</tr>
<tr>
<td>165</td>
<td>Ipomoea</td>
<td>179</td>
<td>19922.1</td>
<td>10.22</td>
</tr>
<tr>
<td>166</td>
<td>Ipomoea</td>
<td>193</td>
<td>21220.5</td>
<td>7.71</td>
</tr>
<tr>
<td>167</td>
<td>Ipomoea</td>
<td>531</td>
<td>56552.3</td>
<td>8.43</td>
</tr>
<tr>
<td>168</td>
<td>Ipomoea</td>
<td>307</td>
<td>33604.7</td>
<td>8.78</td>
</tr>
<tr>
<td>169</td>
<td>Ipomoea</td>
<td>1024</td>
<td>113449.6</td>
<td>5.87</td>
</tr>
<tr>
<td>170</td>
<td>Ipomoea</td>
<td>312</td>
<td>35008.4</td>
<td>8.31</td>
</tr>
<tr>
<td>171</td>
<td>Ipomoea</td>
<td>461</td>
<td>50516.5</td>
<td>8.48</td>
</tr>
<tr>
<td>172</td>
<td>Ipomoea</td>
<td>344</td>
<td>39186.5</td>
<td>7.92</td>
</tr>
<tr>
<td>173</td>
<td>Ipomoea</td>
<td>289</td>
<td>33202.6</td>
<td>9.72</td>
</tr>
<tr>
<td>174</td>
<td>Ipomoea</td>
<td>445</td>
<td>49217.7</td>
<td>5.93</td>
</tr>
<tr>
<td>175</td>
<td>Ipomoea</td>
<td>450</td>
<td>49109.4</td>
<td>8.6</td>
</tr>
<tr>
<td>176</td>
<td>Ipomoea</td>
<td>256</td>
<td>28236.7</td>
<td>8.85</td>
</tr>
<tr>
<td>177</td>
<td>Ipomoea</td>
<td>139</td>
<td>15812.3</td>
<td>7.47</td>
</tr>
<tr>
<td>178</td>
<td>Ipomoea</td>
<td>339</td>
<td>36362.5</td>
<td>7.65</td>
</tr>
<tr>
<td>179</td>
<td>Ipomoea</td>
<td>140</td>
<td>15813.5</td>
<td>9.46</td>
</tr>
<tr>
<td>180</td>
<td>Ipomoea</td>
<td>540</td>
<td>59083.1</td>
<td>8.62</td>
</tr>
<tr>
<td>181</td>
<td>Ipomoea</td>
<td>342</td>
<td>37394.3</td>
<td>8.98</td>
</tr>
<tr>
<td>182</td>
<td>Ipomoea</td>
<td>443</td>
<td>48381.8</td>
<td>8.38</td>
</tr>
<tr>
<td>183</td>
<td>Ipomoea</td>
<td>1002</td>
<td>111088</td>
<td>7.68</td>
</tr>
<tr>
<td>184</td>
<td>Ipomoea</td>
<td>535</td>
<td>59619.7</td>
<td>6.11</td>
</tr>
<tr>
<td>185</td>
<td>Ipomoea</td>
<td>493</td>
<td>55006.7</td>
<td>6.03</td>
</tr>
<tr>
<td>186</td>
<td>Ipomoea</td>
<td>380</td>
<td>40123.2</td>
<td>8.73</td>
</tr>
<tr>
<td>187</td>
<td>Ipomoea</td>
<td>1060</td>
<td>117205.1</td>
<td>7.98</td>
</tr>
<tr>
<td>188</td>
<td>Ipomoea</td>
<td>151</td>
<td>16239.2</td>
<td>9.97</td>
</tr>
<tr>
<td>189</td>
<td>Ipomoea</td>
<td>184</td>
<td>20468.8</td>
<td>10.17</td>
</tr>
<tr>
<td>190</td>
<td>Ipomoea</td>
<td>460</td>
<td>50446.7</td>
<td>8.61</td>
</tr>
<tr>
<td>191</td>
<td>Ipomoea</td>
<td>460</td>
<td>51531</td>
<td>5.82</td>
</tr>
<tr>
<td>192</td>
<td>Ipomoea</td>
<td>287</td>
<td>32309</td>
<td>7.95</td>
</tr>
<tr>
<td>193</td>
<td>Ipomoea</td>
<td>358</td>
<td>40922.4</td>
<td>6.62</td>
</tr>
<tr>
<td>194</td>
<td>Ipomoea</td>
<td>289</td>
<td>32614.7</td>
<td>8.83</td>
</tr>
<tr>
<td>195</td>
<td>Ipomoea</td>
<td>343</td>
<td>36709.8</td>
<td>7.69</td>
</tr>
<tr>
<td>196</td>
<td>Ipomoea</td>
<td>103</td>
<td>12010.4</td>
<td>9.24</td>
</tr>
<tr>
<td>197</td>
<td>Ipomoea</td>
<td>527</td>
<td>57790.8</td>
<td>8.87</td>
</tr>
<tr>
<td>198</td>
<td>Ipomoea</td>
<td>371</td>
<td>39136.1</td>
<td>8.66</td>
</tr>
<tr>
<td>199</td>
<td>Ipomoea</td>
<td>1057</td>
<td>116589.4</td>
<td>7.77</td>
</tr>
<tr>
<td>200</td>
<td>Ipomoea</td>
<td>465</td>
<td>51098.1</td>
<td>8.41</td>
</tr>
<tr>
<td>201</td>
<td>Ipomoea</td>
<td>301</td>
<td>33835.1</td>
<td>8.68</td>
</tr>
<tr>
<td>202</td>
<td>Ipomoea</td>
<td>189</td>
<td>21206.4</td>
<td>10.55</td>
</tr>
<tr>
<td>203</td>
<td>Ipomoea</td>
<td>1022</td>
<td>113412.4</td>
<td>5.82</td>
</tr>
<tr>
<td>204</td>
<td>Ipomoea</td>
<td>309</td>
<td>34090.6</td>
<td>8.94</td>
</tr>
<tr>
<td>205</td>
<td>Ipomoea</td>
<td>404</td>
<td>44384.2</td>
<td>8.73</td>
</tr>
<tr>
<td>206</td>
<td>Ipomoea</td>
<td>993</td>
<td>110136.7</td>
<td>7.23</td>
</tr>
<tr>
<td>207</td>
<td>Ipomoea</td>
<td>512</td>
<td>57476</td>
<td>5.87</td>
</tr>
<tr>
<td>208</td>
<td>Ipomoea</td>
<td>1077</td>
<td>119855.8</td>
<td>7.34</td>
</tr>
<tr>
<td>209</td>
<td>Ipomoea</td>
<td>795</td>
<td>88747</td>
<td>5.84</td>
</tr>
<tr>
<td>210</td>
<td>Ipomoea</td>
<td>449</td>
<td>49071.2</td>
<td>7.89</td>
</tr>
</tbody></table>
<p>随后使用R进行可视化，保存为<strong><em>Figure2b</em></strong>,<strong><em>Figure2c</em></strong>,<strong><em>Figure2d</em></strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br></pre></td><td class="code"><pre><span class="line">setwd(<span class="string">"/Users/sunhaoyun/data/gene_family/plot/Figure1/"</span>)</span><br><span class="line"><span class="keyword">library</span>(ggplot2)</span><br><span class="line"><span class="keyword">library</span>(ggsci)</span><br><span class="line"><span class="comment"># 读入数据</span></span><br><span class="line">Data &lt;- read.csv(<span class="string">"boxplot2.txt"</span>, head=<span class="literal">T</span>,sep = <span class="string">'\t'</span>)</span><br><span class="line"><span class="comment"># 设置X轴标签顺序</span></span><br><span class="line">Data$X  &lt;- factor(Data$Accesion,levels = c(<span class="string">"I. nil"</span>,<span class="string">"I. triloba"</span>,<span class="string">'I. trifida'</span>,<span class="string">'I. batatas'</span>,<span class="string">'Ipomoea'</span>))</span><br><span class="line"><span class="comment"># ggplot匹配表格中的列</span></span><br><span class="line">ggplot(Data, aes(x=X, y=length,color=X)) + </span><br><span class="line"><span class="comment"># 设置y轴标题</span></span><br><span class="line">  labs(y=<span class="string">'Amino acid length'</span>,x=<span class="literal">NULL</span>,fill=<span class="literal">NULL</span>)+</span><br><span class="line"><span class="comment"># 绘制箱线图，边框为虚线</span></span><br><span class="line">  geom_boxplot(linetype=<span class="string">"longdash"</span>,color=<span class="string">'black'</span>)+</span><br><span class="line"><span class="comment"># 计算底和顶</span></span><br><span class="line">  stat_boxplot(aes(ymin=..lower..,ymax=..upper..,fill=`X`),color=<span class="string">"black"</span>)+</span><br><span class="line"><span class="comment"># 添加误差棒，为实心</span></span><br><span class="line">  stat_boxplot(geom = <span class="string">"errorbar"</span>,aes(ymin=..ymax..),</span><br><span class="line">               width=<span class="number">0.4</span>,color=<span class="string">"black"</span>)+</span><br><span class="line"><span class="comment"># 误差棒的尺寸</span></span><br><span class="line">  scale_size_manual(values=c(<span class="number">1</span>, <span class="number">3</span>))+</span><br><span class="line"><span class="comment"># 添加配色</span></span><br><span class="line">  scale_fill_aaas()+theme_classic()+</span><br><span class="line"><span class="comment"># 去除网格线</span></span><br><span class="line">  theme(panel.grid=element_blank())+</span><br><span class="line"><span class="comment"># 设置y轴坐标上限</span></span><br><span class="line">  ylim(<span class="number">0</span>,<span class="number">1200</span>)+</span><br><span class="line"><span class="comment"># 设置字体大小</span></span><br><span class="line">  theme(axis.title.y = element_text(size=<span class="number">24</span>),</span><br><span class="line">        axis.ticks.length=unit(<span class="number">.4</span>, <span class="string">"cm"</span>),</span><br><span class="line">    axis.text.y=element_text(size=<span class="number">18</span>,angle=<span class="number">90</span>,color = <span class="string">'black'</span>),</span><br><span class="line">    axis.text.x=element_text(size=<span class="number">24</span>, face = <span class="string">'italic'</span>,color = <span class="string">'black'</span>))+</span><br><span class="line"><span class="comment"># 去除图例</span></span><br><span class="line">    guides(fill=<span class="literal">F</span>)</span><br><span class="line"><span class="comment"># 下面的同上</span></span><br><span class="line">ggplot(Data, aes(x=X, y=MW,color=X)) + </span><br><span class="line">labs(y=<span class="string">'Molecular weight'</span>,x=<span class="literal">NULL</span>,fill=<span class="literal">NULL</span>)+</span><br><span class="line">  geom_boxplot(linetype=<span class="string">"longdash"</span>,color=<span class="string">'black'</span>)+</span><br><span class="line">  stat_boxplot(aes(ymin=..lower..,ymax=..upper..,fill=`X`),color=<span class="string">"black"</span>)+</span><br><span class="line">  stat_boxplot(geom = <span class="string">"errorbar"</span>,aes(ymin=..ymax..),</span><br><span class="line">               width=<span class="number">0.4</span>,color=<span class="string">"black"</span>)+</span><br><span class="line">  scale_size_manual(values=c(<span class="number">1</span>, <span class="number">3</span>))+</span><br><span class="line">  scale_fill_aaas()+theme_classic()+theme(panel.grid=element_blank())+</span><br><span class="line">  ylim(<span class="number">0</span>,<span class="number">145000</span>)+</span><br><span class="line">  theme(axis.title.y = element_text(size=<span class="number">24</span>),</span><br><span class="line">        axis.ticks.length=unit(<span class="number">.4</span>, <span class="string">"cm"</span>),</span><br><span class="line">        axis.text.y=element_text(size=<span class="number">18</span>,angle=<span class="number">90</span>,color = <span class="string">'black'</span>),</span><br><span class="line">        axis.text.x=element_text(size=<span class="number">24</span>, face = <span class="string">'italic'</span>,color = <span class="string">'black'</span>))+guides(fill=<span class="literal">F</span>)</span><br><span class="line"></span><br><span class="line">ggplot(Data, aes(x=X, y=pI,color=X)) + </span><br><span class="line">  labs(y=<span class="string">'Isoelectric point'</span>,x=<span class="literal">NULL</span>,fill=<span class="literal">NULL</span>)+</span><br><span class="line">  geom_boxplot(linetype=<span class="string">"longdash"</span>,color=<span class="string">'black'</span>)+</span><br><span class="line">  stat_boxplot(aes(ymin=..lower..,ymax=..upper..,fill=`X`),color=<span class="string">"black"</span>)+</span><br><span class="line">  stat_boxplot(geom = <span class="string">"errorbar"</span>,aes(ymin=..ymax..),</span><br><span class="line">               width=<span class="number">0.4</span>,color=<span class="string">"black"</span>)+</span><br><span class="line">  scale_size_manual(values=c(<span class="number">1</span>, <span class="number">3</span>))+</span><br><span class="line">  scale_fill_aaas()+theme_classic()+theme(panel.grid=element_blank())+</span><br><span class="line">  ylim(<span class="number">4</span>,<span class="number">12</span>)+</span><br><span class="line">  theme(axis.title.y = element_text(size=<span class="number">24</span>),</span><br><span class="line">        axis.ticks.length=unit(<span class="number">.4</span>, <span class="string">"cm"</span>),</span><br><span class="line">        axis.text.y=element_text(size=<span class="number">18</span>,angle=<span class="number">90</span>,color = <span class="string">'black'</span>),</span><br><span class="line">        axis.text.x=element_text(size=<span class="number">24</span>, face = <span class="string">'italic'</span>,color = <span class="string">'black'</span>))+guides(fill=<span class="literal">F</span>)</span><br><span class="line"></span><br><span class="line">d &lt;- read.csv(<span class="string">"bar2"</span>,sep =<span class="string">'\t'</span>,header = <span class="literal">T</span>)</span><br><span class="line">d$X  &lt;- factor(d$Accesion,levels = c(<span class="string">"I. nil"</span>,<span class="string">"I. triloba"</span>,<span class="string">'I. trifida'</span>,<span class="string">'I. batatas'</span>))</span><br><span class="line">ggplot(d,aes(X,cl,fill=mir))+</span><br><span class="line">  theme_bw()+</span><br><span class="line">  labs(x=<span class="literal">NULL</span>,y=<span class="literal">NULL</span>)+</span><br><span class="line">  geom_bar(stat=<span class="string">"identity"</span>,position=<span class="string">"fill"</span>,width = <span class="number">0.6</span>)+</span><br><span class="line">  ggtitle(<span class="string">""</span>)+</span><br><span class="line">  scale_fill_nejm()+</span><br><span class="line">  theme(axis.text.y=element_text(size=<span class="number">16</span>),axis.text.x=element_text(size = <span class="number">16</span>,face = <span class="string">"italic"</span>))+guides(color=<span class="literal">F</span>)</span><br></pre></td></tr></table></figure>

<p>使用BUSCA对所有的蛋白序列进行亚细胞定位的预测</p>
<blockquote>
</blockquote>
<p>####2.系统发育树  </p>
<blockquote>
<p>水稻</p>
<blockquote>
<p>query:<a href="http://rice.plantbiology.msu.edu/analyses_search_locus.shtml" target="_blank" rel="noopener">http://rice.plantbiology.msu.edu/analyses_search_locus.shtml</a></p>
</blockquote>
<blockquote>
<p><a href="ftp://ftp.ensemblgenomes.org/pub/plants/release-51/gff3/oryza_sativa/Oryza_sativa.IRGSP-1.0.51.chr.gff3.gz" target="_blank" rel="noopener">ftp://ftp.ensemblgenomes.org/pub/plants/release-51/gff3/oryza_sativa/Oryza_sativa.IRGSP-1.0.51.chr.gff3.gz</a></p>
<p><a href="ftp://ftp.ensemblgenomes.org/pub/plants/release-51/fasta/oryza_sativa/dna//Oryza_sativa.IRGSP-1.0.dna.toplevel.fa.gz" target="_blank" rel="noopener">ftp://ftp.ensemblgenomes.org/pub/plants/release-51/fasta/oryza_sativa/dna//Oryza_sativa.IRGSP-1.0.dna.toplevel.fa.gz</a></p>
</blockquote>
<p>拟南芥</p>
<blockquote>
<p>query:<a href="https://www.arabidopsis.org/tools/bulk/sequences/index.jsp" target="_blank" rel="noopener">https://www.arabidopsis.org/tools/bulk/sequences/index.jsp</a></p>
</blockquote>
<blockquote>
<p><a href="https://www.arabidopsis.org/download_files/Genes/TAIR10_genome_release/TAIR10_chromosome_files/TAIR10_chr_all.fas" target="_blank" rel="noopener">https://www.arabidopsis.org/download_files/Genes/TAIR10_genome_release/TAIR10_chromosome_files/TAIR10_chr_all.fas</a></p>
<p><a href="https://www.arabidopsis.org/download_files/Genes/TAIR10_genome_release/TAIR10_gff3/TAIR10_GFF3_genes.gff" target="_blank" rel="noopener">https://www.arabidopsis.org/download_files/Genes/TAIR10_genome_release/TAIR10_gff3/TAIR10_GFF3_genes.gff</a></p>
</blockquote>
<p>番茄</p>
<blockquote>
<p>query:<a href="https://solgenomics.net/search/locus" target="_blank" rel="noopener">https://solgenomics.net/search/locus</a></p>
</blockquote>
<blockquote>
<p><a href="ftp://ftp.sgn.cornell.edu/genomes/Solanum_lycopersicum//Heinz1706/annotation/ITAG2.4_release/ITAG2.4_gene_models.gff3.gz" target="_blank" rel="noopener">ftp://ftp.sgn.cornell.edu/genomes/Solanum_lycopersicum//Heinz1706/annotation/ITAG2.4_release/ITAG2.4_gene_models.gff3.gz</a></p>
<p><a href="ftp://ftp.sgn.cornell.edu/genomes/Solanum_lycopersicum//Heinz1706/annotation/ITAG2.4_release/ITAG2.4_proteins.fasta" target="_blank" rel="noopener">ftp://ftp.sgn.cornell.edu/genomes/Solanum_lycopersicum//Heinz1706/annotation/ITAG2.4_release/ITAG2.4_proteins.fasta</a></p>
<p><a href="ftp://ftp.sgn.cornell.edu/genomes/Solanum_lycopersicum//Heinz1706/annotation/ITAG2.4_release/ITAG2.4_cds.fasta" target="_blank" rel="noopener">ftp://ftp.sgn.cornell.edu/genomes/Solanum_lycopersicum//Heinz1706/annotation/ITAG2.4_release/ITAG2.4_cds.fasta</a></p>
</blockquote>
<p>莱茵衣藻</p>
<blockquote>
<p>query:<a href="https://www.ncbi.nlm.nih.gov/nuccore/XM_001694888.1" target="_blank" rel="noopener">https://www.ncbi.nlm.nih.gov/nuccore/XM_001694888.1</a></p>
</blockquote>
</blockquote>
<table>
<thead>
<tr>
<th>Gene  Name</th>
<th>Locus ID</th>
</tr>
</thead>
<tbody><tr>
<td>AtSPL1</td>
<td>At2g47070</td>
</tr>
<tr>
<td>AtSPL2</td>
<td>At5g43270</td>
</tr>
<tr>
<td>AtSPL3</td>
<td>At2g33810</td>
</tr>
<tr>
<td>AtSPL4</td>
<td>At1g53160</td>
</tr>
<tr>
<td>AtSPL5</td>
<td>At3g15270</td>
</tr>
<tr>
<td>AtSPL6</td>
<td>At1g69170</td>
</tr>
<tr>
<td>AtSPL7</td>
<td>At5g18830</td>
</tr>
<tr>
<td>AtSPL8</td>
<td>At1g02065</td>
</tr>
<tr>
<td>AtSPL9</td>
<td>At2g42200</td>
</tr>
<tr>
<td>AtSPL10</td>
<td>At1g27370</td>
</tr>
<tr>
<td>AtSPL11</td>
<td>At1g27360</td>
</tr>
<tr>
<td>AtSPL12</td>
<td>At3g60030</td>
</tr>
<tr>
<td>AtSPL13</td>
<td>At5g50570</td>
</tr>
<tr>
<td>AtSPL14</td>
<td>At1g20980</td>
</tr>
<tr>
<td>AtSPL15</td>
<td>At3g57920</td>
</tr>
<tr>
<td>AtSPL16</td>
<td>At1g76580</td>
</tr>
<tr>
<td>OsSPL1</td>
<td>LOC_Os01g18850</td>
</tr>
<tr>
<td>OsSPL2</td>
<td>LOC_Os01g69830</td>
</tr>
<tr>
<td>OsSPL3</td>
<td>LOC_Os02g04680</td>
</tr>
<tr>
<td>OsSPL4</td>
<td>LOC_Os02g07780</td>
</tr>
<tr>
<td>OsSPL5</td>
<td>LOC_Os02g08070</td>
</tr>
<tr>
<td>OsSPL6</td>
<td>LOC_Os03g61760</td>
</tr>
<tr>
<td>OsSPL7</td>
<td>LOC_Os04g46580</td>
</tr>
<tr>
<td>OsSPL8</td>
<td>LOC_Os04g56170</td>
</tr>
<tr>
<td>OsSPL9</td>
<td>LOC_Os05g33810</td>
</tr>
<tr>
<td>OsSPL10</td>
<td>LOC_Os06g44860</td>
</tr>
<tr>
<td>OsSPL11</td>
<td>LOC_Os06g45310</td>
</tr>
<tr>
<td>OsSPL12</td>
<td>LOC_Os06g49010</td>
</tr>
<tr>
<td>OsSPL13</td>
<td>LOC_Os07g32170</td>
</tr>
<tr>
<td>OsSPL14</td>
<td>LOC_Os08g39890</td>
</tr>
<tr>
<td>OsSPL15</td>
<td>LOC_Os08g40260</td>
</tr>
<tr>
<td>OsSPL16</td>
<td>LOC_Os08g41940</td>
</tr>
<tr>
<td>OsSPL17</td>
<td>LOC_Os09g31438</td>
</tr>
<tr>
<td>OsSPL18</td>
<td>LOC_Os09g32944</td>
</tr>
<tr>
<td>OsSPL19</td>
<td>LOC_Os11g30370</td>
</tr>
<tr>
<td>LeSPL-CNR</td>
<td>Solyc02g077920</td>
</tr>
<tr>
<td>SlySBP12a</td>
<td>Solyc01g068100</td>
</tr>
<tr>
<td>SlySBP12b</td>
<td>Solyc05g053240</td>
</tr>
<tr>
<td>SlySBP2</td>
<td>Solyc04g045560</td>
</tr>
<tr>
<td>SlySBP6c</td>
<td>Solyc12g038520</td>
</tr>
<tr>
<td>SlySBP10</td>
<td>Solyc05g015510</td>
</tr>
<tr>
<td>SlySBP13</td>
<td>Solyc05g015840</td>
</tr>
<tr>
<td>SlySBP6a</td>
<td>Solyc03g114850</td>
</tr>
<tr>
<td>SlySBP8a</td>
<td>Solyc10g018780</td>
</tr>
<tr>
<td>SlySBP8b</td>
<td>Solyc01g090730</td>
</tr>
<tr>
<td>SlySBP3</td>
<td>Solyc10g009080</td>
</tr>
<tr>
<td>SlySBP6b</td>
<td>Solyc05g012040</td>
</tr>
<tr>
<td>SlySBP15</td>
<td>Solyc10g078700</td>
</tr>
<tr>
<td>SlySBP4</td>
<td>Solyc07g053810</td>
</tr>
<tr>
<td>SlySBP7</td>
<td>Solyc01g080670</td>
</tr>
<tr>
<td>CrCRR1</td>
<td>CHLRE_09g390023v5</td>
</tr>
</tbody></table>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 合并八个物种的pep文件</span></span><br><span class="line">cat 4fanshu.spl_id.wid70.pep.fa os.spl.pep.fa at.spl.pep.fa sly.pep.fa cr.pep.fa &gt; cr.7merged.wid70.pep.fa</span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用mafft比对pep文件</span></span><br><span class="line">mafft cr.7merged.wid70.pep.fa &gt; cr.7merged.wid70.mafft.pep.fa</span><br><span class="line"><span class="meta">#</span><span class="bash">Gblocks cr.7merged.wid70.mafft.pep.fa -b1=79 -b3=20 -b5=All</span></span><br></pre></td></tr></table></figure>

<blockquote>
<p> MEGA</p>
<blockquote>
<p>File </p>
<blockquote>
<p>Open A File</p>
<blockquote>
<p> Align</p>
</blockquote>
</blockquote>
<p>Data</p>
<blockquote>
<p>Export alignment</p>
<p>Mega format</p>
</blockquote>
<p>Phylogeney</p>
<blockquote>
<p>Construct Neighbor-Joining Tree</p>
<p>yes</p>
<blockquote>
<p>Bootstrap method</p>
<p><em>P</em>-distance</p>
<p>Gamma Distributed (G)</p>
<p>Gamma Parameter 1.00</p>
<p>Pairwise deletion</p>
<p>Number of threads 16</p>
</blockquote>
<p>Reroot CrCRR1</p>
</blockquote>
<p>File</p>
<blockquote>
<p>Export current tree</p>
<p>Branch lengths ✅</p>
<p>bootsrap values ✅</p>
</blockquote>
</blockquote>
</blockquote>
<p>得到nwk格式的树文件</p>
<p><strong><em>cr.7merged.wid70.mafft.pep.fa.mega.nwk</em></strong></p>
<blockquote>
<p>将nwk格式的树上传至<a href="https://evolgenius.info/evolview-v2/#login" target="_blank" rel="noopener">Evolview</a></p>
<blockquote>
<p>设置Leaf Decoration</p>
</blockquote>
</blockquote>
<table>
<thead>
<tr>
<th>AtSPL1</th>
<th>circle,brown</th>
</tr>
</thead>
<tbody><tr>
<td>AtSPL2</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL3</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL4</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL5</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL6</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL7</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL8</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL9</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL10</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL11</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL12</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL13</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL14</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL15</td>
<td>circle,brown</td>
</tr>
<tr>
<td>AtSPL16</td>
<td>circle,brown</td>
</tr>
<tr>
<td>OsSPL1</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL2</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL3</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL4</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL5</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL6</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL7</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL8</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL9</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL10</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL11</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL12</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL13</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL14</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL15</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL16</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL17</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL18</td>
<td>circle,orange</td>
</tr>
<tr>
<td>OsSPL19</td>
<td>circle,orange</td>
</tr>
<tr>
<td>LeSPL-CNR</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP10</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP12a</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP12b</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP13</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP15</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP2</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP3</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP4</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP6a</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP6b</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP6c</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP7</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP8a</td>
<td>circle,pink</td>
</tr>
<tr>
<td>SlySBP8b</td>
<td>circle,pink</td>
</tr>
<tr>
<td>IbSPL1</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL2</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL3</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL4</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL5</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL6</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL7</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL8</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL9</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL10</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL11</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL12</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL13</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL14</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL15</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL16</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL17</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL18</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL19</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL20</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL21</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL22</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL23</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL24</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL25</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL26</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL27</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL28</td>
<td>circle,purple</td>
</tr>
<tr>
<td>IbSPL29</td>
<td>circle,purple</td>
</tr>
<tr>
<td>ItfSPL1</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL2</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL3</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL4</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL5</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL6</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL7</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL8</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL9</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL10</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL11</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL12</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL13</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL14</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL15</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL16</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL17</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL18</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL19</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL20</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL21</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL22</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL23</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL24</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL25</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL26</td>
<td>circle,green</td>
</tr>
<tr>
<td>ItfSPL27</td>
<td>circle,green</td>
</tr>
<tr>
<td>InSPL1</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL2</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL3</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL4</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL5</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL6</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL7</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL8</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL9</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL10</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL11</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL12</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL13</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL14</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL15</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL16</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL17</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL18</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL19</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL20</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL21</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL22</td>
<td>circle,blue</td>
</tr>
<tr>
<td>InSPL23</td>
<td>circle,blue</td>
</tr>
<tr>
<td>ItbSPL1</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL2</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL3</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL4</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL5</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL6</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL7</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL8</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL9</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL10</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL11</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL12</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL13</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL14</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL15</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL16</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL17</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL18</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL19</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL20</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL21</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL22</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL23</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL24</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL25</td>
<td>circle,red</td>
</tr>
<tr>
<td>ItbSPL26</td>
<td>circle,red</td>
</tr>
<tr>
<td>CrCRR1</td>
<td>circle,grey</td>
</tr>
</tbody></table>
<blockquote>
<blockquote>
<p>设置Charts</p>
</blockquote>
</blockquote>
<table>
<thead>
<tr>
<th>OsSPL9,IbSPL20</th>
<th>text=I,fontcolor=#BC3C29,linecolor=#BC3C29,linewidth=6,fontsize=36</th>
</tr>
</thead>
<tbody><tr>
<td>IbSPL11,OsSPL6</td>
<td>text=II,fontcolor=#E64B35,linecolor=#E64B35,linewidth=6,fontsize=36</td>
</tr>
<tr>
<td>ItbSPL7,OsSPL5</td>
<td>text=III,fontcolor=#E18727,linecolor=#E18727,linewidth=6,fontsize=36</td>
</tr>
<tr>
<td>ItbSPL6,AtSPL6</td>
<td>text=IV-a,fontcolor=#6B8E23,linecolor=#6B8E23,linewidth=6,fontsize=36</td>
</tr>
<tr>
<td>IbSPL24,IbSPL23</td>
<td>text=IV-b,fontcolor=#6B8E23,linecolor=#6B8E23,linewidth=6,fontsize=36</td>
</tr>
<tr>
<td>InSPL5,ItfSPL11</td>
<td>text=IV-c,fontcolor=#6B8E23,linecolor=#6B8E23,linewidth=6,fontsize=36</td>
</tr>
<tr>
<td>OsSPL11,IbSPL19</td>
<td>text=V,fontcolor=#6F99AD,linecolor=#6F99AD,linewidth=6,fontsize=36</td>
</tr>
<tr>
<td>ItbSPL5,ItfSPL16</td>
<td>text=VI,fontcolor=#4DBBD5,linecolor=#4DBBD5,linewidth=6,fontsize=36</td>
</tr>
<tr>
<td>OsSPL1,ItbSPL20</td>
<td>text=VII,fontcolor=#0072B5,linecolor=#0072B5,linewidth=6,fontsize=36</td>
</tr>
<tr>
<td>OsSPL7,ItfSPL25</td>
<td>text=VIII,fontcolor=#7876B1,linecolor=#7876B1,linewidth=6,fontsize=36</td>
</tr>
</tbody></table>
<blockquote>
<blockquote>
<p>设置bootstrap value style</p>
</blockquote>
</blockquote>
<table>
<thead>
<tr>
<th>0,50</th>
<th>show=0</th>
</tr>
</thead>
<tbody><tr>
<td>50,100</td>
<td>color=black,size=10</td>
</tr>
</tbody></table>
<p>得到 <strong><em>Figure2.pdf</em></strong></p>
<p>按照clade将spl基因分组后，补充clade信息</p>
<p><strong><em>subfamilies.txt</em></strong></p>
<table>
<thead>
<tr>
<th>SP</th>
<th>GP</th>
<th>NUM</th>
</tr>
</thead>
<tbody><tr>
<td>I. batatas</td>
<td>Clade I</td>
<td>1</td>
</tr>
<tr>
<td>I. batatas</td>
<td>Clade II</td>
<td>4</td>
</tr>
<tr>
<td>I. batatas</td>
<td>Clade III</td>
<td>2</td>
</tr>
<tr>
<td>I. batatas</td>
<td>Clade IV-a</td>
<td>6</td>
</tr>
<tr>
<td>I. batatas</td>
<td>Clade IV-b</td>
<td>3</td>
</tr>
<tr>
<td>I. batatas</td>
<td>Clade IV-c</td>
<td>1</td>
</tr>
<tr>
<td>I. batatas</td>
<td>Clade V</td>
<td>4</td>
</tr>
<tr>
<td>I. batatas</td>
<td>Clade VI</td>
<td>4</td>
</tr>
<tr>
<td>I. batatas</td>
<td>Clade VII</td>
<td>3</td>
</tr>
<tr>
<td>I. batatas</td>
<td>Clade VIII</td>
<td>1</td>
</tr>
<tr>
<td>I. trifida</td>
<td>Clade I</td>
<td>1</td>
</tr>
<tr>
<td>I. trifida</td>
<td>Clade II</td>
<td>4</td>
</tr>
<tr>
<td>I. trifida</td>
<td>Clade III</td>
<td>2</td>
</tr>
<tr>
<td>I. trifida</td>
<td>Clade IV-a</td>
<td>4</td>
</tr>
<tr>
<td>I. trifida</td>
<td>Clade IV-b</td>
<td>2</td>
</tr>
<tr>
<td>I. trifida</td>
<td>Clade IV-c</td>
<td>2</td>
</tr>
<tr>
<td>I. trifida</td>
<td>Clade V</td>
<td>3</td>
</tr>
<tr>
<td>I. trifida</td>
<td>Clade VI</td>
<td>5</td>
</tr>
<tr>
<td>I. trifida</td>
<td>Clade VII</td>
<td>2</td>
</tr>
<tr>
<td>I. trifida</td>
<td>Clade VIII</td>
<td>2</td>
</tr>
<tr>
<td>I. triloba</td>
<td>Clade I</td>
<td>1</td>
</tr>
<tr>
<td>I. triloba</td>
<td>Clade II</td>
<td>4</td>
</tr>
<tr>
<td>I. triloba</td>
<td>Clade III</td>
<td>2</td>
</tr>
<tr>
<td>I. triloba</td>
<td>Clade IV-a</td>
<td>4</td>
</tr>
<tr>
<td>I. triloba</td>
<td>Clade IV-b</td>
<td>2</td>
</tr>
<tr>
<td>I. triloba</td>
<td>Clade IV-c</td>
<td>1</td>
</tr>
<tr>
<td>I. triloba</td>
<td>Clade V</td>
<td>3</td>
</tr>
<tr>
<td>I. triloba</td>
<td>Clade VI</td>
<td>5</td>
</tr>
<tr>
<td>I. triloba</td>
<td>Clade VII</td>
<td>2</td>
</tr>
<tr>
<td>I. triloba</td>
<td>Clade VIII</td>
<td>2</td>
</tr>
<tr>
<td>I. nil</td>
<td>Clade I</td>
<td>1</td>
</tr>
<tr>
<td>I. nil</td>
<td>Clade II</td>
<td>4</td>
</tr>
<tr>
<td>I. nil</td>
<td>Clade III</td>
<td>2</td>
</tr>
<tr>
<td>I. nil</td>
<td>Clade IV-a</td>
<td>4</td>
</tr>
<tr>
<td>I. nil</td>
<td>Clade IV-b</td>
<td>2</td>
</tr>
<tr>
<td>I. nil</td>
<td>Clade IV-c</td>
<td>1</td>
</tr>
<tr>
<td>I. nil</td>
<td>Clade V</td>
<td>3</td>
</tr>
<tr>
<td>I. nil</td>
<td>Clade VI</td>
<td>3</td>
</tr>
<tr>
<td>I. nil</td>
<td>Clade VII</td>
<td>2</td>
</tr>
<tr>
<td>I. nil</td>
<td>Clade VIII</td>
<td>1</td>
</tr>
</tbody></table>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Figure 1a</span></span><br><span class="line"><span class="comment"># 读取上面的表格</span></span><br><span class="line">d &lt;- read.csv(<span class="string">"subfamilies.txt"</span>,sep =<span class="string">'\t'</span>,header = <span class="literal">T</span>)</span><br><span class="line"><span class="comment"># 对表格中的X轴的指定顺序，生成一个新列X</span></span><br><span class="line">d$X  &lt;- factor(d$SP,levels = c(<span class="string">'I. nil'</span>,<span class="string">'I. triloba'</span>,<span class="string">"I. trifida"</span>,<span class="string">"I. batatas"</span>))</span><br><span class="line"></span><br><span class="line"><span class="comment"># 绘图，aes 匹配列名</span></span><br><span class="line">p &lt;- ggplot(d, aes(x = X,y = NUM,fill = GP))+</span><br><span class="line">  <span class="comment">####position="stack"堆叠状</span></span><br><span class="line"><span class="comment"># 添加柱形图</span></span><br><span class="line">  geom_bar(stat =<span class="string">"identity"</span>,width = <span class="number">0.6</span>,position =<span class="string">"stack"</span>)+    </span><br><span class="line"><span class="comment"># 添加色谱</span></span><br><span class="line">  scale_fill_manual(values = c(<span class="string">"#BC3C29FF"</span>,<span class="comment">#1</span></span><br><span class="line">                               <span class="string">"#E64B35B2"</span>,<span class="comment">#2</span></span><br><span class="line">                               <span class="string">'#E18727B2'</span>,<span class="comment">#3</span></span><br><span class="line">                               <span class="string">'#EEE8AAFF'</span>,<span class="comment">#4</span></span><br><span class="line">                               <span class="string">'#6B8E23B2'</span>,<span class="comment">#5</span></span><br><span class="line">                               <span class="string">'#548B54FF'</span>,<span class="comment">#6</span></span><br><span class="line">                               <span class="string">'#6F99ADFF'</span>,<span class="comment">#7</span></span><br><span class="line">                               <span class="string">'#4DBBD5B2'</span>,<span class="comment">#8</span></span><br><span class="line">                               <span class="string">"#0072B5B2"</span>,<span class="comment">#9</span></span><br><span class="line">                               <span class="string">'#7876B1FF'</span>,<span class="comment">#10</span></span><br><span class="line">                               <span class="string">'white'</span>,</span><br><span class="line">                               <span class="string">'#FFDC91FF'</span>,</span><br><span class="line">                               <span class="string">'#3C5488B2'</span>,</span><br><span class="line">                               <span class="string">'#7E6148FF'</span>,</span><br><span class="line">                               <span class="string">'#91D1C2FF'</span>,</span><br><span class="line">                               <span class="string">'#3C5488FF'</span>))+        </span><br><span class="line"><span class="comment"># 去除XY轴标题</span></span><br><span class="line">  labs(x = <span class="string">""</span>,y = <span class="string">""</span>, title = <span class="string">""</span>)+  </span><br><span class="line"><span class="comment"># 选定主题</span></span><br><span class="line">  theme_classic()+</span><br><span class="line"><span class="comment"># 去除网格线</span></span><br><span class="line">theme(panel.grid=element_blank())+</span><br><span class="line"><span class="comment"># 去除图例</span></span><br><span class="line">  guides(fill = guide_legend(reverse = <span class="literal">F</span>))+   </span><br><span class="line"><span class="comment"># 限定y轴高度</span></span><br><span class="line">  ylim(<span class="number">0</span>,<span class="number">30</span>)+</span><br><span class="line"><span class="comment"># 添加标签，位置水平分开</span></span><br><span class="line">  geom_text(aes(label = NUM),size = <span class="number">4</span>, position = position_stack(vjust = <span class="number">0.5</span>))+</span><br><span class="line"><span class="comment"># 设置x字体大小，x轴y轴标签，刻度线</span></span><br><span class="line">  theme(<span class="comment">#plot.title = element_text(size = 12, vjust = 0.5, hjust = 0.5),  </span></span><br><span class="line">        legend.title = element_blank(),    </span><br><span class="line">        axis.ticks.length=unit(<span class="number">.4</span>, <span class="string">"cm"</span>),</span><br><span class="line">        legend.text = element_text(size = <span class="number">12</span>),     </span><br><span class="line">        legend.position = <span class="string">'right'</span>,</span><br><span class="line">        axis.text.y=element_text(size=<span class="number">18</span>,angle=<span class="number">90</span>,color = <span class="string">'black'</span>),</span><br><span class="line">        axis.text.x=element_text(size=<span class="number">24</span>,face = <span class="string">"italic"</span>)</span><br><span class="line">        <span class="comment">#legend.key.size=unit(0.8,'cm')</span></span><br><span class="line">        )</span><br></pre></td></tr></table></figure>

<p>得到 <strong><em>Figure1a</em></strong></p>
<h4 id="3-保守性预测"><a href="#3-保守性预测" class="headerlink" title="3.保守性预测"></a>3.保守性预测</h4><p>#####3.1 甘薯属物种进化树</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 与上述方法一致</span></span><br><span class="line">ln -s ../../1.identification/1.4_sbp_domain/4fanshu.spl_id.wid70.pep.fa .</span><br><span class="line">mafft 4fanshu.spl_id.wid70.pep.fa &gt; 4fanshu.spl_id.wid70.mafft.pep.fa</span><br></pre></td></tr></table></figure>

<p>使用Mega X导出格式为nwk的树文件：</p>
<p><strong><em>4fanshu.spl_id.wid70.mafft.pep.meg.nwk</em></strong></p>
<p>#####3.2 基因结构输入文件准备</p>
<p><strong><em>4.gffread.final.gff</em></strong></p>
<p>原始gff文件需要将transcript 行的起始替换为CDS的起始</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python ../0_scripts/make_cds200_gff.py 4.gffread.final.gff  &gt; manually.created.cds.gff</span><br></pre></td></tr></table></figure>

<p>0_scripts/make_cds200_gff.py</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#make_cds200_gff.py</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"><span class="comment"># 创建最小和最大的字典，保存gene对应的CDS的起始</span></span><br><span class="line">gmin = &#123;&#125;</span><br><span class="line">gmax = &#123;&#125;</span><br><span class="line">chromosome = &#123;&#125;</span><br><span class="line">strand = &#123;&#125;</span><br><span class="line">CDS = &#123;&#125;</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_gff:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_gff:</span><br><span class="line">        <span class="keyword">if</span> <span class="keyword">not</span> row.startswith(<span class="string">'#'</span>):</span><br><span class="line">            lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">            <span class="keyword">if</span> lst1[<span class="number">2</span>] == <span class="string">"transcript"</span> <span class="keyword">or</span> lst1[<span class="number">2</span>] == <span class="string">"mRNA"</span> :</span><br><span class="line">                chromosome[lst1[<span class="number">8</span>].replace(<span class="string">"ID="</span>,<span class="string">""</span>).strip()] = lst1[<span class="number">0</span>]</span><br><span class="line">                strand[lst1[<span class="number">8</span>].replace(<span class="string">"ID="</span>,<span class="string">""</span>).strip()] = lst1[<span class="number">6</span>]</span><br><span class="line">                gmin[lst1[<span class="number">8</span>].replace(<span class="string">"ID="</span>,<span class="string">""</span>).strip()] = <span class="number">9999999999</span></span><br><span class="line">                gmax[lst1[<span class="number">8</span>].replace(<span class="string">"ID="</span>,<span class="string">""</span>).strip()] = <span class="number">0</span></span><br><span class="line">                CDS[lst1[<span class="number">8</span>].replace(<span class="string">"ID="</span>,<span class="string">""</span>).strip()] = <span class="string">""</span></span><br><span class="line">            <span class="keyword">if</span> lst1[<span class="number">2</span>] == <span class="string">"CDS"</span> <span class="keyword">and</span> int(lst1[<span class="number">3</span>])&lt;int(gmin[lst1[<span class="number">8</span>].replace(<span class="string">"Parent="</span>,<span class="string">""</span>).strip()]):</span><br><span class="line">                gmin[lst1[<span class="number">8</span>].replace(<span class="string">"Parent="</span>,<span class="string">""</span>).strip()] = lst1[<span class="number">3</span>]</span><br><span class="line">            <span class="keyword">if</span> lst1[<span class="number">2</span>] == <span class="string">"CDS"</span>:</span><br><span class="line">                CDS[lst1[<span class="number">8</span>].replace(<span class="string">"Parent="</span>,<span class="string">""</span>).strip()] = CDS[lst1[<span class="number">8</span>].replace(<span class="string">"Parent="</span>,<span class="string">""</span>).strip()]+row</span><br><span class="line">            <span class="keyword">if</span> lst1[<span class="number">2</span>] == <span class="string">"CDS"</span> <span class="keyword">and</span> int(lst1[<span class="number">4</span>])&gt;int(gmax[lst1[<span class="number">8</span>].replace(<span class="string">"Parent="</span>,<span class="string">""</span>).strip()]):</span><br><span class="line">                gmax[lst1[<span class="number">8</span>].replace(<span class="string">"Parent="</span>,<span class="string">""</span>).strip()] = lst1[<span class="number">4</span>]</span><br><span class="line"><span class="keyword">for</span> m <span class="keyword">in</span> gmin.keys():</span><br><span class="line">    print(<span class="string">"&#123;&#125;\tTEST\ttranscript\t&#123;&#125;\t&#123;&#125;\t.\t&#123;&#125;\t.\tID=&#123;&#125;\n&#123;&#125;"</span>.format(chromosome[m],int(gmin[m]),int(gmax[m]),strand[m],m,CDS[m]).strip())</span><br></pre></td></tr></table></figure>

<p>使用TBtools</p>
<blockquote>
<p>TBtools</p>
<blockquote>
<p>Graphics</p>
<blockquote>
<p>BioSequence Structure Illustrator</p>
<blockquote>
<p>Gene Structure View (Advanced)</p>
<p>Set Newick Tree String</p>
<p>Set input .gff3 File</p>
<p>❌ Fill in Gradient Mode</p>
<p>Start</p>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
<p>生成 <strong><em>gene_structure.pdf</em></strong></p>
<p>#####3.3 结构域输入文件准备</p>
<p>整理之前SMART结构域预测结果，制作为bed格式：</p>
<p><strong><em>blast_domain.txt</em></strong></p>
<p>同时提取之前得到的蛋白长度，保存为：</p>
<p><strong><em>pep.length</em></strong></p>
<p>使用TBtools中的</p>
<blockquote>
<blockquote>
<p>Graphics</p>
<blockquote>
<p>BioSequence Structure Illustrator</p>
<blockquote>
<p>Gene Structure View (Advanced)</p>
<p>Set Input gene Len in Tab-delimited Format</p>
<p>Set Input ID list</p>
<p>Newic Tree String</p>
<p>❌ Fill in Gradient Mode</p>
<p>Start</p>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
<p>生成 <strong><em>blast_domain.pdf</em></strong></p>
<p>#####3.4 motif输入文件准备</p>
<p>将八个物种的蛋白序列上传到 <a href="https://meme-suite.org/meme/tools/meme" target="_blank" rel="noopener">MEME网站</a></p>
<blockquote>
<p>MEME</p>
<blockquote>
<p>Select the site distribution</p>
<blockquote>
<p>Zero or One Occurrence Per Sequence (zoops)</p>
</blockquote>
<p>Select the number of motifs</p>
<blockquote>
<p>10</p>
</blockquote>
<p>How wide can motifs be</p>
<blockquote>
<p>Minimum width: 6</p>
<p>Maximum width: 100</p>
</blockquote>
<p>Start Search</p>
</blockquote>
</blockquote>
<p>等待30-1小时左右的时间，下载结果中的</p>
<p><strong><em>meme.txt</em></strong></p>
<p>以及</p>
<p><strong><em>meme.xml</em></strong></p>
<p>对于<strong><em>meme.xml</em></strong> 使用TBtools绘制各基因上的motif组成</p>
<blockquote>
<blockquote>
<p>Graphics</p>
<blockquote>
<p>BioSequence Structure Illustrator</p>
<blockquote>
<p>Gene Structure View (Advanced)</p>
<p>Set Input MEME.xml</p>
<p>Set Input ID list</p>
<p>Newic Tree String</p>
<p>❌ Fill in Gradient Mode</p>
<p>Start</p>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
<p>得到 <strong><em>motif.pdf</em></strong></p>
<p>从<strong><em>meme.txt</em></strong>中提取出motif1至motif10的比对序列，如motif1</p>
<table>
<thead>
<tr>
<th>ItbSPL10</th>
<th>(114)</th>
<th>TSAGSVEAWIGLKLGKRTYFE</th>
<th>1</th>
</tr>
</thead>
<tbody><tr>
<td>IbSPL1</td>
<td>(86)</td>
<td>TSAGSVEAWIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>ItfSPL10</td>
<td>(114)</td>
<td>TSAGSVEAWIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>InilSPL13</td>
<td>(114)</td>
<td>TSAGSVEAWIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>ItbSPL14</td>
<td>(107)</td>
<td>TSIGSVEPFIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>IbSPL18</td>
<td>(111)</td>
<td>TSIGSVEPFIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>IbSPL19</td>
<td>(222)</td>
<td>TSIGSVEPFIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>ItfSPL15</td>
<td>(111)</td>
<td>TSIGSVEPFIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>InilSPL23</td>
<td>(111)</td>
<td>TSIGSVEPFIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>IbSPL9</td>
<td>(114)</td>
<td>TLAGSVEPLIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>ItfSPL21</td>
<td>(118)</td>
<td>TLAGSVEPLIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>InilSPL18</td>
<td>(76)</td>
<td>TLVGSVEPLIGLKLGKRTYFE</td>
<td>1</td>
</tr>
<tr>
<td>ItbSPL21</td>
<td>(121)</td>
<td>NLVGSVEPLIGLKLGKRTYFE</td>
<td>1</td>
</tr>
</tbody></table>
<p>然后对这个数据生成fasta文件：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">for i in &#123;1..10&#125;;do echo " python parse_motif.py motif"$&#123;i&#125;" &gt; for_weblogo_motif$&#123;i&#125;" ; done | sh</span><br></pre></td></tr></table></figure>

<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#parse_motif.py</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"><span class="comment"># 只保留I开头的即Ipomoea的物种的motif比对结果</span></span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_motif:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_motif:</span><br><span class="line">        <span class="keyword">if</span> row.startswith(<span class="string">'I'</span>):</span><br><span class="line">            lst1 = row.split(<span class="string">' '</span>)</span><br><span class="line">            print(<span class="string">"&gt;"</span>+lst1[<span class="number">0</span>]+<span class="string">"\n"</span>+lst1[<span class="number">-4</span>])</span><br></pre></td></tr></table></figure>

<p>结果生成了<strong><em>for_weblogo_motif1</em></strong>-<strong><em>motif10</em></strong>的fasta序列，然后在R里面绘图</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(ggseqlogo)</span><br><span class="line"><span class="keyword">library</span>(ggplot2)</span><br><span class="line"><span class="keyword">library</span>(seqinr)</span><br><span class="line">setwd(<span class="string">"~/data/gene_family/plot/Figure3/7sp/"</span>)</span><br><span class="line"></span><br><span class="line">id_list &lt;- c(<span class="string">'1'</span>,<span class="string">'2'</span>,<span class="string">'3'</span>,<span class="string">'4'</span>,<span class="string">'5'</span>,<span class="string">'6'</span>,<span class="string">'7'</span>,<span class="string">'8'</span>,<span class="string">'9'</span>,<span class="string">'10'</span>)</span><br><span class="line"><span class="keyword">for</span> (n <span class="keyword">in</span> id_list) &#123;</span><br><span class="line">in_file =  paste(<span class="string">'for_weblogo_motif'</span>,n,sep=<span class="string">""</span>)</span><br><span class="line"><span class="comment">### 读取  for_weblogo_motif1 - motif10</span></span><br><span class="line"></span><br><span class="line">motif &lt;- read.fasta(in_file)</span><br><span class="line"> </span><br><span class="line">a1_string &lt;- vector(mode = <span class="string">'character'</span>)</span><br><span class="line"><span class="keyword">for</span> (i <span class="keyword">in</span> <span class="number">1</span>:length((motif)))&#123;</span><br><span class="line">  a1_string[i] &lt;- toupper(c2s(motif[[i]]))</span><br><span class="line">&#125;</span><br><span class="line">num &lt;- paste(<span class="string">'motif'</span>,n,sep=<span class="string">""</span>)</span><br><span class="line">motif &lt;- ggseqlogo(a1_string)+</span><br><span class="line">  guides(fill=<span class="literal">F</span>)+</span><br><span class="line">  labs(y=num,x=<span class="literal">NULL</span>,fill=<span class="literal">NULL</span>)+</span><br><span class="line">  theme(axis.line.y=element_line(linetype=<span class="number">1</span>,color=<span class="string">"black"</span>,size=<span class="number">0.5</span>))+</span><br><span class="line">  theme(axis.line.x=element_line(linetype=<span class="number">1</span>,color=<span class="string">"black"</span>,size=<span class="number">0.5</span>))+</span><br><span class="line">  theme(axis.ticks.x=element_line(color=<span class="string">"black"</span>,size=<span class="number">0.5</span>,lineend = <span class="number">10</span>))+</span><br><span class="line">  theme(axis.ticks.y=element_line(color=<span class="string">"black"</span>,size=<span class="number">0.5</span>,lineend = <span class="number">10</span>))</span><br><span class="line">p &lt;- paste(<span class="string">'motif'</span>,n,<span class="string">".pdf"</span>,sep=<span class="string">""</span>)</span><br><span class="line">ggsave(p,motif, width = <span class="number">12</span>, height = <span class="number">4</span>)</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

<p>生成了<strong><em>motif1-motif10.pdf</em></strong>，再拼接出来得到 <strong><em>FigureS4.pdf</em></strong></p>
<p>再将 3.2，3.3，3.4用TBtools得到的 <strong><em>gene_structure.pdf</em></strong>, <strong><em>blast_domain.pdf</em></strong> 和 <strong><em>motif.pdf</em></strong> 拼接，调整配色和位置得到 <strong><em>Figure3.pdf</em></strong></p>
<h4 id="4-基因扩张和选择压力"><a href="#4-基因扩张和选择压力" class="headerlink" title="4. 基因扩张和选择压力"></a>4. 基因扩张和选择压力</h4><h5 id="4-1-直系同源基因的鉴定"><a href="#4-1-直系同源基因的鉴定" class="headerlink" title="4.1 直系同源基因的鉴定"></a>4.1 直系同源基因的鉴定</h5><blockquote>
<p>参考 <a href="https://darencard.net/blog/2018-01-12-orthomcl-tutorial/" target="_blank" rel="noopener">https://darencard.net/blog/2018-01-12-orthomcl-tutorial/</a></p>
</blockquote>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 主要用到orthomcl-pipeline-master</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> orthomcl还会用到mysql，mysql比较容易报错</span></span><br></pre></td></tr></table></figure>

<p>使用MySQL的时候可能会遇到的问题。</p>
<ol>
<li><p>由于Mysql默认要求复杂密码 当出现特殊字符时候 会导致orthomcl运行出错，所以可以先取消复杂密码设置</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">SHOW VARIABLES LIKE &apos;validate_password%&apos;;</span><br><span class="line">+--------------------------------------+-------+  </span><br><span class="line">| Variable_name                        | Value |  </span><br><span class="line">+--------------------------------------+-------+  </span><br><span class="line">| validate_password_dictionary_file    |       |  </span><br><span class="line">| validate_password_length             | 8     |  </span><br><span class="line">| validate_password_mixed_case_count   | 1     |  </span><br><span class="line">| validate_password_number_count       | 3     |  </span><br><span class="line">| validate_password_policy             | LOW   |  </span><br><span class="line">| validate_password_special_char_count | 1     |  </span><br><span class="line">+--------------------------------------+-------+  </span><br><span class="line">set global validate_password_special_char_count=0;</span><br></pre></td></tr></table></figure>



</li>
</ol>
<ol start="2">
<li><p>在设置MySQL时候由于数据结构类型，可能导致超出缓存大小，有两种解决方法</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br></pre></td><td class="code"><pre><span class="line"># 报错信息如下:</span><br><span class="line"># DBD::mysql::st execute failed: The total number of locks exceeds the lock table size at /home/RNAseq/miniconda/bin/orthomclPairs line 709, &lt;F&gt; line 12.</span><br><span class="line">show variables like &quot;%_buffer%&quot;; #查看缓存</span><br><span class="line">+-------------------------------------+----------------+</span><br><span class="line">| Variable_name                       | Value          |</span><br><span class="line">+-------------------------------------+----------------+</span><br><span class="line">| bulk_insert_buffer_size             | 8388608        |</span><br><span class="line">| innodb_buffer_pool_chunk_size       | 134217728      |</span><br><span class="line">| innodb_buffer_pool_dump_at_shutdown | ON             |</span><br><span class="line">| innodb_buffer_pool_dump_now         | OFF            |</span><br><span class="line">| innodb_buffer_pool_dump_pct         | 25             |</span><br><span class="line">| innodb_buffer_pool_filename         | ib_buffer_pool |</span><br><span class="line">| innodb_buffer_pool_in_core_file     | ON             |</span><br><span class="line">| innodb_buffer_pool_instances        | 1              |</span><br><span class="line">| innodb_buffer_pool_load_abort       | OFF            |</span><br><span class="line">| innodb_buffer_pool_load_at_startup  | ON             |</span><br><span class="line">| innodb_buffer_pool_load_now         | OFF            |</span><br><span class="line">| innodb_buffer_pool_size             | 134217728      |</span><br><span class="line">| innodb_change_buffer_max_size       | 25             |</span><br><span class="line">| innodb_change_buffering             | all            |</span><br><span class="line">| innodb_log_buffer_size              | 16777216       |</span><br><span class="line">| innodb_sort_buffer_size             | 1048576        |</span><br><span class="line">| join_buffer_size                    | 262144         |</span><br><span class="line">| key_buffer_size                     | 8388608        |</span><br><span class="line">| myisam_sort_buffer_size             | 8388608        |</span><br><span class="line">| net_buffer_length                   | 16384          |</span><br><span class="line">| preload_buffer_size                 | 32768          |</span><br><span class="line">| read_buffer_size                    | 131072         |</span><br><span class="line">| read_rnd_buffer_size                | 262144         |</span><br><span class="line">| sort_buffer_size                    | 262144         |</span><br><span class="line">| sql_buffer_result                   | OFF            |</span><br><span class="line">+-------------------------------------+----------------+</span><br><span class="line"># 默认innodb_buffer_pool_size太小了</span><br><span class="line">SET GLOBAL innodb_buffer_pool_size=64*1024*1024*1024;</span><br><span class="line">#如果不想更改缓存的话 把储存模式改为myisam</span><br><span class="line"># vim /etc/mysql/my.cnf</span><br><span class="line"># default-storage-engine  = myisam</span><br></pre></td></tr></table></figure>



</li>
</ol>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"># 创建账户orthomcl密码orthomcl</span><br><span class="line">CREATE USER &apos;orthomcl&apos;@&apos;localhost&apos; IDENTIFIED BY &apos;orthomcl&apos;;</span><br><span class="line">SELECT TABLE_NAME, ENGINE FROM information_schema.TABLES where TABLE_SCHEMA = &apos;database&apos;;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># 授权</span><br><span class="line">GRANT ALL PRIVILEGES ON orthomcl.* TO &apos;orthomcl&apos;@&apos;localhost&apos; IDENTIFIED BY &apos;orthomcl&apos;;</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 生成config文件，里面配置了mysql</span></span><br><span class="line">orthomcl-setup-database.pl --user orthomcl --password password --host localhost --database orthomcl --outfile orthomcl.conf</span><br><span class="line"><span class="meta">#</span><span class="bash"> 设置输入文件格式</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 首先准备At Os SLy In Itb Itf Ib 的最长转录本的蛋白文件</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> longest.at.pep.fa      longest.nil.pep.fa    longest.trifida.pep.fa</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> longest.batatas.pep.fa longest.os.pep.fa     longest.sly.pep.fa     </span></span><br><span class="line"><span class="meta">#</span><span class="bash"> longest.triloba.pep.fa</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 批量重命名一下</span></span><br><span class="line">rename 's/longest.//g' *</span><br><span class="line">rename 's/.pep.fa/.fasta/g' *</span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用seqkit和bioawk 整理输入文件的格式</span></span><br><span class="line">for i in &#123;At,Ib,In,Itb,Itf,Os,Sly&#125;</span><br><span class="line">do</span><br><span class="line">echo "seqkit seq -m 50 "$i".fasta | bioawk -c fastx '&#123; print \"&gt;"$i"|\"\$name; print \$seq &#125;' "$i".fasta &gt; "$i.rename.fasta</span><br><span class="line">done | sh</span><br><span class="line"><span class="meta">#</span><span class="bash"> 运行orthomcl</span></span><br><span class="line">orthomcl-pipeline --no-cleanup --compliant -s 10 -m orthomcl.conf -i input -o output</span><br><span class="line"><span class="meta">#</span><span class="bash"> 生成到 output 结果文件夹中  </span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用 parse_orthomcl.py 可以看每个SPL基因位于哪个组</span></span><br><span class="line">python3  ../0_scripts/parse_groups.py spl.id output/6sp_output/groups/groups.txt</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 整理挑选出SPL基因对 6spl.s.result</span></span><br><span class="line">less spl.id | while read line</span><br><span class="line">do</span><br><span class="line"> grep -w $&#123;line&#125; output/6sp_output/groups/groups.txt</span><br><span class="line">done | sort | uniq &gt; spl.og.search.result</span><br></pre></td></tr></table></figure>

<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># parse_groups.py</span></span><br><span class="line"><span class="comment"># parsed Orthomcl result and get speciefic gene id seq</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line">gid = []</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_gene_id:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_gene_id:</span><br><span class="line">        gid.append(row.strip())</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">2</span>]) <span class="keyword">as</span> input_group_all:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_group_all:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        <span class="keyword">for</span> i <span class="keyword">in</span> range(len(lst1)):</span><br><span class="line">            <span class="keyword">if</span> lst1[i] <span class="keyword">in</span> gid:</span><br><span class="line">                print(lst1[i],lst1[<span class="number">0</span>])</span><br></pre></td></tr></table></figure>

<p>整理出各物种的同源基因对为excel格式： <strong><em>OG.xls</em></strong></p>
<h5 id="4-2-甘薯属同源基因之间的KaKs"><a href="#4-2-甘薯属同源基因之间的KaKs" class="headerlink" title="4.2 甘薯属同源基因之间的KaKs"></a>4.2 甘薯属同源基因之间的KaKs</h5><p>计算出每对甘薯属OG之间的Ka和Ks,按照Clade划分，由于Group8缺少甘薯属外的sp</p>
<p>所以对其余的24对OG计算KaKs</p>
<p>首先生成 <strong><em>og.id</em></strong></p>
<p>即每行一个OG</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line">InSPL22 ItbSPL2 ItfSPL2 IbSPL20</span><br><span class="line">InSPL16 InSPL19 ItbSPL8 ItbSPL22        ItfSPL8 ItfSPL22        IbSPL4  IbSPL8</span><br><span class="line">InSPL12 InSPL21 ItbSPL1 ItbSPL26        ItfSPL1 ItfSPL26        IbSPL11 IbSPL21</span><br><span class="line">InSPL15 ItbSPL7 ItfSPL7 IbSPL5</span><br><span class="line">InSPL8  ItbSPL17        ItfSPL17        IbSPL13</span><br><span class="line">InSPL4  ItbSPL24        ItfSPL24        IbSPL6</span><br><span class="line">InSPL20 ItbSPL23        ItfSPL23        IbSPL7</span><br><span class="line">InSPL10 ItbSPL19        ItfSPL19        IbSPL16</span><br><span class="line">InSPL3  ItbSPL6 ItfSPL4 IbSPL26 IbSPL27</span><br><span class="line">InSPL6  ItbSPL13        ItfSPL12        IbSPL22</span><br><span class="line">ItbSPL12        ItfSPL14        IbSPL24</span><br><span class="line">InSPL5  ItbSPL11        ItfSPL11        ItfSPL13        IbSPL25</span><br><span class="line">InSPL13 ItbSPL10        ItfSPL10        IbSPL1</span><br><span class="line">InSPL18 ItbSPL21        ItfSPL21        IbSPL9</span><br><span class="line">InSPL23 ItbSPL14        ItfSPL15        IbSPL18 IbSPL19</span><br><span class="line">InSPL9  ItbSPL18        ItfSPL18        IbSPL14</span><br><span class="line">ItbSPL16        ItfSPL27        IbSPL17</span><br><span class="line">InSPL1  ItbSPL5 ItfSPL6 IbSPL28</span><br><span class="line">InSPL2  ItbSPL4 ItfSPL5 IbSPL29</span><br><span class="line">ItbSPL15        ItfSPL16</span><br><span class="line">InSPL17 ItbSPL20        ItfSPL20        IbSPL2  IbSPL10</span><br><span class="line">InSPL14 ItbSPL9 ItfSPL9 IbSPL3</span><br><span class="line">InSPL11 ItbSPL25        ItfSPL25        IbSPL12</span><br><span class="line">ItbSPL3 ItfSPL3</span><br><span class="line">===============================================</span><br><span class="line">python3 parse_og.py og.id</span><br></pre></td></tr></table></figure>

<p>随后对每一行中的任意两个不重复的基因对进行Ka和Ks的计算</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">for i in &#123;0..23&#125;; do python3 paml2.py $&#123;i&#125; &gt; $&#123;i&#125;.result; tail -n 1 $&#123;i&#125;.result; done</span><br></pre></td></tr></table></figure>

<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># parse_og.py</span></span><br><span class="line"><span class="comment"># 把每一行生成一个文件同时把\t 替换为\n</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line">i = <span class="number">0</span></span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_og:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_og:</span><br><span class="line">        <span class="keyword">with</span> open(str(i),<span class="string">'w'</span>) <span class="keyword">as</span> output_f:</span><br><span class="line">            output_f.write(row.strip().replace(<span class="string">'\t'</span>, <span class="string">'\n'</span>))</span><br><span class="line">            i+=<span class="number">1</span></span><br></pre></td></tr></table></figure>

<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># paml2.py</span></span><br><span class="line"><span class="comment"># 对 id 文件中任选两个id，提取cds和pep，</span></span><br><span class="line"><span class="comment"># </span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"><span class="keyword">import</span> itertools</span><br><span class="line"><span class="keyword">import</span> subprocess</span><br><span class="line">i = <span class="number">0</span></span><br><span class="line">sum_kaks = <span class="number">0</span></span><br><span class="line">sum_ks = <span class="number">0</span></span><br><span class="line">sum_ka = <span class="number">0</span></span><br><span class="line">id_name =[line.strip() <span class="keyword">for</span> line <span class="keyword">in</span> open(sys.argv[<span class="number">1</span>])]</span><br><span class="line"><span class="keyword">for</span> x <span class="keyword">in</span> itertools.combinations(id_name, <span class="number">2</span>):</span><br><span class="line">    <span class="keyword">with</span> open(<span class="string">'temp_tree'</span>,<span class="string">'w'</span>) <span class="keyword">as</span> temp_tree:</span><br><span class="line">        temp_tree.write(<span class="string">'('</span>+x[<span class="number">0</span>]+<span class="string">','</span>+x[<span class="number">1</span>]+<span class="string">')\n'</span>)</span><br><span class="line">    <span class="keyword">with</span> open(<span class="string">'temp_id'</span>,<span class="string">'w'</span>) <span class="keyword">as</span> temp_geneid:</span><br><span class="line">        temp_geneid.write(x[<span class="number">0</span>]+<span class="string">'\n'</span>+x[<span class="number">1</span>]+<span class="string">'\n'</span>)</span><br><span class="line">    extract_cmd1 = <span class="string">'python3 /Users/sunhaoyun/Scripts/get_seq.py temp_id 4fanshu.spl_id.wid70.pep.fa &gt; temp.pep'</span></span><br><span class="line">    extract_cmd2 = <span class="string">'python3 /Users/sunhaoyun/Scripts/get_seq.py temp_id 4fanshu.spl_id.wid70.cds.fa &gt; temp.cds'</span></span><br><span class="line">    subprocess.call(extract_cmd1, shell = <span class="literal">True</span>,stdout = subprocess.PIPE, stderr = subprocess.PIPE)</span><br><span class="line">    subprocess.call(extract_cmd2, shell = <span class="literal">True</span>,stdout = subprocess.PIPE, stderr = subprocess.PIPE)</span><br><span class="line">    mafft_cmd = <span class="string">'mafft --auto temp.pep &gt; temp.pep.ms'</span></span><br><span class="line">    subprocess.call(mafft_cmd, shell = <span class="literal">True</span>,stdout = subprocess.PIPE, stderr = subprocess.PIPE)</span><br><span class="line">    pal2nal_cmd = <span class="string">'perl /Users/sunhaoyun/Downloads/softwares/pal2nal.v14/pal2nal.pl temp.pep.ms temp.cds -output paml &gt; temp.pep.ms.pal2nal'</span></span><br><span class="line">    subprocess.call(pal2nal_cmd, shell = <span class="literal">True</span>,stdout = subprocess.PIPE, stderr = subprocess.PIPE)</span><br><span class="line">    paml_cmd = <span class="string">'codeml'</span></span><br><span class="line">    subprocess.call(paml_cmd, shell = <span class="literal">True</span>, stdout = subprocess.PIPE, stderr = subprocess.PIPE)</span><br><span class="line">    grep_cmd = <span class="string">'tail -n3 BM_mlc| head -n1'</span></span><br><span class="line">    kaks = subprocess.getoutput(grep_cmd)</span><br><span class="line">    rm_cmd = <span class="string">'rm *temp* BM_mlc'</span></span><br><span class="line">    subprocess.call(rm_cmd, shell = <span class="literal">True</span>, stdout = subprocess.PIPE, stderr = subprocess.PIPE)</span><br><span class="line">    lst1 = x[<span class="number">0</span>]+<span class="string">'\t'</span>+x[<span class="number">1</span>]+<span class="string">'\t'</span>+kaks.split(<span class="string">' '</span>)[<span class="number">-3</span>]+<span class="string">'\t'</span>+kaks.split(<span class="string">' '</span>)[<span class="number">-2</span>].replace(<span class="string">'('</span>, <span class="string">''</span>)+<span class="string">'\t'</span>+kaks.split(<span class="string">' '</span>)[<span class="number">-1</span>].replace(<span class="string">')'</span>, <span class="string">''</span>)</span><br><span class="line">    <span class="comment">### 排除掉没有非同义突变的，Ka包含-1的值</span></span><br><span class="line">    <span class="keyword">if</span> <span class="string">'-1'</span> <span class="keyword">not</span> <span class="keyword">in</span> lst1:</span><br><span class="line">        print(lst1)</span><br><span class="line">        sum_kaks += float(kaks.split(<span class="string">' '</span>)[<span class="number">-3</span>])</span><br><span class="line">        sum_ka += float(kaks.split(<span class="string">' '</span>)[<span class="number">-2</span>].replace(<span class="string">'('</span>, <span class="string">''</span>))</span><br><span class="line">        sum_ks += float(kaks.split(<span class="string">' '</span>)[<span class="number">-1</span>].replace(<span class="string">')'</span>, <span class="string">''</span>))</span><br><span class="line">        i += <span class="number">1</span></span><br><span class="line">print(<span class="string">'average\t\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;'</span>.format(round(sum_kaks/float(i),<span class="number">4</span>),round(sum_ka/float(i),<span class="number">4</span>),round(sum_ks/float(i),<span class="number">4</span>)))</span><br></pre></td></tr></table></figure>

<p>将得到的Ka/Ks Ka Ks 保存为 <strong><em>KaKs_OG.xlsx</em></strong></p>
<table>
<thead>
<tr>
<th>Group Number</th>
<th>Clade</th>
<th>I. nil</th>
<th>I. triloba</th>
<th>I. trifida</th>
<th>I. batatas</th>
<th>Ka</th>
<th>Ks</th>
<th>Ka/Ks</th>
</tr>
</thead>
<tbody><tr>
<td>Goup 1</td>
<td>I</td>
<td>InSPL22</td>
<td>ItbSPL2</td>
<td>ItfSPL2</td>
<td>IbSPL20</td>
<td>0.03</td>
<td>0.0613</td>
<td>0.4319</td>
</tr>
<tr>
<td>Goup 2</td>
<td>II</td>
<td>InSPL16/InSPL19</td>
<td>ItbSPL8/ItbSPL22</td>
<td>ItfSPL8/ItfSPL22</td>
<td>IbSPL4/IbSPL8</td>
<td>0.2191</td>
<td>1.3304</td>
<td>0.1828</td>
</tr>
<tr>
<td>Goup 3</td>
<td>II</td>
<td>InSPL12/InSPL21</td>
<td>ItbSPL1/ItbSPL26</td>
<td>ItfSPL1/ItfSPL26</td>
<td>IbSPL11/IbSPL21</td>
<td>0.1258</td>
<td>0.4166</td>
<td>0.3467</td>
</tr>
<tr>
<td>Goup 4</td>
<td>III</td>
<td>InSPL15</td>
<td>ItbSPL7</td>
<td>ItfSPL7</td>
<td>IbSPL5</td>
<td>0.0291</td>
<td>0.16</td>
<td>0.1823</td>
</tr>
<tr>
<td>Goup 5</td>
<td>III</td>
<td>InSPL8</td>
<td>ItbSPL17</td>
<td>ItfSPL17</td>
<td>IbSPL13</td>
<td>0.019</td>
<td>0.1296</td>
<td>0.1328</td>
</tr>
<tr>
<td>Goup 6</td>
<td>IV-a</td>
<td>InSPL4</td>
<td>ItbSPL24</td>
<td>ItfSPL24</td>
<td>IbSPL6</td>
<td>0.0615</td>
<td>0.1288</td>
<td>0.4436</td>
</tr>
<tr>
<td>Goup 7</td>
<td>IV-a</td>
<td>InSPL20</td>
<td>ItbSPL23</td>
<td>ItfSPL23</td>
<td>IbSPL7</td>
<td>0.0227</td>
<td>0.11</td>
<td>0.1871</td>
</tr>
<tr>
<td>Goup 8</td>
<td>IV-a</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>IbSPL15</td>
<td>-</td>
<td>-</td>
<td>-</td>
</tr>
<tr>
<td>Goup 9</td>
<td>IV-a</td>
<td>InSPL10</td>
<td>ItbSPL19</td>
<td>ItfSPL19</td>
<td>IbSPL16</td>
<td>0.0149</td>
<td>0.0741</td>
<td>0.2229</td>
</tr>
<tr>
<td>Goup 10</td>
<td>IV-a</td>
<td>InSPL3</td>
<td>ItbSPL6</td>
<td>ItfSPL4</td>
<td>IbSPL26/IbSPL27</td>
<td>0.0249</td>
<td>0.0777</td>
<td>0.3109</td>
</tr>
<tr>
<td>Goup 11</td>
<td>IV-b</td>
<td>InSPL6</td>
<td>ItbSPL13</td>
<td>ItfSPL12</td>
<td>IbSPL22</td>
<td>0.1084</td>
<td>0.2444</td>
<td>0.4557</td>
</tr>
<tr>
<td>Goup 12</td>
<td>IV-b</td>
<td>-</td>
<td>ItbSPL12</td>
<td>ItfSPL14</td>
<td>IbSPL24</td>
<td>0.0832</td>
<td>0.1617</td>
<td>0.5143</td>
</tr>
<tr>
<td>Goup 13</td>
<td>IV-c</td>
<td>InSPL5</td>
<td>ItbSPL11</td>
<td>ItfSPL11/ItfSPL13</td>
<td>IbSPL25</td>
<td>0.0496</td>
<td>0.0853</td>
<td>0.5593</td>
</tr>
<tr>
<td>Goup 14</td>
<td>V</td>
<td>InSPL13</td>
<td>ItbSPL10</td>
<td>ItfSPL10</td>
<td>IbSPL1</td>
<td>0.0204</td>
<td>0.1255</td>
<td>0.1729</td>
</tr>
<tr>
<td>Goup 15</td>
<td>V</td>
<td>InSPL18</td>
<td>ItbSPL21</td>
<td>ItfSPL21</td>
<td>IbSPL9</td>
<td>0.0268</td>
<td>0.1029</td>
<td>0.2426</td>
</tr>
<tr>
<td>Goup 16</td>
<td>V</td>
<td>InSPL23</td>
<td>ItbSPL14</td>
<td>ItfSPL15</td>
<td>IbSPL18/IbSPL19</td>
<td>0.0299</td>
<td>0.0774</td>
<td>0.388</td>
</tr>
<tr>
<td>Goup 17</td>
<td>VI</td>
<td>InSPL9</td>
<td>ItbSPL18</td>
<td>ItfSPL18</td>
<td>IbSPL14</td>
<td>0.0204</td>
<td>0.0803</td>
<td>0.2492</td>
</tr>
<tr>
<td>Goup 18</td>
<td>VI</td>
<td>-</td>
<td>ItbSPL16</td>
<td>ItfSPL27</td>
<td>IbSPL17</td>
<td>0.0031</td>
<td>0.0564</td>
<td>0.0546</td>
</tr>
<tr>
<td>Goup 19</td>
<td>VI</td>
<td>InSPL1</td>
<td>ItbSPL5</td>
<td>ItfSPL6</td>
<td>IbSPL28</td>
<td>0.0327</td>
<td>0.177</td>
<td>0.1767</td>
</tr>
<tr>
<td>Goup 20</td>
<td>VI</td>
<td>InSPL2</td>
<td>ItbSPL4</td>
<td>ItfSPL5</td>
<td>IbSPL29</td>
<td>0.0369</td>
<td>0.1324</td>
<td>0.2614</td>
</tr>
<tr>
<td>Goup 21</td>
<td>VI</td>
<td>-</td>
<td>ItbSPL15</td>
<td>ItfSPL16</td>
<td>-</td>
<td>0.0292</td>
<td>0.0422</td>
<td>0.692</td>
</tr>
<tr>
<td>Goup 22</td>
<td>VII</td>
<td>InSPL17</td>
<td>ItbSPL20</td>
<td>ItfSPL20</td>
<td>IbSPL2/IbSPL10</td>
<td>0.1286</td>
<td>0.4475</td>
<td>0.3071</td>
</tr>
<tr>
<td>Goup 23</td>
<td>VII</td>
<td>InSPL14</td>
<td>ItbSPL9</td>
<td>ItfSPL9</td>
<td>IbSPL3</td>
<td>0.0219</td>
<td>0.0927</td>
<td>0.1805</td>
</tr>
<tr>
<td>Goup 24</td>
<td>VIII</td>
<td>InSPL11</td>
<td>ItbSPL25</td>
<td>ItfSPL25</td>
<td>IbSPL12</td>
<td>0.0363</td>
<td>0.1218</td>
<td>0.2516</td>
</tr>
<tr>
<td>Goup 25</td>
<td>VIII</td>
<td>-</td>
<td>ItbSPL3</td>
<td>ItfSPL3</td>
<td>-</td>
<td>0.0013</td>
<td>0.0281</td>
<td>0.0457</td>
</tr>
</tbody></table>
<p>绘制每一个同源基因对的柱形图，首先生成每个Group和每个Clade的KaKs文件</p>
<p><strong><em>hist</em></strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">Rscript KaKs_plot.R</span><br></pre></td></tr></table></figure>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#KaKs_plot.R</span></span><br><span class="line"><span class="keyword">library</span>(<span class="string">'ggplot2'</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'reshape2'</span>)</span><br><span class="line"><span class="keyword">library</span>(ggsci)</span><br><span class="line"><span class="keyword">library</span>(ggsignif)</span><br><span class="line"><span class="keyword">library</span>(ggpubr)</span><br><span class="line">predictioncase &lt;- read.csv(<span class="string">"hist"</span>,header = <span class="literal">T</span>,sep = <span class="string">'\t'</span>)  <span class="comment"># 数据存储在csv</span></span><br><span class="line">setwd(<span class="string">"/Users/sunhaoyun/data/gene_family/plot/Figure4/Figure5"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">#qpredictioncase$X &lt;- factor(predictioncase$ID,levels = c('Goup 1',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 2',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 3',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 4',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 5',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 6',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 7',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 8',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 9',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 10',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 11',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 12',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 13',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 14',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 15',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 16',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 17',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 18',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 19',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 20',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 21',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 22',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 23',</span></span><br><span class="line"><span class="comment">#                                                           'Goup 24'))</span></span><br><span class="line">ggplot(predictioncase, aes(x=Clade,y=Ka.Ks,fill=Clade))+</span><br><span class="line">  geom_bar(stat = <span class="string">"summary"</span>, position = <span class="string">"dodge"</span>)+</span><br><span class="line">  scale_fill_manual(values = c(<span class="string">'#BC3C29'</span>,</span><br><span class="line">                    <span class="string">'#0072B5'</span>,</span><br><span class="line">                    <span class="string">'#E18727'</span>,</span><br><span class="line">                    <span class="string">'#20854E'</span>,</span><br><span class="line">                    <span class="string">'#20854E'</span>,</span><br><span class="line">                    <span class="string">'#20854E'</span>,</span><br><span class="line">                    <span class="string">'#7876B1'</span>,</span><br><span class="line">                    <span class="string">'#6F99AD'</span>,</span><br><span class="line">                    <span class="string">'#FFDC91'</span>,</span><br><span class="line">                    <span class="string">'#EE4C97'</span>))+</span><br><span class="line">  theme_classic()+</span><br><span class="line">  labs(x=<span class="literal">NULL</span>,y=<span class="string">'Ka/Ks'</span>)+</span><br><span class="line">  theme(axis.text.x = element_text(color=<span class="string">"black"</span>, size=<span class="number">12</span>,angle = <span class="number">45</span>, hjust = <span class="number">0.8</span>, vjust = <span class="number">0.8</span>))+</span><br><span class="line">  theme(axis.text.y = element_text(color=<span class="string">"black"</span>, size=<span class="number">12</span>))+</span><br><span class="line">  theme(legend.text = element_text(size = <span class="number">12</span>, face = <span class="string">"plain"</span>))+</span><br><span class="line">  theme(legend.title = element_text(size = <span class="number">14</span>, face = <span class="string">"plain"</span>))+</span><br><span class="line">  theme(axis.title.y = element_text(color = <span class="string">"black"</span>,size=<span class="number">20</span>))</span><br></pre></td></tr></table></figure>

<p>得到图 <strong><em>Figure4c.pdf</em></strong></p>
<p>生成batatas、trifida、triloba和nil的蛋白比对文件</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 将每个物种的全部基因与自己比对</span></span><br><span class="line">for i in &#123;batatas,trifida,triloba,nil&#125;</span><br><span class="line">do</span><br><span class="line">makeblastdb -in longest.$i.pep.fa -dbtype prot -parse_seqids </span><br><span class="line">blastp -query longest.$i.pep.fa -db longest.$i.pep.fa -outfmt 6 -evalue 1e-10 -num_alignments 5  -num_threads 6 -out $i.blast</span><br><span class="line">done</span><br><span class="line"><span class="meta">#</span><span class="bash"> 生成MCScanX可以识别的gff格式</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 首先用gffread整理gff文件的标签，去除一些额外信息，在提取出来新的gff</span></span><br><span class="line">ls *gff3 | while read i</span><br><span class="line">do</span><br><span class="line">gffread $i -o $i.reformat</span><br><span class="line">python3 parse_gff.py $i.reformat &gt; $i</span><br><span class="line">done</span><br><span class="line"><span class="meta">#</span><span class="bash"> 将名字重新命名，.gff与.blast应该一致</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 即 batatas.gff3 batatas.blast ......</span></span><br><span class="line">for i in &#123;batatas,trifida,triloba,nil&#125;</span><br><span class="line">do</span><br><span class="line">MCScanX $i</span><br><span class="line">perl ../downstream_analyses/detect_synteny_within_gene_families.pl -i $i.id -d $i.collinearity -o $i.family</span><br><span class="line">done</span><br><span class="line"><span class="meta">#</span><span class="bash"> 得到了所有基因的共线性文件，使用ID提取各个基因属于哪个类型的扩张事件</span></span><br></pre></td></tr></table></figure>

<p>得到 </p>
<p><strong><em>batatas.gene_type</em></strong></p>
<p><strong><em>triloba.gene_type</em></strong></p>
<p><strong><em>triloba.gene_type</em></strong></p>
<p><strong><em>nil.gene_type</em></strong></p>
<p>将里面的扩张事件信息添加至 <strong><em>Table1</em></strong></p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># parse_gff.py</span></span><br><span class="line"><span class="comment"># 以gffread调整过的gff文件为输入，输出MCScanX识别的gff文件</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_gff:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_gff:</span><br><span class="line">        <span class="keyword">if</span> <span class="keyword">not</span> row.startswith(<span class="string">'#'</span>):</span><br><span class="line">            lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">            <span class="keyword">if</span> <span class="string">'mRNA'</span> <span class="keyword">in</span> lst1[<span class="number">2</span>] <span class="keyword">or</span> <span class="string">'transcript'</span> <span class="keyword">in</span> lst1[<span class="number">2</span>]:</span><br><span class="line">                print(lst1[<span class="number">0</span>],lst1[<span class="number">8</span>].split(<span class="string">';'</span>)[<span class="number">0</span>].lstrip(<span class="string">'ID='</span>).strip(),lst1[<span class="number">3</span>],lst1[<span class="number">4</span>],sep=<span class="string">'\t'</span>)</span><br></pre></td></tr></table></figure>

<p>提取SPL基因的扩张事件，并保存为 <strong><em>dup_event.txt</em></strong></p>
<p>同时保存柱状图所需要的文件</p>
<p><strong><em>for_hsit.txt</em></strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># MC_num.R</span></span><br><span class="line">setwd(<span class="string">"~/data/gene_family/plot/Figure4"</span>)</span><br><span class="line"><span class="keyword">library</span>(ggplot2)</span><br><span class="line"><span class="keyword">library</span>(ggsci)</span><br><span class="line">setwd(<span class="string">"/Users/sunhaoyun/data/gene_family/plot/Figure4/"</span>)</span><br><span class="line">d &lt;- read.csv(<span class="string">"for_hsit.txt"</span>,sep =<span class="string">'\t'</span>,header = <span class="literal">T</span>)</span><br><span class="line">d$X  &lt;- factor(d$Sp,levels = c(<span class="string">"I. batatas"</span>,<span class="string">"I. trifida"</span>,<span class="string">'I. triloba'</span>,<span class="string">'I. nil'</span>))</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">p &lt;- ggplot(d, aes(x = X,y = Numver,fill = Type,label =Numver))+</span><br><span class="line">  <span class="comment">####position="stack"堆叠状</span></span><br><span class="line">  geom_bar(stat =<span class="string">"identity"</span>,width = <span class="number">0.6</span>,position =<span class="string">"stack"</span>)+</span><br><span class="line">  geom_text(size = <span class="number">6</span>, position = position_stack(vjust = <span class="number">0.5</span>))+</span><br><span class="line">  labs(x = <span class="string">""</span>,y = <span class="string">""</span>, title = <span class="string">""</span>)+</span><br><span class="line">  theme_classic()+theme(panel.grid=element_blank())+</span><br><span class="line">  guides(fill = guide_legend(reverse = <span class="literal">F</span>))+</span><br><span class="line">  ylim(<span class="number">0</span>,<span class="number">30</span>)+scale_fill_nejm()+</span><br><span class="line">  theme(plot.title = element_text(size = <span class="number">12</span>, vjust = <span class="number">0.5</span>, hjust = <span class="number">0.5</span>),</span><br><span class="line">        legend.title = element_blank(),</span><br><span class="line">        axis.ticks.length=unit(<span class="number">.4</span>, <span class="string">"cm"</span>),</span><br><span class="line">        legend.text = element_text(size = <span class="number">12</span>),</span><br><span class="line">        legend.position = <span class="string">'right'</span>,</span><br><span class="line">        axis.text.y=element_text(size=<span class="number">18</span>,angle=<span class="number">90</span>,color = <span class="string">'black'</span>),</span><br><span class="line">        axis.text.x=element_text(size=<span class="number">24</span>,face = <span class="string">"italic"</span>)</span><br><span class="line">        <span class="comment">#legend.key.size=unit(0.8,'cm')</span></span><br><span class="line">  )</span><br><span class="line">p</span><br></pre></td></tr></table></figure>

<p>保存为 <strong><em>Figure4b.pdf</em></strong></p>
<p>#####4.3 circos作图</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 首先grep出spl基因所在的block，然后把block中第一对和最后一对基因提出来</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 去除block里重复的区块，提取出的为一对一对的block，在使用脚本将circos需要的links的直接做出来<span class="keyword">for</span> x <span class="keyword">in</span> &#123;batatas,trifida,triloba,nil&#125;</span></span><br><span class="line">do</span><br><span class="line">cat $x.id | while read line</span><br><span class="line">do</span><br><span class="line">grep -w $&#123;line&#125; $x.collinearity</span><br><span class="line">done | awk '&#123;print "^"$1&#125;' | sed 's/-//g'  | while read i</span><br><span class="line">do</span><br><span class="line">grep -w $i $x.collinearity | head -n1</span><br><span class="line">grep -w $i $x.collinearity| tail -n1</span><br><span class="line">done |awk '&#123;print $3,$4&#125;' &gt; $x.block</span><br><span class="line">python3 parse_mcx.py $i.block $i.gff &gt; $i.links</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># parse_mcx.py </span></span><br><span class="line"><span class="comment"># 主要就是读取blocks中一对一对的id，提取位置信息，第一行和第二行为一对，最终写入links写在一行</span></span><br><span class="line"><span class="comment"># 运行完这个脚本之后就只需要在每一行后面添加颜色信息就可以放入circos文件夹了</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line">i = <span class="number">1</span></span><br><span class="line">left = []</span><br><span class="line">right = []</span><br><span class="line">l2 = []</span><br><span class="line">r2 = []</span><br><span class="line">zuo = &#123;&#125;</span><br><span class="line">you = &#123;&#125;</span><br><span class="line">ran = &#123;&#125;</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_coll:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_coll:</span><br><span class="line">        lst1 = row.split(<span class="string">' '</span>)</span><br><span class="line">        <span class="keyword">if</span> i%<span class="number">2</span> == <span class="number">0</span>:</span><br><span class="line">            left.append(lst1[<span class="number">0</span>])</span><br><span class="line">            right.append(lst1[<span class="number">1</span>].strip())</span><br><span class="line">            i += <span class="number">1</span></span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            l2.append(lst1[<span class="number">0</span>])</span><br><span class="line">            r2.append(lst1[<span class="number">1</span>].strip())</span><br><span class="line">            i += <span class="number">1</span></span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">2</span>]) <span class="keyword">as</span> input_gff:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_gff:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        zuo[lst1[<span class="number">1</span>]] = lst1[<span class="number">2</span>]</span><br><span class="line">        you[lst1[<span class="number">1</span>]] = lst1[<span class="number">3</span>].strip()</span><br><span class="line">        ran[lst1[<span class="number">1</span>]] = lst1[<span class="number">0</span>]</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> range(len(l2)) :</span><br><span class="line">    p1 = zuo[left[i]]</span><br><span class="line">    p2 = you[l2[i]]</span><br><span class="line">    p3 = zuo[right[i]]</span><br><span class="line">    p4 = you[r2[i]]</span><br><span class="line">    <span class="keyword">if</span> p2 &gt; p1 <span class="keyword">and</span> p4 &gt; p3:</span><br><span class="line">        print(<span class="string">"&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;"</span>.format(ran[left[i]],p2,p1,ran[right[i]],p4,p3))</span><br><span class="line">    <span class="keyword">elif</span> p2 &gt; p1 <span class="keyword">and</span> p4 &lt; p3:</span><br><span class="line">        print(<span class="string">"&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;"</span>.format(ran[left[i]],p2,p1,ran[right[i]],p3,p4,<span class="string">"twist=0,"</span>))</span><br><span class="line">    <span class="keyword">elif</span> p2 &lt; p1 <span class="keyword">and</span> p4 &gt; p3:</span><br><span class="line">        print(<span class="string">"&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;"</span>.format(ran[left[i]],p1,p2,ran[right[i]],p4,p3,<span class="string">"twist=0,"</span>))</span><br><span class="line">    <span class="keyword">else</span>:</span><br><span class="line">        print(<span class="string">"&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;\t&#123;&#125;"</span>.format(ran[left[i]],p2,p1,ran[right[i]],p4,p3))</span><br></pre></td></tr></table></figure>

<p>随后再提取所有SPL基因的位置，包括不在block中的</p>
<p>准备circos其余config文件，生成文件夹分别生成svg格式的文件，将做好的links文件还有染色体信息，和基因文件替换各个文件夹中的原始文件，最终得到 <strong><em>Figure4a.pdf</em></strong> 和 <strong><em>FigureS5.pdf</em></strong></p>
<p>####5. miR156的鉴定</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">ncRNA</span></span><br></pre></td></tr></table></figure>

<p>根据石墨笔记获取了成熟mRNA的序列</p>
<p>得到 <strong><em>Ipomoea_batatas.muture_miRNA.fa</em></strong></p>
<p>随后还需要提取出甘薯属SPL基因的cDNA</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 使用gffread</span></span><br><span class="line">for i in &#123;Ib,Itf,Itb,In&#125;</span><br><span class="line">do</span><br><span class="line">gffread $i.gff3 -g $i.fa -w $i.cdna</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>根据miRbase上的<strong><em>mature.fa</em></strong> blast识别出甘薯中的miR156序列</p>
<p><strong><em>9.identified.mir156.fa</em></strong>，并用于psRNAtarget预测。</p>
<p>将<strong><em>9.identified.mir156.fa</em></strong> 与拟南芥的miR156a 序列比对至一起，使用GenDoc绘制序列比对文件</p>
<p>生成</p>
<blockquote>
<p><a href="http://plantgrn.noble.org/psRNATarget/" target="_blank" rel="noopener">http://plantgrn.noble.org/psRNATarget/</a></p>
<blockquote>
<p><strong>Submit small RNAs and targets</strong></p>
<blockquote>
<p><strong>Upload sRNA file</strong></p>
<p><strong>Upload target file</strong></p>
<p><strong>Schema V2 (2017 release)</strong></p>
</blockquote>
</blockquote>
</blockquote>
<p>将blast到miR156的序列提取出，用作psRNAtarget的输入</p>
<p>得到预测结果 <strong><em>psrnatarget.txt</em></strong></p>
<p>使用如下脚本提取出miRNA的位置位于CDS或者是UTR区域</p>
<figure class="highlight python"><figcaption><span>parse_psRNATarget.py psrnatarget.txt```</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">```python</span><br><span class="line"><span class="comment"># parse_rnatarget.py</span></span><br><span class="line"><span class="comment"># fasta文件的head种包含了CDS区域</span></span><br><span class="line"><span class="comment"># 根据CDS区域</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line">i = <span class="number">0</span></span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_RNA_txt:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_RNA_txt:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        rna_name = lst1[<span class="number">0</span>]</span><br><span class="line">        gene_name = lst1[<span class="number">1</span>]</span><br><span class="line">        target_site_s = lst1[<span class="number">6</span>]</span><br><span class="line">        target_site_e = lst1[<span class="number">7</span>]</span><br><span class="line">        <span class="comment">#print(lst1[1])</span></span><br><span class="line">	    <span class="comment">#print(lst1[12].lstrip('CDS=').split('-'))</span></span><br><span class="line">        cds_region_s = lst1[<span class="number">12</span>].lstrip(<span class="string">'CDS='</span>).split(<span class="string">'-'</span>)[<span class="number">0</span>]</span><br><span class="line">        cds_region_e = lst1[<span class="number">12</span>].lstrip(<span class="string">'CDS='</span>).split(<span class="string">'-'</span>)[<span class="number">1</span>]</span><br><span class="line">        <span class="keyword">if</span> int(target_site_s) &gt; int(cds_region_e):</span><br><span class="line">            print(<span class="string">"&#123;:3&#125; &#123;:1&#125; is in the &#123;:2&#125; 3'UTR"</span>.format(i,rna_name, gene_name))</span><br><span class="line">            i += <span class="number">1</span></span><br><span class="line">        <span class="keyword">elif</span> int(target_site_s) &lt; int(cds_region_s):</span><br><span class="line">            print(<span class="string">"&#123;:3&#125; &#123;:1&#125; is in the &#123;:2&#125; 5'UTR"</span>.format(i,rna_name, gene_name))</span><br><span class="line">            i += <span class="number">1</span></span><br><span class="line">        <span class="keyword">elif</span> int(target_site_s) &gt; int(cds_region_s) <span class="keyword">and</span> int(target_site_e) &lt; int(cds_region_e):</span><br><span class="line">            print(<span class="string">"&#123;:3&#125; &#123;:1&#125; is in the &#123;:2&#125; cds_region"</span>.format(i,rna_name, gene_name))</span><br><span class="line">            i += <span class="number">1</span></span><br><span class="line">        <span class="comment">#if int(target_site_s) &lt; int(cds_region_e) and int(target_site_e) &gt; int(cds_region_e):</span></span><br><span class="line">            <span class="comment">#print("&#123;:1&#125; is in the &#123;:2&#125; cds_region".format(rna_name, gene_name))</span></span><br><span class="line">        <span class="comment">#elif int(target_site_s) &lt; int(cds_region_s) and int(target_site_e) &gt; int(cds_region_s):</span></span><br><span class="line">            <span class="comment">#print("&#123;:1&#125; is in the &#123;:2&#125; cds_region".format(rna_name, gene_name))</span></span><br></pre></td></tr></table></figure>

<p>生成  <strong><em>parse_result.txt</em></strong></p>
<p>然后整理一下其中的结果，生成绘图输入文件 <strong><em>psrnatarget.ipomoea.txt</em></strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">Data &lt;- read.csv(<span class="string">"psrnatarget.ipomoea.txt"</span>, head=<span class="literal">T</span>,sep = <span class="string">'\t'</span>)</span><br><span class="line">Data$X  &lt;- factor(Data$SP,levels = c(<span class="string">"I. nil"</span>,<span class="string">"I. triloba"</span>,<span class="string">'I. trifida'</span>,<span class="string">'I. batatas'</span>))</span><br><span class="line">ggplot(Data, aes(x = X,y = Count,fill = Type))+</span><br><span class="line">  <span class="comment">####position="stack"堆叠状</span></span><br><span class="line">  geom_bar(stat =<span class="string">"identity"</span>,width = <span class="number">0.6</span>,position=<span class="string">"stack"</span>,)+   </span><br><span class="line">  labs(x = <span class="string">""</span>,y = <span class="string">""</span>, title = <span class="string">""</span>)+  </span><br><span class="line">  theme_classic()+theme(panel.grid=element_blank())+</span><br><span class="line">  guides(fill = guide_legend(reverse = <span class="literal">F</span>))+   </span><br><span class="line">  scale_fill_nejm()+</span><br><span class="line">  geom_text(aes(label = Count),size = <span class="number">8</span>, position = position_stack(vjust = <span class="number">0.5</span>))+</span><br><span class="line">  theme(plot.title = element_text(size = <span class="number">12</span>, vjust = <span class="number">0.5</span>, hjust = <span class="number">0.5</span>),  </span><br><span class="line">        legend.title = element_blank(),    </span><br><span class="line">        axis.ticks.length=unit(<span class="number">.4</span>, <span class="string">"cm"</span>),</span><br><span class="line">        legend.text = element_text(size = <span class="number">12</span>),     </span><br><span class="line">        legend.position = <span class="string">'right'</span>,</span><br><span class="line">        axis.text.y=element_text(size=<span class="number">18</span>,angle=<span class="number">90</span>,color = <span class="string">'black'</span>),</span><br><span class="line">        axis.text.x=element_text(size=<span class="number">24</span>,face = <span class="string">"italic"</span>)</span><br><span class="line">        <span class="comment">#legend.key.size=unit(0.8,'cm')</span></span><br><span class="line">  )</span><br></pre></td></tr></table></figure>

<p>然后在R中制作 <strong><em>psRNAtarget.pdf</em></strong></p>
<p>使用ppt制作出excel里面的降解组预测结果</p>
<p><strong><em>jiangjiezu.pdf</em></strong></p>
<h4 id="6-cis-Ruglartory-Elements"><a href="#6-cis-Ruglartory-Elements" class="headerlink" title="6. cis-Ruglartory-Elements"></a>6. cis-Ruglartory-Elements</h4><p>首先需要提取各个物种SPL基因的启动子区域</p>
<p>主要用到 TBtools</p>
<p>上传至PlantCare网站预测后 得到压缩包，解压出</p>
<p><strong><em>PlantCARE_9138_105_plantCARE</em></strong> 文件夹</p>
<p>其中 <strong><em>plantCARE_output_PlantCARE_9138.tab</em></strong> 文件为主要的结果文件,复制该文件命名后缀为.2.tab</p>
<p>对于其中的TATA-box和CAAT-box两个原件，由于匹配到多个物种以及CRE的不同异构体，所以区域会有重叠，首先将有重叠的区域进行一个合并</p>
<ol>
<li><p>调整文件中有一些重复的名字，因为有的名字是重复的</p>
 <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">sed -i &apos;&apos; &apos;s/Box 4/Box IV/g&apos; plantCARE_output_PlantCARE_9138.2.tab</span><br><span class="line">sed -i &apos;&apos; &apos;s/AT-rich sequence/AT-rich/g&apos; plantCARE_output_PlantCARE_9138.2.tab</span><br><span class="line">sed -i &apos;&apos; &apos;s/MBSI/MBS/g&apos; plantCARE_output_PlantCARE_9138.2.tab</span><br><span class="line">sed -i &apos;&apos; &apos;s/TGA-box/TGA-element/g&apos; plantCARE_output_PlantCARE_9138.2.tab</span><br></pre></td></tr></table></figure>
</li>
<li><p>去除重叠在一起的TATAbox和CAATbox</p>
</li>
</ol>
<p>例如以下1781和1782的重复，1788和1789的重复</p>
<p>IbSPL1  TATA-box        TATAA   1781    5       -       Arabidopsis thaliana    core promoter element around -30 of transcription start<br>IbSPL1  TATA-box        TATA    1782    4       -       Arabidopsis thaliana    core promoter element around -30 of transcription start<br>IbSPL1  TATA-box        TATAA   1788    5       -       Arabidopsis thaliana    core promoter element around -30 of transcription start<br>IbSPL1  TATA-box        TATA    1789    4       -       Arabidopsis thaliana    core promoter element around -30 of transcription start</p>
<figure class="highlight plain"><figcaption><span>~/Scripts/parse_plantcare.py plantCARE_output_PlantCARE_9138.2.tab > plantCARE_output_PlantCARE_9138.3.tab   ``` </span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">```python</span><br><span class="line"># parse_plantcare.py </span><br><span class="line">#! /usr/local/opt/python/bin/python3.7</span><br><span class="line">import sys</span><br><span class="line"></span><br><span class="line">idlist = []</span><br><span class="line">idlist2 = []</span><br><span class="line">with open(sys.argv[1]) as input_pcr:</span><br><span class="line">    for row in input_pcr:</span><br><span class="line">        lst1 = row.split(&apos;\t&apos;)</span><br><span class="line">        if lst1[1] != &quot;CAAT-box&quot; and lst1[1] != &quot;TATA-box&quot;:</span><br><span class="line">            print(row.strip())</span><br><span class="line">        if lst1[0] not in idlist and lst1[1] == &quot;CAAT-box&quot;:</span><br><span class="line">            min_loc = int(lst1[3])</span><br><span class="line">            idlist.append(lst1[0])</span><br><span class="line">        if lst1[0] not in idlist2 and lst1[1] == &quot;TATA-box&quot;:</span><br><span class="line">            min_loc = int(lst1[3])</span><br><span class="line">            idlist2.append(lst1[0])</span><br><span class="line">        if lst1[0] in idlist2 and lst1[1] == &quot;TATA-box&quot;:</span><br><span class="line">            if(int(lst1[3]) - int(min_loc) &gt;= int(len(lst1[2]))):</span><br><span class="line">                print(row.strip())</span><br><span class="line">                min_loc = int(lst1[3])</span><br><span class="line">        if lst1[0] in idlist and lst1[1] == &quot;CAAT-box&quot;:</span><br><span class="line">            if(int(lst1[3]) - int(min_loc) &gt;= int(len(lst1[2]))):</span><br><span class="line">                print(row.strip())</span><br><span class="line">                min_loc = int(lst1[3])</span><br></pre></td></tr></table></figure>

<p>得到了合并过位置重复词条的<strong><em>plantCARE_output_PlantCARE_9138.3.tab</em></strong></p>
<p>然后再复制为.4.tab，并合并同一个类型的标签</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 将一些同义项合并</span></span><br><span class="line">sed -i '' 's/MYB recognition site/MYB/g' plantCARE_output_PlantCARE_9138.4.tab\nsed -i '' 's/Myb/MYB/g' plantCARE_output_PlantCARE_9138.4.tab\nsed -i '' 's/Myb-binding site/MYB/g' plantCARE_output_PlantCARE_9138.4.tab\nsed -i '' 's/MYB-like sequence/MYB/g' plantCARE_output_PlantCARE_9138.4.tab\nsed -i '' 's/DRE core/DRE/g' plantCARE_output_PlantCARE_9138.4.tab\nsed -i '' 's/AT-rich element/AT-rich/g' plantCARE_output_PlantCARE_9138.4.tab\nsed -i '' 's/AT-rich sequence/AT-rich sequence/g' plantCARE_output_PlantCARE_9138.4.tab\nsed -i '' 's/MYB recognition site/MYB/g' plantCARE_output_PlantCARE_9138.4.tab\nsed -i '' 's/G-Box/G-box/g' plantCARE_output_PlantCARE_9138.4.tab\nsed -i '' 's/TCA/TCA-element/g' plantCARE_output_PlantCARE_9138.4.tab\nsed -i '' 's/TATA/TATA-box/g' plantCARE_output_PlantCARE_9138.4.tab\n\n</span><br></pre></td></tr></table></figure>

<p>生成一个统计数量的矩阵</p>
<figure class="highlight python"><figcaption><span>count_cre.py plantCARE_output_PlantCARE_9138.4.tab > melt.matrix  ```</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">```python</span><br><span class="line"><span class="comment"># count_cre.py</span></span><br><span class="line"><span class="keyword">import</span> sys </span><br><span class="line">ab = &#123;&#125;</span><br><span class="line">p_list = []</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> input_pcr:</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> input_pcr:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        p = lst1[<span class="number">0</span>]+<span class="string">'\t'</span>+lst1[<span class="number">1</span>]</span><br><span class="line">        <span class="keyword">if</span> p <span class="keyword">not</span> <span class="keyword">in</span> p_list:</span><br><span class="line">            ab[p] = <span class="number">0</span></span><br><span class="line">            p_list.append(p)</span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            ab[p] += <span class="number">1</span></span><br><span class="line"><span class="keyword">for</span> k <span class="keyword">in</span> ab.keys():</span><br><span class="line">    print(k+<span class="string">'\t'</span>+str(ab[k]))</span><br></pre></td></tr></table></figure>

<p>生成分解过后的矩阵 <strong><em>melt.matrix</em></strong></p>
<p>然后把矩阵融合</p>
<figure class="highlight plain"><figcaption><span>martix.R```</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">```R</span><br><span class="line">library(reshape2)</span><br><span class="line">setwd(&quot;~/data/gene_family/plot/Figure5/PlantCARE_9138_105_plantCARE/&quot;)</span><br><span class="line">data1 &lt;- read.csv(&quot;melt.matrix&quot;,sep = &apos;\t&apos;,header = T)</span><br><span class="line">data2 &lt;- dcast(data1,id~CRE)</span><br><span class="line">write.table(data2,file=&quot;dcast.matrix.txt&quot;,sep = &apos;\t&apos;,quote = F)</span><br></pre></td></tr></table></figure>

<p>得到的 <strong><em>dcast.matrix.txt</em></strong> 转存为excel文件</p>
<p>最终整理为<strong><em>Supplementary Table S6</em></strong></p>
<p>使用 <strong><em>all_cre_for_tbtools.txt</em></strong> 和 <strong><em>promoter.length</em></strong> 输入TBtools得到 <strong><em>Figure6a.pdf</em></strong></p>
<p>参考<a href="https://github.com/gahoo/SearchCARE/blob/master/CARE.txt" target="_blank" rel="noopener">https://github.com/gahoo/SearchCARE/blob/master/CARE.txt</a></p>
<p>Information on 88 different cis-regulatory elements identified in 11 CaCCO genes is listed, including their numbers and the function of the respective elements</p>
<p>等文献对CRE的划分</p>
<p>统计各个原件出现的数量，画donut图</p>
<p>制作各个原件所占百分比的文件 <strong><em>donut</em></strong></p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 甜甜圈图做图</span></span><br><span class="line">df &lt;- read.csv(<span class="string">"donut"</span>,header = F,sep = <span class="string">'\t'</span>)</span><br><span class="line">df$X &lt;- factor(df$V1,levels=df$V1)</span><br><span class="line">ggplot(df, aes(x = V2, y = V3, fill =X)) +</span><br><span class="line">  geom_col() +</span><br><span class="line">  scale_x_discrete(limits = c(<span class="string">" "</span>, <span class="string">"IN"</span>,<span class="string">"OUT"</span>)) +</span><br><span class="line">  scale_fill_viridis_d() +</span><br><span class="line">  coord_polar(<span class="string">"y"</span>) +</span><br><span class="line">  theme_void()</span><br></pre></td></tr></table></figure>

<p>结果保存为<strong><em>Figure6b.pdf</em></strong></p>
<h4 id="7-转录组分析"><a href="#7-转录组分析" class="headerlink" title="7. 转录组分析"></a>7. 转录组分析</h4><h5 id="7-1-xuzi3和yan252的表达模式分析"><a href="#7-1-xuzi3和yan252的表达模式分析" class="headerlink" title="7.1 xuzi3和yan252的表达模式分析"></a>7.1 xuzi3和yan252的表达模式分析</h5><p>通过 STAR -&gt; stringtie -&gt; RSEM 得到泰中6号基因组中所有基因的表达量 <strong><em>all.expression.FPKM.txt</em></strong></p>
<p>提取其中的SPL基因的FPKM表达量文件</p>
<p> <strong><em>16.spl.FPKM.matrix</em></strong></p>
<h5 id="7-2-Beauregard的根组织中的表达模式"><a href="#7-2-Beauregard的根组织中的表达模式" class="headerlink" title="7.2  Beauregard的根组织中的表达模式"></a>7.2  Beauregard的根组织中的表达模式</h5><p>使用同样的方法，保存FPKM表达量文件为，过滤依据trinity中的PCA和cluster</p>
<p><strong><em>30.FPKM.noLG.matrix</em></strong></p>
<p>提取出spl基因的表达量，保存为</p>
<p>并制作热图，其中基因的注释文件为 <strong><em>spl.class</em></strong>，样本的注释文件分别为 <strong><em>16.pheno.info</em></strong> 和<strong><em>30.samples.txt</em></strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(pheatmap)</span><br><span class="line"><span class="keyword">library</span>(ggplot2)</span><br><span class="line">setwd(<span class="string">"~/data/gene_family/downstream/9_RSEM/39/30"</span>)</span><br><span class="line"></span><br><span class="line">data2 &lt;- read.table(<span class="string">"spl.average.FPKM.matrix.new"</span>,</span><br><span class="line">                    header = <span class="literal">TRUE</span>,sep= <span class="string">'\t'</span>,</span><br><span class="line">                    c(<span class="string">"Gene"</span>,<span class="string">"10DAT_TR"</span>,<span class="string">"20DAT_TR"</span>,<span class="string">"30DAT_FR"</span>,</span><br><span class="line">                        <span class="string">"40DAT_FR"</span>,<span class="string">"50DAT_FR"</span>,<span class="string">"30DAT_SR"</span>,<span class="string">"40DAT_SR"</span>,</span><br><span class="line">                        <span class="string">"50DAT_SR"</span>),row.names = <span class="number">1</span>)</span><br><span class="line">data1 &lt;- read.table(<span class="string">"16.spl.FPKM.matrix.new"</span>,row.names = <span class="number">1</span>)</span><br><span class="line">gene &lt;- read.table(<span class="string">"spl.class"</span>,row.names = <span class="number">1</span>, header = <span class="literal">TRUE</span>)</span><br><span class="line">pheno &lt;- read.table(<span class="string">"pheno"</span>,row.names = <span class="number">1</span>, header = <span class="literal">T</span>, sep = <span class="string">'\t'</span>)</span><br><span class="line">pheatmap(data1, scale = <span class="string">"row"</span>,</span><br><span class="line">         cluster_row = <span class="literal">FALSE</span>, cluster_cols = <span class="literal">FALSE</span>,labels_col = <span class="string">""</span>,</span><br><span class="line">         cellwidth = <span class="number">6</span>, cellheight = <span class="number">12</span>, fontsize = <span class="number">8</span>, </span><br><span class="line">         annotation_row = gene,</span><br><span class="line">         annotation_col = pheno)</span><br><span class="line">pheatmap(data2, scale = <span class="string">"row"</span>,</span><br><span class="line">         cluster_row = <span class="literal">FALSE</span>, cluster_cols = <span class="literal">FALSE</span>,labels_col = <span class="string">""</span>,</span><br><span class="line">         cellwidth = <span class="number">6</span>, cellheight = <span class="number">12</span>, fontsize = <span class="number">8</span>, </span><br><span class="line">         annotation_row = gene,</span><br><span class="line">         )</span><br></pre></td></tr></table></figure>

<p>结果调色后合并保存为 <strong><em>Figure7.pdf</em></strong></p>
<h4 id="8-徐渝34组织qPCR验证"><a href="#8-徐渝34组织qPCR验证" class="headerlink" title="8. 徐渝34组织qPCR验证"></a>8. 徐渝34组织qPCR验证</h4><p>徐渝34栽培信息和样品保存信息 <strong><em>根的直径统计.xlsx</em></strong> （详细采样时间，采样的样品直径）<strong><em>根的直径统计2.xlsx</em></strong>（各个时间的代表组织） </p>
<p>徐渝34的样品整理后照片 <strong><em>xuyu34_tissues.pdf</em></strong></p>
<p>qPCR整理的原始CT值位于<strong><em>qpcr_计算表格.xlsx</em></strong>的<strong><em>sheet1</em></strong></p>
<p>对 ‘SPL1’,’SPL4’,’SPL5’,’SPL12’,’SPL16’,’SPL17’,’SPL20’,’SPL21’,’SPL27’,’SPL28’,’SPL29’ 进行定量分析。所用到的引物记录在 <strong><em>实验材料.xlsx</em></strong> 中，运行结束后将每一个基因所对应的 delta delta CT计算出来，分别保存为 <strong><em>SPLXX.tissues.0603</em></strong> 文件。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 生成带bar值的，带LSD的标签的柱状图，</span></span><br><span class="line"><span class="keyword">library</span>(dplyr)</span><br><span class="line"><span class="keyword">library</span>(tidyr)</span><br><span class="line"><span class="keyword">library</span>(viridis)</span><br><span class="line"><span class="keyword">library</span>(agricolae)</span><br><span class="line"><span class="keyword">library</span>(ggplot2)</span><br><span class="line"><span class="keyword">library</span>(ggsci)</span><br><span class="line">setwd(<span class="string">'/Users/sunhaoyun/data/gene_family/downstream/14_qpcr/'</span>)</span><br><span class="line"></span><br><span class="line">g_list &lt;- c(<span class="string">'SPL1'</span>,<span class="string">'SPL4'</span>,<span class="string">'SPL5'</span>,<span class="string">'SPL12'</span>,<span class="string">'SPL16'</span>,<span class="string">'SPL17'</span>,<span class="string">'SPL20'</span>,<span class="string">'SPL21'</span>,<span class="string">'SPL27'</span>,<span class="string">'SPL28'</span>,<span class="string">'SPL29'</span>)</span><br><span class="line"><span class="comment">#g_list &lt;- c('SPL27','SPL28','SPL29')</span></span><br><span class="line"><span class="keyword">for</span> (i <span class="keyword">in</span> g_list) &#123;</span><br><span class="line">  </span><br><span class="line">t = paste(i,<span class="string">'.tissues.0603.2'</span>,sep=<span class="string">""</span>)</span><br><span class="line">id = paste(<span class="string">'Ib'</span>,i,sep=<span class="string">""</span>)</span><br><span class="line">data1 &lt;- read.table(t,sep = <span class="string">'\t'</span>)</span><br><span class="line">result &lt;- aov(V2~V1,data = data1)</span><br><span class="line">result2 &lt;- LSD.test(result,<span class="string">"V1"</span>,p.adj = <span class="string">"bonferroni"</span>)</span><br><span class="line">result3 &lt;- result2$means[,<span class="number">1</span>:<span class="number">2</span>]</span><br><span class="line">result4 &lt;-result2$groups</span><br><span class="line">data2=cbind(sample=row.names(result3), result3)</span><br><span class="line">data3=cbind(sample=row.names(result4), result4)</span><br><span class="line">d &lt;-merge(x=data2,y=data3)</span><br><span class="line">d$X = factor(d$sample,levels=c(<span class="string">'stem'</span>,<span class="string">'10D .3mm'</span>,<span class="string">'20D 2mm'</span>,<span class="string">'30D 4mm'</span>,<span class="string">'40D 7mm'</span>,<span class="string">'50D 15mm'</span>,<span class="string">'60D 32mm'</span>,<span class="string">'80D 37mm'</span>,<span class="string">'100D 60mm'</span>,<span class="string">'young leaf'</span>,<span class="string">'mature leaf'</span>,<span class="string">'flower'</span>))</span><br><span class="line">p &lt;- ggplot(d, mapping=aes(x=X, y=V2,fill=X)) + </span><br><span class="line">  labs(title = id,</span><br><span class="line">       x = <span class="string">" "</span>,</span><br><span class="line">       y = <span class="string">"Relative expression level"</span>)+</span><br><span class="line">  guides(fill=<span class="literal">F</span>)+theme_classic()+</span><br><span class="line">  geom_bar(stat = <span class="string">"identity"</span>, width = <span class="number">0.8</span>) + </span><br><span class="line">  geom_errorbar(aes(ymin=V2-std, ymax=V2+std),width=<span class="number">0.2</span>, position=position_dodge(width=<span class="number">0.75</span>))+ </span><br><span class="line">  geom_text(aes(label=groups,vjust=-<span class="number">1.4</span>),size=<span class="number">8</span>)+</span><br><span class="line">  theme(<span class="comment">#axis.text.x=element_text(size=24,angle = 30,hjust=1, vjust=1),</span></span><br><span class="line">        <span class="comment">#axis.text.x = element_blank(),</span></span><br><span class="line">        axis.ticks.x = element_blank(),</span><br><span class="line">        axis.title.y=element_text(size=<span class="number">24</span>),</span><br><span class="line">        axis.text.y=element_text(size=<span class="number">24</span>),</span><br><span class="line">        axis.ticks.length=unit(<span class="number">.4</span>, <span class="string">"cm"</span>),</span><br><span class="line">        plot.title=element_text(size=<span class="number">30</span>,hjust = <span class="number">0.01</span>,vjust = -<span class="number">4</span>)</span><br><span class="line">        )+</span><br><span class="line">  scale_fill_manual(values = c(<span class="string">"#BC3C29FF"</span>,<span class="comment">#1</span></span><br><span class="line">                                      <span class="string">"#E64B35B2"</span>,<span class="comment">#2</span></span><br><span class="line">                                      <span class="string">'#E18727B2'</span>,<span class="comment">#3</span></span><br><span class="line">                                      <span class="string">'#EEE8AAFF'</span>,<span class="comment">#4</span></span><br><span class="line">                                      <span class="string">'#6B8E23B2'</span>,<span class="comment">#5</span></span><br><span class="line">                                      <span class="string">'#548B54FF'</span>,<span class="comment">#6</span></span><br><span class="line">                                      <span class="string">'#6F99ADFF'</span>,<span class="comment">#7</span></span><br><span class="line">                                      <span class="string">'#4DBBD5B2'</span>,<span class="comment">#8</span></span><br><span class="line">                                      <span class="string">"#0072B5B2"</span>,<span class="comment">#9</span></span><br><span class="line">                                      <span class="string">'#7876B1FF'</span>,<span class="comment">#10</span></span><br><span class="line">                                      <span class="string">'#FFDC91FF'</span>,<span class="string">'#3C5488B2'</span>,<span class="string">'#7E6148FF'</span>,<span class="string">'#91D1C2FF'</span>,</span><br><span class="line">                                      <span class="string">'#3C5488FF'</span>))     </span><br><span class="line">pdf &lt;- paste(i,<span class="string">'.0603.pdf'</span>,sep=<span class="string">""</span>)</span><br><span class="line">ggsave(pdf, p, width = <span class="number">10</span>, height = <span class="number">6</span>) </span><br><span class="line"></span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

<p>然后得到  <strong><em>SPLXX.0603.pdf</em></strong> ，再拼接到一起得到了</p>
<h4 id="9-激素处理"><a href="#9-激素处理" class="headerlink" title="9. 激素处理"></a>9. 激素处理</h4><p>所有激素处理样本的计算方式位于<strong><em>1104_1.xlsx</em></strong>，原始的CT值位于 <strong><em>jisu.all.qpcr</em></strong> </p>
<p>各个画图脚本命名为 <strong><em>plot_XXX_qpcr.R</em></strong> 位于result2文件夹中</p>
<p>得到的各激素处理的相对表达量柱状图命名为 <strong><em>jisuXX.pdf</em></strong></p>
<p>拼接得到图片 <strong><em>phytohormone.pdf</em></strong></p>
<h4 id="10-共表达调控网络"><a href="#10-共表达调控网络" class="headerlink" title="10.  共表达调控网络"></a>10.  共表达调控网络</h4><h5 id="10-1-首先对88份甘薯块根材料使用同样的表达量计算方法"><a href="#10-1-首先对88份甘薯块根材料使用同样的表达量计算方法" class="headerlink" title="10.1 首先对88份甘薯块根材料使用同样的表达量计算方法"></a>10.1 首先对88份甘薯块根材料使用同样的表达量计算方法</h5><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># filter by counts &gt;10 in 90 percent samples</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line">ab = &#123;&#125;</span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">1</span>]) <span class="keyword">as</span> count_matrix:</span><br><span class="line">    head_include = count_matrix.readline().split(<span class="string">'\t'</span>)</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> count_matrix:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        m = <span class="number">0</span></span><br><span class="line">        <span class="keyword">for</span> i <span class="keyword">in</span> range(len(lst1)<span class="number">-1</span>):</span><br><span class="line">            <span class="keyword">if</span> float(lst1[i+<span class="number">1</span>].strip()) &gt; <span class="number">10.0</span>:</span><br><span class="line">                m += <span class="number">1</span></span><br><span class="line">        <span class="keyword">if</span> m &gt; int(<span class="number">0.9</span>*len(head_include)):</span><br><span class="line">            ab[lst1[<span class="number">0</span>]] = <span class="string">"yes"</span></span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            ab[lst1[<span class="number">0</span>]] = <span class="string">"bad"</span></span><br><span class="line"><span class="keyword">with</span> open(sys.argv[<span class="number">2</span>]) <span class="keyword">as</span> fpkm_matrix:</span><br><span class="line">    print(fpkm_matrix.readline().rstrip())</span><br><span class="line">    <span class="keyword">for</span> row <span class="keyword">in</span> fpkm_matrix:</span><br><span class="line">        lst1 = row.split(<span class="string">'\t'</span>)</span><br><span class="line">        <span class="keyword">if</span> ab[lst1[<span class="number">0</span>]] == <span class="string">"yes"</span>:</span><br><span class="line">            print(row.strip())</span><br></pre></td></tr></table></figure>

<p>然后使用如上脚本过滤掉在90%的样本中reads count小于10的基因</p>
<figure class="highlight python"><figcaption><span>parse_counnts.py 88.counts.matrix 88.FPKM.noLG.matrix > 88.FPKM.noLG.filtered.matrix```</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">得到表达量文件 ***<span class="number">88.</span>FPKM.noLG.matrix***</span><br><span class="line"></span><br><span class="line">```R</span><br><span class="line">library(WGCNA)</span><br><span class="line">library(tidyverse)</span><br><span class="line">allowWGCNAThreads(<span class="number">12</span>)</span><br><span class="line">options(stringsAsFactors = FALSE)</span><br><span class="line">setwd(<span class="string">"~/data/gene_family/plot/WGCNA/"</span>)</span><br><span class="line">expro=read.table(<span class="string">'88.passed.FPKM'</span>,row.names = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 筛选方差前25%的基因</span></span><br><span class="line"><span class="comment">#m.vars=apply(expro,1,var)</span></span><br><span class="line"><span class="comment">#expro.upper=expro[which(m.vars&gt;quantile(m.vars, probs = seq(0, 1, 0.25))[4]),]</span></span><br><span class="line"><span class="comment">#dim(expro.upper)</span></span><br><span class="line"><span class="comment">#datExpr=as.data.frame(t(expro.upper));</span></span><br><span class="line"><span class="comment">#nGenes = ncol(datExpr)</span></span><br><span class="line"><span class="comment">#nSamples = nrow(datExpr)</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## 不筛选基因 ##</span></span><br><span class="line">datExpr=<span class="keyword">as</span>.data.frame(t(expro));</span><br><span class="line">nGenes = ncol(datExpr)</span><br><span class="line">nSamples = nrow(datExpr)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## 样本聚类检查离群值 ##</span></span><br><span class="line">gsg = goodSamplesGenes(datExpr, verbose = <span class="number">3</span>);</span><br><span class="line">gsg$allOK</span><br><span class="line">sampleTree = hclust(dist(datExpr), method = <span class="string">"average"</span>)</span><br><span class="line">plot(sampleTree, main = <span class="string">"Sample clustering to detect outliers"</span></span><br><span class="line">     , sub=<span class="string">""</span>, xlab=<span class="string">""</span>)</span><br><span class="line"><span class="comment">#save(datExpr, file = "FPKM-01-dataInput.RData")</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 去除CRR022951,CRR022914,CRR022943 ##</span></span><br><span class="line">data_set &lt;- subset(expro, select = -c(CRR022951,CRR022914,CRR022943))</span><br><span class="line">datExpr=<span class="keyword">as</span>.data.frame(t(data_set))</span><br><span class="line">nGenes = ncol(datExpr)</span><br><span class="line">nSamples = nrow(datExpr)</span><br><span class="line">sampleTree = hclust(dist(datExpr), method = <span class="string">"average"</span>)</span><br><span class="line">plot(sampleTree, main = <span class="string">"Sample clustering to detect outliers"</span></span><br><span class="line">     , sub=<span class="string">""</span>, xlab=<span class="string">""</span>)</span><br><span class="line"><span class="comment">#if (!gsg$allOK)&#123;</span></span><br><span class="line">  <span class="comment"># Optionally, print the gene and sample names that were removed:</span></span><br><span class="line"><span class="comment">#  if (sum(!gsg$goodGenes)&gt;0) </span></span><br><span class="line"><span class="comment">#    printFlush(paste("Removing genes:", </span></span><br><span class="line"><span class="comment">#                     paste(names(datExpr)[!gsg$goodGenes], collapse = ",")));</span></span><br><span class="line"><span class="comment">#  if (sum(!gsg$goodSamples)&gt;0) </span></span><br><span class="line"><span class="comment">#    printFlush(paste("Removing samples:", </span></span><br><span class="line"><span class="comment">#                     paste(rownames(datExpr)[!gsg$goodSamples], collapse = ",")));</span></span><br><span class="line">  <span class="comment"># Remove the offending genes and samples from the data:</span></span><br><span class="line"><span class="comment">#  datExpr = datExpr[gsg$goodSamples, gsg$goodGenes]</span></span><br><span class="line"><span class="comment">#&#125;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># datExpr = log2(datExpr+1)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">##软阈值筛选##</span></span><br><span class="line">powers = c(seq(<span class="number">1</span>,<span class="number">25</span>,by = <span class="number">1</span>))</span><br><span class="line">sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = <span class="number">5</span>)</span><br><span class="line">par(mfrow = c(<span class="number">1</span>,<span class="number">2</span>));</span><br><span class="line">cex1 = <span class="number">0.9</span>;</span><br><span class="line">plot(sft$fitIndices[,<span class="number">1</span>], -sign(sft$fitIndices[,<span class="number">3</span>])*sft$fitIndices[,<span class="number">2</span>],</span><br><span class="line">     xlab=<span class="string">"Soft Threshold (power)"</span>,ylab=<span class="string">"Scale Free Topology Model Fit,signed R^2"</span>,type=<span class="string">"n"</span>,</span><br><span class="line">     main = paste(<span class="string">"Scale independence"</span>));</span><br><span class="line">text(sft$fitIndices[,<span class="number">1</span>], -sign(sft$fitIndices[,<span class="number">3</span>])*sft$fitIndices[,<span class="number">2</span>],</span><br><span class="line">     labels=powers,cex=cex1,col=<span class="string">"red"</span>);</span><br><span class="line">abline(h=<span class="number">0.90</span>,col=<span class="string">"red"</span>)</span><br><span class="line">plot(sft$fitIndices[,<span class="number">1</span>], sft$fitIndices[,<span class="number">5</span>],</span><br><span class="line">     xlab=<span class="string">"Soft Threshold (power)"</span>,ylab=<span class="string">"Mean Connectivity"</span>, type=<span class="string">"n"</span>,</span><br><span class="line">     main = paste(<span class="string">"Mean connectivity"</span>))</span><br><span class="line">text(sft$fitIndices[,<span class="number">1</span>], sft$fitIndices[,<span class="number">5</span>], labels=powers, cex=cex1,col=<span class="string">"red"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">#### 运行下列代码，让程序推荐你一个power, 如果程序给的是NA就挑选离0.9最近的就可以</span></span><br><span class="line">sft$powerEstimate</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">##一步法网络构建：One-step network construction and module detection##</span></span><br><span class="line">net = blockwiseModules(</span><br><span class="line">  <span class="comment"># input data</span></span><br><span class="line">  datExpr, </span><br><span class="line">  <span class="comment"># calculate correlation coefficient</span></span><br><span class="line">  power = sft$powerEstimate, </span><br><span class="line">  corType = <span class="string">"pearson"</span>,</span><br><span class="line">  networkType = <span class="string">"unsigned"</span>,</span><br><span class="line">  maxBlockSize = nGenes,</span><br><span class="line">  TOMType = <span class="string">"unsigned"</span>,</span><br><span class="line">  deepSplit = <span class="number">2</span>,</span><br><span class="line">  saveTOMs=TRUE,</span><br><span class="line">  minModuleSize = <span class="number">30</span>,</span><br><span class="line">  reassignThreshold = <span class="number">0</span>,</span><br><span class="line">  <span class="comment"># distance among modules less than 0.2, otherwise correlation among module more than 0.85 will be merged</span></span><br><span class="line">  mergeCutHeight = <span class="number">.25</span>,</span><br><span class="line">  numericLabels = TRUE, </span><br><span class="line">  pamRespectsDendro = FALSE,)</span><br><span class="line">table(net$colors)</span><br><span class="line"><span class="comment">##绘画结果展示### open a graphics window</span></span><br><span class="line"><span class="comment">#sizeGrWindow(12, 9)</span></span><br><span class="line"><span class="comment"># Convert labels to colors for plotting</span></span><br><span class="line">mergedColors = labels2colors(net$colors)</span><br><span class="line"><span class="comment"># Plot the dendrogram and the module colors underneath</span></span><br><span class="line">plotDendroAndColors(net$dendrograms[[<span class="number">1</span>]], mergedColors[net$blockGenes[[<span class="number">1</span>]]],<span class="string">"Module colors"</span>,</span><br><span class="line">                    dendroLabels = FALSE, hang = <span class="number">0.03</span>,</span><br><span class="line">                    addGuide = TRUE, guideHang = <span class="number">0.05</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">##结果保存###</span></span><br><span class="line">moduleLabels = net$colors</span><br><span class="line">moduleColors = labels2colors(net$colors)</span><br><span class="line">table(moduleColors)</span><br><span class="line"><span class="comment"># module eigengene, 可以绘制线图，作为每个模块的基因表达趋势的展示</span></span><br><span class="line">MEs = net$MEs</span><br><span class="line"></span><br><span class="line"><span class="comment">### 不需要重新计算，改下列名字就好</span></span><br><span class="line"><span class="comment">### 官方教程是重新计算的，起始可以不用这么麻烦</span></span><br><span class="line">MEs_col = MEs</span><br><span class="line">colnames(MEs_col) = paste0(<span class="string">"ME"</span>, labels2colors(</span><br><span class="line">  <span class="keyword">as</span>.numeric(str_replace_all(colnames(MEs),<span class="string">"ME"</span>,<span class="string">""</span>))))</span><br><span class="line">MEs_col = orderMEs(MEs_col)</span><br><span class="line"></span><br><span class="line">plotEigengeneNetworks(MEs_col, <span class="string">"Eigengene adjacency heatmap"</span>, </span><br><span class="line">                      marDendro = c(<span class="number">3</span>,<span class="number">3</span>,<span class="number">2</span>,<span class="number">4</span>),</span><br><span class="line">                      marHeatmap = c(<span class="number">3</span>,<span class="number">4</span>,<span class="number">2</span>,<span class="number">2</span>), plotDendrograms = T, </span><br><span class="line">                      xLabelsAngle = <span class="number">90</span>)</span><br><span class="line">dissTOM = <span class="number">1</span>-TOM</span><br><span class="line">hierTOM = hclust(<span class="keyword">as</span>.dist(dissTOM),method=<span class="string">"average"</span>);</span><br><span class="line"><span class="comment"># 基因多的时候使用下面的代码：</span></span><br><span class="line">k &lt;- softConnectivity(datE=datExpr[[<span class="number">1</span>]]$data,power=softPower) </span><br><span class="line">sizeGrWindow(<span class="number">10</span>, <span class="number">5</span>)</span><br><span class="line">par(mfrow=c(<span class="number">1</span>,<span class="number">2</span>))</span><br><span class="line">hist(k)</span><br><span class="line">scaleFreePlot(k,main=<span class="string">"Check Scale free topology\n"</span>)</span><br><span class="line"><span class="comment">#=====================================================================================</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  8. 画出TOM矩阵</span></span><br><span class="line"><span class="comment">#   </span></span><br><span class="line"><span class="comment">#=====================================================================================</span></span><br><span class="line"></span><br><span class="line">load(net$TOMFiles[<span class="number">1</span>], verbose = T)</span><br><span class="line">TOM &lt;- <span class="keyword">as</span>.matrix(TOM)</span><br><span class="line"></span><br><span class="line">dissTOM = <span class="number">1</span>-TOM</span><br><span class="line"></span><br><span class="line"><span class="comment"># Transform dissTOM with a power to make moderately strong</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># connections more visible in the heatmap</span></span><br><span class="line"></span><br><span class="line">plotTOM = dissTOM^softPower</span><br><span class="line"></span><br><span class="line"><span class="comment"># Set diagonal to NA for a nicer plot</span></span><br><span class="line"></span><br><span class="line">diag(plotTOM) = NA</span><br><span class="line"></span><br><span class="line"><span class="comment">#call the plot function</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 这一部分特别耗时，行列同时做层级聚类</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#TOMplot(plotTOM, geneTree, moduleColors, main = "Network heatmap plot, all genes")</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#all genes 耗时太长了，随便选取1000个基因来可视化</span></span><br><span class="line"></span><br><span class="line">nSelect = <span class="number">1000</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># For reproducibility, we set the random seed</span></span><br><span class="line"></span><br><span class="line">set.seed(<span class="number">10</span>)</span><br><span class="line"></span><br><span class="line">select = sample(nGenes, size = nSelect)</span><br><span class="line"></span><br><span class="line">selectTOM = dissTOM[select, select]</span><br><span class="line"></span><br><span class="line"><span class="comment"># There’s no simple way of restricting a clustering tree to a subset of genes, so we must re-cluster</span></span><br><span class="line"></span><br><span class="line">selectTree = hclust(<span class="keyword">as</span>.dist(selectTOM), method = <span class="string">"average"</span>)</span><br><span class="line"></span><br><span class="line">selectColors = moduleColors[select]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Open a graphical window</span></span><br><span class="line"></span><br><span class="line">sizeGrWindow(<span class="number">16</span>,<span class="number">12</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Taking the dissimilarity to a power, say 10, makes the plot more informative by effectively changing</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># the color palette; setting the diagonal to NA also improves the clarity of the plot</span></span><br><span class="line"></span><br><span class="line">plotDiss = selectTOM^softPower;</span><br><span class="line"></span><br><span class="line">diag(plotDiss) = NA;</span><br><span class="line"></span><br><span class="line">TOMplot(plotDiss, selectTree,</span><br><span class="line">        selectColors, main = <span class="string">"Network heatmap plot, selected genes"</span>,</span><br><span class="line">        col=gplots::colorpanel(<span class="number">250</span>,<span class="string">'red'</span>,<span class="string">"orange"</span>,<span class="string">'lemonchiffon'</span>))</span><br><span class="line"></span><br><span class="line"><span class="comment">#=====================================================================================</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  9. 生成 Cytoscape 的输入文件</span></span><br><span class="line"><span class="comment">#   </span></span><br><span class="line"><span class="comment">#=====================================================================================</span></span><br><span class="line"></span><br><span class="line">modProbes = names(datExpr)</span><br><span class="line">moduleLabels = net$colors</span><br><span class="line">moduleColors = labels2colors(net$colors)</span><br><span class="line"><span class="comment"># 重新计算 TOM</span></span><br><span class="line">modules = <span class="string">"all"</span></span><br><span class="line"></span><br><span class="line">cyt = exportNetworkToCytoscape(TOM,</span><br><span class="line">                               edgeFile = paste(<span class="string">"CytoscapeInput-edges-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               nodeFile = paste(<span class="string">"CytoscapeInput-nodes-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               weighted = TRUE,</span><br><span class="line">                               threshold = <span class="number">0.05</span>,</span><br><span class="line">                               nodeNames = modProbes,</span><br><span class="line">                               nodeAttr = moduleColors)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Recalculate topological overlap if needed</span></span><br><span class="line">TOM = TOMsimilarityFromExpr(datExpr, power = sft$powerEstimate);</span><br><span class="line"><span class="comment"># Read in the annotation</span></span><br><span class="line"><span class="comment">#fileannot = read.csv(file = "GeneAnnotation.csv");</span></span><br><span class="line"><span class="comment"># Select modules</span></span><br><span class="line">modules = c(<span class="string">"yellow"</span>);</span><br><span class="line"><span class="comment"># Select module probes</span></span><br><span class="line">probes = names(datExpr)</span><br><span class="line">inModule = <span class="keyword">is</span>.finite(match(moduleColors, modules));</span><br><span class="line">modProbes = probes[inModule];</span><br><span class="line"><span class="comment">#modGenes = annot$gene_symbol[match(modProbes, annot$substanceBXH)];</span></span><br><span class="line"><span class="comment"># Select the corresponding Topological Overlap</span></span><br><span class="line">modTOM = TOM[inModule, inModule];</span><br><span class="line">dimnames(modTOM) = list(modProbes, modProbes)</span><br><span class="line"><span class="comment"># Export the network into edge and node list files Cytoscape can read</span></span><br><span class="line">cyt = exportNetworkToCytoscape(modTOM,</span><br><span class="line">                               edgeFile = paste(<span class="string">"CytoscapeInput-edges-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               nodeFile = paste(<span class="string">"CytoscapeInput-nodes-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                               weighted = TRUE,</span><br><span class="line">                               threshold = <span class="number">0.02</span>,</span><br><span class="line">                               nodeNames = modProbes,</span><br><span class="line">                               <span class="comment">#altNodeNames = modGenes,</span></span><br><span class="line">                               nodeAttr = moduleColors[inModule])</span><br></pre></td></tr></table></figure>

<p>保存在WGCNA运行过程中的 <strong><em>Figure10a-Figure10e.pdf</em></strong></p>
<p>得到的结果文件主要有：</p>
<ol>
<li><p>所有剩余基因的link文件，和分类文件</p>
<p><strong><em>CytoscapeInput-nodes-all.txt</em></strong></p>
<p><strong><em>CytoscapeInput-edges-all.txt</em></strong></p>
<figure class="highlight less"><figcaption><span>spl.id2| while read line\ndo\ngrep $&#123;line&#125; 0.05.CytoscapeInput-edges-all.txt > spl.$&#123;line&#125;.edges\ndone```</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"><span class="selector-tag">2</span>. 提取<span class="selector-tag">SPL</span>基因所在的模块</span><br><span class="line"></span><br><span class="line">   ***<span class="selector-tag">SPL</span><span class="selector-class">.module</span><span class="selector-class">.threshold</span><span class="selector-class">.edges</span>***</span><br><span class="line"></span><br><span class="line"><span class="selector-tag">3</span>. 最终提取<span class="selector-tag">id</span>，保存格式为</span><br><span class="line"></span><br><span class="line">   ***<span class="selector-tag">SPL</span> <span class="selector-tag">ge</span> <span class="selector-tag">ne</span><span class="selector-class">.0</span><span class="selector-class">.05</span><span class="selector-class">.edges</span><span class="selector-class">.id</span>***</span><br><span class="line"></span><br><span class="line">得到了各个模块中的<span class="selector-tag">SPL</span>基因</span><br><span class="line"></span><br><span class="line">随后合并为***<span class="selector-tag">0</span><span class="selector-class">.05</span><span class="selector-class">.cytoscape</span><span class="selector-class">.input</span>***输入至<span class="selector-tag">CytoScape</span>做图得到，做图模版保存为  ***<span class="selector-tag">0</span><span class="selector-class">.05</span><span class="selector-class">.cys</span>***</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">##### <span class="selector-tag">10</span><span class="selector-class">.2</span> 天青色模块的<span class="selector-tag">SPL</span>基因韦恩图</span><br><span class="line"></span><br><span class="line">天青色模块输入以下网站进行韦恩图的绘制，结果保存为 ***<span class="selector-tag">Figure10b</span><span class="selector-class">.csv</span>***</span><br><span class="line"></span><br><span class="line"><span class="selector-tag">http</span>:<span class="comment">//bioinformatics.psb.ugent.be/cgi-bin/liste/Venn/calculate_venn.htpl</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">##### <span class="selector-tag">10</span><span class="selector-class">.3</span> <span class="selector-tag">GO</span>注释</span><br><span class="line"></span><br><span class="line">```<span class="selector-tag">R</span></span><br><span class="line"><span class="selector-tag">library</span>(readr)</span><br><span class="line"><span class="selector-tag">library</span>(clusterProfiler)</span><br><span class="line"><span class="selector-tag">library</span>(DOSE)</span><br><span class="line"><span class="selector-tag">library</span>(purrr)</span><br><span class="line"><span class="selector-tag">library</span>(tidyverse)</span><br><span class="line"><span class="selector-tag">library</span>(org.Ibatatas.eg.db)</span><br><span class="line"><span class="selector-tag">library</span>(openxlsx)</span><br><span class="line"></span><br><span class="line"># 安装构建的<span class="selector-tag">R</span>包</span><br><span class="line"><span class="selector-tag">setwd</span>(<span class="string">"~/data/gene_family/plot/WGCNA/ind/"</span>)</span><br><span class="line"><span class="selector-id">#load</span>(<span class="string">"kegg_info.RData"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">clist &lt;- c("g13567","g13581","g31999","g34536","g4106","g44906","g46868","g54228")</span><br><span class="line">#clist &lt;- c("g13567","g13581")</span><br><span class="line">#clist &lt;- c("g31999","g34536","g46868","g54228")</span><br><span class="line"><span class="selector-tag">for</span> (i in clist) &#123;</span><br><span class="line"></span><br><span class="line">  clist2 &lt;- c("0.05")</span><br><span class="line">  <span class="selector-tag">for</span> (n in clist2) &#123;</span><br><span class="line">    e &lt;- paste(i,".",n,".edges.id",sep="")</span><br><span class="line">    gene_list &lt;- read.table(e, header = FALSE)</span><br><span class="line">    gene_list &lt;- as.character(gene_list$V1)</span><br><span class="line">    </span><br><span class="line">    t &lt;- paste(n,i,sep="_") </span><br><span class="line">    <span class="selector-tag">try</span>(ego &lt;- enrichGO(gene          = gene_list, #差异基因 vector</span><br><span class="line">                keyType       = <span class="string">"GID"</span>,  #差异基因的 ID 类型，需要是 OrgDb 支持的</span><br><span class="line">                OrgDb         = org.Ibatatas.eg.db, #对应的OrgDb</span><br><span class="line">                ont           = <span class="string">"BP"</span>, #GO 分类名称，CC BP MF '</span><br><span class="line">                pvalueCutoff  = <span class="number">0.05</span>, #Pvalue 阈值</span><br><span class="line">                qvalueCutoff  = <span class="number">0.05</span>, #qvalue 阈值</span><br><span class="line">                minGSSize     = <span class="number">10</span>,</span><br><span class="line">                maxGSSize     = <span class="number">500</span>,</span><br><span class="line">                pAdjustMethod = <span class="string">"BH"</span>, #Pvalue 矫正方法</span><br><span class="line">                readable      = FALSE),silent=TRUE) <span class="selector-id">#TRUE</span> 则展示<span class="selector-tag">SYMBOL</span>，<span class="selector-tag">FALSE</span> 则展示原来的<span class="selector-tag">ID</span></span><br><span class="line">#将 <span class="selector-tag">ego</span> 对象转换为<span class="selector-tag">dataframe</span>，新版本可以用<span class="selector-tag">as</span><span class="selector-class">.data</span><span class="selector-class">.frame</span>(ego)</span><br><span class="line">    <span class="selector-tag">try</span>(ego_results&lt;-as.data.frame(ego))</span><br><span class="line">    <span class="selector-tag">try</span>(p &lt;- dotplot(ego, showCategory=<span class="number">20</span>, x = <span class="string">"GeneRatio"</span>)+ggtitle(t))</span><br><span class="line">    o &lt;- paste(i,".",n,".pdf",sep="")</span><br><span class="line">    <span class="selector-id">#pdf</span>(file=o,width = <span class="number">15</span>,height = <span class="number">40</span>) </span><br><span class="line">    <span class="selector-id">#plotGOgraph</span>(ego,firstSigNodes = <span class="number">500</span>)</span><br><span class="line">    <span class="selector-id">#dev</span><span class="selector-class">.off</span>()</span><br><span class="line">    <span class="selector-tag">try</span>(ggsave(o, p, width = <span class="number">8</span>, height = <span class="number">6</span>))</span><br><span class="line"><span class="selector-tag">try</span>(ego &lt;- enrichGO(gene          = gene_list, #差异基因 vector</span><br><span class="line">                keyType       = <span class="string">"GID"</span>,  #差异基因的 ID 类型，需要是 OrgDb 支持的</span><br><span class="line">                OrgDb         = org.Ibatatas.eg.db, #对应的OrgDb</span><br><span class="line">                ont           = <span class="string">"all"</span>, #GO 分类名称，CC BP MF '</span><br><span class="line">                pvalueCutoff  = <span class="number">0.05</span>, #Pvalue 阈值</span><br><span class="line">                qvalueCutoff  = <span class="number">0.05</span>, #qvalue 阈值</span><br><span class="line">                minGSSize     = <span class="number">10</span>,</span><br><span class="line">                maxGSSize     = <span class="number">500</span>,</span><br><span class="line">                pAdjustMethod = <span class="string">"BH"</span>, #Pvalue 矫正方法</span><br><span class="line">                readable      = FALSE),silent=TRUE) <span class="selector-id">#TRUE</span> 则展示<span class="selector-tag">SYMBOL</span>，<span class="selector-tag">FALSE</span> 则展示原来的<span class="selector-tag">ID</span></span><br><span class="line">#将 <span class="selector-tag">ego</span> 对象转换为<span class="selector-tag">dataframe</span>，新版本可以用<span class="selector-tag">as</span><span class="selector-class">.data</span><span class="selector-class">.frame</span>(ego)</span><br><span class="line">    <span class="selector-tag">try</span>(ego_results&lt;-as.data.frame(ego))</span><br><span class="line">  	x &lt;- paste(i,".",n,".xlsx",sep="")</span><br><span class="line">    <span class="selector-tag">write</span><span class="selector-class">.xlsx</span>(ego_results,quote=F,file=x)</span><br><span class="line">&#125;</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

</li>
</ol>
<p>结果会保存每个SPL基因的GOBP主是结果和全部CC MF BP注释的xlsx文件</p>
<h4 id="11-过表达IbSPL17和IbSPL28和MYB以及空载的转录组"><a href="#11-过表达IbSPL17和IbSPL28和MYB以及空载的转录组" class="headerlink" title="11. 过表达IbSPL17和IbSPL28和MYB以及空载的转录组"></a>11. 过表达IbSPL17和IbSPL28和MYB以及空载的转录组</h4><p>文件夹位置 dell服务器</p>
<p>/home/sunhaoyun/data/zlz</p>
<p>第一批样本在part1文件夹中</p>
<p>第二批样本在part2文件夹中</p>
<p>/home/sunhaoyun/data/zlz/part2/clean_fq 存放了所有的clean reads 包括MYB</p>
<p>跑出来的表达量位于</p>
<p>/home/sunhaoyun/data/zlz/part2/04_Reads_count</p>
	  
	</div>

	<div>
  	<center>
	<div class="pagination">

    
    
    <a href="/2021/07/12/Braker/" type="button" class="btn btn-default"><i
                class="fa fa-arrow-circle-o-left"></i> Prev</a>
    

    <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
    
    <a href="/2021/04/26/GATK_WDL/" type="button" class="btn btn-default ">Next<i
                class="fa fa-arrow-circle-o-right"></i></a>
    

    
</div>

    </center>
	</div>
	
	<!-- comment -->
	
<section id="comment">
    <h2 class="title">Comments</h2>

    
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2021-07-12 
	</div>
	

	<!-- categories -->
    

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/Graduate/">Graduate<span>1</span></a></li>
    </ul>
	</div>
		

	<!-- toc -->
	<div class="meta-widget">
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2021 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
