<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <meta http-equiv="pragma" content="no-cache">
  <meta http-equiv="cache-control" content="no-cache">
  <meta http-equiv="expires" content="0">
  
  <title>二代重测序流程 | Shy 的博客</title>
  <meta name="author" content="Shy">
  
  <meta name="description" content="Shy 的博客">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="二代重测序流程">
  <meta property="og:site_name" content="Shy 的博客">

  
    <meta property="og:image" content>
  

  
  
    <link href="/favicon.png" rel="icon">
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>

  <!-- analytics -->
  
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-70812759-1', 'auto');
  ga('send', 'pageview');
</script>



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?cb5448498d7169c668b07c2b255d62c1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
		<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-brand" href="/">Shy 的博客</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class=""></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class=""></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class=""></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class=""></i>About
			</a>
		  </li>
		  
		  <li>
			<a href="/atom.xml" title="Subscribe me.">
			  <i class=""></i>RSS
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header">
			<h1> 二代重测序流程</h1>
		</div>
	



<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <p>首先使用迅雷下载sra文件 速度谁用谁知道</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"># 将ncbi 的 sra格式转化为fastq格式</span><br><span class="line">nohup fastq-dump --split-3 --gzip -O /home/sunhaoyun/data/ERR1347700.1 &amp;</span><br></pre></td></tr></table></figure>

<p>然后call就vans了</p>
<p>还有一点就是之前动物所的老师想要把60X变成20X的深度的reads，那边师兄给的方法是直接把Fq扔掉1/3，</p>
<p>现在学了个聪明点的方法，用rarefraction去除比较有可信度</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#################################</span></span><br><span class="line"><span class="comment">#bwa picard GATK pipeline</span></span><br><span class="line"><span class="comment">#@author: sunhaoyun </span></span><br><span class="line"><span class="comment">#@contact qq: 172293639</span></span><br><span class="line"><span class="comment"># STEP follow the 1000genome project high coverage sequencing wheel</span></span><br><span class="line"><span class="comment"># the specific prameter could check NYGC websites :)</span></span><br><span class="line"><span class="comment"># commond reference webites:</span></span><br><span class="line"><span class="comment"># https://best-practices-for-processing-hts-data.readthedocs.io/en/latest/index.html</span></span><br><span class="line"><span class="comment"># https://www.internationalgenome.org/data-portal/data-collection/30x-grch38</span></span><br><span class="line"><span class="comment"># reference genome could found in this google cloud sites:</span></span><br><span class="line"><span class="comment"># There will be about 3X Volume storage space bigger than Fq files</span></span><br><span class="line"><span class="comment">#1: Trimmomatic</span></span><br><span class="line"><span class="comment">#2: bwa mem</span></span><br><span class="line"><span class="comment">#3. picard fixmate</span></span><br><span class="line"><span class="comment">#4: picard sort sam</span></span><br><span class="line"><span class="comment">#5: picard mark duplicate</span></span><br><span class="line"><span class="comment">#6: gatk realign</span></span><br><span class="line"><span class="comment">#7: gatk BQSR</span></span><br><span class="line"><span class="comment">#8: gatk haltopviewer call gvcf</span></span><br><span class="line"><span class="comment">################################</span></span><br><span class="line"><span class="keyword">import</span> subprocess</span><br><span class="line"><span class="keyword">import</span> argparse</span><br><span class="line"><span class="keyword">import</span> time</span><br><span class="line"><span class="keyword">import</span> os.path</span><br><span class="line"><span class="keyword">from</span> pathlib <span class="keyword">import</span> Path</span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line"><span class="comment">################### SET SOFTWARE PATH HERE ################### </span></span><br><span class="line"></span><br><span class="line">adaptor_path = <span class="string">"/home/gujiaqi/software/adapters/adaptor.fa"</span></span><br><span class="line">trimmomatic_path = <span class="string">"/home/gujiaqi/software/trimmomatic-0.39.jar"</span></span><br><span class="line">bwa_path = <span class="string">"/usr/bin/bwa"</span> </span><br><span class="line">picard_path = <span class="string">"/home/gujiaqi/software/picard.jar"</span></span><br><span class="line">gatk_path = <span class="string">"/home/gujiaqi/software/GenomeAnalysisTK.jar"</span></span><br><span class="line">temp_path = <span class="string">"/data/gujiaqi/temp"</span></span><br><span class="line">db_snp = <span class="string">"/data/gujiaqi/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.dbsnp138.vcf.gz"</span></span><br><span class="line">known_indel = <span class="string">"/data/gujiaqi/reference_genome/resources_broad_hg38_v0_Homo_sapiens_assembly38.known_indels.vcf.gz"</span></span><br><span class="line">mills = <span class="string">"/data/gujiaqi/reference_genome/resources_broad_hg38_v0_Mills_and_1000G_gold_standard.indels.hg38.vcf.gz"</span></span><br><span class="line"><span class="comment">###################    SOFTWARE CHECK   ######################</span></span><br><span class="line"><span class="keyword">if</span> <span class="keyword">not</span> Path(adaptor_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(temp_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(trimmomatic_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(bwa_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(picard_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(gatk_path).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(db_snp).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(known_indel).exists() \</span><br><span class="line"><span class="keyword">or</span> <span class="keyword">not</span> Path(mills).exists():</span><br><span class="line">    print(<span class="string">"SOFTWARE SET FAILED, PLEASE CHECK AGAIN"</span>)</span><br><span class="line">    sys.exit()</span><br><span class="line">    </span><br><span class="line"></span><br><span class="line"></span><br><span class="line">ap = argparse.ArgumentParser()</span><br><span class="line">ap.add_argument(<span class="string">"-p"</span>, <span class="string">"--prefix"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"the  prefix of sample"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-f"</span>, <span class="string">"--refer"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"reference genome path"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-o"</span>, <span class="string">"--outpath"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"Out put file path"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-t"</span>, <span class="string">"--threads"</span>, required=<span class="literal">False</span>,</span><br><span class="line">    help=<span class="string">"process will be used"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-1"</span>, <span class="string">"--reads1"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"Input followed reads"</span>)</span><br><span class="line">ap.add_argument(<span class="string">"-2"</span>, <span class="string">"--reads2"</span>, required=<span class="literal">True</span>,</span><br><span class="line">    help=<span class="string">"Input reverse reads"</span>)</span><br><span class="line"></span><br><span class="line">args = vars(ap.parse_args())</span><br><span class="line"><span class="keyword">print</span> (<span class="string">"The Task id is &#123;&#125;"</span>.format(args[<span class="string">"prefix"</span>]))</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> args[<span class="string">"threads"</span>]:</span><br><span class="line">    cpu = args[<span class="string">"threads"</span>]</span><br><span class="line"><span class="keyword">else</span>:</span><br><span class="line">    cpu = <span class="number">8</span></span><br><span class="line"><span class="keyword">print</span> (<span class="string">"DATE STATUS: The Task Start at: &#123;&#125;"</span>.format(time.ctime())) </span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> <span class="keyword">not</span> Path(args[<span class="string">"prefix"</span>]).exists():</span><br><span class="line">    mk_work_dir = <span class="string">'mkdir &#123;&#125;'</span>.format(args[<span class="string">"prefix"</span>])</span><br><span class="line">    prefix = <span class="string">"&#123;&#125;"</span>.format(args[<span class="string">"prefix"</span>])</span><br><span class="line">    subprocess.call(mk_work_dir, shell=<span class="literal">True</span>)</span><br><span class="line">    print(<span class="string">"Work dictionary &#123;&#125; is created"</span>.format(args[<span class="string">"prefix"</span>]))</span><br><span class="line"><span class="keyword">else</span>:</span><br><span class="line">    prefix = <span class="string">"&#123;&#125;"</span>.format(args[<span class="string">"prefix"</span>])</span><br><span class="line">    print(<span class="string">"Work dictionary  &#123;&#125; is created"</span>.format(args[<span class="string">"prefix"</span>]))</span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 1 ####### </span></span><br><span class="line"><span class="comment">### Trimmomatic software should be downloaded ###</span></span><br><span class="line"><span class="comment">### You could modify the adaptor file and trimmomatic path bellow</span></span><br><span class="line"></span><br><span class="line"><span class="comment">### prepare well ###</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 1 ... filtering the fastq files \n\n\n"</span>)</span><br><span class="line">trim1_pa = <span class="string">"&#123;&#125;/&#123;&#125;_f1.trim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line">trim1_up = <span class="string">"&#123;&#125;/&#123;&#125;_f1.untrim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line">trim2_pa = <span class="string">"&#123;&#125;/&#123;&#125;_r2.trim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line">trim2_up = <span class="string">"&#123;&#125;/&#123;&#125;_r2.untrim.fq.gz"</span>.format(prefix,prefix)</span><br><span class="line"></span><br><span class="line">Trim_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -jar &#123;&#125; \</span></span><br><span class="line"><span class="string">PE -phred33 -threads &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; &#123;&#125; \</span></span><br><span class="line"><span class="string">ILLUMINACLIP:&#123;&#125;:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50"</span>\</span><br><span class="line">.format(cpu,trimmomatic_path,cpu,args[<span class="string">"reads1"</span>],args[<span class="string">"reads2"</span>]\</span><br><span class="line">,trim1_pa,trim1_up,trim2_pa,trim2_up,adaptor_path)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(Trim_cmd))</span><br><span class="line">subprocess.call(Trim_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: Trim Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 2 ####### </span></span><br><span class="line"><span class="comment">### bwa software should be downloaded ###</span></span><br><span class="line"><span class="comment">### You could modify the bwa path bellow</span></span><br><span class="line"><span class="comment">### Check for index file completely ###</span></span><br><span class="line">genome = <span class="string">"&#123;&#125;"</span>.format(args[<span class="string">"refer"</span>])</span><br><span class="line">bwa_index = <span class="string">"&#123;&#125;.amb"</span>.format(genome)</span><br><span class="line">genome_dict = <span class="string">'&#123;&#125;.dict'</span>.format(genome)</span><br><span class="line">fasta_index = <span class="string">'&#123;&#125;.fai'</span>.format(genome)</span><br><span class="line"><span class="keyword">if</span> <span class="keyword">not</span> Path(bwa_index).exists() <span class="keyword">or</span> <span class="keyword">not</span> Path(bwa_index).exists() <span class="keyword">or</span> <span class="keyword">not</span> Path(bwa_index).exists():</span><br><span class="line">    print(<span class="string">"some index file are missing, please offer bwa/samtools/picard index~"</span>)</span><br><span class="line">    print(<span class="string">"DATE STATUS: Failed at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line">    sys.exit()</span><br><span class="line"><span class="keyword">else</span>:</span><br><span class="line">    print(<span class="string">"index file are repared well"</span>)</span><br><span class="line">    RG = <span class="string">r"@RG\tID:&#123;&#125;\tPL:ILLUMINA\tLB:&#123;&#125;\tSM:&#123;&#125;\tPU:1"</span>.format(prefix,prefix,prefix)</span><br><span class="line">    print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 2 ... align the fastq files to &#123;&#125; \n\n\n"</span>.format(genome))</span><br><span class="line">    bwa_cmd = <span class="string">"&#123;&#125; mem -Y -t &#123;&#125; -R '&#123;&#125;' -M  &#123;&#125; &#123;&#125; &#123;&#125; | \</span></span><br><span class="line"><span class="string">    samtools view -Shb -o &#123;&#125;/&#123;&#125;.bam -"</span>\</span><br><span class="line">    .format(bwa_path,cpu,RG,genome,trim1_pa,trim2_pa,prefix,prefix)</span><br><span class="line">    print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(bwa_cmd))</span><br><span class="line">    subprocess.call(bwa_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">    print(<span class="string">"DATE STATUS: BWA MEM Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 3 ####### </span></span><br><span class="line"><span class="comment">#### picard fix mate information</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 3 ... Fix mate information in the BAM \n\n\n"</span>)</span><br><span class="line">fix_mate_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; FixMateInformation MAX_RECORDS_IN_RAM=2000000 \</span></span><br><span class="line"><span class="string">VALIDATION_STRINGENCY=SILENT ADD_MATE_CIGAR=True \</span></span><br><span class="line"><span class="string">ASSUME_SORTED=true I=&#123;&#125;/&#123;&#125;.bam O=&#123;&#125;/&#123;&#125;.fixmate.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,picard_path,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(fix_mate_cmd))</span><br><span class="line">subprocess.call(fix_mate_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: PICARD FIX MATE Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 4 ####### </span></span><br><span class="line"><span class="comment">#### picard sort bam</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 4 ... coordinate sort BAM \n\n\n"</span>)</span><br><span class="line">sort_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; SortSam MAX_RECORDS_IN_RAM=2000000 \</span></span><br><span class="line"><span class="string">VALIDATION_STRINGENCY=SILENT SORT_ORDER=coordinate \</span></span><br><span class="line"><span class="string">CREATE_INDEX=true I=&#123;&#125;/&#123;&#125;.fixmate.bam O=&#123;&#125;/&#123;&#125;.sorted.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,picard_path,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(sort_cmd))</span><br><span class="line">subprocess.call(sort_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: PICARD SORT SAM Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 5 ####### </span></span><br><span class="line"><span class="comment">#### picard mark duplication </span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 5 ... Mark PCR duplicates \n\n\n"</span>)</span><br><span class="line">mark_dump_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; MarkDuplicates MAX_RECORDS_IN_RAM=2000000 \</span></span><br><span class="line"><span class="string">VALIDATION_STRINGENCY=SILENT M=&#123;&#125;/&#123;&#125;.matrix \</span></span><br><span class="line"><span class="string">CREATE_INDEX=true I=&#123;&#125;/&#123;&#125;.sorted.bam O=&#123;&#125;/&#123;&#125;.dedup.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,picard_path,prefix,prefix,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(mark_dump_cmd))</span><br><span class="line">subprocess.call(mark_dump_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: PICARD MARK DUPLICATE Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 6 ####### </span></span><br><span class="line"><span class="comment">#### Realign indel intervels</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 6 ... Realign the indel intervels \n\n\n"</span>)</span><br><span class="line">realign_prepare = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T RealignerTargetCreator \</span></span><br><span class="line"><span class="string">-R &#123;&#125; -I &#123;&#125;/&#123;&#125;.dedup.bam -known &#123;&#125; \</span></span><br><span class="line"><span class="string">-o &#123;&#125;/realignment_targets.list"</span>.format(cpu,temp_path,gatk_path,genome,prefix,prefix,known_indel,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(realign_prepare))</span><br><span class="line">subprocess.call(realign_prepare, shell=<span class="literal">True</span>)</span><br><span class="line">realign_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T IndelRealigner -R &#123;&#125; \</span></span><br><span class="line"><span class="string">-I &#123;&#125;/&#123;&#125;.dedup.bam -targetIntervals &#123;&#125;/realignment_targets.list \</span></span><br><span class="line"><span class="string">-known &#123;&#125; -o &#123;&#125;/&#123;&#125;.realign.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,genome,prefix,prefix,prefix,known_indel,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(realign_cmd))</span><br><span class="line">subprocess.call(realign_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">index_cmd = <span class="string">"samtools index &#123;&#125;/&#123;&#125;.realign.bam"</span>.format(prefix,prefix)</span><br><span class="line">subprocess.call(index_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: GATK REALIGN  Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"><span class="comment">####### STAGE 7 ####### </span></span><br><span class="line"><span class="comment">#### Recalibrate base quality scores</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 7 ... Recalibrate base quality scores using known SNPs \n\n\n"</span>)</span><br><span class="line">recalibrate_prepare = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T BaseRecalibrator \</span></span><br><span class="line"><span class="string">--preserve_qscores_less_than 6 \</span></span><br><span class="line"><span class="string">-nct &#123;&#125; -R &#123;&#125; -I &#123;&#125;/&#123;&#125;.realign.bam \</span></span><br><span class="line"><span class="string">-XL chrX -XL chrY -XL chrM \</span></span><br><span class="line"><span class="string">-knownSites &#123;&#125; -knownSites &#123;&#125; -knownSites &#123;&#125; \</span></span><br><span class="line"><span class="string">-o &#123;&#125;/recal_data.table"</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,cpu,genome,prefix,prefix,mills,known_indel,db_snp,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(recalibrate_prepare))</span><br><span class="line">subprocess.call(recalibrate_prepare, shell=<span class="literal">True</span>)</span><br><span class="line">recalibrate_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; -T PrintReads \</span></span><br><span class="line"><span class="string">--disable_indel_quals \</span></span><br><span class="line"><span class="string">--preserve_qscores_less_than 6 \</span></span><br><span class="line"><span class="string">-nct &#123;&#125; -R &#123;&#125; -I &#123;&#125;/&#123;&#125;.realign.bam \</span></span><br><span class="line"><span class="string">-SQQ 10 -SQQ 20 -SQQ 30 -rf BadCigar \</span></span><br><span class="line"><span class="string">-BQSR &#123;&#125;/recal_data.table -o &#123;&#125;/&#123;&#125;.BQSR.bam"</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,cpu,genome,prefix,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(recalibrate_cmd))</span><br><span class="line">subprocess.call(recalibrate_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">index_cmd = <span class="string">"samtools index &#123;&#125;/&#123;&#125;.BQSR.bam"</span>.format(prefix,prefix)</span><br><span class="line">subprocess.call(index_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: GATK BQSR Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line"><span class="comment">####### STAGE 8 ####### </span></span><br><span class="line"><span class="comment">#### Raw variant calls using HaplotypeCaller on single sample</span></span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tStage 8 ... CALL SNP !!! \n\n\n"</span>)</span><br><span class="line">callsnp_cmd = <span class="string">"java -XX:+UseParallelGC -XX:ParallelGCThreads=&#123;&#125; -Djava.io.tmpdir=&#123;&#125;\</span></span><br><span class="line"><span class="string">-jar &#123;&#125; \</span></span><br><span class="line"><span class="string">-T HaplotypeCaller --genotyping_mode DISCOVERY \</span></span><br><span class="line"><span class="string">-A AlleleBalanceBySample -A DepthPerAlleleBySample \</span></span><br><span class="line"><span class="string">-A DepthPerSampleHC -A InbreedingCoeff \</span></span><br><span class="line"><span class="string">-A MappingQualityZeroBySample -A StrandBiasBySample \</span></span><br><span class="line"><span class="string">-A Coverage -A FisherStrand -A HaplotypeScore \</span></span><br><span class="line"><span class="string">-A MappingQualityRankSumTest -A MappingQualityZero \</span></span><br><span class="line"><span class="string">-A QualByDepth -A RMSMappingQuality \</span></span><br><span class="line"><span class="string">-A ReadPosRankSumTest -A VariantType -l INFO \</span></span><br><span class="line"><span class="string">--emitRefConfidence GVCF -rf BadCigar \</span></span><br><span class="line"><span class="string">--variant_index_parameter 128000 --variant_index_type LINEAR \</span></span><br><span class="line"><span class="string">-R &#123;&#125; -nct 4 -I &#123;&#125;/&#123;&#125;.BQSR.bam -o &#123;&#125;/&#123;&#125;.g.vcf "</span>\</span><br><span class="line">.format(cpu,temp_path,gatk_path,genome,prefix,prefix,prefix,prefix)</span><br><span class="line">print(<span class="string">"commond line was used as follow: &#123;&#125;"</span>.format(callsnp_cmd))</span><br><span class="line">subprocess.call(callsnp_cmd, shell=<span class="literal">True</span>)</span><br><span class="line">print(<span class="string">"DATE STATUS: GATK HaplotypeCaller Finished at : &#123;&#125;"</span>.format(time.ctime()))</span><br><span class="line"></span><br><span class="line">print(<span class="string">"\n\n\n\t\tALl process has been finished, if you want check how many time consume \ you could use commond below"</span>)</span><br><span class="line">print(<span class="string">"\n\n\n\t\t\t\t\t\tgrep 'DATE STATUS:' $log_file"</span>)</span><br></pre></td></tr></table></figure>

	  
	</div>

	<div>
  	<center>
	<div class="pagination">

    
    
    <a type="button" class="btn btn-default disabled"><i class="fa fa-arrow-circle-o-left"></i>Prev</a>
    

    <a href="/" type="button" class="btn btn-default"><i class="fa fa-home"></i>Home</a>
    
    <a href="/2020/12/22/zhuanluzu/" type="button" class="btn btn-default ">Next<i
                class="fa fa-arrow-circle-o-right"></i></a>
    

    
</div>

    </center>
	</div>
	
	<!-- comment -->
	
<section id="comment">
    <h2 class="title">Comments</h2>

    
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2020-12-29 
	</div>
	

	<!-- categories -->
    

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/NGS/">NGS<span>1</span></a></li>
    </ul>
	</div>
		

	<!-- toc -->
	<div class="meta-widget">
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2020 Shy
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a>,<a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>,<a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a> and <a href="http://getbootstrap.com/" target="_blank">BOOTSTRA.386</a>. 
     <br> Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind.386</a>.    
</p>
 </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>⬆︎TOP</span>
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>

</body>
   </html>
